--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-11-06 02:17:57.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-11-25 02:23:49.000000000 -0700
@@ -202,7 +202,9 @@
 
   public Image createImage (ImageProducer producer)
   {
-    return new GtkImage (producer, null);
+    GtkImage image = new GtkImage (producer, null);
+    producer.startProduction (image);
+    return image;
   }
 
   public Image createImage (int width, int height)
