--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkImage.java	2004-07-23 02:23:49.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkImage.java	2004-08-06 02:38:39.000000000 -0600
@@ -89,6 +89,18 @@
     this.observer = observer;
   }
 
+  public synchronized int[]
+  getPixelCache ()
+  {
+    return pixelCache;
+  }
+
+  public synchronized ColorModel
+  getColorModel ()
+  {
+    return model;
+  }
+
   public synchronized int 
   getWidth (ImageObserver observer)
   {
