--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFramePeer.java	2004-09-23 02:37:56.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkFramePeer.java	2004-11-11 02:24:32.000000000 -0700
@@ -170,7 +170,7 @@
   native void nativeSetIconImageFromData (int[] pixels, int width, int height);
   public void setIconImage (Image image) 
   {
-      if (image != null)
+      if (image != null && image instanceof GtkImage)
         {
           GtkImage img = (GtkImage) image;
           // FIXME: Image should be loaded, but if not, do image loading here.
