--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkListPeer.java	2004-10-02 02:25:48.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkListPeer.java	2004-09-04 02:24:27.000000000 -0600
@@ -41,8 +41,8 @@
 import java.awt.AWTEvent;
 import java.awt.Dimension;
 import java.awt.List;
-import java.awt.event.KeyEvent;
 import java.awt.event.MouseEvent;
+import java.awt.event.KeyEvent;
 import java.awt.peer.ListPeer;
 
 public class GtkListPeer extends GtkComponentPeer
