--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/editors/FontEditor.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/beans/editors/FontEditor.java	2002-01-22 16:01:30.000000000 -0700
@@ -38,8 +38,8 @@
 
 package gnu.java.beans.editors;
 
+import java.beans.*;
 import java.awt.Font;
-import java.beans.PropertyEditorSupport;
 
 /**
  ** FontEditor is a property editor for java.awt.Font.
