--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/ExplicitBeanInfo.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/beans/ExplicitBeanInfo.java	2002-01-22 16:01:30.000000000 -0700
@@ -38,11 +38,7 @@
 
 package gnu.java.beans;
 
-import java.beans.BeanDescriptor;
-import java.beans.BeanInfo;
-import java.beans.EventSetDescriptor;
-import java.beans.MethodDescriptor;
-import java.beans.PropertyDescriptor;
+import java.beans.*;
 
 /**
  ** ExplicitBeanInfo lets you specify in the constructor
