--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkMenuBarPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkMenuBarPeer.java	2004-09-13 02:34:14.000000000 -0600
@@ -38,8 +38,10 @@
 
 package gnu.java.awt.peer.gtk;
 
+import java.awt.Font;
 import java.awt.Menu;
 import java.awt.MenuBar;
+import java.awt.MenuComponent;
 import java.awt.peer.MenuBarPeer;
 import java.awt.peer.MenuPeer;
 
@@ -53,18 +55,25 @@
   public GtkMenuBarPeer (MenuBar target)
   {
     super (target);
-    create ();
   }
 
-  /* In Gnome, help menus are no longer right flushed. */
-  public void addHelpMenu (Menu menu)
+  void setFont ()
   {
-    addMenu (menu);
+    MenuComponent mc = (MenuComponent) awtWidget;
+    Font f = mc.getFont ();
+
+    if (f == null)
+      mc.setFont (new Font ("Dialog", Font.PLAIN, 12));
   }
 
-  public void addMenu (Menu menu)
+  // FIXME: remove this method or replace it with one that does
+  // something useful.
+  /* In Gnome, help menus are no longer right flushed. */
+  native void nativeSetHelpMenu(MenuPeer menuPeer);
+
+  public void addHelpMenu (Menu menu)
   {
-    addMenu ((MenuPeer) menu.getPeer ());
+    // nativeSetHelpMenu((MenuPeer) menu.getPeer());
   }
 
   native public void delMenu (int index);
