--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkWindowPeer.java	2004-10-02 02:25:48.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkWindowPeer.java	2004-07-31 02:23:13.000000000 -0600
@@ -40,8 +40,8 @@
 
 import java.awt.Component;
 import java.awt.Dimension;
-import java.awt.Frame;
 import java.awt.Window;
+import java.awt.Frame;
 import java.awt.event.WindowEvent;
 import java.awt.peer.WindowPeer;
 
