--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/ComponentDataBlitOp.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/awt/ComponentDataBlitOp.java	2004-07-22 11:06:18.000000000 -0600
@@ -36,14 +36,9 @@
 
 package gnu.java.awt;
 
+import java.awt.geom.*;
+import java.awt.image.*;
 import java.awt.RenderingHints;
-import java.awt.geom.Point2D;
-import java.awt.geom.Rectangle2D;
-import java.awt.image.ComponentSampleModel;
-import java.awt.image.DataBuffer;
-import java.awt.image.Raster;
-import java.awt.image.RasterOp;
-import java.awt.image.WritableRaster;
 
 /**
  * This raster copy operation assumes that both source and destination
