--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkGraphics2D.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/awt/peer/gtk/GdkGraphics2D.java	2004-07-24 02:22:21.000000000 -0600
@@ -1,5 +1,5 @@
 /* GdkGraphics2D.java
-   Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -48,6 +48,7 @@
 import java.util.Map;
 
 import java.text.AttributedCharacterIterator;
+import java.util.Map;
 import java.util.Stack;
 import java.lang.Integer;
 import gnu.java.awt.ClasspathToolkit;
@@ -87,7 +88,7 @@
 
   private Stack stateStack;
   
-  native private int[] initState (GtkComponentPeer component);
+  native private void initState (GtkComponentPeer component);
   native private void initState (int width, int height);
   native private void copyState (GdkGraphics2D g);
   native public void dispose ();
@@ -168,10 +169,10 @@
   GdkGraphics2D (GtkComponentPeer component)
   {
     this.component = component;
-    int rgb[] = initState (component);
+    initState (component);
 
-    setColor (new Color (rgb[0], rgb[1], rgb[2]));
-    setBackground (new Color (rgb[3], rgb[4], rgb[5]));
+    setColor (component.awtComponent.getForeground ());
+    setBackground (component.awtComponent.getBackground ());
     setPaint (getColor());
     setFont (new Font("SansSerif", Font.PLAIN, 12));
     setTransform (new AffineTransform ());
