--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2004-06-27 02:20:29.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2004-09-04 02:24:27.000000000 -0600
@@ -52,7 +52,7 @@
 
   native void create (int width, int height, int scrollbarVisibility);
 
-  native void gtkSetFont (String name, int style, int size);
+  native void gtkWidgetModifyFont (String name, int style, int size);
   native void gtkWidgetRequestFocus ();
 
   void create ()
