--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFileDialogPeer.java	2004-10-02 02:25:48.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkFileDialogPeer.java	2004-09-04 02:24:27.000000000 -0600
@@ -45,8 +45,8 @@
 import java.awt.Window;
 import java.awt.event.WindowEvent;
 import java.awt.peer.FileDialogPeer;
-import java.io.File;
 import java.io.FilenameFilter;
+import java.io.File;
 
 public class GtkFileDialogPeer extends GtkDialogPeer implements FileDialogPeer
 {
