--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFramePeer.java	2004-11-14 02:20:07.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkFramePeer.java	2004-11-25 02:23:50.000000000 -0700
@@ -89,7 +89,8 @@
         setMenuBarPeer (menuBar);
         int menuBarWidth =
           awtComponent.getWidth () - insets.left - insets.right;
-        setMenuBarWidth (menuBar, menuBarWidth);
+        if (menuBarWidth > 0)
+          setMenuBarWidth (menuBar, menuBarWidth);
         menuBarHeight = getMenuBarHeight ();
         insets.top += menuBarHeight;
         awtComponent.validate ();
@@ -105,7 +106,8 @@
           awtComponent.getWidth () - insets.left - insets.right;
         menuBar = (MenuBarPeer) ((MenuBar) bar).getPeer ();
         setMenuBarPeer (menuBar);
-        setMenuBarWidth (menuBar, menuBarWidth);
+        if (menuBarWidth > 0)
+          setMenuBarWidth (menuBar, menuBarWidth);
         menuBarHeight = getMenuBarHeight ();
         if (oldHeight != menuBarHeight)
           {
@@ -118,8 +120,9 @@
 
   public void setBounds (int x, int y, int width, int height)
   {
-    if (menuBar != null)
-      setMenuBarWidth (menuBar, width - insets.left - insets.right);
+    int menuBarWidth = width - insets.left - insets.right;
+    if (menuBar != null && menuBarWidth > 0)
+      setMenuBarWidth (menuBar, menuBarWidth);
 
     nativeSetBounds (x, y,
 		     width - insets.left - insets.right,
@@ -225,7 +228,8 @@
         || frame_width != awtComponent.getWidth()
         || frame_height != awtComponent.getHeight())
       {
-        if (frame_width != awtComponent.getWidth() && menuBar != null)
+        if (frame_width != awtComponent.getWidth() && menuBar != null
+            && width > 0)
           setMenuBarWidth (menuBar, width);
 
         setBoundsCallback ((Window) awtComponent,
