--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkListPeer.java	2004-06-08 02:24:15.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkListPeer.java	2004-07-31 02:23:13.000000000 -0600
@@ -144,7 +144,7 @@
         // multiple click.
 	MouseEvent me = (MouseEvent) e;
 	if (!me.isConsumed ()
-	    && (me.getModifiers () & MouseEvent.BUTTON1_MASK) != 0
+	    && (me.getModifiersEx () & MouseEvent.BUTTON1_DOWN_MASK) != 0
 	    && me.getClickCount() == 2)
 	  {
             String selectedItem = ((List) awtComponent).getSelectedItem ();
@@ -153,7 +153,7 @@
             // something is selected.
             if (selectedItem != null)
 	      postActionEvent (((List) awtComponent).getSelectedItem (), 
-			       me.getModifiers ());
+			       me.getModifiersEx ());
 	  }
       }
 
@@ -167,7 +167,7 @@
             // Enter only generates an Action event if something is
             // selected.
             if (selectedItem != null)
-	      postActionEvent (selectedItem, ke.getModifiers ());
+	      postActionEvent (selectedItem, ke.getModifiersEx ());
 	  }
       }
 
