--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkGraphics.java	2004-09-06 02:28:53.000000000 -0600
+++ gnu/java/awt/peer/gtk/GdkGraphics.java	2004-09-04 02:24:26.000000000 -0600
@@ -116,7 +116,6 @@
 //    public Graphics create (int x, int y, int width, int height)
 //    {
 //      GdkGraphics g = new GdkGraphics (this);
-//      System.out.println ("translating by: " + x +" " + y);
 //      g.translate (x, y);
 //      g.clipRect (0, 0, width, height);
 
@@ -350,7 +349,6 @@
 
   public Rectangle getClipBounds ()
   {
-//      System.out.println ("returning CLIP: " + clip);
     return new Rectangle (clip.x, clip.y, clip.width, clip.height);
   }
 
