--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/net/protocol/file/Handler.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/net/protocol/file/Handler.java	2003-11-27 02:18:11.000000000 -0700
@@ -1,5 +1,5 @@
 /* Handler.java -- "file" protocol handler for java.net
-   Copyright (C) 1998, 1999, 2000, 2002, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -76,15 +76,17 @@
     String host = url.getHost();
     if ((host != null) && (! host.equals("")))
       {
+        throw new IOException("ftp protocol handler not yet implemented.");
+        /*
 	// Reset the protocol (and implicitly the handler) for this URL.
 	// Then have the URL attempt the connection again, as it will
 	// get the changed handler the next time around.
-	// N.B.: FTP protocol is not supported by default in GNU Classpath
-	// but it does not matter because setURL will keep the current protocol
-	// if "ftp" is not available.
 	setURL (url, "ftp", url.getHost(), url.getPort(), url.getFile(),
-	        url.getRef());
+		url.getRef());
+	// Until the ftp protocol handler is written, this will cause
+	// a NullPointerException.
 	return url.openConnection();
+	*/
       }
 
     return new Connection(url);
