--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-07-31 02:26:41.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-07-31 02:23:13.000000000 -0600
@@ -139,21 +139,21 @@
     setVisibleAndEnabled ();
   }
 
-   void setParent ()
+  void setParent ()
   {
     ComponentPeer p;
     Component component = awtComponent;
     do
       {
-	component = component.getParent ();
-	p = component.getPeer ();
+        component = component.getParent ();
+        p = component.getPeer ();
       }
     while (p instanceof java.awt.peer.LightweightPeer);
-    
+
     if (p != null)
       gtkWidgetSetParent (p);
   }
-  
+
   /*
    * Set the bounds of this peer's AWT Component based on dimensions
    * returned by the native windowing system.  Most Components impose
@@ -178,7 +178,7 @@
     return i.checkImage ();
   }
 
-  public Image createImage (ImageProducer producer) 
+  public Image createImage (ImageProducer producer)
   {
     return new GtkImage (producer, null);
   }
@@ -283,12 +283,12 @@
         break;
       case KeyEvent.KEY_PRESSED:
         ke = (KeyEvent) event;
-        gtkWidgetDispatchKeyEvent (ke.getID (), ke.getWhen (), ke.getModifiers (),
+        gtkWidgetDispatchKeyEvent (ke.getID (), ke.getWhen (), ke.getModifiersEx (),
                                    ke.getKeyCode (), ke.getKeyLocation ());
         break;
       case KeyEvent.KEY_RELEASED:
         ke = (KeyEvent) event;
-        gtkWidgetDispatchKeyEvent (ke.getID (), ke.getWhen (), ke.getModifiers (),
+        gtkWidgetDispatchKeyEvent (ke.getID (), ke.getWhen (), ke.getModifiersEx (),
                                    ke.getKeyCode (), ke.getKeyLocation ());
         break;
       }
@@ -377,8 +377,8 @@
 
   public void requestFocus ()
   {
-    gtkWidgetRequestFocus();
-    postFocusEvent(FocusEvent.FOCUS_GAINED, false);
+    gtkWidgetRequestFocus ();
+    postFocusEvent (FocusEvent.FOCUS_GAINED, false);
   }
 
   public void reshape (int x, int y, int width, int height) 
