--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkGraphicsEnvironment.java	2004-10-02 02:25:48.000000000 -0600
+++ gnu/java/awt/peer/gtk/GdkGraphicsEnvironment.java	2004-07-22 11:06:18.000000000 -0600
@@ -38,17 +38,15 @@
 
 package gnu.java.awt.peer.gtk;
 
-import java.awt.Font;
-import java.awt.Graphics2D;
-import java.awt.GraphicsDevice;
+import java.awt.*;
 import java.awt.GraphicsEnvironment;
-import java.awt.Rectangle;
-import java.awt.Shape;
 import java.awt.image.BufferedImage;
 import java.util.Locale;
 
+
 public class GdkGraphicsEnvironment extends GraphicsEnvironment
 {
+  
   public GdkGraphicsEnvironment ()
   {
   	super();
@@ -83,4 +81,7 @@
   {
     throw new java.lang.UnsupportedOperationException ();
   }
-}
+
+
+} // class GdkGraphicsEnvironment
+
