--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFileDialogPeer.java	2004-07-31 02:26:41.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkFileDialogPeer.java	2004-07-31 02:23:13.000000000 -0600
@@ -114,7 +114,7 @@
 
     // GtkFileChooser requires absolute filenames. If the given filename
     // is not absolute, let's construct it based on current directory.
-        currentFile = fileName;
+    currentFile = fileName;
     if (fileName.indexOf(FS) == 0)
       {
         nativeSetFile (fileName);
