--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/locale/Calendar.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/locale/Calendar.java	2002-01-22 16:01:30.000000000 -0700
@@ -38,9 +38,9 @@
 
 package gnu.java.locale;
 
-import java.util.Date;
 import java.util.ListResourceBundle;
 import java.util.Locale;
+import java.util.Date;
 
 /**
  * This class contains locale data for java.util.Calendar.<br>
