--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/image/ImageDecoder.java	2004-07-23 02:23:49.000000000 -0600
+++ gnu/java/awt/image/ImageDecoder.java	2004-07-22 11:06:18.000000000 -0600
@@ -99,7 +99,7 @@
   public void startProduction (ImageConsumer ic)
   {
     if (!isConsumer(ic))
-    addConsumer (ic);
+      addConsumer(ic);
 
     Vector list = (Vector) consumers.clone ();
     try 
