--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SelectorImpl.java	2004-06-25 02:22:22.000000000 -0600
+++ gnu/java/nio/SelectorImpl.java	2004-07-22 11:06:21.000000000 -0600
@@ -261,7 +261,7 @@
                 if (unhandledWakeup)
                   {
                     unhandledWakeup = false;
-                    Thread.interrupted ();
+                    selectThread.interrupted ();
                   }
                 selectThread = null;
               }
