--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/PipeImpl.java	2004-07-22 02:22:22.000000000 -0600
+++ gnu/java/nio/PipeImpl.java	2004-07-22 11:06:21.000000000 -0600
@@ -168,9 +168,12 @@
     throws IOException
   {
     super();
-    VMPipe.init (this, provider);
+    nativeInit (provider);
   }
 
+  private native void nativeInit (SelectorProvider provider)
+    throws IOException;
+    
   public Pipe.SinkChannel sink()
   {
     return sink;
