--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkScrollPanePeer.java	2004-10-30 02:23:08.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkScrollPanePeer.java	2004-07-22 11:06:19.000000000 -0600
@@ -54,7 +54,7 @@
     create (awtComponent.getWidth (), awtComponent.getHeight ());
   }
 
-  // native void gtkScrolledWindowSetScrollPosition(int x, int y);
+  native void gtkScrolledWindowSetScrollPosition(int x, int y);
   native void gtkScrolledWindowSetHScrollIncrement (int u);
   native void gtkScrolledWindowSetVScrollIncrement (int u);
 
