--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkCheckboxPeer.java	2004-07-31 02:26:41.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkCheckboxPeer.java	2004-09-04 02:24:26.000000000 -0600
@@ -55,7 +55,7 @@
   public native void create (GtkCheckboxGroupPeer group);
   public native void nativeSetCheckboxGroup (GtkCheckboxGroupPeer group);
   public native void connectSignals ();
-  native void gtkSetFont (String name, int style, int size);
+  native void gtkWidgetModifyFont (String name, int style, int size);
   native void gtkButtonSetLabel (String label);
   native void gtkToggleButtonSetActive (boolean is_active);
 
