--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkLabelPeer.java	2004-06-08 02:24:15.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkLabelPeer.java	2004-09-04 02:24:27.000000000 -0600
@@ -47,7 +47,7 @@
     implements LabelPeer
 {
   native void create (String text, float alignment);
-  native void gtkSetFont (String name, int style, int size);
+  native void gtkWidgetModifyFont (String name, int style, int size);
   native void nativeSetAlignment (float alignment);
 
   native public void setText (String text);
