--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/EmbeddedWindow.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/awt/EmbeddedWindow.java	2004-07-31 02:23:12.000000000 -0600
@@ -1,5 +1,5 @@
 /* EmbeddedWindow.java --
-   Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,7 +38,6 @@
 
 package gnu.java.awt;
 
-import gnu.classpath.Configuration;
 import gnu.java.awt.peer.EmbeddedWindowPeer;
 import gnu.java.security.action.SetAccessibleAction;
 
@@ -108,7 +107,7 @@
     catch (NoSuchFieldException e)
       {
 	// This should never happen.
-      }
+  }
 
     super.addNotify();
   }
