--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkButtonPeer.java	2004-10-02 02:25:48.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkButtonPeer.java	2004-09-13 02:34:13.000000000 -0600
@@ -1,5 +1,5 @@
 /* GtkButtonPeer.java -- Implements ButtonPeer with GTK
-   Copyright (C) 1998, 1999, 2004  Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -43,8 +43,8 @@
 import java.awt.Component;
 import java.awt.Font;
 import java.awt.Point;
-import java.awt.event.KeyEvent;
 import java.awt.event.MouseEvent;
+import java.awt.event.KeyEvent;
 import java.awt.peer.ButtonPeer;
 
 public class GtkButtonPeer extends GtkComponentPeer
