--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkImagePainter.java	2004-07-23 02:23:49.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkImagePainter.java	2004-09-04 02:24:27.000000000 -0600
@@ -75,7 +75,7 @@
     s_width = s_height = 0;
     clip = null;
 
-    new Thread (this).start ();
+    run ();
   }
 
   public
@@ -105,7 +105,7 @@
     s_height = Math.abs (sy2 - sy1);
     clip = new Rectangle (sx1, sy1, s_width, s_height);
 
-    new Thread (this).start ();
+    run ();
   }
 
   public void
