--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkMenuItemPeer.java	2004-04-18 02:22:09.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkMenuItemPeer.java	2004-08-26 02:28:56.000000000 -0600
@@ -89,10 +89,8 @@
     setEnabled (true);
   }
 
-  public void setEnabled (boolean b)
-  {
-    // do nothing, for now.
-  }
+  native public void setEnabled (boolean b);
+
   native public void setLabel (String label);
 
   protected void postMenuActionEvent ()
