--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/net/protocol/http/Connection.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/net/protocol/http/Connection.java	2004-07-22 11:06:21.000000000 -0600
@@ -1,6 +1,5 @@
 /* HttpURLConnection.java -- URLConnection class for HTTP protocol
-   Copyright (C) 1998, 1999, 2000, 2002, 2003, 2004
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -53,13 +52,10 @@
 import java.net.Socket;
 import java.net.URL;
 import java.net.URLConnection;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
 import gnu.java.net.HeaderFieldHelper;
-import gnu.java.security.action.GetPropertyAction;
 
 /**
  * This subclass of java.net.URLConnection models a URLConnection via
@@ -80,40 +76,31 @@
    * The socket we are connected to
    */
   private Socket socket;
-  
-  // Properties depeending on system properties settings
-  static int proxyPort = 80;
-  static boolean proxyInUse = false;
-  static String proxyHost = null;
-  static String userAgent;
+  private static int proxyPort = 80;
+  private static boolean proxyInUse = false;
+  private static String proxyHost = null;
 
   static 
   {
     // Recognize some networking properties listed at
     // http://java.sun.com/j2se/1.4/docs/guide/net/properties.html.
     String port = null;
-    GetPropertyAction getProperty = new GetPropertyAction("http.proxyHost");
-    proxyHost = (String) AccessController.doPrivileged(getProperty);
+    proxyHost = System.getProperty("http.proxyHost");
     if (proxyHost != null)
       {
 	proxyInUse = true;
-	getProperty.setParameters("http.proxyPort");
-	port = (String) AccessController.doPrivileged(getProperty);
-	if (port != null)
+	if ((port = System.getProperty("http.proxyPort")) != null)
 	  {
 	    try
 	      {
 		proxyPort = Integer.parseInt(port);
 	      }
-	    catch (NumberFormatException ex)
+	    catch (Throwable t)
 	      {
 		// Nothing.  
 	      }
 	  }
       }
-
-    getProperty.setParameters("http.agent");
-    userAgent = (String) AccessController.doPrivileged(getProperty);
   }
 
   /**
@@ -228,7 +215,8 @@
       setRequestProperty ("Connection", "Close");
     
     if (getRequestProperty ("user-agent") == null)
-      setRequestProperty ("user-agent", userAgent);
+      setRequestProperty ("user-agent", "gnu-libgcj/"
+                          + System.getProperty ("java.vm.version"));
     
     if (getRequestProperty ("accept") == null)
       setRequestProperty ("accept", "*/*");
