--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/net/protocol/http/Connection.java	2004-07-24 02:24:52.000000000 -0600
+++ gnu/java/net/protocol/http/Connection.java	2004-07-22 11:06:21.000000000 -0600
@@ -1,6 +1,5 @@
 /* HttpURLConnection.java -- URLConnection class for HTTP protocol
-   Copyright (C) 1998, 1999, 2000, 2002, 2003, 2004
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -53,8 +52,6 @@
 import java.net.Socket;
 import java.net.URL;
 import java.net.URLConnection;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
@@ -79,45 +76,31 @@
    * The socket we are connected to
    */
   private Socket socket;
-  
-  // Properties depeending on system properties settings
-  static int proxyPort = 80;
-  static boolean proxyInUse = false;
-  static String proxyHost = null;
-  static String userAgent;
+  private static int proxyPort = 80;
+  private static boolean proxyInUse = false;
+  private static String proxyHost = null;
 
   static 
   {
-    // Make sure access control for system properties depends only on
-    // our class ProtectionDomain, not on any (indirect) callers.
-    AccessController.doPrivileged(new PrivilegedAction() {
-	public Object run()
-	{
-	  // Recognize some networking properties listed at
-	  // http://java.sun.com/j2se/1.4/docs/guide/net/properties.html.
-	  String port = null;
-	  proxyHost = System.getProperty("http.proxyHost");
-	  if (proxyHost != null)
-	    {
-	      proxyInUse = true;
-	      if ((port = System.getProperty("http.proxyPort")) != null)
-		{
-		  try
-		    {
-		      proxyPort = Integer.parseInt(port);
-		    }
-		  catch (Throwable t)
-		    {
-		      // Nothing.  
-		    }
-		}
-	    }
-	  
-	  userAgent = System.getProperty("http.agent");
-
-	  return null;
-	}
-      });
+    // Recognize some networking properties listed at
+    // http://java.sun.com/j2se/1.4/docs/guide/net/properties.html.
+    String port = null;
+    proxyHost = System.getProperty("http.proxyHost");
+    if (proxyHost != null)
+      {
+	proxyInUse = true;
+	if ((port = System.getProperty("http.proxyPort")) != null)
+	  {
+	    try
+	      {
+		proxyPort = Integer.parseInt(port);
+	      }
+	    catch (Throwable t)
+	      {
+		// Nothing.  
+	      }
+	  }
+      }
   }
 
   /**
@@ -232,7 +215,8 @@
       setRequestProperty ("Connection", "Close");
     
     if (getRequestProperty ("user-agent") == null)
-      setRequestProperty ("user-agent", userAgent);
+      setRequestProperty ("user-agent", "gnu-libgcj/"
+                          + System.getProperty ("java.vm.version"));
     
     if (getRequestProperty ("accept") == null)
       setRequestProperty ("accept", "*/*");
