--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkMenuBarPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkMenuBarPeer.java	2004-08-26 02:28:55.000000000 -0600
@@ -56,15 +56,12 @@
     create ();
   }
 
+  native void nativeSetHelpMenu(MenuPeer menuPeer);
+
   /* In Gnome, help menus are no longer right flushed. */
   public void addHelpMenu (Menu menu)
   {
-    addMenu (menu);
-  }
-
-  public void addMenu (Menu menu)
-  {
-    addMenu ((MenuPeer) menu.getPeer ());
+    nativeSetHelpMenu((MenuPeer) menu.getPeer());
   }
 
   native public void delMenu (int index);
