--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkToolkit.java	2004-07-31 02:26:41.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkToolkit.java	2004-07-23 02:21:45.000000000 -0600
@@ -266,7 +266,7 @@
   {
     if (useGraphics2D())
       return bufferedImageOrError(GdkPixbufDecoder.createBufferedImage (imagedata,
-                                                   imageoffset, 
+                                                                        imageoffset, 
                                                                         imagelength));
     else
       {
@@ -510,8 +510,12 @@
    */
   private FontPeer getFontPeer (String name, int style, int size) 
   {
-    GtkFontPeer fp = new GtkFontPeer (name, style, size);
-    return fp;
+    try {
+      GtkFontPeer fp = new GtkFontPeer (name, style, size);
+      return fp;
+    } catch (MissingResourceException ex) {
+      return null;
+    }
   }
 
   /**
