--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-10-30 02:23:08.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-10-14 02:21:07.000000000 -0600
@@ -87,6 +87,7 @@
 
   native int[] gtkWidgetGetForeground ();
   native int[] gtkWidgetGetBackground ();
+  native void gtkWidgetSetVisible (boolean b);
   native void gtkWidgetGetDimensions (int[] dim);
   native void gtkWidgetGetPreferredDimensions (int[] dim);
   native void gtkWidgetGetLocationOnScreen (int[] point);
@@ -106,6 +107,7 @@
     throw new RuntimeException ();
   }
 
+  native void connectJObject ();
   native void connectSignals ();
 
   protected GtkComponentPeer (Component awtComponent)
