--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkListPeer.java	2004-08-01 02:25:24.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkListPeer.java	2004-09-04 02:24:27.000000000 -0600
@@ -60,7 +60,7 @@
   native void create (int rows);
   native void connectJObject ();
   native void connectSignals ();
-  native void gtkSetFont (String name, int style, int size);
+  native void gtkWidgetModifyFont (String name, int style, int size);
   native void gtkWidgetRequestFocus ();
 
   native void getSize (int rows, int visibleRows, int dims[]);
