--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/regexp/CharIndexedInputStream.java	2004-07-01 02:24:31.000000000 -0600
+++ gnu/regexp/CharIndexedInputStream.java	2004-07-22 11:06:23.000000000 -0600
@@ -36,9 +36,9 @@
 exception statement from your version. */
 
 package gnu.regexp;
+import java.io.InputStream;
 import java.io.BufferedInputStream;
 import java.io.IOException;
-import java.io.InputStream;
 
 // TODO: move(x) shouldn't rely on calling next() x times
 
