--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/Box.java	2004-08-01 02:25:27.000000000 -0600
+++ javax/swing/Box.java	2004-08-04 02:30:52.000000000 -0600
@@ -198,7 +198,7 @@
   
   public static Box createHorizontalBox()
   {
-    return null;
+    return new Box(BoxLayout.X_AXIS);
   }
   
   /**
@@ -231,12 +231,12 @@
   
   public static Component createRigidArea(Dimension d)
   {
-    return null;
+    return new Filler(d, d, d);
   }
   
   public static Box createVerticalBox()
   {
-    return null;
+    return new Box(BoxLayout.Y_AXIS);
   }
   
   /**
