--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JComponent.java	2004-10-28 02:27:43.000000000 -0600
+++ javax/swing/JComponent.java	2004-11-11 02:25:41.000000000 -0700
@@ -325,6 +325,8 @@
   private ActionMap actionMap;
   private InputVerifier inputVerifier;
 
+  private TransferHandler transferHandler;
+
   /** 
    * A lock held during recursive painting; this is used to serialize
    * access to the double buffer, and also to select the "top level" 
@@ -862,9 +864,10 @@
    */
   protected Graphics getComponentGraphics(Graphics g)
   {    
-    g.setFont(this.getFont());
-    g.setColor(this.getForeground());
-    return g;
+    Graphics g2 = g.create();
+    g2.setFont(this.getFont());
+    g2.setColor(this.getForeground());
+    return g2;
   }
 
 
@@ -2069,6 +2072,32 @@
   }
 
   /**
+   * Get the value of the {@link #transferHandler} property.
+   *
+   * @return The current value of the property
+   *
+   * @see ComponentUI#setTransferHandler
+   */
+
+  public TransferHandler getTransferHandler()
+  {
+    return transferHandler;
+  }
+
+  /**
+   * Set the value of the {@link #transferHandler} property.
+   *
+   * @param newHandler The new value of the property
+   *
+   * @see ComponentUI#getTransferHandler
+   */
+
+  void setTransferHandler (TransferHandler newHandler)
+  {
+    transferHandler = newHandler;
+  }
+
+  /**
    * Set the value of the {@link #opaque} property, revalidate and repaint
    * this component.
    *
