--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JOptionPane.java	2004-07-23 02:24:08.000000000 -0600
+++ javax/swing/JOptionPane.java	2004-07-22 11:06:52.000000000 -0600
@@ -66,7 +66,7 @@
   {
     /** DOCUMENT ME! */
     private static final long serialVersionUID = 686071432213084821L;
-  
+
     /**
      * Creates a new AccessibleJOptionPane object.
      */
@@ -87,7 +87,7 @@
 
   /** DOCUMENT ME! */
   private static final long serialVersionUID = 5231143276678566796L;
-  
+
   /** The value returned when cancel option is selected. */
   public static final int CANCEL_OPTION = 2;
 
