--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JMenuItem.java	2004-10-28 02:27:44.000000000 -0600
+++ javax/swing/JMenuItem.java	2004-11-11 02:25:42.000000000 -0700
@@ -529,19 +529,20 @@
    */
   public void menuSelectionChanged(boolean changed)
   {
+    Component parent = this.getParent();
     if (changed)
       {
 	model.setArmed(true);
 
-	if (this.getParent() instanceof JPopupMenu)
-	  ((JPopupMenu) this.getParent()).setSelected(this);
+	if (parent != null && parent instanceof JPopupMenu)
+	  ((JPopupMenu) parent).setSelected(this);
       }
     else
       {
 	model.setArmed(false);
 
-	if (this.getParent() instanceof JPopupMenu)
-	  ((JPopupMenu) this.getParent()).getSelectionModel().clearSelection();
+	if (parent != null && parent instanceof JPopupMenu)
+	  ((JPopupMenu) parent).getSelectionModel().clearSelection();
       }
   }
 
