--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicPopupMenuSeparatorUI.java	2004-07-09 02:22:40.000000000 -0600
+++ javax/swing/plaf/basic/BasicPopupMenuSeparatorUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -53,7 +53,7 @@
 
 
 /**
- * The Basic Look and Feel UI delegate for JPopupMenu.Separator.
+ * The Basic L&F UI delegate for JPopupMenu.Separator.
  */
 public class BasicPopupMenuSeparatorUI extends BasicSeparatorUI
 {
