--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/SimpleAttributeSet.java	2004-09-04 11:14:01.000000000 -0600
+++ javax/swing/text/SimpleAttributeSet.java	2004-09-14 02:22:07.000000000 -0600
@@ -45,9 +45,9 @@
 public class SimpleAttributeSet
   implements MutableAttributeSet, Serializable, Cloneable
 {
-  Hashtable tab;
+  public static final AttributeSet EMPTY = new SimpleAttributeSet();
 
-  static AttributeSet EMPTY = new SimpleAttributeSet();
+  Hashtable tab;
 
   public SimpleAttributeSet()
   {
