--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/DefaultKeyboardFocusManager.java	2004-09-05 02:26:44.000000000 -0600
+++ java/awt/DefaultKeyboardFocusManager.java	2004-08-26 02:28:58.000000000 -0600
@@ -311,7 +311,7 @@
                         MenuItem item = menu.getItem (j);
                         MenuShortcut shortcut = item.getShortcut ();
 
-                        if (shortcut != null)
+                        if (item.isEnabled() && shortcut != null)
                           {
                             // Dispatch a new ActionEvent if:
                             //
@@ -444,7 +444,7 @@
 
     Component previous = policy.getComponentBefore (focusCycleRoot, focusComp);
     if (previous != null)
-    previous.requestFocusInWindow ();
+      previous.requestFocusInWindow ();
   }
 
   public void focusNextComponent (Component comp)
@@ -455,7 +455,7 @@
 
     Component next = policy.getComponentAfter (focusCycleRoot, focusComp);
     if (next != null)
-    next.requestFocusInWindow ();
+      next.requestFocusInWindow ();
   }
 
   public void upFocusCycle (Component comp)
@@ -468,7 +468,7 @@
         FocusTraversalPolicy policy = focusCycleRoot.getFocusTraversalPolicy ();
         Component defaultComponent = policy.getDefaultComponent (focusCycleRoot);
         if (defaultComponent != null)
-        defaultComponent.requestFocusInWindow ();
+          defaultComponent.requestFocusInWindow ();
       }
     else
       {
@@ -489,7 +489,7 @@
         FocusTraversalPolicy policy = cont.getFocusTraversalPolicy ();
         Component defaultComponent = policy.getDefaultComponent (cont);
         if (defaultComponent != null)
-        defaultComponent.requestFocusInWindow ();
+          defaultComponent.requestFocusInWindow ();        
         setGlobalCurrentFocusCycleRoot (cont);
       }
   }
