--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicArrowButton.java	2004-07-23 02:24:10.000000000 -0600
+++ javax/swing/plaf/basic/BasicArrowButton.java	2004-07-22 11:06:55.000000000 -0600
@@ -171,7 +171,7 @@
 
   /** The top and left edges of the button. */
   private transient Color highlight = Color.BLACK;
-
+  
   /** The border around the ArrowButton. */
   private transient Border tmpBorder = new Border()
   {
