--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/Raster.java	2004-09-23 02:37:57.000000000 -0600
+++ java/awt/image/Raster.java	2004-11-03 02:20:31.000000000 -0700
@@ -128,8 +128,8 @@
 						  int w, int h, int bands,
 						  Point location)
   {
-    // FIXME: Implement;
-    throw new UnsupportedOperationException("not implemented yet");
+    SampleModel sm = new BandedSampleModel(dataType, w, h, bands);
+    return createWritableRaster(sm, location);
   }
 
   public static WritableRaster createBandedRaster(int dataType,
@@ -139,8 +139,9 @@
 						  int[] bandOffsets,
 						  Point location)
   {
-    // FIXME: Implement;
-    throw new UnsupportedOperationException("not implemented yet");
+    SampleModel sm = new BandedSampleModel(dataType, w, h, scanlineStride,
+					   bankIndices, bandOffsets);
+    return createWritableRaster(sm, location);
   }
   
   public static WritableRaster createPackedRaster(int dataType,
@@ -204,8 +205,10 @@
 				    int[] bandOffsets,
 				    Point location)
   {
-    // FIXME: Implement;
-    throw new UnsupportedOperationException("not implemented yet");
+    SampleModel sm = new BandedSampleModel(dataBuffer.getDataType(),
+					   w, h, scanlineStride,
+					   bankIndices, bandOffsets);
+    return createWritableRaster(sm, dataBuffer, location);
   }
   
   public static WritableRaster
@@ -213,7 +216,8 @@
 		     int w, int h,
 		     int scanlineStride,
 		     int[] bandMasks,
-		     Point location) {
+		     Point location)
+ {
     SampleModel sm =
       new SinglePixelPackedSampleModel(dataBuffer.getDataType(),
 				       w, h,
