--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/SpinnerNumberModel.java	2004-09-27 02:19:52.000000000 -0600
+++ javax/swing/SpinnerNumberModel.java	2004-09-14 02:22:03.000000000 -0600
@@ -46,10 +46,8 @@
  * @version 1.0
  */
 public class SpinnerNumberModel extends AbstractSpinnerModel
+  implements Serializable
 {
-  /**
-   * For compatability with Sun's JDK
-   */
   private static final long serialVersionUID = 7279176385485777821L;
 
   /** DOCUMENT ME! */
