--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Window.java	2004-09-23 02:37:57.000000000 -0600
+++ java/awt/Window.java	2004-09-04 02:24:36.000000000 -0600
@@ -714,7 +714,7 @@
    *
    * @deprecated
    */
-  public boolean postEvent(Event e)
+  public boolean postEvent (Event e)
   {
     return handleEvent (e);
   }
@@ -748,7 +748,7 @@
   public void applyResourceBundle(String rbName)
   {
     ResourceBundle rb = ResourceBundle.getBundle(rbName, Locale.getDefault(),
-      ClassLoader.getSystemClassLoader());
+      Window.class.getClassLoader());
     if (rb != null)
       applyResourceBundle(rb);    
   }
