--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Choice.java	2004-10-28 02:27:33.000000000 -0600
+++ java/awt/Choice.java	2004-02-04 02:18:37.000000000 -0700
@@ -1,5 +1,5 @@
 /* Choice.java -- Java choice button widget.
-   Copyright (C) 1999, 2000, 2001, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -169,8 +169,6 @@
       ChoicePeer cp = (ChoicePeer) peer;
       cp.add (item, i);
     }
-  else if (selectedIndex == -1) 
-    select(0);
 }
 
 /*************************************************************************/
@@ -220,8 +218,6 @@
       ChoicePeer cp = (ChoicePeer) peer;
       cp.add (item, index);
     }
-  else if (selectedIndex == -1 || selectedIndex >= index)
-    select(0);
 }
 
 /*************************************************************************/
@@ -265,13 +261,6 @@
       ChoicePeer cp = (ChoicePeer) peer;
       cp.remove (index);
     }
-  else
-    {
-      if (getItemCount() == 0)
-	selectedIndex = -1;
-      else if (index == selectedIndex)
-	select(0);
-    }
 
   if (selectedIndex > index)
     --selectedIndex;
