--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Frame.java	2004-10-12 02:18:11.000000000 -0600
+++ java/awt/Frame.java	2004-09-28 02:20:45.000000000 -0600
@@ -199,7 +199,7 @@
   /*
    * The number used to generate the name returned by getName.
    */
-  private static transient long next_frame_number;
+  private static transient long next_frame_number = 0;
 
 /**
   * Initializes a new instance of <code>Frame</code> that is not visible
