--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultButtonModel.java	2004-09-05 02:26:52.000000000 -0600
+++ javax/swing/DefaultButtonModel.java	2004-08-28 02:26:10.000000000 -0600
@@ -340,7 +340,7 @@
 	fireItemStateChanged(new ItemEvent(this, ItemEvent.ITEM_STATE_CHANGED,
 	                                   null, ItemEvent.SELECTED));
         if (group != null)
-	  group.setSelected(this, true);
+          group.setSelected(this, true);
       }
 
     else if ((oldstate & SELECTED) == SELECTED && (newstate & SELECTED) == 0)
@@ -348,7 +348,7 @@
 	fireItemStateChanged(new ItemEvent(this, ItemEvent.ITEM_STATE_CHANGED,
 	                                   null, ItemEvent.DESELECTED));
         if (group != null)
-	  group.setSelected(this, false);
+          group.setSelected(this, false);
       }
 
     else if (((oldstate & ARMED) == ARMED && (oldstate & PRESSED) == PRESSED)
