--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Point.java	2004-09-13 02:44:25.000000000 -0600
+++ java/awt/Point.java	2002-08-12 10:11:00.000000000 -0600
@@ -95,7 +95,7 @@
    * Initializes a new instance of <code>Point</code> with coordinates
    * identical to the coordinates of the specified points.
    *
-   * @param p the point to copy the coordinates from
+   * @param point the point to copy the coordinates from
    * @throws NullPointerException if p is null
    */
   public Point(Point p)
