--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2004-07-23 02:24:08.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-07-22 11:06:54.000000000 -0600
@@ -498,7 +498,7 @@
                       || vsb.getMaximum() != vs.height
                       || vsb.getValue() != vr.y
                       || vsb.getVisibleAmount() != vr.height))
-                  vsb.setValue(vr.y, vr.height, 0, vs.height);
+                vsb.setValue(vr.y, vr.height, 0, vs.height);
 
               if (hsb != null
                   && (hsb.getMinimum() != 0
