--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ToolTipManager.java	2004-07-23 02:24:09.000000000 -0600
+++ javax/swing/ToolTipManager.java	2004-07-22 11:06:55.000000000 -0600
@@ -47,305 +47,305 @@
 /**
  * ToolTipManager
  * 
- * @author	Andrew Selkirk
+ * @author Andrew Selkirk
  */
 public class ToolTipManager extends MouseAdapter
   implements MouseMotionListener
 {
-	/**
-	 * stillInsideTimerAction
-	 */
+  /**
+   * stillInsideTimerAction
+   */
   protected class stillInsideTimerAction
     implements ActionListener
   {
-		/**
-		 * Constructor stillInsideTimerAction
-		 */
+    /**
+     * Constructor stillInsideTimerAction
+     */
     protected stillInsideTimerAction()
     {
-			// TODO
+      // TODO
     }
 
-		/**
-		 * actionPerformed
-		 * @param event TODO
-		 */
+    /**
+     * actionPerformed
+     * @param event TODO
+     */
     public void actionPerformed(ActionEvent event)
     {
-			// TODO
+      // TODO
     }
   }
 
-	/**
-	 * outsideTimerAction
-	 */
+  /**
+   * outsideTimerAction
+   */
   protected class outsideTimerAction
     implements ActionListener
   {
-		/**
-		 * Constructor outsideTimerAction
-		 */
+    /**
+     * Constructor outsideTimerAction
+     */
     protected outsideTimerAction()
     {
-			// TODO
+      // TODO
     }
 
-		/**
-		 * actionPerformed
-		 * @param value0 TODO
-		 */
+    /**
+     * actionPerformed
+     * @param value0 TODO
+     */
     public void actionPerformed(ActionEvent event)
     {
-			// TODO
+      // TODO
     }
   }
 
-	/**
-	 * insideTimerAction
-	 */
+  /**
+   * insideTimerAction
+   */
   protected class insideTimerAction
     implements ActionListener
   {
-		/**
-		 * Constructor insideTimerAction
-		 */
+    /**
+     * Constructor insideTimerAction
+     */
     protected insideTimerAction()
     {
-			// TODO
+      // TODO
     }
 
-		/**
-		 * actionPerformed
-		 * @param event TODO
-		 */
+    /**
+     * actionPerformed
+     * @param event TODO
+     */
     public void actionPerformed(ActionEvent event)
     {
-			// TODO
+      // TODO
     }
   }
 
-	/**
-	 * enterTimer
-	 */
-	Timer enterTimer;
-
-	/**
-	 * exitTimer
-	 */
-	Timer exitTimer;
-
-	/**
-	 * insideTimer
-	 */
-	Timer insideTimer;
-
-	/**
-	 * toolTipText
-	 */
-	String toolTipText;
-
-	/**
-	 * mouseEvent
-	 */
-	MouseEvent mouseEvent;
-
-	/**
-	 * showImmediately
-	 */
-	boolean showImmediately;
-
-	/**
-	 * tip
-	 */
-	JToolTip tip;
-
-	/**
-	 * enabled
-	 */
-	boolean enabled;
-
-	/**
-	 * timerEnter
-	 */
-	private long timerEnter;
-
-	/**
-	 * lightWeightPopupEnabled
-	 */
-	protected boolean lightWeightPopupEnabled;
-
-	/**
-	 * heavyWeightPopupEnabled
-	 */
-	protected boolean heavyWeightPopupEnabled;
-
-	/**
-	 * Constructor ToolTipManager
-	 */
+  /**
+   * enterTimer
+   */
+  Timer enterTimer;
+
+  /**
+   * exitTimer
+   */
+  Timer exitTimer;
+
+  /**
+   * insideTimer
+   */
+  Timer insideTimer;
+
+  /**
+   * toolTipText
+   */
+  String toolTipText;
+
+  /**
+   * mouseEvent
+   */
+  MouseEvent mouseEvent;
+
+  /**
+   * showImmediately
+   */
+  boolean showImmediately;
+
+  /**
+   * tip
+   */
+  JToolTip tip;
+
+  /**
+   * enabled
+   */
+  boolean enabled;
+
+  /**
+   * timerEnter
+   */
+  private long timerEnter;
+
+  /**
+   * lightWeightPopupEnabled
+   */
+  protected boolean lightWeightPopupEnabled;
+
+  /**
+   * heavyWeightPopupEnabled
+   */
+  protected boolean heavyWeightPopupEnabled;
+
+  /**
+   * Constructor ToolTipManager
+   */
   ToolTipManager()
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * sharedInstance
+  /**
+   * sharedInstance
    * @return ToolTipManager
-	 */
+   */
   public static ToolTipManager sharedInstance()
   {
-		return null; // TODO
+    return null; // TODO
   }
 
-	/**
-	 * setEnabled
-	 * @param enabled TODO
-	 */
+  /**
+   * setEnabled
+   * @param enabled TODO
+   */
   public void setEnabled(boolean enabled)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * isEnabled
+  /**
+   * isEnabled
    * @return boolean
-	 */
+   */
   public boolean isEnabled()
   {
-		return false; // TODO
+    return false; // TODO
   }
 
-	/**
-	 * isLightWeightPopupEnabled
+  /**
+   * isLightWeightPopupEnabled
    * @return boolean
-	 */
+   */
   public boolean isLightWeightPopupEnabled()
   {
-		return false; // TODO
+    return false; // TODO
   }
 
-	/**
-	 * setLightWeightPopupEnabled
-	 * @param enabled TODO
-	 */
+  /**
+   * setLightWeightPopupEnabled
+   * @param enabled TODO
+   */
   public void setLightWeightPopupEnabled(boolean enabled)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * getInitialDelay
+  /**
+   * getInitialDelay
    * @return int
-	 */
+   */
   public int getInitialDelay()
   {
-		return 0; // TODO
+    return 0; // TODO
   }
 
-	/**
-	 * setInitialDelay
-	 * @param delay TODO
-	 */
+  /**
+   * setInitialDelay
+   * @param delay TODO
+   */
   public void setInitialDelay(int delay)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * getDismissDelay
+  /**
+   * getDismissDelay
    * @return int
-	 */
+   */
   public int getDismissDelay()
   {
-		return 0; // TODO
+    return 0; // TODO
   }
 
-	/**
-	 * setDismissDelay
-	 * @param delay TODO
-	 */
+  /**
+   * setDismissDelay
+   * @param delay TODO
+   */
   public void setDismissDelay(int delay)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * getReshowDelay
+  /**
+   * getReshowDelay
    * @return int
-	 */
+   */
   public int getReshowDelay()
   {
-		return 0; // TODO
+    return 0; // TODO
   }
 
-	/**
-	 * setReshowDelay
-	 * @param delay TODO
-	 */
+  /**
+   * setReshowDelay
+   * @param delay TODO
+   */
   public void setReshowDelay(int delay)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * registerComponent
-	 * @param component TODO
-	 */
+  /**
+   * registerComponent
+   * @param component TODO
+   */
   public void registerComponent(JComponent component)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * unregisterComponent
-	 * @param component TODO
-	 */
+  /**
+   * unregisterComponent
+   * @param component TODO
+   */
   public void unregisterComponent(JComponent component)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * mouseEntered
-	 * @param event TODO
-	 */
+  /**
+   * mouseEntered
+   * @param event TODO
+   */
   public void mouseEntered(MouseEvent event)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * mouseExited
-	 * @param event TODO
-	 */
+  /**
+   * mouseExited
+   * @param event TODO
+   */
   public void mouseExited(MouseEvent event)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * mousePressed
-	 * @param event TODO
-	 */
+  /**
+   * mousePressed
+   * @param event TODO
+   */
   public void mousePressed(MouseEvent event)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * mouseDragged
-	 * @param event TODO
-	 */
+  /**
+   * mouseDragged
+   * @param event TODO
+   */
   public void mouseDragged(MouseEvent event)
   {
-		// TODO
+    // TODO
   }
 
-	/**
-	 * mouseMoved
-	 * @param event TODO
-	 */
+  /**
+   * mouseMoved
+   * @param event TODO
+   */
   public void mouseMoved(MouseEvent event)
   {
-		// TODO
+    // TODO
   }
 }
