--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-08-01 02:25:25.000000000 -0600
+++ java/awt/Component.java	2004-08-04 02:30:15.000000000 -0600
@@ -3028,6 +3028,7 @@
           mouseListener.mouseReleased(e);
         break;
       }
+      e.consume();
   }
 
   /**
@@ -3055,6 +3056,7 @@
           mouseMotionListener.mouseMoved(e);
         break;
       }
+      e.consume();
   }
 
   /**
@@ -3073,7 +3075,10 @@
   {
     if (mouseWheelListener != null
         && e.id == MouseEvent.MOUSE_WHEEL)
+    {
       mouseWheelListener.mouseWheelMoved(e);
+      e.consume();
+    }	
   }
 
   /**
