--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicSeparatorUI.java	2004-07-09 02:22:41.000000000 -0600
+++ javax/swing/plaf/basic/BasicSeparatorUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -51,7 +51,7 @@
 import javax.swing.SwingUtilities;
 
 /**
- * The Basic Look and Feel UI delegate for JSeparator.
+ * The Basic L&F UI delegate for JSeparator.
  */
 public class BasicSeparatorUI extends SeparatorUI
 {
@@ -113,7 +113,7 @@
 
   /**
    * This method installs the defaults that are given by
-   * the Basic Look and Feel.
+   * the Basic L&F.
    *
    * @param s The JSeparator that is being installed.
    */
@@ -128,7 +128,7 @@
 
   /**
    * This method removes the defaults that were given
-   * by the Basic Look and Feel.
+   * by the Basic L&F.
    *
    * @param s The JSeparator that is being uninstalled.
    */
