--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JPopupMenu.java	2004-07-23 02:24:08.000000000 -0600
+++ javax/swing/JPopupMenu.java	2004-07-22 11:06:53.000000000 -0600
@@ -179,12 +179,12 @@
   }
 
   /**
-   * Adds given menu item to the popup menu
-   *
-   * @param item menu item to add to the popup menu
-   *
-   * @return menu item that was added to the popup menu
-   */
+  * Adds given menu item to the popup menu
+  *
+  * @param item menu item to add to the popup menu
+  *
+  * @return menu item that was added to the popup menu
+  */
   public JMenuItem add(JMenuItem item)
   {
     this.insert(item, -1);
@@ -463,7 +463,7 @@
     if (label != this.label)
       {
 	String oldLabel = this.label;
-    this.label = label;
+	this.label = label;
 	firePropertyChange(LABEL_CHANGED_PROPERTY, oldLabel, label);
       }
   }
@@ -621,11 +621,11 @@
 	      {
 		// Subtract insets of the top-level container if popup menu's
 		// top-left corner is inside it.
-		    Insets insets = rootContainer.getInsets();
-		    popup.show(popupLocation.x - insets.left,
-		               popupLocation.y - insets.top, size.width,
-		               size.height);
-		  }
+		Insets insets = rootContainer.getInsets();
+		popup.show(popupLocation.x - insets.left,
+		           popupLocation.y - insets.top, size.width,
+		           size.height);
+	      }
 	  }
 	else
 	  {
