--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ScrollPaneLayout.java	2004-07-23 02:24:09.000000000 -0600
+++ javax/swing/ScrollPaneLayout.java	2004-07-22 11:06:54.000000000 -0600
@@ -237,7 +237,7 @@
 
             if (viewportBorder != null)
               {
-              viewportInsets = viewportBorder.getBorderInsets(parent);
+                viewportInsets = viewportBorder.getBorderInsets(parent);
                 if (viewportInsets != null)
                   viewportInsetsSize.setSize(viewportInsets.left + viewportInsets.right,
                                              viewportInsets.top + viewportInsets.bottom);
@@ -308,12 +308,12 @@
 
             if (viewportBorder != null)
               {
-              viewportInsets = viewportBorder.getBorderInsets(parent);
+                viewportInsets = viewportBorder.getBorderInsets(parent);
                 if (viewportInsets != null)
                   viewportInsetsSize.setSize(viewportInsets.left + viewportInsets.right,
                                              viewportInsets.top + viewportInsets.bottom);
               }
-
+            
             if (insets != null)
               insetsSize.setSize(insets.left + insets.right,
                                  insets.top + insets.bottom);
@@ -321,7 +321,7 @@
             maybeSetMinimumSize(viewport, viewportSize);
             maybeSetMinimumSize(colHead, columnHeaderSize);
             maybeSetMinimumSize(rowHead, rowHeaderSize);
-
+            
             if (vsbPolicy != VERTICAL_SCROLLBAR_NEVER)
               maybeSetMinimumSize(vsb, verticalScrollBarSize);
 
@@ -418,7 +418,7 @@
             
             if (!showVsb)
               x3 = x4;
-
+            
             if (!showHsb)
               y3 = y4;
 
@@ -436,7 +436,7 @@
             if (showVsb)
               {
                 vsb.setVisible(true);
-	      vsb.setBounds(new Rectangle(x3, y2, x4-x3, y3-y2));
+                vsb.setBounds(new Rectangle(x3, y2, x4-x3, y3-y2));
               }
             else if (vsb != null)
               vsb.setVisible(false);
@@ -444,7 +444,7 @@
             if (showHsb)
               {
                 hsb.setVisible(true);
-              hsb.setBounds(new Rectangle(x2, y3, x3-x2, y4-y3));
+                hsb.setBounds(new Rectangle(x2, y3, x3-x2, y4-y3));
               }
             else if (hsb != null)
               hsb.setVisible(false);
