--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/MenuItem.java	2004-06-27 02:20:36.000000000 -0600
+++ java/awt/MenuItem.java	2004-08-26 02:28:58.000000000 -0600
@@ -77,7 +77,7 @@
 /**
   * @serial Indicates whether or not this menu item is enabled.
   */
-private boolean enabled;
+private boolean enabled = true;
 
 /**
   * @serial The mask of events that are enabled for this menu item.
