--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/EventDispatchThread.java	2004-10-12 02:18:11.000000000 -0600
+++ java/awt/EventDispatchThread.java	2004-09-28 02:20:45.000000000 -0600
@@ -43,14 +43,14 @@
  */
 class EventDispatchThread extends Thread
 {
-  private static int dispatchThreadNum;
+  private static int dispatchThreadNum = 1;
 
   private EventQueue queue;
 
   EventDispatchThread(EventQueue queue)
   {
     super();
-    setName("AWT-EventQueue-" + ++dispatchThreadNum);
+    setName("AWT-EventQueue-" + dispatchThreadNum++);
     this.queue = queue;
     setPriority(NORM_PRIORITY + 1);
     start();
