--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Menu.java	2004-04-18 02:22:13.000000000 -0600
+++ java/awt/Menu.java	2004-09-13 02:34:35.000000000 -0600
@@ -72,17 +72,20 @@
 /**
   * @serial Flag indicating whether or not this menu is a tear off
   */
-private boolean isTearOff;
+private boolean tearOff;
 
 /**
   * @serial Indicates whether or not this is a help menu.
   */
 private boolean isHelpMenu;
 
-// From the serialization spec.  FIXME: what should it be?
-private int menuSerializedDataVersion;
+  /*
+   * @serial Unused in this implementation, but present in Sun's
+   * serialization spec.  Value obtained via reflection.
+   */
+  private int menuSerializedDataVersion = 1;
 
-static final String separatorLabel = "-";
+static final transient String separatorLabel = "-";
 
 /*************************************************************************/
 
@@ -134,7 +137,7 @@
 {
   super(label);
 
-  this.isTearOff = isTearOff;
+  tearOff = isTearOff;
 
   if (label.equals("Help"))
     isHelpMenu = true;
@@ -158,7 +161,7 @@
 public boolean
 isTearOff()
 {
-  return(isTearOff);
+  return(tearOff);
 }
 
 /*************************************************************************/
@@ -251,8 +254,6 @@
   * @param item The menu item to add.
   * @param index The index of the menu item.
   *
-  * XXX: FIXME
-  *
   * @exception IllegalArgumentException If the index is less than zero.
   * @exception ArrayIndexOutOfBoundsException If the index is otherwise invalid.
   */
@@ -262,12 +263,26 @@
   if (index < 0)
     throw new IllegalArgumentException("Index is less than zero");
 
-  items.insertElementAt(item, index);
+  MenuPeer peer = (MenuPeer) getPeer();
+  if (peer == null)
+    return;
 
-  MenuPeer mp = (MenuPeer)getPeer();
-  // FIXME: Need to add a peer method here.
-//    if (mp != null)
-//      mp.insertItem(item, index);
+  int count = getItemCount ();
+
+  if (index >= count)
+    peer.addItem (item);
+  else
+    {
+      for (int i = count - 1; i >= index; i--)
+        peer.delItem (i);
+
+      peer.addItem (item);
+
+      for (int i = index; i < count; i++)
+        peer.addItem ((MenuItem) items.elementAt (i));
+    }
+
+  items.insertElementAt(item, index);
 }
 
 /*************************************************************************/
@@ -305,8 +320,6 @@
   *
   * @param index The index at which to insert a separator bar.
   *
-  * XXX: FIXME
-  *
   * @exception IllegalArgumentException If the index is less than zero.
   * @exception ArrayIndexOutOfBoundsException If the index is otherwise invalid.
   */
@@ -339,7 +352,7 @@
 
 /**
   * Removes the specifed item from the menu.  If the specified component
-  * does not exist, this method does nothing. // FIXME: Right?
+  * does not exist, this method does nothing.
   *
   * @param item The component to remove.
   */
@@ -415,7 +428,7 @@
 public String
 paramString()
 {
-  return (",isTearOff=" + isTearOff + ",isHelpMenu=" + isHelpMenu
+  return (",tearOff=" + tearOff + ",isHelpMenu=" + isHelpMenu
 	  + super.paramString());
 }
 
