--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2004-07-29 02:25:40.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-07-22 11:06:54.000000000 -0600
@@ -41,6 +41,7 @@
 import java.awt.Component;
 import java.awt.ComponentOrientation;
 import java.awt.Dimension;
+import java.awt.LayoutManager;
 import java.awt.Insets;
 import java.awt.LayoutManager;
 import java.awt.Point;
@@ -497,7 +498,7 @@
                       || vsb.getMaximum() != vs.height
                       || vsb.getValue() != vr.y
                       || vsb.getVisibleAmount() != vr.height))
-                  vsb.setValue(vr.y, vr.height, 0, vs.height);
+                vsb.setValue(vr.y, vr.height, 0, vs.height);
 
               if (hsb != null
                   && (hsb.getMinimum() != 0
