--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/NumberFormat.java	2004-06-21 06:24:01.000000000 -0600
+++ java/text/NumberFormat.java	2004-07-22 11:06:37.000000000 -0600
@@ -310,7 +310,7 @@
     try
       {
 	res = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation",
-		loc, ClassLoader.getSystemClassLoader());
+		loc, NumberFormat.class.getClassLoader());
       }
     catch (MissingResourceException x)
       {
