--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/AbstractButton.java	2004-08-10 02:36:08.000000000 -0600
+++ javax/swing/AbstractButton.java	2004-08-26 02:29:02.000000000 -0600
@@ -234,7 +234,7 @@
   /** The time in miliseconds in which clicks get coalesced into a single
    * <code>ActionEvent</code>. */
   long multiClickThreshhold;
-
+  
   /** Listener the button uses to receive PropertyChangeEvents from its
       Action. */
   PropertyChangeListener actionPropertyChangeListener;
@@ -299,9 +299,9 @@
   /** Fired in a PropertyChangeEvent when the "verticalTextPosition" property changes. */
   public static final String VERTICAL_TEXT_POSITION_CHANGED_PROPERTY = "verticalTextPosition";
 
-    /**
+  /**
    * A Java Accessibility extension of the AbstractButton.
-     */
+   */
   protected abstract class AccessibleAbstractButton
     extends AccessibleJComponent implements AccessibleAction, AccessibleValue,
                                             AccessibleText
@@ -433,7 +433,7 @@
     }
 
     public String getSelectedText()
-  {
+    {
       return null; // TODO
     }
 
@@ -443,33 +443,6 @@
     }
     }
 
-  /**
-   * Helper class used to subscribe to FocusEvents received by the button.
-   */
-  private class ButtonFocusListener implements FocusListener
-    {
-    /**
-     * Possibly repaint the model in response to loss of focus.
-     *
-     * @param event The loss-of-focus event
-     */
-    public void focusLost(FocusEvent event)
-        {
-      if (AbstractButton.this.isFocusPainted())
-        AbstractButton.this.repaint();
-    }
-
-    /**
-     * Possibly repaint the button in response to acquisition of focus.
-     *
-     * @param event The gained-focus event
-     */
-    public void focusGained(FocusEvent event)
-    {
-      if (AbstractButton.this.isFocusPainted())
-        AbstractButton.this.repaint();
-    }
-  }
 
   /**
    * Creates a new AbstractButton object.
@@ -552,12 +525,12 @@
     borderPainted = true;
     contentAreaFilled = true;
 
-    iconTextGap = 4;
+    focusPainted = true;
+    setFocusable(true);
 
     setAlignmentX(LEFT_ALIGNMENT);
     setAlignmentY(CENTER_ALIGNMENT);
 
-    addFocusListener(new ButtonFocusListener());
     setDisplayedMnemonicIndex(-1);    
  }
  
