--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/print/PrinterJob.java	2003-06-16 02:23:49.000000000 -0600
+++ java/awt/print/PrinterJob.java	2004-11-11 02:24:56.000000000 -0700
@@ -37,6 +37,7 @@
 
 
 package java.awt.print;
+import javax.print.attribute.PrintRequestAttributeSet;
 
 /**
   * This class controls printing.
@@ -194,6 +195,8 @@
   * Prints the pages.
   */
 public abstract void print () throws PrinterException;
+public abstract void print (PrintRequestAttributeSet attributes) throws PrinterException;
+
 
 /**
   * Displays a dialog box to the user which allows the print job
@@ -205,6 +208,9 @@
 public abstract boolean
 printDialog();
 
+public abstract boolean 
+printDialog(PrintRequestAttributeSet attributes);
+
 /*************************************************************************/
 
 /**
