--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicToolBarUI.java	2004-10-12 02:18:22.000000000 -0600
+++ javax/swing/plaf/basic/BasicToolBarUI.java	2004-08-12 02:37:23.000000000 -0600
@@ -1331,10 +1331,10 @@
   private static class ToolBarBorder implements Border
   {
     /** The size of the larger, draggable side of the border. */
-    private static final int offset = 10;
+    private static int offset = 10;
 
     /** The other sides. */
-    private static final int regular = 2;
+    private static int regular = 2;
 
     /**
      * This method returns the border insets for the JToolBar.
