--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/TableColumn.java	2004-08-01 02:25:30.000000000 -0600
+++ javax/swing/table/TableColumn.java	2004-09-14 02:22:06.000000000 -0600
@@ -1,5 +1,5 @@
 /* TableColumn.java --
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -505,6 +505,14 @@
   }
 
   /**
+   * @since 1.4
+   */
+  public PropertyChangeListener[] getPropertyChangeListeners()
+  {
+    return changeSupport.getPropertyChangeListeners();
+  }
+
+  /**
    * createDefaultHeaderRenderer
    * @return TableCellRenderer
    */
