--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ScrollPaneLayout.java	2004-10-28 02:27:45.000000000 -0600
+++ javax/swing/ScrollPaneLayout.java	2004-11-11 02:25:43.000000000 -0700
@@ -321,7 +321,6 @@
               insetsSize.setSize(insets.left + insets.right,
                                  insets.top + insets.bottom);
 
-            maybeSetMinimumSize(viewport, viewportSize);
             maybeSetMinimumSize(colHead, columnHeaderSize);
             maybeSetMinimumSize(rowHead, rowHeaderSize);
             
