--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JEditorPane.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/JEditorPane.java	2004-07-22 11:06:52.000000000 -0600
@@ -221,7 +221,7 @@
   }
 
   /**
-   * Establishes the default bindings of type to classname.  
+   * Establishes the default bindings of type to classname.
    */
   public static void registerEditorKitForContentType(String type,
                                                      String classname,
