--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/DefaultTableColumnModel.java	2004-09-05 02:26:55.000000000 -0600
+++ javax/swing/table/DefaultTableColumnModel.java	2004-09-14 02:22:06.000000000 -0600
@@ -204,7 +204,7 @@
    * @return int
    */
   public int getColumnIndexAtX(int x)
-  {
+  {    
     for (int i = 0; i < tableColumns.size(); ++i)
       {
         int w = ((TableColumn)tableColumns.get(i)).getWidth();
@@ -288,7 +288,7 @@
 
   /**
    * addColumnModelListener
-   * @param value0 TODO
+   * @param listener the listener to add
    */
   public void addColumnModelListener(TableColumnModelListener listener)
   {
@@ -297,14 +297,23 @@
 
   /**
    * removeColumnModelListener
-   * @param value0 TODO
+   * @param listener the listener to remove
    */
-  public void removeColumnModelListener(TableColumnModelListener value0)
+  public void removeColumnModelListener(TableColumnModelListener listener)
   {
-    // TODO
+    listenerList.remove(TableColumnModelListener.class, listener);
   }
 
   /**
+   * @since 1.4
+   */
+  public TableColumnModelListener[] getColumnModelListeners()
+  {
+    return (TableColumnModelListener[])
+      listenerList.getListeners(TableColumnModelListener.class);
+  }	  
+
+  /**
    * fireColumnAdded
    * @param value0 TODO
    */
