--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultDesktopManager.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/DefaultDesktopManager.java	2004-07-22 11:06:49.000000000 -0600
@@ -84,7 +84,7 @@
   private transient Container pane;
 
   /**
-   * An array of Rectangles that holds the bounds of the JDesktopIcons in the 
+   * An array of Rectangles that holds the bounds of the JDesktopIcons in the
    * JDesktopPane when looking for where to place a new icon.
    */
   private transient Rectangle[] iconRects;
