--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBox.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/JCheckBox.java	2004-07-22 11:06:51.000000000 -0600
@@ -48,7 +48,7 @@
 public class JCheckBox extends JToggleButton
 {
   private static final long serialVersionUID = -5246739313864538930L;
-  
+
   private boolean borderPaintedFlat;
 
   private void init()
@@ -57,20 +57,20 @@
     contentAreaFilled = false;
   }
   
-    public JCheckBox()
-    {
+  public JCheckBox()
+  {
     super();
     init();
-    }
+  }
 
   public JCheckBox(Action action)
-    {
+  {
     super(action);
     init();
-    }
+  }
 
-    public JCheckBox(Icon icon)
-    { 
+  public JCheckBox(Icon icon)
+  { 
     super(icon);
     init();
   }    
@@ -79,10 +79,10 @@
   { 
     super(icon, selected);
     init();
-    }    
+  }    
   
-    public JCheckBox(String text)
-    {
+  public JCheckBox(String text)
+  {
     super(text);
     init();
   }
@@ -91,41 +91,41 @@
   {
     super(text, selected);
     init();
-    }
+  }
       
-    public JCheckBox(String text, Icon icon)
-    {
-	super(text, icon);
+  public JCheckBox(String text, Icon icon)
+  {
+    super(text, icon);
     init();
-    }
+  }
 
   public JCheckBox(String text, Icon icon, boolean selected)
   {
     super(text, icon, selected);
     init();
   }
-    
+
   /**
    * Gets the AccessibleContext associated with this JCheckBox.
    */
-    public AccessibleContext getAccessibleContext()
-    {
-	return null;
-    }
+  public AccessibleContext getAccessibleContext()
+  {
+    return null;
+  }
   
   /**
    * Returns a string that specifies the name of the L&amp;F class
    * that renders this component.
    */
-    public String getUIClassID()
-    {
-	return "CheckBoxUI";
-    }
+  public String getUIClassID()
+  {
+    return "CheckBoxUI";
+  }
   
-    protected  String paramString()
-    {
-	return "JCheckBox";
-    }
+  protected  String paramString()
+  {
+    return "JCheckBox";
+  }
 
   public boolean isBorderPaintedFlat()
   {
