--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-11-25 02:28:55.000000000 -0700
+++ java/awt/Component.java	2004-11-25 02:24:12.000000000 -0700
@@ -1380,7 +1380,7 @@
             shouldRepaintSelf = parentBounds.intersects(newBounds);
           }
 
-        if (shouldRepaintParent)
+        if (shouldRepaintParent && parent != null)
           parent.repaint(oldx, oldy, oldwidth, oldheight);
         if (shouldRepaintSelf)
           repaint();
