--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/im/InputContext.java	2004-07-16 02:21:45.000000000 -0600
+++ java/awt/im/InputContext.java	2004-07-22 11:06:26.000000000 -0600
@@ -100,8 +100,8 @@
     while (e.hasMoreElements())
       {
         URL url = (URL) e.nextElement();
-        BufferedReader in = null;
-        String line = null;
+        BufferedReader in;
+        String line;
         try
           {
             in = new BufferedReader
