--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ShortViewBufferImpl.java	2004-06-21 06:24:00.000000000 -0600
+++ java/nio/ShortViewBufferImpl.java	2004-07-22 11:06:31.000000000 -0600
@@ -66,13 +66,6 @@
     this.endian = endian;
   }
 
-  /**
-   * Reads the <code>short</code> at this buffer's current position,
-   * and then increments the position.
-   *
-   * @exception BufferUnderflowException If there are no remaining
-   * <code>short</code>s in this buffer.
-   */
   public short get ()
   {
     int p = position();
@@ -81,13 +74,6 @@
     return result;
   }
 
-  /**
-   * Absolute get method. Reads the <code>short</code> at position
-   * <code>index</code>.
-   *
-   * @exception IndexOutOfBoundsException If index is negative or not smaller
-   * than the buffer's limit.
-   */
   public short get (int index)
   {
     return ByteBufferHelper.getShort(bb, (index << 1) + offset, endian);
