--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/DefaultFocusTraversalPolicy.java	2004-06-08 02:24:44.000000000 -0600
+++ java/awt/DefaultFocusTraversalPolicy.java	2004-08-26 02:28:58.000000000 -0600
@@ -92,19 +92,17 @@
 	&& comp.enabled)
       {
 	if (comp.isFocusTraversableOverridden != 0
-	    && comp.isFocusTraversable ())
-	  return true;
-	else
-	  {
-	    if (!(comp instanceof Canvas
-		  || comp instanceof Panel
-		  || comp instanceof Label
-		  || comp instanceof ScrollPane
-		  || comp instanceof Scrollbar
-		  || comp instanceof Window
-		  || comp.isLightweight ()))
-	      return true;
-	  }
+	    && (comp.isFocusTraversable () || comp.isFocusable()))
+          return true;
+        
+	if (!(comp instanceof Canvas
+              || comp instanceof Panel
+              || comp instanceof Label
+              || comp instanceof ScrollPane
+              || comp instanceof Scrollbar
+              || comp instanceof Window
+              || comp.isLightweight ()))
+          return true;
       }
     return false;
   }
