--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/geom/GeneralPath.java	2004-09-13 02:44:26.000000000 -0600
+++ java/awt/geom/GeneralPath.java	2004-08-09 02:22:44.000000000 -0600
@@ -793,11 +793,11 @@
 		y1 = firsty;
 
 		if (y0 == 0.0)
-		  y0 -= epsilon;
+		  y0 += epsilon;
 		if (y1 == 0.0)
-		  y1 -= epsilon;
-		if (Line2D.linesIntersect(x0, y0, x1, y1, 
-					  epsilon, 0.0, distance, 0.0))
+		  y1 += epsilon;
+		if (Line2D.linesIntersect(x0, y0, x1, y1, 0.0, 0.0, distance,
+		                          0.0))
 		  windingNumber += (y1 < y0) ? 1 : negative;
 
 		cx = firstx;
@@ -814,11 +814,10 @@
 	    y1 = firsty;
 
 	    if (y0 == 0.0)
-	      y0 -= epsilon;
+	      y0 += epsilon;
 	    if (y1 == 0.0)
-	      y1 -= epsilon;
-	    if (Line2D.linesIntersect(x0, y0, x1, y1, 
-				      epsilon, 0.0, distance, 0.0))
+	      y1 += epsilon;
+	    if (Line2D.linesIntersect(x0, y0, x1, y1, 0.0, 0.0, distance, 0.0))
 	      windingNumber += (y1 < y0) ? 1 : negative;
 
 	    cx = firstx;
@@ -833,11 +832,10 @@
 	    y1 = ypoints[pos++] - (float) y;
 
 	    if (y0 == 0.0)
-	      y0 -= epsilon;
+	      y0 += epsilon;
 	    if (y1 == 0.0)
-	      y1 -= epsilon;
-	    if (Line2D.linesIntersect(x0, y0, x1, y1, 
-				      epsilon, 0.0, distance, 0.0))
+	      y1 += epsilon;
+	    if (Line2D.linesIntersect(x0, y0, x1, y1, 0.0, 0.0, distance, 0.0))
 	      windingNumber += (y1 < y0) ? 1 : negative;
 
 	    cx = xpoints[pos - 1] - (float) x;
@@ -856,9 +854,9 @@
 	        && (y0 * y1 <= 0 || y1 * y2 <= 0))
 	      {
 		if (y0 == 0.0)
-		  y0 -= epsilon;
+		  y0 += epsilon;
 		if (y2 == 0.0)
-		  y2 -= epsilon;
+		  y2 += epsilon;
 
 		r[0] = y0;
 		r[1] = 2 * (y1 - y0);
@@ -899,9 +897,9 @@
 	        && (y0 * y1 <= 0 || y1 * y2 <= 0 || y2 * y3 <= 0))
 	      {
 		if (y0 == 0.0)
-		  y0 -= epsilon;
+		  y0 += epsilon;
 		if (y3 == 0.0)
-		  y3 -= epsilon;
+		  y3 += epsilon;
 
 		r[0] = y0;
 		r[1] = 3 * (y1 - y0);
@@ -944,4 +942,3 @@
     return (windingNumber);
   }
 } // class GeneralPath
-
