--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2004-07-23 02:24:09.000000000 -0600
+++ javax/swing/UIManager.java	2004-07-22 11:06:55.000000000 -0600
@@ -1,5 +1,5 @@
 /* UIManager.java -- 
-   Copyright (C) 2002, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003, 2004  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -92,7 +92,7 @@
    *
    * @param listener the listener to add
    */
-  public static void addPropertyChangeListener (PropertyChangeListener listener)
+  public static void addPropertyChangeListener(PropertyChangeListener listener)
   {
     // FIXME
   }
@@ -102,7 +102,7 @@
    *
    * @param listener the listener to remove
    */
-  public static void removePropertyChangeListener (PropertyChangeListener listener)
+  public static void removePropertyChangeListener(PropertyChangeListener listener)
   {
     // FIXME
   }
@@ -114,7 +114,7 @@
    *
    * @since 1.4
    */
-  public static PropertyChangeListener[] getPropertyChangeListeners ()
+  public static PropertyChangeListener[] getPropertyChangeListeners()
   {
     // FIXME
     throw new Error ("Not implemented");
@@ -209,7 +209,7 @@
    *
    * @param key an Object that specifies the font. Typically,
    *        this is a String such as
-   *        <code>TitledBorder.font</code>.
+   *        <code>&quot;TitledBorder.font&quot;</code>.
    */
   public static Font getFont(Object key)
   {
