--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/Timer.java	2004-08-01 02:25:28.000000000 -0600
+++ javax/swing/Timer.java	2004-08-26 02:29:06.000000000 -0600
@@ -172,7 +172,7 @@
         }
       catch (Exception e)
         {
-	  System.out.println("swing.Timer::" + e);
+//	  System.out.println("swing.Timer::" + e);
         }
     }
   }
@@ -374,7 +374,7 @@
   public void start()
   {
     if (isRunning())
-	return;
+      return;
     waker = new Waker();
     waker.start();
   }
@@ -394,7 +394,8 @@
   public void stop()
   {
     running = false;
-    waker.interrupt();
+    if (waker != null)
+      waker.interrupt();
     synchronized (queueLock)
       {
 	queue = 0;
