--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2004-07-31 02:23:38.000000000 -0600
@@ -881,7 +881,7 @@
       "ToolBar.font", new FontUIResource("Dialog", Font.PLAIN, 12),
       "ToolBar.foreground", new ColorUIResource(Color.black),
       "ToolBar.separatorSize", new DimensionUIResource(20, 20),
-      "ToolTip.background", new ColorUIResource(Color.white),
+      "ToolTip.background", new ColorUIResource(122, 178, 241),
       "ToolTip.border", new BorderUIResource.LineBorderUIResource(Color.lightGray),
       "ToolTip.font", new FontUIResource("SansSerif", Font.PLAIN, 12),
       "ToolTip.foreground", new ColorUIResource(Color.black),
