--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTable.java	2004-10-28 02:27:45.000000000 -0600
+++ javax/swing/JTable.java	2004-11-11 02:25:42.000000000 -0700
@@ -675,6 +675,11 @@
     return renderer;
   }
 
+  public void setDefaultRenderer(Class columnClass, TableCellRenderer rend)
+  {
+    defaultRenderersByColumnClass.put(columnClass, rend);
+  }
+
   public TableCellRenderer getDefaultRenderer(Class columnClass)
   {
     if (defaultRenderersByColumnClass.containsKey(columnClass))
