--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/SecurityManager.java	2004-06-08 02:25:04.000000000 -0600
+++ java/lang/SecurityManager.java	2004-07-22 11:06:29.000000000 -0600
@@ -45,8 +45,6 @@
 import java.lang.reflect.Member;
 import java.net.InetAddress;
 import java.net.SocketPermission;
-import java.security.AccessControlContext;
-import java.security.AccessController;
 import java.security.AllPermission;
 import java.security.Permission;
 import java.security.Security;
@@ -309,7 +307,8 @@
    */
   public Object getSecurityContext()
   {
-    return AccessController.getContext();
+    // XXX Should be: return AccessController.getContext();
+    return new SecurityContext(getClassContext());
   }
 
   /**
@@ -324,7 +323,8 @@
    */
   public void checkPermission(Permission perm)
   {
-    AccessController.checkPermission(perm);
+    // XXX Should be: AccessController.checkPermission(perm);
+    throw new SecurityException("Operation not allowed");
   }
 
   /**
@@ -345,9 +345,11 @@
    */
   public void checkPermission(Permission perm, Object context)
   {
-    if (! (context instanceof AccessControlContext))
-      throw new SecurityException("Missing context");
-    ((AccessControlContext) context).checkPermission(perm);
+    // XXX Should be:
+    // if (! (context instanceof AccessControlContext))
+    //   throw new SecurityException("Missing context");
+    // ((AccessControlContext) context).checkPermission(perm);
+    throw new SecurityException("Operation not allowed");
   }
 
   /**
@@ -546,10 +548,12 @@
    */
   public void checkRead(String filename, Object context)
   {
-    if (! (context instanceof AccessControlContext))
-      throw new SecurityException("Missing context");
-    AccessControlContext ac = (AccessControlContext) context;
-    ac.checkPermission(new FilePermission(filename, "read"));
+    // XXX Should be:
+    // if (! (context instanceof AccessControlContext))
+    //   throw new SecurityException("Missing context");
+    // AccessControlContext ac = (AccessControlContext) context;
+    // ac.checkPermission(new FilePermission(filename, "read"));
+    throw new SecurityException("Cannot read files via file names.");
   }
 
   /**
@@ -663,15 +667,17 @@
    */
   public void checkConnect(String host, int port, Object context)
   {
-    if (! (context instanceof AccessControlContext))
-      throw new SecurityException("Missing context");
-    AccessControlContext ac = (AccessControlContext) context;
-    if (port == -1)
-      ac.checkPermission(new SocketPermission(host, "resolve"));
-    else
-      // Use the toString() hack to do the null check.
-      ac.checkPermission(new SocketPermission(host.toString() + ":" + port,
-                                              "connect"));
+    // XXX Should be:
+    // if (! (context instanceof AccessControlContext))
+    //   throw new SecurityException("Missing context");
+    // AccessControlContext ac = (AccessControlContext) context;
+    // if (port == -1)
+    //   ac.checkPermission(new SocketPermission(host, "resolve"));
+    // else
+    //   // Use the toString() hack to do the null check.
+    //   ac.checkPermission(new SocketPermission(host.toString + ":" +port,
+    //                                           "connect"));
+    throw new SecurityException("Cannot make network connections.");
   }
 
   /**
