--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JDesktopPane.java	2004-10-28 02:27:43.000000000 -0600
+++ javax/swing/JDesktopPane.java	2004-11-11 02:25:41.000000000 -0700
@@ -154,7 +154,10 @@
 
     // FIXME: Unsupported mode.
     if (mode == OUTLINE_DRAG_MODE)
-      throw new IllegalArgumentException("Outline drag modes are unsupported.");
+      {
+        // throw new IllegalArgumentException("Outline drag modes are unsupported.");
+        mode = LIVE_DRAG_MODE;
+      }
 
     dragMode = mode;
   }
