--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/PopupMenu.java	2004-09-28 02:23:07.000000000 -0600
+++ java/awt/PopupMenu.java	2004-11-17 02:21:29.000000000 -0700
@@ -40,6 +40,9 @@
 
 import java.awt.peer.PopupMenuPeer;
 
+import javax.accessibility.AccessibleContext;
+import javax.accessibility.AccessibleRole;
+
 /**
   * This class implement an AWT popup menu widget
   *
@@ -135,5 +138,23 @@
     }
 }
 
+  protected class AccessibleAWTPopupMenu extends AccessibleAWTMenu
+  {
+    protected AccessibleAWTPopupMenu()
+    {
+    }
+    
+    public AccessibleRole getAccessibleRole()
+    {
+      return AccessibleRole.POPUP_MENU;
+    }
+    
+  }
+  
+  public AccessibleContext getAccessibleContext()
+  {
+    return new AccessibleAWTPopupMenu();
+  }
+
 } // class PopupMenu
 
