--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicPasswordFieldUI.java	2004-09-04 11:14:01.000000000 -0600
+++ javax/swing/plaf/basic/BasicPasswordFieldUI.java	2004-09-13 02:35:52.000000000 -0600
@@ -41,6 +41,7 @@
 import javax.swing.JComponent;
 import javax.swing.plaf.ComponentUI;
 import javax.swing.text.Element;
+import javax.swing.text.PasswordView;
 import javax.swing.text.View;
 
 public class BasicPasswordFieldUI extends BasicTextFieldUI
@@ -49,6 +50,11 @@
   {
   }
 
+  public View create(Element elem)
+  {
+    return new PasswordView(elem);
+  }
+  
   public static ComponentUI createUI(JComponent c)
   {
     return new BasicPasswordFieldUI();
