--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicRootPaneUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicRootPaneUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -49,17 +49,17 @@
 public class BasicRootPaneUI extends RootPaneUI
   implements PropertyChangeListener
 {
-    public static ComponentUI createUI(JComponent x) 
-    {
-        return new BasicRootPaneUI();
-    }
+  public static ComponentUI createUI(JComponent x) 
+  {
+    return new BasicRootPaneUI();
+  }
 
-    public void installUI(JComponent c)
-    {
-      c.setOpaque(true);
+  public void installUI(JComponent c)
+  {
+    c.setOpaque(true);
     c.setBackground(UIManager.getColor("control"));
-      super.installUI(c);
-    }
+    super.installUI(c);
+  }
 
   public void propertyChange(PropertyChangeEvent event)
   {
