--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JRootPane.java	2004-07-23 02:24:08.000000000 -0600
+++ javax/swing/JRootPane.java	2004-07-22 11:06:53.000000000 -0600
@@ -47,6 +47,7 @@
 import javax.accessibility.AccessibleRole;
 import javax.swing.plaf.RootPaneUI;
 
+
 /**
  * This class is where JComponents are added to. Unlike awt where you could
  * just say frame.add(), with swing you need to say frame.getRootPane()
@@ -96,7 +97,7 @@
     protected RootLayout()
     {
     }
-    
+
     /**
      * DOCUMENT ME!
      *
@@ -226,8 +227,8 @@
 	    {
 	      glassPane.setBounds(0, 0, containerSize.width,
 	                          containerSize.height);
-	    contentPane.setBounds(0, 0, containerSize.width,
-	                          containerSize.height);
+	      contentPane.setBounds(0, 0, containerSize.width,
+	                            containerSize.height);
 	    }
 
 	  layeredPane.setSize(containerSize.width, containerSize.height);
