--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicOptionPaneUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicOptionPaneUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -850,17 +850,17 @@
     addIcon(messageArea);
 
     JPanel rightSide = new JPanel()
-    {
-    public Dimension getPreferredSize()
-    {
+      {
+	public Dimension getPreferredSize()
+	{
 	  int w = Math.max(optionPane.getSize().width, minimumWidth);
-      Insets i = optionPane.getInsets();
-      Dimension orig = super.getPreferredSize();
-      Dimension value = new Dimension(w - i.left - i.right - iconSize,
-                                      orig.height);
-      return value;
-    }
-    };    
+	  Insets i = optionPane.getInsets();
+	  Dimension orig = super.getPreferredSize();
+	  Dimension value = new Dimension(w - i.left - i.right - iconSize,
+	                                  orig.height);
+	  return value;
+	}
+      };
     rightSide.setLayout(new GridBagLayout());
     GridBagConstraints con = createConstraints();
 
