--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/DirectColorModel.java	2003-06-28 02:19:15.000000000 -0600
+++ java/awt/image/DirectColorModel.java	2004-07-29 02:22:31.000000000 -0600
@@ -60,7 +60,6 @@
    * @param rmask the bits describing the red component of a pixel
    * @param gmask the bits describing the green component of a pixel
    * @param bmask the bits describing the blue component of a pixel 
-   * @param amask the bits describing the alpha component of a pixel 
    */
   public DirectColorModel(int pixelBits, int rmask, int gmask, int bmask)
   {
@@ -82,6 +81,7 @@
    * @param rmask the bits describing the red component of a pixel
    * @param gmask the bits describing the green component of a pixel
    * @param bmask the bits describing the blue component of a pixel 
+   * @param amask the bits describing the alpha component of a pixel 
    */
   public DirectColorModel(int pixelBits,
 			  int rmask, int gmask, int bmask, int amask)
