--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-09-04 02:25:04.000000000 -0600
@@ -1,5 +1,5 @@
 /* BasicGraphicsUtils.java
-   Copyright (C) 2003 Free Software Foundation, Inc.
+   Copyright (C) 2003, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -455,7 +455,9 @@
 
     drawUnderline = (underlinedIndex >= 0) && (underlinedIndex < textLength);
 
-    if (!(g instanceof Graphics2D))
+    // XXX - FIXME we now always use this fall-back since TextLayout is
+    // almost completely not implemented.
+    if (!(g instanceof Graphics2D) || true)
     {
       /* Fall-back. This is likely to produce garbage for any text
        * containing right-to-left (Hebrew or Arabic) characters, even
