--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/Utilities.java	2004-10-10 02:18:02.000000000 -0600
+++ javax/swing/text/Utilities.java	2004-11-11 02:25:55.000000000 -0700
@@ -98,15 +98,17 @@
     for (int offset = s.offset; offset < (s.offset + s.count); ++offset)
       {
         char c = buffer[offset];
-        if (len > 0 && (c == '\t' || c == '\n'))
+        if (c == '\t' || c == '\n')
           {
-	    g.drawChars(buffer, pos, len, pixelX, pixelY + ascent);            
-            pixelX += pixelWidth;
-            pixelWidth = 0;
+            if (len > 0) {
+              g.drawChars(buffer, pos, len, pixelX, pixelY + ascent);            
+              pixelX += pixelWidth;
+              pixelWidth = 0;
+            }
             pos = offset+1;
             len = 0;
           }
-          
+        
 	switch (c)
 	  {
 	  case '\t':
