--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSpinner.java	2004-09-13 02:44:30.000000000 -0600
+++ javax/swing/JSpinner.java	2004-09-13 02:35:40.000000000 -0600
@@ -114,11 +114,6 @@
                                                               LayoutManager
   {
     /**
-     * For compatability with Sun's JDK 1.4.2 rev. 5
-     */
-    private static final long serialVersionUID = -5317788736173368172L;
-
-    /**
      * Creates a new DefaultEditor object.
      *
      * @param spinner DOCUMENT ME!
@@ -220,11 +215,6 @@
   public static class NumberEditor extends DefaultEditor
   {
     /**
-     * For compatability with Sun's JDK
-     */
-    private static final long serialVersionUID = 3791956183098282942L;
-
-    /**
      * Creates a new NumberEditor object.
      *
      * @param spinner DOCUMENT ME!
@@ -347,6 +337,29 @@
   }
 
   /**
+   * Sets a new underlying model.
+   *
+   * @param newModel the new model to set
+   *
+   * @exception IllegalArgumentException if newModel is <code>null</code>
+   */
+  public void setModel(SpinnerModel newModel)
+  {
+    if (newModel == null)
+      throw new IllegalArgumentException();
+    
+    if (model == newModel)
+      return;
+
+    SpinnerModel oldModel = model;
+    model = newModel;
+    firePropertyChange("model", oldModel, newModel);
+
+    if (editor == null)
+      setEditor(createEditor(model));
+  }
+
+  /**
    * Gets the next value without changing the current value.
    *
    * @return the next value
