--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/GregorianCalendar.java	2004-06-21 06:24:02.000000000 -0600
+++ java/util/GregorianCalendar.java	2004-07-22 11:06:37.000000000 -0600
@@ -91,7 +91,7 @@
   private static ResourceBundle getBundle(Locale locale) 
   {
     return ResourceBundle.getBundle(bundleName, locale,
-      ClassLoader.getSystemClassLoader());
+      GregorianCalendar.class.getClassLoader());
   }
 
   /**
