--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2004-09-05 02:26:53.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-09-13 02:35:40.000000000 -0600
@@ -593,7 +593,6 @@
     return new JViewport();
   }
 
-
   public String getUIClassID()
   {
     return "ScrollPaneUI";
@@ -605,6 +604,25 @@
     setUI(b);
   }  
 
+  /**
+   * This method returns the scrollpane's UI delegate.
+   *
+   * @return The scrollpane's UI delegate.
+   */
+  public ScrollPaneUI getUI()
+  {
+    return (ScrollPaneUI) ui;
+  }
+
+  /**
+   * This method sets the scrollpane's UI delegate.
+   *
+   * @param ui The scrollpane's UI delegate.
+   */
+  public void setUI(ScrollPaneUI ui)
+  {
+    super.setUI(ui);
+  }
 
   class ScrollBar 
     extends JScrollBar
