--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/datatransfer/ClipboardOwner.java	2003-10-13 02:20:16.000000000 -0600
+++ java/awt/datatransfer/ClipboardOwner.java	2004-09-13 02:34:37.000000000 -0600
@@ -53,6 +53,5 @@
    * @param contents The contents of the clipboard which are no longer owned.
    */
   void lostOwnership (Clipboard clipboard, Transferable contents); 
-
-} // interface ClipboardOwner
+}
 
