--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/AbstractDocument.java	2004-07-23 02:24:14.000000000 -0600
+++ javax/swing/text/AbstractDocument.java	2004-07-22 11:06:57.000000000 -0600
@@ -133,7 +133,7 @@
   protected void fireUndoableEditUpdate(UndoableEditEvent event)
   {
     UndoableEditListener[] listeners = getUndoableEditListeners();
-    
+
     for (int index = 0; index < listeners.length; ++index)
       listeners[index].undoableEditHappened(event);
   }
@@ -199,9 +199,9 @@
   }
 
   public String getText(int offset, int length) throws BadLocationException
-      {
-	return content.getString(offset, length);
-      }
+  {
+    return content.getString(offset, length);
+  }
 
   public void getText(int offset, int length, Segment txt)
     throws BadLocationException
