--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBox.java	2004-08-01 02:25:28.000000000 -0600
+++ javax/swing/JCheckBox.java	2004-09-14 02:22:02.000000000 -0600
@@ -49,6 +49,9 @@
 {
   private static final long serialVersionUID = -5246739313864538930L;
 
+  public static final String BORDER_PAINTED_FLAT_CHANGED_PROPERTY =
+    "borderPaintedFlat";
+  
   private boolean borderPaintedFlat;
 
   private void init()
