--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicInternalFrameUI.java	2004-07-23 02:24:10.000000000 -0600
+++ javax/swing/plaf/basic/BasicInternalFrameUI.java	2004-07-31 02:23:38.000000000 -0600
@@ -765,7 +765,7 @@
 	                                         lastComponentEntered);
 	  MouseEvent exited = new MouseEvent(lastComponentEntered,
 	                                     MouseEvent.MOUSE_EXITED,
-	                                     me.getWhen(), me.getModifiers(),
+	                                     me.getWhen(), me.getModifiersEx(),
 	                                     tp.x, tp.y, me.getClickCount(),
 	                                     me.isPopupTrigger(),
 	                                     me.getButton());
@@ -787,7 +787,7 @@
 	      MouseEvent entered = new MouseEvent(lastComponentEntered,
 	                                          MouseEvent.MOUSE_ENTERED,
 	                                          me.getWhen(),
-	                                          me.getModifiers(), cp.x,
+	                                          me.getModifiersEx(), cp.x,
 	                                          cp.y, me.getClickCount(),
 	                                          me.isPopupTrigger(),
 	                                          me.getButton());
@@ -828,7 +828,7 @@
       if (e instanceof MouseEvent)
         {
 	  MouseEvent me = SwingUtilities.convertMouseEvent(frame.getRootPane()
-                                                                .getGlassPane(),
+	                                                        .getGlassPane(),
 	                                                   (MouseEvent) e,
 	                                                   frame.getRootPane()
 	                                                        .getGlassPane());
@@ -1109,7 +1109,7 @@
 
 	installDefaults();
 	installListeners();
-	installComponents();	
+	installComponents();
 	installKeyboardActions();
 
 	frame.setOpaque(true);
@@ -1126,7 +1126,7 @@
   public void uninstallUI(JComponent c)
   {
     uninstallKeyboardActions();
-    uninstallComponents();    
+    uninstallComponents();
     uninstallListeners();
     uninstallDefaults();
 
