--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBoxMenuItem.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/JCheckBoxMenuItem.java	2004-07-22 11:06:51.000000000 -0600
@@ -205,7 +205,7 @@
     * This method overrides JComponent.requestFocus with an empty
     * implementation, since JCheckBoxMenuItems should not
     * receve focus in general.
-   */
+    */
   public void requestFocus()
   {
     //  Should do nothing here
