--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/DefaultTableColumnModel.java	2004-09-05 02:26:55.000000000 -0600
+++ javax/swing/table/DefaultTableColumnModel.java	2004-08-26 02:29:09.000000000 -0600
@@ -204,7 +204,7 @@
    * @return int
    */
   public int getColumnIndexAtX(int x)
-  {
+  {    
     for (int i = 0; i < tableColumns.size(); ++i)
       {
         int w = ((TableColumn)tableColumns.get(i)).getWidth();
