--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JDesktopPane.java	2004-08-17 02:29:01.000000000 -0600
+++ javax/swing/JDesktopPane.java	2004-07-22 11:06:51.000000000 -0600
@@ -64,13 +64,13 @@
    * This specifies that when dragged, a JInternalFrame should be completely
    * visible.
    */
-  public static final int LIVE_DRAG_MODE = 0;
+  public static int LIVE_DRAG_MODE = 0;
 
   /**
    * This specifies that when dragged, a JInternalFrame should only be visible
    * as an outline.
    */
-  public static final int OUTLINE_DRAG_MODE = 1;
+  public static int OUTLINE_DRAG_MODE = 1;
 
   /** The selected frame in the JDesktopPane. */
   private transient JInternalFrame selectedFrame;
