--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/RepaintManager.java	2004-10-28 02:27:45.000000000 -0600
+++ javax/swing/RepaintManager.java	2004-11-11 02:25:43.000000000 -0700
@@ -309,6 +309,9 @@
   public synchronized void addDirtyRegion(JComponent component, int x, int y,
                                           int w, int h)
   {
+    if (w == 0 || h == 0)
+      return;
+
     Rectangle r = new Rectangle(x, y, w, h);
     if (dirtyComponents.containsKey(component))
       r = r.union((Rectangle)dirtyComponents.get(component));
