--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Panel.java	2004-07-23 02:23:50.000000000 -0600
+++ java/awt/Panel.java	2004-09-04 02:24:36.000000000 -0600
@@ -38,7 +38,6 @@
 
 package java.awt;
 
-import java.awt.event.PaintEvent;
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
@@ -98,36 +97,6 @@
   }
 
   /**
-   * Consume initial application-requested paint event if it has not
-   * already been consumed, and if the initial system-requested paint
-   * event has not already been handled.  Otherwise, call
-   * super.dispatchEventImpl.  These extra steps are required to
-   * prevent a Panel from being painted twice when it is initially
-   * shown.
-   *
-   * @param e the event to dispatch
-   */
-  void dispatchEventImpl (AWTEvent e)
-  {
-    if (e instanceof PaintEvent)
-      {
-        if (e.id == PaintEvent.UPDATE)
-          {
-            if (!initialUpdateConsumed
-                && !initialSystemUpdateDone)
-              {
-                e.consume ();
-                initialUpdateConsumed = true;
-              }
-          }
-        else if (e.id == PaintEvent.PAINT)
-          initialSystemUpdateDone = true;
-      }
-    else
-      super.dispatchEventImpl (e);
-  }
-
-  /**
    * Notifies this object to create its native peer.
    *
    * @see #isDisplayable()
