--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ComponentInputMap.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/ComponentInputMap.java	2004-07-22 11:06:49.000000000 -0600
@@ -39,17 +39,17 @@
 
 
 /**
- * @author	Andrew Selkirk
+ * @author Andrew Selkirk
  * @author Michael Koch
  */
 public class ComponentInputMap extends InputMap
 {
-	/**
+  /**
    * The component to notify.
-	 */
-	private JComponent component;
+   */
+  private JComponent component;
 
-	/**
+  /**
    * Creates <code>ComponentInputMap</code> object that notifies the given
    * component about changes to it.
    *
