--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocket.java	2004-07-17 02:21:34.000000000 -0600
+++ java/net/DatagramSocket.java	2004-07-22 11:06:30.000000000 -0600
@@ -1,6 +1,5 @@
 /* DatagramSocket.java -- A class to model UDP sockets
-   Copyright (C) 1998, 1999, 2000, 2002, 2003, 2004
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -283,7 +282,7 @@
    */
   public InetAddress getLocalAddress()
   {
-    if (! isBound())
+    if (isClosed())
       return null;
 
     InetAddress localAddr;
