--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/JTextComponent.java	2004-07-23 02:24:14.000000000 -0600
+++ javax/swing/text/JTextComponent.java	2004-07-22 11:06:57.000000000 -0600
@@ -386,7 +386,7 @@
   }
 
   /**
-   * Returns a string that specifies the name of the Look and Feel class
+   * Returns a string that specifies the name of the l&amp;f class
    * that renders this component.
    *
    * @return the string "TextComponentUI"
