--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PrintWriter.java	2004-07-21 02:19:57.000000000 -0600
+++ java/io/PrintWriter.java	2004-07-22 11:06:28.000000000 -0600
@@ -86,7 +86,7 @@
    */
   public PrintWriter(Writer wr)
   {
-    super(wr.lock);
+    super(wr);
     this.out = wr;
   }
 
@@ -102,7 +102,7 @@
    */
   public PrintWriter(Writer wr, boolean autoflush)
   {
-    super(wr.lock);
+    super(wr);
     this.out = wr;
     this.autoflush = autoflush;
   }
