--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Set.java	2004-07-30 02:31:35.000000000 -0600
+++ java/util/Set.java	2002-01-22 16:01:42.000000000 -0700
@@ -118,10 +118,6 @@
    *
    * @param o the object to look for
    * @return true if it is found in the set
-   * @throws ClassCastException if the type of o is not a valid type
-   *         for this set.
-   * @throws NullPointerException if o is null and this set doesn't
-   *         support null values.
    */
   boolean contains(Object o);
 
@@ -133,10 +129,6 @@
    * @param c the collection to check membership in
    * @return true if all elements in this set are in c
    * @throws NullPointerException if c is null
-   * @throws ClassCastException if the type of any element in c is not
-   *         a valid type for this set.
-   * @throws NullPointerException if some element of c is null and this
-   *         set doesn't support null values.
    * @see #contains(Object)
    */
   boolean containsAll(Collection c);
@@ -156,7 +148,6 @@
    * equals, this is the sum of the hashcode of all elements in the set.
    *
    * @return the sum of the hashcodes of all set elements
-   * @see #equals(Object)
    */
   int hashCode();
 
@@ -183,10 +174,6 @@
    * @param o the object to remove
    * @return true if the set changed (an object was removed)
    * @throws UnsupportedOperationException if this operation is not allowed
-   * @throws ClassCastException if the type of o is not a valid type
-   *         for this set.
-   * @throws NullPointerException if o is null and this set doesn't allow
-   *         the removal of a null value.
    */
   boolean remove(Object o);
 
@@ -199,10 +186,6 @@
    * @return true if this set changed as a result
    * @throws UnsupportedOperationException if this operation is not allowed
    * @throws NullPointerException if c is null
-   * @throws ClassCastException if the type of any element in c is not
-   *         a valid type for this set.
-   * @throws NullPointerException if some element of c is null and this
-   *         set doesn't support removing null values.
    * @see #remove(Object)
    */
   boolean removeAll(Collection c);
@@ -216,10 +199,6 @@
    * @return true if this set was modified
    * @throws UnsupportedOperationException if this operation is not allowed
    * @throws NullPointerException if c is null
-   * @throws ClassCastException if the type of any element in c is not
-   *         a valid type for this set.
-   * @throws NullPointerException if some element of c is null and this
-   *         set doesn't support retaining null values.
    * @see #remove(Object)
    */
   boolean retainAll(Collection c);
