--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ViewportLayout.java	2004-07-23 02:24:09.000000000 -0600
+++ javax/swing/ViewportLayout.java	2004-07-22 11:06:55.000000000 -0600
@@ -67,7 +67,7 @@
   {
     JViewport vp = (JViewport)parent;
     Component view = vp.getView();
-      return view.getPreferredSize();
+    return view.getPreferredSize();
   }
   public Dimension minimumLayoutSize(Container parent) 
   {
