--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/MemoryImageSource.java	2004-07-23 02:23:51.000000000 -0600
+++ java/awt/image/MemoryImageSource.java	2004-07-22 11:06:26.000000000 -0600
@@ -166,9 +166,9 @@
 	Vector list = (Vector) consumers.clone();
 	for(int i = 0; i < list.size(); i++) {
 	    ic = (ImageConsumer) list.elementAt(i);
-		sendPicture( ic );
+	    sendPicture( ic );
 	    ic.imageComplete( ImageConsumer.STATICIMAGEDONE );
-	    }	
+	}	
     }
 
     /**
@@ -217,7 +217,7 @@
 			ic = (ImageConsumer) list.elementAt(i);
 			sendPicture( ic );
 			ic.imageComplete( ImageConsumer.SINGLEFRAME );
-		    }	
+		}	
 	    }
     }
 
