--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DoubleBufferImpl.java	2004-06-21 06:23:59.000000000 -0600
+++ java/nio/DoubleBufferImpl.java	2004-07-22 11:06:31.000000000 -0600
@@ -1,5 +1,5 @@
 /* DoubleBufferImpl.java -- 
-   Copyright (C) 2002, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -98,16 +98,10 @@
   }
 
   /**
-   * Reads the <code>double</code> at this buffer's current position,
-   * and then increments the position.
-   *
-   * @exception BufferUnderflowException If there are no remaining
-   * <code>double</code>s in this buffer.
+   * Relative get method. Reads the next <code>double</code> from the buffer.
    */
   public double get ()
   {
-    checkForUnderflow();
-
     double result = backing_buffer [position ()];
     position (position () + 1);
     return result;
@@ -116,15 +110,13 @@
   /**
    * Relative put method. Writes <code>value</code> to the next position
    * in the buffer.
-   *
-   * @exception BufferOverflowException If there no remaining 
-   * space in this buffer.
+   * 
    * @exception ReadOnlyBufferException If this buffer is read-only.
    */
   public DoubleBuffer put (double value)
   {
-    checkIfReadOnly();
-    checkForOverflow();
+    if (readOnly)
+      throw new ReadOnlyBufferException ();
 	  	    
     backing_buffer [position ()] = value;
     position (position () + 1);
@@ -140,8 +132,6 @@
    */
   public double get (int index)
   {
-    checkIndex(index);
-
     return backing_buffer [index];
   }
   
@@ -155,9 +145,9 @@
    */
   public DoubleBuffer put (int index, double value)
   {
-    checkIfReadOnly();
-    checkIndex(index);
-
+    if (readOnly)
+      throw new ReadOnlyBufferException ();
+    	    
     backing_buffer [index] = value;
     return this;
   }
