--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicTextUI.java	2004-07-23 02:24:12.000000000 -0600
+++ javax/swing/plaf/basic/BasicTextUI.java	2004-07-22 11:06:57.000000000 -0600
@@ -85,21 +85,21 @@
     {
     }
   }
-
+  
   private class RootView extends View
   {
     private JTextComponent textComponent;
     private View view;
     
     public RootView(JTextComponent parent)
-      {
-          super(null);
+    {
+      super(null);
       textComponent = parent;
-      }
+    }
 
     public void setView(View v)
-      {
-          if (view != null)
+    {
+      if (view != null)
 	view.setParent(null);
       
       if (v != null)
@@ -109,17 +109,17 @@
     }
 
     public Container getContainer()
-              {
+    {
       return textComponent;
     }
-
+    
     public float getPreferredSpan(int axis)
     {
       if (view != null)
 	return view.getPreferredSpan(axis);
 
       return Integer.MAX_VALUE;
-              }
+    }
 
     public void paint(Graphics g, Shape s)
     {
@@ -127,7 +127,7 @@
       
       if (view != null)
 	view.paint(g, s);
-      }
+    }
   }
   
   RootView rootView;
@@ -153,7 +153,7 @@
   {
     return textComponent;
   }
-
+  
   public void installUI(final JComponent c)
   {
     super.installUI(c);
@@ -166,8 +166,8 @@
       {
 	doc = new PlainDocument();
 	textComponent.setDocument(doc);
-  }
-
+      }
+    
     rootView = new RootView(textComponent);
     setView(create(doc.getDefaultRootElement()));
     
@@ -221,7 +221,7 @@
 
     return new Dimension((int) w, (int) h);
   }
-
+  
   public final void paint(Graphics g, JComponent c)
   {
     paintSafely(g);
