--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/TableColumnModel.java	2004-07-23 02:24:13.000000000 -0600
+++ javax/swing/table/TableColumnModel.java	2004-07-22 11:06:57.000000000 -0600
@@ -48,118 +48,118 @@
  */
 public interface TableColumnModel
 {
-	/**
-	 * addColumn
-	 * @param column TableColumn
-	 */
-	void addColumn(TableColumn column);
-
-	/**
-	 * removeColumn
-	 * @param column TableColumn
-	 */
-	void removeColumn(TableColumn column);
-
-	/**
-	 * moveColumn
-	 * @param columnIndex Index of column to move
-	 * @param newIndex New index of column
-	 */
-	void moveColumn(int columnIndex, int newIndex);
-
-	/**
-	 * setColumnMargin
-	 * @param margin Margin of column
-	 */
-	void setColumnMargin(int margin);
+  /**
+   * addColumn
+   * @param column TableColumn
+   */
+  void addColumn(TableColumn column);
+
+  /**
+   * removeColumn
+   * @param column TableColumn
+   */
+  void removeColumn(TableColumn column);
+
+  /**
+   * moveColumn
+   * @param columnIndex Index of column to move
+   * @param newIndex New index of column
+   */
+  void moveColumn(int columnIndex, int newIndex);
+
+  /**
+   * setColumnMargin
+   * @param margin Margin of column
+   */
+  void setColumnMargin(int margin);
 
-	/**
-	 * getColumnCount
+  /**
+   * getColumnCount
    * @return Column count
-	 */
-	int getColumnCount();
+   */
+  int getColumnCount();
 
-	/**
-	 * getColumns
+  /**
+   * getColumns
    * @return Enumeration of columns
-	 */
-	Enumeration getColumns();
+   */
+  Enumeration getColumns();
 
-	/**
-	 * getColumnIndex
-	 * @param columnIdentifier Column id
-	 */
-	int getColumnIndex(Object columnIdentifier);
-
-	/**
-	 * getColumn
-	 * @param columnIndex Index of column
-	 */
-	TableColumn getColumn(int columnIndex);
+  /**
+   * getColumnIndex
+   * @param columnIdentifier Column id
+   */
+  int getColumnIndex(Object columnIdentifier);
+
+  /**
+   * getColumn
+   * @param columnIndex Index of column
+   */
+  TableColumn getColumn(int columnIndex);
 
-	/**
-	 * getColumnMargin
+  /**
+   * getColumnMargin
    * @return Column margin
-	 */
-	int getColumnMargin();
+   */
+  int getColumnMargin();
 
-	/**
-	 * getColumnIndexAtX
+  /**
+   * getColumnIndexAtX
    * @return Column index as position x
-	 */
-	int getColumnIndexAtX(int xPosition);
+   */
+  int getColumnIndexAtX(int xPosition);
 
-	/**
-	 * getTotalColumnWidth
+  /**
+   * getTotalColumnWidth
    * @return Total column width
-	 */
-	int getTotalColumnWidth();
+   */
+  int getTotalColumnWidth();
 
-	/**
-	 * setColumnSelectionAllowed
-	 * @param value Set column selection
-	 */
-	void setColumnSelectionAllowed(boolean value);
+  /**
+   * setColumnSelectionAllowed
+   * @param value Set column selection
+   */
+  void setColumnSelectionAllowed(boolean value);
 
-	/**
-	 * getColumnSelectionAllowed
+  /**
+   * getColumnSelectionAllowed
    * @return true if column selection allowed, false otherwise
-	 */
-	boolean getColumnSelectionAllowed();
+   */
+  boolean getColumnSelectionAllowed();
 
-	/**
-	 * getSelectedColumns
+  /**
+   * getSelectedColumns
    * @return Selected columns
-	 */
-	int[] getSelectedColumns();
+   */
+  int[] getSelectedColumns();
 
-	/**
-	 * getSelectedColumnCount
+  /**
+   * getSelectedColumnCount
    * @return Count of selected columns
-	 */
-	int getSelectedColumnCount();
+   */
+  int getSelectedColumnCount();
 
-	/**
-	 * setSelectionModel
-	 * @param model ListSelectionModel
-	 */
-	void setSelectionModel(ListSelectionModel model);
-
-	/**
-	 * getSelectionModel
-	 * @param column TableColumn
-	 */
-	ListSelectionModel getSelectionModel();
-
-	/**
-	 * addColumnModelListener
-	 * @param listener TableColumnModelListener
-	 */
-	void addColumnModelListener(TableColumnModelListener listener);
-
-	/**
-	 * removeColumnModelListener
-	 * @param listener TableColumnModelListener
-	 */
-	void removeColumnModelListener(TableColumnModelListener listener);
+  /**
+   * setSelectionModel
+   * @param model ListSelectionModel
+   */
+  void setSelectionModel(ListSelectionModel model);
+
+  /**
+   * getSelectionModel
+   * @param column TableColumn
+   */
+  ListSelectionModel getSelectionModel();
+
+  /**
+   * addColumnModelListener
+   * @param listener TableColumnModelListener
+   */
+  void addColumnModelListener(TableColumnModelListener listener);
+
+  /**
+   * removeColumnModelListener
+   * @param listener TableColumnModelListener
+   */
+  void removeColumnModelListener(TableColumnModelListener listener);
 }
