--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Double.java	2004-04-18 02:22:16.000000000 -0600
+++ java/lang/Double.java	2004-07-22 11:06:28.000000000 -0600
@@ -428,10 +428,9 @@
    * @return the bits of the <code>double</code>
    * @see #longBitsToDouble(long)
    */
-  public static long doubleToLongBits(double value)
-  {
-    return VMDouble.doubleToLongBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native long doubleToLongBits(double value);
+  // END GCJ LOCAL
 
   /**
    * Convert the double to the IEEE 754 floating-point "double format" bit
@@ -447,10 +446,9 @@
    * @return the bits of the <code>double</code>
    * @see #longBitsToDouble(long)
    */
-  public static long doubleToRawLongBits(double value)
-  {
-    return VMDouble.doubleToRawLongBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native long doubleToRawLongBits(double value);
+  // END GCJ LOCAL
 
   /**
    * Convert the argument in IEEE 754 floating-point "double format" bit
@@ -465,10 +463,9 @@
    * @see #doubleToLongBits(double)
    * @see #doubleToRawLongBits(double)
    */
-  public static double longBitsToDouble(long bits)
-  {
-    return VMDouble.longBitsToDouble(bits);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native double longBitsToDouble(long bits);
+  // END GCJ LOCAL
 
   /**
    * Compare two Doubles numerically by comparing their <code>double</code>
