--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicTabbedPaneUI.java	2004-07-23 02:24:12.000000000 -0600
+++ javax/swing/plaf/basic/BasicTabbedPaneUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -1162,8 +1162,7 @@
    * This is a helper class that implements UIResource so it is not added as a
    * tab when an object of this class is added to the JTabbedPane.
    */
-  private static class ScrollingButton extends BasicArrowButton
-    implements UIResource
+  private class ScrollingButton extends BasicArrowButton implements UIResource
   {
     /**
      * Creates a ScrollingButton given the direction.
