--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicInternalFrameTitlePane.java	2004-09-21 02:40:13.000000000 -0600
+++ javax/swing/plaf/basic/BasicInternalFrameTitlePane.java	2004-07-22 11:06:56.000000000 -0600
@@ -439,22 +439,22 @@
   }
 
   /** The action command for the Close action. */
-  protected static final String CLOSE_CMD = "Close";
+  protected static String CLOSE_CMD = "Close";
 
   /** The action command for the Minimize action. */
-  protected static final String ICONIFY_CMD = "Minimize";
+  protected static String ICONIFY_CMD = "Minimize";
 
   /** The action command for the Maximize action. */
-  protected static final String MAXIMIZE_CMD = "Maximize";
+  protected static String MAXIMIZE_CMD = "Maximize";
 
   /** The action command for the Move action. */
-  protected static final String MOVE_CMD = "Move";
+  protected static String MOVE_CMD = "Move";
 
   /** The action command for the Restore action. */
-  protected static final String RESTORE_CMD = "Restore";
+  protected static String RESTORE_CMD = "Restore";
 
   /** The action command for the Size action. */
-  protected static final String SIZE_CMD = "Size";
+  protected static String SIZE_CMD = "Size";
 
   /** The action associated with closing the JInternalFrame. */
   protected Action closeAction;
