--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/JTableHeader.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/table/JTableHeader.java	2004-07-22 11:06:57.000000000 -0600
@@ -60,4 +60,21 @@
   protected TableColumn resizingColumn;
   protected JTable table;
   protected boolean updateTableInRealTime;
+
+  public JTableHeader()
+  {
+    this(null);
+  }
+
+  public JTableHeader(TableColumnModel cm)
+  {
+    super();
+
+    this.columnModel = cm == null ? createDefaultColumnModel() : cm;
+  }
+
+  protected TableColumnModel createDefaultColumnModel()
+  {
+    return new DefaultTableColumnModel();
+  }
 }
