--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicMenuItemUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicMenuItemUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -37,6 +37,8 @@
 
 package javax.swing.plaf.basic;
 
+import java.awt.AWTKeyStroke;
+import java.awt.BasicStroke;
 import java.awt.Color;
 import java.awt.Component;
 import java.awt.Dimension;
@@ -68,6 +70,8 @@
 import javax.swing.SwingUtilities;
 import javax.swing.UIDefaults;
 import javax.swing.UIManager;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
 import javax.swing.event.MenuDragMouseEvent;
 import javax.swing.event.MenuDragMouseListener;
 import javax.swing.event.MenuKeyEvent;
@@ -630,9 +634,9 @@
 	                                               textRect.y
 	                                               + fm.getAscent());
 	else
-    BasicGraphicsUtils.drawString(g, text, 0, textRect.x,
-                                  textRect.y + fm.getAscent());
-  }
+	  BasicGraphicsUtils.drawString(g, text, 0, textRect.x,
+	                                textRect.y + fm.getAscent());
+      }
   }
 
   /**
@@ -771,7 +775,7 @@
     FontMetrics fm = g.getFontMetrics(acceleratorFont);
 
     if (menuItem.isEnabled())
-    g.setColor(acceleratorForeground);
+      g.setColor(acceleratorForeground);
     else
       // FIXME: should fix this to use 'disabledForeground', but its
       // default value in BasicLookAndFeel is null.
