--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicMenuUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicMenuUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -309,7 +309,7 @@
        if (i) if this menu is a submenu in some other menu
           (ii) or if this menu is in a menu bar and some other menu in a menu bar was just
                selected. (If nothing was selected, menu should be pressed before
-               it will be selected)      
+               it will be selected)
       */
       JMenu menu = (JMenu) menuItem;
       if (! menu.isTopLevelMenu()
@@ -338,15 +338,15 @@
       MenuSelectionManager manager = MenuSelectionManager.defaultManager();
       JMenu menu = (JMenu) menuItem;
       manager.processMouseEvent(e);
-      
+
       // Menu should be displayed when the menu is pressed only if 
       // it is top-level menu
       if (menu.isTopLevelMenu())
         {
 	  if (menu.getPopupMenu().isVisible())
-	      // If menu is visible and menu button was pressed.. 
-	      // then need to cancel the menu
-	      manager.clearSelectedPath();
+	    // If menu is visible and menu button was pressed.. 
+	    // then need to cancel the menu
+	    manager.clearSelectedPath();
 	  else
 	    {
 	      // Display the menu
