--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JMenu.java	2004-07-31 02:26:42.000000000 -0600
+++ javax/swing/JMenu.java	2004-09-04 02:25:01.000000000 -0600
@@ -140,7 +140,8 @@
    */
   public JMenu(String text, boolean tearoff)
   {
-    throw new Error("not implemented");
+    // FIXME: tearoff not implemented
+    this(text);
   }
 
   private void writeObject(ObjectOutputStream stream) throws IOException
