--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/DefaultHighlighter.java	2004-09-05 02:26:55.000000000 -0600
+++ javax/swing/text/DefaultHighlighter.java	2004-09-14 02:22:06.000000000 -0600
@@ -168,11 +168,22 @@
   
   private JTextComponent textComponent;
   private Vector highlights = new Vector();
+  private boolean drawsLayeredHighlights = true;
   
   public DefaultHighlighter()
   {
   }
 
+  public boolean getDrawsLayeredHighlights()
+  {
+    return drawsLayeredHighlights;
+  }
+
+  public void setDrawsLayeredHighlights(boolean newValue)
+  {
+    drawsLayeredHighlights = newValue;
+  }
+  
   private void checkPositions(int p0, int p1)
     throws BadLocationException
   {
