--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicScrollBarUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicScrollBarUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -534,18 +534,18 @@
 
   /** The Icon that points up. */
   private static Icon upIcon = new arrowIcon(new Polygon(new int[] { 2, 5, 8 },
-                                                            new int[] { 7, 3, 7 },
-                                                            3));
+                                                         new int[] { 7, 3, 7 },
+                                                         3));
 
   /** The Icon that points down. */
   private static Icon downIcon = new arrowIcon(new Polygon(new int[] { 2, 5, 8 },
-                                                              new int[] { 3, 7, 3 },
-                                                              3));
+                                                           new int[] { 3, 7, 3 },
+                                                           3));
 
   /** The Icon that points left. */
   private static Icon leftIcon = new arrowIcon(new Polygon(new int[] { 7, 3, 7 },
-                                                              new int[] { 2, 5, 8 },
-                                                              3));
+                                                           new int[] { 2, 5, 8 },
+                                                           3));
 
   /** The Icon that points right. */
   private static Icon rightIcon = new arrowIcon(new Polygon(new int[] { 3, 7, 3},
@@ -566,7 +566,7 @@
 
   /**
    * This method configures the scrollbar's colors. This can be 
-   * done by looking up the standard colors from the Look and Feel defaults.
+   * done by looking up the standard colors from the L&F defaults.
    */
   protected void configureScrollBarColors()
   {
@@ -602,7 +602,7 @@
   {
     if (incrButton == null)
       {
-      incrButton = new JButton();
+        incrButton = new JButton();      
         incrButton.setMargin(new Insets(0,0,0,0));
         incrButton.setHorizontalAlignment(SwingConstants.CENTER);
         incrButton.setHorizontalTextPosition(SwingConstants.CENTER);
@@ -630,7 +630,7 @@
   {
     if (decrButton == null)
       {
-      decrButton = new JButton();
+        decrButton = new JButton();
         decrButton.setMargin(new Insets(0,0,0,0));
         decrButton.setHorizontalAlignment(SwingConstants.CENTER);
         decrButton.setHorizontalTextPosition(SwingConstants.CENTER);
@@ -902,7 +902,7 @@
 
   /**
    * This method installs the defaults for the scrollbar specified
-   * by the Basic Look and Feel.
+   * by the Basic L&F.
    */
   protected void installDefaults()
   {
@@ -1268,7 +1268,7 @@
 
   /**
    * This method uninstalls any defaults that this
-   * scrollbar acquired from the Basic Look and Feel defaults.
+   * scrollbar acquired from the Basic L&F defaults.
    */
   protected void uninstallDefaults()
   {
