--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/LongBufferImpl.java	2004-06-21 06:24:00.000000000 -0600
+++ java/nio/LongBufferImpl.java	2004-07-22 11:06:31.000000000 -0600
@@ -1,5 +1,5 @@
 /* LongBufferImpl.java -- 
-   Copyright (C) 2002, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -98,16 +98,10 @@
   }
 
   /**
-   * Reads the <code>long</code> at this buffer's current position,
-   * and then increments the position.
-   *
-   * @exception BufferUnderflowException If there are no remaining
-   * <code>longs</code> in this buffer.
+   * Relative get method. Reads the next <code>long</code> from the buffer.
    */
   public long get ()
   {
-    checkForUnderflow();
-
     long result = backing_buffer [position ()];
     position (position () + 1);
     return result;
@@ -116,16 +110,14 @@
   /**
    * Relative put method. Writes <code>value</code> to the next position
    * in the buffer.
-   *
-   * @exception BufferOverflowException If there is insufficient space in this
-   * buffer.
+   * 
    * @exception ReadOnlyBufferException If this buffer is read-only.
    */
   public LongBuffer put (long value)
   {
-    checkIfReadOnly();
-    checkForOverflow();
-
+    if (readOnly)
+      throw new ReadOnlyBufferException ();
+	  	    
     backing_buffer [position ()] = value;
     position (position () + 1);
     return this;
@@ -140,8 +132,6 @@
    */
   public long get (int index)
   {
-    checkIndex(index);
-
     return backing_buffer [index];
   }
   
@@ -155,9 +145,9 @@
    */
   public LongBuffer put (int index, long value)
   {
-    checkIfReadOnly();
-    checkIndex(index);
-
+    if (readOnly)
+      throw new ReadOnlyBufferException ();
+    	    
     backing_buffer [index] = value;
     return this;
   }
