--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JRootPane.java	2004-09-13 02:44:30.000000000 -0600
+++ javax/swing/JRootPane.java	2004-09-24 02:23:51.000000000 -0600
@@ -309,6 +309,16 @@
   /** DOCUMENT ME! */
   private static final long serialVersionUID = 8690748000348575668L;
 
+  public static final int NONE = 0;
+  public static final int FRAME = 1;
+  public static final int PLAIN_DIALOG = 2;
+  public static final int INFORMATION_DIALOG = 3;
+  public static final int ERROR_DIALOG = 4;
+  public static final int COLOR_CHOOSER_DIALOG = 5;
+  public static final int FILE_CHOOSER_DIALOG = 6;
+  public static final int QUESTION_DIALOG = 7;
+  public static final int WARNING_DIALOG = 8;
+	  
   /** DOCUMENT ME! */
   protected Component glassPane;
 
@@ -321,6 +331,8 @@
   /** DOCUMENT ME! */
   protected Container contentPane;
 
+  protected JButton defaultButton;
+  
   /**
    * DOCUMENT ME!
    *
@@ -541,4 +553,19 @@
   {
     return "RootPaneUI";
   }
+
+  public JButton getDefaultButton()
+  {
+    return defaultButton;
+  }
+  
+  public void setDefaultButton(JButton newButton)
+  {
+    if (defaultButton == newButton)
+      return;
+    
+    JButton oldButton = defaultButton;
+    defaultButton = newButton;
+    firePropertyChange("defaultButton", oldButton, newButton);
+  }
 }
