--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/MenuSelectionManager.java	2004-07-23 02:24:09.000000000 -0600
+++ javax/swing/MenuSelectionManager.java	2004-07-22 11:06:54.000000000 -0600
@@ -272,10 +272,10 @@
 	MenuElement[] subComponents = ((MenuElement) mouseOverMenuComp)
 	                              .getSubElements();
 
-    for (int i = 0; i < subComponents.length; i++)
-      {
-	  subComponents[i].processMouseEvent(event, path, manager);
-      }
+	for (int i = 0; i < subComponents.length; i++)
+	 {
+	      subComponents[i].processMouseEvent(event, path, manager);
+	 }
 	*/
       }
   }
