--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/ColorConvertOp.java	2004-10-29 10:33:44.000000000 -0600
+++ java/awt/image/ColorConvertOp.java	2004-11-03 02:20:31.000000000 -0700
@@ -300,16 +300,16 @@
         RenderingHints.VALUE_COLOR_RENDER_QUALITY)
     {
       // use cie for accuracy
-      for (int y = src.getMinY(); y < src.getHeight() - src.getMinY(); y++)
-        for (int x = src.getMinX(); x < src.getWidth() - src.getMinX(); x++)
+      for (int y = src.getMinY(); y < src.getHeight() + src.getMinY(); y++)
+        for (int x = src.getMinX(); x < src.getWidth() + src.getMinX(); x++)
           dst.setPixel(x, y,
 		       dcs.fromCIEXYZ(scs.toCIEXYZ(src.getPixel(x, y, sbuf))));
     }
     else
     {
       // use rgb - it's probably faster
-      for (int y = src.getMinY(); y < src.getHeight() - src.getMinY(); y++)
-        for (int x = src.getMinX(); x < src.getWidth() - src.getMinX(); x++)
+      for (int y = src.getMinY(); y < src.getHeight() + src.getMinY(); y++)
+        for (int x = src.getMinX(); x < src.getWidth() + src.getMinX(); x++)
           dst.setPixel(x, y,
 		       dcs.fromRGB(scs.toRGB(src.getPixel(x, y, sbuf))));
     }
