--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormatSymbols.java	2004-06-21 06:24:01.000000000 -0600
+++ java/text/DecimalFormatSymbols.java	2004-07-22 11:06:36.000000000 -0600
@@ -130,7 +130,7 @@
     try
       {
 	res = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation",
-		loc, ClassLoader.getSystemClassLoader());
+		loc, DecimalFormatSymbols.class.getClassLoader());
       }
     catch (MissingResourceException x)
       {
