--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/AttributeSet.java	2004-07-23 02:24:14.000000000 -0600
+++ javax/swing/text/AttributeSet.java	2004-07-22 11:06:57.000000000 -0600
@@ -57,13 +57,13 @@
   {
   }
   
-    boolean containsAttribute(Object name, Object value);
-    boolean containsAttributes(AttributeSet attributes);
-    AttributeSet copyAttributes();
-    Object getAttribute(Object key);
-    int getAttributeCount();
-    Enumeration getAttributeNames();
-    AttributeSet getResolveParent();
-    boolean isDefined(Object attrName);
-    boolean isEqual(AttributeSet attr);     
+  boolean containsAttribute(Object name, Object value);
+  boolean containsAttributes(AttributeSet attributes);
+  AttributeSet copyAttributes();
+  Object getAttribute(Object key);
+  int getAttributeCount();
+  Enumeration getAttributeNames();
+  AttributeSet getResolveParent();
+  boolean isDefined(Object attrName);
+  boolean isEqual(AttributeSet attr);     
 }
