--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ImageIcon.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/ImageIcon.java	2004-07-22 11:06:50.000000000 -0600
@@ -1,4 +1,4 @@
-/* ImageIcon.java -- 
+/* ImageIcon.java --
    Copyright (C) 2002, 2004  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -57,7 +57,7 @@
   public ImageIcon()
   {
   }
-
+  
   public ImageIcon(String file)
   {
     this(file, file);
@@ -98,12 +98,12 @@
     this.image = Toolkit.getDefaultToolkit().createImage(image.getSource());
     this.description = description;
   }
-
+    
   public ImageObserver getImageObserver()
   {
     return observer;
   }
-
+  
   public void setImageObserver(ImageObserver newObserver)
   {
     observer = newObserver;
