--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/Position.java	2004-07-23 02:24:14.000000000 -0600
+++ javax/swing/text/Position.java	2004-07-22 11:06:57.000000000 -0600
@@ -40,8 +40,8 @@
 
 public interface Position
 {
-    static class Bias
-    {
+  static class Bias
+  {
     public static final Bias Backward = new Bias("backward");
     public static final Bias Forward = new Bias("forward");
 
@@ -56,7 +56,7 @@
     {
       return name;
     }
-    }
+  }
     
-    int getOffset();
+  int getOffset();
 }
