--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Panel.java	2004-10-12 02:18:11.000000000 -0600
+++ java/awt/Panel.java	2004-09-28 02:20:46.000000000 -0600
@@ -73,7 +73,7 @@
   /*
    * The number used to generate the name returned by getName.
    */
-  private static transient long next_panel_number;
+  private static transient long next_panel_number = 0;
 
   /**
    * Initializes a new instance of <code>Panel</code> that has a default
