--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/JTextComponent.java	2004-09-05 02:26:55.000000000 -0600
+++ javax/swing/text/JTextComponent.java	2004-08-26 02:29:09.000000000 -0600
@@ -557,7 +557,7 @@
   
   public static final String DEFAULT_KEYMAP = "default";
   public static final String FOCUS_ACCELERATOR_KEY = "focusAcceleratorKey";
-
+  
   private static Hashtable keymaps = new Hashtable();
   private Keymap keymap;
   
