--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ComboBoxModel.java	2003-10-13 02:22:32.000000000 -0600
+++ javax/swing/ComboBoxModel.java	2004-08-26 02:29:02.000000000 -0600
@@ -37,28 +37,32 @@
 
 package javax.swing;
 
+
 /**
- * ComboBoxModel
- * @author	Andrew Selkirk
- * @version	1.0
+ * ComboBoxModel is a data model for JComboBox. This model keeps
+ * track of elements contained in the JComboBox as well as the current
+ * combo box selection. Whenever selection in the JComboBox changes, the
+ * ComboBoxModel should fire ListDataEvents to ComboBox's ListDataListeners.
+ *
+ * @author        Andrew Selkirk
+ * @version        1.0
  */
-public interface ComboBoxModel extends ListModel {
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * setSelectedItem
-	 * @param item TODO
-	 */
-	void setSelectedItem(Object item);
-
-	/**
-	 * getSelectedItem
-	 * @returns Object
-	 */
-	Object getSelectedItem();
-
-
+public interface ComboBoxModel extends ListModel
+{
+  /**
+   * This method sets the selected item in the combo box. Class
+   * implementing this interface should fire ListDataEvents to
+   * all registered ListDataListeners to indicated that the
+   * selection has changed.
+   *
+   * @param item item in the combo box that should be selected
+   */
+  void setSelectedItem(Object item);
+
+  /**
+   * The method returns currently selected item in the combo box
+   *
+   * @returns item that is currently selected in the combo box.
+   */
+  Object getSelectedItem();
 } // ComboBoxModel
