--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicToolBarUI.java	2004-08-01 02:25:30.000000000 -0600
+++ javax/swing/plaf/basic/BasicToolBarUI.java	2004-08-12 02:37:23.000000000 -0600
@@ -157,10 +157,8 @@
 
   /**
    * This method creates a new BasicToolBarUI object for the given JToolBar.
-   *
-   * @param b The JToolBar to represent with this UI.
    */
-  public BasicToolBarUI(JToolBar b)
+  public BasicToolBarUI()
   {
     super();
   }
@@ -353,7 +351,7 @@
    */
   public static ComponentUI createUI(JComponent c)
   {
-    return new BasicToolBarUI((JToolBar) c);
+    return new BasicToolBarUI();
   }
 
   /**
