--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Graphics.java	2004-03-23 02:22:29.000000000 -0700
+++ java/awt/Graphics.java	2004-07-24 02:22:24.000000000 -0600
@@ -732,14 +732,14 @@
 /*************************************************************************/
 
 /**
-  * Returns a string representation of this object.
-  *
-  * @param A string representation of this object. 
-  */
+ * Returns a string representation of this object.
+ *
+ * @param A string representation of this object. 
+ */
 public String
 toString()
 {
-  return(super.toString());
+  return getClass ().getName () + "[font=" + getFont () + ",color=" + getColor () + "]";
 }
 
 public boolean
