--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicCheckBoxMenuItemUI.java	2004-07-23 02:24:10.000000000 -0600
+++ javax/swing/plaf/basic/BasicCheckBoxMenuItemUI.java	2004-07-22 11:06:55.000000000 -0600
@@ -97,7 +97,7 @@
    */
   public void processMouseEvent(JMenuItem item, MouseEvent e,
                                 MenuElement[] path,
-                         MenuSelectionManager manager)
+                                MenuSelectionManager manager)
   {
   }
 }
