--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JDialog.java	2004-10-12 02:18:19.000000000 -0600
+++ javax/swing/JDialog.java	2004-07-31 02:23:35.000000000 -0600
@@ -77,7 +77,7 @@
   private int close_action = HIDE_ON_CLOSE;
   
   /** Whether JDialogs are decorated by the Look and Feel. */
-  private static boolean decorated;
+  private static boolean decorated = false;
 
   /**
    * Creates a new non-modal JDialog with no title 
