--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicToggleButtonUI.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/plaf/basic/BasicToggleButtonUI.java	2004-08-13 02:30:58.000000000 -0600
@@ -43,12 +43,10 @@
 
 public class BasicToggleButtonUI extends BasicButtonUI
 {
-  
-    public static ComponentUI createUI(final JComponent c)  {
-	return new BasicToggleButtonUI();
-    }    
-}
-
-
+  public static ComponentUI createUI(final JComponent component)
+  {
+    return new BasicToggleButtonUI();
+  }    
 
+}
 
