--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/SinglePixelPackedSampleModel.java	2004-04-13 02:20:18.000000000 -0600
+++ java/awt/image/SinglePixelPackedSampleModel.java	2004-09-17 02:19:39.000000000 -0600
@@ -59,6 +59,16 @@
 				      int scanlineStride, int[] bitMasks)
   {
     super(dataType, w, h, bitMasks.length);
+
+    switch (dataType)
+      {
+      case DataBuffer.TYPE_BYTE:
+      case DataBuffer.TYPE_USHORT:
+      case DataBuffer.TYPE_INT:
+	break;
+      default:
+	throw new IllegalArgumentException("SinglePixelPackedSampleModel unsupported dataType");
+      }
     
     this.scanlineStride = scanlineStride;
     this.bitMasks = bitMasks;
