--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/ColorModel.java	2004-09-28 02:23:08.000000000 -0600
+++ java/awt/image/ColorModel.java	2004-10-28 02:23:42.000000000 -0600
@@ -43,6 +43,7 @@
 import java.awt.Point;
 import java.awt.Transparency;
 import java.awt.color.ColorSpace;
+import java.lang.reflect.Constructor;
 import java.util.Arrays;
 
 /**
@@ -158,6 +159,28 @@
     this.transferType = transferType;
   }
 
+  // This is a hook for ColorConvertOp to create a colormodel with
+  // a new colorspace
+  ColorModel cloneColorModel(ColorSpace cspace)
+  {
+    Class cls = this.getClass();
+    ColorModel cm;
+    try {
+      // This constructor will exist.
+      Constructor ctor =
+        cls.getConstructor(new Class[]{int.class, int[].class, ColorSpace.class,
+          boolean.class, boolean.class, int.class, int.class});
+      cm = (ColorModel)ctor.newInstance(new Object[]{new Integer(pixel_bits), bits,
+          cspace, new Boolean(hasAlpha), new Boolean(isAlphaPremultiplied),
+          new Integer(transparency), new Integer(transferType)});
+    }
+    catch (Exception e)
+    {
+      throw new IllegalArgumentException();
+    }
+    return cm;
+  }
+  
   public void finalize()
   {
     // Do nothing here.
