--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTextField.java	2004-07-23 02:24:08.000000000 -0600
+++ javax/swing/JTextField.java	2004-07-22 11:06:54.000000000 -0600
@@ -42,6 +42,7 @@
 import java.awt.FontMetrics;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+
 import javax.accessibility.AccessibleStateSet;
 import javax.swing.text.Document;
 import javax.swing.text.JTextComponent;
