--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JPopupMenu.java	2004-07-31 02:26:43.000000000 -0600
+++ javax/swing/JPopupMenu.java	2004-08-26 02:29:05.000000000 -0600
@@ -144,6 +144,9 @@
   /* Location of the popup */
   private Point popupLocation;
 
+  /* Field indicating if popup menu is visible or not */
+  private boolean visible = false;
+  
   /* Bound Property indicating visibility of the popup menu*/
   public static final String VISIBLE_CHANGED_PROPERTY = "visible";
 
@@ -564,7 +567,7 @@
    */
   public boolean isVisible()
   {
-    return super.isVisible();
+    return visible;
   }
 
   /**
@@ -577,7 +580,7 @@
   public void setVisible(boolean visible)
   {
     boolean old = isVisible();
-    super.setVisible(visible);
+    this.visible = visible;
     if (old != isVisible())
       {
 	firePropertyChange(VISIBLE_CHANGED_PROPERTY, old, (boolean) isVisible());
@@ -615,7 +618,7 @@
 		                            .getLayeredPane();
 		Point p = new Point(popupLocation.x, popupLocation.y);
 		SwingUtilities.convertPointFromScreen(p, layeredPane);
-		popup.show(p.x, p.y, size.width, size.height);
+		popup.show(p.x, p.y, size.width, size.height);  
 	      }
 	    else
 	      {
