--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Frame.java	2004-07-23 02:23:50.000000000 -0600
+++ java/awt/Frame.java	2004-07-22 11:06:24.000000000 -0600
@@ -197,9 +197,9 @@
    */
   private boolean undecorated = false;
 
-/*
+  /*
    * The number used to generate the name returned by getName.
- */
+   */
   private static transient long next_frame_number = 0;
 
 /**
@@ -408,13 +408,13 @@
   super.removeNotify();
 }
 
-/**
-  * Returns a debugging string describing this window.
-  *
-  * @return A debugging string describing this window.
-  */
+  /**
+   * Returns a debugging string describing this window.
+   *
+   * @return A debugging string describing this window.
+   */
   protected String paramString ()
-{
+  {
     String title = getTitle ();
 
     String resizable = "";
@@ -442,7 +442,7 @@
       }
 
     return super.paramString () + ",title=" + title + resizable + state;
-}
+  }
 
 public static Frame[]
 getFrames()
