--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicListUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicListUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -685,7 +685,7 @@
       {
         Rectangle bounds = getCellBounds(list, row, row);
         if (bounds.intersects(clip))
-        paintCell(g, row, bounds, render, model, sel, lead);
+          paintCell(g, row, bounds, render, model, sel, lead);
       }
   }
 
