--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Polygon.java	2004-09-13 02:44:25.000000000 -0600
+++ java/awt/Polygon.java	2004-07-22 11:06:24.000000000 -0600
@@ -684,7 +684,7 @@
    * path iterator is not either.
    *
    * @param transform an optional transform to apply to the iterator
-   * @param flatness the maximum distance for deviation from the real boundary
+   * @param double the maximum distance for deviation from the real boundary
    * @return a new iterator over the boundary
    * @since 1.2
    */
