--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/colorchooser/ColorChooserComponentFactory.java	2003-06-16 02:24:08.000000000 -0600
+++ javax/swing/colorchooser/ColorChooserComponentFactory.java	2004-08-26 02:29:06.000000000 -0600
@@ -35,49 +35,51 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package javax.swing.colorchooser;
 
 import javax.swing.JComponent;
 
+
 /**
  * ColorChooserComponentFactory
- * @author	Andrew Selkirk
- * @version	1.0
+ *
+ * @author Andrew Selkirk
+ * @version 1.0
  */
-public class ColorChooserComponentFactory {
-
-	//-------------------------------------------------------------
-	// Initialization ---------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * Constructor ColorChooserComponentFactory
-	 */
-	private ColorChooserComponentFactory() {
-		// TODO
-	} // ColorChooserComponentFactory()
-
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * getDefaultChooserPanels
-	 * @returns AbstractColorChooserPanel[]
-	 */
-	public static AbstractColorChooserPanel[] getDefaultChooserPanels() {
-		return null; // TODO
-	} // getDefaultChooserPanels()
-
-	/**
-	 * getPreviewPanel
-	 * @returns JComponent
-	 */
-	public static JComponent getPreviewPanel() {
-		return null; // TODO
-	} // getPreviewPanel()
-
-
+public class ColorChooserComponentFactory
+{
+  /**
+   * Constructor ColorChooserComponentFactory
+   */
+  private ColorChooserComponentFactory()
+  {
+  } // ColorChooserComponentFactory()
+
+  /**
+   * This method returns the three default chooser panels to be used in
+   * JColorChooser.
+   *
+   * @return The default chooser panels.
+   */
+  public static AbstractColorChooserPanel[] getDefaultChooserPanels()
+  {
+    AbstractColorChooserPanel[] values = 
+                                         {
+                                           new DefaultSwatchChooserPanel(),
+                                           new DefaultHSBChooserPanel(),
+                                           new DefaultRGBChooserPanel()
+                                         };
+    return values;
+  }
+
+  /**
+   * This method returns the default preview panel to be used with
+   * JColorChoosers.
+   *
+   * @return The default preview panel.
+   */
+  public static JComponent getPreviewPanel()
+  {
+    return new DefaultPreviewPanel();
+  } // getPreviewPanel()
 } // ColorChooserComponentFactory
