--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JPasswordField.java	2004-06-27 02:20:46.000000000 -0600
+++ javax/swing/JPasswordField.java	2004-09-13 02:35:39.000000000 -0600
@@ -44,6 +44,7 @@
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Document;
 
+
 /**
  * JPasswordField
  * @author	Andrew Selkirk
@@ -75,160 +76,160 @@
     }
   }
 
-	/**
-	 * uiClassID
-	 */
-	private static final String uiClassID = "PasswordFieldUI";
-
-	/**
-	 * echoChar.  Default is 0
-	 */
-	private char echoChar = 0;
-
-
-	//-------------------------------------------------------------
-	// Initialization ---------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * Constructor JPasswordField
-	 */
-	public JPasswordField() {
-		// TODO
-	} // JPasswordField()
-
-	/**
-	 * Constructor JPasswordField
-	 * @param text TODO
-	 */
-	public JPasswordField(String text) {
-		// TODO
-	} // JPasswordField()
-
-	/**
-	 * Constructor JPasswordField
-	 * @param columns TODO
-	 */
-	public JPasswordField(int columns) {
-		// TODO
-	} // JPasswordField()
-
-	/**
-	 * Constructor JPasswordField
-	 * @param text TODO
-	 * @param columns TODO
-	 */
-	public JPasswordField(String text, int columns) {
-		// TODO
-	} // JPasswordField()
-
-	/**
-	 * Constructor JPasswordField
-	 * @param document TODO
-	 * @param text TODO
-	 * @param columns TODO
-	 */
-	public JPasswordField(Document document, String text, int columns) {
-		// TODO
-	} // JPasswordField()
-
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * writeObject
-	 * @param stream TODO
-	 * @exception IOException TODO
-	 */
-	private void writeObject(ObjectOutputStream stream) throws IOException {
-		// TODO
-	} // writeObject()
-
-	/**
-	 * copy
-	 */
-	public void copy() {
-		// TODO
-	} // copy()
-
-	/**
-	 * getUIClassID
-	 * @returns String
-	 */
-	public String getUIClassID() {
-		return uiClassID;
-	} // getUIClassID()
-
-	/**
-	 * getEchoChar
-	 * @returns char
-	 */
-	public char getEchoChar() {
-		return echoChar;
-	} // getEchoChar()
-
-	/**
-	 * setEchoChar
-	 * @param echo TODO
-	 */
-	public void setEchoChar(char echo) {
-		this.echoChar = echo;
-		// TODO
-	} // setEchoChar()
-
-	/**
-	 * echoCharIsSet
-	 * @returns boolean
-	 */
-	public boolean echoCharIsSet() {
-		return (echoChar == 0);
-	} // echoCharIsSet()
-
-	/**
-	 * cut
-	 */
-	public void cut() {
-		// TODO
-	} // cut()
-
-	/**
-	 * getText
-	 * @returns String
-         * @deprecated
-	 */
-	public String getText() {
-		return null; // TODO
-	} // getText()
-
-	/**
-	 * getText
-	 * @param offset TODO
-	 * @param length TODO
-	 * @exception BadLocationException TODO
-	 * @returns String
-         * @deprecated
-	 */
-	public String getText(int offset, int length) throws BadLocationException {
-		return null; // TODO
-	} // getText()
-
-	/**
-	 * getPassword
-	 * @returns char[]
-	 */
-	public char[] getPassword() {
-		return null; // TODO
-	} // getPassword()
-
-	/**
-	 * paramString
-	 * @returns String
-	 */
-	protected String paramString() {
-		return null; // TODO
-	} // paramString()
+  /**
+   * echoChar.  Default is 0.
+   */
+  private char echoChar = 0;
+
+  /**
+   * Constructor JPasswordField
+   */
+  public JPasswordField()
+  {
+    this(null, null, 0);
+  }
+
+  /**
+   * Constructor JPasswordField
+   * @param text TODO
+   */
+  public JPasswordField(String text)
+  {
+    this(null, text, 0);
+  }
+
+  /**
+   * Constructor JPasswordField
+   * @param columns TODO
+   */
+  public JPasswordField(int columns)
+  {
+    this(null, null, columns);
+  }
+
+  /**
+   * Constructor JPasswordField
+   * @param text TODO
+   * @param columns TODO
+   */
+  public JPasswordField(String text, int columns)
+  {
+    this(null, text, columns);
+  }
+
+  /**
+   * Constructor JPasswordField
+   * @param document TODO
+   * @param text TODO
+   * @param columns TODO
+   */
+  public JPasswordField(Document document, String text, int columns)
+  {
+    super(document, text, columns);
+  }
+
+  /**
+   * writeObject
+   * @param stream TODO
+   * @exception IOException TODO
+   */
+  private void writeObject(ObjectOutputStream stream) throws IOException
+  {
+    // TODO
+  }
+
+  /**
+   * getUIClassID
+   * @return String
+   */
+  public String getUIClassID()
+  {
+    return "PasswordFieldUI";
+  }
+
+  /**
+   * getEchoChar
+   * @return char
+   */
+  public char getEchoChar()
+  {
+    return echoChar;
+  }
+
+  /**
+   * setEchoChar
+   * @param echo TODO
+   */
+  public void setEchoChar(char echo)
+  {
+    this.echoChar = echo;
+  }
+
+  /**
+   * echoCharIsSet
+   * @return boolean
+   */
+  public boolean echoCharIsSet()
+  {
+    return echoChar == 0;
+  }
+
+  /**
+   * copy
+   */
+  public void copy()
+  {
+    UIManager.getLookAndFeel().provideErrorFeedback(this);
+  }
+
+  /**
+   * cut
+   */
+  public void cut()
+  {
+    UIManager.getLookAndFeel().provideErrorFeedback(this);
+  }
+
+  /**
+   * getText
+   * @return String
+   * @deprecated
+   */
+  public String getText()
+  {
+    return null; // TODO
+  }
+
+  /**
+   * getText
+   * @param offset TODO
+   * @param length TODO
+   * @return String
+   * @exception BadLocationException TODO
+   * @deprecated
+   */
+  public String getText(int offset, int length) throws BadLocationException
+  {
+    return null; // TODO
+  }
+
+  /**
+   * getPassword
+   * @return char[]
+   */
+  public char[] getPassword()
+  {
+    return null; // TODO
+  }
+
+  /**
+   * paramString
+   * @return String
+   */
+  protected String paramString()
+  {
+    return null; // TODO
+  }
 
   /**
    * getAccessibleContext
