--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JDialog.java	2004-07-09 02:22:37.000000000 -0600
+++ javax/swing/JDialog.java	2004-07-22 11:06:52.000000000 -0600
@@ -76,7 +76,7 @@
   /** The default action taken when closed. */
   private int close_action = HIDE_ON_CLOSE;
   
-  /** Whether JDialogs are decorated by the Look and Feel. */
+  /** Whether JDialogs are decorated by the L&F. */
   private static boolean decorated = false;
 
   /**
@@ -237,7 +237,7 @@
    * This method returns whether JDialogs will have their
    * window decorations provided by the Look and Feel.
    *
-   * @return Whether the window decorations are Look and Feel provided.
+   * @return Whether the window decorations are L&F provided.
    */
   public static boolean isDefaultLookAndFeelDecorated()
   {
@@ -249,7 +249,7 @@
    * window decorations provided by the Look and Feel.
    *
    * @param defaultLookAndFeelDecorated Whether the window
-   * decorations are Look and Feel provided.
+   * decorations are L&F provided.
    */
   public static void setDefaultLookAndFeelDecorated(boolean defaultLookAndFeelDecorated)
   {
