--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Container.java	2004-07-23 02:23:50.000000000 -0600
+++ java/awt/Container.java	2004-07-31 02:23:19.000000000 -0600
@@ -1863,8 +1863,9 @@
       {
         candidate =
           SwingUtilities.getDeepestComponentAt(parent, p.x, p.y);
-        if (candidate == null)
+        if (candidate == null || (candidate.eventMask & me.getID()) == 0)
         {
+          candidate = null;
           p = SwingUtilities.convertPoint(parent, p.x, p.y, parent.parent);
           parent = parent.parent;
         }
@@ -1887,7 +1888,7 @@
         MouseEvent exited = new MouseEvent (lastComponentEntered, 
                                             MouseEvent.MOUSE_EXITED,
                                             me.getWhen (), 
-                                            me.getModifiers (), 
+                                            me.getModifiersEx (), 
                                             tp.x, tp.y,
                                             me.getClickCount (),
                                             me.isPopupTrigger (),
@@ -1911,7 +1912,7 @@
             MouseEvent entered = new MouseEvent (lastComponentEntered, 
                                                  MouseEvent.MOUSE_ENTERED,
                                                  me.getWhen (), 
-                                                 me.getModifiers (), 
+                                                 me.getModifiersEx (), 
                                                  cp.x, cp.y,
                                                  me.getClickCount (),
                                                  me.isPopupTrigger (),
