--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/Box.java	2004-07-09 02:22:37.000000000 -0600
+++ javax/swing/Box.java	2004-07-22 11:06:49.000000000 -0600
@@ -55,8 +55,9 @@
 {
   private static final long serialVersionUID = 1525417495883046342L;
   
+  protected class AccessibleBox 
   // FIXME: disable to make libjava compile; visibility rules are broken
-  protected class AccessibleBox // extends Container.AccessibleAWTContainer
+  // extends Container.AccessibleAWTContainer
   {
     private static final long serialVersionUID = -7775079816389931944L;
   
@@ -74,8 +75,9 @@
   {
     private static final long serialVersionUID = -1204263191910183998L;
   
+    protected class AccessibleBoxFiller 
     // FIXME: disable to make libjava compile; visibility rules are broken
-    protected class AccessibleBoxFiller // extends Component.AccessibleAWTComponent
+    // extends Component.AccessibleAWTComponent
     {
       private static final long serialVersionUID = 164963348357479321L;
       
