--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JViewport.java	2004-09-05 02:26:53.000000000 -0600
+++ javax/swing/JViewport.java	2004-09-13 02:35:42.000000000 -0600
@@ -253,9 +253,9 @@
       remove(0);
     if (v != null)
       {
-    add(v);
-    fireStateChanged();
-  }
+        add(v);
+        fireStateChanged();
+      }
   }
 
   public void revalidate()
@@ -305,23 +305,11 @@
     return false;
   }
 
-  public ChangeListener[] getChangeListeners() 
-  {
-    return (ChangeListener[]) getListeners(ChangeListener.class);
-  }
-
   public void paint(Graphics g)
   {
     paintComponent(g);
   }
 
-  void fireStateChanged()
-  {
-    ChangeListener[] listeners = getChangeListeners();
-    for (int i = 0; i < listeners.length; ++i)
-      listeners[i].stateChanged(changeEvent);
-  }
-
   public void addChangeListener(ChangeListener listener)
   {
     listenerList.add(ChangeListener.class, listener);
@@ -332,6 +320,18 @@
     listenerList.remove(ChangeListener.class, listener);
   }
 
+  public ChangeListener[] getChangeListeners() 
+  {
+    return (ChangeListener[]) getListeners(ChangeListener.class);
+  }
+
+  protected void fireStateChanged()
+  {
+    ChangeListener[] listeners = getChangeListeners();
+    for (int i = 0; i < listeners.length; ++i)
+      listeners[i].stateChanged(changeEvent);
+  }
+
   /**
    * This method returns the String ID of the UI class of  Separator.
    *
@@ -349,4 +349,24 @@
   {
     setUI((ViewportUI) UIManager.getUI(this));
   }            
+
+  /**
+   * This method returns the viewport's UI delegate.
+   *
+   * @return The viewport's UI delegate.
+   */
+  public ViewportUI getUI()
+  {
+    return (ViewportUI) ui;
+  }
+
+  /**
+   * This method sets the viewport's UI delegate.
+   *
+   * @param ui The viewport's UI delegate.
+   */
+  public void setUI(ViewportUI ui)
+  {
+    super.setUI(ui);
+  }
 }
