--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2004-08-15 02:27:46.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2004-08-31 02:30:09.000000000 -0600
@@ -41,6 +41,7 @@
 import java.awt.Color;
 import java.awt.Dimension;
 import java.awt.Font;
+import java.awt.event.InputEvent;
 import java.awt.event.KeyEvent;
 import java.io.Serializable;
 import java.util.Enumeration;
@@ -247,6 +248,7 @@
         "SPACE",  "pressed",
         "released SPACE", "released"
       }),
+      "Button.focus", midPurple,
       "Button.font", new FontUIResource("Dialog", Font.PLAIN, 12),
       "Button.foreground", new ColorUIResource(Color.black),
       "Button.highlight", new ColorUIResource(Color.white),
@@ -821,7 +823,14 @@
       "TextField.keyBindings", new JTextComponent.KeyBinding[] {
         new JTextComponent.KeyBinding(KeyStroke.getKeyStroke(KeyEvent.VK_ENTER,
                                                              0),
-                                      "notify-field-accept")},
+                                      "notify-field-accept"),
+        new JTextComponent.KeyBinding(KeyStroke.getKeyStroke(KeyEvent.VK_LEFT,
+							     InputEvent.SHIFT_DOWN_MASK),
+							     "selection-backward"),
+        new JTextComponent.KeyBinding(KeyStroke.getKeyStroke(KeyEvent.VK_RIGHT,
+							     InputEvent.SHIFT_DOWN_MASK),
+							     "selection-forward"),
+          },
       "TextField.margin", new InsetsUIResource(0, 0, 0, 0),
       "TextField.selectionBackground", new ColorUIResource(lightPurple),
       "TextField.selectionForeground", new ColorUIResource(Color.black),
