--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ScrollPaneLayout.java	2004-07-31 02:26:43.000000000 -0600
+++ javax/swing/ScrollPaneLayout.java	2004-08-26 02:29:06.000000000 -0600
@@ -82,6 +82,8 @@
 
   public void syncWithScrollPane(JScrollPane scrollPane) {
     viewport = scrollPane.getViewport();
+    rowHead = scrollPane.getRowHeader();
+    colHead = scrollPane.getColumnHeader();
     vsb = scrollPane.getVerticalScrollBar();
     hsb = scrollPane.getHorizontalScrollBar();
     vsbPolicy = scrollPane.getVerticalScrollBarPolicy();
@@ -389,12 +391,12 @@
             y4 = scrollPaneBounds.y + scrollPaneBounds.height;
             
             if (colHead != null)
-              y2 = colHead.getPreferredSize().height;
+              y2 = y1 + colHead.getPreferredSize().height;
             else
               y2 = y1;
 
             if (rowHead != null)
-              x2 = rowHead.getPreferredSize().width;
+              x2 = x1 + rowHead.getPreferredSize().width;
             else
               x2 = x1;
 
