--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JLabel.java	2004-09-05 02:26:53.000000000 -0600
+++ javax/swing/JLabel.java	2004-08-26 02:29:05.000000000 -0600
@@ -368,7 +368,7 @@
   public void setDisplayedMnemonic(int mnemonic)
   {
     if (displayedMnemonic != mnemonic)
-  {
+      {
 	firePropertyChange(DISPLAYED_MNEMONIC_CHANGED_PROPERTY,
 	                   displayedMnemonic, mnemonic);
 	displayedMnemonic = mnemonic;
@@ -386,7 +386,7 @@
    * @param mnemonic The character to use for the mnemonic.
    */
   public void setDisplayedMnemonic(char mnemonic)
-      {
+  {
     setDisplayedMnemonic((int) mnemonic);
   }
 
@@ -416,16 +416,16 @@
   {
     if (newIndex < -1 || (text != null && newIndex >= text.length()))
       throw new IllegalArgumentException();
-      
+
     if (text == null || text.charAt(newIndex) != displayedMnemonic)
       newIndex = -1;
-      
+
     if (newIndex != displayedMnemonicIndex)
-    {
-      firePropertyChange(DISPLAYED_MNEMONIC_INDEX_CHANGED_PROPERTY,
-                           displayedMnemonicIndex, newIndex);
-        displayedMnemonicIndex = newIndex;
-    }
+      {
+	firePropertyChange(DISPLAYED_MNEMONIC_INDEX_CHANGED_PROPERTY,
+	                   displayedMnemonicIndex, newIndex);
+	displayedMnemonicIndex = newIndex;
+      }
   }
 
   /**
