--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSpinner.java	2004-09-27 02:19:52.000000000 -0600
+++ javax/swing/JSpinner.java	2004-09-24 02:23:51.000000000 -0600
@@ -116,11 +116,6 @@
     private JSpinner spinner;
     
     /**
-     * For compatability with Sun's JDK 1.4.2 rev. 5
-     */
-    private static final long serialVersionUID = -5317788736173368172L;
-
-    /**
      * Creates a new <code>DefaultEditor</code> object.
      *
      * @param spinner the <code>JSpinner</code> associated with this editor
