--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/Raster.java	2004-11-07 02:18:24.000000000 -0700
+++ java/awt/image/Raster.java	2004-11-13 02:17:23.000000000 -0700
@@ -225,6 +225,19 @@
 				       bandMasks);
     return createWritableRaster(sm, dataBuffer, location);
   }
+  
+  public static WritableRaster
+  createPackedRaster(DataBuffer dataBuffer,
+		     int w, int h,
+		     int bitsPerPixel,
+		     Point location)
+  {
+    SampleModel sm =
+      new MultiPixelPackedSampleModel(dataBuffer.getDataType(),
+				       w, h,
+				       bitsPerPixel);
+    return createWritableRaster(sm, dataBuffer, location);
+  }
     
   public static Raster createRaster(SampleModel sm, DataBuffer db,
 				    Point location)
