--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/AttributeSet.java	2004-08-01 02:25:30.000000000 -0600
+++ javax/swing/text/AttributeSet.java	2004-08-11 02:26:03.000000000 -0600
@@ -56,7 +56,10 @@
   static interface ParagraphAttribute
   {
   }
-  
+
+  static Object NameAttribute = StyleConstants.NameAttribute;
+  static Object ResolveAttribute = StyleConstants.ResolveAttribute;
+
   boolean containsAttribute(Object name, Object value);
   boolean containsAttributes(AttributeSet attributes);
   AttributeSet copyAttributes();
