--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/GapContent.java	2004-06-27 02:20:48.000000000 -0600
+++ javax/swing/text/GapContent.java	2004-11-11 02:25:54.000000000 -0700
@@ -93,7 +93,7 @@
 
   public String getString(int where, int len) throws BadLocationException
   {
-    return buf.toString();
+    return buf.substring(where, where+len);
   }
 
   public void getChars(int where, int len, Segment txt)
