--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/geom/Rectangle2D.java	2004-09-13 02:44:26.000000000 -0600
+++ java/awt/geom/Rectangle2D.java	2003-09-26 02:18:35.000000000 -0600
@@ -368,7 +368,7 @@
    *
    * @param r the rectangle to add to this rectangle
    * @throws NullPointerException if r is null
-   * @see #union(Rectangle2D, Rectangle2D, Rectangle2D)
+   * @see #union(Rectangle2D)
    */
   public void add(Rectangle2D r)
   {
