--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicPopupMenuUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicPopupMenuUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -229,11 +229,11 @@
 
   /**
   * This method returns the minimum size of the JPopupMenu.
-   *
+  *
   * @param c The JComponent to find a size for.
-   *
+  *
   * @return The minimum size.
-   */
+  */
   public Dimension getMinimumSize(JComponent c)
   {
     return null;
