--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JButton.java	2004-07-31 02:26:42.000000000 -0600
+++ javax/swing/JButton.java	2004-07-31 02:23:35.000000000 -0600
@@ -99,8 +99,9 @@
 
   public String getUIClassID()
   {
-    //Returns a string that specifies the name of the L&F class that renders this component.  
-    return "ButtonUI";
+	//Returns a string that specifies the name of the Look and Feel
+	//class that renders this component.
+	return "ButtonUI";
   }
 
   public boolean isDefaultButton()
