--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2004-07-31 02:26:43.000000000 -0600
+++ javax/swing/UIManager.java	2004-07-31 02:23:37.000000000 -0600
@@ -278,7 +278,7 @@
   }
 
   /**
-   * Returns the L&amp;F object that renders the target component.
+   * Returns the Look and Feel object that renders the target component.
    */
   public static ComponentUI getUI(JComponent target)
   {
