--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/SpringLayout.java	2004-07-22 13:45:39.000000000 -0600
+++ javax/swing/SpringLayout.java	2004-09-13 02:35:44.000000000 -0600
@@ -91,7 +91,7 @@
    *
    * @author Roman Kennke (roman@ontographics.com)
    */
-  public final static class Constraints
+  public static class Constraints
   {
 
     // The constraints for each edge, and width and height.
