--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Button.java	2004-10-12 02:18:11.000000000 -0600
+++ java/awt/Button.java	2004-07-26 02:22:05.000000000 -0600
@@ -84,7 +84,7 @@
   /*
    * The number used to generate the name returned by getName.
    */
-  private static transient long next_button_number;
+  private static transient long next_button_number = 0;
 
 /*************************************************************************/
 
