--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicSplitPaneDivider.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicSplitPaneDivider.java	2004-07-22 11:06:56.000000000 -0600
@@ -156,7 +156,7 @@
   /** Keeps track of where the divider should be placed when using one touch expand
    * buttons. */
   private transient int currentDividerLocation = 1;
-
+  
   private transient Border tmpBorder = new Border()
   {
     public Insets getBorderInsets(Component c)
