--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2004-07-30 02:27:51.000000000 -0600
@@ -1,5 +1,5 @@
 /* BasicLookAndFeel.java --
-   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -228,7 +228,7 @@
   {
     Object[] uiDefaults;
     
-    // The default Look and Feel happens to use these three purple shades
+    // The JDK's default L&F happens to use these three purple shades
     // extensively.
     Color lightPurple = new Color(0xCC, 0xCC, 0xFF);
     Color midPurple = new Color(0x99, 0x99, 0xCC);
@@ -881,7 +881,7 @@
       "ToolBar.font", new FontUIResource("Dialog", Font.PLAIN, 12),
       "ToolBar.foreground", new ColorUIResource(Color.black),
       "ToolBar.separatorSize", new DimensionUIResource(20, 20),
-      "ToolTip.background", new ColorUIResource(Color.white),
+      "ToolTip.background", new ColorUIResource(122, 178, 241),
       "ToolTip.border", new BorderUIResource.LineBorderUIResource(Color.lightGray),
       "ToolTip.font", new FontUIResource("SansSerif", Font.PLAIN, 12),
       "ToolTip.foreground", new ColorUIResource(Color.black),
