--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLabelUI.java	2004-07-09 02:22:40.000000000 -0600
+++ javax/swing/plaf/basic/BasicLabelUI.java	2004-07-31 02:23:38.000000000 -0600
@@ -176,12 +176,12 @@
     if (icon != null)
       icon.paintIcon(b, g, ir.x, ir.y);
     if (text != null && ! text.equals(""))
-    {
-      if (b.isEnabled())
-        paintEnabledText(b, g, text, tr.x, tr.y + fm.getAscent());
-      else
-        paintDisabledText(b, g, text, tr.x, tr.y + fm.getAscent());
-    }
+      {
+	if (b.isEnabled())
+	  paintEnabledText(b, g, text, tr.x, tr.y + fm.getAscent());
+	else
+	  paintDisabledText(b, g, text, tr.x, tr.y + fm.getAscent());
+      }
     g.setFont(saved_font);
   }
 
