--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-09-13 02:44:25.000000000 -0600
+++ java/awt/Component.java	2004-09-04 02:24:34.000000000 -0600
@@ -293,7 +293,7 @@
    * @see #getLocale()
    * @see #setLocale(Locale)
    */
-  Locale locale;
+  Locale locale = Locale.getDefault ();
 
   /**
    * True if the object should ignore repaint events (usually because it is
@@ -4505,7 +4505,7 @@
    *
    * @return an AWT 1.0 event representing e
    */
-  static Event translateEvent (AWTEvent e)
+  private Event translateEvent (AWTEvent e)
   {
     Component target = (Component) e.getSource ();
     Event translated = null;
@@ -4728,8 +4728,8 @@
                     return;
               }
           }
-      processEvent (e);
-  }
+        processEvent (e);
+      }
   }
 
   /**
