--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTextArea.java	2004-09-27 02:19:52.000000000 -0600
+++ javax/swing/JTextArea.java	2004-09-27 02:18:10.000000000 -0600
@@ -195,15 +195,7 @@
    */
   public void append(String toAppend)
   {
-      try
-	  {
-	      getDocument().insertString(getText().length(), toAppend, null);
-	  }
-      catch (BadLocationException exception)
-	  {
-	      /* This shouldn't happen in theory -- but, if it does...  */
-	      throw new RuntimeException("Unexpected exception occurred.", exception);
-	  }
+    setText(getText() + toAppend);
   }
 
   /**
@@ -436,25 +428,18 @@
    */
   public void insert(String string, int position)
   {
-      // Retrieve the document model.
-      Document document = getDocument();
-      
-      // Check the model and string for validity.
-      if (document == null
-	  || string == null
-	  || string.length() == 0)
-	return;
-      
-      // Insert the text into the model.
-      try
-	{
-	  document.insertString(position, string, null);
-	}
-      catch (BadLocationException exception)
-	{
-	  throw new IllegalArgumentException("The supplied position, "
-					     + position + ", was invalid.");
-	}
+    if (position < doc.getStartPosition().getOffset()
+	|| position >= doc.getEndPosition().getOffset())
+      throw new IllegalArgumentException();
+
+    try
+      {
+	doc.insertString(position, string, null);
+      }
+    catch (BadLocationException e)
+      {
+	// This cannot happen as we check position above.
+      }
   }
 
   public void replaceRange(String text, int start, int end)
