--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ToolTipManager.java	2004-08-01 02:25:28.000000000 -0600
+++ javax/swing/ToolTipManager.java	2004-08-26 02:29:06.000000000 -0600
@@ -460,6 +460,8 @@
   public void mouseMoved(MouseEvent event)
   {
     currentPoint = event.getPoint();
+    if (currentTip != null)
+      currentTip.setTipText(((JComponent) currentComponent).getToolTipText(event));
     if (enterTimer.isRunning())
       enterTimer.restart();
   }
