--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/geom/GeneralPath.java	2004-10-30 02:23:12.000000000 -0600
+++ java/awt/geom/GeneralPath.java	2004-09-28 02:20:48.000000000 -0600
@@ -642,7 +642,7 @@
 	  if (transform != null)
 	    transform.transform( /* src */
 	    coords, /* srcOffset */
-	    0, /* dest */ coords, /* destOffset */
+	    pos, /* dest */ coords, /* destOffset */
 	    0, /* numPoints */ numCoords);
         }
       return seg;
