--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicRadioButtonMenuItemUI.java	2004-07-23 02:24:11.000000000 -0600
+++ javax/swing/plaf/basic/BasicRadioButtonMenuItemUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -65,11 +65,11 @@
   /**
     * Factory method to create a BasicRadioButtonMenuItemUI for the given {@link
     * JComponent}, which should be a JRadioButtonMenuItem.
-   *
+    *
     * @param b The {@link JComponent} a UI is being created for.
-   *
+    *
     * @return A BasicRadioButtonMenuItemUI for the {@link JComponent}.
-   */
+    */
   public static ComponentUI createUI(JComponent b)
   {
     return new BasicRadioButtonMenuItemUI();
@@ -95,7 +95,7 @@
    */
   public void processMouseEvent(JMenuItem item, MouseEvent e,
                                 MenuElement[] path,
-                         MenuSelectionManager manager)
+                                MenuSelectionManager manager)
   {
   }
 }
