--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JMenu.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/JMenu.java	2004-07-22 11:06:52.000000000 -0600
@@ -349,7 +349,7 @@
     // display popup associated with this menu	
     if (selected)
       {
-    super.setArmed(true);
+	super.setArmed(true);
 	super.setSelected(true);
 
 	// FIXME: The popup menu should be shown on the screen after certain
@@ -357,7 +357,7 @@
 	// this amount of seconds. 'delay' property is 0 by default.
 	if (this.isShowing())
 	  {
-    fireMenuSelected();
+	    fireMenuSelected();
 
 	    int x = 0;
 	    int y = 0;
