--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/RepaintManager.java	2004-07-23 02:24:09.000000000 -0600
+++ javax/swing/RepaintManager.java	2004-07-22 11:06:54.000000000 -0600
@@ -263,7 +263,7 @@
         && ancestor instanceof JComponent
         && ((JComponent) ancestor).isValidateRoot())
       component = (JComponent) ancestor;
-    
+
     if (invalidComponents.contains(component))
       return;
 
