--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JFrame.java	2004-10-12 02:18:19.000000000 -0600
+++ javax/swing/JFrame.java	2004-09-14 02:22:02.000000000 -0600
@@ -60,7 +60,7 @@
   implements WindowConstants, RootPaneContainer
 {
   private static final long serialVersionUID = -3362141868504252139L;
-  private static boolean defaultLookAndFeelDecorated;
+  private static boolean defaultLookAndFeelDecorated = false;
   private int close_action = HIDE_ON_CLOSE;
   protected AccessibleContext accessibleContext;
   protected JRootPane rootPane;
