--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicSplitPaneUI.java	2004-08-17 02:29:10.000000000 -0600
+++ javax/swing/plaf/basic/BasicSplitPaneUI.java	2004-07-22 11:06:56.000000000 -0600
@@ -892,8 +892,7 @@
    * The constraints object used when adding the non-continuous divider to the
    * JSplitPane.
    */
-  protected static final String NON_CONTINUOUS_DIVIDER
-    = "nonContinuousDivider";
+  protected static String NON_CONTINUOUS_DIVIDER;
 
   /** The dark divider used when dragging in non-continuous layout mode. */
   protected Component nonContinuousLayoutDivider;
