--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicTextUI.java	2004-09-05 02:26:55.000000000 -0600
+++ javax/swing/plaf/basic/BasicTextUI.java	2004-08-31 02:30:10.000000000 -0600
@@ -159,7 +159,7 @@
 	}
     }
   }
-
+  
   static EditorKit kit = new DefaultEditorKit();
 
   RootView rootView = new RootView();
@@ -254,7 +254,7 @@
   }
 
   protected void installKeyboardActions()
-  {
+  {    
     // load any bindings for the older Keymap interface
     Keymap km = JTextComponent.getKeymap(getKeymapName());
     if (km == null)
