--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/geom/PathIterator.java	2004-09-13 02:44:26.000000000 -0600
+++ java/awt/geom/PathIterator.java	2004-01-09 02:18:04.000000000 -0700
@@ -46,8 +46,8 @@
  *
  * @author Tom Tromey <tromey@cygnus.com>
  * @author Eric Blake <ebb9@email.byu.edu>
- * @see java.awt.Shape
- * @see java.awt.Stroke
+ * @see Shape
+ * @see Stroke
  * @see FlatteningPathIterator
  * @since 1.2
  * @status updated to 1.4
