--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/DefaultFocusTraversalPolicy.java	2004-09-05 02:26:44.000000000 -0600
+++ java/awt/DefaultFocusTraversalPolicy.java	2004-08-26 02:28:58.000000000 -0600
@@ -93,17 +93,17 @@
       {
 	if (comp.isFocusTraversableOverridden != 0
 	    && (comp.isFocusTraversable () || comp.isFocusable()))
-	  return true;
+          return true;
         
-	    if (!(comp instanceof Canvas
-		  || comp instanceof Panel
-		  || comp instanceof Label
-		  || comp instanceof ScrollPane
-		  || comp instanceof Scrollbar
-		  || comp instanceof Window
-		  || comp.isLightweight ()))
-	      return true;
-	  }
+	if (!(comp instanceof Canvas
+              || comp instanceof Panel
+              || comp instanceof Label
+              || comp instanceof ScrollPane
+              || comp instanceof Scrollbar
+              || comp instanceof Window
+              || comp.isLightweight ()))
+          return true;
+      }
     return false;
   }
 }
