--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Font.java	2004-07-23 02:23:50.000000000 -0600
+++ java/awt/Font.java	2004-07-24 02:22:24.000000000 -0600
@@ -1265,12 +1265,28 @@
 public String
 toString()
 {
-  return(getClass().getName() 
-         + "(logical=" + getName () 
-         + ",family=" + getFamily ()
-         + ",name=" + getFontName ()
-         + ",style=" + getStyle ()
-         + ",size=" + getSize ());
+  String styleString = "";
+
+  switch (getStyle ())
+    {
+    case 0:
+      styleString = "plain";
+      break;
+    case 1:
+      styleString = "bold";
+      break;
+    case 2:
+      styleString = "italic";
+      break;
+    default:
+      styleString = "unknown";
+    }
+
+  return getClass ().getName () 
+    + "[family=" + getFamily ()
+    + ",name=" + getFontName ()
+    + ",style=" + styleString
+    + ",size=" + getSize () + "]";
 }
 
 
