--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-07-23 02:23:50.000000000 -0600
+++ java/awt/Component.java	2004-07-22 11:06:24.000000000 -0600
@@ -2231,9 +2231,9 @@
    * calls {@link #postEvent}.
    *
    * @param e the event to deliver
-   * @deprecated use {@link #dispatchEvent(AWTEvent)} instead
+   * @deprecated use {@link #dispatchEvent (AWTEvent)} instead
    */
-  public void deliverEvent(Event e)
+  public void deliverEvent (Event e)
   {
     postEvent (e);
   }
@@ -2265,7 +2265,7 @@
    * @return the result of handling <code>e</code>
    * @deprecated use {@link #dispatchEvent(AWTEvent)} instead
    */
-  public boolean postEvent(Event e)
+  public boolean postEvent (Event e)
   {
     boolean handled = handleEvent (e);
 
@@ -3162,9 +3162,9 @@
    *
    * @param evt the event to handle
    * @return true if the event was handled, false otherwise
-   * @deprecated use {@link #processEvent(AWTEvent)} instead
+   * @deprecated use {@link #processEvent (AWTEvent)} instead
    */
-  public boolean handleEvent(Event evt)
+  public boolean handleEvent (Event evt)
   {
     switch (evt.id)
       {
@@ -3809,10 +3809,10 @@
                                                          currentFocusOwner));
                           }
                         else
-                    eq.postEvent (new FocusEvent(this, FocusEvent.FOCUS_GAINED, temporary));
+                          eq.postEvent (new FocusEvent(this, FocusEvent.FOCUS_GAINED, temporary));
+                      }
                   }
               }
-              }
             else
               // FIXME: need to add a focus listener to our top-level
               // ancestor, so that we can post this event when it becomes
@@ -3926,10 +3926,10 @@
                                                              currentFocusOwner));
                               }
                             else
-                        eq.postEvent (new FocusEvent(this, FocusEvent.FOCUS_GAINED, temporary));
+                              eq.postEvent (new FocusEvent(this, FocusEvent.FOCUS_GAINED, temporary));
+                          }
                       }
                   }
-                  }
                 else
                   return false;
               }
@@ -4673,7 +4673,7 @@
    *
    * @param e the event to dispatch
    */
-  void dispatchEventImpl(AWTEvent e)
+  void dispatchEventImpl (AWTEvent e)
   {
     Event oldEvent = translateEvent (e);
 
@@ -4681,7 +4681,7 @@
       postEvent (oldEvent);
 
     if (eventTypeEnabled (e.id))
-      processEvent(e);
+      processEvent (e);
   }
 
   /**
