--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIDefaults.java	2004-08-01 02:25:28.000000000 -0600
+++ javax/swing/UIDefaults.java	2004-09-13 02:35:44.000000000 -0600
@@ -477,7 +477,7 @@
     propertyChangeSupport.addPropertyChangeListener(listener);
   }
 
-  void removePropertyChangeListener(PropertyChangeListener listener)
+  public void removePropertyChangeListener(PropertyChangeListener listener)
   {
     propertyChangeSupport.removePropertyChangeListener(listener);
   }
