--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JFormattedTextField.java	2004-07-23 02:24:06.000000000 -0600
+++ javax/swing/JFormattedTextField.java	2004-07-22 11:06:52.000000000 -0600
@@ -135,7 +135,7 @@
   
   public JFormattedTextField ()
   {
-    throw new InternalError ("not implemented");
+    this((AbstractFormatterFactory) null);
   }
 
   public JFormattedTextField (Format format)
@@ -150,7 +150,7 @@
 
   public JFormattedTextField (AbstractFormatterFactory factory)
   {
-    throw new InternalError ("not implemented");
+    this(factory, null);
   }
 
   public JFormattedTextField (AbstractFormatterFactory factory, Object value)
