--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/Inet4Address.java	2004-04-30 02:23:42.000000000 -0600
+++ java/net/Inet4Address.java	2004-07-22 11:06:30.000000000 -0600
@@ -62,19 +62,7 @@
    */
   private Object writeReplace() throws ObjectStreamException
   {
-    return new InetAddress(addr, hostName, null);
-  }
-  
-  /**
-   * Initializes this object's addr instance variable from the passed in
-   * byte array. Note that this constructor is package-private and is called
-   * only by static methods in InetAddress.
-   * 
-   * @param addr
-   */
-  Inet4Address(byte[] addr)
-  {
-    this(addr, null, null);
+    return new InetAddress(addr, hostName);
   }
 
   /**
@@ -85,25 +73,10 @@
    */
   Inet4Address(byte[] addr, String host)
   {
-    this(addr, host, null);
+    super(addr, host);
   }
 
   /**
-   * Initializes this object's addr instance variable from the passed in
-   * byte array.  Note that this constructor is protected and is called
-   * only by static methods in this class.
-   *
-   * @param addr The IP number of this address as an array of bytes
-   * @param hostname The hostname of this IP address.
-   * @param hostname_alias A backup hostname to use if hostname is null to
-   * prevent reverse lookup failures
-   */
-  Inet4Address(byte[] addr, String hostname, String hostname_alias)
-  {
-    super(addr, hostname, hostname_alias);
-  }
-  
-  /**
    * Checks if the address is a multicast address
    *
    * @since 1.1
