--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Throwable.java	2004-05-21 02:26:21.000000000 -0600
+++ java/lang/Throwable.java	2004-07-22 11:06:29.000000000 -0600
@@ -1,5 +1,5 @@
 /* java.lang.Throwable -- Root class for all Exceptions and Errors
-   Copyright (C) 1998, 1999, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -396,21 +396,7 @@
     pw.print(stackTraceString());
   }
 
-  /*
-   * We use inner class to avoid a static initializer in this basic class.
-   */
-  private static class StaticData
-  {
-
-    private final static String nl;
-
-    static
-    {
-      // Access package private properties field to prevent Security check.
-      nl = System.properties.getProperty("line.separator");
-    }
-  }
-
+  private static final String nl = System.getProperty("line.separator");
   // Create whole stack trace in a stringbuffer so we don't have to print
   // it line by line. This prevents printing multiple stack traces from
   // different threads to get mixed up when written to the same PrintWriter.
@@ -463,7 +449,6 @@
   private static void stackTraceStringBuffer(StringBuffer sb, String name,
 					StackTraceElement[] stack, int equal)
   {
-    String nl = StaticData.nl;
     // (finish) first line
     sb.append(name);
     sb.append(nl);
