--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ComboBoxEditor.java	2004-01-10 02:20:47.000000000 -0700
+++ javax/swing/ComboBoxEditor.java	2004-08-26 02:29:02.000000000 -0600
@@ -42,49 +42,56 @@
 
 /**
  * ComboBoxEditor
- * @author	Andrew Selkirk
- * @version	1.0
+ *
+ * @author Andrew Selkirk
+ * @author Olga Rodimina
+ * @version 1.0
  */
-public interface ComboBoxEditor {
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * getEditorComponent
-	 * @returns Component
-	 */
-	Component getEditorComponent();
-
-	/**
-	 * setItem
-	 * @param item TODO
-	 */
-	void setItem(Object item);
-
-	/**
-	 * getItem
-	 * @returns Object
-	 */
-	Object getItem();
-
-	/**
-	 * selectAll
-	 */
-	void selectAll();
-
-	/**
-	 * addActionListener
-	 * @param listener TODO
-	 */
-	void addActionListener(ActionListener listener);
-
-	/**
-	 * removeActionListener
-	 * @param listener TODO
-	 */
-	void removeActionListener(ActionListener listener);
-
-
+public interface ComboBoxEditor
+{
+  /**
+   * This method returns component that will be used by the combo box to
+   * display/edit currently selected item in the combo box.
+   *
+   * @return Component that will be used by the combo box to display/edit
+   *         currently selected item
+   */
+  Component getEditorComponent();
+
+  /**
+   * Sets item that should be editted when any editting operation is performed
+   * by the user. The value is always equal to the currently selected value
+   * in the combo box. Thus, whenever a different value is selected from the
+   * combo box list then this method should be called to change editting item
+   * to the new selected item.
+   *
+   * @param selectedItem item that is currently selected in the combo box
+   */
+  void setItem(Object item);
+
+  /**
+   * This method returns item that is currently editable.
+   *
+   * @return Item in the combo box that is currently editable
+   */
+  Object getItem();
+
+  /**
+   * selectAll
+   */
+  void selectAll();
+
+  /**
+   * This method adds specified ActionListener to this ComboBoxEditor.
+   *
+   * @param listener
+   */
+  void addActionListener(ActionListener listener);
+
+  /**
+   * This method removes given ActionListener from this ComboBoxEditor.
+   *
+   * @param listener TODO
+   */
+  void removeActionListener(ActionListener listener);
 } // ComboBoxEditor
