--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/InputVerifier.java	2002-05-20 21:03:48.000000000 -0600
+++ javax/swing/InputVerifier.java	2004-09-14 02:22:02.000000000 -0600
@@ -1,5 +1,5 @@
 /* InputVerifier.java --
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,43 +37,39 @@
 
 package javax.swing;
 
+
 /**
  * InputVerifier
- * @author	Andrew Selkirk
- * @version	1.0
+ * @author Andrew Selkirk
+ * @version 1.0
  */
-public abstract class InputVerifier {
-
-	//-------------------------------------------------------------
-	// Initialization ---------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * Constructor InputVerifier
-	 */
-	public InputVerifier() {
-	} // InputVerifier()
-
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * verify
-	 * @param component TODO
-	 * @returns boolean
-	 */
-	public abstract boolean verify(JComponent component);
-
-	/**
-	 * shouldYieldFocus
-	 * @param component TODO
-	 * @returns boolean
-	 */
-	public boolean shouldYieldFocus(JComponent component) {
-		return verify(component);
-	} // shouldYieldFocus()
-
-
-} // InputVerifier
+public abstract class InputVerifier
+{
+  /**
+   * Creates a <code>InputVerifier</code>
+   */
+  public InputVerifier()
+  {
+  }
+
+  /**
+   * verify
+   *
+   * @param component the component to verify
+   *
+   * @return <code>true</code> if valid, <code>false</code> otherwise.
+   */
+  public abstract boolean verify(JComponent component);
+
+  /**
+   * shouldYieldFocus
+   *
+   * @param component the component to verify
+   *
+   * @return <code>true</code> if valid, <code>false</code> otherwise.
+   */
+  public boolean shouldYieldFocus(JComponent component)
+  {
+    return verify(component);
+  }
+}
