--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/Kernel.java	2004-08-07 02:29:22.000000000 -0600
+++ java/awt/image/Kernel.java	2004-07-13 22:46:52.000000000 -0600
@@ -117,7 +117,7 @@
     throws IllegalArgumentException
   {
     if (data == null)
-      return (float[])this.data.clone();
+	return (float[])this.data.clone();
 
     if (data.length < this.data.length)
       throw new IllegalArgumentException();
@@ -131,13 +131,6 @@
    */
   public Object clone()
   {
-    try
-      {
-	return super.clone();
-      }
-    catch (CloneNotSupportedException e)
-      {
-        throw (Error) new InternalError().initCause(e); // Impossible
-      }
+    return new Kernel(width, height, data);
   }
 }
