--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/FieldView.java	2004-07-22 13:45:39.000000000 -0600
+++ javax/swing/text/FieldView.java	2004-08-27 02:28:50.000000000 -0600
@@ -90,8 +90,14 @@
     return axis = axis == X_AXIS ? 1 : 0;
   }
   
+  public Shape modelToView(int pos, Shape a, Position.Bias bias)
+    throws BadLocationException
+  {
+    return super.modelToView(pos, a, bias);
+  }
+  
   public void paint(Graphics g, Shape s)
   {
-    drawLine(0, g, 0, 0);
+    super.paint(g, s);
   }
 }
