--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/SampleModel.java	2004-04-13 02:20:18.000000000 -0600
+++ java/awt/image/SampleModel.java	2004-11-03 02:20:31.000000000 -0700
@@ -456,6 +456,17 @@
 
     public abstract SampleModel createCompatibleSampleModel(int w, int h);
 
+    /**
+     * Return a SampleModel with a subset of the bands in this model.
+     * 
+     * Selects bands.length bands from this sample model.  The bands chosen
+     * are specified in the indices of bands[].  This also permits permuting
+     * the bands as well as taking a subset.  Thus, giving an array with
+     * 1, 2, 3, ..., numbands, will give an identical sample model.
+     * 
+     * @param bands Array with band indices to include.
+     * @return A new sample model
+     */
     public abstract SampleModel createSubsetSampleModel(int[] bands);
 
     public abstract DataBuffer createDataBuffer();
