--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/DefaultKeyboardFocusManager.java	2004-07-23 02:23:50.000000000 -0600
+++ java/awt/DefaultKeyboardFocusManager.java	2004-07-22 11:06:24.000000000 -0600
@@ -197,7 +197,7 @@
               setGlobalPermanentFocusOwner (null);
           }
 
-          target.dispatchEvent (e);
+        target.dispatchEvent (e);
 
         return true;
       }
@@ -219,7 +219,7 @@
         Component focusOwner = getGlobalPermanentFocusOwner ();
 
         if (focusOwner != null)
-        processKeyEvent (focusOwner, (KeyEvent) e);
+          processKeyEvent (focusOwner, (KeyEvent) e);
 
         if (e.isConsumed ())
           return true;
@@ -258,7 +258,7 @@
     Component focusOwner = getGlobalPermanentFocusOwner ();
 
     if (focusOwner != null)
-    focusOwner.dispatchEvent (e);
+      focusOwner.dispatchEvent (e);
 
     // Loop through all registered KeyEventPostProcessors, giving
     // each a chance to process this event.
