--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMainThread.c	2004-07-23 02:24:16.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMainThread.c	2004-08-01 02:22:55.000000000 -0600
@@ -265,14 +265,12 @@
       int int_dpi;
       g_object_get (settings, "gtk-xft-dpi", &int_dpi, NULL);
       /* If int_dpi == -1 gtk-xft-dpi returns the default value. So we
-	 have to do approximate calculation here.  */
-      if (int_dpi < 0)
-	dpi_conversion_factor = PANGO_SCALE * 72.0 / 96.;
-      else
-	dpi_conversion_factor = PANGO_SCALE * 72.0 / (int_dpi / PANGO_SCALE);
-
+         have to do approximate calculation here.  The value is 1024 *
+         dots.  We take 96 for dots.  */
+      if (int_dpi < 0)  int_dpi = 98304;
+      dpi_conversion_factor = PANGO_SCALE * 72.0 / (int_dpi / PANGO_SCALE);
       g_signal_connect (settings, "notify::gtk-xft-dpi",
-			G_CALLBACK (dpi_changed_cb), NULL);
+                        G_CALLBACK (dpi_changed_cb), NULL);
     }
   else
     /* Approximate. */
@@ -281,12 +279,9 @@
 
 static void
 dpi_changed_cb (GtkSettings  *settings,
-		GParamSpec *pspec __attribute__((unused)))
+                GParamSpec *pspec __attribute__((unused)))
 {
   int int_dpi;
   g_object_get (settings, "gtk-xft-dpi", &int_dpi, NULL);
-  if (int_dpi < 0)
-    dpi_conversion_factor = PANGO_SCALE * 72.0 / 96.;
-  else
-    dpi_conversion_factor = PANGO_SCALE * 72.0 / (int_dpi / PANGO_SCALE);
+  dpi_conversion_factor = PANGO_SCALE * 72.0 / (int_dpi / PANGO_SCALE);
 }
