--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuPeer.c	2004-08-01 02:25:30.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuPeer.c	2004-09-13 02:36:04.000000000 -0600
@@ -73,7 +73,6 @@
   GtkWidget *menu_title, *menu, *toplevel;
   const char *str;
 
-  /* Create global reference and save it for future use */
   NSA_SET_GLOBAL_REF (env, obj);
 
   str = (*env)->GetStringUTFChars (env, label, NULL);
