--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkGenericPeer.c	2004-10-30 02:23:23.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkGenericPeer.c	2004-09-04 02:25:06.000000000 -0600
@@ -39,8 +39,7 @@
 #include "gtkpeer.h"
 #include "gnu_java_awt_peer_gtk_GtkGenericPeer.h"
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkGenericPeer_dispose
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkGenericPeer_dispose
   (JNIEnv *env, jobject obj)
 {
   void *ptr;
