--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextFieldPeer.c	2004-08-01 02:25:30.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextFieldPeer.c	2004-09-04 02:25:07.000000000 -0600
@@ -175,7 +175,7 @@
 }
 
 JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkTextFieldPeer_gtkSetFont
+Java_gnu_java_awt_peer_gtk_GtkTextFieldPeer_gtkWidgetModifyFont
   (JNIEnv *env, jobject obj, jstring name, jint style, jint size)
 {
   const char *font_name;
