--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMainThread.c	2004-10-28 02:27:50.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMainThread.c	2004-10-10 02:17:14.000000000 -0600
@@ -76,7 +76,7 @@
 
 double dpi_conversion_factor;
 
-static void init_dpi_conversion_factor (void);
+static void init_dpi_conversion_factor ();
 static void dpi_changed_cb (GtkSettings  *settings,
                             GParamSpec   *pspec);
 
