--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextComponentPeer.c	2004-07-23 02:24:16.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextComponentPeer.c	2004-07-31 02:23:41.000000000 -0600
@@ -151,7 +151,6 @@
   GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextIter iter;
-  GtkTextMark *mark;
   GtkTextMark *oldmark;
   GtkTextIter olditer;
   int oldpos;
