--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkTextLayout.c	2004-10-28 02:27:50.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkTextLayout.c	2004-10-10 02:17:14.000000000 -0600
@@ -65,7 +65,7 @@
   gdk_threads_leave ();
 }
 
-JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkTextLayout_setText
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkGlyphVector_setText
   (JNIEnv *env, jobject self, jstring text)
 {
   struct textlayout *tl;
