--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkListPeer.c	2004-08-01 02:25:30.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkListPeer.c	2004-09-04 02:25:06.000000000 -0600
@@ -163,7 +163,7 @@
 }
 
 JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkListPeer_gtkSetFont
+Java_gnu_java_awt_peer_gtk_GtkListPeer_gtkWidgetModifyFont
   (JNIEnv *env, jobject obj, jstring name, jint style, jint size)
 {
   const char *font_name;
