--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkPixbufDecoder.c	2004-08-10 02:36:09.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkPixbufDecoder.c	2004-09-24 02:24:12.000000000 -0600
@@ -99,7 +99,7 @@
   JNIEnv *env;
   union env_union e;
   jint stride_bytes, stride_pixels, n_channels, n_pixels;
-  int i, px;
+  int i;
   jintArray jpixels;  
   jint *java_pixels;
   guchar *gdk_pixels;
@@ -129,21 +129,13 @@
 	  gdk_pixels + (y * stride_bytes), 
 	  (height * stride_bytes));
 
+#ifndef WORDS_BIGENDIAN
+  /* convert pixels from 0xBBGGRRAA to 0xAARRGGBB */
   for (i = 0; i < n_pixels; ++i)
     {
-      px = java_pixels[i];
-
-      /* move alpha around (GdkPixbufLoader results are AGBR not GBRA, in
-	 the lsb sense) */
-      /* px = ((px >> 24) & 0xff) | ((px << 8) & 0xffffff00); */
-
-      /* it appears to require a full byte swap, now, not just a shift to
-	 the A channel. why did this change? don't know. */
-      px = ((px >>  8) & 0x00ff00ff) | ((px <<  8) & 0xff00ff00); 
-      px = ((px >> 16) & 0x0000ffff) | ((px << 16) & 0xffff0000); 
-
-      java_pixels[i] = px;
+      java_pixels[i] = SWAPU32 ((unsigned)java_pixels[i]);
     }
+#endif
 
   g_object_unref (pixbuf);
 
