--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkFontPeer.c	2004-10-30 02:23:23.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkFontPeer.c	2004-10-28 02:24:26.000000000 -0600
@@ -69,15 +69,13 @@
   java_awt_font_HANGING_BASELINE = 2
 };
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkFontPeer_initStaticState 
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkFontPeer_initStaticState 
   (JNIEnv *env, jclass clazz)
 {
   NSA_FONT_INIT (env, clazz);
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkFontPeer_initState
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkFontPeer_initState
   (JNIEnv *env, jobject self)
 {
   struct peerfont *pfont = NULL;
@@ -90,8 +88,7 @@
 }
 
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkFontPeer_dispose
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkFontPeer_dispose
   (JNIEnv *env, jobject self)
 {
   struct peerfont *pfont = NULL;
@@ -111,8 +108,7 @@
   gdk_threads_leave ();
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkFontPeer_setFont
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkFontPeer_setFont
   (JNIEnv *env, jobject self, jstring family_name_str, jint style_int, jint size, jboolean useGraphics2D)
 {
   struct peerfont *pfont = NULL;
