--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-10-28 02:27:50.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-10-28 02:24:27.000000000 -0600
@@ -127,8 +127,8 @@
    This function extracts the filename from the GtkFileFilterInfo object,
    and passes it to the Java method.  The Java method will call the filter's
    accept() method and will give back the return value. */
-static gboolean filenameFilterCallback (const GtkFileFilterInfo *filter_info,
-					gpointer obj)
+gboolean filenameFilterCallback (const GtkFileFilterInfo *filter_info,
+                                 gpointer obj)
 {
   jclass cx;
   jmethodID id;
