--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuBarPeer.c	2004-04-18 02:22:35.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuBarPeer.c	2004-08-26 02:29:09.000000000 -0600
@@ -70,6 +70,29 @@
   gdk_threads_leave ();
 }
 
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkMenuBarPeer_nativeSetHelpMenu
+  (JNIEnv *env, jobject obj, jobject menupeer)
+{
+  static void *helpmenu;
+  void *mbar, *menu;
+  GList *list;
+
+  mbar = NSA_GET_PTR (env, obj);
+  menu = NSA_GET_PTR (env, menupeer);
+
+  gdk_threads_enter ();
+  if (helpmenu != NULL)
+    {
+      list = gtk_container_children (GTK_CONTAINER (mbar));
+      while (list != NULL && list->data != helpmenu)
+        list = list->next;
+      if (list != NULL && list->data == helpmenu)
+        gtk_container_remove (GTK_CONTAINER (mbar), GTK_WIDGET (list->data));
+    }
+  helpmenu = menu;
+  gdk_threads_leave ();
+}
+
 JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkMenuBarPeer_delMenu
   (JNIEnv *env, jobject obj, jint index)
 {
