--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkGlyphVector.c	2004-09-13 02:44:35.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkGlyphVector.c	2004-09-04 02:25:06.000000000 -0600
@@ -273,8 +273,7 @@
   
   pango_context_set_font_description (vec->ctx, vec->desc);
 
-  if (vec->glyphitems != NULL)
-    items = pango_itemize (vec->ctx, str, 0, len, attrs, NULL);
+  items = pango_itemize (vec->ctx, str, 0, len, attrs, NULL);
   if (items != NULL)
     {
       
@@ -304,15 +303,12 @@
 	  gi->glyphs = pango_glyph_string_new ();
 	  g_assert (gi->glyphs != NULL);
 	  
-	  if (gi->glyphs->num_glyphs > 0)
-	    {
-	      pango_shape (str + gi->item->offset, 
-			   gi->item->length, 
-			   &(gi->item->analysis), 
-			   gi->glyphs);
-	      
-	      vec->glyphitems = g_list_append (vec->glyphitems, gi);
-	    }
+	  pango_shape (str + gi->item->offset, 
+		       gi->item->length, 
+		       &(gi->item->analysis), 
+		       gi->glyphs);
+	  
+	  vec->glyphitems = g_list_append (vec->glyphitems, gi);
 	}
       
       /* 
