--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuItemPeer.c	2004-10-30 02:23:23.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuItemPeer.c	2004-09-13 02:36:03.000000000 -0600
@@ -43,8 +43,7 @@
 static void item_activate (GtkMenuItem *item __attribute__((unused)),
                            jobject peer_obj);
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkMenuItemPeer_create
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkMenuItemPeer_create
   (JNIEnv *env, jobject obj, jstring label)
 {
   GtkWidget *widget;
@@ -125,8 +124,7 @@
   (*env)->ReleaseStringUTFChars (env, name, font_name);
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkMenuItemPeer_setEnabled
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkMenuItemPeer_setEnabled
   (JNIEnv *env, jobject obj, jboolean enabled)
 {
   void *ptr;
@@ -138,8 +136,7 @@
   gdk_threads_leave ();
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkMenuItemPeer_setLabel
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkMenuItemPeer_setLabel
   (JNIEnv *env, jobject obj, jstring label)
 {
   void *ptr;
