--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkGraphics2D.c	2004-08-15 02:27:48.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkGraphics2D.c	2004-09-04 02:25:06.000000000 -0600
@@ -159,18 +159,7 @@
   g_assert (draw != NULL);
   g_assert (win != NULL);
 
-  if (GTK_IS_WINDOW (widget))
-    {
-      *win = find_gtk_layout (widget)->bin_window;
-    }
-  else if (GTK_IS_LAYOUT (widget))
-    {
-      *win = GTK_LAYOUT (widget)->bin_window;
-    }
-  else
-    {
-      *win = widget->window;
-    }
+  *win = widget->window;
 
   *draw = *win;
   gdk_window_get_internal_paint_info (*win, draw, 0, 0); 
