--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextAreaPeer.c	2004-06-27 02:20:49.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextAreaPeer.c	2004-09-04 02:25:07.000000000 -0600
@@ -141,7 +141,7 @@
 }
 
 JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkTextAreaPeer_gtkSetFont
+Java_gnu_java_awt_peer_gtk_GtkTextAreaPeer_gtkWidgetModifyFont
   (JNIEnv *env, jobject obj, jstring name, jint style, jint size)
 {
   const char *font_name;
