--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer.c	2004-10-30 02:23:23.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer.c	2004-07-22 11:06:59.000000000 -0600
@@ -39,8 +39,7 @@
 #include "gtkpeer.h"
 #include "gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer.h"
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer_dispose
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer_dispose
   (JNIEnv *env, jobject obj)
 {
   /* The actual underlying widget is owned by a different class.  So
@@ -48,8 +47,7 @@
   NSA_DEL_PTR (env, obj);
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer_remove
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkCheckboxGroupPeer_remove
   (JNIEnv *env, jobject obj, jobject checkbox)
 {
   GtkRadioButton *button;
