--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkTextLayout.c	2004-10-30 02:23:23.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkTextLayout.c	2004-10-10 02:17:14.000000000 -0600
@@ -44,15 +44,13 @@
 
 struct state_table *native_text_layout_state_table;
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkTextLayout_initStaticState 
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkTextLayout_initStaticState 
   (JNIEnv *env, jclass clazz)
 {
   NSA_TEXT_LAYOUT_INIT (env, clazz);
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkTextLayout_initState
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkTextLayout_initState
   (JNIEnv *env, jobject self)
 {
   struct textlayout *tl;
@@ -67,8 +65,7 @@
   gdk_threads_leave ();
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkTextLayout_setText
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkGlyphVector_setText
   (JNIEnv *env, jobject self, jstring text)
 {
   struct textlayout *tl;
@@ -93,8 +90,7 @@
   gdk_threads_leave ();  
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkTextLayout_indexToPos
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkTextLayout_indexToPos
   (JNIEnv *env, jobject self, jint idx, jdoubleArray javaPos)
 {
   struct textlayout *tl;
@@ -124,8 +120,7 @@
   gdk_threads_leave ();  
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkTextLayout_getExtents
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkTextLayout_getExtents
   (JNIEnv *env, jobject self, jdoubleArray javaInkExtents, jdoubleArray javaLogExtents)
 {
   struct textlayout *tl;
@@ -166,8 +161,7 @@
   gdk_threads_leave ();    
 }
 
-JNIEXPORT void JNICALL
-Java_gnu_java_awt_peer_gtk_GdkTextLayout_dispose
+JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GdkTextLayout_dispose
   (JNIEnv *env, jobject self)
 {
   struct textlayout *tl;
