--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/IntrospectionIncubator.java	2004-11-09 02:37:21.000000000 -0700
+++ gnu/java/beans/IntrospectionIncubator.java	2004-12-01 02:15:47.000000000 -0700
@@ -99,7 +99,8 @@
 	 * @param method The method instance to examine.
 	 */
 	public void addMethod(Method method) {
-		if(Modifier.isPublic(method.getModifiers())) {
+		if(Modifier.isPublic(method.getModifiers()) &&
+			!Modifier.isStatic(method.getModifiers())) {
 			String name = ClassHelper.getTruncatedName(method.getName());
 			Class retType = method.getReturnType();
 			Class[] params = method.getParameterTypes();
