--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileChannelImpl.java	2003-04-07 02:24:15.000000000 -0600
+++ gnu/java/nio/FileChannelImpl.java	2003-03-23 02:18:01.000000000 -0700
@@ -52,7 +52,7 @@
 import java.nio.channels.NonWritableChannelException;
 import java.nio.channels.ReadableByteChannel;
 import java.nio.channels.WritableByteChannel;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
@@ -64,6 +64,8 @@
 
 public class FileChannelImpl extends FileChannel
 {
+  // GCJ LOCAL: This variable stores a pointer to the memory
+  // where the file is mapped.
   RawData map_address;
   
   int length;
@@ -84,7 +86,7 @@
 
   public FileChannelImpl ()
   {
-    this (new FileDescriptor (), true, null);
+    this (new FileDescriptor (-1), true, null);
   }
 
   private native long implPosition ();
@@ -215,12 +217,19 @@
                                long size)
     throws IOException
   {
-//     int cmode = mode.m;
-//     address = nio_mmap_file (fd, position, size, cmode);
-//     length = size;
-//     buf = new MappedByteFileBuffer (this);
-//     return buf;
-    return null;
+    if ((mode != MapMode.READ_ONLY
+         && mode != MapMode.READ_WRITE
+         && mode != MapMode.PRIVATE)
+        || position < 0
+        || size < 0
+        || size > Integer.MAX_VALUE)
+      throw new IllegalArgumentException ();
+    
+    int cmode = mode.m;
+    map_address = nio_mmap_file (position, size, cmode);
+    length = (int) size;
+    buf = new MappedByteFileBuffer (this);
+    return buf;
   }
 
   static MappedByteBuffer create_direct_mapped_buffer (RawData map_address,
