--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/charset/UTF_16Decoder.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/nio/charset/UTF_16Decoder.java	2002-11-11 00:36:41.000000000 -0700
@@ -144,6 +144,24 @@
       }
   }
 
+  /**
+   * Writes <code>c</code> to <code>out</code> in the byte order
+   * specified by <code>byteOrder</code>.
+   **/
+  private void put (ByteBuffer out, char c)
+  {
+    if (byteOrder == BIG_ENDIAN)
+      {
+        out.put ((byte) (c >> 8));
+        out.put ((byte) (c & 0xFF));
+      }
+    else
+      {
+        out.put ((byte) (c & 0xFF));
+        out.put ((byte) (c >> 8));
+      }
+  }
+
   protected void implReset ()
   {
     byteOrder = originalByteOrder;
