--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastServerRef.java	2003-07-14 02:18:55.000000000 -0600
+++ gnu/java/rmi/server/UnicastServerRef.java	2003-12-27 02:18:02.000000000 -0700
@@ -254,8 +254,11 @@
 			throw new NoSuchMethodException();
 		}
 		UnicastRemoteCall call = new UnicastRemoteCall(conn);
-		skel.dispatch(myself, call, method, hash);
-		return (call.returnValue());
+		skel.dispatch(myself, call, method, hash);		  
+		if (!call.isReturnValue())
+		  return RMIVoidValue.INSTANCE;
+		else
+		  return (call.returnValue());
 	}
 }
 
