--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkMenuBarPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkMenuBarPeer.java	2003-01-06 20:37:27.000000000 -0700
@@ -37,9 +37,7 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.Menu;
-import java.awt.MenuBar;
+import java.awt.*;
 import java.awt.peer.MenuBarPeer;
 import java.awt.peer.MenuPeer;
 
