--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/security/provider/DSAKeyPairGenerator.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/security/provider/DSAKeyPairGenerator.java	2002-08-12 10:10:53.000000000 -0600
@@ -39,18 +39,20 @@
 package gnu.java.security.provider;
 
 import java.math.BigInteger;
-import java.security.AlgorithmParameterGenerator;
 import java.security.AlgorithmParameters;
-import java.security.InvalidAlgorithmParameterException;
-import java.security.InvalidParameterException;
-import java.security.KeyPair;
+import java.security.AlgorithmParameterGenerator;
 import java.security.KeyPairGeneratorSpi;
+import java.security.KeyPair;
 import java.security.SecureRandom;
+import java.security.InvalidParameterException;
+import java.security.InvalidAlgorithmParameterException;
 import java.security.interfaces.DSAParams;
 import java.security.spec.AlgorithmParameterSpec;
 import java.security.spec.DSAParameterSpec;
 import java.util.Random;
 
+import gnu.java.security.util.Prime;
+
 public class DSAKeyPairGenerator extends KeyPairGeneratorSpi 
 	implements java.security.interfaces.DSAKeyPairGenerator
 {
