--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2003-02-20 02:17:56.000000000 -0700
@@ -37,11 +37,8 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.Dimension;
-import java.awt.Font;
-import java.awt.TextArea;
-import java.awt.peer.TextAreaPeer;
+import java.awt.peer.*;
+import java.awt.*;
 
 public class GtkTextAreaPeer extends GtkTextComponentPeer
   implements TextAreaPeer
