--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkWindowPeer.java	2002-11-09 18:32:32.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkWindowPeer.java	2003-07-09 02:18:35.000000000 -0600
@@ -37,8 +37,11 @@
 
 
 package gnu.java.awt.peer.gtk;
-import java.awt.peer.*;
-import java.awt.*;
+
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Window;
+import java.awt.peer.WindowPeer;
 
 public class GtkWindowPeer extends GtkContainerPeer
   implements WindowPeer
