--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-07-23 02:18:42.000000000 -0600
@@ -61,6 +61,7 @@
 import java.awt.Window;
 import java.awt.event.FocusEvent;
 import java.awt.event.ItemEvent;
+import java.awt.event.FocusEvent;
 import java.awt.event.KeyEvent;
 import java.awt.event.MouseEvent;
 import java.awt.event.PaintEvent;
