--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/server/UnicastServer.java	Thu Oct 31 12:43:09 2002
+++ gnu/java/rmi/server/UnicastServer.java	Mon Sep 30 16:13:32 2002
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -45,7 +45,6 @@
 import java.net.InetAddress;
 import java.util.Hashtable;
 import java.net.UnknownHostException;
-import java.rmi.Remote;
 import java.rmi.server.ObjID;
 import java.rmi.server.UnicastRemoteObject;
 import java.rmi.server.UID;
@@ -57,36 +56,27 @@
 public class UnicastServer
 	implements ProtocolConstants {
 
-static private Hashtable objects = new Hashtable();  //mapping OBJID to server ref
-static private Hashtable refcache = new Hashtable(); //mapping obj itself to server ref
+static private Hashtable objects = new Hashtable();
 static private DGCImpl dgc;
 
 public static void exportObject(UnicastServerRef obj) {
 	startDGC();
 	objects.put(obj.objid, obj);
-	refcache.put(obj.myself, obj);
 	obj.manager.startServer();
 }
 
 // FIX ME: I haven't handle force parameter
 public static boolean unexportObject(UnicastServerRef obj, boolean force) {
 	objects.remove(obj.objid);
-	refcache.remove(obj.myself);
 	obj.manager.stopServer();
 	return true;
 }
 
-public static UnicastServerRef getExportedRef(Remote remote){
-    return (UnicastServerRef)refcache.get(remote);
-}
-
 private static synchronized void startDGC() {
 	if (dgc == null) {
 		try {
 			dgc = new DGCImpl();
-			// Changed DGCImpl to inherit UnicastServerRef directly
-			//((UnicastServerRef)dgc.getRef()).exportObject(dgc);
-			dgc.exportObject(dgc);
+			((UnicastServerRef)dgc.getRef()).exportObject(dgc);
 		}
 		catch (RemoteException e) {
 			e.printStackTrace();
