--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileLockImpl.java	2004-01-26 02:24:05.000000000 -0700
+++ gnu/java/nio/FileLockImpl.java	2003-12-03 02:18:10.000000000 -0700
@@ -1,5 +1,5 @@
-/* FileLockImpl.java -- 
-   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+/* FileChannelImpl.java -- 
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -60,29 +60,20 @@
   }
   
   private FileDescriptor fd;
+  private boolean released;
   
   public FileLockImpl (FileDescriptor fd, FileChannel channel, long position,
                        long size, boolean shared)
   {
     super (channel, position, size, shared);
     this.fd = fd;
-  }
-
-  public void finalize()
-  {
-    try
-      {
-	release();
-      }
-    catch (IOException e)
-      {
-	// Ignore this.
-      }
+    this.released = false;
   }
   
   public boolean isValid ()
   {
-    return !channel().isOpen();
+    return (released
+            || !channel ().isOpen ());
   }
 
   private native void releaseImpl () throws IOException;
@@ -90,5 +81,6 @@
   public synchronized void release () throws IOException
   {
     releaseImpl ();
+    released = true;
   }
 }
