--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkCheckboxPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkCheckboxPeer.java	2003-01-06 20:37:27.000000000 -0700
@@ -37,11 +37,8 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.Checkbox;
-import java.awt.CheckboxGroup;
-import java.awt.Component;
-import java.awt.peer.CheckboxPeer;
+import java.awt.peer.*;
+import java.awt.*;
 
 public class GtkCheckboxPeer extends GtkComponentPeer
   implements CheckboxPeer
