--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/charset/UTF_16Encoder.java	2004-10-16 02:22:15.000000000 -0600
+++ gnu/java/nio/charset/UTF_16Encoder.java	2002-11-11 00:36:41.000000000 -0700
@@ -63,7 +63,7 @@
   UTF_16Encoder (Charset cs, int byteOrder, boolean useByteOrderMark)
   {
     super (cs, 2.0f,
-           useByteOrderMark ? 2.0f : 4.0f,
+           useByteOrderMark ? 4.0f : 2.0f,
            byteOrder == BIG_ENDIAN
              ? new byte[] { (byte) 0xFF, (byte) 0xFD }
              : new byte[] { (byte) 0xFD, (byte) 0xFF });
