--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkPanelPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkPanelPeer.java	2003-01-06 20:37:27.000000000 -0700
@@ -37,9 +37,8 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.Panel;
-import java.awt.peer.PanelPeer;
+import java.awt.peer.*;
+import java.awt.*;
 
 public class GtkPanelPeer extends GtkContainerPeer
   implements PanelPeer
