--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextComponentPeer.java	2003-07-08 02:19:08.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextComponentPeer.java	2003-07-08 02:18:21.000000000 -0600
@@ -59,7 +59,7 @@
   public native void select (int start, int end);
   public native void setEditable (boolean state);
   public native void setText (String text);
-
+  
   public int getIndexAtPoint(int x, int y)
   {
     return 0;  // FIXME
