--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkButtonPeer.java	2003-10-09 02:19:18.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkButtonPeer.java	2003-12-12 02:18:27.000000000 -0700
@@ -50,6 +50,8 @@
     implements ButtonPeer
 {
   native void create ();
+  public native void connectJObject ();
+  public native void connectSignals ();
 
   native void gtkSetFont(String name, int style, int size);
   native void gtkWidgetSetForeground (int red, int green, int blue);
