--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkGraphics2D.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/awt/peer/gtk/GdkGraphics2D.java	2004-06-18 02:18:49.000000000 -0600
@@ -1,5 +1,5 @@
 /* GdkGraphics2D.java
-   Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -48,6 +48,7 @@
 import java.util.Map;
 
 import java.text.AttributedCharacterIterator;
+import java.util.Map;
 import java.util.Stack;
 import java.lang.Integer;
 import gnu.java.awt.ClasspathToolkit;
