--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastServer.java	2004-09-21 02:40:00.000000000 -0600
+++ gnu/java/rmi/server/UnicastServer.java	2004-03-21 02:18:06.000000000 -0700
@@ -46,7 +46,6 @@
 import java.util.Hashtable;
 import java.net.UnknownHostException;
 import java.rmi.Remote;
-import java.rmi.ServerError;
 import java.rmi.server.ObjID;
 import java.rmi.server.UnicastRemoteObject;
 import java.rmi.server.UID;
@@ -137,10 +136,6 @@
 			returnval = e;
 			returncode = RETURN_NACK;
 		}
-                catch (Error e) {
-			returnval = new ServerError ("An Error is thrown while processing the invocation on the server", e);
-			returncode = RETURN_NACK;
-                }
 	}
 	else {
 		returnval = new NoSuchObjectException("");
