--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SocketChannelImpl.java	2003-09-26 02:20:57.000000000 -0600
+++ gnu/java/nio/SocketChannelImpl.java	2003-09-26 02:18:34.000000000 -0600
@@ -98,6 +98,11 @@
       }
   }
 
+  int getNativeFD()
+  {
+    return socket.getImpl().getNativeFD();
+  }
+
   protected void implCloseSelectableChannel () throws IOException
   {
     connected = false;
