--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/registry/RegistryImpl.java	Thu Oct 31 12:43:09 2002
+++ gnu/java/rmi/registry/RegistryImpl.java	Mon Mar 25 15:08:26 2002
@@ -64,8 +64,7 @@
 
 public RegistryImpl(int port, RMIClientSocketFactory cf, RMIServerSocketFactory sf) throws RemoteException {
 	super(new UnicastServerRef(new ObjID(ObjID.REGISTRY_ID), port, sf));
-	// The following is unnecessary, because UnicastRemoteObject export itself automatically.
-	//((UnicastServerRef)getRef()).exportObject(this);
+	((UnicastServerRef)getRef()).exportObject(this);
 }
 
 public Remote lookup(String name) throws RemoteException, NotBoundException, AccessException {
