--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/security/x509/X509Certificate.java	2003-04-30 02:20:24.000000000 -0600
+++ gnu/java/security/x509/X509Certificate.java	2003-04-30 01:23:41.000000000 -0600
@@ -421,7 +421,7 @@
     return subjectKey;
   }
 
-  protected Object writeReplace() throws java.io.ObjectStreamException
+  public Object writeReplace() throws java.io.ObjectStreamException
   {
     return super.writeReplace();
   }
