--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/ComponentDataBlitOp.java	2002-01-22 15:26:56.000000000 -0700
+++ gnu/java/awt/ComponentDataBlitOp.java	2004-02-27 02:18:20.000000000 -0700
@@ -1,4 +1,4 @@
-/* Copyright (C) 2000, 2002  Free Software Foundation
+/* Copyright (C) 2000, 2002, 2004  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -52,7 +52,7 @@
  */
 public class ComponentDataBlitOp implements RasterOp
 {
-  public static ComponentDataBlitOp INSTANCE = new ComponentDataBlitOp();
+  public static final ComponentDataBlitOp INSTANCE = new ComponentDataBlitOp();
 
   public WritableRaster filter(Raster src, WritableRaster dest)
   {
