--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/image/ImageDecoder.java	2003-12-03 02:20:00.000000000 -0700
+++ gnu/java/awt/image/ImageDecoder.java	2004-02-27 02:18:20.000000000 -0700
@@ -1,5 +1,5 @@
 /* ImageDecoder.java
-   Copyright (C) 1999, 2000 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000, 2004  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,11 +37,14 @@
 
 package gnu.java.awt.image;
 
-import java.awt.*;
-import java.awt.image.*;
-import java.util.*;
-import java.io.*;
+import java.awt.image.ImageConsumer;
+import java.awt.image.ImageProducer;
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.IOException;
 import java.net.URL;
+import java.util.Vector;
 
 public abstract class ImageDecoder implements ImageProducer 
 {
@@ -53,8 +56,6 @@
   int length;
   InputStream input;
 
-  public static ColorModel cm;
-
   static
   {
     // FIXME: there was some broken code here that looked like
