--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/IntBufferImpl.java	2003-03-11 10:27:46.000000000 -0700
+++ gnu/java/nio/IntBufferImpl.java	2003-05-09 02:26:56.000000000 -0600
@@ -70,26 +70,6 @@
     readOnly = copy.isReadOnly ();
   }
 
-  private static native int[] nio_cast (byte[] copy);
-
-  IntBufferImpl (byte[] copy)
-  {
-    super (copy.length, copy.length, 0, 0);
-    this.backing_buffer = copy != null ? nio_cast (copy) : null;
-    readOnly = false;
-  }
-
-  private static native byte nio_get_Byte (IntBufferImpl b, int index, int limit);
-
-  private static native void nio_put_Byte (IntBufferImpl b, int index, int limit, byte value);
-
-  public ByteBuffer asByteBuffer ()
-  {
-    ByteBufferImpl res = new ByteBufferImpl (backing_buffer);
-    res.limit ((limit () * 1) / 4);
-    return res;
-  }
-
   public boolean isReadOnly()
   {
     return readOnly;
