--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/NIOSocket.java	2003-10-16 02:18:55.000000000 -0600
+++ gnu/java/nio/NIOSocket.java	2003-11-23 02:18:08.000000000 -0700
@@ -59,14 +59,14 @@
     this.channel = channel;
   }
 
-  public final PlainSocketImpl getImpl()
+  public final PlainSocketImpl getPlainSocketImpl()
   {
     return impl;
   }
 
   final void setChannel (SocketChannelImpl channel)
   {
-    this.impl = channel.getImpl();
+    this.impl = channel.getPlainSocketImpl();
     this.channel = channel;
   }
   
