--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/DatagramChannelImpl.java	2003-09-27 02:18:59.000000000 -0600
+++ gnu/java/nio/DatagramChannelImpl.java	2003-11-23 02:18:08.000000000 -0700
@@ -67,7 +67,7 @@
 
   public int getNativeFD()
   {
-    return socket.getImpl().getNativeFD();
+    return socket.getPlainDatagramSocketImpl().getNativeFD();
   }
     
   public DatagramSocket socket ()
