--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/IntrospectionIncubator.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/beans/IntrospectionIncubator.java	2003-11-07 02:19:01.000000000 -0700
@@ -38,22 +38,10 @@
 
 package gnu.java.beans;
 
-import gnu.java.lang.ArrayHelper;
-import gnu.java.lang.ClassHelper;
-
-import java.beans.BeanInfo;
-import java.beans.EventSetDescriptor;
-import java.beans.IndexedPropertyDescriptor;
-import java.beans.IntrospectionException;
-import java.beans.Introspector;
-import java.beans.MethodDescriptor;
-import java.beans.PropertyDescriptor;
-import java.lang.reflect.Array;
-import java.lang.reflect.Method;
-import java.lang.reflect.Modifier;
-import java.util.Enumeration;
-import java.util.Hashtable;
-import java.util.Vector;
+import java.beans.*;
+import java.util.*;
+import java.lang.reflect.*;
+import gnu.java.lang.*;
 
 /**
  ** IntrospectionIncubator takes in a bunch of Methods, and
