--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkOffScreenImage.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkOffScreenImage.java	2003-01-06 20:37:27.000000000 -0700
@@ -35,13 +35,10 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package gnu.java.awt.peer.gtk;
 
-import java.awt.Graphics;
-import java.awt.Image;
-import java.awt.image.ImageObserver;
-import java.awt.image.ImageProducer;
+import java.awt.*;
+import java.awt.image.*;
 
 public class GtkOffScreenImage extends Image
 {
