--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkLabelPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkLabelPeer.java	2003-01-06 20:37:27.000000000 -0700
@@ -37,10 +37,8 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.Component;
-import java.awt.Label;
-import java.awt.peer.LabelPeer;
+import java.awt.*;
+import java.awt.peer.*;
 
 public class GtkLabelPeer extends GtkComponentPeer
     implements LabelPeer
