--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFramePeer.java	2004-01-28 02:26:59.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkFramePeer.java	2004-01-22 02:21:46.000000000 -0700
@@ -1,5 +1,5 @@
 /* GtkFramePeer.java -- Implements FramePeer with GTK
-   Copyright (C) 1999, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -45,7 +45,6 @@
 import java.awt.Image;
 import java.awt.MenuBar;
 import java.awt.Rectangle;
-import java.awt.Window;
 import java.awt.event.PaintEvent;
 import java.awt.peer.FramePeer;
 import java.awt.peer.MenuBarPeer;
@@ -53,74 +52,17 @@
 public class GtkFramePeer extends GtkWindowPeer
     implements FramePeer
 {
-  private int menuBarHeight;
-  private MenuBarPeer menuBar;
-  native int getMenuBarHeight (MenuBarPeer bar);
-
-  native void setMenuBarPeer (MenuBarPeer bar);
-  native void removeMenuBarPeer ();
-  native void moveLayout (int offset);
+  int menuBarHeight = 0;
+  native int getMenuBarHeight ();
+
+  native public void setMenuBarPeer (MenuBarPeer bar);
 
   public void setMenuBar (MenuBar bar)
   {
     if (bar == null)
-    {    
-      if (menuBar != null)
-      {
-        removeMenuBarPeer(); 
-        menuBar = null;
-        moveLayout(menuBarHeight);
-        insets.top -= menuBarHeight;
-        menuBarHeight = 0;      
-        awtComponent.doLayout();
-      }
-    }
+      setMenuBarPeer (null);
     else
-    {
-      int oldHeight = 0;
-      if (menuBar != null)
-      {
-        removeMenuBarPeer();
-        oldHeight = menuBarHeight;
-        insets.top -= menuBarHeight;
-      }
-      menuBar = (MenuBarPeer) ((MenuBar) bar).getPeer();
-      setMenuBarPeer(menuBar);
-      menuBarHeight = getMenuBarHeight (menuBar);
-      if (oldHeight != menuBarHeight)
-        moveLayout(oldHeight - menuBarHeight);
-      insets.top += menuBarHeight;
-      awtComponent.doLayout();
-    }
-  }
-
-  public void setBounds (int x, int y, int width, int height)
-  {
-    nativeSetBounds (x, y,
-		     width - insets.left - insets.right,
-		     height - insets.top - insets.bottom
-		     + menuBarHeight);
-  }  
-  
-  public void setResizable (boolean resizable)
-  {
-    // Call setSize; otherwise when resizable is changed from true to
-    // false the frame will shrink to the dimensions it had before it
-    // was resizable.
-    setSize (awtComponent.getWidth() - insets.left - insets.right,
-             awtComponent.getHeight() - insets.top - insets.bottom
-             + menuBarHeight);
-    set ("allow_shrink", resizable);
-    set ("allow_grow", resizable);
-  }  
-  
-  protected void postInsetsChangedEvent (int top, int left,
-					 int bottom, int right)
-  {
-    insets.top = top + menuBarHeight;
-    insets.left = left;
-    insets.bottom = bottom;
-    insets.right = right;
+      setMenuBarPeer ((MenuBarPeer) bar.getPeer ());
   }
 
   public GtkFramePeer (Frame frame)
@@ -132,7 +74,6 @@
   {
     // Create a normal decorated window.
     create (GDK_WINDOW_TYPE_HINT_NORMAL, true);
-    setMenuBar(((Frame) awtComponent).getMenuBar());
   }
 
   public void getArgs (Component component, GtkArgList args)
@@ -161,32 +102,10 @@
     g.translate (-insets.left, -insets.top);
     return g;
   }
-  
-  protected void postConfigureEvent (int x, int y, int width, int height)
-  {
-    int frame_x = x - insets.left;
-    // Since insets.top includes the MenuBar height, we need to add back
-    // the MenuBar height to the frame's y position.
-    // If no MenuBar exists in this frame, the MenuBar height will be 0.
-    int frame_y = y - insets.top + menuBarHeight;
-    int frame_width = width + insets.left + insets.right;
-    // Ditto as above. Since insets.top already includes the MenuBar's height,
-    // we need to subtract the MenuBar's height from the top inset.
-    int frame_height = height + insets.top + insets.bottom - menuBarHeight;
-    if (frame_x != awtComponent.getX()
-        || frame_y != awtComponent.getY()
-        || frame_width != awtComponent.getWidth()
-        || frame_height != awtComponent.getHeight())
-      {
-        setBoundsCallback ((Window) awtComponent,
-                           frame_x,
-                           frame_y,
-                           frame_width,
-                           frame_height);
-      }
-    awtComponent.validate();
-  }
-  
+
+  // FIXME: When MenuBars work, override postConfigureEvent and
+  // setBounds to account for MenuBar dimensions.
+
   protected void postMouseEvent(int id, long when, int mods, int x, int y, 
 				int clickCount, boolean popupTrigger)
   {
