--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastConnection.java	2004-10-03 02:35:32.000000000 -0600
+++ gnu/java/rmi/server/UnicastConnection.java	2004-03-21 02:18:06.000000000 -0700
@@ -37,6 +37,9 @@
 
 package gnu.java.rmi.server;
 
+import java.lang.Runnable;
+import java.net.Socket;
+import java.net.ServerSocket;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.ObjectInputStream;
@@ -46,8 +49,6 @@
 import java.io.ObjectOutput;
 import java.io.ObjectInput;
 import java.io.IOException;
-import java.net.Socket;
-import java.net.ServerSocket;
 import java.rmi.RemoteException;
 
 public class UnicastConnection 
