--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SelectorImpl.java	2004-08-01 02:25:24.000000000 -0600
+++ gnu/java/nio/SelectorImpl.java	2004-07-10 02:19:10.000000000 -0600
@@ -53,6 +53,14 @@
 
 public class SelectorImpl extends AbstractSelector
 {
+  static
+  {
+    // load the shared library needed for native methods.
+    if (Configuration.INIT_LOAD_LIBRARY)
+      {
+        System.loadLibrary ("javanio");
+      }
+  }
   
   private Set keys;
   private Set selected;
@@ -132,6 +140,11 @@
     return select (0);
   }
 
+  // A timeout value of 0 means block forever.
+  private static native int implSelect (int[] read, int[] write,
+                                        int[] except, long timeout)
+    throws IOException;
+
   private final int[] getFDsAsArray (int ops)
   {
     int[] result;
@@ -224,7 +237,7 @@
             try
               {
                 begin();
-                result = VMSelector.select (read, write, except, timeout);
+                result = implSelect (read, write, except, timeout);
               }
             finally
               {
