--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileLockImpl.java	2003-10-21 02:18:57.000000000 -0600
+++ gnu/java/nio/FileLockImpl.java	2003-06-13 02:27:27.000000000 -0600
@@ -42,7 +42,6 @@
 import java.io.IOException;
 import java.nio.channels.FileChannel;
 import java.nio.channels.FileLock;
-import gnu.classpath.Configuration;
 
 /**
  * @author Michael Koch
@@ -50,15 +49,6 @@
  */
 public class FileLockImpl extends FileLock
 {
-  static
-  {
-    // load the shared library needed for native methods.
-    if (Configuration.INIT_LOAD_LIBRARY)
-      {
-        System.loadLibrary ("javanio");
-      }
-  }
-  
   private FileDescriptor fd;
   private boolean released;
   
