--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkContainerPeer.java	2004-01-28 02:26:59.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkContainerPeer.java	2004-01-14 02:18:24.000000000 -0700
@@ -100,6 +100,41 @@
     return new GdkGraphics (this);
   }
 
+  public void handleEvent (AWTEvent event)
+  {
+    int id = event.getID();
+      
+    switch (id)
+      {
+      case PaintEvent.PAINT:
+      case PaintEvent.UPDATE:
+	{
+	  try 
+	    {
+	      Graphics g = getGraphics ();
+
+	      // Some peers like GtkFileDialogPeer are repainted by Gtk itself
+	      if (g == null)
+	        break;
+
+	      g.setClip (((PaintEvent)event).getUpdateRect());
+
+	      if (id == PaintEvent.PAINT)
+		awtComponent.paint (g);
+	      else
+		awtComponent.update (g);
+	      
+	      g.dispose ();
+	    } 
+	  catch (InternalError e)
+	    { 
+	      System.err.println (e);
+	    }
+	}
+	break;
+      }
+  }
+
   public void beginLayout () { }
   public void endLayout () { }
   public boolean isPaintPending () { return false; }
