--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkPixbufDecoder.java	2003-11-04 04:24:08.000000000 -0700
+++ gnu/java/awt/peer/gtk/GdkPixbufDecoder.java	2003-11-12 02:18:12.000000000 -0700
@@ -41,6 +41,7 @@
 import java.awt.image.*;
 import java.io.FileDescriptor;
 import java.io.FileInputStream;
+import java.io.InputStream;
 import java.io.IOException;
 import java.net.URL;
 import java.util.Vector;
@@ -85,6 +86,12 @@
     initState ();
   }
 
+  public GdkPixbufDecoder (byte[] imagedata, int imageoffset, int imagelength)
+  {
+    super (imagedata, imageoffset, imagelength);
+    initState ();
+  }
+
   // called back by native side
   void areaPrepared (int width, int height)
   {
@@ -122,7 +129,7 @@
   // this object, feeding back decoded pixel blocks, which we pass to each
   // of the ImageConsumers in the provided Vector.
 
-  public void produce (Vector v, FileInputStream is) throws IOException
+  public void produce (Vector v, InputStream is) throws IOException
   {
     curr = v;
 
@@ -208,13 +215,9 @@
 
   public static BufferedImage createBufferedImage (String filename)
   {
-    /*
     BufferedImageBuilder bb = new BufferedImageBuilder ();
     GdkPixbufDecoder dec = new GdkPixbufDecoder (filename);
     dec.startProduction (bb);
     return bb.getBufferedImage ();
-    */
-    throw new Error ("not implemented");
   }
-
 }
