--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileChannelImpl.java	2003-06-19 02:34:36.000000000 -0600
+++ gnu/java/nio/FileChannelImpl.java	2003-06-15 02:21:55.000000000 -0600
@@ -46,7 +46,6 @@
 import java.io.RandomAccessFile;
 import java.nio.ByteBuffer;
 import java.nio.MappedByteBuffer;
-import java.nio.MappedByteFileBuffer;
 import java.nio.channels.ClosedChannelException;
 import java.nio.channels.FileChannel;
 import java.nio.channels.FileLock;
@@ -54,7 +53,7 @@
 import java.nio.channels.NonWritableChannelException;
 import java.nio.channels.ReadableByteChannel;
 import java.nio.channels.WritableByteChannel;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
@@ -66,7 +65,7 @@
 
 public class FileChannelImpl extends FileChannel
 {
-  public RawData map_address;
+  RawData map_address;
   
   int length;
   FileDescriptor fd;
@@ -86,7 +85,7 @@
 
   public FileChannelImpl ()
   {
-    this (new FileDescriptor (), true, null);
+    this (new FileDescriptor (-1), true, null);
   }
 
   private native long implPosition ();
