--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkDialogPeer.java	2004-01-14 02:21:29.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkDialogPeer.java	2004-05-27 02:18:56.000000000 -0600
@@ -41,7 +41,10 @@
 import java.awt.AWTEvent;
 import java.awt.Component;
 import java.awt.Dialog;
+import java.awt.Graphics;
 import java.awt.peer.DialogPeer;
+import java.awt.Rectangle;
+import java.awt.event.PaintEvent;
 
 public class GtkDialogPeer extends GtkWindowPeer
   implements DialogPeer
@@ -50,6 +53,33 @@
   {
     super (dialog);
   }
+  
+  public Graphics getGraphics ()
+  {
+    Graphics g;
+    if (GtkToolkit.useGraphics2D ())
+      g = new GdkGraphics2D (this);
+    else
+      g = new GdkGraphics (this);
+    g.translate (-insets.left, -insets.top);
+    return g;
+  }  
+  
+  protected void postMouseEvent(int id, long when, int mods, int x, int y, 
+				int clickCount, boolean popupTrigger)
+  {
+    super.postMouseEvent (id, when, mods, 
+			  x + insets.left, y + insets.top, 
+			  clickCount, popupTrigger);
+  }
+
+  protected void postExposeEvent (int x, int y, int width, int height)
+  {
+    q.postEvent (new PaintEvent (awtComponent, PaintEvent.PAINT,
+				 new Rectangle (x + insets.left, 
+						y + insets.top, 
+						width, height)));
+  }  
 
   void create ()
   {
