--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2004-04-30 02:23:29.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2004-01-14 02:18:24.000000000 -0700
@@ -105,6 +105,8 @@
   public Dimension minimumSize (int rows, int cols)
   {
     TextArea ta = ((TextArea) awtComponent);
+    int hScrollbarHeight = 0;
+    int vScrollbarWidth = 0;
     int height = 0;
     int width = 0;
 
@@ -135,6 +137,8 @@
   public Dimension preferredSize (int rows, int cols)
   {
     TextArea ta = ((TextArea) awtComponent);
+    int hScrollbarHeight = 0;
+    int vScrollbarWidth = 0;
     int height = 0;
     int width = 0;
 
