--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/dgc/DGCImpl.java	Sat Mar 23 09:42:53 2002
+++ gnu/java/rmi/dgc/DGCImpl.java	Tue Jan 22 15:39:58 2002
@@ -49,16 +49,12 @@
 public class DGCImpl
 	extends UnicastRemoteObject implements DGC {
 
-private static final long leaseValue = 600000L;
-
 public DGCImpl() throws RemoteException {
 	super(new UnicastServerRef(new ObjID(ObjID.DGC_ID), 0, RMISocketFactory.getSocketFactory()));
 }
 
 public Lease dirty(ObjID[] ids, long sequenceNum, Lease lease) throws RemoteException {
-	VMID vmid = lease.getVMID();
-    lease = new Lease(vmid, leaseValue);
-	System.out.println("DGCImpl.dirty - not completely implemented");
+	System.out.println("DGCImpl.dirty - not implemented");
 	return (lease);
 }
 
