--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastConnectionManager.java	2004-10-03 02:35:32.000000000 -0600
+++ gnu/java/rmi/server/UnicastConnectionManager.java	2004-03-21 02:18:06.000000000 -0700
@@ -1,6 +1,5 @@
 /*
-  Copyright (c) 1996, 1997, 1998, 1999, 2002i, 2004
-  Free Software Foundation, Inc.
+  Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -46,10 +45,13 @@
 import java.io.ObjectOutput;
 import java.io.ObjectInput;
 import java.io.DataInputStream;
+import java.lang.Thread;
+import java.lang.Runnable;
 import java.net.InetAddress;
 import java.net.Socket;
 import java.net.ServerSocket;
 import java.net.UnknownHostException;
+
 import java.util.ArrayList;
 import java.util.ConcurrentModificationException;
 import java.util.Enumeration;
