--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SelectorProviderImpl.java	2003-03-23 02:18:38.000000000 -0700
+++ gnu/java/nio/SelectorProviderImpl.java	2003-10-13 02:17:50.000000000 -0600
@@ -1,5 +1,5 @@
 /* SelectorProviderImpl.java -- 
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -60,7 +60,7 @@
   public Pipe openPipe ()
     throws IOException
   {
-    return new PipeImpl ();
+    return new PipeImpl (this);
   }
     
   public AbstractSelector openSelector ()
