--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/locale/LocaleInformation_en.java	2003-06-28 02:19:11.000000000 -0600
+++ gnu/java/locale/LocaleInformation_en.java	2002-01-22 16:01:30.000000000 -0700
@@ -62,7 +62,7 @@
   // and probably other things as well.
   private static final String collation_rules = 
   "<0<1<2<3<4<5<6<7<8<9<A,a<b,B<c,C<d,D<e,E<f,F<g,G<h,H<i,I<j,J<k,K" +
-  "<l,L<m,M<n,N<o,O<p,P<q,Q<r,R<s,S<t,T<u,U<v,V<w,W<x,X<y,Y<z,Z";
+  "<l,L<m,M<n,N<o,O<p,P<q,Q<r,R<s,S<t,T<u,U<v,V<w,W<x,X<y,Y,z<Z";
 
   /**
    * This is the list of months, fully spelled out
