--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastConnectionManager.java	2002-10-31 11:35:21.000000000 -0700
+++ gnu/java/rmi/server/UnicastConnectionManager.java	2003-12-27 02:18:02.000000000 -0700
@@ -152,6 +152,9 @@
             if (debug) System.out.println("************* exit scavenger.");
         }
     });
+    // As it is used for client connection, we may put this thread
+    // in daemon state to prevent the VM from blocking when exiting.
+    scavenger.setDaemon(true);
     scavenger.start();
 }
 
