--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/util/prefs/NodeReader.java	2004-04-28 02:20:15.000000000 -0600
+++ gnu/java/util/prefs/NodeReader.java	2003-02-13 00:45:58.000000000 -0700
@@ -38,13 +38,13 @@
 package gnu.java.util.prefs;
 
 import java.io.BufferedReader;
-import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.io.IOException;
 import java.io.Reader;
-import java.util.prefs.InvalidPreferencesFormatException;
-import java.util.prefs.Preferences;
-import java.util.prefs.PreferencesFactory;
+
+import java.util.prefs.*;
 
 /**
  * Reads and imports preferences nodes from files.
