--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SocketChannelImpl.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/nio/SocketChannelImpl.java	2004-02-06 02:18:31.000000000 -0700
@@ -38,12 +38,12 @@
 
 package gnu.java.nio;
 
-import gnu.java.net.PlainSocketImpl;
-
-import java.io.IOException;
 import java.io.InputStream;
+import java.io.IOException;
 import java.io.OutputStream;
+import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import gnu.java.net.PlainSocketImpl;
 import java.net.Socket;
 import java.net.SocketAddress;
 import java.net.SocketTimeoutException;
@@ -53,12 +53,13 @@
 import java.nio.channels.ConnectionPendingException;
 import java.nio.channels.NoConnectionPendingException;
 import java.nio.channels.NotYetConnectedException;
-import java.nio.channels.SelectionKey;
-import java.nio.channels.Selector;
-import java.nio.channels.SocketChannel;
 import java.nio.channels.UnresolvedAddressException;
 import java.nio.channels.UnsupportedAddressTypeException;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.Selector;
+import java.nio.channels.SelectionKey;
 import java.nio.channels.spi.SelectorProvider;
+import gnu.classpath.Configuration;
 
 public final class SocketChannelImpl extends SocketChannel
 {
