--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/BitwiseXORComposite.java	2004-04-30 02:23:29.000000000 -0600
+++ gnu/java/awt/BitwiseXORComposite.java	2004-01-12 12:11:00.000000000 -0700
@@ -232,7 +232,7 @@
                         WritableRaster dstOut)
     {
       int aX, bX, dstX, aY, bY, dstY, width, height;
-      int xorPixel;
+      int xorPixel, transferType;
       int[] srcLine, dstLine;
 
       aX = src.getMinX();
