--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkPopupMenuPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkPopupMenuPeer.java	2003-03-02 02:17:42.000000000 -0700
@@ -37,12 +37,8 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.Component;
-import java.awt.Event;
-import java.awt.MenuItem;
-import java.awt.Point;
-import java.awt.PopupMenu;
+import java.awt.*;
+import java.awt.peer.ComponentPeer;
 import java.awt.peer.PopupMenuPeer;
 
 public class GtkPopupMenuPeer extends GtkMenuPeer
