--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkContainerPeer.java	2003-02-26 02:18:42.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkContainerPeer.java	2003-07-09 02:18:35.000000000 -0600
@@ -37,8 +37,12 @@
 
 
 package gnu.java.awt.peer.gtk;
-import java.awt.*;
-import java.awt.event.*;
+
+import java.awt.AWTEvent;
+import java.awt.Container;
+import java.awt.Graphics;
+import java.awt.Insets;
+import java.awt.event.PaintEvent;
 import java.awt.peer.ContainerPeer;
 
 public class GtkContainerPeer extends GtkComponentPeer
