--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFramePeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkFramePeer.java	2003-09-20 02:18:07.000000000 -0600
@@ -84,6 +84,7 @@
     args.add ("allow_shrink", frame.isResizable ());
     args.add ("allow_grow", frame.isResizable ());
   }
+
   public void setIconImage (Image image) 
   {
       /* TODO: Waiting on Toolkit Image routines */
