--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2003-08-06 02:18:11.000000000 -0600
@@ -350,13 +350,13 @@
 				 new Rectangle (x, y, width, height)));
   }
 
-  protected void postKeyEvent (int id, long when, int mods, 
-			       int keyCode, char keyChar)
+  protected void postKeyEvent (int id, long when, int mods,
+			       int keyCode, char keyChar, int keyLocation)
   {
-    q.postEvent (new KeyEvent (awtComponent, id, when, mods, 
-			       keyCode, keyChar));
+    q.postEvent (new KeyEvent (awtComponent, id, when, mods,
+			       keyCode, keyChar, keyLocation));
   }
-  
+
   protected void postFocusEvent (int id, boolean temporary)
   {
     q.postEvent (new FocusEvent (awtComponent, id, temporary));
