--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2003-10-09 02:19:19.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2003-11-13 02:18:09.000000000 -0700
@@ -242,13 +242,12 @@
       PrepareImage (GtkImage image, ImageObserver observer)
       {
 	this.image = image;
-	this.observer = observer;
+	image.setObserver (observer);
       }
       
       public void run ()
       {
-	// XXX: need to return data to image observer
-	image.source.startProduction (null);
+	image.source.startProduction (image);
       }
     }
 
