--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkDialogPeer.java	2003-08-23 02:18:43.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkDialogPeer.java	2003-09-20 02:18:07.000000000 -0600
@@ -53,7 +53,9 @@
 
   void create ()
   {
-    create (GTK_WINDOW_POPUP);
+    create (GTK_WINDOW_TOPLEVEL,
+	    awtComponent.getWidth(),
+	    awtComponent.getHeight());
   }
 
   public void getArgs (Component component, GtkArgList args)
@@ -62,6 +64,7 @@
 
     Dialog dialog = (Dialog) component;
 
+    args.add ("title", dialog.getTitle ());
     args.add ("modal", dialog.isModal ());
     args.add ("allow_shrink", dialog.isResizable ());
     args.add ("allow_grow", dialog.isResizable ());
