--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileChannelImpl.java	2003-06-28 02:19:11.000000000 -0600
+++ gnu/java/nio/FileChannelImpl.java	2003-06-28 02:18:19.000000000 -0600
@@ -54,7 +54,7 @@
 import java.nio.channels.NonWritableChannelException;
 import java.nio.channels.ReadableByteChannel;
 import java.nio.channels.WritableByteChannel;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
