--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-06-08 02:24:15.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkComponentPeer.java	2004-06-18 02:18:49.000000000 -0600
@@ -199,7 +199,10 @@
 
   public Graphics getGraphics ()
   {
-    return null;
+    if (GtkToolkit.useGraphics2D ())
+        return new GdkGraphics2D (this);
+    else
+        return new GdkGraphics (this);
   }
 
   public Point getLocationOnScreen () 
