--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkToolkit.java	2003-11-07 02:20:15.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkToolkit.java	2003-07-28 02:18:18.000000000 -0600
@@ -52,7 +52,9 @@
 import java.util.Map;
 import java.util.MissingResourceException;
 import java.util.Properties;
-import gnu.java.awt.peer.gtk.GdkPixbufDecoder;
+import gnu.java.awt.EmbeddedWindow;
+import gnu.java.awt.EmbeddedWindowSupport;
+import gnu.java.awt.peer.EmbeddedWindowPeer;
 import gnu.classpath.Configuration;
 
 /* This class uses a deprecated method java.awt.peer.ComponentPeer.getPeer().
@@ -63,6 +65,7 @@
    that will keep up with every window's peer, but for now this is faster. */
 
 public class GtkToolkit extends Toolkit
+  implements EmbeddedWindowSupport
 {
   GtkMainThread main;
   Hashtable containers = new Hashtable();
@@ -96,12 +99,16 @@
 
   public Image createImage (String filename)
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image createImage (URL url)
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public Image createImage (ImageProducer producer) 
@@ -137,12 +144,16 @@
 
   public Image getImage (String filename) 
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image getImage (URL url) 
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public PrintJob getPrintJob (Frame frame, String jobtitle, Properties props) 
@@ -292,6 +303,11 @@
     return new GtkWindowPeer (w);
   }
 
+  public EmbeddedWindowPeer createEmbeddedWindow (EmbeddedWindow w)
+  {
+    return new GtkEmbeddedWindowPeer (w);
+  }
+
   protected FontPeer getFontPeer (String name, int style) 
   {
     try {
