--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastServerRef.java	2004-10-03 02:35:32.000000000 -0600
+++ gnu/java/rmi/server/UnicastServerRef.java	2004-09-25 02:16:22.000000000 -0600
@@ -1,6 +1,5 @@
-/* UnicastServerRef.java --
-   Copyright (c) 1996, 1997, 1998, 1999, 2002, 2003, 2004
-   Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -58,6 +57,8 @@
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
 import java.lang.reflect.InvocationTargetException;
+import java.lang.Thread;
+import java.lang.Exception;
 import java.io.IOException;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
