--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkContainerPeer.java	2004-03-20 02:21:14.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkContainerPeer.java	2004-06-18 02:18:49.000000000 -0600
@@ -107,10 +107,7 @@
 
   public Graphics getGraphics ()
   {
-    if (GtkToolkit.useGraphics2D ())
-        return new GdkGraphics2D (this);
-    else
-    return new GdkGraphics (this);
+    return super.getGraphics();
   }
 
   public void beginLayout () { }
