--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextComponentPeer.java	2003-08-07 02:18:42.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextComponentPeer.java	2003-12-12 02:18:27.000000000 -0700
@@ -56,7 +56,7 @@
     setText (tc.getText ());
   }
 
-  public native void connectHooks ();
+  public native void connectSignals ();
 
   public native int getCaretPosition ();
   public native void setCaretPosition (int pos);
