--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/javax/rmi/CORBA/DelegateFactory.java	2004-03-12 02:21:18.000000000 -0700
+++ gnu/javax/rmi/CORBA/DelegateFactory.java	2003-03-29 14:23:25.000000000 -0700
@@ -60,9 +60,7 @@
       }
     try
       {
-	Class dclass = Class.forName(dcname, 
-				     true,
-				     Thread.currentThread().getContextClassLoader());
+	Class dclass = Class.forName(dcname);
 	r = dclass.newInstance();
 	cache.put(type, r);
 	return r;
