--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkButtonPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkButtonPeer.java	2003-02-20 02:17:56.000000000 -0700
@@ -37,14 +37,10 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.AWTEvent;
-import java.awt.Button;
-import java.awt.Component;
-import java.awt.Font;
+import java.awt.*;
 import java.awt.event.MouseEvent;
 import java.awt.event.KeyEvent;
-import java.awt.peer.ButtonPeer;
+import java.awt.peer.*;
 
 public class GtkButtonPeer extends GtkComponentPeer
     implements ButtonPeer
