--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkDialogPeer.java	2002-11-09 16:17:30.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkDialogPeer.java	2003-07-09 02:18:35.000000000 -0600
@@ -37,9 +37,11 @@
 
 
 package gnu.java.awt.peer.gtk;
-import java.awt.*;
-import java.awt.peer.*;
-import java.awt.event.*;
+
+import java.awt.AWTEvent;
+import java.awt.Component;
+import java.awt.Dialog;
+import java.awt.peer.DialogPeer;
 
 public class GtkDialogPeer extends GtkWindowPeer
   implements DialogPeer
