--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkToolkit.java	2003-07-02 02:19:04.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkToolkit.java	2003-07-18 02:18:00.000000000 -0600
@@ -52,7 +52,6 @@
 import java.util.Map;
 import java.util.MissingResourceException;
 import java.util.Properties;
-import gnu.java.awt.image.GdkPixbufDecoder;
 import gnu.classpath.Configuration;
 
 /* This class uses a deprecated method java.awt.peer.ComponentPeer.getPeer().
@@ -96,12 +95,16 @@
 
   public Image createImage (String filename)
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image createImage (URL url)
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public Image createImage (ImageProducer producer) 
@@ -137,12 +140,16 @@
 
   public Image getImage (String filename) 
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image getImage (URL url) 
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public PrintJob getPrintJob (Frame frame, String jobtitle, Properties props) 
