--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileChannelImpl.java	2003-05-10 02:20:17.000000000 -0600
+++ gnu/java/nio/FileChannelImpl.java	2003-05-02 02:20:04.000000000 -0600
@@ -52,7 +52,7 @@
 import java.nio.channels.NonWritableChannelException;
 import java.nio.channels.ReadableByteChannel;
 import java.nio.channels.WritableByteChannel;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
@@ -64,6 +64,8 @@
 
 public class FileChannelImpl extends FileChannel
 {
+  // GCJ LOCAL: This variable stores a pointer to the memory
+  // where the file is mapped.
   RawData map_address;
   
   int length;
@@ -84,7 +86,7 @@
 
   public FileChannelImpl ()
   {
-    this (new FileDescriptor (), true, null);
+    this (new FileDescriptor (-1), true, null);
   }
 
   private native long implPosition ();
