--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/GLightweightPeer.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/awt/peer/GLightweightPeer.java	2004-01-11 02:18:04.000000000 -0700
@@ -1,5 +1,5 @@
 /* GLightweightPeer.java --
-   Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -45,6 +45,7 @@
 import java.awt.Cursor;
 import java.awt.Dimension;
 import java.awt.Font;
+import java.awt.Font;
 import java.awt.FontMetrics;
 import java.awt.Graphics;
 import java.awt.GraphicsConfiguration;
