--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/info/ComponentBeanInfo.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/beans/info/ComponentBeanInfo.java	2002-01-22 16:01:30.000000000 -0700
@@ -38,9 +38,8 @@
 
 package gnu.java.beans.info;
 
-import java.beans.IntrospectionException;
-import java.beans.PropertyDescriptor;
-import java.beans.SimpleBeanInfo;
+import gnu.java.beans.*;
+import java.beans.*;
 
 /** BeanInfo class for java.awt.Component.
  ** This provides a few properties, but that's
