--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/registry/RegistryImpl_Stub.java	Thu Oct 31 12:43:09 2002
+++ gnu/java/rmi/registry/RegistryImpl_Stub.java	Mon Sep 30 16:11:38 2002
@@ -67,7 +67,7 @@
     static {
         try {
             java.rmi.server.RemoteRef.class.getMethod("invoke", new java.lang.Class[] { java.rmi.Remote.class, java.lang.reflect.Method.class, java.lang.Object[].class, long.class });
-            useNewInvoke = false;
+            useNewInvoke = true;
             $method_bind_0 = gnu.java.rmi.registry.RegistryImpl.class.getMethod("bind", new java.lang.Class[] {java.lang.String.class, java.rmi.Remote.class});
             $method_list_1 = gnu.java.rmi.registry.RegistryImpl.class.getMethod("list", new java.lang.Class[] {});
             $method_lookup_2 = gnu.java.rmi.registry.RegistryImpl.class.getMethod("lookup", new java.lang.Class[] {java.lang.String.class});
