--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/security/der/DERReader.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/security/der/DERReader.java	2003-04-30 01:23:40.000000000 -0600
@@ -335,7 +335,7 @@
               Integer.parseInt(str.substring( 6,  8)),    // hour
               Integer.parseInt(str.substring( 8, 10)));   // minute
             if (date.length() == 12);
-              calendar.set(Calendar.SECOND,
+              calendar.set(calendar.SECOND,
                 Integer.parseInt(str.substring(10, 12)));
           }
         catch (NumberFormatException nfe)
@@ -362,13 +362,13 @@
                 case 18:
                 case 17:
                 case 16:
-                  calendar.set(Calendar.MILLISECOND,
+                  calendar.set(calendar.MILLISECOND,
                     Integer.parseInt(date.substring(15)));
                 case 14:
-                  calendar.set(Calendar.SECOND,
+                  calendar.set(calendar.SECOND,
                     Integer.parseInt(date.substring(12, 14)));
                 case 12:
-                  calendar.set(Calendar.MINUTE,
+                  calendar.set(calendar.MINUTE,
                     Integer.parseInt(date.substring(10, 12)));
               }
           }
