--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkCanvasPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkCanvasPeer.java	2003-01-06 20:37:27.000000000 -0700
@@ -35,15 +35,10 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package gnu.java.awt.peer.gtk;
-
-import java.awt.AWTEvent;
-import java.awt.Canvas;
-import java.awt.Dimension;
-import java.awt.Graphics;
+import java.awt.*;
 import java.awt.event.PaintEvent;
-import java.awt.peer.CanvasPeer;
+import java.awt.peer.*;
 
 public class GtkCanvasPeer extends GtkComponentPeer implements CanvasPeer
 {
