--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkChoicePeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkChoicePeer.java	2003-12-24 02:18:18.000000000 -0700
@@ -84,10 +84,10 @@
   }
   */
 
-  protected void postItemEvent (Object item, int stateChange)
+  protected void choicePostItemEvent (String label, int stateChange)
   {
     if (stateChange == ItemEvent.SELECTED)
-      ((Choice) awtComponent).select ((String) item);
-    super.postItemEvent (item, stateChange);
+      ((Choice) awtComponent).select (label);
+    super.postItemEvent (label, stateChange);
   }
 }
