--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/ServerSocketChannelImpl.java	2003-10-16 02:18:55.000000000 -0600
+++ gnu/java/nio/ServerSocketChannelImpl.java	2003-12-03 02:18:10.000000000 -0700
@@ -54,9 +54,9 @@
 
 public final class ServerSocketChannelImpl extends ServerSocketChannel
 {
-  NIOServerSocket serverSocket;
-  boolean blocking = true;
-  boolean connected = false;
+  private NIOServerSocket serverSocket;
+  private boolean blocking = true;
+  private boolean connected;
 
   protected ServerSocketChannelImpl (SelectorProvider provider)
     throws IOException
