--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/channels/FileChannelImpl.java	2004-05-02 02:20:13.000000000 -0600
+++ gnu/java/nio/channels/FileChannelImpl.java	2004-04-23 02:18:16.000000000 -0600
@@ -94,6 +94,7 @@
   // efficient way to accomplish that.
   private int fd = -1;
 
+  private long pos;
   private int mode;
 
   public FileChannelImpl ()
@@ -413,9 +414,7 @@
     if ((mode & WRITE) == 0)
        throw new NonWritableChannelException ();
 
-    if (size < size ())
-      implTruncate (size);
-
+    implTruncate (size);
     return this;
   }
 }
