--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SocketChannelImpl.java	Thu Dec  5 09:57:53 2002
+++ gnu/java/nio/SocketChannelImpl.java	Fri Nov 29 02:57:05 2002
@@ -46,6 +46,7 @@
 import java.nio.channels.AlreadyConnectedException;
 import java.nio.channels.SocketChannel;
 import java.nio.channels.spi.SelectorProvider;
+import gnu.classpath.Configuration;
 
 public class SocketChannelImpl extends SocketChannel
 {
@@ -75,7 +76,7 @@
         System.err.println("failed to create socket:"+fd);
 	    }
   }
-    
+
   public void finalizer()
   {
     if (connected)
