--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/TestAWT.java	2004-01-27 02:25:12.000000000 -0700
+++ gnu/java/awt/peer/gtk/TestAWT.java	2004-01-22 02:21:46.000000000 -0700
@@ -272,7 +272,6 @@
         public void windowClosing (WindowEvent e)
         {
           System.out.println ("Window Closing");
-	  text.setVisible (false);
           hide ();
         }
       });
@@ -306,8 +305,7 @@
     cb.addActionListener(new ActionListener () {
       public void actionPerformed (ActionEvent e) 
 	{
-	  text.setVisible (false);
-	  hide();
+	  dispose();
 	}
     });
     
