--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/ServerSocketChannelImpl.java	2003-06-19 02:34:36.000000000 -0600
+++ gnu/java/nio/ServerSocketChannelImpl.java	2003-10-10 02:18:09.000000000 -0600
@@ -38,6 +38,7 @@
 
 package gnu.java.nio;
 
+import gnu.java.net.PlainSocketImpl;
 import java.io.IOException;
 import java.net.InetSocketAddress;
 import java.net.ServerSocket;
@@ -47,9 +48,10 @@
 import java.nio.channels.SocketChannel;
 import java.nio.channels.spi.SelectorProvider;
 
-class ServerSocketChannelImpl extends ServerSocketChannel
+public final class ServerSocketChannelImpl extends ServerSocketChannel
 {
   ServerSocket serverSocket;
+  PlainSocketImpl impl;
   boolean blocking = true;
   boolean connected = false;
 
@@ -57,7 +59,20 @@
     throws IOException
   {
     super (provider);
-    serverSocket = new ServerSocket ();
+    impl = new PlainSocketImpl();
+    initServerSocket();
+  }
+
+  /*
+   * This method is only need to call a package private constructor
+   * of java.net.ServerSocket. It only initializes the member variables
+   * "serverSocket".
+   */
+  private native void initServerSocket() throws IOException;
+
+  public int getNativeFD()
+  {
+    return impl.getNativeFD();
   }
  
   public void finalizer()
