--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/server/UnicastServer.java	Sat Mar 23 09:42:53 2002
+++ gnu/java/rmi/server/UnicastServer.java	Tue Jan 22 15:40:01 2002
@@ -65,13 +65,6 @@
 	obj.manager.startServer();
 }
 
-// FIX ME: I haven't handle force parameter
-public static boolean unexportObject(UnicastServerRef obj, boolean force) {
-	objects.remove(obj.objid);
-	obj.manager.stopServer();
-	return true;
-}
-
 private static synchronized void startDGC() {
 	if (dgc == null) {
 		try {
@@ -107,14 +100,10 @@
 	UnicastServerRef uref = (UnicastServerRef)objects.get(objid);
 	Object returnval;
 	int returncode = RETURN_ACK;
-	// returnval is from Method.invoke(), so we must check the return class to see
-	// if it's primitive type
-	Class returncls = null;
 	if (uref != null) {
 		try {
 			// Dispatch the call to it.
 			returnval = uref.incomingMessageCall(conn, method, hash);
-			returncls = uref.getMethodReturnType(method, hash);
 		}
 		catch (Exception e) {
 			returnval = e;
@@ -132,10 +121,7 @@
 
 	out.writeByte(returncode);
 	(new UID()).write(out);
-	if(returnval != null && returncls != null)
-	    ((RMIObjectOutputStream)out).writeValue(returnval, returncls);
-	else
-	    out.writeObject(returnval);
+	out.writeObject(returnval);
 
 	out.flush();
 }
