--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/editors/NativeBooleanEditor.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/beans/editors/NativeBooleanEditor.java	2002-03-24 16:11:02.000000000 -0700
@@ -38,7 +38,7 @@
 
 package gnu.java.beans.editors;
 
-import java.beans.PropertyEditorSupport;
+import java.beans.*;
 
 /**
  ** NativeBooleanEditor is a property editor for the
