--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkPixbufDecoder.java	2003-11-04 04:24:08.000000000 -0700
+++ gnu/java/awt/peer/gtk/GdkPixbufDecoder.java	2003-09-17 14:03:02.000000000 -0600
@@ -122,7 +122,7 @@
   // this object, feeding back decoded pixel blocks, which we pass to each
   // of the ImageConsumers in the provided Vector.
 
-  public void produce (Vector v, FileInputStream is) throws IOException
+  void produce (Vector v, FileInputStream is) throws IOException
   {
     curr = v;
 
@@ -208,13 +208,10 @@
 
   public static BufferedImage createBufferedImage (String filename)
   {
-    /*
     BufferedImageBuilder bb = new BufferedImageBuilder ();
     GdkPixbufDecoder dec = new GdkPixbufDecoder (filename);
     dec.startProduction (bb);
     return bb.getBufferedImage ();
-    */
-    throw new Error ("not implemented");
   }
 
 }
