--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileLockImpl.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/nio/FileLockImpl.java	2004-04-21 02:18:07.000000000 -0600
@@ -38,10 +38,10 @@
 
 package gnu.java.nio;
 
-import gnu.java.nio.channels.FileChannelImpl;
-
 import java.io.IOException;
 import java.nio.channels.FileLock;
+import gnu.classpath.Configuration;
+import gnu.java.nio.channels.FileChannelImpl;
 
 /**
  * @author Michael Koch
