--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/UnicastServerRef.java	2003-07-04 02:19:07.000000000 -0600
+++ gnu/java/rmi/server/UnicastServerRef.java	2003-01-03 15:42:10.000000000 -0700
@@ -225,9 +225,7 @@
 			 * lets us know that.
 			 */
 			try {
-				// need to handle primitive types
-				args[i] = ((RMIObjectInputStream)in).readValue(meth.getParameterTypes()[i]);
-				
+				args[i] = in.readObject();
 			}
 			catch (Exception t) {
 				t.printStackTrace();
@@ -259,5 +257,3 @@
 }
 
 }
-
-
