--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/server/UnicastServerRef.java	Thu Oct 31 12:43:09 2002
+++ gnu/java/rmi/server/UnicastServerRef.java	Mon Sep 30 16:13:33 2002
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -63,8 +63,6 @@
 import java.io.DataOutputStream;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
-import java.io.ObjectInput;
-import java.io.ObjectOutput;
 import java.util.Hashtable;
 
 public class UnicastServerRef
@@ -73,18 +71,11 @@
 
 final static private Class[] stubprototype = new Class[] { RemoteRef.class };
 
-Remote myself; //save the remote object itself
+Remote myself;
 private Skeleton skel;
 private RemoteStub stub;
 private Hashtable methods = new Hashtable();
 
-/**
- * Used by serialization.
- */
-UnicastServerRef()
-{
-}
-
 public UnicastServerRef(ObjID id, int port, RMIServerSocketFactory ssf) {
 	super(id);
 	manager = UnicastConnectionManager.getInstance(port, ssf);
@@ -93,9 +84,6 @@
 public RemoteStub exportObject(Remote obj) throws RemoteException {
 	if (myself == null) {
 		myself = obj;
-		// Save it to server manager, to let client calls in the same VM to issue
-		//  local call
-		manager.serverobj = obj;
 
 		// Find and install the stub
 		Class cls = obj.getClass();
@@ -124,10 +112,6 @@
 	return exportObject(remote);
 }
 
-public RemoteStub getStub(){
-    return stub;
-}
-
 
 public boolean unexportObject(Remote obj, boolean force) throws RemoteException {
     // Remove all hashes of methods which may be called.
