--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkCheckboxGroupPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkCheckboxGroupPeer.java	2003-01-06 20:37:27.000000000 -0700
@@ -37,9 +37,9 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.CheckboxGroup;
-import java.util.WeakHashMap;
+import java.awt.*;
+import java.awt.event.*;
+import java.util.*;
 
 // Note that there is no peer interface for a CheckboxGroup.  We
 // introduce our own in order to make it easier to keep a piece of
