--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/CharBufferImpl.java	Tue Mar 11 10:27:46 2003
+++ gnu/java/nio/CharBufferImpl.java	Wed Mar 12 02:17:46 2003
@@ -58,7 +58,6 @@
   
   public CharBufferImpl(char[] array, int offset, int length)
   {
-    // FIXME: check if IndexOutOfBoundsException is thrown correctly
     super (array.length, length, offset, 0);
     this.backing_buffer = array;
     readOnly = false;
