--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkToolkit.java	Fri Jan 31 09:56:27 2003
+++ gnu/java/awt/peer/gtk/GtkToolkit.java	Fri Jan 31 09:59:29 2003
@@ -94,12 +94,16 @@
 
   public Image createImage (String filename)
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image createImage (URL url)
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public Image createImage (ImageProducer producer) 
@@ -135,12 +139,16 @@
 
   public Image getImage (String filename) 
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image getImage (URL url) 
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public PrintJob getPrintJob (Frame frame, String jobtitle, Properties props) 
