--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkImagePainter.java	2004-06-08 02:24:15.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkImagePainter.java	2004-06-18 02:18:49.000000000 -0600
@@ -104,7 +104,7 @@
     s_width = Math.abs (sx2 - sx1);
     s_height = Math.abs (sy2 - sy1);
     clip = new Rectangle (sx1, sy1, s_width, s_height);
-    
+
     new Thread (this).start ();
   }
 
@@ -170,8 +170,6 @@
 
 	offset += r.y * scansize + r.x;
 
-	r.translate (-Math.abs (clip.x - startX), -Math.abs (clip.y - startY));
-
 	width = r.width;
 	height = r.height;
 	x = r.x;
