--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/DatagramChannelImpl.java	2003-09-26 02:20:57.000000000 -0600
+++ gnu/java/nio/DatagramChannelImpl.java	2003-09-26 02:18:34.000000000 -0600
@@ -64,6 +64,11 @@
     super (provider);
     socket = new NIODatagramSocket (new PlainDatagramSocketImpl(), this);
   }
+
+  public int getNativeFD()
+  {
+    return socket.getImpl().getNativeFD();
+  }
     
   public DatagramSocket socket ()
   {
