--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/rmi/server/RMIHashes.java	2004-10-03 02:35:32.000000000 -0600
+++ gnu/java/rmi/server/RMIHashes.java	2002-09-30 21:43:08.000000000 -0600
@@ -38,6 +38,7 @@
 package gnu.java.rmi.server;
 
 import java.lang.reflect.Method;
+import java.lang.Class;
 import gnu.java.io.NullOutputStream;
 import gnu.java.lang.reflect.TypeSignature;
 import java.security.MessageDigest;
