--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SelectionKeyImpl.java	Sat Nov 16 08:48:26 2002
+++ gnu/java/nio/SelectionKeyImpl.java	Fri Nov 29 02:57:05 2002
@@ -44,7 +44,9 @@
 
 public class SelectionKeyImpl extends AbstractSelectionKey
 {
-  int fd, ops;
+  int fd;
+  int readyOps;
+  int interestOps;
   SelectorImpl impl;
   SelectableChannel ch;
 
@@ -62,17 +64,23 @@
 
   public int readyOps ()
   {
-    return 0;
+    return readyOps;
+  }
+
+  public SelectionKey readyOps (int ops)
+  {
+    readyOps = ops;
+    return this;
   }
 
   public int interestOps ()
   {
-    return ops;    
+    return interestOps;    
   }
 
   public SelectionKey interestOps (int ops)
   {
-    this.ops = ops;
+    interestOps = ops;
     return this;
   }
     
