--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/DatagramChannelImpl.java	2003-11-27 02:20:12.000000000 -0700
+++ gnu/java/nio/DatagramChannelImpl.java	2003-12-03 02:18:10.000000000 -0700
@@ -56,7 +56,7 @@
 public final class DatagramChannelImpl extends DatagramChannel
 {
   private NIODatagramSocket socket;
-  private boolean blocking = false;
+  private boolean blocking = true;
   
   protected DatagramChannelImpl (SelectorProvider provider)
     throws IOException
