--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkFontPeer.java	2004-03-26 02:20:26.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkFontPeer.java	2004-03-18 02:18:19.000000000 -0700
@@ -43,7 +43,6 @@
 import java.awt.font.*;
 import java.util.Locale;
 import java.util.ResourceBundle;
-import java.util.MissingResourceException;
 import java.text.*;
 import gnu.java.awt.peer.ClasspathFontPeer;
 
@@ -75,20 +74,9 @@
   {
     super(name, style, size);
 
-    String Xname = null;
     if (bundle != null)
-      {
-	try
-	  {
-	    Xname = bundle.getString (name.toLowerCase () + "." + style);
-	  }
-	catch (MissingResourceException mre)
-	  {
-	    // ignored
-	  }
-      }
-
-    if (Xname == null)
+      Xname = bundle.getString (name.toLowerCase () + "." + style);
+    else
       {
 	String weight;
 	String slant;
@@ -110,8 +98,6 @@
 
         Xname = "-*-*-" + weight + "-" + slant + "-normal-*-*-" + size + "-*-*-" + spacing + "-*-*-*";
       }
-
-    this.Xname = Xname;
   }
 
   public String getXLFD ()
