--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/FileLockImpl.java	2004-04-13 02:20:14.000000000 -0600
+++ gnu/java/nio/FileLockImpl.java	2004-03-01 02:18:07.000000000 -0700
@@ -38,8 +38,9 @@
 
 package gnu.java.nio;
 
+import java.io.FileDescriptor;
 import java.io.IOException;
-import java.nio.channels.FileLock;
+import java.nio.channels.*;
 import gnu.classpath.Configuration;
 import gnu.java.nio.channels.FileChannelImpl;
 
@@ -49,6 +50,15 @@
  */
 public class FileLockImpl extends FileLock
 {
+  static
+  {
+    // load the shared library needed for native methods.
+    if (Configuration.INIT_LOAD_LIBRARY)
+      {
+        System.loadLibrary ("javanio");
+      }
+  }
+  
   private FileChannelImpl ch;
   
   public FileLockImpl (FileChannelImpl channel, long position,
