--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/text/CharacterBreakIterator.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/text/CharacterBreakIterator.java	2004-05-04 02:18:21.000000000 -0600
@@ -160,6 +160,7 @@
     if (iter.getIndex() == iter.getBeginIndex())
       return DONE;
 
+    int start = iter.getBeginIndex();
     while (iter.getIndex() >= iter.getBeginIndex())
       {
 	char c = iter.previous();
