--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/BeanInfoEmbryo.java	2004-05-21 02:25:38.000000000 -0600
+++ gnu/java/beans/BeanInfoEmbryo.java	2004-05-04 02:18:21.000000000 -0600
@@ -85,9 +85,9 @@
 
 		PropertyDescriptor[] Aproperties = new PropertyDescriptor[properties.size()];
 		int i = 0;
-		Enumeration e = properties.elements();
-		while (e.hasMoreElements()) {
-			Aproperties[i] = (PropertyDescriptor) e.nextElement();
+		Enumeration enum = properties.elements();
+		while(enum.hasMoreElements()) {
+			Aproperties[i] = (PropertyDescriptor)enum.nextElement();
 			if(defaultPropertyName != null && Aproperties[i].getName().equals(defaultPropertyName)) {
 				defaultProperty = i;
 			}
@@ -96,9 +96,9 @@
 
 		EventSetDescriptor[] Aevents = new EventSetDescriptor[events.size()];
 		i = 0;
-		e = events.elements();
-		while (e.hasMoreElements()) {
-			Aevents[i] = (EventSetDescriptor) e.nextElement();
+		enum = events.elements();
+		while(enum.hasMoreElements()) {
+			Aevents[i] = (EventSetDescriptor)enum.nextElement();
 			if(defaultEventName != null && Aevents[i].getName().equals(defaultEventName)) {
 				defaultEvent = i;
 			}
