--- /home/tromey/gnu/classpath/classpath/gnu/java/security/provider/MD5.java	Fri Jul 26 18:56:22 2002
+++ gnu/java/security/provider/MD5.java	Mon Aug 12 10:10:53 2002
@@ -1,5 +1,5 @@
 /* MD5.java -- Class implementing the MD5 algorithm as specified in RFC1321.
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,14 +37,14 @@
 
 
 package gnu.java.security.provider;
-import java.security.MessageDigest;
+import java.security.MessageDigestSpi;
 
 /**
    This class implements the MD5 algorithm as described in RFC1321.
 
    @see java.security.MessageDigest
 */
-public class MD5 extends MessageDigest implements Cloneable
+public class MD5 extends MessageDigestSpi implements Cloneable
 {
   private final int W[] = new int[16];
   private long bytecount;
@@ -55,7 +55,7 @@
 
   public MD5()
   {
-    super("MD5");
+    super();
     engineReset ();
   }
 
