--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/beans/BeanInfoEmbryo.java	2004-04-28 02:20:14.000000000 -0600
+++ gnu/java/beans/BeanInfoEmbryo.java	2002-03-24 16:11:02.000000000 -0700
@@ -38,17 +38,10 @@
 
 package gnu.java.beans;
 
-import java.beans.BeanDescriptor;
-import java.beans.BeanInfo;
-import java.beans.EventSetDescriptor;
-import java.beans.IndexedPropertyDescriptor;
-import java.beans.MethodDescriptor;
-import java.beans.PropertyDescriptor;
-import java.lang.reflect.Method;
-import java.util.Arrays;
-import java.util.Enumeration;
-import java.util.Hashtable;
-import java.util.Vector;
+import java.beans.*;
+import java.util.*;
+import gnu.java.lang.*;
+import java.lang.reflect.*;
 
 /**
  ** A BeanInfoEmbryo accumulates information about a Bean
