--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/EmbeddedWindow.java	2004-07-29 02:25:34.000000000 -0600
+++ gnu/java/awt/EmbeddedWindow.java	2004-07-18 02:18:48.000000000 -0600
@@ -40,13 +40,11 @@
 
 import gnu.classpath.Configuration;
 import gnu.java.awt.peer.EmbeddedWindowPeer;
-import gnu.java.security.action.SetAccessibleAction;
 
 import java.awt.Component;
 import java.awt.Frame;
 import java.awt.Toolkit;
 import java.lang.reflect.Field;
-import java.security.AccessController;
 
 /**
  * Represents an AWT window that can be embedded into another
@@ -90,15 +88,13 @@
 
     if (! (tk instanceof EmbeddedWindowSupport))
       throw new UnsupportedOperationException
-        ("Embedded windows are not supported by the current peers: "
-	 + tk.getClass());
+        ("Embedded windows are not supported by the current peers: " + tk.getClass());
 
     // Circumvent the package-privateness of the AWT internal
     // java.awt.Component.peer member variable.
     try
       {
 	Field peerField = Component.class.getDeclaredField("peer");
-	AccessController.doPrivileged(new SetAccessibleAction(peerField));
 	peerField.set(this, ((EmbeddedWindowSupport) tk).createEmbeddedWindow (this));
       }
     catch (IllegalAccessException e)
