--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/image/XBMDecoder.java	2002-01-22 15:26:56.000000000 -0700
+++ gnu/java/awt/image/XBMDecoder.java	2003-11-12 02:18:12.000000000 -0700
@@ -62,7 +62,7 @@
     super (url);
   }
 
-  public void produce (Vector v, FileInputStream is) throws IOException
+  public void produce (Vector v, InputStream is) throws IOException
   {
     reader = new BufferedReader (new InputStreamReader (is));
     int width = -1, height = -1;
