--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkMenuPeer.java	2004-07-24 02:24:52.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkMenuPeer.java	2004-04-17 02:18:07.000000000 -0600
@@ -52,15 +52,11 @@
   native void create (String label);
   native void addItem (MenuItemPeer item, int key, boolean shiftModifier);
   native void setupAccelGroup (GtkGenericPeer container);
-  native void addTearOff ();
 
   public GtkMenuPeer (Menu menu)
   {
     super (menu);
     
-    if (menu.isTearOff())
-      addTearOff();
-
     MenuContainer parent = menu.getParent ();
     if (parent instanceof Menu)
       setupAccelGroup ((GtkGenericPeer)((Menu)parent).getPeer ());
