--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GdkFontMetrics.java	2003-12-02 02:19:59.000000000 -0700
+++ gnu/java/awt/peer/gtk/GdkFontMetrics.java	2004-06-18 02:18:49.000000000 -0600
@@ -50,19 +50,21 @@
                        MAX_ADVANCE = 4;
 
   private int[] metrics;
-  private native int[] initState (String fname, int size);
+  private native int[] initState (String fname, int style, int size);
 
   public GdkFontMetrics (Font font)
   {
     super (font);
-    metrics = initState (font.getName (), font.getSize ());
+    metrics = initState (font.getName (), font.getStyle (), font.getSize ());
   }
 
-  native public int stringWidth (String fname, int size, String str);
+  native public int stringWidth (String fname, int style, int size,
+                                 String str);
 
   public int stringWidth (String str)
   {
-    return stringWidth (font.getName (), font.getSize (), str);
+    return stringWidth (font.getName (), font.getStyle (), font.getSize (),
+                        str);
   }
 
   public int charWidth (char ch)
