--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SocketChannelImpl.java	2003-06-16 02:23:37.000000000 -0600
+++ gnu/java/nio/SocketChannelImpl.java	2003-06-15 02:21:55.000000000 -0600
@@ -35,6 +35,7 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package gnu.java.nio;
 
 import java.io.IOException;
@@ -50,7 +51,7 @@
 
 public class SocketChannelImpl extends SocketChannel
 {
-  Socket sock_object;
+  Socket socket;
   int fd;
   int local_port;
   boolean blocking = true;
@@ -141,20 +142,17 @@
     
   public boolean isConnectionPending ()
   {
-    if (blocking)
-	    return true;
-
-    return false;
+    return blocking ? true : false;
   }
     
   public Socket socket ()
   {
-    if (sock_object != null)
+    if (socket != null)
 	    {
-        //sock_object.ch = this;
+        //socket.ch = this;
 	    }
 
-    return sock_object;
+    return socket;
   }
 
   public int read (ByteBuffer dst) throws IOException
