--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkImagePainter.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkImagePainter.java	2003-01-06 20:37:27.000000000 -0700
@@ -35,14 +35,11 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package gnu.java.awt.peer.gtk;
 
-import java.awt.Color;
-import java.awt.Rectangle;
-import java.awt.image.ColorModel;
-import java.awt.image.ImageConsumer;
-import java.util.Hashtable;
+import java.awt.*;
+import java.awt.image.*;
+import java.util.*;
 
 public class GtkImagePainter implements Runnable, ImageConsumer
 {
