--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/net/protocol/http/Connection.java	2004-03-21 02:20:43.000000000 -0700
+++ gnu/java/net/protocol/http/Connection.java	2004-01-29 02:18:49.000000000 -0700
@@ -215,8 +215,8 @@
       setRequestProperty ("Connection", "Close");
     
     if (getRequestProperty ("user-agent") == null)
-      setRequestProperty ("user-agent", "gnu-classpath/"
-                          + System.getProperty ("classpath.version"));
+      setRequestProperty ("user-agent", "gnu-libgcj/"
+                          + System.getProperty ("java.vm.version"));
     
     if (getRequestProperty ("accept") == null)
       setRequestProperty ("accept", "*/*");
