--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextFieldPeer.java	2003-07-10 02:18:54.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextFieldPeer.java	2003-07-08 02:18:21.000000000 -0600
@@ -37,13 +37,9 @@
 
 
 package gnu.java.awt.peer.gtk;
-
-import java.awt.AWTEvent;
-import java.awt.Dimension;
-import java.awt.Font;
-import java.awt.TextField;
 import java.awt.event.KeyEvent;
-import java.awt.peer.TextFieldPeer;
+import java.awt.peer.*;
+import java.awt.*;
 
 public class GtkTextFieldPeer extends GtkTextComponentPeer
   implements TextFieldPeer
@@ -107,4 +103,18 @@
   {
     gtkSetFont(((GtkFontPeer)f.getPeer()).getXLFD(), f.getSize());
   }
+
+  public void handleEvent (AWTEvent e)
+  {
+    if (e.getID () == KeyEvent.KEY_PRESSED)
+      {
+        KeyEvent ke = (KeyEvent)e;
+
+        if (!ke.isConsumed()
+            && ke.getKeyCode() == KeyEvent.VK_ENTER)
+          postActionEvent (getText(), ke.getModifiers ());
+      }
+
+    super.handleEvent (e);
+  }
 }
