--- /home/tromey/gnu/classpath/classpath/gnu/java/rmi/server/RMIObjectOutputStream.java	Thu Oct 31 12:43:09 2002
+++ gnu/java/rmi/server/RMIObjectOutputStream.java	Mon Sep 30 16:13:16 2002
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -72,9 +72,10 @@
         throws IOException
 {
     if((obj instanceof Remote) && !(obj instanceof RemoteStub)){
-	    UnicastServerRef ref = UnicastServer.getExportedRef((Remote)obj);
-	    if (ref != null)
-		    return ref.getStub();
+	    UnicastServerRef ref = new UnicastServerRef(new ObjID(), 0, null);
+		try{
+		    return ref.exportObject((Remote)obj);
+		}catch(Exception e){}
     }
     return obj;
 }
