--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkToolkit.java	2004-03-26 02:20:26.000000000 -0700
+++ gnu/java/awt/peer/gtk/GtkToolkit.java	2004-03-18 02:18:19.000000000 -0700
@@ -377,8 +377,12 @@
    */
   private FontPeer getFontPeer (String name, int style, int size) 
   {
-    GtkFontPeer fp = new GtkFontPeer (name, style, size);
-    return fp;
+    try {
+      GtkFontPeer fp = new GtkFontPeer (name, style, size);
+      return fp;
+    } catch (MissingResourceException ex) {
+      return null;
+    }
   }
 
   /**
