--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkToolkit.java	2003-07-02 02:19:04.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkToolkit.java	2003-01-31 09:59:29.000000000 -0700
@@ -39,20 +39,18 @@
 package gnu.java.awt.peer.gtk;
 
 import java.awt.*;
-import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.*;
 import java.awt.dnd.DragGestureEvent;
 import java.awt.dnd.peer.DragSourceContextPeer;
 import java.awt.im.InputMethodHighlight;
-import java.awt.image.ColorModel;
-import java.awt.image.ImageObserver;
-import java.awt.image.ImageProducer;
+import java.awt.image.*;
 import java.awt.peer.*;
-import java.net.URL;
+import java.net.*;
 import java.util.Hashtable;
 import java.util.Map;
 import java.util.MissingResourceException;
 import java.util.Properties;
-import gnu.java.awt.image.GdkPixbufDecoder;
+import gnu.java.awt.image.*;
 import gnu.classpath.Configuration;
 
 /* This class uses a deprecated method java.awt.peer.ComponentPeer.getPeer().
@@ -96,12 +94,16 @@
 
   public Image createImage (String filename)
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image createImage (URL url)
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public Image createImage (ImageProducer producer) 
@@ -137,12 +139,16 @@
 
   public Image getImage (String filename) 
   {
-    return new GtkImage (new GdkPixbufDecoder (filename), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (filename), null);
+    return null;
   }
 
   public Image getImage (URL url) 
   {
-    return new GtkImage (new GdkPixbufDecoder (url), null);
+    // FIXME - gcj local: GdkPixbufDecoder doesn't work.
+    // return new GtkImage (new GdkPixbufDecoder (url), null);
+    return null;
   }
 
   public PrintJob getPrintJob (Frame frame, String jobtitle, Properties props) 
