--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/nio/SelectorImpl.java	2003-09-27 02:18:59.000000000 -0600
+++ gnu/java/nio/SelectorImpl.java	2003-10-10 02:18:09.000000000 -0600
@@ -253,17 +253,17 @@
     if (ch instanceof SocketChannelImpl)
       {
         SocketChannelImpl sc = (SocketChannelImpl) ch;
-        result = new SocketChannelSelectionKey (ch, this); // FIXME: last argument
+        result = new SocketChannelSelectionKey (ch, this);
       }
     else if (ch instanceof DatagramChannelImpl)
       {
         DatagramChannelImpl dc = (DatagramChannelImpl) ch;
-        result = new DatagramChannelSelectionKey (ch, this); // FIXME: last argument
+        result = new DatagramChannelSelectionKey (ch, this);
       }
     else if (ch instanceof ServerSocketChannelImpl)
       {
         ServerSocketChannelImpl ssc = (ServerSocketChannelImpl) ch;
-        result = new SocketChannelSelectionKey (ch, this); // FIXME: last argument
+        result = new ServerSocketChannelSelectionKey (ch, this);
       }
     else
       {
