--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/locale/LocaleInformation_en.java	2004-07-26 02:25:01.000000000 -0600
+++ gnu/java/locale/LocaleInformation_en.java	2003-07-14 02:18:13.000000000 -0600
@@ -39,6 +39,8 @@
 package gnu.java.locale;
 
 import java.util.ListResourceBundle;
+import java.util.Calendar;
+import java.util.Date;
 
 /**
  * This class contains locale data for English.
