--- /home/tromey/gnu/Nightly/classpath/classpath/gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2004-04-30 02:23:29.000000000 -0600
+++ gnu/java/awt/peer/gtk/GtkTextAreaPeer.java	2004-05-27 02:18:56.000000000 -0600
@@ -50,6 +50,7 @@
   native void create (int width, int height, int scrollbarVisibility);
 
   native void gtkSetFont (String name, int style, int size);
+  native void gtkWidgetRequestFocus ();
 
   void create ()
   {
@@ -171,9 +172,4 @@
   {
     insert (str, pos);
   }
-
-  public void setFont (Font f)
-  {
-    gtkSetFont (f.getName (), f.getStyle (), f.getSize ());
-  }
 }
