--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSlider.java	2004-04-20 02:21:39.000000000 -0600
+++ javax/swing/JSlider.java	2004-04-17 02:18:07.000000000 -0600
@@ -71,7 +71,7 @@
  * </p>
  * 
  * <table>
- * <tr><th> Property         </th><th> Stored in </th><th> Bound? </th></tr>
+ * <tr><th> Property         </td><th> Stored in </td><th> Bound? </td></tr>
  * <tr><td> extent           </td><td> model     </td><td> no     </td></tr>
  * <tr><td> inverted         </td><td> slider    </td><td> yes    </td></tr>
  * <tr><td> labelTable       </td><td> slider    </td><td> yes    </td></tr>
