--- /home/tromey/gnu/egcs/classpath/classpath/java/util/ResourceBundle.java	Sun Nov  4 14:58:58 2001
+++ java/util/ResourceBundle.java	Mon Nov  5 14:36:58 2001
@@ -93,7 +93,7 @@
   /**
    * We override SecurityManager in order to access getClassContext(). 
    */
-  class Security extends SecurityManager
+  static class Security extends SecurityManager
   {
     /** Return the ClassLoader of the class which called into this
         ResourceBundle, or null if it cannot be determined. */
