--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2003-06-28 02:19:17.000000000 -0600
+++ java/net/URLStreamHandler.java	2003-09-12 02:17:59.000000000 -0600
@@ -196,7 +196,11 @@
             // need to canonicalise the file path.
             try
               {
+		boolean endsWithSlash = file.charAt(file.length() - 1) == '/';
                 file = new File (file).getCanonicalPath ();
+		if (endsWithSlash
+		    && file.charAt(file.length() - 1) != '/')
+		  file += '/';
               }
             catch (IOException e)
               {
