--- /home/tromey/gnu/Nightly/classpath/classpath/javax/crypto/MacSpi.java	2004-09-29 02:44:30.000000000 -0600
+++ javax/crypto/MacSpi.java	2004-09-21 12:37:44.000000000 -0600
@@ -52,7 +52,7 @@
  * then provide an entry pointing to this implementation in the master
  * {@link java.security.Provider} class.
  *
- * <p>Implementations may optionally implement the {@link
+ * <p>Implemetations may optionally implement the {@link
  * java.lang.Cloneable} interface.
  *
  * @author Casey Marshall (csm@gnu.org)
