--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/HashAttributeSet.java	2004-04-28 02:20:18.000000000 -0600
+++ javax/print/attribute/HashAttributeSet.java	2004-04-21 02:18:12.000000000 -0600
@@ -1,5 +1,5 @@
 /* HashAttributeSet.java -- 
-   Copyright (C) 2003, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -232,7 +232,7 @@
    *
    * @param category the category to test for
    *
-   * @return true if the category exists in this attribute set, false otherwise.
+   * @result true if the category exists in this attribute set, false otherwise.
    */
   public boolean containsKey(Class category)
   {
@@ -244,7 +244,7 @@
    *
    * @param attribute the attribute to test for
    *
-   * @return true if the attribute exists in this attribute set,
+   * @result true if the attribute exists in this attribute set,
    * false otherwise.
    */
   public boolean containsValue(Attribute attribute)
@@ -257,7 +257,7 @@
    *
    * @param obj the object to test
    *
-   * @return true if both objects are equal, false otherwise.
+   * @returns true if both objects are equal, false otherwise.
    */
   public boolean equals(Object obj)
   {
