--- /home/tromey/gnu/egcs/classpath/classpath/java/util/AbstractSet.java	Sun Nov 25 12:30:06 2001
+++ java/util/AbstractSet.java	Sat Dec 15 09:15:59 2001
@@ -115,7 +115,7 @@
     Iterator i;
     if (oldsize < count)
       {
-        for (i = iterator(), count = oldsize; count > 0; count--)
+	for (i = iterator(), count = oldsize; count > 0; count--)
           if (c.contains(i.next()))
             i.remove();
       }
