--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RMIClassLoaderSpi.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/server/RMIClassLoaderSpi.java	2003-01-01 16:53:58.000000000 -0700
@@ -1,5 +1,5 @@
-/* RMIClassLoaderSpi.java --
-   Copyright (c) 2002 Free Software Foundation, Inc.
+/* RMIClassLoaderSpi.java
+  Copyright (c) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
