--- /home/tromey/gnu/Nightly/classpath/classpath/java/beans/PropertyEditor.java	2004-04-18 02:22:14.000000000 -0600
+++ java/beans/PropertyEditor.java	2003-10-13 02:17:50.000000000 -0600
@@ -77,7 +77,7 @@
  **       have isPaintable() return true and implement the paintValue() method.
  **       This method does not determine in any way how the value is edited;
  **       merely how it is displayed.</LI>
- **   <LI>Let the caller of the PropertyEditor give the user a text input.  Do
+ **   <LU>Let the caller of the PropertyEditor give the user a text input.  Do
  **       this by returning a non-null String from getAsText().  If you support
  **       text input, you *must* support setAsText().</LI>
  **   <LI>Give the caller a set of possible values, such as "true"/"false", that
