--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/MediaTracker.java	2003-07-25 02:18:53.000000000 -0600
+++ java/awt/MediaTracker.java	2003-11-12 02:18:13.000000000 -0700
@@ -88,8 +88,8 @@
       {
 	MediaTracker.this.notifyAll();
       }
-      
-      return ((status & COMPLETE) != 0);
+      // If status is not COMPLETE then we need more updates.
+      return (status & COMPLETE) == 0;
     }
   }
 
