--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/PlainDocument.java	2002-04-30 15:37:27.000000000 -0600
+++ javax/swing/text/PlainDocument.java	2004-06-18 02:18:59.000000000 -0600
@@ -1,5 +1,5 @@
-/* PlainDocument.java -- 
-   Copyright (C) 2002 Free Software Foundation, Inc.
+/* PlainDocument.java --
+   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,22 +37,33 @@
 
 package javax.swing.text;
 
-
 public class PlainDocument extends AbstractDocument
 {
-    PlainDocument()
-    {
-	super(new GapContent());
-    }
-
-    public Element getDefaultRootElement()
-    {
-	return null;
-    }
-
-    public Element getParagraphElement(int  pos)
-    {
-	return null;
-    }
+  private static final long serialVersionUID = 4758290289196893664L;
+    
+  public static final String lineLimitAttribute = "lineLimit";
+  public static final String tabSizeAttribute = "tabSize";
+
+  private int tabSize;
+  
+  public PlainDocument()
+  {
+    this(new GapContent());
+  }
+
+  public PlainDocument(AbstractDocument.Content content)
+  {
+    super(content);
+    tabSize = 8;
+  }
+
+  public Element getDefaultRootElement()
+  {
+    return null;
+  }
+
+  public Element getParagraphElement(int pos)
+  {
+    return null;
+  }
 }
-
