--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/InflaterInputStream.java	2004-06-21 06:24:02.000000000 -0600
+++ java/util/zip/InflaterInputStream.java	2004-07-31 02:18:21.000000000 -0600
@@ -70,10 +70,6 @@
    */
   protected int len;
 
-  /*
-   * We just use this if we are decoding one byte at a time with the read() call
-   */
-  private byte[] onebytebuffer = new byte[1];
 
   /**
    * Create an InflaterInputStream with the default decompresseor
@@ -139,9 +135,8 @@
    */
   public synchronized void close() throws IOException
   {
-    if (in != null)
-      in.close();
-    in = null;
+    inf = null;
+    super.close();
   }
 
   /**
@@ -161,21 +156,6 @@
   }
 
   /**
-   * Reads one byte of decompressed data.
-   *
-   * The byte is in the lower 8 bits of the int.
-   */
-  public int read() throws IOException
-  { 
-    int nread = read(onebytebuffer, 0, 1); //read one byte
-    
-    if (nread > 0)
-      return onebytebuffer[0] & 0xff;
-    
-    return -1;
-  }
-
-  /**
    * Decompresses data into the byte array
    *
    * @param b the array to read and decompress data into
@@ -188,14 +168,28 @@
       throw new IOException("stream closed");
     if (len == 0)
       return 0;
+    if (inf.finished())
+      return -1;
 
     int count = 0;
     for (;;)
       {
+	if (inf.needsInput())
+	  fill();
 	
 	try
 	  {
 	    count = inf.inflate(b, off, len);
+	    if (count == 0)
+	      {
+		if (this.len == -1)
+		  {
+		    // Couldn't get any more data to feed to the Inflater
+		    return -1;
+		  }
+		if (inf.needsDictionary())
+		  throw new ZipException("Inflater needs Dictionary");
+	      }
 	  } 
 	catch (DataFormatException dfe) 
 	  {
@@ -204,14 +198,6 @@
 
 	if (count > 0)
 	  return count;
-	
-	if (inf.needsDictionary()
-	    | inf.finished())
-	  return -1;
-	else if (inf.needsInput())
-	  fill();
-	else
-	  throw new InternalError("Don't know what to do");
       }
   }
 
