--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-01-28 02:27:14.000000000 -0700
+++ javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-02-05 02:18:45.000000000 -0700
@@ -614,8 +614,8 @@
       b.getToolkit().getFontMetrics(b.getFont()), // see comment above
       b.getText(),
       b.getIcon(),
-      b.getVerticalAlignment(),
-      b.getHorizontalAlignment(),
+      SwingUtilities.TOP,    // important:
+      SwingUtilities.LEFT,   // large vrect, stick to the top left
       b.getVerticalTextPosition(),
       b.getHorizontalTextPosition(),
       viewRect, iconRect, textRect,
