--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/metal/MetalLookAndFeel.java	2003-07-02 02:19:18.000000000 -0600
+++ javax/swing/plaf/metal/MetalLookAndFeel.java	2003-05-10 02:19:19.000000000 -0600
@@ -36,14 +36,14 @@
 exception statement from your version. */
 
 
-
 package javax.swing.plaf.metal;
 
-import javax.swing.UIDefaults;
-import javax.swing.plaf.basic.BasicDefaults;
-import javax.swing.plaf.basic.BasicLookAndFeel;
+import javax.swing.*;
+import javax.swing.plaf.*;
+import javax.swing.plaf.basic.*;
+
 
-public class MetalLookAndFeel extends BasicLookAndFeel
+public class MetalLookAndFeel extends LookAndFeel
  {	   
      public boolean isNativeLookAndFeel()        { return true; }
      public boolean isSupportedLookAndFeel()     { return true; }
