--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/HashAttributeSet.java	2004-04-19 02:20:15.000000000 -0600
+++ javax/print/attribute/HashAttributeSet.java	2003-12-24 02:18:19.000000000 -0700
@@ -151,7 +151,7 @@
    * @exception ClassCastException if any element of attributes is not an
    * interface of interfaceName
    */
-  protected HashAttributeSet(AttributeSet attributes, Class interfaceName)
+  public HashAttributeSet(AttributeSet attributes, Class interfaceName)
   {
     this(interfaceName);
     
