--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/MutableAttributeSet.java	2004-01-11 02:20:39.000000000 -0700
+++ javax/swing/text/MutableAttributeSet.java	2004-06-18 02:18:59.000000000 -0600
@@ -44,48 +44,42 @@
  * @author	Andrew Selkirk
  * @version	1.0
  */
-public interface MutableAttributeSet extends AttributeSet {
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * addAttribute
-	 * @param name TODO
-	 * @param value TODO
-	 */
-	void addAttribute(Object name, Object value);
-
-	/**
-	 * addAttributes
-	 * @param attributes TODO
-	 */
-	void addAttributes(AttributeSet attributes);
-
-	/**
-	 * removeAttribute
-	 * @param name TODO
-	 */
-	void removeAttribute(Object name);
-
-	/**
-	 * removeAttributes
-	 * @param names TODO
-	 */
-	void removeAttributes(Enumeration names);
-
-	/**
-	 * removeAttributes
-	 * @param attributes TODO
-	 */
-	void removeAttributes(AttributeSet attributes);
-
-	/**
-	 * setResolveParent
-	 * @param parent TODO
-	 */
-	void setResolveParent(AttributeSet parent);
-
-
-} // MutableAttributeSet
+public interface MutableAttributeSet extends AttributeSet
+{
+  /**
+   * addAttribute
+   * @param name TODO
+   * @param value TODO
+   */
+  void addAttribute(Object name, Object value);
+
+  /**
+   * addAttributes
+   * @param attributes TODO
+   */
+  void addAttributes(AttributeSet attributes);
+
+  /**
+   * removeAttribute
+   * @param name TODO
+   */
+  void removeAttribute(Object name);
+
+  /**
+   * removeAttributes
+   * @param names TODO
+   */
+  void removeAttributes(Enumeration names);
+
+  /**
+   * removeAttributes
+   * @param attributes TODO
+   */
+  void removeAttributes(AttributeSet attributes);
+
+  /**
+   * setResolveParent
+   * @param parent TODO
+   */
+  void setResolveParent(AttributeSet parent);
+}
