--- /home/tromey/gnu/classpath/classpath/java/awt/TextArea.java	Mon Dec 30 15:39:04 2002
+++ java/awt/TextArea.java	Tue Jan 22 16:49:58 2002
@@ -109,8 +109,6 @@
   * Initialize a new instance of <code>TextArea</code> that is empty
   * and is one row and one column.  Both horizontal and vertical
   * scrollbars will be used.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextArea()
@@ -126,8 +124,6 @@
   * scrollbars will be used.
   *
   * @param text The text to display in this text area.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextArea(String text)
@@ -144,8 +140,6 @@
   *
   * @param rows The number of rows in this text area.
   * @param columns The number of columns in this text area.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextArea(int rows, int columns)
@@ -162,8 +156,6 @@
   * @param text The text to display in this text area.
   * @param rows The number of rows in this text area.
   * @param columns The number of columns in this text area.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextArea(String text, int rows, int columns)
@@ -182,17 +174,12 @@
   * @param rows The number of rows in this text area.
   * @param columns The number of columns in this text area.
   * @param scrollbarVisibility Which scrollbars to display.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextArea(String text, int rows, int columns, int scrollbarVisibility)
 {
   super(text);
 
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
-
   if ((rows < 1) || (columns < 0))
     throw new IllegalArgumentException("Bad row or column value");
 
