--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultBoundedRangeModel.java	2002-10-02 06:58:40.000000000 -0600
+++ javax/swing/DefaultBoundedRangeModel.java	2003-03-31 02:17:51.000000000 -0700
@@ -337,5 +337,14 @@
 		return listenerList.getListeners(c);
 	} // getListeners()
 
+	/**
+	 * getChangeListeners
+	 */
+	public ChangeListener[] getChangeListeners()
+	{
+	  // FIXME: implement this
+	  return null;
+	}
+
 
 } // DefaultBoundedRangeModel
