--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Observer.java	2002-02-21 19:09:40.000000000 -0700
+++ java/util/Observer.java	2003-10-13 02:17:51.000000000 -0600
@@ -56,5 +56,5 @@
    * @param observable the Observable object that changed
    * @param arg arbitrary information, usually relating to the change
    */
-  public void update(Observable observable, Object arg);
+  void update(Observable observable, Object arg);
 }
