--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/border/CompoundBorder.java	2003-06-17 02:18:48.000000000 -0600
+++ javax/swing/border/CompoundBorder.java	2003-06-13 02:27:43.000000000 -0600
@@ -51,10 +51,6 @@
 public class CompoundBorder
   extends AbstractBorder
 {
-  /**
-   * Determined using the <code>serialver</code> tool
-   * of Apple/Sun JDK 1.3.1 on MacOS X 10.1.5.
-   */
   static final long serialVersionUID = 9054540377030555103L;
 
 
@@ -94,14 +90,14 @@
    * outside borders.
    *
    * @param outsideBorder the outside border, which is painted to the
-   *        outside of both <code>insideBorder</code> and the enclosed
-   *        component. It is acceptable to pass <code>null</code>, in
-   *        which case no outside border is painted.
+   *        outside of both <code>insideBorder</code> and the bordered
+   *        compoonent. It is acceptable to pass <code>null</code>, in
+   *        which no outside border is painted.
    *
    * @param insideBorder the inside border, which is painted to
-   *        between <code>outsideBorder</code> and the enclosed
+   *        between <code>outsideBorder</code> and the bordered
    *        component. It is acceptable to pass <code>null</code>, in
-   *        which case no inside border is painted.
+   *        which no intside border is painted.
    */
   public CompoundBorder (Border outsideBorder, Border insideBorder)
   {
