--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectOutputStream.java	2004-10-03 02:35:37.000000000 -0600
+++ java/io/ObjectOutputStream.java	2004-07-24 02:18:38.000000000 -0600
@@ -1,6 +1,5 @@
 /* ObjectOutputStream.java -- Class used to write serialized objects
-   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003, 2004
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,18 +38,19 @@
 
 package java.io;
 
-import gnu.classpath.Configuration;
-import gnu.java.io.ObjectIdentityWrapper;
-import gnu.java.lang.reflect.TypeSignature;
-import gnu.java.security.action.SetAccessibleAction;
-
 import java.lang.reflect.Array;
 import java.lang.reflect.Field;
-import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.lang.reflect.InvocationTargetException;
+import java.security.PrivilegedAction;
 import java.security.AccessController;
 import java.util.Hashtable;
 
+import gnu.java.io.ObjectIdentityWrapper;
+import gnu.java.lang.reflect.TypeSignature;
+import gnu.java.security.action.SetAccessibleAction;
+import gnu.classpath.Configuration;
+
 /**
  * An <code>ObjectOutputStream</code> can be used to write objects
  * as well as primitive data in a platform-independent manner to an
