--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultListModel.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/DefaultListModel.java	2004-03-18 02:18:22.000000000 -0700
@@ -1,5 +1,5 @@
 /* DefaultListModel.java --
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,6 +38,7 @@
 package javax.swing;
 
 import java.util.Enumeration;
+import java.util.NoSuchElementException;
 import java.util.Vector;
 
 /**
@@ -47,7 +48,7 @@
  * java.util.Vector}.
  *
  * @author Andrew Selkirk
- * @author Graydon Hoare (graydon@redhat.com)
+ * @author Graydon Hoare (graydon&064;redhat.com)
  */
 
 public class DefaultListModel extends AbstractListModel
