--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/SecurityManager.java	2004-01-07 02:20:41.000000000 -0700
+++ java/lang/SecurityManager.java	2004-04-02 02:18:41.000000000 -0700
@@ -861,8 +861,7 @@
    */
   public void checkAwtEventQueueAccess()
   {
-    // Should be: checkPermission(new AWTPermission("accessEventQueue"));
-    throw new SecurityException("Cannot access the AWT event queue.");
+    checkPermission(new AWTPermission("accessEventQueue"));
   }
 
   /**
