--- /home/tromey/gnu/egcs/classpath/classpath/java/util/TimerTask.java	Mon Jun  3 14:01:56 2002
+++ java/util/TimerTask.java	Tue Jan 22 16:01:44 2002
@@ -44,11 +44,11 @@
  * it should have been scheduled and cancel itself when no longer needed.
  * <p>
  * Example:
- * <pre>
+ * <code>
  *  Timer timer = new Timer();
  *  TimerTask task = new TimerTask() {
  *      public void run() {
- *      if (this.scheduledExecutionTime() &lt; System.currentTimeMillis() + 500)
+ *      if (this.scheduledExecutionTime() < System.currentTimeMillis() + 500)
  *          // Do something
  *      else
  *          // Complain: We are more then half a second late!
@@ -56,7 +56,7 @@
  *          this.cancel(); // This was our last execution
  *  };
  *  timer.scheduleAtFixedRate(task, 1000, 1000); // schedule every second
- * </pre>
+ * </code>
  * <p>
  * Note that a TimerTask object is a one shot object and can only given once
  * to a Timer. (The Timer will use the TimerTask object for bookkeeping,
