--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/NetworkInterface.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/NetworkInterface.java	2004-05-04 02:18:23.000000000 -0600
@@ -130,12 +130,15 @@
   }
 
   /**
-   *  Returns an network interface by name
+   * Returns an network interface by name
    *
-   *  @param name The name of the interface to return
+   * @param name The name of the interface to return
+   * 
+   * @return a <code>NetworkInterface</code> object representing the interface,
+   * or null if there is no interface with that name.
    *
-   *  @exception SocketException If an error occurs
-   *  @exception NullPointerException If the specified name is null
+   * @exception SocketException If an error occurs
+   * @exception NullPointerException If the specified name is null
    */
   public static NetworkInterface getByName(String name)
     throws SocketException
@@ -150,7 +153,8 @@
 	  return tmp;
       }
 
-    throw new SocketException("no network interface with this name exists");
+    // No interface with the given name found.
+    return null;
   }
 
   /**
