--- /home/tromey/gnu/classpath/classpath/java/nio/channels/ScatteringByteChannel.java	Tue Apr 30 17:58:37 2002
+++ java/nio/channels/ScatteringByteChannel.java	Mon Oct  7 01:59:12 2002
@@ -37,11 +37,43 @@
 
 package java.nio.channels;
 
-import java.nio.*;
-import java.io.*;
+import java.nio.ByteBuffer;
+import java.io.IOException;
 
-public interface ScatteringByteChannel extends ReadableByteChannel
+public interface ScatteringByteChannel
+  extends ReadableByteChannel
 {
-    public long write(ByteBuffer[] srcs, int offset, int length) throws IOException;
-    public int write(ByteBuffer[] srcs) throws IOException;
+  /**
+   * Reads a sequence of bytes from this channel into a subsequence of the
+   * given buffers
+   *
+   * @exception AsynchronousCloseException If another thread closes this
+   * channel while the write operation is in progress
+   * @exception ClosedByInterruptException If another thread interrupts the
+   * current thread while the write operation is in progress, thereby closing
+   * the channel and setting the current thread's interrupt status
+   * @exception ClosedChannelException If this channel is closed
+   * @exception IndexOutOfBoundsException If the preconditions on the offset
+   * and length parameters do not hold
+   * @exception IOException If an error occurs
+   * @exception NonReadableChannelException If this channel was not opened for
+   * reading
+   */
+  public long read(ByteBuffer[] srcs, int offset, int length)
+    throws IOException;
+
+  /**
+   * Reads a sequence of bytes from this channel into the given buffers
+   *
+   * @exception AsynchronousCloseException If another thread closes this
+   * channel while the write operation is in progress
+   * @exception ClosedByInterruptException If another thread interrupts the
+   * current thread while the write operation is in progress, thereby closing
+   * the channel and setting the current thread's interrupt status
+   * @exception ClosedChannelException If this channel is closed
+   * @exception IOException If an error occurs
+   * @exception NonReadableChannelException If this channel was not opened for
+   * reading
+   */
+  public long read(ByteBuffer[] srcs) throws IOException;
 }
