--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/InternalError.java	Sat Mar  2 12:13:12 2002
+++ java/lang/InternalError.java	Tue Jan 22 16:01:34 2002
@@ -1,5 +1,5 @@
-/* InternalError.java -- thrown when the VM encounters an internal error
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* InternalError.java 
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,35 +38,39 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct. 
+ */
+
 /**
  * An <code>InternalError</code> is thrown when a mystical error has
  * occurred in the Java Virtual Machine.
  *
+ * @since JDK 1.0
+ * 
  * @author Brian Jones
  * @author Tom Tromey <tromey@cygnus.com>
- * @status updated to 1.4
+ * @date October 1, 1998
  */
 public class InternalError extends VirtualMachineError
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = -9062593416125562365L;
+  static final long serialVersionUID = -9062593416125562365L;
 
   /**
    * Create an error without a message.
    */
   public InternalError()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an error with a message.
-   *
-   * @param s the message
    */
   public InternalError(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
