--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FilterInputStream.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/FilterInputStream.java	2003-03-24 02:17:53.000000000 -0700
@@ -104,7 +104,7 @@
     */
   public boolean markSupported()
   {
-    return in.markSupported() ;
+    return(in.markSupported());
   }
 
   /**
@@ -126,7 +126,7 @@
     */
   public int available() throws IOException
   {
-    return in.available() ;
+    return(in.available());
   }
 
   /**
@@ -138,9 +138,9 @@
     *
     * @exception IOException If an error occurs
     */
-  public long skip(long numBytes) throws IOException
+  public long skip(long num_bytes) throws IOException
   {
-    return in.skip(numBytes);
+    return(in.skip(num_bytes));
   }
 
   /**
@@ -152,7 +152,7 @@
     */
   public int read() throws IOException
   {
-    return in.read();
+    return(in.read());
   }
 
   /**
@@ -170,7 +170,7 @@
     */
   public int read(byte[] buf) throws IOException
   {
-    return read(buf, 0, buf.length);
+    return(read(buf, 0, buf.length));
   }
 
   /**
@@ -186,7 +186,7 @@
     */
   public int read(byte[] buf, int offset, int len) throws IOException
   {
-    return in.read(buf, offset, len);
+    return(in.read(buf, offset, len));
   }
 
   /**
@@ -200,4 +200,6 @@
   {
     in.close();
   }
-}
+
+} // class FilterInputStream
+
