--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/StringTokenizer.java	2004-04-18 02:22:27.000000000 -0600
+++ java/util/StringTokenizer.java	2002-06-17 23:01:42.000000000 -0600
@@ -196,7 +196,7 @@
 
   /**
    * This does the same as hasMoreTokens. This is the
-   * <code>Enumeration</code> interface method.
+   * <code>Enumeration</code interface method.
    *
    * @return true, if the next call of nextElement() will succeed
    * @see #hasMoreTokens()
@@ -208,7 +208,7 @@
 
   /**
    * This does the same as nextTokens. This is the
-   * <code>Enumeration</code> interface method.
+   * <code>Enumeration</code interface method.
    *
    * @return the next token with respect to the current delimiter characters
    * @throws NoSuchElementException if there are no more tokens
