--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/SecureClassLoader.java	2004-06-08 02:25:15.000000000 -0600
+++ java/security/SecureClassLoader.java	2004-07-10 02:19:15.000000000 -0600
@@ -84,7 +84,7 @@
     if (cs != null)
       {
 	ProtectionDomain protectionDomain
-          = new ProtectionDomain(cs, getPermissions(cs), this, null);
+		= new ProtectionDomain(cs, getPermissions(cs));
 	return super.defineClass(name, b, off, len, protectionDomain);
       } 
     else
@@ -102,7 +102,7 @@
    */
   protected PermissionCollection getPermissions(CodeSource cs)
   {
-    Policy policy = Policy.getCurrentPolicy();
+    Policy policy = Policy.getPolicy();
     return policy.getPermissions(cs);
   }
 }
