--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Provider.java	2003-05-07 02:23:22.000000000 -0600
+++ java/security/Provider.java	2003-05-10 02:19:17.000000000 -0600
@@ -56,7 +56,7 @@
  */
 public abstract class Provider extends Properties implements Serializable
 {
-  static final long serialVersionUID = -4298000515446427739L;
+  private static final long serialVersionUID = -4298000515446427739L;
 
   /**
    * This is a textual description of the provider
