--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBoxMenuItem.java	2004-04-18 02:22:29.000000000 -0600
+++ javax/swing/JCheckBoxMenuItem.java	2004-05-27 02:19:03.000000000 -0600
@@ -39,135 +39,210 @@
 
 import java.io.IOException;
 import java.io.ObjectOutputStream;
-
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
 
 
+/**
+ * DOCUMENT ME!
+ *
+ * @author $author$
+ * @version $Revision: 1.5 $
+ */
 public class JCheckBoxMenuItem extends JMenuItem implements SwingConstants,
                                                             Accessible
 {
-  //-------------------------------------------------------------
-  // Variables --------------------------------------------------
-  //-------------------------------------------------------------
   private static final String uiClassID = "CheckBoxMenuItemUI";
   private boolean state;
   private Object[] selectedObjects;
 
-  //-------------------------------------------------------------
-  // Initialization ---------------------------------------------
-  //-------------------------------------------------------------
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   */
   public JCheckBoxMenuItem()
   {
     this(null, null);
-  } // JCheckBoxMenuItem()
+  }
 
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   *
+   * @param icon DOCUMENT ME!
+   */
   public JCheckBoxMenuItem(Icon icon)
   {
     this(null, icon);
-  } // JCheckBoxMenuItem()
+  }
 
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   */
   public JCheckBoxMenuItem(String text)
   {
     this(text, null);
-  } // JCheckBoxMenuItem()
+  }
 
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   *
+   * @param action DOCUMENT ME!
+   */
   public JCheckBoxMenuItem(Action action)
   {
     this();
     setAction(action);
-  } // JCheckBoxMenuItem()
+  }
 
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   * @param icon DOCUMENT ME!
+   */
   public JCheckBoxMenuItem(String text, Icon icon)
   {
     this(text, icon, false);
-  } // JCheckBoxMenuItem()
+  }
 
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   * @param state DOCUMENT ME!
+   */
   public JCheckBoxMenuItem(String text, boolean state)
   {
     this(text, null, state);
-  } // JCheckBoxMenuItem()
+  }
 
+  /**
+   * Creates a new JCheckBoxMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   * @param icon DOCUMENT ME!
+   * @param state DOCUMENT ME!
+   */
   public JCheckBoxMenuItem(String text, Icon icon, boolean state)
   {
     super(text, icon);
     setModel(new JToggleButton.ToggleButtonModel());
     this.state = state;
-  } // JCheckBoxMenuItem()
+  }
 
-  //-------------------------------------------------------------
-  // Methods ----------------------------------------------------
-  //-------------------------------------------------------------
+  /**
+   * DOCUMENT ME!
+   *
+   * @param stream DOCUMENT ME!
+   *
+   * @throws IOException DOCUMENT ME!
+   */
   private void writeObject(ObjectOutputStream stream) throws IOException
   {
     // TODO
-  } // writeObject()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   public String getUIClassID()
   {
     return uiClassID;
-  } // getUIClassID()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   public boolean getState()
   {
     return state;
-  } // getState()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @param state DOCUMENT ME!
+   */
   public synchronized void setState(boolean state)
   {
     this.state = state;
-  } // setState()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   public Object[] getSelectedObjects()
   {
     return selectedObjects;
-  } // getSelectedObjects()
+  }
 
+  /**
+   * DOCUMENT ME!
+   */
   public void requestFocus()
   {
     // TODO
-  } // requestFocus()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   protected String paramString()
   {
     return "JCheckBoxMenuItem";
-  } // paramString()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      {
-        accessibleContext = new AccessibleJCheckBoxMenuItem(this);
-      }
+      accessibleContext = new AccessibleJCheckBoxMenuItem(this);
 
     return accessibleContext;
-  } // getAccessibleContext()
+  }
 
-  //-------------------------------------------------------------
-  // Classes ----------------------------------------------------
-  //-------------------------------------------------------------
+  /**
+   * DOCUMENT ME!
+   *
+   * @author $author$
+   * @version $Revision: 1.5 $
+   */
   protected class AccessibleJCheckBoxMenuItem extends AccessibleJMenuItem
   {
-    //-------------------------------------------------------------
-    // Variables --------------------------------------------------
-    //-------------------------------------------------------------
-    //-------------------------------------------------------------
-    // Initialization ---------------------------------------------
-    //-------------------------------------------------------------
+    /**
+     * Creates a new AccessibleJCheckBoxMenuItem object.
+     *
+     * @param component DOCUMENT ME!
+     */
     protected AccessibleJCheckBoxMenuItem(JCheckBoxMenuItem component)
     {
       super(component);
 
       // TODO
-    } // AccessibleJCheckBoxMenuItem()
+    }
 
-    //-------------------------------------------------------------
-    // Methods ----------------------------------------------------
-    //-------------------------------------------------------------
+    /**
+     * DOCUMENT ME!
+     *
+     * @return $returnType$ DOCUMENT ME!
+     */
     public AccessibleRole getAccessibleRole()
     {
       return AccessibleRole.CHECK_BOX;
-    } // getAccessibleRole()
-  } // AccessibleJCheckBoxMenuItem
-} // JCheckBoxMenuItem
+    }
+  }
+}
