--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/PasswordAuthentication.java	2004-10-16 02:22:18.000000000 -0600
+++ java/net/PasswordAuthentication.java	2004-04-21 02:18:10.000000000 -0600
@@ -39,54 +39,69 @@
 
 
 /**
- * This class serves a container for username/password pairs.
- *
- * @author Aaron M. Renn (arenn@urbanophile.com)
- *
- * @since 1.2
- */
+  * This class serves a container for username/password pairs.
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @since 1.2
+  */
 public final class PasswordAuthentication
 {
-  /**
-   * The username
+  /*
+   * Instance Variables
    */
+
+  /**
+    * The username
+    */
   private String username;
 
   /**
-   * The password
-   */
+    * The password
+    */
   private char[] password;
 
-  /**
-   * Creates a new <code>PasswordAuthentication</code> object from the
-   * specified username and password.
-   *
-   * @param username The username for this object
-   * @param password The password for this object
+  /*************************************************************************/
+
+  /*
+   * Constructors
    */
+
+  /**
+    * Creates a new <code>PasswordAuthentication</code> object from the
+    * specified username and password.
+    *
+    * @param username The username for this object
+    * @param password The password for this object
+    */
   public PasswordAuthentication(String username, char[] password)
   {
     this.username = username;
     this.password = password;
   }
 
-  /**
-   * Returns the username associated with this object
-   *
-   * @return The username
+  /*************************************************************************/
+
+  /*
+   * Instance Methods
    */
+
+  /**
+    * Returns the username associated with this object
+    *
+    * @return The username
+    */
   public String getUserName()
   {
-    return username;
+    return (username);
   }
 
   /**
-   * Returns the password associated with this object
-   *
-   * @return The password
-   */
+    * Returns the password associated with this object
+    *
+    * @return The password
+    */
   public char[] getPassword()
   {
-    return password;
+    return (password);
   }
-}
+} // class PasswordAuthentication
