--- /home/tromey/gnu/classpath/classpath/java/awt/image/MemoryImageSource.java	Fri Jul 26 18:56:24 2002
+++ java/awt/image/MemoryImageSource.java	Sat Jan 26 19:37:59 2002
@@ -116,7 +116,7 @@
        Constructs an ImageProducer from memory using the default RGB ColorModel
     */
     public MemoryImageSource(int w, int h,
-			     int pix[], int off, int scan)
+			     byte pix[], int off, int scan)
     {
 	this ( w, h, ColorModel.getRGBdefault(), pix, off, scan, null);
     }
@@ -312,10 +312,7 @@
 				       int scansize)
 
     {
-	if( animated == true )
-	    {
-		//FIXME
-	    }
+      // FIXME
     }
 
     public synchronized void newPixels(int newpix[],
@@ -324,10 +321,7 @@
 				       int scansize)
 
     {
-	if( animated == true )
-	    {
-		//FIXME
-	    }
+      // FIXME
     }
 
 }
