--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Properties.java	2004-03-28 02:19:57.000000000 -0700
+++ java/util/Properties.java	2003-08-27 02:17:52.000000000 -0600
@@ -188,14 +188,17 @@
       {
         char c = 0;
         int pos = 0;
-	// Leading whitespaces must be deleted first.
+        // If empty line or begins with a comment character, skip this line.
+        if (line.length() == 0
+	    || line.charAt(0) == '#' || line.charAt(0) == '!')
+          continue;
+
         while (pos < line.length()
                && Character.isWhitespace(c = line.charAt(pos)))
           pos++;
 
-        // If empty line or begins with a comment character, skip this line.
-        if ((line.length() - pos) == 0
-	    || line.charAt(pos) == '#' || line.charAt(pos) == '!')
+        // If line is empty skip this line.
+        if (pos == line.length())
           continue;
 
         // The characters up to the next Whitespace, ':', or '='
