--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Collections.java	2004-10-28 02:27:38.000000000 -0600
+++ java/util/Collections.java	2004-11-08 02:15:38.000000000 -0700
@@ -3606,7 +3606,7 @@
    * Set s2 = m2.keySet(); // safe outside a synchronized block
    * synchronized (m) // synch on m, not m2, s or s2
    *   {
-3~3~   *     Iterator i = s.iterator();
+   *     Iterator i = s.iterator();
    *     while (i.hasNext())
    *       foo(i.next());
    *     i = s2.iterator();
