--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/Channel.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/Channel.java	2003-10-13 02:17:50.000000000 -0600
@@ -1,4 +1,4 @@
-/* Channel.java --
+/* Channel.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -39,7 +39,6 @@
 
 import java.io.IOException;
 
-
 public interface Channel
 {
   /**
