--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBox.java	2004-02-15 02:21:04.000000000 -0700
+++ javax/swing/JCheckBox.java	2004-06-18 02:18:51.000000000 -0600
@@ -1,5 +1,5 @@
 /* JCheckBox.java -- 
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -47,6 +47,8 @@
  */
 public class JCheckBox extends JToggleButton
 {
+  private static final long serialVersionUID = -5246739313864538930L;
+  
     public JCheckBox()
     {
 	this(null, null);
