--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2004-11-09 02:37:45.000000000 -0700
+++ java/net/URLStreamHandler.java	2004-10-28 02:16:11.000000000 -0600
@@ -214,9 +214,23 @@
 	  file = (file.substring(0, lastSlash)
 		  + '/' + spec.substring(start, end));
 
-	// For URLs constructed relative to a context, we
-	// need to canonicalise the file path.
-	file = canonicalizeFilename(file);
+	if (url.getProtocol().equals("file"))
+	  {
+	    // For "file" URLs constructed relative to a context, we
+	    // need to canonicalise the file path.
+	    try
+	      {
+		boolean endsWithSlash = file.charAt(file.length() - 1) == '/';
+		file = new File(file).getCanonicalPath();
+		file = file.replace(separator, '/');
+		if (endsWithSlash && file.charAt(file.length() - 1) != '/')
+		  file += '/';
+	      }
+	    catch (IOException e)
+	      {
+		// Do nothing.
+	      }
+	  }
 
 	ref = null;
       }
