--- /home/tromey/gnu/classpath/classpath/java/awt/event/KeyEvent.java	Sat Nov  9 16:09:31 2002
+++ java/awt/event/KeyEvent.java	Thu Aug  8 22:26:15 2002
@@ -754,18 +754,18 @@
   public static final int VK_EURO_SIGN = 516;
 
   /**
-   * The virtual key VK_EXCLAMATION_MARK.
+   * The virtual key VK_EXCLAMATION_POINT.
    *
    * @since 1.2
    */
-  public static final int VK_EXCLAMATION_MARK = 517;
+  public static final int VK_EXCLAMATION_POINT = 517;
 
   /**
-   * The virtual key VK_INVERTED_EXCLAMATION_MARK.
+   * The virtual key VK_INVERTED_EXCLAMATION_POINT.
    *
    * @since 1.2
    */
-  public static final int VK_INVERTED_EXCLAMATION_MARK = 518;
+  public static final int VK_INVERTED_EXCLAMATION_POINT = 518;
 
   /**
    * The virtual key VK_LEFT_PARENTHESIS.
@@ -1410,9 +1410,9 @@
         return "Dollar";
       case VK_EURO_SIGN:
         return "Euro";
-      case VK_EXCLAMATION_MARK:
+      case VK_EXCLAMATION_POINT:
         return "Exclamation Mark";
-      case VK_INVERTED_EXCLAMATION_MARK:
+      case VK_INVERTED_EXCLAMATION_POINT:
         return "Inverted Exclamation Mark";
       case VK_LEFT_PARENTHESIS:
         return "Left Parenthesis";
