--- /home/tromey/gnu/classpath/classpath/java/net/NetworkInterface.java	Sun Feb 16 17:03:45 2003
+++ java/net/NetworkInterface.java	Thu Feb 13 08:56:23 2003
@@ -1,5 +1,5 @@
 /* NetworkInterface.java
-   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -59,12 +59,8 @@
     this.inetAddresses.add (address);
   }
 
-  private static Vector getRealNetworkInterfaces ()
-    throws SocketException
-  {
-    throw new InternalError (
-      "NetworkInterface::getRealNetworkInterfaces not implemented");
-  }
+  private native static Vector getRealNetworkInterfaces ()
+    throws SocketException;
 
   /**
    *  Returns the name of the network interface
@@ -204,8 +200,9 @@
       return false;
    
     NetworkInterface tmp = (NetworkInterface) obj;
+    
     return (name.equals (tmp.name)
-	    && inetAddresses.equals (tmp.inetAddresses));
+            && inetAddresses.equals (tmp.inetAddresses));
   }
 
   /**
