--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Container.java	2003-07-25 02:18:53.000000000 -0600
+++ java/awt/Container.java	2003-07-09 02:18:35.000000000 -0600
@@ -473,7 +473,7 @@
   {
     synchronized (getTreeLock ())
       {
-        if (! isValid() && peer != null)
+        if (! isValid())
           {
             validateTree();
           }
@@ -658,7 +658,6 @@
    */
   public void update(Graphics g)
   {
-    g.clearRect(0, 0, width, height);
     super.update(g);
   }
 
