--- /home/tromey/gnu/classpath/classpath/java/beans/PropertyChangeListener.java	Sun Mar 31 23:08:10 2002
+++ java/beans/PropertyChangeListener.java	Tue Jan 22 16:01:32 2002
@@ -1,5 +1,5 @@
-/* PropertyChangeListener.java -- listen for changes in a bound property
-   Copyright (C) 1998, 2000, 2002 Free Software Foundation, Inc.
+/* java.beans.PropertyChangeListener
+   Copyright (C) 1998, 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -41,21 +41,22 @@
 import java.util.EventListener;
 
 /**
- * PropertyChangeListener allows a class to monitor properties of a Bean for
- * changes. A propertyChange() event will only be fired <em>after</em> the
- * property has changed.
- *
- * @author John Keiser
- * @see PropertyChangeSupport
- * @since 1.1
- * @status updated to 1.4
- */
+ ** PropertyChangeListener allows a class to monitor
+ ** properties of a Bean for changes.<P>
+ **
+ ** A propertyChange() event will only be fired
+ ** <EM>after</EM> the property has changed.
+ **
+ ** @author John Keiser
+ ** @since JDK1.1
+ ** @version 1.1.0, 29 Jul 1998
+ ** @see java.beans.PropertyChangeSupport
+ **/
+
 public interface PropertyChangeListener extends EventListener
 {
-  /**
-   * Fired after a Bean's property has changed.
-   *
-   * @param e the change (containing the old and new values)
-   */
-  void propertyChange(PropertyChangeEvent e);
-} // interface PropertyChangeListener
+  /** Fired after a Bean's property has changed.
+   ** @param e the change (containing the old and new values)
+   **/
+  public abstract void propertyChange(PropertyChangeEvent e);
+}
