--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectOutputStream.java	Mon Mar 24 02:18:31 2003
+++ java/io/ObjectOutputStream.java	Mon Mar 24 02:17:54 2003
@@ -841,16 +841,24 @@
    */
   public static abstract class PutField
   {
-    public abstract void put (String name, boolean value);
-    public abstract void put (String name, byte value);
-    public abstract void put (String name, char value);
-    public abstract void put (String name, double value);
-    public abstract void put (String name, float value);
-    public abstract void put (String name, int value);
-    public abstract void put (String name, long value);
-    public abstract void put (String name, short value);
-    public abstract void put (String name, Object value);
-    /** @deprecated */
+    public abstract void put (String name, boolean value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, byte value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, char value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, double value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, float value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, int value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, long value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, short value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, Object value)
+      throws IOException, IllegalArgumentException;
     public abstract void write (ObjectOutput out) throws IOException;
   }
 
@@ -866,6 +874,7 @@
 	= new Object[currentObjectStreamClass.objectFieldCount];
 
 	public void put (String name, boolean value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -874,6 +883,7 @@
 	}
 
 	public void put (String name, byte value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -882,6 +892,7 @@
 	}
 
 	public void put (String name, char value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -892,6 +903,7 @@
 	}
 
 	public void put (String name, double value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -909,6 +921,7 @@
 	}
 
 	public void put (String name, float value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -922,6 +935,7 @@
 	}
 
 	public void put (String name, int value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -934,6 +948,7 @@
 	}
 
 	public void put (String name, long value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -950,6 +965,7 @@
 	}
 
 	public void put (String name, short value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -960,6 +976,7 @@
 	}
 
 	public void put (String name, Object value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
