--- /home/tromey/gnu/egcs/classpath/classpath/java/security/KeyException.java	Sat Mar  2 12:13:17 2002
+++ java/security/KeyException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
 /* KeyException.java -- Thrown when there is a problem with a key
-   Copyright (C) 1998, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,30 +40,26 @@
 /**
  * This exception is thrown when there is a problem with a key.
  *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @see Key
- * @status updated to 1.4
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
  */
 public class KeyException extends GeneralSecurityException
 {
   /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -7483676942812432108L;
-
-  /**
    * This method initializes a new instance of <code>KeyException</code>
    * with no descriptive message.
    */
   public KeyException()
   {
+    super();
   }
 
   /**
    * This method initializes a new instance of <code>KeyException</code>
    * with a descriptive message.
    *
-   * @param msg the descriptive message
+   * @param msg The descriptive message.
    */
   public KeyException(String msg)
   {
