--- /home/tromey/gnu/classpath/classpath/java/nio/channels/DatagramChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/DatagramChannel.java	Tue Feb 11 09:41:07 2003
@@ -171,7 +171,8 @@
    * @exception SecurityException If a security manager has been installed and
    * it does not permit datagrams to be sent to the given address.
    */
-  public abstract int send (ByteBuffer src, SocketAddress target) throws IOException;
+  public abstract int send (ByteBuffer src, SocketAddress target)
+    throws IOException;
  
   /**
    * Retrieves the channel's socket.
