--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Container.java	2004-06-08 02:24:44.000000000 -0600
+++ java/awt/Container.java	2004-06-18 02:18:50.000000000 -0600
@@ -876,11 +876,6 @@
   {
     return locate (x, y);
   }
-  
-  public Component getComponentAt(int index)
-  {
-    return component[index];
-  }
 
   /**
    * Returns the component located at the specified point.  This is done
