--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/BasicPermission.java	2004-10-28 02:27:35.000000000 -0600
+++ java/security/BasicPermission.java	2004-10-28 02:16:12.000000000 -0600
@@ -72,9 +72,8 @@
  * @since 1.1
  * @status updated to 1.4
  */
-public abstract class BasicPermission extends java.security.Permission
+public abstract class BasicPermission extends Permission
   implements Serializable
-  // FIXME extends with fully qualified classname as workaround for gcj 3.3.
 {
   /**
    * Compatible with JDK 1.1+.
