--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Float.java	2004-04-18 02:22:16.000000000 -0600
+++ java/lang/Float.java	2004-04-21 02:18:09.000000000 -0600
@@ -426,10 +426,9 @@
    * @return the bits of the <code>float</code>
    * @see #intBitsToFloat(int)
    */
-  public static int floatToIntBits(float value)
-  {
-    return VMFloat.floatToIntBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native int floatToIntBits(float value);
+  // END GCJ LOCAL
 
   /**
    * Convert the float to the IEEE 754 floating-point "single format" bit
@@ -444,10 +443,9 @@
    * @return the bits of the <code>float</code>
    * @see #intBitsToFloat(int)
    */
-  public static int floatToRawIntBits(float value)
-  {
-    return VMFloat.floatToRawIntBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native int floatToRawIntBits(float value);
+  // END GCJ LOCAL
 
   /**
    * Convert the argument in IEEE 754 floating-point "single format" bit
@@ -462,10 +460,9 @@
    * @see #floatToIntBits(float)
    * @see #floatToRawIntBits(float)
    */
-  public static float intBitsToFloat(int bits)
-  {
-    return VMFloat.intBitsToFloat(bits);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native float intBitsToFloat(int bits);
+  // END GCJ LOCAL
 
   /**
    * Compare two Floats numerically by comparing their <code>float</code>
