--- /home/tromey/gnu/classpath/classpath/java/awt/Choice.java	Mon May  6 10:39:58 2002
+++ java/awt/Choice.java	Sun Mar 24 16:08:55 2002
@@ -234,7 +234,7 @@
   *
   * @param index The index of the item to remove.
   *
-  * @exception ArrayIndexOutOfBoundsException If the index is not valid.
+  * @exception ArrayIndexOutOfBoundException If the index is not valid.
   */
 public synchronized void
 remove(int index)
