--- /home/tromey/gnu/classpath/classpath/java/nio/channels/spi/AbstractChannel.java	Mon Aug 12 11:47:31 2002
+++ java/nio/channels/spi/AbstractChannel.java	Fri Nov  1 05:03:39 2002
@@ -1,4 +1,4 @@
-/* AbstractChannel.java
+/* AbstractChannel.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -35,27 +35,23 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package java.nio.channels.spi;
 
-import java.io.*;
-import java.nio.channels.*;
+import java.io.IOException;
+import java.nio.channels.Channel;
 
 public abstract class AbstractChannel implements Channel
 {
-    boolean opened;
+  boolean opened;
 
-    public boolean isOpen()
-    {
-	return opened;
-    }
-
-    public void close() throws IOException
-    {
-	if (! isOpen())
-	    {
-		return;
-	    }
-    }
+  public boolean isOpen()
+  {
+    return opened;
+  }
+
+  public void close() throws IOException
+  {
+    if (! isOpen())
+      return;
+  }
 }
-
