--- /home/tromey/gnu/egcs/classpath/classpath/java/util/Dictionary.java	Tue Apr 30 17:58:37 2002
+++ java/util/Dictionary.java	Tue Jan 22 16:01:40 2002
@@ -1,6 +1,6 @@
 /* Dictionary.java -- an abstract (and essentially worthless) 
    class which is Hashtable's superclass
-   Copyright (C) 1998, 2001, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -47,21 +47,18 @@
  * People at Javasoft are probably embarrassed by it.  At this point,
  * it might as well be an interface rather than a class, but it remains
  * this poor, laughable skeleton for the sake of backwards compatibility.
- * At any rate, this was what came before the {@link Map} interface 
+ * At any rate, this was what came before the <pre>Map</pre> interface 
  * in the Collections framework.
  *
  * @author Jon Zeppieri
- * @author Eric Blake (ebb9@email.byu.edu)
+ * @author Eric Blake <ebb9@email.byu.edu>
  * @see Map
  * @see Hashtable
  * @since 1.0
  * @status updated to 1.4
  */
-public abstract class Dictionary
+public abstract class Dictionary extends Object
 {
-  // WARNING: Dictionary is a CORE class in the bootstrap cycle. See the
-  // comments in vm/reference/java/lang/Runtime for implications of this fact.
-
   /**
    * Sole constructor (often called implicitly).
    */
@@ -133,4 +130,4 @@
    * @return the number of keys in the Dictionary
    */
   public abstract int size();
-} // class Dictionary
+}
