--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/LogManager.java	2004-04-18 02:22:29.000000000 -0600
+++ java/util/logging/LogManager.java	2004-01-09 02:18:04.000000000 -0700
@@ -67,12 +67,13 @@
  * <code>java.util.logging.LogManager</code> is initialized.
  * The configuration process includes the subsequent steps:
  *
- * <ul>
+ * <ol>
  * <li>If the system property <code>java.util.logging.manager</code>
  *     is set to the name of a subclass of
  *     <code>java.util.logging.LogManager</code>, an instance of
  *     that subclass is created and becomes the global LogManager.
  *     Otherwise, a new instance of LogManager is created.</li>
+ *     
  * <li>The <code>LogManager</code> constructor tries to create
  *     a new instance of the class specified by the system
  *     property <code>java.util.logging.config.class</code>.
@@ -90,13 +91,14 @@
  *     {@link #readConfiguration(java.io.InputStream)}.
  *     The name and location of this file are specified by the system
  *     property <code>java.util.logging.config.file</code>.</li>
+ *
  * <li>If the system property <code>java.util.logging.config.file</code>
  *     is not set, however, the contents of the URL
  *     "{gnu.classpath.home.url}/logging.properties" are passed to
  *     {@link #readConfiguration(java.io.InputStream)}.
  *     Here, "{gnu.classpath.home.url}" stands for the value of
  *     the system property <code>gnu.classpath.home.url</code>.</li>
- * </ul>
+ * </ol>
  *
  * @author Sascha Brawer (brawer@acm.org)
  */
@@ -257,10 +259,10 @@
    *
    * @param logger the logger to be added.
    *
-   * @return <code>true</code>if <code>logger</code> was added,
+   * @return <code>true<code>if <code>logger</code> was added,
    *         <code>false</code> otherwise.
    *
-   * @throws NullPointerException if <code>name</code> is
+   * @throws NullPointerException if <code>name<code> is
    *         <code>null</code>.
    */
   public synchronized boolean addLogger(Logger logger)
