--- /home/tromey/gnu/Nightly/classpath/classpath/javax/security/auth/PrivateCredentialPermission.java	2004-11-17 02:24:09.000000000 -0700
+++ javax/security/auth/PrivateCredentialPermission.java	2004-09-25 02:16:22.000000000 -0600
@@ -63,7 +63,7 @@
  * <i>PrincipalName</i> is a string representing the principal, or the
  * wildcard character.</p>
  */
-public final class PrivateCredentialPermission extends Permission
+public class PrivateCredentialPermission extends Permission
   implements Serializable
 {
   /**
