--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ByteBufferHelper.java	2004-10-12 02:18:13.000000000 -0600
+++ java/nio/ByteBufferHelper.java	2004-07-10 02:19:14.000000000 -0600
@@ -327,7 +327,7 @@
 
   public static void putDouble (ByteBuffer buffer, double value, ByteOrder order)
   {
-    putLong (buffer, Double.doubleToRawLongBits (value), order);
+    putLong (buffer, Double.doubleToLongBits (value), order);
   }
   
   public static double getDouble (ByteBuffer buffer, int index, ByteOrder order)
@@ -338,7 +338,6 @@
   public static void putDouble (ByteBuffer buffer, int index,
 				double value, ByteOrder order)
   {
-    putLong (buffer, index, Double.doubleToRawLongBits (value), order);
+    putLong (buffer, index, Double.doubleToLongBits (value), order);
   }
 } // ByteBufferHelper
-
