--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/registry/LocateRegistry.java	2004-11-17 02:24:08.000000000 -0700
+++ java/rmi/registry/LocateRegistry.java	2004-10-28 02:16:12.000000000 -0600
@@ -50,10 +50,6 @@
 import java.rmi.server.RemoteRef;
 
 public final class LocateRegistry {
-  /**
-   * This class isn't intended to be instantiated.
-   */
-  private LocateRegistry() {}
 
 public static Registry getRegistry() throws RemoteException {
 	return (getRegistry("localhost", Registry.REGISTRY_PORT));
