--- /home/tromey/gnu/classpath/classpath/java/text/Format.java	Wed Oct  2 11:21:38 2002
+++ java/text/Format.java	Tue Jan 22 15:40:37 2002
@@ -59,8 +59,6 @@
  */
 public abstract class Format implements Serializable, Cloneable
 {
-  static final long serialVersionUID = -299282585814624189L;
-
   /**
    * This method initializes a new instance of <code>Format</code>.
    * It performs no actions, but acts as a default constructor for
