--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/UnresolvedPermission.java	2004-04-28 02:20:17.000000000 -0600
+++ java/security/UnresolvedPermission.java	2002-05-31 12:14:24.000000000 -0600
@@ -37,6 +37,7 @@
 
 package java.security;
 
+import java.io.Serializable;
 // All uses of Certificate in this file refer to the one in the listed
 // package, not this one.
 import java.security.cert.Certificate;
