--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/ServerSocket.java	2004-01-08 02:20:40.000000000 -0700
+++ java/net/ServerSocket.java	2004-01-31 02:18:50.000000000 -0700
@@ -1,5 +1,6 @@
 /* ServerSocket.java -- Class for implementing server side sockets
-   Copyright (C) 1998, 1999, 2000, 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2002, 2003, 2004
+   Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -345,9 +346,14 @@
     if (isClosed())
       throw new SocketException("ServerSocket is closed");
     
+    // The Sun spec says that if we have an associated channel and
+    // it is in non-blocking mode, we throw an IllegalBlockingModeException.
+    // However, in our implementation if the channel itself initiated this
+    // operation, then we must honor it regardless of its blocking mode.
     if (getChannel() != null
-        && !getChannel().isBlocking())
-      throw new IllegalBlockingModeException();
+        && !getChannel().isBlocking ()
+        && !((PlainSocketImpl) getImpl()).isInChannelOperation())
+      throw new IllegalBlockingModeException ();
 	    
     impl.accept(socket.getImpl());
   }
