--- /home/tromey/gnu/classpath/classpath/java/awt/Toolkit.java	Thu May  9 19:18:48 2002
+++ java/awt/Toolkit.java	Mon Sep 16 03:46:37 2002
@@ -96,11 +96,11 @@
 {
   /** The default toolkit name. */
   private static String default_toolkit_name
-    = "gnu.java.awt.peer.gtk.GtkToolkit";
+    = "gnu.awt.gtk.GtkToolkit";
 
   /**
    * The toolkit in use.  Once we load it, we don't ever change it
-   * if the awt.toolkit propert is set.
+   * if the awt.toolkit property is set.
    */
   private static Toolkit toolkit;
 
