--- /home/tromey/gnu/classpath/classpath/java/net/Socket.java	Mon Apr  8 12:05:09 2002
+++ java/net/Socket.java	Fri Jul 26 18:40:47 2002
@@ -38,7 +38,6 @@
 package java.net;
 
 import java.io.*;
-import java.nio.channels.*;
 
 /* Written using on-line Java Platform 1.2 API Specification.
  * Status:  I believe all methods are implemented.
@@ -668,26 +667,4 @@
 
     factory = fac;
   }
-
-  public void shutdownInput() throws IOException
-  {
-  }
-
-  public void shutdownOutput() throws IOException
-  {
-  }
-
-
-    /**
-     * returns the socket channel associated with 
-     * this socket.
-     * It returns null if no associated socket exists.
-     */
-    
-    SocketChannel ch; // this field must have been set if created by SocketChannel
-
-    public SocketChannel getChannel() 
-    {
-	return ch;
-    }
 }
