--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/SimpleDateFormat.java	2004-04-18 02:22:26.000000000 -0600
+++ java/text/SimpleDateFormat.java	2003-12-03 02:18:10.000000000 -0700
@@ -358,13 +358,13 @@
    * object.  This will be true if and only if the specified object:
    * <p>
    * <ul>
-   * <li>Is not <code>null</code>.</li>
-   * <li>Is an instance of <code>SimpleDateFormat</code>.</li>
+   * <li>Is not <code>null</code>.
+   * <li>Is an instance of <code>SimpleDateFormat</code>.
    * <li>Is equal to this object at the superclass (i.e., <code>DateFormat</code>)
-   *     level.</li>
-   * <li>Has the same formatting pattern.</li>
-   * <li>Is using the same formatting symbols.</li>
-   * <li>Is using the same century for two digit years.</li>
+   *     level.
+   * <li>Has the same formatting pattern.
+   * <li>Is using the same formatting symbols.
+   * <li>Is using the same century for two digit years.
    * </ul>
    *
    * @param obj The object to compare for equality against.
