--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/ColorModel.java	2004-04-28 02:20:15.000000000 -0600
+++ java/awt/image/ColorModel.java	2003-06-28 02:18:20.000000000 -0600
@@ -166,7 +166,7 @@
    */
   public static ColorModel getRGBdefault()
   {
-    return new DirectColorModel(32, 0xff0000, 0xff00, 0xff, 0xff000000);
+    return new DirectColorModel(8, 0xff0000, 0xff00, 0xff, 0xff000000);
   }
 
   public final boolean hasAlpha()
@@ -597,11 +597,7 @@
     return null;
   }
     
-  /**
-   * Checks if the given raster has a compatible data-layout (SampleModel).
-   * @param raster The Raster to test.
-   * @return true if raster is compatible.
-   */ 
+  // Typically overridden
   public boolean isCompatibleRaster(Raster raster)
   {
     SampleModel sampleModel = raster.getSampleModel();
