--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/AbstractListModel.java	Wed Oct  2 06:58:40 2002
+++ javax/swing/AbstractListModel.java	Mon Mar 31 02:17:51 2003
@@ -191,6 +191,15 @@
 	} // getListeners()
 
 	/**
+	 * getListDataListeners
+	 */
+	public ListDataListener[] getListDataListeners()
+	{
+          // FIXME: implement this
+	  return null;
+	}
+
+	/**
 	 * getElementAt
 	 * @param index TODO
 	 * @returns Object
