--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/CompoundName.java	2003-07-14 02:19:07.000000000 -0600
+++ javax/naming/CompoundName.java	2003-11-12 02:18:14.000000000 -0700
@@ -240,7 +240,7 @@
 
   public int compareTo (Object obj)
   {
-    if (obj == null || ! (obj instanceof CompoundName))
+    if (! (obj instanceof CompoundName))
       throw new ClassCastException ("CompoundName.compareTo() expected CompoundName");
     CompoundName cn = (CompoundName) obj;
     int last = Math.min (cn.elts.size (), elts.size ());
