--- /home/tromey/gnu/Nightly/classpath/classpath/java/math/BigInteger.java	2003-10-10 02:19:17.000000000 -0600
+++ java/math/BigInteger.java	2003-11-12 02:18:14.000000000 -0700
@@ -1565,7 +1565,7 @@
   /* Assumes this and obj are both canonicalized. */
   public boolean equals(Object obj)
   {
-    if (obj == null || ! (obj instanceof BigInteger))
+    if (! (obj instanceof BigInteger))
       return false;
     return equals(this, (BigInteger) obj);
   }
