--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2003-07-02 02:19:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2003-04-01 02:18:01.000000000 -0700
@@ -64,7 +64,7 @@
 public abstract class BasicLookAndFeel extends LookAndFeel
   implements Serializable
 {
-  static final long serialVersionUID = -6096995660290287879L;
+  static final long serialVersionUID = 5484702182266873258L;
 
   /**
    * Constructor BasicLookAndFeel
