--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Provider.java	2002-11-16 16:28:25.000000000 -0700
+++ java/security/Provider.java	2003-05-04 02:19:12.000000000 -0600
@@ -45,7 +45,7 @@
  * The services provided by a such a provider can range from security
  * algorithms to key generation.
  * <p>
- * Providers are installed by name and verion number.  There is one
+ * Providers are installed by name and version number.  There is one
  * standard provider supplied with the class library.  This is the
  * "GNU" provider, which can also be accessed by the alias "SUN" for
  * compatibility with the JDK.
