--- /home/tromey/gnu/egcs/classpath/classpath/java/security/interfaces/DSAKey.java	Wed Mar 22 12:26:27 2000
+++ java/security/interfaces/DSAKey.java	Wed Apr 25 11:13:37 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,27 +24,22 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security.interfaces;
 
 /**
-  * This interface is implemented by a class to return the parameters
-  * of a Digital Signature Algorithm (DSA) public or private key.
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This interface is implemented by a class to return the parameters
+ * of a Digital Signature Algorithm (DSA) public or private key.
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public interface DSAKey
 {
-
-/**
-  * This method returns non-secret parameters of the DSA key
-  *
-  * @return The DSA parameters
-  */
-public abstract DSAParams
-getParams();
-
-} // interface DSAKey
-
+  /**
+   * This method returns non-secret parameters of the DSA key
+   *
+   * @return The DSA parameters
+   */
+  public abstract DSAParams getParams();
+}
