--- /home/tromey/gnu/classpath/classpath/java/nio/Buffer.java	Fri Nov 15 14:10:10 2002
+++ java/nio/Buffer.java	Tue Feb 11 09:41:06 2003
@@ -39,14 +39,33 @@
 
 public abstract class Buffer
 {
-  /** Capacity of the buffer.
-   * XXX - FIXME - should not be protected but (package) private
-   * */
-  protected int cap = 0;
-  int limit = 0;
-  int pos = 0;
-  int mark = -1;
+  private int cap = 0;
+  private int limit = 0;
+  private int pos = 0;
+  private int mark = -1;
 
+  // Creates a new Buffer.
+  //
+  // Should be package private.
+  //
+  Buffer (int capacity, int limit, int position, int mark)
+  {
+    if (capacity < 0)
+      throw new IllegalArgumentException ();
+    
+    cap = capacity;
+    limit (limit);
+    position (position);
+    
+    if (mark > 0)
+    {
+      if (mark > pos)
+        throw new IllegalArgumentException ();
+      
+      this.mark = mark;
+    }
+  }
+  
   /**
    * Retrieves the capacity of the buffer.
    */
