--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ActionMap.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/ActionMap.java	2004-01-10 02:18:19.000000000 -0700
@@ -42,6 +42,7 @@
 import java.io.ObjectOutputStream;
 import java.io.Serializable;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
