--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/KeyStore.java	2004-04-18 02:22:22.000000000 -0600
+++ java/security/KeyStore.java	2003-04-30 02:19:05.000000000 -0600
@@ -208,7 +208,7 @@
 
   /**
    * Returns the default KeyStore type. This method looks up the
-   * type in &lt;JAVA_HOME&gt;/lib/security/java.security with the 
+   * type in <JAVA_HOME>/lib/security/java.security with the 
    * property "keystore.type" or if that fails then "jks" .
    */
   public static final String getDefaultType()
