--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JLabel.java	2004-04-18 02:22:30.000000000 -0600
+++ javax/swing/JLabel.java	2004-06-18 02:18:52.000000000 -0600
@@ -331,9 +331,8 @@
    */
   public Icon getDisabledIcon()
   {
-    //FIXME: We should be gray-scaling the active icon and then returning it
     if (disabledIcon == null && activeIcon instanceof ImageIcon)
-      setDisabledIcon(activeIcon);
+      disabledIcon = new ImageIcon(GrayFilter.createDisabledImage(((ImageIcon) activeIcon).getImage()));
     return disabledIcon;
   }
 
