--- /home/tromey/gnu/egcs/classpath/classpath/java/security/SecureRandomSpi.java	Wed Mar 22 12:26:22 2000
+++ java/security/SecureRandomSpi.java	Wed Apr 25 09:45:12 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,43 +24,40 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 import java.io.Serializable;
-//import SecureRandom;
+
 /**
    SecureRandomSpi is the Service Provider Interface for SecureRandom
    providers. It provides an interface for providers to the 
    SecureRandom engine to write their own pseudo-random number
    generator.
 
-   @since JDK 1.2	
+   @since JDK 1.2       
 
    @author Mark Benvenuto <ivymccough@worldnet.att.net>
-*/
-
+ */
 public abstract class SecureRandomSpi implements Serializable
 {
-
   /**
      Default Constructor for SecureRandomSpi
-  */
+   */
   public SecureRandomSpi()
-  {}
+  {
+  }
 
   /**
      Updates the seed for SecureRandomSpi but does not reset seed. 
      It does to this so repeated called never decrease randomness.
-  */
+   */
   protected abstract void engineSetSeed(byte[] seed);
 
-
   /**
      Gets a user specified number of bytes depending on the length
      of the array?
 
      @param bytes array to fill with random bytes
-  */
+   */
   protected abstract void engineNextBytes(byte[] bytes);
 
   /**
@@ -70,6 +67,6 @@
      @param numBytes number of random bytes to generate
 
      @return an array full of random bytes
-  */
+   */
   protected abstract byte[] engineGenerateSeed(int numBytes);
 }
