--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Locale.java	2003-08-13 02:19:50.000000000 -0600
+++ java/util/Locale.java	2004-03-12 02:18:25.000000000 -0700
@@ -436,7 +436,7 @@
    */
   public String getISO3Language()
   {
-    if (language == "")
+    if (language.equals(""))
       return "";
     int index
       = ("aa,ab,af,am,ar,as,ay,az,ba,be,bg,bh,bi,bn,bo,br,ca,co,cs,cy,da,"
@@ -472,7 +472,7 @@
    */
   public String getISO3Country()
   {
-    if (country == "")
+    if (country.equals(""))
       return "";
     int index
       = ("AD,AE,AF,AG,AI,AL,AM,AN,AO,AQ,AR,AS,AT,AU,AW,AZ,BA,BB,BD,BE,BF,"
