--- /home/tromey/gnu/Nightly/classpath/classpath/java/sql/Date.java	Sat Apr 12 02:21:30 2003
+++ java/sql/Date.java	Tue Apr  1 02:18:00 2003
@@ -37,7 +37,6 @@
 
 package java.sql;
 
-import java.text.ParseException;
 import java.text.SimpleDateFormat;
 
 /**
@@ -83,66 +82,6 @@
   }
 
   /**
-   * This method always throws an IllegalArgumentException.
-   *
-   * @throws IllegalArgumentException when it's called.
-   * @deprecated
-   */
-  public int getHours() throws IllegalArgumentException {
-    throw new IllegalArgumentException();
-  }
-
-  /**
-   * This method always throws an IllegalArgumentException.
-   *
-   * @throws IllegalArgumentException when it's called.
-   * @deprecated
-   */
-  public int getMinutes() throws IllegalArgumentException {
-    throw new IllegalArgumentException();
-  }
-
-  /**
-   * This method always throws an IllegalArgumentException.
-   *
-   * @throws IllegalArgumentException when it's called.
-   * @deprecated
-   */
-  public int getSeconds() throws IllegalArgumentException {
-    throw new IllegalArgumentException();
-  }
-
-  /**
-   * This method always throws an IllegalArgumentException.
-   *
-   * @throws IllegalArgumentException when it's called.
-   * @deprecated
-   */
-  public void setHours(int newValue) throws IllegalArgumentException {
-    throw new IllegalArgumentException();
-  }
-
-  /**
-   * This method always throws an IllegalArgumentException.
-   *
-   * @throws IllegalArgumentException when it's called.
-   * @deprecated
-   */
-  public void setMinutes(int newValue) throws IllegalArgumentException {
-    throw new IllegalArgumentException();
-  }
-
-  /**
-   * This method always throws an IllegalArgumentException.
-   *
-   * @throws IllegalArgumentException when it's called.
-   * @deprecated
-   */
-  public void setSeconds(int newValue) throws IllegalArgumentException {
-    throw new IllegalArgumentException();
-  }
-
-  /**
    * This method returns a new instance of this class by parsing a
    * date in JDBC format into a Java date.
    *
@@ -153,20 +92,14 @@
    */
   public static Date valueOf (String str)
   {
-    try 
+    try
       {
 	java.util.Date d = (java.util.Date) sdf.parseObject(str);
-
-	if (d == null) {
-	  throw new IllegalArgumentException(str);
-	}
-	else {
-	  return(new Date(d.getTime()));
-	}
+	return(new Date(d.getTime()));
       }
-    catch (ParseException e)
+    catch(Exception e)
       {
-	throw new IllegalArgumentException(str);
+	return(null);
       }
   }
 
