--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/CharBufferImpl.java	2003-06-28 02:19:18.000000000 -0600
+++ java/nio/CharBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -52,7 +52,9 @@
   
   CharBufferImpl (char[] buffer, int offset, int capacity, int limit, int position, int mark, boolean readOnly)
   {
-    super (buffer, offset, capacity, limit, position, mark);
+    super (capacity, limit, position, mark);
+    this.backing_buffer = buffer;
+    this.array_offset = offset;
     this.readOnly = readOnly;
   }
   
