--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/MulticastSocket.java	2003-06-16 02:23:59.000000000 -0600
+++ java/net/MulticastSocket.java	2003-09-11 02:18:05.000000000 -0600
@@ -36,8 +36,10 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package java.net;
 
+import gnu.java.net.PlainDatagramSocketImpl;
 import java.io.IOException;
 import java.util.Enumeration;
 
