--- /home/tromey/gnu/classpath/classpath/java/awt/Container.java	Mon Sep 30 13:37:09 2002
+++ java/awt/Container.java	Thu Aug  8 22:26:13 2002
@@ -205,7 +205,7 @@
    *
    * @return The same component that was added.
    *
-   * @throws ArrayIndexOutOfBounds If the specified index is invalid.
+   * @param throws ArrayIndexOutOfBounds If the specified index is invalid.
    */
   public Component add(Component comp, int index)
   {
@@ -236,7 +236,7 @@
    * @param index The index in the component list to insert this child
    * at, or -1 to add at the end of the list.
    *
-   * @throws ArrayIndexOutOfBounds If the specified index is invalid.
+   * @param throws ArrayIndexOutOfBounds If the specified index is invalid.
    */
   public void add(Component comp, Object constraints, int index)
   {
@@ -256,7 +256,7 @@
    * @param index The index in the component list to insert this child
    * at, or -1 to add at the end of the list.
    *
-   * @throws ArrayIndexOutOfBounds If the specified index is invalid.
+   * @param throws ArrayIndexOutOfBounds If the specified index is invalid.
    */
   protected void addImpl(Component comp, Object constraints, int index)
   {
