--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/directory/Attribute.java	2003-10-13 02:22:18.000000000 -0600
+++ javax/naming/directory/Attribute.java	2003-12-06 02:18:23.000000000 -0700
@@ -38,8 +38,11 @@
 
 package javax.naming.directory;
 
-import javax.naming.*;
 import java.io.Serializable;
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
 
 /**
  * @author Warren Levy <warrenl@redhat.com>
