--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JRootPane.java	2003-11-22 02:22:38.000000000 -0700
+++ javax/swing/JRootPane.java	2004-02-12 02:18:27.000000000 -0700
@@ -75,7 +75,6 @@
       public Dimension preferredLayoutSize ( Container c )
 	{	    
 	  Dimension p = super.preferredLayoutSize(c);
-	  System.out.println("              PREF-SIZE from RootLayout = " + p);
 	  return p;
 	}        
     }
@@ -97,7 +96,7 @@
     /********************************************************/
 
     public String getUIClassID()
-    {	return "JPanel";    }
+    {	return "RootPaneUI";    }
 
     
     void setJMenuBar(JMenuBar m)
@@ -169,7 +168,7 @@
     JRootPane()
     {
 	setLayout(createRootLayout());
-	
+	setBackground(UIManager.getColor("control"));
 	getGlassPane();
 	getLayeredPane();
 	getContentPane();
