--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/LineNumberReader.java	2003-09-01 02:18:44.000000000 -0600
+++ java/io/LineNumberReader.java	2003-06-28 02:18:20.000000000 -0600
@@ -227,7 +227,7 @@
 	  {
 	    if (markPos >= 0 && limit == buffer.length)
 	      markPos = -1;
-	    if (markPos < 0)
+	    if (markPos <= 0)
 	      pos = limit = 0;
 	    int count = in.read(buffer, limit, buffer.length - limit);
 	    if (count <= 0)
