--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/dgc/Lease.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/dgc/Lease.java	2002-01-22 16:01:36.000000000 -0700
@@ -1,5 +1,5 @@
-/* Lease.java
-   Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,6 +38,7 @@
 package java.rmi.dgc;
 
 import java.io.Serializable;
+import java.rmi.dgc.VMID;
 
 public final class Lease
 	implements Serializable {
