--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/MappedByteBufferImpl.java	2003-06-27 16:11:17.000000000 -0600
+++ java/nio/MappedByteBufferImpl.java	2003-06-20 06:01:54.000000000 -0600
@@ -40,7 +40,7 @@
 
 import gnu.java.nio.FileChannelImpl;
 import java.io.IOException;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 public class MappedByteBufferImpl extends MappedByteBuffer
 {
