--- /home/tromey/gnu/egcs/classpath/classpath/java/security/cert/CertificateFactory.java	Tue Apr 30 11:50:10 2002
+++ java/security/cert/CertificateFactory.java	Tue Jan 22 16:01:37 2002
@@ -1,5 +1,5 @@
-/* CertificateFactory.java -- Certificate Factory Class
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+/* CertificateFactory.java --- Certificate Factory Class
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -52,9 +52,9 @@
    java.security.cert.X509Certificate class, and CRLs of the 
    java.security.cert.X509CRL class. 
    
-   @author Mark Benvenuto
    @since JDK 1.2
-   @status still missing full 1.4 support
+
+   @author Mark Benvenuto
 */
 public class CertificateFactory
 {
@@ -267,9 +267,4 @@
     return certFacSpi.engineGenerateCRLs( inStream );
   }
 
-  public final CertPath generateCertPath(InputStream inStream)
-    throws CertificateException
-  {
-    throw new CertificateException("not implemented");
-  }
-} // class CertificateFactory
+}
