--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/MappedByteBuffer.java	Tue Mar 11 10:27:52 2003
+++ java/nio/MappedByteBuffer.java	Tue Feb 11 09:41:07 2003
@@ -43,7 +43,7 @@
  */
 public abstract class MappedByteBuffer extends ByteBuffer
 {
-  protected MappedByteBuffer (int capacity, int limit, int position, int mark)
+  MappedByteBuffer (int capacity, int limit, int position, int mark)
   {
     super (capacity, limit, position, mark);
   }
