--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Window.java	2004-06-21 06:23:48.000000000 -0600
+++ java/awt/Window.java	2004-06-15 02:19:28.000000000 -0600
@@ -704,7 +704,7 @@
   public void applyResourceBundle(String rbName)
   {
     ResourceBundle rb = ResourceBundle.getBundle(rbName, Locale.getDefault(),
-      ClassLoader.getSystemClassLoader());
+      Window.class.getClassLoader());
     if (rb != null)
       applyResourceBundle(rb);    
   }
