--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipOutputStream.java	2004-10-28 02:27:40.000000000 -0600
+++ java/util/zip/ZipOutputStream.java	2004-06-08 02:19:58.000000000 -0600
@@ -1,5 +1,5 @@
-/* ZipOutputStream.java --
-   Copyright (C) 2001, 2004  Free Software Foundation, Inc.
+/* java.util.zip.ZipOutputStream
+   Copyright (C) 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -71,18 +71,17 @@
   /**
    * Our Zip version is hard coded to 1.0 resp. 2.0
    */
-  private static final int ZIP_STORED_VERSION = 10;
-  private static final int ZIP_DEFLATED_VERSION = 20;
+  private final static int ZIP_STORED_VERSION   = 10;
+  private final static int ZIP_DEFLATED_VERSION = 20;
 
   /**
    * Compression method.  This method doesn't compress at all.
    */
-  public static final int STORED = 0;
-  
+  public final static int STORED      =  0;
   /**
    * Compression method.  This method uses the Deflater.
    */
-  public static final int DEFLATED = 8;
+  public final static int DEFLATED    =  8;
 
   /**
    * Creates a new Zip output stream, writing a zip archive.
@@ -137,7 +136,7 @@
   /**
    * Write an unsigned short in little endian byte order.
    */
-  private void writeLeShort(int value) throws IOException 
+  private final void writeLeShort(int value) throws IOException 
   {
     out.write(value & 0xff);
     out.write((value >> 8) & 0xff);
@@ -146,7 +145,7 @@
   /**
    * Write an int in little endian byte order.
    */
-  private void writeLeInt(int value) throws IOException 
+  private final void writeLeInt(int value) throws IOException 
   {
     writeLeShort(value);
     writeLeShort(value >> 16);
