--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormat.java	2004-04-28 02:20:17.000000000 -0600
+++ java/text/DecimalFormat.java	2004-04-21 02:18:11.000000000 -0600
@@ -37,10 +37,12 @@
 
 package java.text;
 
-import java.io.IOException;
-import java.io.ObjectInputStream;
 import java.util.Currency;
 import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.io.ObjectInputStream;
+import java.io.IOException;
 
 /**
  * @author Tom Tromey <tromey@cygnus.com>
