--- /home/tromey/gnu/egcs/classpath/classpath/java/security/PrivateKey.java	Sun Mar 31 23:08:12 2002
+++ java/security/PrivateKey.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* PrivateKey.java -- tagging interface for all private keys
-   Copyright (C) 1998, 2001, 2002 Free Software Foundation, Inc.
+/* PrivateKey.java -- "Super-interface" for all private keys
+   Copyright (C) 1998, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,24 +39,13 @@
 package java.security;
 
 /**
- * This interface specified no methods.  In simply provides a common
- * super-interface for all algorithm specific private key values.
- *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @see Key
- * @see PublicKey
- * @see Certificate
- * @see Signature#initVerify(PublicKey)
- * @see DSAPrivateKey
- * @see RSAPrivateKey
- * @see RSAPrivateCrtKey
- * @since 1.1
- * @status updated to 1.4
+   * This interface specified no methods.  In simply provides a common
+   * super-interface for all algorithm specific private key values.
+   *
+   * @version 0.0
+   *
+   * @author Aaron M. Renn (arenn@urbanophile.com)
  */
 public interface PrivateKey extends Key
 {
-  /**
-   * The verion identifier used for serialization.
-   */
-  long serialVersionUID = 6034044314589513430L;
-} // interface PrivateKey
+}
