--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JLayeredPane.java	2004-04-19 02:20:15.000000000 -0600
+++ javax/swing/JLayeredPane.java	2004-01-10 02:18:19.000000000 -0700
@@ -82,15 +82,16 @@
  * <dd>An offset into a layer's "logical drawing order". Layer position 0
  * is drawn last. Layer position -1 is a synonym for the first layer
  * position (the logical "bottom").</dd>
- * </dl>
  *
  * <p><b>Note:</b> the layer numbering order is the <em>reverse</em> of the
  * component indexing and position order</p>
  *
  * @author Graydon Hoare <graydon@redhat.com>
  */
+
 public class JLayeredPane extends JComponent implements Accessible
 {
+
   public static String LAYER_PROPERTY = "LAYER_PROPERTY";
 
   public static Integer FRAME_CONTENT_LAYER = new Integer (-30000);
