--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Properties.java	2003-08-17 02:19:12.000000000 -0600
+++ java/util/Properties.java	2003-03-03 02:17:50.000000000 -0700
@@ -56,7 +56,7 @@
  * here.  This extends the example given in ListResourceBundle.
  * Create a file MyResource_de.properties with the following contents
  * and put it in the CLASSPATH.  (The character
- * <code>\</code><code>u00e4</code> is the german umlaut)
+ * <code>\</code><code>u00e4</code> is the german &auml;)
  *
  * 
 <pre>s1=3
