--- /home/tromey/gnu/classpath/classpath/java/net/DatagramSocketImpl.java	Sun Jul 14 23:54:26 2002
+++ java/net/DatagramSocketImpl.java	Fri Feb 15 00:17:12 2002
@@ -42,15 +42,14 @@
 import java.io.FileDescriptor;
 
 /**
- * This abstract class models a datagram socket implementation.  An
- * actual implementation class would implement these methods, probably
- * via redirecting them to native code.
- * <p>
  * Written using on-line Java Platform 1.2 API Specification, as well
  * as "The Java Class Libraries", 2nd edition (Addison-Wesley, 1998).
- * <p>
  * Status:  Believed complete and correct.
  *
+ * <p>This abstract class models a datagram socket implementation.  An
+ * actual implementation class would implement these methods, probably
+ * via redirecting them to native code.
+ *
  * @author Aaron M. Renn (arenn@urbanophile.com)
  * @author Warren Levy <warrenl@cygnus.com>
  */
