--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/datatransfer/DataFlavor.java	2003-11-16 02:19:36.000000000 -0700
+++ java/awt/datatransfer/DataFlavor.java	2003-12-09 02:18:21.000000000 -0700
@@ -459,7 +459,7 @@
   if (idx == -1)
     return(null);
 
-  String value = mimeString.substring(idx + paramName.length() + 2);
+  String value = mimeString.substring(idx + paramName.length() + 1);
 
   idx = value.indexOf(" ");
   if (idx == -1)
