--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-01-28 02:27:14.000000000 -0700
+++ javax/swing/plaf/basic/BasicGraphicsUtils.java	2003-08-02 02:18:05.000000000 -0600
@@ -595,7 +595,6 @@
     Rectangle iconRect = new Rectangle();
     Rectangle textRect = new Rectangle();
     Insets insets = b.getInsets();
-    Insets margin = b.getMargin();
     
     /* For determining the ideal size, do not assume a size restriction. */
     viewRect = new Rectangle(0, 0,
@@ -631,11 +630,7 @@
      */
     contentRect = textRect.union(iconRect);
 
-    return new Dimension(insets.left + margin.left
-			 + contentRect.width 
-			 + insets.right + margin.right,
-                         insets.top + margin.top
-			 + contentRect.height 
-			 + insets.bottom + margin.bottom);
+    return new Dimension(insets.left + contentRect.width + insets.right,
+                         insets.top + contentRect.height + insets.bottom);
   }
 }
