--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Security.java	2003-12-01 02:24:44.000000000 -0700
+++ java/security/Security.java	2003-10-09 02:17:59.000000000 -0600
@@ -475,7 +475,7 @@
    *    provider that supplied a CertificateFactory implementation for X.509
    *    certificates.</p></li>
    *
-   *    <li><p>&lt;crypto_service>.&lt;algorithm_or_type> &lt;attribute_name>:&lt;attribute_value></p>
+   *    <li><p>&lt;crypto_service>.&lt;algorithm_or_type>&nbsp;&lt;attribute_name>:&lt;attribute_value></p>
    *    <p>The cryptographic service name must not contain any dots. There must
    *    be one or more space charaters between the the &lt;algorithm_or_type>
    *    and the &lt;attribute_name>.</p>
