--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/AlreadyConnectedException.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/AlreadyConnectedException.java	2002-12-10 19:20:33.000000000 -0700
@@ -1,4 +1,4 @@
-/* AlreadyConnectedException.java --
+/* AlreadyConnectedException.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
