--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipFile.java	2003-11-26 02:20:09.000000000 -0700
+++ java/util/zip/ZipFile.java	2003-07-14 02:18:14.000000000 -0600
@@ -105,7 +105,7 @@
   public ZipFile(File file) throws ZipException, IOException
   {
     this.raf = new RandomAccessFile(file, "r");
-    this.name = file.getPath();
+    this.name = file.getName();
   }
 
   /**
@@ -134,7 +134,7 @@
 	  ("OPEN_DELETE mode not supported yet in java.util.zip.ZipFile");
       }
     this.raf = new RandomAccessFile(file, "r");
-    this.name = file.getPath();
+    this.name = file.getName();
   }
 
   /**
@@ -438,7 +438,7 @@
   }
   
   /**
-   * Returns the (path) name of this zip file.
+   * Returns the name of this zip file.
    */
   public String getName()
   {
