--- /home/tromey/gnu/classpath/classpath/java/awt/Choice.java	Mon Sep 30 13:37:09 2002
+++ java/awt/Choice.java	Thu Aug  8 22:26:13 2002
@@ -215,7 +215,7 @@
   *
   * @param item The item to remove.
   *
-  * @exception IllegalArgumentException If the specified item doesn't exist.
+  * @param IllegalArgumentException If the specified item doesn't exist.
   */
 public synchronized void
 remove(String item)
@@ -234,7 +234,7 @@
   *
   * @param index The index of the item to remove.
   *
-  * @exception IndexOutOfBoundsException If the index is not valid.
+  * @exception ArrayIndexOutOfBoundsException If the index is not valid.
   */
 public synchronized void
 remove(int index)
@@ -325,7 +325,7 @@
   *
   * @param index The index of the row to make selected.
   *
-  * @exception IllegalArgumentException If the specified index is invalid.
+  * @param IllegalArgumentException If the specified index is invalid.
   */
 public synchronized void
 select(int index)
