--- /home/tromey/gnu/classpath/classpath/java/nio/MappedByteBuffer.java	Wed Nov 13 10:15:53 2002
+++ java/nio/MappedByteBuffer.java	Mon Oct  7 07:12:42 2002
@@ -37,26 +37,6 @@
 
 package java.nio;
 
-
 public abstract class MappedByteBuffer extends ByteBuffer
 {
-    private static native void sync();
-
-    public final MappedByteBuffer force()
-    {
-	sync();
-	return this;
-    }
-    
-    public final boolean isLoaded()
-    {
-	return true;
-    }
-    
-    public final MappedByteBuffer load()
-    {
-	sync();
-	return this;
-    }
- 
 }
