--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/NumberFormat.java	2004-04-28 02:20:17.000000000 -0600
+++ java/text/NumberFormat.java	2004-03-27 02:18:31.000000000 -0700
@@ -38,8 +38,8 @@
 
 package java.text;
 
-import java.io.IOException;
 import java.io.InvalidObjectException;
+import java.io.IOException;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 import java.util.Currency;
