--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/SimpleFormatter.java	2003-08-13 02:19:51.000000000 -0600
+++ java/util/logging/SimpleFormatter.java	2003-06-21 04:31:55.000000000 -0600
@@ -106,9 +106,7 @@
 
     buf.append(dateFormat.format(new Date(record.getMillis())));
     buf.append(' ');
-    buf.append(record.getSourceClassName());
-    buf.append(' ');
-    buf.append(record.getSourceMethodName());
+    buf.append(record.getLoggerName());
     buf.append(lineSep);
 
     buf.append(record.getLevel());
