--- /home/tromey/gnu/classpath/classpath/java/awt/Graphics2D.java	Wed Jan  1 15:49:07 2003
+++ java/awt/Graphics2D.java	Wed Feb 19 14:18:32 2003
@@ -36,6 +36,8 @@
 
 package java.awt;
 
+import java.awt.font.FontRenderContext;
+import java.awt.font.GlyphVector;
 import java.awt.geom.AffineTransform;
 import java.awt.image.BufferedImage;
 import java.awt.image.BufferedImageOp;
@@ -151,5 +153,7 @@
 
   public abstract void clip(Shape s);
 
-  // public abstract FontRenderContext getFontRenderContext();
+  public abstract FontRenderContext getFontRenderContext ();
+
+  public abstract void drawGlyphVector (GlyphVector g, float x, float y);
 }
