--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicMenuUI.java	2004-06-27 02:20:48.000000000 -0600
+++ javax/swing/plaf/basic/BasicMenuUI.java	2004-06-18 02:18:58.000000000 -0600
@@ -342,6 +342,7 @@
 	    {
 	      // If menu is visible and menu button was pressed.. 
 	      // then need to cancel the menu
+	      menu.fireMenuCanceled();
 	      manager.clearSelectedPath();
 	    }
 	  else
@@ -350,6 +351,7 @@
 	      int x = 0;
 	      int y = menu.getHeight();
 
+	      menu.fireMenuSelected();
 	      manager.setSelectedPath(getPath());
 
 	      JMenuBar mb = (JMenuBar) menu.getParent();
