--- /home/tromey/gnu/egcs/classpath/classpath/java/rmi/UnexpectedException.java	Sat Mar  2 12:13:14 2002
+++ java/rmi/UnexpectedException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* UnexpectedException.java -- an unexpected checked exception was received
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,39 +37,15 @@
 
 package java.rmi;
 
-/**
- * Thrown if an unexpected checked exception was received in a remote
- * procedure call.
- *
- * @author unknown
- * @since 1.1
- * @status updated to 1.4
- */
-public class UnexpectedException extends RemoteException
-{
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = 1800467484195073863L;
-
-  /**
-   * Create an exception with a message.
-   *
-   * @param s the message
-   */
-  public UnexpectedException(String s)
-  {
-    super(s);
-  }
-
-  /**
-   * Create an exception with a message and a cause.
-   *
-   * @param s the message
-   * @param e the cause
-   */
-  public UnexpectedException(String s, Exception e)
-  {
-    super(s, e);
-  }
+public class UnexpectedException
+	extends RemoteException {
+
+public UnexpectedException(String s) {
+	super(s);
+}
+
+public UnexpectedException(String s, Exception e) {
+	super(s, e);
+}
+
 }
