--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultButtonModel.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/DefaultButtonModel.java	2004-02-12 02:18:27.000000000 -0700
@@ -37,13 +37,15 @@
 
 package javax.swing;
 
+import java.awt.AWTEvent;
+import java.awt.AWTEventMulticaster;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 import java.awt.event.ItemEvent;
 import java.awt.event.ItemListener;
 import java.io.Serializable;
 import java.util.EventListener;
-
+import java.util.Vector;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
 import javax.swing.event.EventListenerList;
