--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/LogManager.java	2004-10-12 02:18:15.000000000 -0600
+++ java/util/logging/LogManager.java	2004-04-21 02:18:12.000000000 -0600
@@ -351,7 +351,6 @@
   private synchronized Logger findAncestor(Logger child)
   {
     String childName = child.getName();
-    int    childNameLength = childName.length();
     Logger best = rootLogger;
     int    bestNameLength = 0;
 
@@ -367,10 +366,9 @@
       candName = (String) iter.next();
       candNameLength = candName.length();
 
-      if (candNameLength > bestNameLength
-	  && childNameLength > candNameLength
+      if ((candNameLength > bestNameLength)
 	  && childName.startsWith(candName)
-	  && childName.charAt(candNameLength) == '.')
+	  && (childName.charAt(candNameLength) == '.'))
       {
         cand = (Logger) ((WeakReference) loggers.get(candName)).get();
 	if ((cand == null) || (cand == child))
