--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectStreamConstants.java	Mon Apr  7 02:24:27 2003
+++ java/io/ObjectStreamConstants.java	Tue Dec 10 19:20:25 2002
@@ -1,6 +1,6 @@
 /* ObjectStreamConstants.java -- Interface containing constant values
    used in reading and writing serialized objects
-   Copyright (C) 1998, 1999, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,15 +40,14 @@
 package java.io;
 
 /**
- * This interface contains constants that are used in object
- * serialization.  This interface is used by <code>ObjectOutputStream</code>,
- * <code>ObjectInputStream</code>, and <code>ObjectStreamClass</code>.
- * The values for these constants are specified by the Java library
- * specification.
- */
+   This interface contains constants that are used in object
+   serialization.  This interface is used by ObjectOutputStream,
+   ObjectInputStream, ObjectStreamClass, and possibly other classes.
+   The values for these constants are specified in Javasoft's "Object
+   Serialization Specification" TODO: add reference
+*/
 public interface ObjectStreamConstants
 {
-  // FIXME: Javadoc comment these values.
   public final static int PROTOCOL_VERSION_1 = 1;
   public final static int PROTOCOL_VERSION_2 = 2;
 
@@ -86,4 +85,3 @@
   final static SerializablePermission SUBCLASS_IMPLEMENTATION_PERMISSION
     = new SerializablePermission("enableSubclassImplementation");
 }
-
