--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/AlgorithmParameters.java	2003-04-24 02:22:11.000000000 -0600
+++ java/security/AlgorithmParameters.java	2003-04-20 02:20:19.000000000 -0600
@@ -41,8 +41,6 @@
 import java.security.spec.AlgorithmParameterSpec;
 import java.io.IOException;
 
-import gnu.java.security.Engine;
-
 /**
  * <p>This class is used as an opaque representation of cryptographic
  * parameters.</p>
@@ -206,10 +204,6 @@
 	  Engine.getInstance(ALGORITHM_PARAMETERS, algorithm, provider),
 	  provider, algorithm);
       }
-    catch (java.lang.reflect.InvocationTargetException ite)
-      {
-	throw new NoSuchAlgorithmException(algorithm);
-      }
     catch (ClassCastException cce)
       {
 	throw new NoSuchAlgorithmException(algorithm);
