--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ByteOrder.java	2004-02-12 02:22:34.000000000 -0700
+++ java/nio/ByteOrder.java	2003-10-22 02:21:48.000000000 -0600
@@ -63,7 +63,7 @@
    */
   public static ByteOrder nativeOrder ()
   {
-    return (System.getProperty ("gnu.cpu.endian").equals("big")
+    return (System.getProperty ("gnu.cpu.endian") == "big"
             ? BIG_ENDIAN : LITTLE_ENDIAN);
   }
 
