--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormat.java	2004-04-18 02:22:24.000000000 -0600
+++ java/text/DateFormat.java	2004-06-15 02:19:32.000000000 -0600
@@ -325,7 +325,7 @@
     try
       {
 	res = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation",
-				       loc);
+				       loc, DateFormat.class.getClassLoader());
       }
     catch (MissingResourceException x)
       {
