--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/StrictMath.java	2003-07-30 02:20:25.000000000 -0600
+++ java/lang/StrictMath.java	2002-02-14 20:21:47.000000000 -0700
@@ -1,5 +1,5 @@
 /* java.lang.StrictMath -- common mathematical functions, strict Java
-   Copyright (C) 1998, 2001, 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1998, 2001, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -1053,7 +1053,7 @@
    * double to <code>x / y</code> (ties go to the even n); for a zero
    * remainder, the sign is that of <code>x</code>. If either argument is NaN,
    * the first argument is infinite, or the second argument is zero, the result
-   * is NaN; if x is finite but y is infinite, the result is x.
+   * is NaN; if x is finite but y is infinte, the result is x.
    *
    * @param x the dividend (the top half)
    * @param y the divisor (the bottom half)
