--- /home/tromey/gnu/classpath/classpath/java/rmi/activation/ActivationGroupID.java	Wed Oct  2 11:21:38 2002
+++ java/rmi/activation/ActivationGroupID.java	Tue Jan 22 15:40:25 2002
@@ -39,9 +39,8 @@
 
 import java.io.Serializable;
 
-public class ActivationGroupID implements Serializable
-{
-  static final long serialVersionUID = -1648432278909740833L;
+public class ActivationGroupID
+	implements Serializable {
 
 private ActivationSystem system;
 
