--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PrintStream.java	Tue Apr  1 02:19:07 2003
+++ java/io/PrintStream.java	Tue Apr  1 02:18:00 2003
@@ -38,6 +38,8 @@
 
 package java.io;
 
+import gnu.gcj.convert.UnicodeToBytes;
+
 /* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
  * "The Java Language Specification", ISBN 0-201-63451-1
  * Status:  Believed complete and correct to 1.3
@@ -64,23 +66,24 @@
  */
 public class PrintStream extends FilterOutputStream
 {
-
-  /*
-   * Ok, why is this class deprecated?  It could easily have been extended
-   * to support character encodings.  In fact, PrintWriter is basically a
-   * superset of this except for the write() methods.  So let's do something
-   * tricky here and just redirect calls in this class to a hidden PrintWriter
-   * instance.  All the functionality goes there since that is the 'real'
-   * class.  The big win of doing this way is that the default character 
-   * encoding is done automagicially by the PrintWriter tree!
-   */
-
-  /**
-   * This boolean indicates whether or not an error has ever occured
-   * on this stream.
-   */
-  private boolean error_occurred;
-
+  /* Notice the implementation is quite similar to OutputStreamWriter.
+   * This leads to some minor duplication, because neither inherits
+   * from the other, and we want to maximize performance. */
+
+  // Line separator string.
+  private static final char[] line_separator
+    = System.getProperty("line.separator").toCharArray();
+  
+  UnicodeToBytes converter;
+
+  // Work buffer of characters for converter.
+  char[] work = new char[100];
+  // Work buffer of bytes where we temporarily keep converter output.
+  byte[] work_bytes = new byte[100];
+
+  // True if error occurred.
+  private boolean error;
+  // True if auto-flush.
   /**
    * This is <code>true</code> if auto-flush is enabled, 
    * <code>false</code> otherwise
@@ -88,16 +91,6 @@
   private boolean auto_flush;
 
   /**
-   * The PrintWriter instance this object writes to
-   */
-  private PrintWriter pw;
-
-  /**
-   * Lets us know if the stream is closed
-   */
-  private boolean closed;
-
-  /**
    * This method intializes a new <code>PrintStream</code> object to write
    * to the specified output sink.  Note that this class is deprecated in
    * favor of <code>PrintWriter</code>.
@@ -129,36 +122,12 @@
   {
     super(out);
 
-    pw = new PrintWriter(out, auto_flush);
+    converter = UnicodeToBytes.getDefaultEncoder();
+    error = false;
     this.auto_flush = auto_flush;
   }
 
   /**
-   * This method intializes a new <code>PrintStream</code> object to write
-   * to the specified output sink.  This constructor also allows "auto-flush"
-   * functionality to be specified where the stream will be flushed after
-   * every line is terminated or newline character is written.
-   * <p>
-   * Note that this class is deprecated in favor of <code>PrintWriter</code>.
-   *
-   * @param out The <code>OutputStream</code> to write to.
-   * @param autoFlush <code>true</code> to flush the stream after every 
-   * line, <code>false</code> otherwise
-   * @param encoding The name of the character encoding to use for this
-   * object.
-   *
-   * @deprecated
-   */
-  public PrintStream (OutputStream out, boolean autoFlush, String encoding)
-    throws UnsupportedEncodingException
-  {
-    super (out);
-
-    pw = new PrintWriter (new OutputStreamWriter (out, encoding), autoFlush);
-    this.auto_flush = autoFlush;
-  }
-
-  /**
    * This method checks to see if an error has occurred on this stream.  Note
    * that once an error has occurred, this method will continue to report
    * <code>true</code> forever for this stream.  Before checking for an
@@ -169,13 +138,8 @@
    */
   public boolean checkError ()
   {
-    if (!closed)
-      pw.flush ();
-
-    if (pw.checkError () | error_occurred)
-      return true;
-    else
-      return false;
+    flush();
+    return error;
   }
 
   /**
@@ -184,16 +148,27 @@
    */
   protected void setError ()
   {
-    error_occurred = true;
+    error = true;
   }
 
   /**
    * This method closes this stream and all underlying streams.
    */
-  public synchronized void close ()
+  public void close ()
   {
-    pw.close ();
-    closed = true;
+    try
+      {
+	flush();
+	out.close();
+      }
+    catch (InterruptedIOException iioe)
+      {
+	Thread.currentThread().interrupt();
+      }
+    catch (IOException e)
+      {
+	setError ();
+      }
   }
 
   /**
@@ -202,7 +177,85 @@
    */
   public void flush ()
   {
-    pw.flush();
+    try
+      {
+	out.flush();
+      }
+    catch (InterruptedIOException iioe)
+      {
+	Thread.currentThread().interrupt();
+      }
+    catch (IOException e)
+      {
+	setError ();
+      }
+  }
+
+  private synchronized void print (String str, boolean println)
+  {
+    try
+      {
+        writeChars(str, 0, str.length());
+	if (println)
+	  writeChars(line_separator, 0, line_separator.length);
+	if (auto_flush)
+	  flush();
+      }
+    catch (InterruptedIOException iioe)
+      {
+	Thread.currentThread().interrupt();
+      }
+    catch (IOException e)
+      {
+	setError ();
+      }
+  }
+
+  private synchronized void print (char[] chars, int pos, int len,
+				   boolean println)
+  {
+    try
+      {
+        writeChars(chars, pos, len);
+	if (println)
+	  writeChars(line_separator, 0, line_separator.length);
+	if (auto_flush)
+	  flush();
+      }
+    catch (InterruptedIOException iioe)
+      {
+	Thread.currentThread().interrupt();
+      }
+    catch (IOException e)
+      {
+	setError ();
+      }
+  }
+
+  private void writeChars(char[] buf, int offset, int count)
+    throws IOException
+  {
+    while (count > 0)
+      {
+	converter.setOutput(work_bytes, 0);
+	int converted = converter.write(buf, offset, count);
+	offset += converted;
+	count -= converted;
+	out.write(work_bytes, 0, converter.count);
+      }
+  }
+
+  private void writeChars(String str, int offset, int count)
+    throws IOException
+  {
+    while (count > 0)
+      {
+	converter.setOutput(work_bytes, 0);
+	int converted = converter.write(str, offset, count, work);
+	offset += converted;
+	count -= converted;
+	out.write(work_bytes, 0, converter.count);
+      }
   }
 
   /**
@@ -214,7 +267,7 @@
    */
   public void print (boolean bool)
   {
-    pw.print (bool);
+    print(String.valueOf(bool), false);
   }
 
   /**
@@ -225,7 +278,7 @@
    */
   public void print (int inum)
   {
-    pw.print (inum);
+    print(String.valueOf(inum), false);
   }
 
   /**
@@ -236,7 +289,7 @@
    */
   public void print (long lnum)
   {
-    pw.print (lnum);
+    print(String.valueOf(lnum), false);
   }
 
   /**
@@ -247,7 +300,7 @@
    */
   public void print (float fnum)
   {
-    pw.print (fnum);
+    print(String.valueOf(fnum), false);
   }
 
   /**
@@ -258,7 +311,7 @@
    */
   public void print (double dnum)
   {
-    pw.print (dnum);
+    print(String.valueOf(dnum), false);
   }
 
   /**
@@ -270,9 +323,7 @@
    */
   public void print (Object obj)
   {
-    // Don't call pw directly.  Convert to String so we scan for newline
-    // characters on auto-flush;
-    print(String.valueOf(obj));
+    print(obj == null ? "null" : obj.toString(), false);
   }
 
   /**
@@ -283,11 +334,7 @@
    */
   public void print (String str)
   {
-    pw.print (str);
-
-    if (auto_flush)
-      if ((str.indexOf('\r') != -1) || (str.indexOf('\n') != -1))
-        flush();
+    print(str == null ? "null" : str, false);
   }
 
   /**
@@ -296,14 +343,10 @@
    *
    * @param ch The <code>char</code> value to be printed
    */
-  public void print (char ch)
+  public synchronized void print (char ch)
   {
-    pw.print (ch);
-
-    if (auto_flush)
-      if ((ch == '\r')
-          || (ch == '\n'))
-        flush ();    
+    work[0] = ch;
+    print(work, 0, 1, false);
   }
 
   /**
@@ -314,15 +357,7 @@
    */
   public void print (char[] charArray)
   {
-    pw.print(charArray);
-
-    if (auto_flush)
-      for (int i = 0; i < charArray.length; i++)
-        if ((charArray[i] == '\r') || (charArray[i] == '\n'))
-          {
-            flush();
-            break;
-          }
+    print(charArray, 0, charArray.length, false);
   }
 
   /**
@@ -332,7 +367,7 @@
    */
   public void println ()
   {
-    pw.println ();
+    print(line_separator, 0, line_separator.length, false);
   }
 
   /**
@@ -346,7 +381,7 @@
    */
   public void println (boolean bool)
   {
-    pw.println (bool);
+    print(String.valueOf(bool), true);
   }
 
   /**
@@ -359,7 +394,7 @@
    */
   public void println (int inum)
   {
-    pw.println (inum);
+    print(String.valueOf(inum), true);
   }
 
   /**
@@ -372,7 +407,7 @@
    */
   public void println (long lnum)
   {
-    pw.println (lnum);
+    print(String.valueOf(lnum), true);
   }
 
   /**
@@ -385,7 +420,7 @@
    */
   public void println (float fnum)
   {
-    pw.println (fnum);
+    print(String.valueOf(fnum), true);
   }
 
   /**
@@ -398,7 +433,7 @@
    */
   public void println (double dnum)
   {
-    pw.println (dnum);
+    print(String.valueOf(dnum), true);
   }
 
   /**
@@ -412,7 +447,7 @@
    */
   public void println (Object obj)
   {
-    pw.println (obj);
+    print(obj == null ? "null" : obj.toString(), true);
   }
 
   /**
@@ -425,7 +460,7 @@
    */
   public void println (String str)
   {
-    pw.println (str);
+    print (str == null ? "null" : str, true);
   }
 
   /**
@@ -436,9 +471,10 @@
    *
    * @param ch The <code>char</code> value to be printed
    */
-  public void println (char ch)
+  public synchronized void println (char ch)
   {
-    pw.println (ch);
+    work[0] = ch;
+    print(work, 0, 1, true);
   }
 
   /**
@@ -451,7 +487,7 @@
    */
   public void println (char[] charArray)
   {
-    pw.println (charArray);
+    print(charArray, 0, charArray.length, true);
   }
 
   /**
@@ -461,23 +497,21 @@
    * 
    * @param b The byte to be written
    */
-  public synchronized void write (int oneByte)
+  public void write (int oneByte)
   {
-    // Sigh, we actually have to implement this method. Flush first so that
-    // things get written in the right order.
-    flush ();
-
     try
       {
-        out.write (oneByte);
-
-        if (auto_flush)
-          if ((oneByte == '\n') || (oneByte == '\n'))
-            flush ();
+	out.write(oneByte);
+	if (auto_flush && oneByte == '\n')
+	  flush();
+      }
+    catch (InterruptedIOException iioe)
+      {
+	Thread.currentThread().interrupt();
       }
     catch (IOException e)
       {
-        error_occurred = true;
+	setError ();
       }
   }
 
@@ -489,27 +523,22 @@
    * @param offset The index into the array to start writing from
    * @param len The number of bytes to write
    */
-  public synchronized void write (byte[] buffer, int offset, int len)
+  public void write (byte[] buffer, int offset, int len)
   {
-    // We actually have to implement this method too. Flush first so that
-    // things get written in the right order.
-    flush();
-
     try
       {
-        out.write(buffer, offset, len);
-
-        if (auto_flush)
-          for (int i = offset; i < len; i++)
-            if ((buffer[i] == '\r') || (buffer[i] == '\n'))
-              {
-                flush();
-                break;
-              }
+	out.write (buffer, offset, len);
+        
+	if (auto_flush)
+	  flush();
+      }
+    catch (InterruptedIOException iioe)
+      {
+	Thread.currentThread().interrupt();
       }
     catch (IOException e)
       {
-        error_occurred = true;
+	setError ();
       }
   }
 
