--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormat.java	2004-11-17 02:24:08.000000000 -0700
+++ java/text/DecimalFormat.java	2004-10-28 02:16:12.000000000 -0600
@@ -447,8 +447,8 @@
 	    && useExponentialNotation == dup.useExponentialNotation);
   }
 
-  private void formatInternal (double number, FormatBuffer dest,
-			       FieldPosition fieldPos)
+  protected void formatInternal (double number, FormatBuffer dest,
+				 FieldPosition fieldPos)
   {
     // A very special case.
     if (Double.isNaN(number))
