--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicButtonUI.java	2004-06-08 02:25:41.000000000 -0600
+++ javax/swing/plaf/basic/BasicButtonUI.java	2004-06-18 02:18:57.000000000 -0600
@@ -304,11 +304,7 @@
     Icon i = currentIcon(b);
 
     if (i != null)
-      {
-        int x = iconRect.x;
-        int y = iconRect.y;
-        i.paintIcon(c, g, x, y);
-      }
+      i.paintIcon(c, g, iconRect.x, iconRect.y);
   }
 
   /**
@@ -340,7 +336,7 @@
    */
   protected void paintButtonNormal(Graphics g, Rectangle area, JComponent b)
   {
-    if (((AbstractButton)b).isContentAreaFilled())
+    if (((AbstractButton)b).isContentAreaFilled() && b.isOpaque())
       {
         g.setColor(b.getBackground());
         g.fillRect(area.x, area.y, area.width, area.height);
@@ -362,9 +358,18 @@
     Font f = c.getFont();
     g.setFont(f);
     FontMetrics fm = g.getFontMetrics(f);
-    g.setColor(c.getForeground());
-    BasicGraphicsUtils.drawString(g, text, 0,
-                                  textRect.x, 
-                                  textRect.y + fm.getAscent());
+
+    if (c.isEnabled())
+      {
+	g.setColor(c.getForeground());
+	g.drawString(text, textRect.x, textRect.y + fm.getAscent());
+      }
+    else
+      {
+	g.setColor(c.getBackground().brighter());
+	g.drawString(text, textRect.x, textRect.y + fm.getAscent());
+	g.setColor(c.getBackground().darker());
+	g.drawString(text, textRect.x + 1, textRect.y + fm.getAscent() + 1);
+      }
   } 
 }
