--- /home/tromey/gnu/classpath/classpath/java/awt/List.java	Mon Dec 30 15:39:04 2002
+++ java/awt/List.java	Tue Dec 10 19:20:24 2002
@@ -113,8 +113,6 @@
 /**
   * Initializes a new instance of <code>List</code> with no visible lines
   * and multi-select disabled.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 List()
@@ -129,8 +127,6 @@
   * number of visible lines and multi-select disabled.
   *
   * @param lines The number of visible lines in the list.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 List(int rows)
@@ -147,17 +143,12 @@
   * @param lines The number of visible lines in the list.
   * @param multipleMode <code>true</code> if multiple lines can be selected
   * simultaneously, <code>false</code> otherwise.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public 
 List(int rows, boolean multipleMode)
 {
   this.rows = rows;
   this.multipleMode = multipleMode;
-
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
