--- /home/tromey/gnu/Nightly/classpath/classpath/java/beans/BeanDescriptor.java	2004-03-09 02:21:52.000000000 -0700
+++ java/beans/BeanDescriptor.java	2003-12-05 02:18:14.000000000 -0700
@@ -1,5 +1,5 @@
 /* java.beans.BeanDescriptor
-   Copyright (C) 1998, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -67,14 +67,6 @@
 	public BeanDescriptor(Class beanClass, Class customizerClass) {
 		this.beanClass = beanClass;
 		this.customizerClass = customizerClass;
-
-		// Set the FeatureDescriptor programmatic name.
-		String name = beanClass.getName();
-		int lastInd = name.lastIndexOf('.');
-		if (lastInd != -1)
-		  name = name.substring(lastInd + 1);
-
-		setName(name);
 	}
 
 	/** Get the Bean's class. **/
