--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/ScatteringByteChannel.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/ScatteringByteChannel.java	2003-10-13 02:17:50.000000000 -0600
@@ -1,4 +1,4 @@
-/* ScatteringByteChannel.java --
+/* ScatteringByteChannel.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -37,11 +37,11 @@
 
 package java.nio.channels;
 
-import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.io.IOException;
 
-
-public interface ScatteringByteChannel extends ReadableByteChannel
+public interface ScatteringByteChannel
+  extends ReadableByteChannel
 {
   /**
    * Reads a sequence of bytes from this channel into a subsequence of the
@@ -59,7 +59,7 @@
    * @exception NonReadableChannelException If this channel was not opened for
    * reading
    */
-  long read(ByteBuffer[] srcs, int offset, int length)
+  long read (ByteBuffer[] srcs, int offset, int length)
     throws IOException;
 
   /**
@@ -75,5 +75,5 @@
    * @exception NonReadableChannelException If this channel was not opened for
    * reading
    */
-  long read(ByteBuffer[] srcs) throws IOException;
+  long read (ByteBuffer[] srcs) throws IOException;
 }
