--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Calendar.java	2004-11-03 02:21:41.000000000 -0700
+++ java/util/Calendar.java	2004-10-28 02:16:12.000000000 -0600
@@ -347,13 +347,6 @@
   private int minimalDaysInFirstWeek;
 
   /**
-   * Is set to true if DST_OFFSET is explicitly set. In that case
-   * it's value overrides the value computed from the current
-   * time and the timezone. 
-   */
-  private boolean explicitDSTOffset = false;
-
-  /**
    * The version of the serialized data on the stream. 
    * <dl><dt>0 or not present</dt>
    * <dd> JDK 1.1.5 or later.</dd>
@@ -582,9 +575,9 @@
    */
   public void setTimeInMillis(long time)
   {
-    clear();
     this.time = time;
     isTimeSet = true;
+    computeFields();
   }
 
   /**
@@ -657,12 +650,10 @@
       case HOUR:
 	isSet[HOUR_OF_DAY] = false;
 	break;
-      case DST_OFFSET:
-        explicitDSTOffset = true;
       }
 
     // May have crossed over a DST boundary.
-    if (!explicitDSTOffset && (field != DST_OFFSET && field != ZONE_OFFSET))
+    if (field != DST_OFFSET && field != ZONE_OFFSET)
       isSet[DST_OFFSET] = false;
   }
 
@@ -685,8 +676,7 @@
     isSet[DAY_OF_WEEK] = false;
     isSet[DAY_OF_WEEK_IN_MONTH] = false;
 
-    if (!explicitDSTOffset)
-      isSet[DST_OFFSET] = false;  // May have crossed a DST boundary.
+    isSet[DST_OFFSET] = false;  // May have crossed a DST boundary.
   }
 
   /**
