--- /home/tromey/gnu/classpath/classpath/java/nio/channels/DatagramChannel.java	Sun Nov 24 16:32:34 2002
+++ java/nio/channels/DatagramChannel.java	Wed Nov 13 09:30:36 2002
@@ -61,8 +61,6 @@
 
   /**
    * Opens a datagram channel.
-   *
-   * @exception IOException If an error occurs
    */
   public static DatagramChannel open () throws IOException
   {
@@ -84,9 +82,6 @@
   
   /**
    * Writes data to this channel.
-   *
-   * @exception IOException If an error occurs
-   * @exception NotYetConnectedException The channel's socket is not connected.
    */
   public final long write (ByteBuffer[] srcs)
   {
@@ -100,23 +95,16 @@
     
   /**
    * Connects this channel's socket.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract DatagramChannel connect (SocketAddress remote);
 
   /**
    * Disonnects this channel's socket.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract DatagramChannel disconnect ();
 
   /**
    * Tells whether or not this channel's socket is connected.
-   *
-   * @exception IOException If an error occurs.
-   * @exception NotYetConnectedException The channel's socket is not connected.
    */
   public abstract boolean isConnected ();
   
@@ -127,23 +115,16 @@
   
   /**
    * Reads data from this channel.
-   *
-   * @exception IOException If an error occurs.
-   * @exception NotYetConnectedException The channel's socket is not connected.
    */
   public abstract long read (ByteBuffer[] dsts, int offset, int length);
  
   /**
    * Receives a datagram via this channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract SocketAddress receive (ByteBuffer dst);
  
   /**
    * Sends a datagram via this channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract int send (ByteBuffer src, SocketAddress target);
  
@@ -154,25 +135,16 @@
   
   /**
    * Writes data to this channel.
-   *
-   * @exception IOException If an error occurs.
-   * @exception NotYetConnectedException The channel's socket is not connected.
    */
   public abstract int write (ByteBuffer src);
   
   /**
    * Writes data to this channel.
-   *
-   * @exception IOException If an error occurs.
-   * @exception NotYetConnectedException The channel's socket is not connected.
    */
   public abstract long write (ByteBuffer[] srcs, int offset, int length);
 
   /**
    * Retrieves the valid operations for this channel.
-   *
-   * @exception IOException If an error occurs.
-   * @exception NotYetConnectedException The channel's socket is not connected.
    */
   public final int validOps ()
   {
