--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/Handler.java	2003-07-17 02:19:22.000000000 -0600
+++ java/util/logging/Handler.java	2003-06-21 04:31:55.000000000 -0600
@@ -379,7 +379,7 @@
    */
   public boolean isLoggable(LogRecord record)
   {
-    if (record.getLevel().intValue() <= level.intValue())
+    if (record.getLevel().intValue() < level.intValue())
       return false;
     
     if (filter != null)
