--- /home/tromey/gnu/Nightly/classpath/classpath/java/beans/IndexedPropertyDescriptor.java	2003-10-17 02:18:39.000000000 -0600
+++ java/beans/IndexedPropertyDescriptor.java	2002-01-22 16:01:32.000000000 -0700
@@ -1,5 +1,5 @@
 /* java.beans.IndexedPropertyDescriptor
-   Copyright (C) 1998, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -216,7 +216,7 @@
 		this.getMethod = getMethod;
 		this.setMethod = setMethod;
 		this.getIndex = getIndex;
-		this.setIndex = setIndex;
+		this.setIndex = getIndex;
 		this.indexedPropertyType = getIndex != null ? getIndex.getReturnType() : setIndex.getParameterTypes()[1];
 		this.propertyType = getMethod != null ? getMethod.getReturnType() : (setMethod != null ? setMethod.getParameterTypes()[0] : Array.newInstance(this.indexedPropertyType,0).getClass());
 	}
