--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectOutputStream.java	2004-10-12 02:18:12.000000000 -0600
+++ java/io/ObjectOutputStream.java	2004-10-10 02:15:37.000000000 -0600
@@ -893,7 +893,7 @@
    *
    * XXX: finish up comments
    */
-  public abstract static class PutField
+  public static abstract class PutField
   {
     public abstract void put (String name, boolean value);
     public abstract void put (String name, byte value);
@@ -1552,7 +1552,7 @@
   }
 
   // this value comes from 1.2 spec, but is used in 1.1 as well
-  private static final int BUFFER_SIZE = 1024;
+  private final static int BUFFER_SIZE = 1024;
 
   private static int defaultProtocolVersion = PROTOCOL_VERSION_2;
 
