--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/DigestInputStream.java	2003-11-29 02:20:16.000000000 -0700
+++ java/security/DigestInputStream.java	2002-01-22 16:01:36.000000000 -0700
@@ -1,5 +1,5 @@
 /* DigestInputStream.java --- An Input stream tied to a message digest
-   Copyright (C) 1999, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -138,7 +138,7 @@
     int temp = in.read(b, off, len);
 
     if (state == true && temp != -1)
-      digest.update(b, off, temp);
+      digest.update(b, off, len);
 
     return temp;
   }
