--- /home/tromey/gnu/Nightly/classpath/classpath/java/math/BigDecimal.java	2004-04-28 02:20:16.000000000 -0600
+++ java/math/BigDecimal.java	2003-08-02 02:18:05.000000000 -0600
@@ -271,6 +271,7 @@
       return newScale == 0 ? ZERO : new BigDecimal (ZERO.intVal, newScale);
     
     // Ensure that pow gets a non-negative value.
+    int valScale = val.scale;
     BigInteger valIntVal = val.intVal;
     int power = newScale - (scale - val.scale);
     if (power < 0)
