--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormatSymbols.java	2004-03-09 02:21:57.000000000 -0700
+++ java/text/DateFormatSymbols.java	2003-08-27 02:17:52.000000000 -0600
@@ -1,4 +1,4 @@
-/* DateFormatSymbols.java -- Format over a range of numbers
+/* ChoiceFormat.java -- Format over a range of numbers
    Copyright (C) 1998, 1999, 2000, 2001, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
