--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/ServerSocket.java	2004-02-06 02:21:25.000000000 -0700
+++ java/net/ServerSocket.java	2004-02-12 02:18:26.000000000 -0700
@@ -471,7 +471,7 @@
     if (isClosed())
       throw new SocketException("ServerSocket is closed");
     
-    impl.setOption (SocketOptions.SO_REUSEADDR, new Boolean (on));
+    impl.setOption (SocketOptions.SO_REUSEADDR, Boolean.valueOf(on));
   }
 
   /**
