--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Package.java	2004-04-18 02:22:18.000000000 -0600
+++ java/lang/Package.java	2004-01-06 02:18:20.000000000 -0700
@@ -227,10 +227,7 @@
    * <code>getSpecificationVersion</code>.
    *
    * @param version the (minimal) desired version of the specification
-   *
-   * @return true if the version is compatible, false otherwise
-   *
-   * @Throws NumberFormatException if either version string is invalid
+   * @throws NumberFormatException if either version string is invalid
    * @throws NullPointerException if either version string is null
    */
   public boolean isCompatibleWith(String version)
