--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/Popup.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/Popup.java	2003-06-27 06:41:52.000000000 -0600
@@ -163,7 +163,7 @@
 
 
     /**
-     * Displays the popup's <code>JWindow</code> on the screen.
+     * Displays the popup&#x2019;s <code>JWindow</code> on the screen.
      * Nothing happens if it is already visible.
      */
     public void show()
@@ -173,7 +173,7 @@
     
     
     /**
-     * Removes the popup's <code>JWindow</code> from the
+     * Removes the popup&#x2019;s <code>JWindow</code> from the
      * screen.  Nothing happens if it is currently not visible.
      */
     public void hide()
