--- /home/tromey/gnu/egcs/classpath/classpath/java/util/jar/Attributes.java	Mon Mar 25 15:38:05 2002
+++ java/util/jar/Attributes.java	Tue Jan 22 16:01:45 2002
@@ -430,7 +430,7 @@
    * @returns the old value of the attribute name or null if it didn't exist
    * yet
    */
-  String putValue(Name name, String value)
+  private String putValue(Name name, String value)
   {
     return (String) put(name, value);
   }
