--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Byte.java	2004-04-18 02:22:16.000000000 -0600
+++ java/lang/Byte.java	2002-06-15 13:26:39.000000000 -0600
@@ -147,7 +147,7 @@
    *
    * @param s the <code>String</code> to convert
    * @param radix the radix (base) to use in the conversion
-   * @return the <code>String</code> argument converted to <code>byte</code>
+   * @return the <code>String</code> argument converted to </code>byte</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as a
    *         <code>byte</code>
    */
