--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2004-07-02 02:22:56.000000000 -0600
+++ java/net/URLStreamHandler.java	2004-07-07 02:18:39.000000000 -0600
@@ -131,8 +131,12 @@
     String query = null;
     
     // On Windows we need to change \ to / for file URLs
-    if (url.getProtocol().equals("file"))
-      spec = spec.replace(File.separatorChar, '/');
+    char separator = File.separatorChar;
+    if (url.getProtocol().equals("file") && separator != '/')
+      {
+	file = file.replace(separator, '/');
+	spec = spec.replace(separator, '/');
+      }
 
     if (spec.regionMatches(start, "//", 0, 2))
       {
@@ -204,11 +208,9 @@
       {
 	// Context is available, but only override it if there is a new file.
 	int lastSlash = file.lastIndexOf('/');
-	if (lastSlash < 0)
-	  file = spec.substring(start, end);
-	else
-	  file = (file.substring(0, lastSlash)
-		  + '/' + spec.substring(start, end));
+
+	file =
+	  file.substring(0, lastSlash) + '/' + spec.substring(start, end);
 
 	if (url.getProtocol().equals("file"))
 	  {
@@ -218,7 +220,7 @@
 	      {
 		boolean endsWithSlash = file.charAt(file.length() - 1) == '/';
 		file = new File(file).getCanonicalPath();
-		file = file.replace(File.separatorChar, '/');
+		file = file.replace(separator, '/');
 		if (endsWithSlash && file.charAt(file.length() - 1) != '/')
 		  file += '/';
 	      }
