--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/CollationKey.java	2004-01-08 02:20:41.000000000 -0700
+++ java/text/CollationKey.java	2004-01-08 02:18:06.000000000 -0700
@@ -78,13 +78,24 @@
   /**
    * This is the bit value for this key.
    */
-  private byte[] key;
+  private int[] key;
 
-  CollationKey (Collator collator, String originalText, byte[] key)
+  CollationKey(Collator collator, CollationElementIterator iter,
+	       String originalText, int strength)
   {
     this.collator = collator;
     this.originalText = originalText;
-    this.key = key;
+
+    // Compute size of required array.
+    int size = 0;
+    while (RuleBasedCollator.next(iter, strength)
+	   != CollationElementIterator.NULLORDER)
+      ++size;
+
+    iter.reset();
+    key = new int[size];
+    for (int i = 0; i < size; i++)
+      key[i] = RuleBasedCollator.next(iter, strength);
   }
 
   /**
@@ -194,6 +205,15 @@
    */
   public byte[] toByteArray()
   {
-    return key;
+    byte[] r = new byte[4 * key.length];
+    int off = 0;
+    for (int i = 0; i < key.length; ++i)
+      {
+	r[off++] = (byte) ((key[i] >>> 24) & 255);
+	r[off++] = (byte) ((key[i] >>> 16) & 255);
+	r[off++] = (byte) ((key[i] >>>  8) & 255);
+	r[off++] = (byte) ((key[i]       ) & 255);
+      }
+    return r;
   }
 }
