--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JViewport.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JViewport.java	2004-06-18 02:18:54.000000000 -0600
@@ -45,6 +45,7 @@
 import java.awt.Point;
 import java.awt.Rectangle;
 
+import javax.accessibility.Accessible;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
 import javax.swing.plaf.ViewportUI;
@@ -91,19 +92,20 @@
  * the underlying child at position <code>(-VX,-VY)</code></p>
  *
  */
-
 public class JViewport extends JComponent
 {
-  public static int BACKINGSTORE_SCROLL_MODE = 1;
-  public static int BLIT_SCROLL_MODE = 2;
-  public static int SIMPLE_SCROLL_MODE = 3;
+  private static final long serialVersionUID = -6925142919680527970L;
+  
+  public static final int SIMPLE_SCROLL_MODE = 0;
+  public static final int BLIT_SCROLL_MODE = 1;
+  public static final int BACKINGSTORE_SCROLL_MODE = 2;
 
   ChangeEvent changeEvent = new ChangeEvent(this);
 
   int scrollMode;
 
-  boolean scrollUnderway;
-  boolean isViewSizeSet;
+  protected boolean scrollUnderway;
+  protected boolean isViewSizeSet;
 
   /** 
    * The width and height of the Viewport's area in terms of view
@@ -133,11 +135,23 @@
   public Dimension getExtentSize()
   {
     if (extentSize == null)
-      return getPreferredSize();
+      return toViewCoordinates(getSize());
     else
       return extentSize;
   }
 
+  public Dimension toViewCoordinates(Dimension size)
+  {
+    return size;
+  }
+
+  public Point toViewCoordinates(Point p)
+  {
+    Point pos = getViewPosition();
+    return new Point(p.x + pos.x,
+                     p.y + pos.y);
+  }
+
   public void setExtentSize(Dimension newSize)
   {
     extentSize = newSize;
@@ -146,10 +160,10 @@
 
   public Dimension getViewSize()
   {
-    if (viewSize == null)
-      return getView().getPreferredSize();
-    else
+    if (isViewSizeSet)
       return viewSize;
+    else
+      return getView().getSize();
   }
 
 
@@ -158,10 +172,17 @@
     viewSize = newSize;
     Component view = getView();
     if (view != null)
-      view.setSize(newSize);
+      view.setSize(viewSize);
+    isViewSizeSet = true;
     fireStateChanged();
   }
 
+  /**
+   * Get the viewport's position in view space. Despite confusing name,
+   * this really does return the viewport's (0,0) position in view space,
+   * not the view's position.
+   */
+
   public Point getViewPosition()
   {
     Component view = getView();
