--- /home/tromey/gnu/classpath/classpath/java/util/zip/DataFormatException.java	Sat Mar  2 12:13:22 2002
+++ java/util/zip/DataFormatException.java	Tue Jan 22 16:01:46 2002
@@ -1,5 +1,5 @@
-/* DataformatException.java -- thrown when compressed data is corrupt
-   Copyright (C) 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
+/* DataformatException.java - Exception thrown when compressed data is corrupt
+   Copyright (C) 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,34 +37,25 @@
 
 package java.util.zip;
 
+/* Written using on-line Java Platform 1.2 API Specification.
+ * Believed complete and correct.
+ */
+
 /**
  * Exception thrown when compressed data is corrupt.
  *
  * @author Tom Tromey
  * @author John Leuner
- * @since 1.1
- * @status updated to 1.4
+ * @since JDK 1.1
  */
 public class DataFormatException extends Exception
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = 2219632870893641452L;
-
-  /**
-   * Create an exception without a message.
-   */
-  public DataFormatException()
+  public DataFormatException ()
   {
+    super();
   }
 
-  /**
-   * Create an exception with a message.
-   *
-   * @param msg the message
-   */
-  public DataFormatException(String msg)
+  public DataFormatException (String msg)
   {
     super(msg);
   }
