--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/ServerSocket.java	2004-10-16 02:22:18.000000000 -0600
+++ java/net/ServerSocket.java	2004-04-22 02:18:15.000000000 -0600
@@ -39,7 +39,6 @@
 package java.net;
 
 import gnu.java.net.PlainSocketImpl;
-
 import java.io.IOException;
 import java.nio.channels.IllegalBlockingModeException;
 import java.nio.channels.ServerSocketChannel;
@@ -341,7 +340,6 @@
 	  }
 	catch (IOException e2)
 	  {
-	    // Ignore.
 	  }
 
 	throw e;
