--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLabelUI.java	2004-07-09 02:22:40.000000000 -0600
+++ javax/swing/plaf/basic/BasicLabelUI.java	2004-05-27 02:19:07.000000000 -0600
@@ -59,7 +59,7 @@
 
 /**
  * This is the Basic Look and Feel class for the JLabel.  One BasicLabelUI
- * object is used to paint all JLabels that utilize the Basic Look and Feel.
+ * object is used to paint all JLabels that utilize the Basic L&F.
  */
 public class BasicLabelUI extends LabelUI implements PropertyChangeListener
 {
@@ -81,7 +81,7 @@
    *
    * @param c The {@link JComponent} that a UI is being created for.
    *
-   * @return A label UI for the Basic Look and Feel.
+   * @return A label UI for the Basic L&F.
    */
   public static ComponentUI createUI(JComponent c)
   {
