--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicRadioButtonUI.java	2003-06-19 02:34:52.000000000 -0600
+++ javax/swing/plaf/basic/BasicRadioButtonUI.java	2003-05-10 02:19:19.000000000 -0600
@@ -60,7 +60,14 @@
     public Dimension getPreferredSize(JComponent c) 
     {
 	AbstractButton b = (AbstractButton)c;
-	Dimension d = BasicGraphicsUtils.getPreferredButtonSize(b, gap);
+	Dimension d = BasicGraphicsUtils.getPreferredSize(b, 
+							  gap,
+							  b.getText(),
+							  b.getIcon(),
+							  b.getVerticalAlignment(),
+							  b.getHorizontalAlignment(),
+							  b.getHorizontalTextPosition(),
+							  b.getVerticalTextPosition());
 	
 	// and add a little something for the circles:
 
