--- /home/tromey/gnu/classpath/classpath/java/awt/Scrollbar.java	Mon Dec 30 15:39:04 2002
+++ java/awt/Scrollbar.java	Sun Mar 24 16:08:55 2002
@@ -40,6 +40,7 @@
 
 import java.awt.peer.ScrollbarPeer;
 import java.awt.peer.ComponentPeer;
+
 import java.awt.event.AdjustmentListener;
 import java.awt.event.AdjustmentEvent;
 
@@ -126,8 +127,6 @@
 /**
   * Initializes a new instance of <code>Scrollbar</code> with a
   * veritical orientation and default values for all other parameters.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 Scrollbar()
@@ -146,7 +145,6 @@
   *
   * @param orientation The orientation of this scrollbar.
   *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   * @exception IllegalArgumentException If the orientation value is not valid.
   */
 public
@@ -170,16 +168,12 @@
   * @param minimum The minimum value of the scrollbar.
   * @param maximum The maximum value of the scrollbar.
   *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   * @exception IllegalArgumentException If the orientation value is not valid.
   */
 public 
 Scrollbar(int orientation, int value, int visibleAmount, int minimum, 
           int maximum) throws IllegalArgumentException
 {
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
-
   if ((orientation != HORIZONTAL) && (orientation != VERTICAL))
     throw new IllegalArgumentException("Bad orientation value: "
 				       + orientation);
