--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2003-07-25 02:18:52.000000000 -0600
+++ java/awt/Component.java	2003-03-03 02:17:48.000000000 -0700
@@ -1452,7 +1452,12 @@
    */
   public Dimension getPreferredSize()
   {
-    return preferredSize();
+    if (prefSize == null)
+      if (peer == null)
+	return new Dimension(width, height);
+      else 
+        prefSize = peer.getPreferredSize();
+    return prefSize;
   }
 
   /**
@@ -1463,12 +1468,7 @@
    */
   public Dimension preferredSize()
   {
-    if (prefSize == null)
-      if (peer == null)
-	return new Dimension(width, height);
-      else 
-        prefSize = peer.getPreferredSize();
-    return prefSize;
+    return getPreferredSize();
   }
 
   /**
@@ -1480,7 +1480,10 @@
    */
   public Dimension getMinimumSize()
   {
-    return minimumSize();
+    if (minSize == null)
+      minSize = (peer != null ? peer.getMinimumSize()
+                 : new Dimension(width, height));
+    return minSize;
   }
 
   /**
@@ -1491,10 +1494,7 @@
    */
   public Dimension minimumSize()
   {
-    if (minSize == null)
-      minSize = (peer != null ? peer.getMinimumSize()
-                 : new Dimension(width, height));
-    return minSize;
+    return getMinimumSize();
   }
 
   /**
@@ -1941,10 +1941,7 @@
   public boolean prepareImage(Image image, int width, int height,
                               ImageObserver observer)
   {
-    if (peer != null)
-	return peer.prepareImage(image, width, height, observer);
-    else
-	return getToolkit().prepareImage(image, width, height, observer);
+    return peer.prepareImage(image, width, height, observer);
   }
 
   /**
@@ -1960,7 +1957,8 @@
    */
   public int checkImage(Image image, ImageObserver observer)
   {
-    return checkImage(image, -1, -1, observer);
+    return checkImage(image, image.getWidth(observer),
+                      image.getHeight(observer), observer);
   }
 
   /**
