--- /home/tromey/gnu/classpath/classpath/java/rmi/ServerRuntimeException.java	Sat Mar  2 12:13:14 2002
+++ java/rmi/ServerRuntimeException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* ServerRuntimeException.java -- wraps an exception while creating the server
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,31 +37,11 @@
 
 package java.rmi;
 
-/**
- * Wraps any runtime exception thrown while processing the server of a
- * remote call. Note, this exception is no longer used.
- *
- * @author unknown
- * @since 1.1
- * @deprecated no replacement
- * @status updated to 1.4
- */
-public class ServerRuntimeException extends RemoteException
-{
-  /**
-   * Compatible with JDK 1.1.
-   */
-  private static final long serialVersionUID = 7054464920481467219L;
-
-  /**
-   * Create an exception with a message and a cause.
-   *
-   * @param s the message
-   * @param e the cause
-   * @deprecated no longer needed
-   */
-  public ServerRuntimeException(String s, Exception e)
-  {
-    super(s, e);
-  }
+public class ServerRuntimeException
+	extends RemoteException {
+
+public ServerRuntimeException(String s, Exception e) {
+	super(s, e);
+}
+
 }
