--- /home/tromey/gnu/classpath/classpath/java/rmi/ConnectIOException.java	Sat Mar  2 12:13:14 2002
+++ java/rmi/ConnectIOException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* ConnectIOException.java -- thrown if an IO exception occurs during connect
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,38 +37,16 @@
 
 package java.rmi;
 
-/**
- * Wraps an I/O Exception thrown while connecting for a remote call.
- *
- * @author unknown
- * @since 1.1
- * @status updated to 1.4
- */
-public class ConnectIOException extends RemoteException
-{
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -8087809532704668744L;
-
-  /**
-   * Create an exception with a message.
-   *
-   * @param s the message
-   */
-  public ConnectIOException(String s)
-  {
-    super(s);
-  }
-
-  /**
-   * Create an exception with a message and a cause.
-   *
-   * @param s the message
-   * @param e the cause
-   */
-  public ConnectIOException(String s, Exception e)
-  {
-    super(s, e);
-  }
+public class ConnectIOException 
+	extends RemoteException {
+
+
+public ConnectIOException(String s) {
+	super(s);
+}
+
+public ConnectIOException(String s, Exception e) {
+	super (s, e);
+}
+
 }
