--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/cert/CertStore.java	2003-04-23 17:09:18.000000000 -0600
+++ java/security/cert/CertStore.java	2003-11-12 02:18:14.000000000 -0700
@@ -215,8 +215,7 @@
     catch (java.lang.reflect.InvocationTargetException ite)
       {
         Throwable cause = ite.getCause();
-        if (cause != null &&
-           (cause instanceof InvalidAlgorithmParameterException))
+        if (cause instanceof InvalidAlgorithmParameterException)
           throw (InvalidAlgorithmParameterException) cause;
         else
           throw new NoSuchAlgorithmException(type);
