--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/AlgorithmParameterGenerator.java	2004-04-18 02:22:21.000000000 -0600
+++ java/security/AlgorithmParameterGenerator.java	2003-04-30 02:19:05.000000000 -0600
@@ -67,7 +67,7 @@
  *  system parameters, for example, the parameter generation values usually
  *  consist of the size of the prime modulus and the size of the random
  *  exponent, both specified in number of bits.</li>
- * </ul>
+ * <ul>
  *
  * <p>In case the client does not explicitly initialize the
  * <code>AlgorithmParameterGenerator</code> (via a call to an <code>init()</code>
