--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Byte.java	Mon Jul 23 14:21:55 2001
+++ java/lang/Byte.java	Fri Oct 26 16:18:42 2001
@@ -57,7 +57,7 @@
    * The primitive type <code>byte</code> is represented by this 
    * <code>Class</code> object.
    */
-  public static final Class TYPE = VMClassLoader.getPrimitiveClass("byte");
+  public static final Class TYPE = VMClassLoader.getPrimitiveClass('B');
 
   /**
    * The immutable value of this Byte.
