--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/ClassLoader.java	2004-04-28 02:20:16.000000000 -0600
+++ java/lang/ClassLoader.java	2004-04-21 02:18:09.000000000 -0600
@@ -38,11 +38,9 @@
 
 package java.lang;
 
-import gnu.java.util.DoubleEnumeration;
-import gnu.java.util.EmptyEnumeration;
-
-import java.io.IOException;
 import java.io.InputStream;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
 import java.net.URL;
 import java.security.CodeSource;
 import java.security.PermissionCollection;
@@ -51,6 +49,8 @@
 import java.util.Enumeration;
 import java.util.HashMap;
 import java.util.Map;
+import gnu.java.util.DoubleEnumeration;
+import gnu.java.util.EmptyEnumeration;
 
 /**
  * The ClassLoader is a way of customizing the way Java gets its classes
