--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/spec/DSAPublicKeySpec.java	2003-04-25 02:26:30.000000000 -0600
+++ java/security/spec/DSAPublicKeySpec.java	2002-12-10 19:20:35.000000000 -0700
@@ -62,7 +62,7 @@
      @param q the sub-prime
      @param g the base
   */
-  public DSAPublicKeySpec(BigInteger y, BigInteger p, BigInteger q, BigInteger g) 
+  public DSAPublicKeySpec(BigInteger x, BigInteger p, BigInteger q, BigInteger g) 
   {
     this.y = y;
     this.p = p;
