--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JApplet.java	2004-06-08 02:25:39.000000000 -0600
+++ javax/swing/JApplet.java	2004-06-18 02:18:51.000000000 -0600
@@ -1,5 +1,5 @@
 /* JApplet.java -- 
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -51,6 +51,7 @@
 
 public class JApplet extends Applet implements RootPaneContainer
 {
+  private static final long serialVersionUID = 7269359214497372587L;
 
     public final static int HIDE_ON_CLOSE        = 0;
     public final static int EXIT_ON_CLOSE        = 1;
@@ -136,10 +137,10 @@
     {    return close_action;   }
 
     
-    JMenuBar getJMenuBar()
+    public JMenuBar getJMenuBar()
     {    return getRootPane().getJMenuBar();   }
     
-    void setJMenuBar(JMenuBar menubar)
+    public void setJMenuBar(JMenuBar menubar)
     {    getRootPane().setJMenuBar(menubar); }
     
     
