--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2004-04-28 02:20:17.000000000 -0600
+++ java/net/URLStreamHandler.java	2004-04-22 02:18:15.000000000 -0600
@@ -502,9 +502,12 @@
   protected String toExternalForm(URL url)
   {
     String protocol;
+    String host;
     String file;
     String ref;
+    String user;
     String authority;
+    int port;
 
     protocol = url.getProtocol();
     authority = url.getAuthority();
