--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/ThreadGroup.java	2004-03-15 02:21:10.000000000 -0700
+++ java/lang/ThreadGroup.java	2003-08-27 02:17:52.000000000 -0600
@@ -684,7 +684,7 @@
   {
     if (groups == null)
       return;
-    System.out.println(indentation + this);
+    System.out.print(indentation + this);
     indentation += "    ";
     int i = threads.size();
     while (--i >= 0)
