--- /home/tromey/gnu/classpath/classpath/java/awt/datatransfer/Clipboard.java	Mon Dec 30 15:39:04 2002
+++ java/awt/datatransfer/Clipboard.java	Wed Jan 23 18:32:09 2002
@@ -103,8 +103,6 @@
   * Returns the contents of the clipboard.
   *
   * @param requestor The object requesting the contents.
-  *
-  * @exception IllegalStateException If the clipboard is currently unavailable
   */
 public synchronized Transferable
 getContents(Object requestor)
@@ -122,8 +120,6 @@
   *
   * @param contents The new clipboard contents.
   * @param owner The new clipboard owner
-  *
-  * @exception IllegalStateException If the clipboard is currently unavailable
   */
 public synchronized void
 setContents(Transferable contents, ClipboardOwner owner)
