--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JList.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/JList.java	2004-03-20 02:18:22.000000000 -0700
@@ -99,7 +99,7 @@
  * <tr><td>visibleRowCount                </td><td>list     </td><td>no    </td></tr>
  * </table> 
  *
- * @author Graydon Hoare (graydon@redhat.com)
+ * @author Graydon Hoare (graydon&064;redhat.com)
  */
 
 public class JList extends JComponent implements Accessible, Scrollable
