--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Map.java	2004-07-30 02:31:35.000000000 -0600
+++ java/util/Map.java	2004-04-21 02:18:11.000000000 -0600
@@ -106,10 +106,6 @@
    *
    * @param value the value to search for
    * @return true if the map contains the value
-   * @throws ClassCastException if the type of the value is not a valid type
-   *         for this map.
-   * @throws NullPointerException if the value is null and the map doesn't
-   *         support null values.
    */
   boolean containsValue(Object value);
 
@@ -168,8 +164,7 @@
    * @throws ClassCastException if the key or value is of the wrong type
    * @throws IllegalArgumentException if something about this key or value
    *         prevents it from existing in this map
-   * @throws NullPointerException if either the key or the value is null,
-   *         and the map forbids null keys or values
+   * @throws NullPointerException if the map forbids null keys or values
    * @see #containsKey(Object)
    */
   Object put(Object key, Object value);
@@ -229,12 +224,8 @@
    * @param key the key to remove
    * @return the value the key mapped to, or null if not present
    * @throws UnsupportedOperationException if deletion is unsupported
-   * @throws NullPointerException if the key is null and this map doesn't
-   *         support null keys.
-   * @throws ClassCastException if the type of the key is not a valid type
-   *         for this map.
    */
-  Object remove(Object key);
+  Object remove(Object o);
 
   /**
    * Returns the number of key-value mappings in the map. If there are more
