--- /home/tromey/gnu/classpath/classpath/java/nio/channels/spi/SelectorProvider.java	Sun Nov 17 14:22:22 2002
+++ java/nio/channels/spi/SelectorProvider.java	Wed Nov 13 09:30:36 2002
@@ -37,7 +37,7 @@
 
 package java.nio.channels.spi;
 
-import gnu.java.nio.SelectorProviderImpl;
+/* import gnu.java.nio.channels.SelectorProviderImpl; */
 import java.nio.channels.DatagramChannel;
 import java.nio.channels.Pipe;
 import java.nio.channels.ServerSocketChannel;
@@ -95,8 +95,8 @@
    */
   public static SelectorProvider provider ()
   {
-    if (pr == null)
-      pr = new SelectorProviderImpl ();
+/*    if (pr == null) */
+/*      pr = new SelectorProviderImpl (); */
     
     return pr;
   }
