--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Runnable.java	Wed Mar 20 17:01:40 2002
+++ java/lang/Runnable.java	Tue Jan 22 16:01:35 2002
@@ -1,5 +1,5 @@
-/* Runnable -- interface for a method tied to an Object; often for Threads
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* java.lang.Runnable
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,6 +38,12 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Complete.
+ */
+
 /**
  * Runnable is an interface you implement to indicate that your class can be
  * executed as the main part of a Thread, among other places.  When you want
@@ -46,17 +52,14 @@
  *
  * @author Paul Fisher
  * @author Tom Tromey <tromey@cygnus.com>
- * @see Thread
- * @since 1.0
- * @status updated to 1.4
  */
+
 public interface Runnable
 {
   /**
    * This method will be called by whoever wishes to run your class
-   * implementing Runnable. Note that there are no restrictions on what
-   * you are allowed to do in the run method, except that you cannot
-   * throw a checked exception.
+   * implementing Runnable.
+   * @since JDK1.0
    */
   void run();
 }
