--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Package.java	Mon Oct  1 14:44:13 2001
+++ java/lang/Package.java	Wed Apr 25 09:45:12 2001
@@ -202,22 +202,23 @@
     StringTokenizer versionTokens = new StringTokenizer(version, ".");
     StringTokenizer specTokens = new StringTokenizer(specVersion, ".");
     try
+      {
+	while (versionTokens.hasMoreElements())
 	  {
-        while (versionTokens.hasMoreElements()) {
-        int vers = Integer.parseInt(versionTokens.nextToken());
-        int spec = Integer.parseInt(specTokens.nextToken());
-        if (spec < vers) {
-          return false;
-        } else if (spec > vers) {
-          return true;
-        }
-        // They must be equal, next Token please!
+	    int vers = Integer.parseInt(versionTokens.nextToken());
+	    int spec = Integer.parseInt(specTokens.nextToken());
+	    if (spec < vers)
+	      return false;
+	    else if (spec > vers)
+	      return true;
+	    // They must be equal, next Token please!
+	  }
+      }
+    catch (NoSuchElementException e)
+      {
+	// this must have been thrown by spec.netToken() so return false
+	return false;
       }
- }
-    catch (NoSuchElementException e) {
-      // this must have been thrown by spec.netToken() so return false
-      return false;
-    }
 
     // They must have been exactly the same version.
     // Or the specVersion has more subversions. That is also good.
@@ -233,10 +234,11 @@
    */
   public static Package getPackage(String name)
   {
-    // get the callers classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    // Get the caller's classloader
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
-    
+
     if (cl != null)
       return cl.getPackage(name);
     else
@@ -249,10 +251,11 @@
    */
   public static Package[] getPackages()
   {
-    // get the callers classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    // Get the caller's classloader
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
-      
+
     if (cl != null)
       return cl.getPackages();
     else
