--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectStreamClass.java	2003-08-16 02:19:44.000000000 -0600
+++ java/io/ObjectStreamClass.java	2003-03-24 02:17:54.000000000 -0700
@@ -621,7 +621,7 @@
       {
 	// Use getDeclaredField rather than getField for the same reason
 	// as above in getDefinedSUID.
-	Field f = clazz.getDeclaredField ("serialPersistentFields");
+	Field f = clazz.getDeclaredField ("getSerialPersistentFields");
 	f.setAccessible(true);
 	o = (ObjectStreamField[])f.get (null);
       }
