--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormat.java	2003-11-16 02:19:39.000000000 -0700
+++ java/text/DecimalFormat.java	2003-11-07 02:19:02.000000000 -0700
@@ -37,7 +37,6 @@
 
 package java.text;
 
-import java.util.Currency;
 import java.util.Locale;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
@@ -638,19 +637,6 @@
     return dest;
   }
 
-  /**
-   * Returns the currency corresponding to the currency symbol stored
-   * in the instance of <code>DecimalFormatSymbols</code> used by this
-   * <code>DecimalFormat</code>.
-   *
-   * @return A new instance of <code>Currency</code> if
-   * the currency code matches a known one, null otherwise.
-   */
-  public Currency getCurrency()
-  {
-    return symbols.getCurrency();
-  }
-
   public DecimalFormatSymbols getDecimalFormatSymbols ()
   {
     return symbols;
@@ -870,16 +856,6 @@
     return result;
   }
 
-  /**
-   * Sets the <code>Currency</code> on the
-   * <code>DecimalFormatSymbols</code> used, which also sets the
-   * currency symbols on those symbols.
-   */
-  public void setCurrency(Currency currency)
-  {
-    symbols.setCurrency(currency);
-  }
-
   public void setDecimalFormatSymbols (DecimalFormatSymbols newSymbols)
   {
     symbols = newSymbols;
