--- /home/tromey/gnu/classpath/classpath/java/nio/MappedByteBuffer.java	Mon Dec 30 15:39:05 2002
+++ java/nio/MappedByteBuffer.java	Tue Feb 11 09:41:07 2003
@@ -43,23 +43,23 @@
  */
 public abstract class MappedByteBuffer extends ByteBuffer
 {
-    private static native void sync();
-
-    public final MappedByteBuffer force()
-    {
-	sync();
-	return this;
-    }
+  MappedByteBuffer (int capacity, int limit, int position, int mark)
+  {
+    super (capacity, limit, position, mark);
+  }
+  
+  public final MappedByteBuffer force ()
+  {
+    return this;
+  }
     
-    public final boolean isLoaded()
-    {
-	return true;
-    }
+  public final boolean isLoaded ()
+  {
+    return true;
+  }
     
-    public final MappedByteBuffer load()
-    {
-	sync();
-	return this;
-    }
- 
+  public final MappedByteBuffer load ()
+  {
+    return this;
+  }
 }
