--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBoxMenuItem.java	2004-06-08 02:25:39.000000000 -0600
+++ javax/swing/JCheckBoxMenuItem.java	2004-06-18 02:18:51.000000000 -0600
@@ -50,6 +50,8 @@
 public class JCheckBoxMenuItem extends JMenuItem implements SwingConstants,
                                                             Accessible
 {
+  private static final long serialVersionUID = -6676402307973384715L;
+
   private static final String uiClassID = "CheckBoxMenuItemUI";
   private boolean state;
   private Object[] selectedObjects;
@@ -207,7 +209,7 @@
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      accessibleContext = new AccessibleJCheckBoxMenuItem(this);
+      accessibleContext = new AccessibleJCheckBoxMenuItem();
 
     return accessibleContext;
   }
@@ -217,16 +219,13 @@
    */
   protected class AccessibleJCheckBoxMenuItem extends AccessibleJMenuItem
   {
+    private static final long serialVersionUID = 1079958073579370777L;
+
     /**
      * Creates a new AccessibleJCheckBoxMenuItem object.
-     *
-     * @param component DOCUMENT ME!
      */
-    protected AccessibleJCheckBoxMenuItem(JCheckBoxMenuItem component)
+    protected AccessibleJCheckBoxMenuItem()
     {
-      super(component);
-
-      // TODO
     }
 
     /**
