--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/CompoundName.java	2004-04-28 02:20:18.000000000 -0600
+++ javax/naming/CompoundName.java	2004-04-21 02:18:12.000000000 -0600
@@ -264,7 +264,7 @@
     int delta = elts.size () - cn.elts.size ();
     for (int i = 0; i < cn.elts.size (); ++i)
       {
-	String f = canonicalize ((String) elts.get (delta + i));
+	String f = canonicalize ((String) elts.get (i));
 	if (! f.equals (canonicalize ((String) cn.elts.get (i))))
 	  return false;
       }
