--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectStreamException.java	Mon Apr  7 02:24:27 2003
+++ java/io/ObjectStreamException.java	Sat Jun 15 13:01:33 2002
@@ -1,5 +1,5 @@
 /* ObjectStreamException.java -- Superclass of all serialization exceptions
-   Copyright (C) 1998, 2000, 2001, 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1998, 2000, 2001, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,7 +40,7 @@
 
 /**
  * This exception is thrown when a problem occurs during serialization.
- * There are more specific subclasses that give more fine grained
+ * There are more specific subclasses than give more fine grained
  * indications of the precise failure.
  *
  * @author Aaron M. Renn (arenn@urbanophile.com)
