--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileOutputStream.java	2004-04-17 02:20:15.000000000 -0600
+++ java/io/FileOutputStream.java	2004-03-01 02:18:07.000000000 -0700
@@ -81,7 +81,13 @@
   public FileOutputStream (String path, boolean append)
     throws SecurityException, FileNotFoundException
   {
-    this (new File(path), append);
+    SecurityManager s = System.getSecurityManager();
+    if (s != null)
+      s.checkWrite(path);
+    ch = new FileChannelImpl (path, (append
+				     ? FileChannelImpl.WRITE
+				     | FileChannelImpl.APPEND
+				     : FileChannelImpl.WRITE));
   }
 
   /**
@@ -124,7 +130,7 @@
   public FileOutputStream (File file)
     throws SecurityException, FileNotFoundException
   {
-    this (file, false);
+    this (file.getPath(), false);
   }
 
   /**
@@ -150,17 +156,7 @@
   public FileOutputStream (File file, boolean append)
     throws FileNotFoundException
   {
-    SecurityManager s = System.getSecurityManager();
-    if (s != null)
-      s.checkWrite(file.getPath());
-
-    if (file.isDirectory())
-      throw new FileNotFoundException(file.getPath() + " is a directory");
-
-   ch = new FileChannelImpl (file.getPath(), (append
-				     ? FileChannelImpl.WRITE
-				     | FileChannelImpl.APPEND
-				     : FileChannelImpl.WRITE));
+    this (file.getPath(), append);
   }
 
   /**
