--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/Inet6Address.java	2003-06-28 02:19:17.000000000 -0600
+++ java/net/Inet6Address.java	2003-11-12 02:18:14.000000000 -0700
@@ -243,7 +243,7 @@
    */
   public boolean equals (Object obj)
   {
-    if (obj == null || ! (obj instanceof Inet6Address))
+    if (! (obj instanceof Inet6Address))
       return false;
 
     Inet6Address tmp = (Inet6Address) obj;
