--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocket.java	2004-08-07 02:29:24.000000000 -0600
+++ java/net/DatagramSocket.java	2004-07-18 02:18:49.000000000 -0600
@@ -554,7 +554,7 @@
    * @exception IllegalBlockingModeException If this socket has an associated
    * channel, and the channel is in non-blocking mode.
    * @exception SecurityException If a security manager exists and its
-   * checkAccept method doesn't allow the receive.
+   * checkAccept ethod doesn't allow the receive.
    */
   public synchronized void receive(DatagramPacket p) throws IOException
   {
