--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/spi/ObjectFactoryBuilder.java	2003-10-13 02:22:24.000000000 -0600
+++ javax/naming/spi/ObjectFactoryBuilder.java	2003-12-06 02:18:23.000000000 -0700
@@ -37,8 +37,9 @@
 
 
 package javax.naming.spi;
-import javax.naming.*;
+
 import java.util.Hashtable;
+import javax.naming.NamingException;
  
 /**
  * @author Warren Levy <warrenl@redhat.com>
