--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/Collator.java	2004-04-28 02:20:17.000000000 -0600
+++ java/text/Collator.java	2004-06-15 02:19:32.000000000 -0600
@@ -303,7 +303,7 @@
     try
       {
 	res = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation",
-				       loc);
+				       loc, Collator.class.getClassLoader());
 	pattern = res.getString("collation_rules");
       }
     catch (MissingResourceException x)
@@ -380,12 +380,7 @@
   }
 
   // Decompose a single character and append results to the buffer.
-  // FIXME: for libgcj this is a native method which handles
-  // decomposition.  For Classpath, for now, it does nothing.
-  final void decomposeCharacter (char c, StringBuffer buf)
-  {
-    buf.append (c);
-  }
+  native final void decomposeCharacter (char c, StringBuffer buf);
 
   /**
    * This is the current collation decomposition setting.
