--- /home/tromey/gnu/classpath/classpath/java/lang/ArithmeticException.java	Sat Mar  2 12:13:10 2002
+++ java/lang/ArithmeticException.java	Tue Jan 22 16:01:33 2002
@@ -1,6 +1,6 @@
 /* ArithmeticException.java -- exception thrown to indicate conditions
    like divide by zero.
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -8,7 +8,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -39,39 +39,41 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
+ */
+
 /**
- * Thrown when a math error has occured, such as trying to divide an
- * integer by zero. For example:<br>
- * <pre>
- * int i = 0;
- * int j = 2 / i;
- * </pre>
+ * Exceptions may be thrown by one part of a Java program and caught
+ * by another in order to deal with exceptional circumstances.
+ * In this case an ArithmeticException is thrown when things like trying
+ * to divide a number by zero.
+ *
+ * @since JDK 1.0
  *
  * @author Brian Jones
  * @author Warren Levy <warrenl@cygnus.com>
- * @status updated to 1.4
+ * @date September 18, 1998.
  */
 public class ArithmeticException extends RuntimeException
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = 2256477558314496007L;
+  static final long serialVersionUID = 2256477558314496007L;
 
   /**
    * Create an exception without a message.
    */
   public ArithmeticException()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an exception with a message.
-   *
-   * @param s the message
    */
   public ArithmeticException(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
