--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/interfaces/DSAKeyPairGenerator.java	2004-04-28 02:20:17.000000000 -0600
+++ java/security/interfaces/DSAKeyPairGenerator.java	2003-10-13 02:17:51.000000000 -0600
@@ -1,5 +1,5 @@
 /* DSAKeyPairGenerator.java -- Initialize a DSA key generator
-   Copyright (C) 1998, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,8 +37,8 @@
 
 package java.security.interfaces;
 
-import java.security.InvalidParameterException;
 import java.security.SecureRandom;
+import java.security.InvalidParameterException;
 
 /**
  * This interface contains methods for intializing a Digital Signature
