--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/SimpleDateFormat.java	2004-05-21 02:26:38.000000000 -0600
+++ java/text/SimpleDateFormat.java	2004-05-05 02:18:55.000000000 -0600
@@ -189,7 +189,6 @@
     numberFormat = NumberFormat.getInstance(locale);
     numberFormat.setGroupingUsed (false);
     numberFormat.setParseIntegerOnly (true);
-    numberFormat.setMaximumFractionDigits (0);
   }
   
   /**
@@ -217,7 +216,6 @@
     numberFormat = NumberFormat.getInstance(locale);
     numberFormat.setGroupingUsed (false);
     numberFormat.setParseIntegerOnly (true);
-    numberFormat.setMaximumFractionDigits (0);
   }
 
   /**
@@ -236,7 +234,6 @@
     numberFormat = NumberFormat.getInstance();
     numberFormat.setGroupingUsed (false);
     numberFormat.setParseIntegerOnly (true);
-    numberFormat.setMaximumFractionDigits (0);
   }
 
   // What is the difference between localized and unlocalized?  The
