--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/AbstractListModel.java	2002-10-02 06:58:40.000000000 -0600
+++ javax/swing/AbstractListModel.java	2003-03-31 02:17:51.000000000 -0700
@@ -191,6 +191,15 @@
 	} // getListeners()
 
 	/**
+	 * getListDataListeners
+	 */
+	public ListDataListener[] getListDataListeners()
+	{
+          // FIXME: implement this
+	  return null;
+	}
+
+	/**
 	 * getElementAt
 	 * @param index TODO
 	 * @returns Object
