--- /home/tromey/gnu/classpath/classpath/java/awt/image/PixelGrabber.java	Mon May  6 10:40:02 2002
+++ java/awt/image/PixelGrabber.java	Thu Jan 24 22:15:23 2002
@@ -151,7 +151,7 @@
 
        @return true if successful
 
-       @throws InterruptedException if interrupted by another thread.
+       @throws InterruptedExcpetion if interrupted by another thread.
      */
     public boolean grabPixels() throws InterruptedException
     {
@@ -163,7 +163,7 @@
 
        @return true if successful
 
-       @throws InterruptedException if interrupted by another thread.
+       @throws InterruptedExcpetion if interrupted by another thread.
                or time runs out
      */
     public synchronized boolean grabPixels(long ms) throws InterruptedException
