--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/event/UndoableEditListener.java	2003-10-13 02:22:49.000000000 -0600
+++ javax/swing/event/UndoableEditListener.java	2004-06-18 02:18:57.000000000 -0600
@@ -1,5 +1,5 @@
 /* UndoableEditListener.java --
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,21 +37,20 @@
 
 package javax.swing.event;
 
-// Imports
 import java.util.EventListener;
 
+
 /**
  * UndoableEditListener public interface
  * @author Andrew Selkirk
  * @author Ronald Veldema
  */
-public interface UndoableEditListener extends EventListener {
-
-	/**
-	 * Undoable edit has happened
-	 * @param event Undoable Edit Event
-	 */
-	void undoableEditHappened(UndoableEditEvent event);
-
-
-} // UndoableEditListener
+public interface UndoableEditListener extends EventListener
+{
+  /**
+   * Undoable edit has happened
+   *
+   * @param event Undoable Edit Event
+   */
+  void undoableEditHappened(UndoableEditEvent event);
+}
