--- /home/tromey/gnu/egcs/classpath/classpath/java/net/SocketException.java	Sat Mar  2 12:13:13 2002
+++ java/net/SocketException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
 /* SocketException.java -- An exception occurred while performing a socket op
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,38 +37,49 @@
 
 package java.net;
 
-import java.io.IOException;
+/* Written using on-line Java Platform 1.2 API Specification.
+ * Believed complete and correct.
+ */
 
 /**
- * This exception indicates that a generic error occurred related to an
- * operation on a socket.  Check the descriptive message (if any) for
- * details on the nature of this error
- *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @author Per Bothner
- * @status updated to 1.4
+  * This exception indicates that a generic error occurred related to an
+  * operation on a socket.  Check the descriptive message (if any) for
+  * details on the nature of this error
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @author Per Bothner 
+  * @date January 6, 1999.
+  */
+public class SocketException extends java.io.IOException
+{
+
+/*
+ * Constructors
  */
-public class SocketException extends IOException
+
+/**
+  * Initializes a new instance of <code>SocketException</code> without
+  * a descriptive error message.
+  */
+public
+SocketException()
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = -5935874303556886934L;
-
-  /**
-   * Create a new instance without a descriptive error message.
-   */
-  public SocketException()
-  {
-  }
-
-  /**
-   * Create a new instance with a descriptive error message.
-   *
-   * @param message a message describing the error that occurred
-   */
-  public SocketException(String message)
-  {
-    super(message);
-  }
+  super();
+}
+
+/*************************************************************************/
+
+/**
+  * Initializes a new instance of <code>SocketException</code> without
+  * a descriptive error message.
+  *
+  * @param message A message describing the error that occurred.
+  */
+public
+SocketException(String message)
+{
+  super(message);
+}
+
 } // class SocketException
+
