--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Float.java	Sun Oct 21 10:39:27 2001
+++ java/lang/Float.java	Fri Oct 26 16:18:42 2001
@@ -79,7 +79,7 @@
    * The primitive type <code>float</code> is represented by this 
    * <code>Class</code> object.
    */
-  public static final Class TYPE = VMClassLoader.getPrimitiveClass ("float");
+  public static final Class TYPE = VMClassLoader.getPrimitiveClass('F');
 
   /**
    * The immutable value of this Float.
