--- /home/tromey/gnu/egcs/classpath/classpath/java/util/List.java	Tue Apr 30 17:58:37 2002
+++ java/util/List.java	Tue Jan 22 16:01:41 2002
@@ -201,7 +201,7 @@
    * @see Object#equals(Object)
    * @see #hashCode()
    */
-  boolean equals(Object o);
+ /* boolean equals(Object o);*/
 
   /**
    * Get the element at a given index in this list.
@@ -213,20 +213,18 @@
   Object get(int index);
 
   /**
-   * Obtains a hash code for this list. In order to obey the general
-   * contract of the hashCode method of class Object, this value is
-   * calculated as follows:
-   * 
-<p><pre>hashCode = 1;
-Iterator i = list.iterator();
-while (i.hasNext())
-{
-  Object obj = i.next();
-  hashCode = 31 * hashCode + (obj == null ? 0 : obj.hashCode());
-}</pre>
-   *
-   * <p>This ensures that the general contract of Object.hashCode()
-   * is adhered to.
+   * Obtain a hash code for this list. In order to obey the general contract of
+   * the hashCode method of class Object, this value is calculated as follows:
+   * <pre>
+   *   hashCode = 1;
+   *   Iterator i = list.iterator();
+   *   while (i.hasNext())
+   *     {
+   *       Object obj = i.next();
+   *       hashCode = 31 * hashCode + (obj == null ? 0 : obj.hashCode());
+   *     }
+   * </pre>
+   * This ensures that the general contract of Object.hashCode() is adhered to.
    *
    * @return the hash code of this list
    * @see Object#hashCode()
