--- /home/tromey/gnu/egcs/classpath/classpath/java/util/Arrays.java	Fri Oct 26 16:06:57 2001
+++ java/util/Arrays.java	Sat Dec 15 09:16:00 2001
@@ -46,6 +46,7 @@
  * sets that cause other quicksorts to degrade to quadratic performance.
  *
  * @author Original author unknown
+ * @author Bryce McKinlay
  * @author Eric Blake <ebb9@email.byu.edu>
  * @see Comparable
  * @see Comparator
@@ -2334,8 +2335,6 @@
       }
   }
 
-
-// asList
   /**
    * Returns a list "view" of the specified array. This method is intended to
    * make it easy to use the Collections API with existing array-based APIs and
