--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JRadioButtonMenuItem.java	2004-04-18 02:22:30.000000000 -0600
+++ javax/swing/JRadioButtonMenuItem.java	2004-05-27 02:19:05.000000000 -0600
@@ -39,118 +39,188 @@
 
 import java.io.IOException;
 import java.io.ObjectOutputStream;
-
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
 
 
+/**
+ * DOCUMENT ME!
+ *
+ * @author $author$
+ * @version $Revision: 1.5 $
+ */
 public class JRadioButtonMenuItem extends JMenuItem implements Accessible
 {
-  //-------------------------------------------------------------
-  // Variables --------------------------------------------------
-  //-------------------------------------------------------------
   private static final String uiClassID = "RadioButtonMenuItemUI";
 
-  //-------------------------------------------------------------
-  // Initialization ---------------------------------------------
-  //-------------------------------------------------------------
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   */
   public JRadioButtonMenuItem()
   {
     this(null, null);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param icon DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(Icon icon)
   {
     this(null, icon);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(String text)
   {
     this(text, null);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param action DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(Action action)
   {
     this();
     setAction(action);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   * @param icon DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(String text, Icon icon)
   {
     this(text, icon, false);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   * @param selected DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(String text, boolean selected)
   {
     this(text, null, selected);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param icon DOCUMENT ME!
+   * @param selected DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(Icon icon, boolean selected)
   {
     this(null, icon, selected);
-  } // JRadioButtonMenuItem()
+  }
 
+  /**
+   * Creates a new JRadioButtonMenuItem object.
+   *
+   * @param text DOCUMENT ME!
+   * @param icon DOCUMENT ME!
+   * @param selected DOCUMENT ME!
+   */
   public JRadioButtonMenuItem(String text, Icon icon, boolean selected)
   {
-    super(text, icon);  
-    setModel(new JToggleButton.ToggleButtonModel());	
+    super(text, icon);
+    setModel(new JToggleButton.ToggleButtonModel());
     model.setSelected(selected);
-  } // JRadioButtonMenuItem()
+  }
 
-  //-------------------------------------------------------------
-  // Methods ----------------------------------------------------
-  //-------------------------------------------------------------
+  /**
+   * DOCUMENT ME!
+   *
+   * @param stream DOCUMENT ME!
+   *
+   * @throws IOException DOCUMENT ME!
+   */
   private void writeObject(ObjectOutputStream stream) throws IOException
   {
     // TODO
-  } // writeObject()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   public String getUIClassID()
   {
     return uiClassID;
-  } // getUIClassID()
+  }
 
+  /**
+   * DOCUMENT ME!
+   */
   public void requestFocus()
   {
     // TODO
-  } // requestFocus()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   protected String paramString()
   {
     return "JRadioButtonMenuItem";
-  } // paramString()
+  }
 
+  /**
+   * DOCUMENT ME!
+   *
+   * @return $returnType$ DOCUMENT ME!
+   */
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      {
-        accessibleContext = new AccessibleJRadioButtonMenuItem(this);
-      }
+      accessibleContext = new AccessibleJRadioButtonMenuItem(this);
+
     return accessibleContext;
-  } // getAccessibleContext()
+  }
 
-  //-------------------------------------------------------------
-  // Classes ----------------------------------------------------
-  //-------------------------------------------------------------
+  /**
+   * DOCUMENT ME!
+   *
+   * @author $author$
+   * @version $Revision: 1.5 $
+   */
   protected class AccessibleJRadioButtonMenuItem extends AccessibleJMenuItem
   {
-    //-------------------------------------------------------------
-    // Initialization ---------------------------------------------
-    //-------------------------------------------------------------
+    /**
+     * Creates a new AccessibleJRadioButtonMenuItem object.
+     *
+     * @param component DOCUMENT ME!
+     */
     protected AccessibleJRadioButtonMenuItem(JRadioButtonMenuItem component)
     {
       super(component);
 
       // TODO
-    } // AccessibleJRadioButtonMenuItem()
+    }
 
-    //-------------------------------------------------------------
-    // Methods ----------------------------------------------------
-    //-------------------------------------------------------------
+    /**
+     * DOCUMENT ME!
+     *
+     * @return $returnType$ DOCUMENT ME!
+     */
     public AccessibleRole getAccessibleRole()
     {
       return AccessibleRole.RADIO_BUTTON;
-    } // getAccessibleRole()
-  } // AccessibleJRadioButtonMenuItem
-} // JRadioButtonMenuItem
+    }
+  }
+}
