--- /home/tromey/gnu/egcs/classpath/classpath/java/net/UnknownServiceException.java	Sun Apr 15 11:57:25 2001
+++ java/net/UnknownServiceException.java	Fri Nov  2 21:44:51 2001
@@ -1,4 +1,4 @@
-/* UnknownServiceException.java -- A service error occured
+/* UnknownServiceException.java -- A service error occurred
    Copyright (C) 1998, 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
