--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/AttributedStringIterator.java	2004-02-29 02:21:57.000000000 -0700
+++ java/text/AttributedStringIterator.java	2002-01-22 16:01:37.000000000 -0700
@@ -179,12 +179,8 @@
   if (attribs == null)
     return(s);
 
-    for (int i = 0; i < attribs.length; i++)
+  for (int i = 0; i < attribs.length; i++)
     {
-      if (attribs[i].begin_index > getEndIndex()
-	  || attribs[i].end_index <= getBeginIndex())
-	continue;
-
       Set key_set = attribs[i].attribs.keySet();
       Iterator iter = key_set.iterator();
       while (iter.hasNext())
@@ -331,7 +327,7 @@
           // Check for attribute match and range match
           if (obj.equals(attrib))
             if ((ci.getIndex() >= attribs[i].begin_index) &&
-                (ci.getIndex() < attribs[i].end_index))
+                (ci.getIndex() <= attribs[i].end_index))
               return(attribs[i].attribs.get(obj));
         }
     }
@@ -355,7 +351,7 @@
   for (int i = 0; i < attribs.length; i++)
     {
        if ((ci.getIndex() >= attribs[i].begin_index) &&
-           (ci.getIndex() < attribs[i].end_index))
+           (ci.getIndex() <= attribs[i].end_index))
          m.putAll(attribs[i].attribs);
     }
 
