--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/TextArea.java	2004-02-04 02:21:19.000000000 -0700
+++ java/awt/TextArea.java	2004-05-27 02:18:59.000000000 -0600
@@ -39,6 +39,9 @@
 
 import java.awt.peer.ComponentPeer;
 import java.awt.peer.TextAreaPeer;
+import java.awt.event.KeyEvent;
+import java.util.HashSet;
+import java.util.Set;
 
 
 /**
@@ -193,11 +196,19 @@
     this.rows = rows;
     this.columns = columns;
     this.scrollbarVisibility = scrollbarVisibility;
-  }
 
-  /*
-   * Instance Variables
-   */
+    // TextAreas need to receive tab key events so we override the
+    // default forward and backward traversal key sets.
+    Set s = new HashSet ();
+    s.add (AWTKeyStroke.getAWTKeyStroke (KeyEvent.VK_TAB,
+                                         KeyEvent.CTRL_DOWN_MASK));
+    setFocusTraversalKeys (KeyboardFocusManager.FORWARD_TRAVERSAL_KEYS, s);
+    s = new HashSet ();
+    s.add (AWTKeyStroke.getAWTKeyStroke (KeyEvent.VK_TAB,
+                                         KeyEvent.SHIFT_DOWN_MASK
+                                         | KeyEvent.CTRL_DOWN_MASK));
+    setFocusTraversalKeys (KeyboardFocusManager.BACKWARD_TRAVERSAL_KEYS, s);
+  }
 
   /**
    * Retrieve the number of columns that this text area would prefer
@@ -431,10 +442,8 @@
    */
   public void addNotify ()
   {
-    if (getPeer () != null)
-      return;
-
-    setPeer ((ComponentPeer) getToolkit().createTextArea (this));
+    if (getPeer () == null)
+      setPeer ((ComponentPeer) getToolkit().createTextArea (this));
   }
 
   /**
@@ -458,10 +467,9 @@
   public void appendText (String str)
   {
     TextAreaPeer peer = (TextAreaPeer) getPeer ();
-    if (peer == null)
-      return;
 
-    peer.insert (str, peer.getText().length ());
+    if (peer != null)
+      peer.insert (str, peer.getText().length ());
   }
 
   /**
@@ -489,10 +497,9 @@
   public void insertText (String str, int pos)
   {
     TextAreaPeer peer = (TextAreaPeer) getPeer ();
-    if (peer == null)
-      return;
 
-    peer.insert (str, pos);
+    if (peer != null)
+      peer.insert (str, pos);
   }
 
   /**
@@ -530,10 +537,9 @@
   public void replaceText (String str, int start, int end)
   {
     TextAreaPeer peer = (TextAreaPeer) getPeer ();
-    if (peer == null)
-      return;
 
-    peer.replaceRange (str, start, end);
+    if (peer != null)
+      peer.replaceRange (str, start, end);
   }
 
   /**
