--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicProgressBarUI.java	2004-04-18 02:22:34.000000000 -0600
+++ javax/swing/plaf/basic/BasicProgressBarUI.java	2004-05-27 02:19:08.000000000 -0600
@@ -712,6 +712,7 @@
     progressBar.setForeground(defaults.getColor("ProgressBar.foreground"));
     progressBar.setBackground(defaults.getColor("ProgressBar.background"));
     progressBar.setBorder(defaults.getBorder("ProgressBar.border"));
+    progressBar.setOpaque(true);
 
     selectionForeground = defaults.getColor("ProgressBar.selectionForeground");
     selectionBackground = defaults.getColor("ProgressBar.selectionBackground");
