--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/spi/NamingManager.java	2004-10-12 02:18:18.000000000 -0600
+++ javax/naming/spi/NamingManager.java	2003-12-06 02:18:23.000000000 -0700
@@ -55,10 +55,10 @@
 {
   public static final String CPE = "java.naming.spi.CannotProceedException";
 
-  private static InitialContextFactoryBuilder icfb;
+  private static InitialContextFactoryBuilder icfb = null;
 
   // Package private so DirectoryManager can access it.
-  static ObjectFactoryBuilder ofb;
+  static ObjectFactoryBuilder ofb = null;
 
   // This class cannot be instantiated.
   NamingManager ()
