--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/StackTraceElement.java	2004-11-17 02:24:08.000000000 -0700
+++ java/lang/StackTraceElement.java	2004-10-28 02:16:10.000000000 -0600
@@ -51,7 +51,7 @@
  * @since 1.4
  * @status updated to 1.4
  */
-public final class StackTraceElement implements Serializable
+public class StackTraceElement implements Serializable
 {
   /**
    * Compatible with JDK 1.4+.
