--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-03-18 02:18:22.000000000 -0700
@@ -40,16 +40,17 @@
 
 import java.awt.Component;
 import java.awt.ComponentOrientation;
-import java.awt.Insets;
 import java.awt.LayoutManager;
+import java.awt.Insets;
 import java.awt.Point;
 import java.awt.Rectangle;
 
-import javax.accessibility.Accessible;
 import javax.swing.border.Border;
+import javax.accessibility.Accessible;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
 import javax.swing.plaf.ScrollPaneUI;
+import javax.swing.plaf.UIResource;
 
 /**
  * <table>
