--- /home/tromey/gnu/egcs/classpath/classpath/java/awt/Transparency.java	Wed Mar 20 17:01:39 2002
+++ java/awt/Transparency.java	Tue Jan 22 16:01:31 2002
@@ -1,5 +1,4 @@
-/* Transparency.java -- common transparency modes in graphics
-   Copyright (C) 2000, 2002 Free Software Foundation
+/* Copyright (C) 2000, 2002  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -38,30 +37,21 @@
 package java.awt;
 
 /**
- * A common transparency mode for layering graphics.
- *
  * @author Warren Levy <warrenl@cygnus.com>
- * @since 1.1
- * @status updated to 1.4
+ * @date March 15, 2000.
  */
+
+/**
+ * Written using on-line Java Platform 1.2 API Specification, as well
+ * as "The Java Class Libraries", 2nd edition (Addison-Wesley, 1998).
+ * Status:  Believed complete and correct.
+ */
+
 public interface Transparency
 {
-  /** Image data which is completely opaque, for an alpha value of 1.0. */
-  int OPAQUE = 1;
+  public static final int OPAQUE = 1;
+  public static final int BITMASK = 2;
+  public static final int TRANSLUCENT = 3;
 
-  /**
-   * Image data which is either completely opaque or transparent, for an
-   * exact integer alpha value.
-   */
-  int BITMASK = 2;
-
-  /** Image data which is translucent, for a non-integer alpha value. */
-  int TRANSLUCENT = 3;
-
-  /**
-   * Return the transparency type.
-   *
-   * @return One of {@see #OPAQUE}, {@see #BITMASK}, or {@see #TRANSLUCENT}.
-   */
-  int getTransparency();
-} // interface Transparency
+  public int getTransparency();
+}
