--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/AuthenticationException.java	2004-04-20 02:21:36.000000000 -0600
+++ javax/naming/AuthenticationException.java	2002-12-10 19:20:36.000000000 -0700
@@ -1,5 +1,5 @@
 /* AuthenticationException.java --
-   Copyright (C) 2000, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,6 +38,7 @@
  
 package javax.naming;
 
+import java.lang.Exception;
  
 public class AuthenticationException extends NamingSecurityException
 {
