--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/AttributedCharacterIterator.java	Tue Jan 22 15:27:01 2002
+++ java/text/AttributedCharacterIterator.java	Sun Mar 23 02:18:02 2003
@@ -70,6 +70,8 @@
 public static class Attribute implements Serializable
 {
 
+  private static final long serialVersionUID = -9142742483513960612L;
+
 /*************************************************************************/
 
 /*
