--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/ProtectionDomain.java	2004-06-08 02:25:15.000000000 -0600
+++ java/security/ProtectionDomain.java	2003-04-20 02:20:19.000000000 -0600
@@ -1,5 +1,5 @@
 /* ProtectionDomain.java -- A security domain
-   Copyright (C) 1998, 2003, 2004  Free Software Foundation, Inc.
+   Copyright (C) 1998, 2003, Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -83,7 +83,7 @@
    */
   public ProtectionDomain(CodeSource codesource, PermissionCollection permissions)
   {
-    this(codesource, permissions, null, null, true);
+    this(codesource, permissions, null, null, false);
   }
 
   /**
