--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ByteArrayOutputStream.java	Mon Mar 10 02:19:32 2003
+++ java/io/ByteArrayOutputStream.java	Tue Feb 25 02:17:41 2003
@@ -65,6 +65,8 @@
   * application specified character encoding.  Thus it can handle 
   * multibyte character encodings.
   *
+  * @version 0.0
+  *
   * @author Aaron M. Renn (arenn@urbanophile.com)
   * @author Tom Tromey <tromey@cygnus.com>
   * @date September 24, 1998 
