--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/DocFlavor.java	2004-11-17 02:24:09.000000000 -0700
+++ javax/print/DocFlavor.java	2004-10-28 02:16:13.000000000 -0600
@@ -217,9 +217,7 @@
   
   private static final long serialVersionUID = -4512080796965449721L;
   
-  // FIXME: Get the host encoding from somewhere. Note that the new String is to make
-  // sure the field won't be a compile time constant.
-  public static final String hostEncoding = new String("US-ASCII");
+  public static final String hostEncoding = "US-ASCII";
 
   private String mediaSubtype;
   private String mediaType;
