--- /home/tromey/gnu/Nightly/classpath/classpath/javax/imageio/ImageIO.java	2004-11-17 02:24:09.000000000 -0700
+++ javax/imageio/ImageIO.java	2004-11-06 11:08:58.000000000 -0700
@@ -51,11 +51,6 @@
 
 public final class ImageIO
 {
-  /**
-   * This class isn't intended to be instantiated.
-   */
-  private ImageIO() {}
-
   private static final class ReaderFormatFilter implements ServiceRegistry.Filter
   {
     private String formatName;
