--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JButton.java	2004-01-10 02:20:47.000000000 -0700
+++ javax/swing/JButton.java	2004-02-12 02:18:27.000000000 -0700
@@ -98,7 +98,7 @@
     public String getUIClassID()
     {
 	//Returns a string that specifies the name of the L&F class that renders this component.  
-	return "JButton";
+	return "ButtonUI";
     }
   
     public boolean isDefaultButton()
