--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Identity.java	2004-04-18 02:22:22.000000000 -0600
+++ java/security/Identity.java	2003-05-10 02:19:17.000000000 -0600
@@ -151,7 +151,7 @@
    * @throws KeyManagementException if another identity in the identity's scope
    * has the same public key, or if another exception occurs.
    * @throws SecurityException if a security manager exists and its
-   * <code>checkSecurityAccess()</code> method doesn't allow setting the public
+   * <code>checkSecurityAccess()<code> method doesn't allow setting the public
    * key.
    * @see #getPublicKey()
    * @see SecurityManager#checkSecurityAccess(String)
