--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DirectByteBufferImpl.java	2003-10-21 02:19:05.000000000 -0600
+++ java/nio/DirectByteBufferImpl.java	2003-09-25 02:18:11.000000000 -0600
@@ -38,20 +38,10 @@
 
 package java.nio;
 
-import gnu.classpath.Configuration;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 class DirectByteBufferImpl extends ByteBuffer
 {
-  static
-  {
-    // load the shared library needed for native methods.
-    if (Configuration.INIT_LOAD_LIBRARY)
-      {
-        System.loadLibrary ("javanio");
-      }
-  }
-  
   RawData address;
   private int offset;
   private boolean readOnly;
