--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/ProtectionDomain.java	2004-06-08 02:25:15.000000000 -0600
+++ java/security/ProtectionDomain.java	2004-09-13 02:17:04.000000000 -0600
@@ -83,7 +83,7 @@
    */
   public ProtectionDomain(CodeSource codesource, PermissionCollection permissions)
   {
-    this(codesource, permissions, null, null, true);
+    this(codesource, permissions, null, null, false);
   }
 
   /**
