--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Menu.java	2003-06-16 02:23:42.000000000 -0600
+++ java/awt/Menu.java	2004-01-20 02:18:20.000000000 -0700
@@ -1,5 +1,5 @@
 /* Menu.java -- A Java AWT Menu
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -294,7 +294,8 @@
 public void
 addSeparator()
 {
-  add(separator);
+  if (peer != null)
+    ((MenuPeer) peer).addSeparator();
 }
 
 /*************************************************************************/
@@ -376,8 +377,14 @@
 public void
 addNotify()
 {
-  if (peer != null)
+  if (peer == null)
     peer = getToolkit().createMenu(this);
+  java.util.Enumeration e = items.elements();
+  while (e.hasMoreElements())
+  {
+    MenuItem mi = (MenuItem)e.nextElement();
+    mi.addNotify();
+  }    
   super.addNotify ();
 }
 
