--- /home/tromey/gnu/Nightly/classpath/classpath/javax/rmi/CORBA/UtilDelegate.java	2004-04-20 02:21:38.000000000 -0600
+++ javax/rmi/CORBA/UtilDelegate.java	2003-03-29 14:23:25.000000000 -0700
@@ -38,10 +38,9 @@
 
 package javax.rmi.CORBA;
 
-import java.io.InputStream;
-import java.io.OutputStream;
 import java.rmi.Remote;
 import java.rmi.RemoteException;
+import java.io.*;
 //import org.omg.CORBA.ORB;
 //import org.omg.CORBA.SystemException;
 //import org.omg.CORBA.portable.InputStream;
