--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2003-06-16 02:24:07.000000000 -0600
+++ javax/swing/UIManager.java	2003-06-13 02:27:43.000000000 -0600
@@ -174,21 +174,11 @@
 	System.out.println("UIManager.getDim");
 	return new Dimension(200,100);
     }
-
-
-  /**
-   * Retrieves a font from the defaults table of the current
-   * LookAndFeel.
-   *
-   * @param key an Object that specifies the font. Typically,
-   *        this is a String such as
-   *        <code>&quot;TitledBorder.font&quot;</code>.
-   */
-  public static Font getFont(Object key)
-  {
-    return (Font) getLookAndFeel().getDefaults().get(key);
-  }
-
+    static  Font getFont(Object key)
+    //      Returns a drawing font from the defaults table. 
+    {
+	return (Font) getLookAndFeel().getDefaults().get(key);
+    }
     static  Icon getIcon(Object key)
     //      Returns an Icon from the defaults table. 
     {
