--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicDefaults.java	2003-11-21 02:20:37.000000000 -0700
+++ javax/swing/plaf/basic/BasicDefaults.java	2003-11-18 02:18:14.000000000 -0700
@@ -116,9 +116,6 @@
 	put("JToggleButton.border", "javax.swing.plaf.basic.PanelBorder");
 	put("JCheckBox.border", "javax.swing.plaf.basic.PanelBorder");
 	put("JRadioButton.border", "javax.swing.plaf.basic.PanelBorder");
-
-	put("AbstractUndoableEdit.undoText", "Undo");
-	put("AbstractUndoableEdit.redoText", "Redo");
     }
     
 }
