--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/CompoundName.java	2004-04-20 02:21:36.000000000 -0600
+++ javax/naming/CompoundName.java	2003-11-12 02:18:14.000000000 -0700
@@ -1,5 +1,5 @@
 /* CompoundName.java --
-   Copyright (C) 2001, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,8 +40,8 @@
 
 import java.io.Serializable;
 import java.util.Enumeration;
-import java.util.NoSuchElementException;
 import java.util.Properties;
+import java.util.NoSuchElementException;
 import java.util.Vector;
 
 /**
