--- /home/tromey/gnu/egcs/classpath/classpath/java/util/Timer.java	Fri Feb  9 16:31:22 2001
+++ java/util/Timer.java	Fri Nov  2 21:44:52 2001
@@ -39,8 +39,8 @@
  * scheduling guarantees more or less that the task will be executed at a
  * specific time, but if there is ever a delay in execution then the period
  * between successive executions will be shorter. The first method of
- * repeated scheduling is prefered for repeated tasks in response to user
- * interaction, the second method of repeated scheduling is prefered for tasks
+ * repeated scheduling is preferred for repeated tasks in response to user
+ * interaction, the second method of repeated scheduling is preferred for tasks
  * that act like alarms.
  * <p>
  * The Timer keeps a binary heap as a task priority queue which means that
@@ -65,7 +65,7 @@
     /** Default size of this queue */
     private final int DEFAULT_SIZE = 32;
 
-    /** Wheter to return null when there is nothing in the queue */
+    /** Whether to return null when there is nothing in the queue */
     private boolean nullOnEmpty;
 
     /**
@@ -375,7 +375,7 @@
   private boolean canceled;
 
   /**
-   * Creates a new Timer with a non deamon Thread as Scheduler, with normal
+   * Creates a new Timer with a non daemon Thread as Scheduler, with normal
    * priority and a default name.
    */
   public Timer()
@@ -384,7 +384,7 @@
   }
 
   /**
-   * Creates a new Timer with a deamon Thread as scheduler if deamon is true,
+   * Creates a new Timer with a daemon Thread as scheduler if daemon is true,
    * with normal priority and a default name.
    */
   public Timer(boolean daemon)
@@ -393,7 +393,7 @@
   }
 
   /**
-   * Creates a new Timer with a deamon Thread as scheduler if deamon is true,
+   * Creates a new Timer with a daemon Thread as scheduler if daemon is true,
    * with the priority given and a default name.
    */
   private Timer(boolean daemon, int priority)
@@ -402,7 +402,7 @@
   }
 
   /**
-   * Creates a new Timer with a deamon Thread as scheduler if deamon is true,
+   * Creates a new Timer with a daemon Thread as scheduler if daemon is true,
    * with the priority and name given.E
    */
   private Timer(boolean daemon, int priority, String name)
