--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/CharArrayReader.java	2002-01-22 15:26:59.000000000 -0700
+++ java/io/CharArrayReader.java	2003-12-29 02:18:04.000000000 -0700
@@ -228,7 +228,7 @@
 
 	/* Don't need to check pos value, arraycopy will check it. */
 	if (off < 0 || len < 0 || off + len > b.length)
-	  throw new ArrayIndexOutOfBoundsException();
+	  throw new IndexOutOfBoundsException();
 
 	if (pos >= count)
 	  return -1;
