--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/ResolutionSyntax.java	2004-04-19 02:20:15.000000000 -0600
+++ javax/print/attribute/ResolutionSyntax.java	2003-12-28 02:18:10.000000000 -0700
@@ -104,7 +104,7 @@
    *
    * @return the resolution
    *
-   * @exception IllegalArgumentException if units < 1
+   * @exception IllegalArgumenException if units < 1
    */
   public int getCrossFeedResolution(int units)
   {
@@ -130,7 +130,7 @@
    *
    * @return the resolution
    *
-   * @exception IllegalArgumentException if units < 1
+   * @exception IllegalArgumenException if units < 1
    */
   public int getFeedResolution(int units)
   {
