--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PipedReader.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/PipedReader.java	2003-03-23 02:18:02.000000000 -0700
@@ -256,7 +256,7 @@
     * @param offset The index into the buffer at which to start writing.
     * @param len The maximum number of chars to read.
     *
-    * @exception IOException If <code>close()</code> was called on this Piped
+    * @exception IOException If <code>close()/code> was called on this Piped
     *                        Reader.
     */  
   public int read(char[] buf, int offset, int len)
