--- /home/tromey/gnu/egcs/classpath/classpath/java/security/KeyPairGeneratorSpi.java	Wed Mar 22 12:26:19 2000
+++ java/security/KeyPairGeneratorSpi.java	Wed Apr 25 11:13:33 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,7 +24,6 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 import java.security.spec.AlgorithmParameterSpec;
 
@@ -33,24 +32,23 @@
    for security algorithms.
 
    @author Mark Benvenuto
-*/
+ */
 public abstract class KeyPairGeneratorSpi
 {
-
   /**
      Constructs a new KeyPairGeneratorSpi
-  */
+   */
   public KeyPairGeneratorSpi()
-  {}
-
+  {
+  }
 
   /**
      Initialize the KeyPairGeneratorSpi with the specified
      key size and source of randomness
 
      @param keysize size of the key to generate
-     @param random A SecureRandom source of randomness	
-  */
+     @param random A SecureRandom source of randomness  
+   */
   public abstract void initialize(int keysize, SecureRandom random);
 
   /**
@@ -63,11 +61,12 @@
      method just throws UnsupportedOperationException.
 
      @param params A AlgorithmParameterSpec to intialize with
-     @param random A SecureRandom source of randomness	
+     @param random A SecureRandom source of randomness  
 
      @throws InvalidAlgorithmParameterException
-  */
-  public void initialize(AlgorithmParameterSpec params, SecureRandom random) throws InvalidAlgorithmParameterException
+   */
+  public void initialize(AlgorithmParameterSpec params, SecureRandom random)
+    throws InvalidAlgorithmParameterException
   {
     throw new java.lang.UnsupportedOperationException();
   }
@@ -78,7 +77,6 @@
      creates a unique key pair each time.
 
      @return a key pair
-  */
+   */
   public abstract KeyPair generateKeyPair();
-
 }
