--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocket.java	2004-02-06 02:21:24.000000000 -0700
+++ java/net/DatagramSocket.java	2004-02-12 02:18:26.000000000 -0700
@@ -798,7 +798,7 @@
     if (isClosed())
       throw new SocketException("socket is closed");
 
-    getImpl().setOption (SocketOptions.SO_REUSEADDR, new Boolean (on));
+    getImpl().setOption (SocketOptions.SO_REUSEADDR, Boolean.valueOf(on));
   }
 
   /**
@@ -837,7 +837,7 @@
     if (isClosed())
       throw new SocketException("socket is closed");
 
-    getImpl().setOption(SocketOptions.SO_BROADCAST, new Boolean(enable));
+    getImpl().setOption(SocketOptions.SO_BROADCAST, Boolean.valueOf(enable));
   }
 
   /**
