--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RemoteCall.java	Sun Mar 23 02:18:41 2003
+++ java/rmi/server/RemoteCall.java	Mon Mar 31 02:17:49 2003
@@ -46,14 +46,41 @@
 /**
  * @deprecated
  */
-public interface RemoteCall {
+public interface RemoteCall
+{
+  /**
+   * @deprecated
+   */
+  public ObjectOutput getOutputStream () throws IOException;
 
-public ObjectOutput getOutputStream() throws IOException;
-public void releaseOutputStream() throws IOException;
-public ObjectInput getInputStream() throws IOException;
-public void releaseInputStream() throws IOException;
-public ObjectOutput getResultStream(boolean success) throws IOException, StreamCorruptedException;
-public void executeCall() throws Exception;
-public void done() throws IOException;
+  /**
+   * @deprecated
+   */
+  public void releaseOutputStream () throws IOException;
 
+  /**
+   * @deprecated
+   */
+  public ObjectInput getInputStream () throws IOException;
+
+  /**
+   * @deprecated
+   */
+  public void releaseInputStream () throws IOException;
+
+  /**
+   * @deprecated
+   */
+  public ObjectOutput getResultStream (boolean success)
+    throws IOException, StreamCorruptedException;
+
+  /**
+   * @deprecated
+   */
+  public void executeCall () throws Exception;
+
+  /**
+   * @deprecated
+   */
+  public void done () throws IOException;
 }
