--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultSingleSelectionModel.java	Wed Oct  2 06:58:40 2002
+++ javax/swing/DefaultSingleSelectionModel.java	Mon Mar 31 02:17:51 2003
@@ -177,5 +177,14 @@
 		return listenerList.getListeners(listenerClass);
 	} // getListeners()
 
+	/**
+	 * getChangeListeners
+	 */
+	public ChangeListener[] getChangeListeners()
+	{
+	  // FIXME: implement this
+	  return null;
+	}
+
 
 } // DefaultSingleSelectionModel
