--- /home/tromey/gnu/egcs/classpath/classpath/java/util/ResourceBundle.java	Wed Sep 26 08:18:48 2001
+++ java/util/ResourceBundle.java	Fri Sep  7 12:43:38 2001
@@ -28,7 +28,6 @@
 package java.util;
 import java.lang.ref.Reference;
 import java.lang.ref.SoftReference;
-import gnu.classpath.Configuration;
 
 /**
  * A resource bundle contains locale-specific data.  If you need
@@ -74,14 +73,6 @@
  * @author Jochen Hoenicke */
 public abstract class ResourceBundle
 {
-  static 
-  {
-    if (Configuration.INIT_LOAD_LIBRARY)
-      {
-	System.loadLibrary ("javautil");
-      }
-  }
-
   /**
    * The parent bundle.  This is consulted when you call getObject
    * and there is no such resource in the current bundle.  This
