--- /home/tromey/gnu/classpath/classpath/java/net/ServerSocket.java	Mon Nov 18 10:34:28 2002
+++ java/net/ServerSocket.java	Wed Nov 20 08:55:06 2002
@@ -151,6 +151,7 @@
     throws IOException
   {
     this();
+
     if (impl == null)
       throw new IOException("Cannot initialize Socket implementation");
 
@@ -181,19 +182,7 @@
   public void bind (SocketAddress endpoint)
     throws IOException
   {
-    if (impl == null)
-      throw new IOException ("Cannot initialize Socket implementation");
-
-    if (! (endpoint instanceof InetSocketAddress))
-      throw new IllegalArgumentException ("Address type not supported");
-
-    InetSocketAddress tmp = (InetSocketAddress) endpoint;
-    
-    SecurityManager s = System.getSecurityManager ();
-    if (s != null)
-      s.checkListen (tmp.getPort ());
-
-    impl.bind (tmp.getAddress (), tmp.getPort ());
+    bind (endpoint, 50);
   }
  
   /**
@@ -363,6 +352,17 @@
   }
 
   /**
+   * Returns true if the socket is closed, otherwise false
+   * 
+   * @since 1.4
+   */
+  public boolean isClosed()
+  {
+    // FIXME: implement this
+    return false;
+  }
+
+  /**
    * Sets the value of SO_TIMEOUT.  A value of 0 implies that SO_TIMEOUT is
    * disabled (ie, operations never time out).  This is the number of 
    * milliseconds a socket operation can block before an
