--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectInput.java	Mon Apr  7 02:24:27 2003
+++ java/io/ObjectInput.java	Mon Mar 24 02:17:53 2003
@@ -45,8 +45,6 @@
   * <code>InputStream</code>
   *
   * @author Aaron M. Renn (arenn@urbanophile.com)
-  *
-  * @see DataInput
   */
 public interface ObjectInput extends DataInput
 {
@@ -62,8 +60,7 @@
 
   /**
     * This method reading a byte of data from a stream.  It returns that byte
-    * as an <code>int</code>.  This method blocks if no data is available 
-    * to be read.
+    * as an int.  This method blocks if no data is available to be read.
     * 
     * @return The byte of data read
     *
@@ -79,7 +76,7 @@
     *
     * @param buf The byte array to receive the data read
     *
-    * @return The actual number of bytes read or -1 if end of stream
+    * @return The actual number fo bytes read or -1 if end of stream
     *
     * @exception IOException If an error occurs
     */
@@ -95,10 +92,10 @@
     * possible.
     *
     * @param buf The byte array to receive the data read
-    * @param offset The offset into <code>buf</code> to start storing data
+    * @param offset The offset into @code{buf} to start storing data
     * @param len The maximum number of bytes to read
     *
-    * @return The actual number of bytes read or -1 if end of stream
+    * @return The actual number fo bytes read or -1 if end of stream
     *
     * @exception IOException If an error occurs
     */
@@ -106,14 +103,14 @@
 
   /**
     * Reads an object instance and returns it.  If the class for the object
-    * being read cannot be found, then a <code>ClassNotFoundException</code>
-    * will be thrown.
+    * being read cannot be found, then a ClassNotFoundException will
+    * be thrown.
     *
     * @return The object instance that was read
     *
     * @exception ClassNotFoundException If a class for the object cannot be 
     * found
-    * @exception IOException If any other error occurs
+    * @exception IOException If an error occurs
     */
   public abstract Object readObject() 
     throws ClassNotFoundException, IOException;
@@ -129,7 +126,7 @@
     *
     * @exception IOException If an error occurs
     */
-  public abstract long skip(long numBytes) throws IOException;
+  public abstract long skip(long num_bytes) throws IOException;
 
   /**
     * This method closes the input source
