--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/Naming.java	2004-11-17 02:24:08.000000000 -0700
+++ java/rmi/Naming.java	2004-10-28 02:16:12.000000000 -0600
@@ -44,10 +44,6 @@
 import java.rmi.registry.Registry;
 
 public final class Naming {
-  /**
-   * This class isn't intended to be instantiated.
-   */
-  private Naming() {}
 
 /**
  * Looks for the remote object that is associated with the named service.
