--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/SocketAddress.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/SocketAddress.java	2003-01-15 15:49:46.000000000 -0700
@@ -1,4 +1,4 @@
-/* SocketAddress.java --
+/* SocketAddress.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -39,8 +39,7 @@
 
 import java.io.Serializable;
 
-
-/**
+/** 
  * Abstract base class for InetSocketAddress.
  * InetSocketAddress is to my knowledge the only derived
  * class. [Ronald]
