--- /home/tromey/gnu/Nightly/classpath/classpath/java/sql/ResultSet.java	2004-04-19 02:20:15.000000000 -0600
+++ java/sql/ResultSet.java	2003-10-13 02:17:51.000000000 -0600
@@ -48,14 +48,14 @@
 /**
  * This interface provides access to the data set returned by a SQL
  * statement.  An instance of this interface is returned by the various
- * execution methods in the <code>Statement</code>.
- *
- * <p> This class models a cursor, which can be stepped through one row at a
+ * execution methods in the <code>Statement</code.
+ * <p>
+ * This class models a cursor, which can be stepped through one row at a
  * time.  Methods are provided for accessing columns by column name or by
- * index.</p>
- *
- * <p> Note that a result set is invalidated if the statement that returned
- * it is closed.</p>
+ * index.
+ * <p>
+ * Note that a result set is invalidated if the statement that returned
+ * it is closed.
  *
  * @author Aaron M. Renn (arenn@urbanophile.com)
  */
