--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/prefs/AbstractPreferences.java	2003-02-21 02:18:50.000000000 -0700
+++ java/util/prefs/AbstractPreferences.java	2003-06-18 02:22:33.000000000 -0600
@@ -35,17 +35,15 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package java.util.prefs;
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.InputStream;
 import java.io.OutputStream;
-
 import java.util.Iterator;
 import java.util.HashMap;
 import java.util.TreeSet;
-
 import gnu.java.util.prefs.NodeWriter;
 
 /**
