--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ShortViewBufferImpl.java	2004-02-12 02:22:34.000000000 -0700
+++ java/nio/ShortViewBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -45,12 +45,21 @@
   private ByteBuffer bb;
   private boolean readOnly;
   private ByteOrder endian;
+
+  ShortViewBufferImpl (ByteBuffer bb, int capacity)
+  {
+    super (capacity, capacity, 0, -1);
+    this.bb = bb;
+    this.offset = bb.position();
+    this.readOnly = bb.isReadOnly();
+    this.endian = bb.order();
+  }
   
   public ShortViewBufferImpl (ByteBuffer bb, int offset, int capacity,
 			      int limit, int position, int mark,
 			      boolean readOnly, ByteOrder endian)
   {
-    super (limit >> 1, limit >> 1, position >> 1, mark >> 1);
+    super (capacity, limit, position, mark);
     this.bb = bb;
     this.offset = offset;
     this.readOnly = readOnly;
