--- /home/tromey/gnu/classpath/classpath/java/awt/DisplayMode.java	Mon Sep 30 13:37:09 2002
+++ java/awt/DisplayMode.java	Thu Aug  8 22:26:14 2002
@@ -143,8 +143,11 @@
    * @param o the object to compare to
    * @return true if it is equal
    */
-  public boolean equals(DisplayMode m)
+  public boolean equals(Object o)
   {
+    if (! (o instanceof DisplayMode))
+      return false;
+    DisplayMode m = (DisplayMode) o;
     return width == m.width && height == m.height && bitDepth == m.bitDepth
       && refreshRate == m.refreshRate;
   }
