--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/LineNumberReader.java	Mon Mar 24 02:18:31 2003
+++ java/io/LineNumberReader.java	Tue Mar 25 02:17:55 2003
@@ -376,13 +376,6 @@
     */
   public long skip(long count) throws IOException
   {
-    return(super.skip(count)); 
-    // FIXME: This method not merged from gcj. The code below is the original
-    // gcj code that doesn't work.  I'm not sure if the method above (from
-    // Classpath) is broken or not, but it seems to work correctly in all
-    // methods I've tested.  If somebody wants to investigate why the code
-    // below fails, feel free.
-/*
     if (count <= 0)
       return 0;
     long to_do = count;
@@ -412,7 +405,6 @@
       }
     while (to_do > 0);
     return count - to_do;
-*/
   }
 }
 
