--- /home/tromey/gnu/classpath/classpath/java/nio/ByteOrder.java	Fri Jul 26 19:15:37 2002
+++ java/nio/ByteOrder.java	Fri Nov  1 05:03:38 2002
@@ -40,8 +40,8 @@
 
 public final class ByteOrder
 {
-  static final ByteOrder BIG_ENDIAN     = new ByteOrder();
-  static final ByteOrder LITTLE_ENDIAN  = new ByteOrder();
+  public static final ByteOrder BIG_ENDIAN     = new ByteOrder();
+  public static final ByteOrder LITTLE_ENDIAN  = new ByteOrder();
 
   public static ByteOrder nativeOrder()
   {
