--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JWindow.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JWindow.java	2004-05-28 02:23:42.000000000 -0600
@@ -58,7 +58,7 @@
  *
  * @author Ronald Veldema (rveldema@cs.vu.nl)
  */
-public class JWindow extends Window implements Accessible
+public class JWindow extends Window implements Accessible, RootPaneContainer
 {
     public final static int HIDE_ON_CLOSE        = 0;
     public final static int EXIT_ON_CLOSE        = 1;
@@ -79,8 +79,8 @@
      *************/
 
     public JWindow()
-    {
-      this(null);
+    {      
+     	super(null); // FIXME: This throws an exception.
     }
 
     // huuu ?
@@ -124,20 +124,20 @@
   public  void setLayout(LayoutManager manager)
   {    super.setLayout(manager);  }
 
-    void setLayeredPane(JLayeredPane layeredPane) 
+    public void setLayeredPane(JLayeredPane layeredPane) 
     {   getRootPane().setLayeredPane(layeredPane);   }
   
-    JLayeredPane getLayeredPane()
+    public JLayeredPane getLayeredPane()
     {   return getRootPane().getLayeredPane();     }
   
-    JRootPane getRootPane()
+    public JRootPane getRootPane()
     {
 	if (rootPane == null)
 	    setRootPane(createRootPane());
 	return rootPane;          
     }
 
-    void setRootPane(JRootPane root)
+    public void setRootPane(JRootPane root)
     {
 	if (rootPane != null)
 	    remove(rootPane);
@@ -146,19 +146,19 @@
 	add(rootPane, BorderLayout.CENTER);
     }
 
-    JRootPane createRootPane()
+    public JRootPane createRootPane()
     {   return new JRootPane();    }
 
-    Container getContentPane()
+    public Container getContentPane()
     {    return getRootPane().getContentPane();     }
 
-    void setContentPane(Container contentPane)
+    public void setContentPane(Container contentPane)
     {    getRootPane().setContentPane(contentPane);    }
   
-    Component getGlassPane()
+    public Component getGlassPane()
     {    return getRootPane().getGlassPane();   }
   
-    void setGlassPane(Component glassPane)
+    public void setGlassPane(Component glassPane)
     {   getRootPane().setGlassPane(glassPane);   }
 
     
