--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLClassLoader.java	2003-07-17 02:19:18.000000000 -0600
+++ java/net/URLClassLoader.java	2003-06-20 02:21:15.000000000 -0600
@@ -311,9 +311,6 @@
       if (jarfile == null)
 	return null;
 
-      if (name.startsWith("/"))
-        name = name.substring(1);
-
       JarEntry je = jarfile.getJarEntry(name);
       if(je != null)
 	return new JarURLResource(this, name, je);
