--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/event/AncestorEvent.java	2003-07-02 02:19:17.000000000 -0600
+++ javax/swing/event/AncestorEvent.java	2003-06-28 02:18:23.000000000 -0600
@@ -50,9 +50,9 @@
 {
   private static final long serialVersionUID = -8079801679695605002L;
   
-  public static final int ANCESTOR_ADDED = 1;
+  public static final int ANCESTOR_ADDED = 0;
+  public static final int ANCESTOR_MOVED = 1;
   public static final int ANCESTOR_REMOVED = 2;
-  public static final int ANCESTOR_MOVED = 3;
 
   private JComponent sourceComponent;
   private Container ancestor;
