--- /home/tromey/gnu/Nightly/classpath/classpath/javax/security/auth/PrivateCredentialPermission.java	2004-09-13 02:44:29.000000000 -0600
+++ javax/security/auth/PrivateCredentialPermission.java	2004-09-21 02:33:33.000000000 -0600
@@ -66,10 +66,6 @@
 public class PrivateCredentialPermission extends Permission
   implements Serializable
 {
-  /**
-   * For compatability with Sun's JDK 1.4.2 rev. 5
-   */
-  private static final long serialVersionUID = 5284372143517237068L;
 
   // Fields.
   // -------------------------------------------------------------------------
