--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectStreamField.java	2003-12-03 02:20:26.000000000 -0700
+++ java/io/ObjectStreamField.java	2003-12-17 02:18:16.000000000 -0700
@@ -108,7 +108,7 @@
         type = Object.class; //FIXME: ???
       }
   }
-
+  
   /**
    * There are many cases you can not get java.lang.Class from typename 
    * if your context class loader cann not load it, then use typename to
@@ -205,8 +205,6 @@
     offset = off;
   }
 
-  /**
-   */
   public boolean isUnshared ()
   {
     return unshared;
