--- /home/tromey/gnu/classpath/classpath/javax/naming/BinaryRefAddr.java	Tue Jan 22 15:50:12 2002
+++ javax/naming/BinaryRefAddr.java	Wed Sep 25 10:10:50 2002
@@ -97,7 +97,7 @@
    *         the addrType is the same as this addrType and the bytes of the
    *         content are the same.
    */
-  public boolean equal(Object o)
+  public boolean equals(Object o)
   {
     if (o instanceof BinaryRefAddr)
       {
