--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/FieldPosition.java	2003-11-28 02:22:20.000000000 -0700
+++ java/text/FieldPosition.java	2003-11-28 02:18:12.000000000 -0700
@@ -168,7 +168,7 @@
    * <p>
    * <ul>
    * <li>The specified object is not <code>null</code>.
-   * <li>The specified object has the same class as this object.
+   * <li>The specified object is an instance of <code>FieldPosition</code>.
    * <li>The specified object has the same field identifier, field attribute 
    * and beginning and ending index as this object.
    * </ul>
