--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DirectByteBufferImpl.java	2003-05-19 03:05:13.000000000 -0600
+++ java/nio/DirectByteBufferImpl.java	2003-06-18 02:22:30.000000000 -0600
@@ -38,13 +38,18 @@
 
 package java.nio;
 
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 public class DirectByteBufferImpl extends ByteBuffer
 {
-  private RawData address;
+  RawData address;
   private int offset;
   private boolean readOnly;
+
+  public DirectByteBufferImpl (RawData address, long len)
+  {
+    this (address, 0, (int) len, (int) len, 0, -1, false);
+  }
   
   public DirectByteBufferImpl (RawData address, int offset, int capacity,
                                int limit, int position, int mark,
