--- /home/tromey/gnu/egcs/classpath/classpath/java/rmi/server/ServerNotActiveException.java	Sat Mar  2 12:13:15 2002
+++ java/rmi/server/ServerNotActiveException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* ServerNotActiveException.java -- the method is not servicing a remote call
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,36 +37,17 @@
 
 package java.rmi.server;
 
-/**
- * Thrown during <code>RemoteServer.getClientHost</code> if the host is
- * not servicing a remote method call.
- *
- * @author unknown
- * @see RemoteServer#getClientHost()
- * @since 1.1
- * @status updated to 1.4
- */
-public class ServerNotActiveException extends Exception
-{
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = 4687940720827538231L;
-
-  /**
-   * Create an exception with no message.
-   */
-  public ServerNotActiveException()
-  {
-  }
-
-  /**
-   * Create an exception with a message.
-   *
-   * @param s the message
-   */
-  public ServerNotActiveException(String s)
-  {
-    super(s);
-  }
+import java.lang.Exception;
+
+public class ServerNotActiveException
+	extends Exception {
+
+public ServerNotActiveException() {
+	super();
+}
+
+public ServerNotActiveException(String s) {
+	super(s);
+}
+
 }
