--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/ComponentUI.java	2003-07-29 02:20:01.000000000 -0600
+++ javax/swing/plaf/ComponentUI.java	2003-06-28 02:18:23.000000000 -0600
@@ -50,7 +50,7 @@
  * should not need to access this class; it is internal to Swing
  * and the look-and-feel implementations.
  *
- * <p><img src="doc-files/ComponentUI-1.png" width="700" height="550"
+ * <p><img src="ComponentUI-1.png" width="700" height="550"
  * alt="[UML diagram illustrating the architecture for pluggable
  * look and feels]" />
  *
