--- /home/tromey/gnu/egcs/classpath/classpath/java/security/NoSuchAlgorithmException.java	Sat Mar  2 12:13:18 2002
+++ java/security/NoSuchAlgorithmException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* NoSuchAlgorithmException.java -- an algorithm was not available
-   Copyright (C) 1998, 2002 Free Software Foundation, Inc.
+/* NoSuchAlgorithmException.java -- What it says
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -41,27 +41,26 @@
  * This exception is thrown when the requested security algorithm is
  * not available
  *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @status updated to 1.4
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
  */
 public class NoSuchAlgorithmException extends GeneralSecurityException
 {
   /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -7443947487218346562L;
-
-  /**
-   * Create a new instance with no descriptive error message.
+   * This method initializes a new instance of <code>NoSuchAlgorithmException</code>
+   * with no descriptive error message.
    */
   public NoSuchAlgorithmException()
   {
+    super();
   }
 
   /**
-   * Create a new instance with a descriptive error message.
+   * This method initializes a new instance of <code>NoSuchAlgorithmException</code>
+   * with a descriptive error message.
    *
-   * @param msg the descriptive error message
+   * @param msg The descriptive error message
    */
   public NoSuchAlgorithmException(String msg)
   {
