--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2004-07-29 02:25:40.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-07-23 02:18:47.000000000 -0600
@@ -41,6 +41,7 @@
 import java.awt.Component;
 import java.awt.ComponentOrientation;
 import java.awt.Dimension;
+import java.awt.LayoutManager;
 import java.awt.Insets;
 import java.awt.LayoutManager;
 import java.awt.Point;
