--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectInputStream.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/ObjectInputStream.java	2004-04-21 02:18:08.000000000 -0600
@@ -1802,7 +1802,11 @@
    * @param sm SecurityManager instance which should be called.
    * @return The current class loader in the calling stack.
    */
-  private static native ClassLoader currentClassLoader (SecurityManager sm);
+  private static ClassLoader currentClassLoader (SecurityManager sm)
+  {
+    // FIXME: This is too simple.
+    return ClassLoader.getSystemClassLoader ();
+  }
 
   private void callReadMethod (Method readObject, Class klass, Object obj) throws IOException
   {
