--- /home/tromey/gnu/egcs/classpath/classpath/java/security/interfaces/DSAKey.java	Wed Sep 26 08:18:45 2001
+++ java/security/interfaces/DSAKey.java	Fri Nov  2 21:44:51 2001
@@ -1,4 +1,4 @@
-/* DSAKey.java -- Interface for Digital Signature Algorith key
+/* DSAKey.java -- Interface for Digital Signature Algorithm key
    Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
