--- /home/tromey/gnu/classpath/classpath/java/awt/GridLayout.java	Wed Oct  2 11:21:37 2002
+++ java/awt/GridLayout.java	Sun Mar 10 11:15:58 2002
@@ -58,8 +58,6 @@
  */
 public class GridLayout implements LayoutManager, Serializable
 {
-  static final long serialVersionUID = -7411804673224730901L;
-  
   /** Add a new component to the layout.  This particular implementation
    * does nothing.
    * @param name The name of the component to add.
