--- /home/tromey/gnu/classpath/classpath/java/nio/channels/Channel.java	Tue Apr 30 17:58:37 2002
+++ java/nio/channels/Channel.java	Mon Oct  7 01:59:12 2002
@@ -41,6 +41,15 @@
 
 public interface Channel
 {
-    public boolean isOpen();
-    public void close() throws IOException;
+  /**
+   * Tells whether this channel is open or not
+   */
+  public boolean isOpen();
+
+  /**
+   * Closes this channel
+   *
+   * @exception IOException If an error occurs
+   */
+  public void close() throws IOException;
 }
