--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Window.java	2004-06-08 02:24:45.000000000 -0600
+++ java/awt/Window.java	2004-06-15 02:19:28.000000000 -0600
@@ -703,7 +703,8 @@
    */
   public void applyResourceBundle(String rbName)
   {
-    ResourceBundle rb = ResourceBundle.getBundle(rbName);
+    ResourceBundle rb = ResourceBundle.getBundle(rbName, Locale.getDefault(),
+      Window.class.getClassLoader());
     if (rb != null)
       applyResourceBundle(rb);    
   }
