--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/TextComponent.java	2004-04-28 02:20:15.000000000 -0600
+++ java/awt/TextComponent.java	2004-06-18 02:18:50.000000000 -0600
@@ -141,6 +141,7 @@
   TextComponentPeer tcp = (TextComponentPeer)getPeer();
   if (tcp != null)
     tcp.setText(text);
+  setCaretPosition(0);
 }
 
 /*************************************************************************/
