--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicViewportUI.java	2004-06-08 02:25:43.000000000 -0600
+++ javax/swing/plaf/basic/BasicViewportUI.java	2004-06-18 02:18:59.000000000 -0600
@@ -173,10 +173,18 @@
         g.clearRect(0, 0, portBounds.width, portBounds.height);
       }
 
-    view.paint(g2);
+    g2.translate(-pos.x, -pos.y);
+    try 
+      {
+        view.paint(g2);
+      }
+    finally
+      {
+        g2.translate(pos.x, pos.y);
+      }
     g2 = null;
     g.drawImage(backingStoreImage, 
-                -pos.x, -pos.y, 
+                0, 0, 
                 (ImageObserver)null);
   }
 }
