--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectOutputStream.java	Sun Mar 23 02:18:39 2003
+++ java/io/ObjectOutputStream.java	Fri Mar 14 15:14:39 2003
@@ -843,16 +843,24 @@
   */
   public static abstract class PutField
   {
-    public abstract void put (String name, boolean value);
-    public abstract void put (String name, byte value);
-    public abstract void put (String name, char value);
-    public abstract void put (String name, double value);
-    public abstract void put (String name, float value);
-    public abstract void put (String name, int value);
-    public abstract void put (String name, long value);
-    public abstract void put (String name, short value);
-    public abstract void put (String name, Object value);
-    /** @deprecated */
+    public abstract void put (String name, boolean value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, byte value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, char value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, double value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, float value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, int value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, long value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, short value)
+      throws IOException, IllegalArgumentException;
+    public abstract void put (String name, Object value)
+      throws IOException, IllegalArgumentException;
     public abstract void write (ObjectOutput out) throws IOException;
   }
 
@@ -869,6 +877,7 @@
 	= new Object[currentObjectStreamClass.objectFieldCount];
 
 	public void put (String name, boolean value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -877,6 +886,7 @@
 	}
 
 	public void put (String name, byte value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -885,6 +895,7 @@
 	}
 
 	public void put (String name, char value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -895,6 +906,7 @@
 	}
 
 	public void put (String name, double value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -912,6 +924,7 @@
 	}
 
 	public void put (String name, float value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -925,6 +938,7 @@
 	}
 
 	public void put (String name, int value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -937,6 +951,7 @@
 	}
 
 	public void put (String name, long value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -953,6 +968,7 @@
 	}
 
 	public void put (String name, short value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
@@ -963,6 +979,7 @@
 	}
 
 	public void put (String name, Object value)
+	  throws IOException, IllegalArgumentException
 	{
 	  ObjectStreamField field
 	    = currentObjectStreamClass.getField (name);
