--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultCellEditor.java	2004-01-10 02:20:47.000000000 -0700
+++ javax/swing/DefaultCellEditor.java	2004-06-18 02:18:51.000000000 -0600
@@ -58,50 +58,34 @@
 {
   static final long serialVersionUID = 3564035141373880027L;
 
-	//-------------------------------------------------------------
-	// Classes ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * EditorDelegate
-	 */
-	protected class EditorDelegate implements ActionListener, 
-			ItemListener, Serializable {
-
-		//-------------------------------------------------------------
-		// Variables --------------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * value
-		 */
-		protected Object value;
-
-
-		//-------------------------------------------------------------
-		// Initialization ---------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * Constructor EditorDelegate
-		 * @param value0 TODO
-		 */
-		protected EditorDelegate(DefaultCellEditor editor) {
-			// TODO
-		} // EditorDelegate()
-
-
-		//-------------------------------------------------------------
-		// Methods ----------------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * setValue
-		 * @param event TODO
-		 */
-		public void setValue(Object event) {
-			// TODO
-		} // setValue()
+  /**
+   * EditorDelegate
+   */
+  protected class EditorDelegate
+    implements ActionListener, ItemListener, Serializable
+  {
+    private static final long serialVersionUID = -1420007406015481933L;
+
+    /**
+     * value
+     */
+    protected Object value;
+
+    /**
+     * Constructor EditorDelegate
+     * @param value0 TODO
+     */
+    protected EditorDelegate()
+    {
+    }
+
+    /**
+     * setValue
+     * @param event TODO
+     */
+    public void setValue(Object event)
+    {
+    }
 
 		/**
 		 * getCellEditorValue
@@ -172,11 +156,6 @@
 
 	} // EditorDelegate
 
-
-	//-------------------------------------------------------------
-	// Variables --------------------------------------------------
-	//-------------------------------------------------------------
-
 	/**
 	 * editorComponent
 	 */
@@ -192,11 +171,6 @@
 	 */
 	protected int clickCountToStart;
 
-
-	//-------------------------------------------------------------
-	// Initialization ---------------------------------------------
-	//-------------------------------------------------------------
-
 	/**
 	 * Constructor DefaultCellEditor
 	 * @param textfield TODO
@@ -221,11 +195,6 @@
 		// TODO
 	} // DefaultCellEditor()
 
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
-
 	/**
 	 * getComponent
 	 * @returns Component
@@ -320,6 +289,4 @@
 			Object value, boolean isSelected, int row, int column) {
 		return null; // TODO
 	} // getTableCellEditorComponent()
-
-
-} // DefaultCellEditor
+}
