--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/BufferedWriter.java	Tue Apr  8 02:23:45 2003
+++ java/io/BufferedWriter.java	Sun Mar 23 02:18:02 2003
@@ -61,7 +61,7 @@
   /**
    * This method initializes a new <code>BufferedWriter</code> instance
    * that will write to the specified subordinate <code>Writer</code>
-   * and which will use a default buffer size of 8192 chars.
+   * and which will use a default buffer size of 512 chars.
    *
    * @param out The underlying <code>Writer</code> to write data to
    */
