--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormatSymbols.java	2004-04-21 02:21:19.000000000 -0600
+++ java/text/DateFormatSymbols.java	2004-06-15 02:19:32.000000000 -0600
@@ -98,7 +98,8 @@
   public DateFormatSymbols (Locale locale) throws MissingResourceException
   {
     ResourceBundle res
-      = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation", locale);
+      = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation", locale,
+      				 getClass().getClassLoader());
 
     ampms = res.getStringArray ("ampms");
     eras = res.getStringArray ("eras");
