--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileDescriptor.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/FileDescriptor.java	2004-03-01 02:18:07.000000000 -0700
@@ -83,14 +83,6 @@
   /**
    * This method is used to initialize an invalid FileDescriptor object.
    */
-  public FileDescriptor()
-  {
-    channel = null;
-  }
-
-  /**
-   * This method is used to initialize a FileDescriptor object.
-   */
   FileDescriptor(ByteChannel channel)
   {
     this.channel = channel;
@@ -133,6 +125,6 @@
    */
   public boolean valid ()
   {
-    return channel != null && channel.isOpen();
+    return channel.isOpen();
   }
 }
