--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocket.java	2003-09-19 02:19:00.000000000 -0600
+++ java/net/DatagramSocket.java	2003-09-19 02:17:51.000000000 -0600
@@ -153,8 +153,21 @@
     if (s != null)
       s.checkListen(port);
 
-    // Why is there no factory for this?
-    impl = new PlainDatagramSocketImpl();
+    String propVal = System.getProperty("impl.prefix");
+    if (propVal == null || propVal.equals(""))
+      impl = new PlainDatagramSocketImpl();
+    else
+      try
+	{
+          impl = (DatagramSocketImpl) Class.forName
+            ("java.net." + propVal + "DatagramSocketImpl").newInstance();
+	}
+      catch (Exception e)
+	{
+	  System.err.println("Could not instantiate class: java.net." +
+	    propVal + "DatagramSocketImpl");
+	  impl = new PlainDatagramSocketImpl();
+	}
     impl.create();
 
     if (laddr == null)
@@ -249,29 +262,28 @@
    */
   public InetAddress getLocalAddress()
   {
-    if (impl == null)
+    if (impl == null
+	|| closed)
       return null;
     
-    // FIXME: According to libgcj, checkConnect() is supposed to be called
-    // before performing this operation.  Problems: 1) We don't have the
-    // addr until after we do it, so we do a post check.  2). The docs I
-    // see don't require this in the Socket case, only DatagramSocket, but
-    // we'll assume they mean both.
-
     InetAddress localAddr;
     
     try
       {
 	localAddr = (InetAddress) impl.getOption (SocketOptions.SO_BINDADDR);
+
+	SecurityManager s = System.getSecurityManager();
+	if (s != null)
+	  s.checkConnect (localAddr.getHostName(), -1);
+      }
+    catch (SecurityException e)
+      {
+	localAddr = InetAddress.ANY_IF;
       }
     catch (SocketException e)
       {
         return null;
       }
-    
-    SecurityManager sm = System.getSecurityManager();
-    if (sm != null)
-      sm.checkConnect(localAddr.getHostName(), getLocalPort());
 
     return localAddr;
   }
