--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Void.java	Mon Oct  1 14:44:13 2001
+++ java/lang/Void.java	Fri Oct 26 16:18:42 2001
@@ -47,7 +47,7 @@
    * The return type <code>void</code> is represented by this 
    * <code>Class</code> object.
    */
-  public static final Class TYPE = VMClassLoader.getPrimitiveClass("void");
+  public static final Class TYPE = VMClassLoader.getPrimitiveClass('V');
 
   /**
    * Don't allow Void objects to be made.
