--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicScrollBarUI.java	2004-07-09 02:22:41.000000000 -0600
+++ javax/swing/plaf/basic/BasicScrollBarUI.java	2004-06-18 02:18:58.000000000 -0600
@@ -571,7 +571,7 @@
 
   /**
    * This method configures the scrollbar's colors. This can be 
-   * done by looking up the standard colors from the Look and Feel defaults.
+   * done by looking up the standard colors from the L&F defaults.
    */
   protected void configureScrollBarColors()
   {
@@ -891,7 +891,7 @@
 
   /**
    * This method installs the defaults for the scrollbar specified
-   * by the Basic Look and Feel.
+   * by the Basic L&F.
    */
   protected void installDefaults()
   {
@@ -1257,7 +1257,7 @@
 
   /**
    * This method uninstalls any defaults that this
-   * scrollbar acquired from the Basic Look and Feel defaults.
+   * scrollbar acquired from the Basic L&F defaults.
    */
   protected void uninstallDefaults()
   {
