--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicProgressBarUI.java	2004-07-09 02:22:40.000000000 -0600
+++ javax/swing/plaf/basic/BasicProgressBarUI.java	2004-06-18 02:18:58.000000000 -0600
@@ -139,7 +139,7 @@
   // The total number of frames must be an even number.
   // The total number of frames is calculated from
   // the cycleTime and repaintInterval given by
-  // the basic Look and Feel defaults.
+  // the basic L&F's defaults.
   //
   // +-----------------------------------------------+
   // | frame0 | frame1 | frame2 | frame 3 | frame 4  |
