--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/ClosedSelectorException.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/ClosedSelectorException.java	2002-11-18 06:22:55.000000000 -0700
@@ -1,4 +1,4 @@
-/* ClosedSelectorException.java --
+/* ClosedSelectorException.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -37,7 +37,6 @@
 
 package java.nio.channels;
 
-
 /**
  * @author Michael Koch
  * @since 1.4
