--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Short.java	Mon Jul 23 14:15:25 2001
+++ java/lang/Short.java	Fri Oct 26 16:18:42 2001
@@ -56,7 +56,7 @@
    * The primitive type <code>short</code> is represented by this 
    * <code>Class</code> object.
    */
-  public static final Class TYPE = VMClassLoader.getPrimitiveClass("short");
+  public static final Class TYPE = VMClassLoader.getPrimitiveClass('S');
 
   /**
    * The immutable value of this Short.
