--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/IntViewBufferImpl.java	2004-02-12 02:22:34.000000000 -0700
+++ java/nio/IntViewBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -46,11 +46,20 @@
   private boolean readOnly;
   private ByteOrder endian;
   
+  IntViewBufferImpl (ByteBuffer bb, int capacity)
+  {
+    super (capacity, capacity, 0, -1);
+    this.bb = bb;
+    this.offset = bb.position();
+    this.readOnly = bb.isReadOnly();
+    this.endian = bb.order();
+  }
+  
   public IntViewBufferImpl (ByteBuffer bb, int offset, int capacity,
 			    int limit, int position, int mark,
 			    boolean readOnly, ByteOrder endian)
   {
-    super (limit >> 2, limit >> 2, position >> 2, mark >> 2);
+    super (capacity, limit, position, mark);
     this.bb = bb;
     this.offset = offset;
     this.readOnly = readOnly;
