--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/tree/DefaultMutableTreeNode.java	2004-05-21 02:27:14.000000000 -0600
+++ javax/swing/tree/DefaultMutableTreeNode.java	2003-10-13 02:17:51.000000000 -0600
@@ -997,17 +997,17 @@
 	public int getLeafCount() {
 
 		// Variables
-		Enumeration	e;
+		Enumeration	enum;
 		int			count;
 		TreeNode	current;
 
 		// Get Enumeration of all descendants
-		e = depthFirstEnumeration();
+		enum = depthFirstEnumeration();
 
 		// Process Nodes
 		count = 0;
-		while (e.hasMoreElements() == true) {
-			current = (TreeNode) e.nextElement();
+		while (enum.hasMoreElements() == true) {
+			current = (TreeNode) enum.nextElement();
 			if (current.isLeaf() == true) {
 				count++;
 			} // if
