--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/TreeUI.java	2003-08-17 02:19:18.000000000 -0600
+++ javax/swing/plaf/TreeUI.java	2003-08-02 02:18:05.000000000 -0600
@@ -126,7 +126,7 @@
    * Finds the path that is closest to the specified position.
    *
    * <p><img src="doc-files/TreeUI-1.png" width="300" height="250"
-   * alt="[A screen shot of a JTree]" />
+   * alt="[A screen shot of a JTree] />
    *
    * <p>As shown by the above illustration, the bounds of the
    * closest path do not necessarily need to contain the passed
