--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2004-01-03 02:20:34.000000000 -0700
+++ java/net/URLStreamHandler.java	2004-02-12 02:18:26.000000000 -0700
@@ -415,7 +415,7 @@
     InetAddress addr1 = getHostAddress (url1);
     InetAddress addr2 = getHostAddress (url2);
 
-    if (addr1 != null || addr2 != null)
+    if (addr1 != null && addr2 != null)
       return addr1.equals (addr2);
 
     String host1 = url1.getHost();
@@ -511,7 +511,7 @@
     int size = protocol.length() + host.length() + file.length() + 24;
     StringBuffer sb = new StringBuffer(size);
 
-    if (protocol != null && protocol.length() > 0)
+    if (protocol.length() != 0)
       {
 	sb.append(protocol);
 	sb.append(":");
