--- /home/tromey/gnu/egcs/classpath/classpath/java/security/SignatureException.java	Wed Mar 22 12:26:23 2000
+++ java/security/SignatureException.java	Wed Apr 25 11:13:35 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,41 +24,33 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 
 /**
-  * This exception is thrown when a problem is encountered with a
-  * digital signature.
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This exception is thrown when a problem is encountered with a
+ * digital signature.
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public class SignatureException extends GeneralSecurityException
 {
-
-/**
-  * This method initializes a new instance of <code>SignatureException</code>
-  * with no descriptive error message.
-  */
-public
-SignatureException()
-{
-  super();
-}
-
-/*************************************************************************/
-
-/**
-  * This method initializes a new instance of <code>SignatureException</code>
-  * with a descriptive error message.
-  */
-public
-SignatureException(String msg)
-{
-  super(msg);
+  /**
+   * This method initializes a new instance of <code>SignatureException</code>
+   * with no descriptive error message.
+   */
+  public SignatureException()
+  {
+    super();
+  }
+
+  /**
+   * This method initializes a new instance of <code>SignatureException</code>
+   * with a descriptive error message.
+   */
+  public SignatureException(String msg)
+  {
+    super(msg);
+  }
 }
-
-} // class SignatureException
-
