--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Graphics.java	2004-03-23 02:22:29.000000000 -0700
+++ java/awt/Graphics.java	2003-06-13 02:27:30.000000000 -0600
@@ -1,5 +1,5 @@
 /* Graphics.java -- Abstract Java drawing class
-   Copyright (C) 1999, 2000, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -417,7 +417,7 @@
   drawLine(x1, y2, x1, y1);
   setColor(br);
   drawLine(x2, y1, x2, y2);
-  drawLine(x2, y2, x1, y2);
+  drawLine(x2, y1, x1, y2);
   setColor(color);
 }
 
