--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormat.java	2004-04-18 02:22:24.000000000 -0600
+++ java/text/DateFormat.java	2003-12-03 02:18:10.000000000 -0700
@@ -196,9 +196,9 @@
    * object:
    * <P>
    * <ul>
-   * <li>Is not <code>null</code>.</li>
-   * <li>Is an instance of <code>DateFormat</code>.</li>
-   * <li>Has the same numberFormat field value as this object.</li>
+   * <li>Is not <code>null</code>.
+   * <li>Is an instance of <code>DateFormat</code>.
+   * <li>Has the same numberFormat field value as this object.
    * </ul>
    *
    * @param obj The object to test for equality against.
