--- /home/tromey/gnu/classpath/classpath/java/net/Socket.java	Fri Jan 10 15:27:27 2003
+++ java/net/Socket.java	Thu Feb 13 08:56:23 2003
@@ -692,10 +692,10 @@
    * this option implies that there is no timeout (ie, operations will 
    * block forever).  On systems that have separate read and write timeout
    * values, this method returns the read timeout.  This
-   * value is in thousandths of a second.
+   * value is in thousandths of a second (****????*****)
    *
-   * @param timeout The length of the timeout in thousandths of a second, or 
-   * 0 to indicate no timeout.
+   * @param timeout The length of the timeout in thousandth's of a second or 
+   * 0 if not set
    *
    * @exception SocketException If an error occurs or Socket not connected
    *
