--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/ReferralException.java	2004-04-20 02:21:37.000000000 -0600
+++ javax/naming/ReferralException.java	2003-06-28 02:18:23.000000000 -0600
@@ -1,5 +1,5 @@
 /* ReferralException.java --
-   Copyright (C) 2001, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,6 +38,7 @@
 
 package javax.naming;
 
+import java.lang.Exception;
 import java.util.Hashtable;
  
 /**
