--- /home/tromey/gnu/classpath/classpath/java/net/UnknownHostException.java	Sat Mar  2 12:13:13 2002
+++ java/net/UnknownHostException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* UnknownHostException.java -- The hostname is unknown
-   Copyright (C) 1998, 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
+/* UnknownHostException.java -- The hostname is not unknown
+   Copyright (C) 1998, 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,40 +37,52 @@
 
 package java.net;
 
-import java.io.IOException;
+/*
+ * Written using on-line Java Platform 1.2 API Specification.
+ * Status:  Believed complete and correct.
+ */
 
 /**
- * This exception indicates that an attempt was made to reference a hostname
- * or IP address that is not valid.  This could possibly indicate that a
- * DNS problem has occurred, but most often means that the host was not
- * correctly specified.
- *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @author Per Bothner
- * @status updated to 1.4
+  * This exception indicates that an attempt was made to reference a hostname
+  * or IP address that is not valid.  This could possibly indicate that a
+  * DNS problem has occurred, but most often means that the host was not
+  * correctly specified.
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @author Per Bothner 
+  * @date January 6, 1999.
+  */
+public class UnknownHostException extends java.io.IOException
+{
+
+/*
+ * Constructors
  */
-public class UnknownHostException extends IOException
+
+/**
+  * Initializes a new instance of <code>UnknownHostException</code>
+  * without a descriptive error message.
+  */
+public
+UnknownHostException()
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = -4639126076052875403L;
-
-  /**
-   * Create a new instance without a descriptive error message.
-   */
-  public UnknownHostException()
-  {
-  }
-
-  /**
-   * Create a new instance with a descriptive error message, such as the
-   * name of the host that could not be resolved.
-   *
-   * @param message a message describing the error that occurred
-   */
-  public UnknownHostException(String message)
-  {
-    super(message);
-  }
+  super();
+}
+
+/*************************************************************************/
+
+/**
+  * Initializes a new instance of <code>UnknownHostException</code>
+  * with a descriptive error message, such as the name of the host
+  * that could not be resolved.
+  *
+  * @param message A message describing the error that occurrred.
+  */
+public
+UnknownHostException(String message)
+{
+  super(message);
+}
+
 } // class UnknownHostException
+
