--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/AWTPermission.java	2004-04-18 02:22:10.000000000 -0600
+++ java/awt/AWTPermission.java	2002-08-12 10:10:54.000000000 -0600
@@ -79,7 +79,7 @@
  * <tr>
  *   <td><code>fullScreenExclusive</code></td>
  *   <td>enter full-screen exclusive mode</td>
- *   <td>malicious code could masquerade as a trusted program</td></tr>
+ *   <td>malicious code could masquerade as a trusted program</td><tr>
  * </table>
  *
  * @author Tom Tromey <tromey@redhat.com>
