--- /home/tromey/gnu/egcs/classpath/classpath/java/text/DecimalFormatSymbols.java	Sun Sep  2 13:37:58 2001
+++ java/text/DecimalFormatSymbols.java	Fri Nov  2 21:44:51 2001
@@ -476,7 +476,7 @@
   }
 
   /**
-   * This method sets the charcter used to represen the digit zero.
+   * This method sets the character used to represent the digit zero.
    *
    * @param zeroDigit The character used to represent the digit zero.
    */
