--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/SocketHandler.java	2004-04-18 02:22:29.000000000 -0600
+++ java/util/logging/SocketHandler.java	2003-06-21 04:31:55.000000000 -0600
@@ -68,7 +68,7 @@
  *
  * <li><code>java.util.SocketHandler.encoding</code> - specifies
  *     the name of the character encoding. Default value:
- *     the default platform encoding.</li>
+ *     the default platform encoding.
  *
  * <li><code>java.util.SocketHandler.host</code> - specifies
  *     the name of the host to which records are published.
