--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/event/MouseEvent.java	2004-03-23 02:22:31.000000000 -0700
+++ java/awt/event/MouseEvent.java	2002-08-12 10:11:06.000000000 -0600
@@ -220,11 +220,11 @@
     if ((modifiers & EventModifier.OLD_MASK) != 0)
       {
         if ((modifiers & BUTTON1_MASK) != 0)
-          this.button = BUTTON1;
+          button = BUTTON1;
         else if ((modifiers & BUTTON2_MASK) != 0)
-          this.button = BUTTON2;
+          button = BUTTON2;
         else if ((modifiers & BUTTON3_MASK) != 0)
-          this.button = BUTTON3;
+          button = BUTTON3;
       }
   }
 
