--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/FileNameMap.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/FileNameMap.java	2003-10-13 02:17:50.000000000 -0600
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -35,14 +35,15 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-package java.net;
 
+package java.net;
 
 /**
  * Written using on-line Java Platform 1.2 API Specification, as well
  * as "The Java Class Libraries", 2nd edition (Addison-Wesley, 1998).
  * Status:  Believed complete and correct.
  */
+
 /**
   * This interface has one method which, when passed a filename, returns
   * the MIME type associated with that filename.
@@ -62,4 +63,6 @@
     * @return The MIME type for the filename passed in.
     */
   String getContentTypeFor(String filename);
+
 } // interface FileNameMap
+
