--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/prefs/Preferences.java	2003-10-07 02:20:33.000000000 -0600
+++ java/util/prefs/Preferences.java	2003-12-10 02:18:06.000000000 -0700
@@ -45,7 +45,7 @@
 import java.security.Permission;
 import java.security.PrivilegedAction;
 
-import gnu.java.util.prefs.*;
+import gnu.java.util.prefs.NodeReader;
 
 /**
  * Preference node containing key value entries and subnodes.
