--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/PropertyResourceBundle.java	2003-08-17 02:19:12.000000000 -0600
+++ java/util/PropertyResourceBundle.java	2003-06-18 02:22:32.000000000 -0600
@@ -61,8 +61,8 @@
  * An example of a properties file for the german language is given
  * here. This extends the example given in ListResourceBundle.
  * Create a file MyResource_de.properties with the following contents
- * and put it in the CLASSPATH. (The char <code>\u00e4</code> is the
- * german umlaut)
+ * and put it in the CLASSPATH. (The char <code>\u00e4<char> is the
+ * german &auml;)
  *
  *
 <pre>
