--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Double.java	2003-07-17 02:19:17.000000000 -0600
+++ java/lang/Double.java	2003-09-24 02:18:21.000000000 -0600
@@ -1,6 +1,5 @@
 /* Double.java -- object wrapper for double
-   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -36,6 +35,7 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package java.lang;
 
 import gnu.classpath.Configuration;
@@ -428,10 +428,9 @@
    * @return the bits of the <code>double</code>
    * @see #longBitsToDouble(long)
    */
-  public static long doubleToLongBits(double value)
-  {
-    return VMDouble.doubleToLongBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native long doubleToLongBits(double value);
+  // END GCJ LOCAL
 
   /**
    * Convert the double to the IEEE 754 floating-point "double format" bit
@@ -447,10 +446,9 @@
    * @return the bits of the <code>double</code>
    * @see #longBitsToDouble(long)
    */
-  public static long doubleToRawLongBits(double value)
-  {
-    return VMDouble.doubleToRawLongBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native long doubleToRawLongBits(double value);
+  // END GCJ LOCAL
 
   /**
    * Convert the argument in IEEE 754 floating-point "double format" bit
@@ -465,10 +463,9 @@
    * @see #doubleToLongBits(double)
    * @see #doubleToRawLongBits(double)
    */
-  public static double longBitsToDouble(long bits)
-  {
-    return VMDouble.longBitsToDouble(bits);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native double longBitsToDouble(long bits);
+  // END GCJ LOCAL
 
   /**
    * Compare two Doubles numerically by comparing their <code>double</code>
