--- /home/tromey/gnu/egcs/classpath/classpath/java/security/InvalidAlgorithmParameterException.java	Sat Mar  2 12:13:16 2002
+++ java/security/InvalidAlgorithmParameterException.java	Tue Jan 22 16:01:36 2002
@@ -1,6 +1,5 @@
-/* InvalidAlgorithmParameterException.java -- an invalid parameter to a
-   security algorithm
-   Copyright (C) 2000, 2002 Free Software Foundation, Inc.
+/* InvalidAlgorithmParameterException.java
+   Copyright (C) 2000  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,33 +38,23 @@
 package java.security;
 
 /**
- * Thrown for an invalid security algorithm parameter.
- *
  * @author Warren Levy <warrenl@cygnus.com>
- * @since 1.2
- * @status updated to 1.4
+ * @date February 2, 2000.
  */
-public class InvalidAlgorithmParameterException
-  extends GeneralSecurityException
+
+/**
+ * Written using on-line Java Platform 1.2 API Specification.
+ * Status:  Believed complete and correct.
+ */
+
+// JDK1.2
+public class InvalidAlgorithmParameterException extends GeneralSecurityException
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = 2864672297499471472L;
-
-  /**
-   * Construct an exception with no message.
-   */
   public InvalidAlgorithmParameterException()
   {
     super();
   }
 
-  /**
-   * Construct an exception with a message.
-   *
-   * @param msg the message
-   */
   public InvalidAlgorithmParameterException(String msg)
   {
     super(msg);
