--- /home/tromey/gnu/egcs/classpath/classpath/java/security/DigestOutputStream.java	Mon Mar 25 00:05:18 2002
+++ java/security/DigestOutputStream.java	Tue Jan 22 16:01:36 2002
@@ -112,7 +112,7 @@
     if (state)
       digest.update((byte) b);
 
-    out.write(b);
+    super.write(b);
   }
 
   /**
@@ -131,7 +131,7 @@
     if (state)
       digest.update(b, off, len);
 
-    out.write(b, off, len);
+    super.write(b, off, len);
   }
 
   /**
