--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/KeyStroke.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/KeyStroke.java	2003-06-13 02:27:43.000000000 -0600
@@ -1,5 +1,5 @@
 /* KeyStroke.java --
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,6 +37,7 @@
 
 package javax.swing;
 
+import java.awt.AWTKeyStroke;
 import java.awt.event.KeyEvent;
 import java.io.Serializable;
 
