--- /home/tromey/gnu/egcs/classpath/classpath/java/io/PipedReader.java	Thu Aug 30 23:25:40 2001
+++ java/io/PipedReader.java	Fri Nov  2 21:44:51 2001
@@ -181,7 +181,7 @@
 	  else
 	    copylen = Math.min (len, buffer.length - in);
 
-	  // Copy chars until the pipe is filled, wrapping if neccessary.
+	  // Copy chars until the pipe is filled, wrapping if necessary.
 	  System.arraycopy(buf, bufpos, buffer, in, copylen);
 	  len -= copylen;
 	  bufpos += copylen;
