--- /home/tromey/gnu/classpath/classpath/java/awt/GridBagConstraints.java	Wed Oct  2 11:21:37 2002
+++ java/awt/GridBagConstraints.java	Wed Jan 23 18:05:10 2002
@@ -45,8 +45,6 @@
  * GridBagLayout layout manager.  */
 public class GridBagConstraints implements Cloneable, Serializable
 {
-  static final long serialVersionUID = -1000070633030801713L;
-
   /** Fill in both directions.  */
   public static final int BOTH = 1;
   /** Don't fill.  */
