--- /home/tromey/gnu/classpath/classpath/java/awt/TextField.java	Mon Dec 30 15:39:04 2002
+++ java/awt/TextField.java	Sun Mar 24 16:08:55 2002
@@ -84,11 +84,9 @@
  * Constructors
  */
 
-/**
+/*
  * Initializes a new instance of <code>TextField</code> that is empty
  * and has one column.
- *
- * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
  */
 public
 TextField()
@@ -104,8 +102,6 @@
   * length of the text string.
   *
   * @param text The text to display in the field.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextField(String text)
@@ -120,8 +116,6 @@
   * and has the specified number of columns.
   *
   * @param columns The number of columns in the text field.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextField(int columns)
@@ -137,17 +131,12 @@
   *
   * @param text The text to display in the field.
   * @param columns The number of columns in the field.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true,
   */
 public
 TextField(String text, int columns)
 {
   super(text);
   this.columns = columns;
-
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
