--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URI.java	2002-12-21 05:37:01.000000000 -0700
+++ java/net/URI.java	2002-10-05 01:49:08.000000000 -0600
@@ -164,7 +164,7 @@
    * @exception NullPointerException If str is null
    */
   public static URI create (String str)
-    throws IllegalArgumentException
+    throws IllegalArgumentException, URISyntaxException
   {
     return null;
   }
