--- /home/tromey/gnu/classpath/classpath/java/awt/Button.java	Mon Dec 30 15:39:03 2002
+++ java/awt/Button.java	Mon Aug 12 10:10:54 2002
@@ -90,9 +90,6 @@
 
 /**
   * Initializes a new instance of <code>Button</code> with no label.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true
   */
 public
 Button()
@@ -107,18 +104,12 @@
   * label.  The action command name is also initialized to this value.
   *
   * @param label The label to display on the button.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true
   */
 public
 Button(String label)
 {
   this.label = label;
   actionCommand = label;
-
-  if (GraphicsEnvironment.isHeadless ())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
@@ -220,10 +211,6 @@
 /** Returns all registered EventListers of the given listenerType. 
  * listenerType must be a subclass of EventListener, or a 
  * ClassClassException is thrown.
- *
- * @exception ClassCastException If listenerType doesn't specify a class or
- * interface that implements @see java.util.EventListener.
- *
  * @since 1.3 
  */
   public EventListener[] getListeners(Class listenerType)
