--- /home/tromey/gnu/egcs/classpath/classpath/java/util/WeakHashMap.java	Sun Feb 10 17:34:22 2002
+++ java/util/WeakHashMap.java	Tue Jan 22 16:01:44 2002
@@ -1,6 +1,6 @@
 /* java.util.WeakHashMap -- a hashtable that keeps only weak references
    to its keys, allowing the virtual machine to reclaim them
-   Copyright (C) 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -170,13 +170,6 @@
   private final class WeakEntrySet extends AbstractSet
   {
     /**
-     * Non-private constructor to reduce bytecode emitted.
-     */
-    WeakEntrySet()
-    {
-    }
-
-    /**
      * Returns the size of this set.
      *
      * @return the set size
@@ -490,7 +483,7 @@
      */
     WeakEntry getEntry()
     {
-      final Object key = this.get();
+      final Object key = get();
       if (key == null)
         return null;
       return new WeakEntry(key);
