--- /home/tromey/gnu/classpath/classpath/java/awt/MenuComponent.java	Mon Dec 30 15:39:04 2002
+++ java/awt/MenuComponent.java	Tue Jan 22 16:01:31 2002
@@ -92,14 +92,10 @@
 
 /**
   * Default constructor for subclasses.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 protected
 MenuComponent()
 {
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
