--- /home/tromey/gnu/classpath/classpath/java/lang/NoSuchMethodException.java	Sat Mar  2 12:13:12 2002
+++ java/lang/NoSuchMethodException.java	Tue Jan 22 16:01:35 2002
@@ -1,5 +1,6 @@
-/* NoSuchMethodException.java -- thrown when reflecting a non-existant method
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* NoSuchMethodException.java -- exception thrown to indicate the class
+   does not have the specified method.
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +8,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,35 +39,40 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
+ */
+
 /**
- * Thrown to indicate the class does not have the specified method. This is
- * caused by a variety of reflection methods, when looking up a method by name.
+ * Exceptions may be thrown by one part of a Java program and caught
+ * by another in order to deal with exceptional conditions.  
+ * Thrown to indicate the class does not have the specified method.
  *
+ * @since JDK 1.0
+ * 
  * @author Brian Jones
  * @author Warren Levy <warrenl@cygnus.com>
- * @status updated to 1.4
+ * @date September 18, 1998.
  */
 public class NoSuchMethodException extends Exception
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = 5034388446362600923L;
+  static final long serialVersionUID = 5034388446362600923L;
 
   /**
    * Create an exception without a message.
    */
   public NoSuchMethodException()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an exception with a message.
-   *
-   * @param s the message
    */
   public NoSuchMethodException(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
