--- /home/tromey/gnu/classpath/classpath/java/rmi/activation/ActivationID.java	Wed Oct  2 11:21:38 2002
+++ java/rmi/activation/ActivationID.java	Tue Jan 22 15:40:25 2002
@@ -41,9 +41,8 @@
 import java.rmi.Remote;
 import java.rmi.RemoteException;
 
-public class ActivationID implements Serializable
-{
-  static final long serialVersionUID = -4608673054848209235L;
+public class ActivationID
+	implements Serializable {
 
 private Activator activator;
 
