--- /home/tromey/gnu/classpath/classpath/java/util/zip/ZipInputStream.java	Fri Nov 29 21:37:14 2002
+++ java/util/zip/ZipInputStream.java	Sun Nov  3 14:05:25 2002
@@ -151,7 +151,7 @@
 	return null;
       }
     if (header != LOCSIG)
-      throw new ZipException("Wrong Local header signature: "
+      throw new ZipException("Wrong Local header signature"
 			     + Integer.toHexString(header));
     /* skip version */
     readLeShort();
