--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/reflect/Proxy.java	2003-08-27 02:18:56.000000000 -0600
+++ java/lang/reflect/Proxy.java	2003-08-27 02:17:52.000000000 -0600
@@ -269,9 +269,18 @@
                               ? getProxyData0(loader, interfaces)
                               : ProxyData.getProxyData(pt));
 
-            clazz = (Configuration.HAVE_NATIVE_GENERATE_PROXY_CLASS
-		     ? generateProxyClass0(loader, data)
-                     : new ClassFactory(data).generate(loader));
+            // FIXME workaround for bug in gcj 3.0.x
+            // Not needed with the latest gcj from cvs
+            //clazz = (Configuration.HAVE_NATIVE_GENERATE_PROXY_CLASS
+            //	       ? generateProxyClass0(loader, data)
+            //         : new ClassFactory(data).generate(loader));
+            if (Configuration.HAVE_NATIVE_GENERATE_PROXY_CLASS)
+              clazz = generateProxyClass0(loader, data);
+            else
+              {
+                ClassFactory cf = new ClassFactory(data);
+                clazz = cf.generate(loader);
+              }
           }
 
         Object check = proxyClasses.put(pt, clazz);
