--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/metal/MetalLookAndFeel.java	2003-10-13 02:22:58.000000000 -0600
+++ javax/swing/plaf/metal/MetalLookAndFeel.java	2004-02-12 02:18:27.000000000 -0700
@@ -40,7 +40,6 @@
 package javax.swing.plaf.metal;
 
 import javax.swing.UIDefaults;
-import javax.swing.plaf.basic.BasicDefaults;
 import javax.swing.plaf.basic.BasicLookAndFeel;
 
 public class MetalLookAndFeel extends BasicLookAndFeel
@@ -61,7 +60,7 @@
      public UIDefaults getDefaults()
 	 {
 	   if (LAF_defaults == null)
-	     LAF_defaults = new BasicDefaults();
+	     LAF_defaults = super.getDefaults();
 	     
 	     //      Returns the default values for this look and feel. 
 	     return LAF_defaults;
