--- /home/tromey/gnu/Nightly/classpath/classpath/javax/rmi/CORBA/Stub.java	2004-04-20 02:21:38.000000000 -0600
+++ javax/rmi/CORBA/Stub.java	2003-03-29 14:23:25.000000000 -0700
@@ -1,5 +1,5 @@
 /* Stub.java -- 
-   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,14 +38,16 @@
 
 package javax.rmi.CORBA;
 
-import gnu.javax.rmi.CORBA.DelegateFactory;
-import gnu.javax.rmi.CORBA.GetDelegateInstanceException;
-
 import java.io.IOException;
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 import java.io.Serializable;
 import java.rmi.RemoteException;
+//import org.omg.CORBA.ORB;
+//import org.omg.CORBA_2_3.portable.ObjectImpl;
+//import org.omg.CORBA.portable.ObjectImpl;
+import gnu.javax.rmi.CORBA.DelegateFactory;
+import gnu.javax.rmi.CORBA.GetDelegateInstanceException;
 
 public abstract class Stub extends ObjectImpl
   implements Serializable
