--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/StringBuffer.java	Tue Jan 22 15:49:57 2002
+++ java/lang/StringBuffer.java	Tue Jan 22 16:01:35 2002
@@ -111,10 +111,7 @@
    *  @return this <code>StringBuffer</code>.
    *  @see java.lang.String#valueOf(int)
    */
-  public StringBuffer append (int inum)
-  {
-    return append (String.valueOf(inum));
-  }
+  public native StringBuffer append (int inum);
 
   /** Append the <code>String</code> value of the argument to this <code>StringBuffer</code>.
    *  Uses <code>String.valueOf()</code> to convert to
