--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JToggleButton.java	2004-07-09 02:22:38.000000000 -0600
+++ javax/swing/JToggleButton.java	2004-06-18 02:18:54.000000000 -0600
@@ -115,8 +115,8 @@
   }
   
   /**
-   * Returns a string that specifies the name of the Look and Feel
-   * class that renders this component.
+   * Returns a string that specifies the name of the L&amp;F class that renders
+   * this component.
    */
   public String getUIClassID()
   {
