--- /home/tromey/gnu/classpath/classpath/java/net/Socket.java	Wed Nov 13 10:15:52 2002
+++ java/net/Socket.java	Thu Oct  3 11:50:56 2002
@@ -1028,6 +1028,15 @@
   }
 
   /**
+   * Checks if the socket is closed.
+   */
+  public boolean isClosed ()
+  {
+    // FIXME: implement this.
+    return false;
+  }
+
+  /**
    * Checks if the socket's input stream is shutdown
    */
   public boolean isInputShutdown ()
