--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/spi/AbstractSelectionKey.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/spi/AbstractSelectionKey.java	2004-03-19 02:19:08.000000000 -0700
@@ -1,4 +1,4 @@
-/* AbstractSelectionKey.java --
+/* AbstractSelectionKey.java -- 
    Copyright (C) 2002, 2003, 2004  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -39,40 +39,38 @@
 
 import java.nio.channels.SelectionKey;
 
-
 /**
  * @since 1.4
  */
-public abstract class AbstractSelectionKey extends SelectionKey
+public abstract class AbstractSelectionKey
+  extends SelectionKey
 {
-  private boolean cancelled;
+  private boolean cancelled = false;
 
   /**
    * Initializes the key.
    */
-  protected AbstractSelectionKey()
+  protected AbstractSelectionKey ()
   {
   }
 
   /**
    * Cancels this key.
    */
-  public final void cancel()
+  public final void cancel ()
   {
     if (isValid())
       {
 	((AbstractSelector) selector()).cancelKey(this);
-	cancelled = true;
+        cancelled = true;
       }
   }
 
   /**
    * Tells whether this key is valid or not.
-   *
-   * @return true if this key is valid, false otherwise
    */
-  public final boolean isValid()
+  public final boolean isValid ()
   {
-    return ! cancelled;
+    return !cancelled;
   }
 }
