--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/Collator.java	2004-04-18 02:22:24.000000000 -0600
+++ java/text/Collator.java	2002-10-08 18:10:16.000000000 -0600
@@ -183,10 +183,10 @@
    * object.  This will be true if and only if the following conditions are
    * met:
    * <ul>
-   * <li>The specified object is not <code>null</code>.</li>
-   * <li>The specified object is an instance of <code>Collator</code>.</li>
+   * <li>The specified object is not <code>null</code>.
+   * <li>The specified object is an instance of <code>Collator</code>.
    * <li>The specified object has the same strength and decomposition
-   * settings as this object.</li>
+   * settings as this object.
    * </ul>
    *
    * @param obj The <code>Object</code> to test for equality against
@@ -380,12 +380,7 @@
   }
 
   // Decompose a single character and append results to the buffer.
-  // FIXME: for libgcj this is a native method which handles
-  // decomposition.  For Classpath, for now, it does nothing.
-  final void decomposeCharacter (char c, StringBuffer buf)
-  {
-    buf.append (c);
-  }
+  native final void decomposeCharacter (char c, StringBuffer buf);
 
   /**
    * This is the current collation decomposition setting.
