--- /home/tromey/gnu/Nightly/classpath/classpath/java/applet/Applet.java	2004-04-30 02:23:40.000000000 -0600
+++ java/applet/Applet.java	2004-04-17 02:18:07.000000000 -0600
@@ -483,7 +483,7 @@
    */
   public Dimension preferredSize()
   {
-    return stub == null ? super.preferredSize () : getDimensions ();
+    return getDimensions ();
   }
 
   /**
@@ -494,7 +494,7 @@
    */
   public Dimension minimumSize()
   {
-    return stub == null ? super.minimumSize () : getDimensions ();
+    return getDimensions ();
   }
 
   /**
