--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2004-06-08 02:25:42.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2004-06-18 02:18:58.000000000 -0600
@@ -110,6 +110,7 @@
       "DesktopIconUI", "javax.swing.plaf.basic.BasicDesktopIconUI",
       "DesktopPaneUI", "javax.swing.plaf.basic.BasicDesktopPaneUI",
       "EditorPaneUI", "javax.swing.plaf.basic.BasicEditorPaneUI",
+      "FormattedTextFieldUI", "javax.swing.plaf.basic.BasicFormattedTextFieldUI",
       "InternalFrameUI", "javax.swing.plaf.basic.BasicInternalFrameUI",
       "LabelUI", "javax.swing.plaf.basic.BasicLabelUI",
       "ListUI", "javax.swing.plaf.basic.BasicListUI",
@@ -205,15 +206,15 @@
   private void loadResourceBundle(UIDefaults defaults)
   {
     ResourceBundle bundle;
-    Enumeration enum;
+    Enumeration e;
     String key;
     String value;
     bundle = ResourceBundle.getBundle("resources/basic");
     // Process Resources
-    enum = bundle.getKeys();
-    while (enum.hasMoreElements())
+    e = bundle.getKeys();
+    while (e.hasMoreElements())
       {
-        key = (String) enum.nextElement();
+        key = (String) e.nextElement();
         value = bundle.getString(key);
         defaults.put(key, value);
       }
@@ -346,7 +347,7 @@
         "ctrl F10", "maximize",
         "ctrl alt shift F6","selectPreviousFrame"
       }),
-      "Desktop.background", new ColorUIResource(0, 92, 92),
+      "Desktop.background", new ColorUIResource(175, 163, 236),
       "DesktopIcon.border", new BorderUIResource.CompoundBorderUIResource(null,
                                                                           null),
       "EditorPane.background", new ColorUIResource(Color.white),
@@ -419,7 +420,7 @@
       "FocusManagerClassName", "TODO",
       "FormView.resetButtonText", "Reset",
       "FormView.submitButtonText", "Submit Query",
-      "InternalFrame.activeTitleBackground", new ColorUIResource(lightPurple),
+      "InternalFrame.activeTitleBackground", new ColorUIResource(162, 167, 241),
       "InternalFrame.activeTitleForeground", new ColorUIResource(Color.black),
       "InternalFrame.border", new BorderUIResource.CompoundBorderUIResource(null,
                                                                             null),
@@ -427,8 +428,8 @@
       // XXX Don't use gif
       "InternalFrame.icon", new IconUIResource(new ImageIcon("icons/JavaCup.gif")),
       "InternalFrame.iconifyIcon", BasicIconFactory.createEmptyFrameIcon(),
-      "InternalFrame.inactiveTitleBackground", new ColorUIResource(Color.gray),
-      "InternalFrame.inactiveTitleForeground", new ColorUIResource(Color.lightGray),
+      "InternalFrame.inactiveTitleBackground", new ColorUIResource(Color.lightGray),
+      "InternalFrame.inactiveTitleForeground", new ColorUIResource(Color.black),
       "InternalFrame.maximizeIcon", BasicIconFactory.createEmptyFrameIcon(),
       "InternalFrame.minimizeIcon", BasicIconFactory.createEmptyFrameIcon(),
       "InternalFrame.titleFont", new FontUIResource("Dialog", Font.PLAIN, 12),
@@ -703,7 +704,7 @@
         "ctrl UP", "requestFocus",
         "ctrl KP_UP", "requestFocus"
       }),
-      "TabbedPane.background", new ColorUIResource(Color.GRAY),
+      "TabbedPane.background", new ColorUIResource(Color.LIGHT_GRAY),
       "TabbedPane.contentBorderInsets", new InsetsUIResource(2, 2, 3, 3),
       "TabbedPane.darkShadow", new ColorUIResource(Color.darkGray),
       "TabbedPane.focus", new ColorUIResource(Color.black),
