--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTable.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JTable.java	2004-06-18 02:18:54.000000000 -0600
@@ -58,6 +58,8 @@
   implements TableModelListener, Scrollable, TableColumnModelListener,
              ListSelectionListener, CellEditorListener, Accessible
 {
+  private static final long serialVersionUID = 3876025080382781659L;
+  
   public static final int AUTO_RESIZE_ALL_COLUMNS = 4;
   public static final int AUTO_RESIZE_LAST_COLUMN = 3;
   public static final int AUTO_RESIZE_NEXT_COLUMN = 1;
