--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/ServerSocket.java	2003-05-01 02:23:11.000000000 -0600
+++ java/net/ServerSocket.java	2003-02-13 02:17:38.000000000 -0700
@@ -164,36 +164,9 @@
     if (bindAddr == null)
       bindAddr = InetAddress.ANY_IF;
 
-    // create socket
     impl.create(true);
-
-    // bind to address/port
-    try {
-      impl.bind(bindAddr, port);
-    } catch (IOException exception) {
-      impl.close();
-      throw exception;
-    } catch (RuntimeException exception) {
-      impl.close();
-      throw exception;
-    } catch (Error error) {
-      impl.close();
-      throw error;
-    }
-
-    // listen on socket
-    try {
-      impl.listen(backlog);
-    } catch (IOException exception) {
-      impl.close();
-      throw exception;
-    } catch (RuntimeException exception) {
-      impl.close();
-      throw exception;
-    } catch (Error error) {
-      impl.close();
-      throw error;
-    }
+    impl.bind(bindAddr, port);
+    impl.listen(backlog);
   }
 
   /**
