--- /home/tromey/gnu/egcs/classpath/classpath/java/util/TooManyListenersException.java	Fri Feb 22 16:38:25 2002
+++ java/util/TooManyListenersException.java	Tue Jan 22 16:01:44 2002
@@ -1,6 +1,5 @@
-/* TooManyListenersException.java -- thrown when a unicast event can't accept
-   another Listener
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* java.util.TooManyListenersException
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,24 +38,26 @@
 
 package java.util;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
+ */
+
 /**
  * This exception is part of the java event model.  It is thrown if an
  * event listener is added via the addXyzEventListener method, but the
  * object doesn't support any more listeners, e.g. it only supports a
  * single event listener.
  *
- * @author Jochen Hoenicke
- * @author Warren Levy <warrenl@cygnus.com>
  * @see EventListener
  * @see EventObject
- * @since 1.1
- * @status updated to 1.4
+ * @author Jochen Hoenicke 
+ * @author Warren Levy <warrenl@cygnus.com>
  */
+
 public class TooManyListenersException extends Exception
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
   private static final long serialVersionUID = 5074640544770687831L;
 
   /**
@@ -68,8 +69,7 @@
 
   /**
    * Constructs a TooManyListenersException with a detail message.
-   *
-   * @param detail the detail message
+   * @param detail the detail message.
    */
   public TooManyListenersException(String detail)
   {
