--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ToggleButtonModel.java	2004-01-28 02:27:14.000000000 -0700
+++ javax/swing/ToggleButtonModel.java	2003-06-20 02:21:15.000000000 -0600
@@ -40,6 +40,11 @@
 
 public class ToggleButtonModel extends DefaultButtonModel
 {
+    ToggleButtonModel(JComponent c)
+    {
+	super(c);
+    }
+
     public void setPressed(boolean b)  
     {
 	if (! isEnabled())
