--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Package.java	Sun Oct  7 14:45:48 2001
+++ java/lang/Package.java	Fri Nov  2 21:44:51 2001
@@ -39,7 +39,7 @@
  * <a href="http://java.sun.com/products/jdk/1.3/docs/guide/versioning/spec/VersioningSpecification.html">Product Versioning Specification</a>.
  * It also allows packages to be sealed with respect to the originating URL.
  * <p>
- * The most usefull method is the <code>isCompatibleWith()</code> method that
+ * The most useful method is the <code>isCompatibleWith()</code> method that
  * compares a desired version of a specification with the version of the
  * specification as implemented by a package. A package is considered
  * compatible with another version if the version of the specification is
@@ -235,7 +235,8 @@
   public static Package getPackage(String name)
   {
     // Get the caller's classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
 
     if (cl != null)
@@ -251,7 +252,8 @@
   public static Package[] getPackages()
   {
     // Get the caller's classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
 
     if (cl != null)
