--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/AttributedCharacterIterator.java	2004-04-03 02:21:03.000000000 -0700
+++ java/text/AttributedCharacterIterator.java	2003-10-16 02:17:56.000000000 -0600
@@ -1,5 +1,5 @@
 /* AttributedCharacterIterator.java -- Iterate over attributes
-   Copyright (C) 1998, 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -60,7 +60,7 @@
   /**
    * This class defines attribute keys that are used as text attributes.
    */
-  class Attribute implements Serializable
+  public static class Attribute implements Serializable
   {
     private static final long serialVersionUID = -9142742483513960612L;
 
