--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandlerFactory.java	2004-04-13 02:20:24.000000000 -0600
+++ java/net/URLStreamHandlerFactory.java	2003-10-13 02:17:50.000000000 -0600
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,15 +37,15 @@
 
 package java.net;
 
-
 /**
  * Written using on-line Java Platform 1.2 API Specification, as well
  * as "The Java Class Libraries", 2nd edition (Addison-Wesley, 1998).
  * Status:  Believed complete and correct.
  */
+
 /**
  * This interface contains one method which maps the protocol portion of
- * a URL (eg, "http" in "http://www.urbanophile.com/arenn/") to a
+ * a URL (eg, "http" in "http://www.urbanophile.com/arenn/") to a 
  * <code>URLStreamHandler</code> object.
  *
  * @author Aaron M. Renn (arenn@urbanophile.com)
@@ -54,12 +54,14 @@
 public interface URLStreamHandlerFactory
 {
   /**
-    * This method maps the protocol portion of a URL to a
+    * This method maps the protocol portion of a URL to a 
     * <code>URLStreamHandler</code> object.
     *
     * @param protocol The protocol name to map ("http", "ftp", etc).
     *
     * @return The <code>URLStreamHandler</code> for the specified protocol
     */
-  URLStreamHandler createURLStreamHandler(String protocol);
+  URLStreamHandler createURLStreamHandler (String protocol);
+
 } // interface URLStreamHandlerFactory
+
