--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/ThreadGroup.java	2003-07-18 02:18:37.000000000 -0600
+++ java/lang/ThreadGroup.java	2002-06-18 09:34:07.000000000 -0600
@@ -718,7 +718,6 @@
     if (groups == null)
       return;
     threads.remove(t);
-    t.group = null;
     // Daemon groups are automatically destroyed when all their threads die.
     if (daemon_flag && groups.size() == 0 && threads.size() == 0)
       {
