--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/undo/StateEditable.java	2003-11-15 02:20:02.000000000 -0700
+++ javax/swing/undo/StateEditable.java	2004-01-11 02:18:05.000000000 -0700
@@ -80,7 +80,7 @@
    * Sun&#x2019;s reference implementation, GNU Classpath also has to
    * provide this field. However, we do not try to match its value.
    */
-  static final String RCSID = "";
+  String RCSID = "";
 
 
   /**
