--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSeparator.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JSeparator.java	2004-03-18 02:18:22.000000000 -0700
@@ -37,6 +37,8 @@
 
 package javax.swing;
 
+import java.io.IOException;
+import java.io.ObjectOutputStream;
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
