--- /home/tromey/gnu/Nightly/classpath/classpath/javax/imageio/spi/ImageInputStreamSpi.java	2004-10-04 02:33:45.000000000 -0600
+++ javax/imageio/spi/ImageInputStreamSpi.java	2004-04-21 02:20:31.000000000 -0600
@@ -40,9 +40,9 @@
 
 import java.io.File;
 import java.io.IOException;
-
 import javax.imageio.stream.ImageInputStream;
 
+
 /**
  * An abstract superclass for service providers that create
  * {@linkplain javax.imageio.stream.ImageInputStream image input
