--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JViewport.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JViewport.java	2004-05-27 02:19:06.000000000 -0600
@@ -114,7 +114,13 @@
    *
    * @see #toViewCoordinates
    */
-  Dimension viewExtent;
+  Dimension extentSize;
+
+  /**
+   * The width and height of the view in its own coordinate space.
+   */
+
+  Dimension viewSize;
 
   Point lastPaintPosition;
 
@@ -124,17 +130,35 @@
     updateUI();
   }
 
-  public Dimension getViewSize()
+  public Dimension getExtentSize()
   {
-    if (viewExtent == null)
+    if (extentSize == null)
       return getPreferredSize();
     else
-      return viewExtent;
+      return extentSize;
+  }
+
+  public void setExtentSize(Dimension newSize)
+  {
+    extentSize = newSize;
+    fireStateChanged();
+  }
+
+  public Dimension getViewSize()
+  {
+    if (viewSize == null)
+      return getView().getPreferredSize();
+    else
+      return viewSize;
   }
 
+
   public void setViewSize(Dimension newSize)
   {
-    viewExtent = newSize;
+    viewSize = newSize;
+    Component view = getView();
+    if (view != null)
+      view.setSize(newSize);
     fireStateChanged();
   }
 
@@ -166,7 +190,7 @@
   public Rectangle getViewRect()
   {
     return new Rectangle(getViewPosition(), 
-                         getViewSize());
+                         getExtentSize());
   }
 
   public boolean isBackingStoreEnabled()
@@ -207,7 +231,24 @@
     add(v);
     fireStateChanged();
   }
-    
+
+  public void revalidate()
+  {
+    fireStateChanged();
+    super.revalidate();
+  }
+
+  public void reshape(int x, int y, int w, int h)
+  {
+    boolean changed = 
+      (x != getX()) 
+      || (y != getY()) 
+      || (w != getWidth())
+      || (h != getHeight());
+    super.reshape(x, y, w, h);
+    if (changed)
+      fireStateChanged();
+  }
     
   public void addImpl(Component comp, Object constraints, int index)
   {
