--- /home/tromey/gnu/classpath/classpath/java/awt/BufferCapabilities.java	Mon Dec 30 15:39:03 2002
+++ java/awt/BufferCapabilities.java	Thu Aug  8 22:26:13 2002
@@ -67,11 +67,6 @@
   private final ImageCapabilities back;
   private final FlipContents flip;
 
-  /**
-   * Creates a buffer capabilities object.
-   *
-   * @exception IllegalArgumentException If frontCaps or backCaps are null.
-   */
   public BufferCapabilities(ImageCapabilities front, ImageCapabilities back,
                             FlipContents flip)
   {
