--- /home/tromey/gnu/egcs/classpath/classpath/java/io/InvalidObjectException.java	Sat Mar  2 12:13:09 2002
+++ java/io/InvalidObjectException.java	Tue Jan 22 16:01:33 2002
@@ -1,5 +1,5 @@
-/* InvalidObjectException.java -- deserialization failed verification
-   Copyright (C) 1998, 2002 Free Software Foundation, Inc.
+/* InvalidObjectException.java -- An I/O operation was interrupted.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -39,28 +39,30 @@
 package java.io;
 
 /**
- * This exception is thrown when an object fails a validation test
- * during serialization.
- *
- * @author Aaron M. Renn (arenn@urbanophile.com)
- * @since 1.1
- * @status updated to 1.4
- */
+  * This exception is thrown when an object fails a validation test
+  * during serialization.
+  *
+  * @version 0.0
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  */
 public class InvalidObjectException extends ObjectStreamException
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = 3233174318281839583L;
-
-  /**
-   * Create an exception with a descriptive error message String. This should
-   * be the cause of the verification failure.
-   *
-   * @param message the descriptive error message
-   */
-  public InvalidObjectException(String message)
-  {
-    super(message);
-  }
+
+/*
+ * Constructors
+ */
+
+/**
+  * Create a new InvalidObjectException with a descriptive error message String
+  *
+  * @param message The descriptive error message
+  */
+public
+InvalidObjectException(String message)
+{
+  super(message);
+}
+
 } // class InvalidObjectException
+
