--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Frame.java	2003-09-18 02:19:48.000000000 -0600
+++ java/awt/Frame.java	2003-09-20 02:18:08.000000000 -0600
@@ -438,25 +438,6 @@
 /*************************************************************************/
 
 /**
-  * Destroys any resources associated with this frame.  This includes
-  * all components in the frame and all owned toplevel windows.
-  */
-public void
-dispose()
-{
-  Enumeration e = ownedWindows.elements();
-  while(e.hasMoreElements())
-    {
-      Window w = (Window)e.nextElement();
-      w.dispose();
-    }
-
-  super.dispose();
-}
-
-/*************************************************************************/
-
-/**
   * Returns a debugging string describing this window.
   *
   * @return A debugging string describing this window.
@@ -472,8 +453,6 @@
 {
   //Frame[] array = new Frames[frames.size()];
   //return frames.toArray(array);
-    
-    // see finalize() comment
   String msg = "FIXME: can't be implemented without weak references";
   throw new UnsupportedOperationException(msg);
 }
