--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Toolkit.java	2003-06-28 02:19:14.000000000 -0600
+++ java/awt/Toolkit.java	2003-06-13 02:27:31.000000000 -0600
@@ -96,7 +96,7 @@
 {
   /** The default toolkit name. */
   private static String default_toolkit_name
-    = "gnu.java.awt.peer.gtk.GtkToolkit";
+    = gnu.classpath.Configuration.default_awt_peer_toolkit;
 
   /**
    * The toolkit in use.  Once we load it, we don't ever change it
