--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/PrivateKey.java	2002-03-29 01:30:51.000000000 -0700
+++ java/security/PrivateKey.java	2003-05-04 02:19:12.000000000 -0600
@@ -56,7 +56,7 @@
 public interface PrivateKey extends Key
 {
   /**
-   * The verion identifier used for serialization.
+   * The version identifier used for serialization.
    */
   long serialVersionUID = 6034044314589513430L;
 } // interface PrivateKey
