--- /home/tromey/gnu/egcs/classpath/classpath/java/util/SimpleTimeZone.java	Tue Apr 30 17:58:37 2002
+++ java/util/SimpleTimeZone.java	Tue Jan 22 16:01:42 2002
@@ -52,8 +52,7 @@
  *
  * @see Calendar
  * @see GregorianCalender 
- * @author Jochen Hoenicke
- */
+ * @author Jochen Hoenicke */
 public class SimpleTimeZone extends TimeZone
 {
   /**
@@ -437,9 +436,9 @@
    * In the standard JDK the results given by this method may result in
    * inaccurate results at the end of February or the beginning of March.
    * To avoid this, you should use Calendar instead:
-   * <code>offset = cal.get(Calendar.ZONE_OFFSET)
-   * + cal.get(Calendar.DST_OFFSET);</code>
-   *
+   * <pre>
+   * offset = cal.get(Calendar.ZONE_OFFSET) + cal.get(Calendar.DST_OFFSET);
+   * </pre>
    * You could also use in
    *
    * This version doesn't suffer this inaccuracy.
