--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JProgressBar.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JProgressBar.java	2004-04-17 02:18:07.000000000 -0600
@@ -38,7 +38,8 @@
 package javax.swing;
 
 import java.awt.Graphics;
-
+import java.io.IOException;
+import java.io.ObjectOutputStream;
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
@@ -46,6 +47,7 @@
 import javax.accessibility.AccessibleValue;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
+import javax.swing.event.EventListenerList;
 import javax.swing.plaf.ProgressBarUI;
 
 
