--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicButtonListener.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicButtonListener.java	2004-03-18 02:18:23.000000000 -0700
@@ -45,12 +45,13 @@
 import java.awt.event.MouseMotionListener;
 import java.beans.PropertyChangeEvent;
 import java.beans.PropertyChangeListener;
-
 import javax.swing.AbstractButton;
 import javax.swing.ButtonModel;
-import javax.swing.JComponent;
+import javax.swing.UIDefaults;
+import javax.swing.UIManager;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
+import javax.swing.JComponent;
 
 public class BasicButtonListener
   implements MouseListener, MouseMotionListener, FocusListener, 
