--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Currency.java	2004-06-21 06:24:02.000000000 -0600
+++ java/util/Currency.java	2004-06-15 02:19:32.000000000 -0600
@@ -55,7 +55,7 @@
   {
     this.locale = loc;
     this.res = ResourceBundle.getBundle ("gnu.java.locale.LocaleInformation", 
-      locale, ClassLoader.getSystemClassLoader());
+      locale, Currency.class.getClassLoader());
   }
 
   /**
