--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JEditorPane.java	2004-02-06 02:21:26.000000000 -0700
+++ javax/swing/JEditorPane.java	2004-02-12 02:18:27.000000000 -0700
@@ -137,7 +137,7 @@
     { return super.getText();    }
     
     public String getUIClassID()
-    {    return "JEditorPane";  }
+    {    return "EditorPaneUI";  }
 
     public boolean isFocusCycleRoot()
     { return focus_root;    }
