--- /home/tromey/gnu/Nightly/classpath/classpath/java/sql/Time.java	Thu Jun 20 23:34:12 2002
+++ java/sql/Time.java	Tue Apr  1 02:18:00 2003
@@ -61,9 +61,11 @@
    * date in JDBC format into a Java date.
    *
    * @param str The string to parse.
-   * @return The resulting <code>java.sql.Time</code> value. 
+   * @return The resulting <code>java.sql.Time</code> value.
+   *
+   * @deprecated
    */
-  public static Time valueOf(String str)
+  public static Time valueOf (String str)
   {
     try
       {
@@ -110,11 +112,12 @@
    * This method returns this date in JDBC format.
    *
    * @return This date as a string.
+   *
+   * @deprecated
    */
-  public String
-  toString()
+  public String toString ()
   {
-    return sdf.format(this);
+    return sdf.format (this);
   }
 
 }
