--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/WriteAbortedException.java	2004-04-13 02:20:22.000000000 -0600
+++ java/io/WriteAbortedException.java	2002-06-15 13:07:49.000000000 -0600
@@ -81,7 +81,7 @@
   /**
    * This method returns a message indicating what went wrong, in this
    * format:
-   * <code>super.getMessage() + (detail == null ? "" : "; " + detail)</code>.
+   * <code>super.getMessage() + (detail == null ? "" : "; " + detail)<code>.
    *
    * @return the chained message
    */
