--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/RandomAccessFile.java	2004-05-02 02:20:13.000000000 -0600
+++ java/io/RandomAccessFile.java	2004-04-21 02:18:08.000000000 -0600
@@ -194,14 +194,12 @@
   }
 
   /**
-   * This method sets the length of the file to the specified length.
-   * If the currently length of the file is longer than the specified
-   * length, then the file is truncated to the specified length (the
-   * file position is set to the end of file in this case).  If the
-   * current length of the file is shorter than the specified length,
-   * the file is extended with bytes of an undefined value (the file
-   * position is unchanged in this case).
-   * <p>
+   * This method sets the length of the file to the specified length.  If
+   * the currently length of the file is longer than the specified length,
+   * then the file is truncated to the specified length.  If the current
+   * length of the file is shorter than the specified length, the file
+   * is extended with bytes of an undefined value.
+   *  <p>
    * The file must be open for write access for this operation to succeed.
    *
    * @param newlen The new length of the file
@@ -210,17 +208,7 @@
    */
   public void setLength (long newLen) throws IOException
   {
-    // FileChannel.truncate() can only shrink a file.
-    // To expand it we need to seek forward and write at least one byte.
-    if (newLen < length())
-      ch.truncate (newLen);
-    else
-      {
-	long pos = getFilePointer();
-	seek(newLen - 1);
-	write(0);
-	seek(pos);
-      }
+    ch.truncate (newLen);
   }
 
   /**
