--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicScrollBarUI.java	2004-06-08 02:25:42.000000000 -0600
+++ javax/swing/plaf/basic/BasicScrollBarUI.java	2004-06-18 02:18:58.000000000 -0600
@@ -791,7 +791,7 @@
    */
   public Dimension getPreferredSize(JComponent c)
   {
-    layoutContainer(scrollbar);
+    calculatePreferredSize();
     return preferredSize;
   }
 
@@ -957,8 +957,7 @@
 	trackRect = new Rectangle();
 	thumbRect = new Rectangle();
 
-	scrollTimer = new Timer();
-	scrollTimer.setDelay(200);
+	scrollTimer = new Timer(200, null);
 	scrollTimer.setRepeats(true);
 
 	installComponents();
