--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/MediaTracker.java	2003-07-25 02:18:53.000000000 -0600
+++ java/awt/MediaTracker.java	2002-03-06 09:29:56.000000000 -0700
@@ -76,9 +76,9 @@
 			       int width, int height)
     {
       if ((flags & ABORT) != 0)
-        status = ABORTED | COMPLETE;
+        status = ABORTED & COMPLETE;
       else if ((flags & ERROR) != 0)
-        status = ERRORED | COMPLETE;
+        status = ERRORED & COMPLETE;
       else if ((flags & ALLBITS) != 0)
         status = COMPLETE;
       else
