--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Toolkit.java	Wed Mar 26 02:18:41 2003
+++ java/awt/Toolkit.java	Tue Mar 25 02:17:55 2003
@@ -96,7 +96,7 @@
 {
   /** The default toolkit name. */
   private static String default_toolkit_name
-    = "gnu.java.awt.peer.gtk.GtkToolkit";
+    = gnu.classpath.Configuration.default_awt_peer_toolkit;
 
   /**
    * The toolkit in use.  Once we load it, we don't ever change it
