--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTree.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JTree.java	2004-02-12 02:18:27.000000000 -0700
@@ -1,5 +1,5 @@
 /* JTree.java -- 
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,10 +40,23 @@
 
 import java.awt.Dimension;
 import java.awt.Rectangle;
-
+import java.io.Serializable;
+import java.util.Hashtable;
+import java.util.Vector;
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
+import javax.swing.event.TreeModelEvent;
+import javax.swing.event.TreeModelListener;
+import javax.swing.event.TreeSelectionEvent;
+import javax.swing.event.TreeSelectionListener;
 import javax.swing.plaf.TreeUI;
+import javax.swing.tree.DefaultTreeSelectionModel;
+import javax.swing.tree.TreeCellEditor;
+import javax.swing.tree.TreeCellRenderer;
+import javax.swing.tree.TreeModel;
+import javax.swing.tree.TreeNode;
+import javax.swing.tree.TreePath;
+import javax.swing.tree.TreeSelectionModel;
 
 public class JTree extends JComponent implements Scrollable, Accessible
 {
