--- /home/tromey/gnu/classpath/classpath/java/net/HttpURLConnection.java	Sun Nov 24 16:32:33 2002
+++ java/net/HttpURLConnection.java	Sun Nov 17 14:01:15 2002
@@ -142,11 +142,6 @@
    * a conditional GET.
    */
   public static final int HTTP_NOT_MODIFIED = 304;
-
-  /**
-   * The requested resource needs to be accessed through a proxy.
-   */
-  public static final int HTTP_USE_PROXY = 305;
   
   
   /* HTTP Client Error Response Codes */
@@ -249,7 +244,7 @@
    * The server encountered an unexpected error (such as a CGI script crash)
    * that prevents the request from being fulfilled.
    */
-  public static final int HTTP_INTERNAL_ERROR   = 501;
+  public static final int HTTP_INTERNAL_ERROR   = 500;
 
   /**
    * The server does not support the requested functionality.  
@@ -464,36 +459,31 @@
       connect();
       
     gotResponseVals = true;
+    // Response is the first header received from the connection.
+    String respField = getHeaderField(0);
+    
+    if (respField == null || ! respField.startsWith("HTTP/"))
+      {
+	// Set to default values on failure.
+        responseCode = -1;
+	responseMessage = null;
+	return;
+      }
 
-    // If responseCode not yet explicitly set by subclass
-    if (responseCode == -1)
+    int firstSpc, nextSpc;
+    firstSpc = respField.indexOf(' ');
+    nextSpc = respField.indexOf(' ', firstSpc + 1);
+    responseMessage = respField.substring(nextSpc + 1);
+    String codeStr = respField.substring(firstSpc + 1, nextSpc);
+    try
+      {
+	responseCode = Integer.parseInt(codeStr);
+      }
+    catch (NumberFormatException e)
       {
-	// Response is the first header received from the connection.
-	String respField = getHeaderField(0);
-	
-	if (respField == null || ! respField.startsWith("HTTP/"))
-	  {
-	    // Set to default values on failure.
-	    responseCode = -1;
-	    responseMessage = null;
-	    return;
-	  }
-
-	int firstSpc, nextSpc;
-	firstSpc = respField.indexOf(' ');
-	nextSpc = respField.indexOf(' ', firstSpc + 1);
-	responseMessage = respField.substring(nextSpc + 1);
-	String codeStr = respField.substring(firstSpc + 1, nextSpc);
-	try
-	  {
-	    responseCode = Integer.parseInt(codeStr);
-	  }
-	catch (NumberFormatException e)
-	  {
-	    // Set to default values on failure.
-	    responseCode = -1;
-	    responseMessage = null;
-	  }
+	// Set to default values on failure.
+        responseCode = -1;
+	responseMessage = null;
       }
   }
 
