--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/event/SwingPropertyChangeSupport.java	2004-01-11 02:20:39.000000000 -0700
+++ javax/swing/event/SwingPropertyChangeSupport.java	2004-05-27 02:19:07.000000000 -0600
@@ -210,8 +210,9 @@
 		PropertyChangeListener	listener;
 
 		// Check Values if they are equal
-		if (event.getOldValue() == null || event.getNewValue() == null ||
-			event.getOldValue().equals(event.getNewValue()) == true) {
+		if (event.getOldValue() == null && event.getNewValue() == null ||
+		    (event.getOldValue() != null && event.getNewValue() != null &&
+	            event.getOldValue().equals(event.getNewValue()))) {
 			return;
 		} // if
 
