--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/JTableHeader.java	2004-01-28 02:27:14.000000000 -0700
+++ javax/swing/table/JTableHeader.java	2003-06-11 10:36:11.000000000 -0600
@@ -60,7 +60,7 @@
 import javax.swing.event.TableColumnModelListener;
 import javax.swing.plaf.TableHeaderUI;
 
-public class JTableHeader extends JComponent
+public class JTableHeader
 {
   protected class AccessibleJTableHeader
   {
