--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/spi/NamingManager.java	2003-11-07 02:21:37.000000000 -0700
+++ javax/naming/spi/NamingManager.java	2003-12-06 02:18:23.000000000 -0700
@@ -38,8 +38,18 @@
 
 package javax.naming.spi;
 
-import java.util.*;
-import javax.naming.*;
+import java.util.Enumeration;
+import java.util.Hashtable;
+import java.util.StringTokenizer;
+import javax.naming.CannotProceedException;
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingException;
+import javax.naming.NoInitialContextException;
+import javax.naming.RefAddr;
+import javax.naming.Reference;
+import javax.naming.Referenceable;
+import javax.naming.StringRefAddr;
 
 public class NamingManager
 {
