--- /home/tromey/gnu/egcs/classpath/classpath/java/text/Collator.java	Tue Jan 22 15:50:04 2002
+++ java/text/Collator.java	Tue Jan 22 16:01:37 2002
@@ -380,12 +380,7 @@
   }
 
   // Decompose a single character and append results to the buffer.
-  // FIXME: for libgcj this is a native method which handles
-  // decomposition.  For Classpath, for now, it does nothing.
-  final void decomposeCharacter (char c, StringBuffer buf)
-  {
-    buf.append (c);
-  }
+  native final void decomposeCharacter (char c, StringBuffer buf);
 
   /**
    * This is the current collation decomposition setting.
