--- /home/tromey/gnu/Nightly/classpath/classpath/java/beans/Introspector.java	2004-04-18 02:22:14.000000000 -0600
+++ java/beans/Introspector.java	2003-12-05 02:18:14.000000000 -0700
@@ -65,7 +65,7 @@
  * When you call getBeanInfo(class c), the Introspector
  * first searches for BeanInfo class to see if you
  * provided any explicit information.  It searches for a
- * class named &lt;bean class name&gt;BeanInfo in different
+ * class named <bean class name>BeanInfo in different
  * packages, first searching the bean class's package
  * and then moving on to search the beanInfoSearchPath.<P>
  *
@@ -113,14 +113,14 @@
  *     type &lt;type&gt;.  There may also be a
  *     <CODE>public &lt;type&gt;[] getXXX()</CODE> and a
  *     <CODE>public void setXXX(&lt;type&gt;)</CODE>
- *     method as well.</LI>
+ *     method as well.</CODE></LI>
  * <LI>If there is a
  *     <CODE>public void setXXX(int,&lt;type&gt;)</CODE>
  *     method, then it is a write-only indexed property of
  *     type &lt;type&gt;.  There may also be a
  *     <CODE>public &lt;type&gt;[] getXXX()</CODE> and a
  *     <CODE>public void setXXX(&lt;type&gt;)</CODE>
- *     method as well.</LI>
+ *     method as well.</CODE></LI>
  * <LI>If there is a
  *     <CODE>public &lt;type&gt; getXXX()</CODE> method,
  *     then XXX is a read-only property of type
