--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/MediaTracker.java	2003-12-03 02:20:11.000000000 -0700
+++ java/awt/MediaTracker.java	2004-06-18 02:18:50.000000000 -0600
@@ -81,12 +81,17 @@
         status = ERRORED | COMPLETE;
       else if ((flags & ALLBITS) != 0)
         status = COMPLETE;
-      else
+      else if ((flags & SOMEBITS) != 0)
         status = LOADING;
-      
-      synchronized (MediaTracker.this)
+      else
+        status = 0;
+
+      if ((status & COMPLETE) == COMPLETE)
       {
-	MediaTracker.this.notifyAll();
+        synchronized (MediaTracker.this)
+        {
+          MediaTracker.this.notifyAll();
+        }
       }
       // If status is not COMPLETE then we need more updates.
       return (status & COMPLETE) == 0;
@@ -106,7 +111,8 @@
     e.next = head;
     head = e;
     // Start tracking image status.
-    target.checkImage(image, e);
+    int flags = target.checkImage(image, e);
+    e.imageUpdate(image, flags, -1, -1, -1, -1);
   }
 
   public void addImage(Image image, int id, int width, int height)
@@ -119,7 +125,8 @@
     e.height = height;
     head = e;
     // Start tracking image status.
-    target.checkImage(image, width, height, e);
+    int flags = target.checkImage(image, width, height, e);
+    e.imageUpdate(image, flags, -1, -1, width, height);
   }
 
   public boolean checkAll()
