--- /home/tromey/gnu/classpath/classpath/java/rmi/activation/ActivationDesc.java	Wed Oct  2 11:21:38 2002
+++ java/rmi/activation/ActivationDesc.java	Tue Jan 22 15:40:25 2002
@@ -40,9 +40,8 @@
 import java.io.Serializable;
 import java.rmi.MarshalledObject;
 
-public final class ActivationDesc implements Serializable
-{
-  static final long serialVersionUID = 7455834104417690957L;
+public final class ActivationDesc
+	implements Serializable {
 
 private ActivationGroupID groupid;
 private String classname;
