--- /home/tromey/gnu/classpath/classpath/java/lang/ref/Reference.java	Tue Jan 22 15:49:58 2002
+++ java/lang/ref/Reference.java	Tue Jan 22 16:01:35 2002
@@ -75,8 +75,21 @@
   /**
    * The underlying object.  This field is handled in a special way by
    * the garbage collection.
+   * GCJ LOCAL:
+   * This is a RawData because it must be disguised from the GC.
+   * END GCJ LOCAL
    */
-  Object referent;
+  gnu.gcj.RawData referent;
+
+  /**
+   * This is like REFERENT but is not scanned by the GC.  We keep a
+   * copy around so that we can see when clear() has been called.
+   * GCJ LOCAL:
+   * This field doesn't exist in Classpath; we use it to detect
+   * clearing.
+   * END GCJ LOCAL
+   */
+  gnu.gcj.RawData copy;
 
   /**
    * The queue this reference is registered on. This is null, if this
@@ -108,7 +121,7 @@
    */
   Reference(Object ref)
   {
-    referent = ref;
+    create (ref);
   }
 
   /**
@@ -123,11 +136,16 @@
   {
     if (q == null)
       throw new NullPointerException();
-    referent = ref;
     queue = q;
+    create (ref);
   }
 
   /**
+   * Notifies the VM that a new Reference has been created.
+   */
+  private native void create (Object o);
+
+  /**
    * Returns the object, this reference refers to.
    * @return the object, this reference refers to, or null if the 
    * reference was cleared.
@@ -149,6 +167,7 @@
   public void clear()
   {
     referent = null;
+    copy = null;
   }
 
   /**
