--- /home/tromey/gnu/classpath/classpath/java/nio/channels/ServerSocketChannel.java	Tue Apr 30 17:58:37 2002
+++ java/nio/channels/ServerSocketChannel.java	Wed Sep 25 10:10:49 2002
@@ -37,29 +37,9 @@
 
 package java.nio.channels;
 
-import java.nio.channels.spi.*;
-import java.net.*;
-import java.nio.*;
-import java.io.*;
+import java.nio.channels.spi.AbstractSelectableChannel;
 
-public abstract class ServerSocketChannel
-    extends AbstractSelectableChannel
+public class ServerSocketChannel
+  extends AbstractSelectableChannel
 {
-    protected ServerSocketChannel(SelectorProvider provider)
-    {
-    }
- 
-
-    public abstract  SocketChannel accept();
-    public abstract  ServerSocket socket();
-    
-    public static ServerSocketChannel open() throws IOException
-    {
-	return SelectorProvider.provider().openServerSocketChannel();
-    }
-
-    public int validOps()
-    {
-	return SelectionKey.OP_ACCEPT;
-    } 
 }
