--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/event/InvocationEvent.java	2004-07-02 02:22:25.000000000 -0600
+++ java/awt/event/InvocationEvent.java	2002-08-12 10:11:05.000000000 -0600
@@ -189,13 +189,8 @@
         }
     else
       runnable.run();
-
-    Object o = notifier;
-    if (o != null)
-      synchronized(o)
-	{
-	  o.notifyAll();
-	}
+    if (notifier != null)
+      notifier.notifyAll();
   }
 
   /**
