--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ByteBuffer.java	Tue Mar 11 10:27:52 2003
+++ java/nio/ByteBuffer.java	Wed Mar 12 02:17:47 2003
@@ -46,8 +46,8 @@
 {
   private ByteOrder endian = ByteOrder.BIG_ENDIAN;
 
-  protected int offset;
-  protected byte[] backing_buffer;
+  int offset;
+  byte[] backing_buffer;
   
   /**
    * Allocates a new direct byte buffer.
@@ -84,7 +84,7 @@
     return wrap (array, 0, array.length);
   }
 
-  protected ByteBuffer (int capacity, int limit, int position, int mark)
+  ByteBuffer (int capacity, int limit, int position, int mark)
   {
     super (capacity, limit, position, mark);
   }
