--- /home/tromey/gnu/classpath/classpath/java/nio/charset/CodingErrorAction.java	Tue Apr 30 17:58:37 2002
+++ java/nio/charset/CodingErrorAction.java	Fri Nov  1 05:03:39 2002
@@ -40,4 +40,15 @@
 
 class CodingErrorAction
 {
+  public static final CodingErrorAction IGNORE;
+  public static final CodingErrorAction REPLACE;
+  public static final CodingErrorAction REPORT;
+
+  /**
+   * FIXME
+   */
+  public String toString ()
+  {
+    return "";
+  }
 }
