--- /home/tromey/gnu/egcs/classpath/classpath/java/security/PrivilegedExceptionAction.java	Wed Mar 22 12:26:21 2000
+++ java/security/PrivilegedExceptionAction.java	Wed Apr 25 09:45:12 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,34 +24,29 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 
 /**
-  * This interface defines a method that is called by 
-  * <code>AccessController.doPrivileged()</code> in order to perform a
-  * privileged operation with higher privileges enabled.  This interface
-  * differs from <code>PrivilegedAction</code> in that the <code>run</code>
-  * method in this interface may throw a checked exception.
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This interface defines a method that is called by 
+ * <code>AccessController.doPrivileged()</code> in order to perform a
+ * privileged operation with higher privileges enabled.  This interface
+ * differs from <code>PrivilegedAction</code> in that the <code>run</code>
+ * method in this interface may throw a checked exception.
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public interface PrivilegedExceptionAction
 {
-
-/**
-  * This method performs an operation that requires higher privileges to
-  * successfully complete.  It is called when a section of code invokes
-  * <code>AccessController.doPrivileged()</code>.
-  *
-  * @return obj An implementation defined return value.
-  *
-  * @exception Exception An implementation specific exception.
-  */
-public abstract Object
-run() throws Exception;
-
-} // interface PrivilegedExceptionAction
-
+  /**
+   * This method performs an operation that requires higher privileges to
+   * successfully complete.  It is called when a section of code invokes
+   * <code>AccessController.doPrivileged()</code>.
+   *
+   * @return obj An implementation defined return value.
+   *
+   * @exception Exception An implementation specific exception.
+   */
+  public abstract Object run() throws Exception;
+}
