--- /home/tromey/gnu/classpath/classpath/java/nio/charset/CharacterCodingException.java	Tue Apr 30 17:58:37 2002
+++ java/nio/charset/CharacterCodingException.java	Mon Oct  7 01:59:12 2002
@@ -37,16 +37,14 @@
 
 package java.nio.charset;
 
-import java.io.*;
+import java.io.IOException;
 
 class CharacterCodingException extends IOException
 {
-    CharacterCodingException()
-    {
-    }
-
-    CharacterCodingException(String s)
-    {
-	super(s);
-    }
+  /**
+   * Creates the exception
+   */
+  public CharacterCodingException()
+  {
+  }
 }
