--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/AbstractDocument.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/text/AbstractDocument.java	2004-01-11 02:18:05.000000000 -0700
@@ -41,9 +41,9 @@
 import java.util.Enumeration;
 import java.util.EventListener;
 import java.util.Vector;
-
 import javax.swing.event.DocumentEvent;
 import javax.swing.event.DocumentListener;
+import javax.swing.event.EventListenerList;
 import javax.swing.event.UndoableEditEvent;
 import javax.swing.event.UndoableEditListener;
 import javax.swing.tree.TreeNode;
