--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/PropertyPermissionCollection.java	2004-06-08 02:25:18.000000000 -0600
+++ java/util/PropertyPermissionCollection.java	2004-09-13 02:17:04.000000000 -0600
@@ -147,7 +147,7 @@
               return true;
           }
 
-        prefixLength = name.lastIndexOf('.', prefixLength - 1);
+        prefixLength = name.lastIndexOf('.', prefixLength);
         if (prefixLength < 0)
           return false;
         name = name.substring(0, prefixLength + 1) + '*';
