--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2003-09-03 02:18:53.000000000 -0600
+++ java/awt/Component.java	2003-08-26 02:18:05.000000000 -0600
@@ -1870,12 +1870,12 @@
   {
     Image returnValue = null;
     if (!GraphicsEnvironment.isHeadless ())
-      {
-	if (isLightweight () && parent != null)
-	  returnValue = parent.createImage (width, height);
-	else if (peer != null)
-	  returnValue = peer.createImage (width, height);
-      }
+    {
+      if (isLightweight () && parent != null)
+        returnValue = parent.createImage (width, height);
+      else if (peer != null)
+        returnValue = peer.createImage (width, height);
+    }
     return returnValue;
   }
 
