--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Arrays.java	2004-04-28 02:20:17.000000000 -0600
+++ java/util/Arrays.java	2003-09-11 02:18:05.000000000 -0600
@@ -2400,7 +2400,7 @@
     {
       int size = a.length;
       for (int i = 0; i < size; i++)
-        if (ArrayList.equals(o, a[i]))
+        if (this.equals(o, a[i]))
           return i;
       return -1;
     }
@@ -2409,7 +2409,7 @@
     {
       int i = a.length;
       while (--i >= 0)
-        if (ArrayList.equals(o, a[i]))
+        if (this.equals(o, a[i]))
           return i;
       return -1;
     }
