--- /home/tromey/gnu/egcs/classpath/classpath/java/security/Policy.java	Mon Mar 25 00:05:18 2002
+++ java/security/Policy.java	Tue Jan 22 16:01:36 2002
@@ -85,23 +85,9 @@
   // FIXME: The class name of the Policy provider should really be sourced 
   // from the "java.security" configuration file. For now, just hard-code 
   // a stub implementation.
-  static private Policy currentPolicy = null;
-  static 
-  {
-    String pp = System.getProperty ("policy.provider");
-    if (pp != null)
-      try
-	{
-	  currentPolicy = (Policy)Class.forName(pp).newInstance();
-	} 
-      catch (Exception _) 
-	{
-	  currentPolicy = null;
-	}
-    if (currentPolicy == null)
-      currentPolicy = new gnu.java.security.provider.DefaultPolicy();
-  }
-  
+  static private Policy currentPolicy 
+    = new gnu.java.security.provider.DefaultPolicy();
+
   /**
      Constructs a new Policy class.
    */
