--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/border/LineBorder.java	2003-08-17 02:19:16.000000000 -0600
+++ javax/swing/border/LineBorder.java	2003-08-02 02:18:05.000000000 -0600
@@ -129,7 +129,7 @@
    * it has rounded corners.
    * 
    * <p><img src="doc-files/LineBorder-1.png" width="500" height="200"
-   * alt="[An illustration of two LineBorders]" />
+   * alt="[An illustration of two LineBorders] />
    *
    * <p>Note that the enlarged view in the right-hand picture shows
    * that the implementation draws one more pixel than specified,
