--- /home/tromey/gnu/classpath/classpath/java/awt/BasicStroke.java	Mon Dec 30 15:39:03 2002
+++ java/awt/BasicStroke.java	Thu Aug  8 22:26:13 2002
@@ -59,31 +59,12 @@
   private final float[] dash;
   private final float phase;
 
-  /**
-   * Creates a basic stroke.
-   *
-   * @param width May not be negative .
-   * @param cap May be either CAP_BUTT, CAP_ROUND or CAP_SQUARE.
-   * @param join May be either JOIN_ROUND, JOIN_BEVEL, or JOIN_MITER.
-   * @param miterlimit the limit to trim the miter join. The miterlimit must be
-   * greater than or equal to 1.0f.
-   * @param dash The array representing the dashing pattern.
-   * @param dash_phase is negative and dash is not null.
-   *
-   * @exception IllegalArgumentException If one input parameter doesn't meet
-   * its needs.
-   */
   public BasicStroke(float width, int cap, int join, float miterlimit,
                      float[] dash, float dashPhase)
   {
-    if (width < 0 ||
-        miterlimit < 1.0f ||
-        cap < CAP_BUTT ||
-        cap > CAP_SQUARE ||
-        join < JOIN_MITER ||
-        join > JOIN_BEVEL)
+    if (width < 0 || miterlimit < 1 || cap < CAP_BUTT || cap > CAP_SQUARE
+        || join < JOIN_MITER || join > JOIN_BEVEL)
       throw new IllegalArgumentException();
-
     this.width = width;
     this.cap = cap;
     this.join = join;
@@ -92,54 +73,21 @@
     phase = dashPhase;
   }
 
-  /**
-   * Creates a basicc stroke.
-   *
-   * @param width The width of the BasicStroke. May not be negative .
-   * @param cap May be either CAP_BUTT, CAP_ROUND or CAP_SQUARE.
-   * @param join May be either JOIN_ROUND, JOIN_BEVEL, or JOIN_MITER.
-   * @param miterlimit the limit to trim the miter join. The miterlimit must be
-   * greater than or equal to 1.0f.
-   * 
-   * @exception IllegalArgumentException If one input parameter doesn't meet
-   * its needs.
-   */
   public BasicStroke(float width, int cap, int join, float miterlimit)
   {
     this(width, cap, join, miterlimit, null, 0);
   }
 
-  /**
-   * Creates a basicc stroke.
-   *
-   * @param width The width of the BasicStroke. May not be nehative.
-   * @param cap May be either CAP_BUTT, CAP_ROUND or CAP_SQUARE.
-   * @param join May be either JOIN_ROUND, JOIN_BEVEL, or JOIN_MITER.
-   * 
-   * @exception IllegalArgumentException If one input parameter doesn't meet
-   * its needs.
-   * @exception IllegalArgumentException FIXME
-   */
   public BasicStroke(float width, int cap, int join)
   {
     this(width, cap, join, 10, null, 0);
   }
 
-  /**
-   * Creates a basicc stroke.
-   *
-   * @param width The width of the BasicStroke.
-   * 
-   * @exception IllegalArgumentException If width is negative.
-   */
   public BasicStroke(float width)
   {
     this(width, CAP_SQUARE, JOIN_MITER, 10, null, 0);
   }
 
-  /**
-   * Creates a basicc stroke.
-   */
   public BasicStroke()
   {
     this(1, CAP_SQUARE, JOIN_MITER, 10, null, 0);
