--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/AttributedStringIterator.java	2004-04-28 02:20:17.000000000 -0600
+++ java/text/AttributedStringIterator.java	2004-04-17 02:18:07.000000000 -0600
@@ -1,5 +1,5 @@
 /* AttributedStringIterator.java -- Class to iterate over AttributedString
-   Copyright (C) 1998, 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,11 +38,11 @@
 
 package java.text;
 
-import java.util.HashMap;
+import java.util.Set;
 import java.util.HashSet;
-import java.util.Iterator;
 import java.util.Map;
-import java.util.Set;
+import java.util.HashMap;
+import java.util.Iterator;
 
 /**
   * This class implements the AttributedCharacterIterator interface.  It
