--- /home/tromey/gnu/classpath/classpath/java/awt/dnd/DropTargetContext.java	Wed Oct  2 11:21:37 2002
+++ java/awt/dnd/DropTargetContext.java	Thu Oct  3 08:40:43 2002
@@ -48,8 +48,6 @@
 
 public class DropTargetContext implements Serializable
 {
-  static final long serialVersionUID = -634158968993743371L;
-
   protected class TransferableProxy implements Transferable
   {
     protected boolean isLocal;
