--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/FileChannelImpl.java	2003-10-21 02:19:06.000000000 -0600
+++ java/nio/channels/FileChannelImpl.java	2003-08-03 02:17:56.000000000 -0600
@@ -47,8 +47,7 @@
 import java.nio.ByteBuffer;
 import java.nio.MappedByteBuffer;
 import java.nio.MappedByteBufferImpl;
-import gnu.classpath.Configuration;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
@@ -60,15 +59,6 @@
 
 public class FileChannelImpl extends FileChannel
 {
-  static
-  {
-    // load the shared library needed for native methods.
-    if (Configuration.INIT_LOAD_LIBRARY)
-      {
-        System.loadLibrary ("javanio");
-      }
-  }
-  
   public RawData map_address;
   
   int length;
