--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Container.java	2003-12-02 02:20:02.000000000 -0700
+++ java/awt/Container.java	2003-08-13 02:17:59.000000000 -0600
@@ -159,25 +159,6 @@
   }
 
   /**
-   * Swaps the components at position i and j, in the container.
-   */
-
-  protected void swapComponents (int i, int j)
-  {   
-    synchronized (getTreeLock ())
-      {
-        if (i < 0 
-            || i >= component.length
-            || j < 0 
-            || j >= component.length)
-          throw new ArrayIndexOutOfBoundsException ();
-        Component tmp = component[i];
-        component[i] = component[j];
-        component[j] = tmp;
-      }
-  }
-
-  /**
    * Returns the insets for this container, which is the space used for
    * borders, the margin, etc.
    *
@@ -1225,11 +1206,7 @@
                           Component comp)
   {
     Rectangle bounds = comp.getBounds();
-    Rectangle clip = gfx.getClipBounds();
-    if (clip == null)
-      clip = bounds;
-    else
-      clip = clip.intersection(bounds);
+    Rectangle clip = gfx.getClipBounds().intersection(bounds);
 
     if (clip.isEmpty()) return;
 
