--- /home/tromey/gnu/classpath/classpath/java/awt/MenuBar.java	Mon Dec 30 15:39:04 2002
+++ java/awt/MenuBar.java	Tue Jan 22 16:18:21 2002
@@ -86,14 +86,10 @@
 
 /**
   * Initializes a new instance of <code>MenuBar</code>.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 MenuBar()
 {
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
