--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2003-06-19 02:34:51.000000000 -0600
+++ javax/swing/UIManager.java	2003-06-13 02:27:43.000000000 -0600
@@ -174,21 +174,11 @@
 	System.out.println("UIManager.getDim");
 	return new Dimension(200,100);
     }
-
-
-  /**
-   * Retrieves a font from the defaults table of the current
-   * LookAndFeel.
-   *
-   * @param key an Object that specifies the font. Typically,
-   *        this is a String such as
-   *        <code>&quot;TitledBorder.font&quot;</code>.
-   */
-  public static Font getFont(Object key)
-  {
-    return (Font) getLookAndFeel().getDefaults().get(key);
-  }
-
+    static  Font getFont(Object key)
+    //      Returns a drawing font from the defaults table. 
+    {
+	return (Font) getLookAndFeel().getDefaults().get(key);
+    }
     static  Icon getIcon(Object key)
     //      Returns an Icon from the defaults table. 
     {
@@ -217,17 +207,11 @@
 	return look_and_feel;
     }
 
-
-  /**
-   * Returns the <code>UIDefaults</code> table of the currently active
-   * look and feel.
-   */
-  public static UIDefaults getLookAndFeelDefaults()
-  {
-    return getLookAndFeel().getDefaults();
-  }
-
-
+    static  UIDefaults getLookAndFeelDefaults()
+    //      Returns the default values for this look and feel. 
+    {
+	return getLookAndFeel().getDefaults();
+    }
     static  String getString(Object key)
     //      Returns a string from the defaults table. 
     {
