--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/regex/PatternSyntaxException.java	Mon May  6 17:10:38 2002
+++ java/util/regex/PatternSyntaxException.java	Tue Mar 25 02:17:56 2003
@@ -44,6 +44,7 @@
  */
 public class PatternSyntaxException extends IllegalArgumentException
 {
+  private static final long serialVersionUID = -3864639126226059218L;
 
   /**
    * Human readable escription of the syntax error.
