--- /home/tromey/gnu/classpath/classpath/java/nio/channels/FileChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/FileChannel.java	Mon Nov 18 06:56:59 2002
@@ -108,26 +108,12 @@
     return result;
   }
   
-  public abstract int write (ByteBuffer src) throws IOException;
-
   /**
    * Writes data to the channel.
    */
   public abstract long write(ByteBuffer[] srcs, int offset, int length)
     throws IOException;
   
-  public final long read (ByteBuffer[] dsts) throws IOException
-  {
-    long result = 0;
-    
-    for (int i = 0; i < dsts.length; i++)
-      {
-        read (dsts [i]);
-      }
-
-    return result;
-  }
-
   /**
    * Reads data from the channel.
    */
@@ -145,5 +131,5 @@
   /**
    * msync with the disk
    */
-  public abstract void force(boolean metaData) throws IOException;
+  public abstract void force(boolean metaData);    
 }
