--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/UID.java	2004-10-16 02:22:19.000000000 -0600
+++ java/rmi/server/UID.java	2004-08-31 02:19:53.000000000 -0600
@@ -1,5 +1,5 @@
-/* UID.java --
-   Copyright (c) 1996, 1997, 1998, 1999, 2004 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,13 +37,17 @@
 
 package java.rmi.server;
 
+import java.io.Serializable;
 import java.io.DataOutput;
 import java.io.DataInput;
 import java.io.IOException;
-import java.io.Serializable;
+import java.util.Random;
+import java.lang.Thread;
+import java.lang.InterruptedException;
+
+public final class UID
+	implements Serializable {
 
-public final class UID implements Serializable
-{
 public static final long serialVersionUID = 1086053664494604050L;
 
 private static final Object lock = UID.class;
