--- /home/tromey/gnu/egcs/classpath/classpath/java/util/TimeZone.java	Wed Sep 26 08:18:48 2001
+++ java/util/TimeZone.java	Tue Oct 16 09:16:19 2001
@@ -27,7 +27,6 @@
 
 package java.util;
 import java.text.DateFormatSymbols;
-import gnu.classpath.Configuration;
 
 /**
  * This class represents a time zone offset and handles daylight savings.
@@ -743,16 +742,9 @@
   /* Look up default timezone */
   static
   {
-    if (Configuration.INIT_LOAD_LIBRARY)
-      {
-	System.loadLibrary("javautil");
-      }
-    String tzid = System.getProperty("user.timezone");
+    // System.loadLibrary("javautil");
 
-    // XXX: Remove call to getDefaultTimeZoneId after java.lang.System
-    // is merged with libgcj.
-    if (tzid == null)
-      tzid = getDefaultTimeZoneId();
+    String tzid = System.getProperty("user.timezone");
 
     if (tzid == null)
       tzid = "GMT";
@@ -760,19 +752,6 @@
     defaultZone = getTimeZone(tzid);
   }
 
-  /* This method returns us a time zone id string which is in the
-     form <standard zone name><GMT offset><daylight time zone name>.
-     The GMT offset is in seconds, except where it is evenly divisible
-     by 3600, then it is in hours.  If the zone does not observe
-     daylight time, then the daylight zone name is omitted.  Examples:
-     in Chicago, the timezone would be CST6CDT.  In Indianapolis 
-     (which does not have Daylight Savings Time) the string would
-     be EST5
-   */
-  // XXX: Remove getDefaultTimeZoneId after java.lang.System
-  // is merged with libgcj.
-  private static native String getDefaultTimeZoneId();
-
   /**
    * Gets the time zone offset, for current date, modified in case of 
    * daylight savings.  This is the offset to add to UTC to get the local
@@ -959,6 +938,7 @@
    * @return The time zone for the identifier or GMT, if no such time
    * zone exists.
    */
+  // FIXME: XXX: JCL indicates this and other methods are synchronized.
   public static TimeZone getTimeZone(String ID)
   {
     // First check timezones hash
