--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Window.java	2003-07-25 02:18:53.000000000 -0600
+++ java/awt/Window.java	2003-07-26 02:18:06.000000000 -0600
@@ -38,6 +38,7 @@
 
 package java.awt;
 
+import gnu.java.awt.EmbeddedWindowSupport;
 import java.awt.event.WindowEvent;
 import java.awt.event.WindowFocusListener;
 import java.awt.event.WindowListener;
@@ -92,6 +93,19 @@
     graphicsConfiguration = gc;
   }
 
+  Window(int window_id, int width, int height)
+  {
+    this();
+
+    Toolkit tk = getToolkit();
+    if (!(tk instanceof EmbeddedWindowSupport))
+      throw new UnsupportedOperationException
+	("Embedded windows not supported by the current peers: " + tk.getClass());
+    
+    peer = ((EmbeddedWindowSupport) getToolkit())
+	    .createEmbeddedWindow (window_id, width, height);
+  }
+  
   /**
    * Initializes a new instance of <code>Window</code> with the specified
    * parent.  The window will initially be invisible.
