--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/MessageFormat.java	2003-03-29 02:18:40.000000000 -0700
+++ java/text/MessageFormat.java	2003-12-20 02:18:39.000000000 -0700
@@ -454,11 +454,25 @@
    * Creates a new MessageFormat object with
    * the specified pattern
    *
-   * @param aPattern The Pattern
+   * @param pattern The Pattern
    */
-  public MessageFormat (String pattern)
+  public MessageFormat(String pattern)
   {
-    locale = Locale.getDefault();
+    this(pattern, Locale.getDefault());
+  }
+
+  /**
+   * Creates a new MessageFormat object with
+   * the specified pattern
+   *
+   * @param pattern The Pattern
+   * @param locale The Locale to use
+   *
+   * @since 1.4
+   */
+  public MessageFormat(String pattern, Locale locale)
+  {
+    this.locale = locale;
     applyPattern (pattern);
   }
 
