--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/DefaultCaret.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/text/DefaultCaret.java	2004-01-11 02:18:05.000000000 -0700
@@ -41,12 +41,16 @@
 import java.awt.Graphics;
 import java.awt.Point;
 import java.awt.Rectangle;
+import java.awt.Shape;
+import java.awt.event.FocusEvent;
 import java.awt.event.FocusListener;
+import java.awt.event.MouseEvent;
 import java.awt.event.MouseListener;
 import java.awt.event.MouseMotionListener;
+import java.util.EventListener;
 import java.util.Vector;
-
 import javax.swing.event.ChangeListener;
+import javax.swing.event.EventListenerList;
 
 
 public class DefaultCaret extends Rectangle implements Caret, FocusListener, MouseListener, MouseMotionListener
