--- /home/tromey/gnu/Nightly/classpath/classpath/java/sql/Date.java	Thu Jun 20 23:34:12 2002
+++ java/sql/Date.java	Tue Apr  1 02:18:00 2003
@@ -86,9 +86,11 @@
    * date in JDBC format into a Java date.
    *
    * @param str The string to parse.
-   * @return The resulting <code>java.sql.Date</code> value. 
+   * @return The resulting <code>java.sql.Date</code> value.
+   *
+   * @deprecated
    */
-  public static Date valueOf(String str)
+  public static Date valueOf (String str)
   {
     try
       {
@@ -105,9 +107,11 @@
    * This method returns this date in JDBC format.
    *
    * @return This date as a string.
+   *
+   * @deprecated
    */
   public String toString()
   {
-    return(sdf.format(this));
+    return sdf.format(this);
   }
 }
