--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Locale.java	2003-06-19 02:34:49.000000000 -0600
+++ java/util/Locale.java	2003-06-18 02:22:32.000000000 -0600
@@ -231,9 +231,9 @@
     // default locale.
     if (defaultLocale != null)
       {
-        language = convertLanguage(language).intern();
-        country = country.toUpperCase().intern();
-        variant = variant.toUpperCase().intern();
+        language = convertLanguage(language);
+        country = country.toUpperCase();
+        variant = variant.toUpperCase();
       }
     this.language = language;
     this.country = country;
@@ -436,7 +436,7 @@
    */
   public String getISO3Language()
   {
-    if (language == "")
+    if ("".equals(language))
       return "";
     int index
       = ("aa,ab,af,am,ar,as,ay,az,ba,be,bg,bh,bi,bn,bo,br,ca,co,cs,cy,da,"
@@ -472,7 +472,7 @@
    */
   public String getISO3Country()
   {
-    if (country == "")
+    if ("".equals(country))
       return "";
     int index
       = ("AD,AE,AF,AG,AI,AL,AM,AN,AO,AQ,AR,AS,AT,AU,AW,AZ,BA,BB,BD,BE,BF,"
@@ -729,13 +729,9 @@
       return false;
     Locale l = (Locale) obj;
 
-    // ??? We might also want to add:
-    //        hashCode() == l.hashCode()
-    // But this is a synchronized method.  Is the overhead worth it?
-    // Measure this to make a decision.
-    return (language == l.language
-            && country == l.country
-            && variant == l.variant);
+    return (language.equals(l.language)
+            && country.equals(l.country)
+            && variant.equals(l.variant));
   }
 
   /**
