--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectInputStream.java	2003-06-28 02:19:16.000000000 -0600
+++ java/io/ObjectInputStream.java	2003-06-28 02:18:20.000000000 -0600
@@ -551,6 +551,9 @@
     if (sm == null)
       sm = new SecurityManager () {};
 
+    // FIXME: currentClassLoader doesn't yet do anything useful. We need
+    // to call forName() with the classloader of the class which called 
+    // readObject(). See SecurityManager.getClassContext().
     ClassLoader cl = currentClassLoader (sm);
 
     if (cl == null)
@@ -707,142 +710,52 @@
 
   public boolean readBoolean () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 1)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    boolean value = this.dataInputStream.readBoolean ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readBoolean ();
   }
 
   public byte readByte () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 1)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    byte value = this.dataInputStream.readByte ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readByte ();
   }
 
   public int readUnsignedByte () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 1)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    int value = this.dataInputStream.readUnsignedByte ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readUnsignedByte ();
   }
 
   public short readShort () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 2)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    short value = this.dataInputStream.readShort ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readShort ();
   }
 
   public int readUnsignedShort () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 2)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    int value = this.dataInputStream.readUnsignedShort ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readUnsignedShort ();
   }
 
   public char readChar () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 2)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    char value = this.dataInputStream.readChar ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readChar ();
   }
 
   public int readInt () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 4)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    int value = this.dataInputStream.readInt ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readInt ();
   }
 
   public long readLong () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 8)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    long value = this.dataInputStream.readLong ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readLong ();
   }
 
   public float readFloat () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 4)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    float value = this.dataInputStream.readFloat ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readFloat ();
   }
 
   public double readDouble () throws IOException
   {
-    boolean switchmode = true;
-    boolean oldmode = this.readDataFromBlock;
-    if (!oldmode || this.blockDataBytes - this.blockDataPosition >= 8)
-      switchmode = false;
-    if (switchmode)
-      oldmode = setBlockDataMode (true);
-    double value = this.dataInputStream.readDouble ();
-    if (switchmode)
-      setBlockDataMode (oldmode);
-    return value;
+    return this.dataInputStream.readDouble ();
   }
 
   public void readFully (byte data[]) throws IOException
@@ -1494,7 +1407,11 @@
 
   // this native method is used to get access to the protected method
   // of the same name in SecurityManger
-  private static native ClassLoader currentClassLoader (SecurityManager sm);
+  private static ClassLoader currentClassLoader (SecurityManager sm)
+  {
+    // FIXME: This is too simple.
+    return ClassLoader.getSystemClassLoader ();
+  }
 
   private static Field getField (Class klass, String name)
     throws java.lang.NoSuchFieldException
