--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/CharViewBufferImpl.java	2004-06-21 06:23:59.000000000 -0600
+++ java/nio/CharViewBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -66,13 +66,6 @@
     this.endian = endian;
   }
 
-  /**
-   * Reads the <code>char</code> at this buffer's current position,
-   * and then increments the position.
-   *
-   * @exception BufferUnderflowException If there are no remaining
-   * <code>char</code>s in this buffer.
-   */
   public char get ()
   {
     int p = position();
@@ -81,15 +74,6 @@
     return result;
   }
 
-  /**
-   * Absolute get method. Reads the <code>char</code> at position
-   * <code>index</code>.
-   *
-   * @param index Position to read the <code>char</code> from.
-   *
-   * @exception IndexOutOfBoundsException If index is negative or not smaller
-   * than the buffer's limit.
-   */
   public char get (int index)
   {
     return ByteBufferHelper.getChar(bb, (index << 1) + offset, endian);
