--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JFileChooser.java	2004-02-06 02:21:26.000000000 -0700
+++ javax/swing/JFileChooser.java	2004-06-18 02:18:51.000000000 -0600
@@ -61,55 +61,30 @@
 
   private static final long serialVersionUID = 3162921138695327837L;
 
-	//-------------------------------------------------------------
-	// Classes ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * AccessibleJFileChooser
-	 */
-	protected class AccessibleJFileChooser extends AccessibleJComponent {
-
-          private static final long serialVersionUID = 3318922050345221200L;
-
-		//-------------------------------------------------------------
-		// Variables --------------------------------------------------
-		//-------------------------------------------------------------
-
-
-		//-------------------------------------------------------------
-		// Initialization ---------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * Constructor AccessibleJFileChooser
-		 * @param component TODO
-		 */
-		protected AccessibleJFileChooser(JFileChooser component) {
-			super(component);
-			// TODO
-		} // AccessibleJFileChooser()
-
-
-		//-------------------------------------------------------------
-		// Methods ----------------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * getAccessibleRole
-		 * @returns AccessibleRole
-		 */
-		public AccessibleRole getAccessibleRole() {
-			return AccessibleRole.FILE_CHOOSER;
-		} // getAccessibleRole()
-
-
-	} // AccessibleJFileChooser
-
-
-	//-------------------------------------------------------------
-	// Variables --------------------------------------------------
-	//-------------------------------------------------------------
+  /**
+   * AccessibleJFileChooser
+   */
+  protected class AccessibleJFileChooser extends AccessibleJComponent
+  {
+    private static final long serialVersionUID = 8205148454060169244L;
+
+    /**
+     * Constructor AccessibleJFileChooser
+     * @param component TODO
+     */
+    protected AccessibleJFileChooser()
+    {
+    }
+
+    /**
+     * getAccessibleRole
+     * @return AccessibleRole
+     */
+    public AccessibleRole getAccessibleRole()
+    {
+      return AccessibleRole.FILE_CHOOSER;
+    }
+  }
 
 	/**
 	 * uiClassID
@@ -969,16 +944,15 @@
 		return null; // TODO
 	} // paramString()
 
-	/**
-	 * getAccessibleContext
-	 * @returns AccessibleContext
-	 */
-	public AccessibleContext getAccessibleContext() {
-		if (accessibleContext == null) {
-			accessibleContext = new AccessibleJFileChooser(this);
-		} // if
-		return accessibleContext;
-	} // getAccessibleContext()
-
-
-} // JFileChooser
+  /**
+   * getAccessibleContext
+   * @returns AccessibleContext
+   */
+  public AccessibleContext getAccessibleContext()
+  {
+    if (accessibleContext == null)
+      accessibleContext = new AccessibleJFileChooser();
+
+    return accessibleContext;
+  }
+}
