--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/event/KeyEvent.java	2003-08-07 02:18:44.000000000 -0600
+++ java/awt/event/KeyEvent.java	2003-08-06 02:18:11.000000000 -0600
@@ -1652,6 +1652,7 @@
   public String paramString()
   {
     StringBuffer s = new StringBuffer();
+
     switch (id)
       {
       case KEY_PRESSED:
@@ -1665,7 +1666,7 @@
         break;
       default:
         s.append("unknown type");
-          }
+      }
 
     s.append(",keyCode=").append(keyCode);
 
@@ -1698,6 +1699,7 @@
                                                         & CONVERT_MASK));
     if (modifiers != 0)
       s.append(",extModifiers=").append(getModifiersExText(modifiers));
+
     s.append(",keyLocation=KEY_LOCATION_");
     switch (keyLocation)
       {
@@ -1716,6 +1718,7 @@
       case KEY_LOCATION_NUMPAD:
         s.append("NUMPAD");
       }
+
     return s.toString();
   }
 
