--- /home/tromey/gnu/egcs/classpath/classpath/java/security/DummyMessageDigest.java	Wed Mar 22 12:26:19 2000
+++ java/security/DummyMessageDigest.java	Wed Apr 25 09:45:12 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -34,29 +34,29 @@
 {
   private MessageDigestSpi mdSpi = null;
 
-  public DummyMessageDigest (MessageDigestSpi mdSpi, String algorithm)
+  public DummyMessageDigest(MessageDigestSpi mdSpi, String algorithm)
   {
-    super (algorithm);
+    super(algorithm);
     this.mdSpi = mdSpi;
   }
 
-  protected void engineUpdate (byte input)
+  protected void engineUpdate(byte input)
   {
-    mdSpi.engineUpdate (input);
+    mdSpi.engineUpdate(input);
   }
-  
-  protected void engineUpdate (byte[] input, int offset, int len)
+
+  protected void engineUpdate(byte[]input, int offset, int len)
   {
-    mdSpi.engineUpdate (input, offset, len);
+    mdSpi.engineUpdate(input, offset, len);
   }
-  
+
   protected byte[] engineDigest()
   {
-    return mdSpi.engineDigest ();
+    return mdSpi.engineDigest();
   }
 
-  protected void engineReset ()
+  protected void engineReset()
   {
-    mdSpi.engineReset ();
+    mdSpi.engineReset();
   }
 }
