--- /home/tromey/gnu/classpath/classpath/java/nio/charset/CharsetEncoder.java	Sat Nov  9 16:09:33 2002
+++ java/nio/charset/CharsetEncoder.java	Sat Feb 15 18:59:11 2003
@@ -319,6 +319,11 @@
     return this;
   }
 
+  public CodingErrorAction unmappableCharacterAction ()
+  {
+    return unmappableCharacterAction;
+  }
+
   public final CharsetEncoder onUnmappableCharacter
     (CodingErrorAction newAction)
   {
