--- /home/tromey/gnu/egcs/classpath/classpath/java/security/InvalidParameterException.java	Sat Mar  2 12:13:17 2002
+++ java/security/InvalidParameterException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,4 @@
-/* InvalidParameterException.java -- an invalid parameter in the JCA/JCE engine
-   Copyright (C) 2000, 2002 Free Software Foundation
+/* Copyright (C) 2000  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -38,31 +37,22 @@
 package java.security;
 
 /**
- * Thrown when an invalid parameter is passed to a method of the JCA/JCE
- * engine classes.
- *
  * @author Warren Levy <warrenl@cygnus.com>
- * @status updated to 1.4
+ * @date February 1, 2000.
  */
+
+/**
+ * Written using on-line Java Platform 1.2 API Specification.
+ * Status:  Believed complete and correct.
+ */
+
 public class InvalidParameterException extends IllegalArgumentException
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -857968536935667808L;
-
-  /**
-   * Construct an exception with no message.
-   */
   public InvalidParameterException()
   {
+    super();
   }
 
-  /**
-   * Construct an exception with a message.
-   *
-   * @param msg the message
-   */
   public InvalidParameterException(String msg)
   {
     super(msg);
