--- /home/tromey/gnu/classpath/classpath/java/sql/DatabaseMetaData.java	Mon May  6 10:40:03 2002
+++ java/sql/DatabaseMetaData.java	Tue Jan 22 16:01:37 2002
@@ -500,7 +500,7 @@
   * This method tests whether or not the database treats mixed case
   * identifiers as all upper case.
   *
-  * @return <code>true</code> if the database treats all identifiers as
+  * @exception <code>true</code> if the database treats all identifiers as
   * upper case, <code>false</code> otherwise.
   *
   * @exception SQLException If an error occurs.
@@ -514,7 +514,7 @@
   * This method tests whether or not the database treats mixed case
   * identifiers as all lower case.
   *
-  * @return <code>true</code> if the database treats all identifiers as
+  * @exception <code>true</code> if the database treats all identifiers as
   * lower case, <code>false</code> otherwise.
   *
   * @exception SQLException If an error occurs.
@@ -556,7 +556,7 @@
   * This method tests whether or not the database treats mixed case
   * quoted identifiers as all upper case.
   *
-  * @return <code>true</code> if the database treats all quoted identifiers 
+  * @exception <code>true</code> if the database treats all quoted identifiers 
   * as upper case, <code>false</code> otherwise.
   *
   * @exception SQLException If an error occurs.
@@ -570,7 +570,7 @@
   * This method tests whether or not the database treats mixed case
   * quoted identifiers as all lower case.
   *
-  * @return <code>true</code> if the database treats all quoted identifiers 
+  * @exception <code>true</code> if the database treats all quoted identifiers 
   * as lower case, <code>false</code> otherwise.
   *
   * @exception SQLException If an error occurs.
