--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/KeyPairGeneratorSpi.java	2004-11-17 02:24:08.000000000 -0700
+++ java/security/KeyPairGeneratorSpi.java	2002-12-10 19:20:35.000000000 -0700
@@ -92,11 +92,18 @@
   public abstract KeyPair generateKeyPair();
 
   /**
-   * We override clone here to make it accessible for use by
-   * DummyKeyPairGenerator.
+     Returns a clone of this class.
+
+     If cloning is not supported, then by default the class throws a
+     CloneNotSupportedException.  The MessageDigestSpi provider
+     implementation has to overload this class in order to be
+     cloneable.
    */
-  protected Object clone() throws CloneNotSupportedException
+  public Object clone() throws CloneNotSupportedException
   {
-    return super.clone();
+    if (this instanceof Cloneable)
+      return super.clone();
+    else
+      throw new CloneNotSupportedException();
   }
 }
