--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/FileChannel.java	2003-07-14 02:19:03.000000000 -0600
+++ java/nio/channels/FileChannel.java	2004-02-27 02:18:22.000000000 -0700
@@ -53,9 +53,9 @@
   {
     int m;
 
-    public static MapMode READ_ONLY  = new MapMode(0);
-    public static MapMode READ_WRITE = new MapMode(1);
-    public static MapMode PRIVATE    = new MapMode(2);
+    public static final MapMode READ_ONLY  = new MapMode(0);
+    public static final MapMode READ_WRITE = new MapMode(1);
+    public static final MapMode PRIVATE    = new MapMode(2);
 
     /**
      * Initializes the MapMode.
