--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Long.java	Mon Jul 23 14:15:25 2001
+++ java/lang/Long.java	Fri Oct 26 16:18:42 2001
@@ -60,7 +60,7 @@
    * The primitive type <code>long</code> is represented by this 
    * <code>Class</code> object.
    */
-  public static final Class TYPE = VMClassLoader.getPrimitiveClass ("long");
+  public static final Class TYPE = VMClassLoader.getPrimitiveClass ('J');
 
   /**
    * The immutable value of this Long.
