--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileInputStream.java	2004-10-03 02:35:36.000000000 -0600
+++ java/io/FileInputStream.java	2004-03-01 02:18:07.000000000 -0700
@@ -38,9 +38,8 @@
 
 package java.io;
 
-import gnu.java.nio.channels.FileChannelImpl;
-
 import java.nio.channels.FileChannel;
+import gnu.java.nio.channels.FileChannelImpl;
 
 /* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
  * "The Java Language Specification", ISBN 0-201-63451-1
