--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JButton.java	2004-07-09 02:22:37.000000000 -0600
+++ javax/swing/JButton.java	2004-06-18 02:18:51.000000000 -0600
@@ -98,8 +98,7 @@
   
     public String getUIClassID()
     {
-	//Returns a string that specifies the name of the Look and Feel
-	//class that renders this component.  
+	//Returns a string that specifies the name of the L&F class that renders this component.  
 	return "ButtonUI";
     }
   
