--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicIconFactory.java	2004-04-18 02:22:32.000000000 -0600
+++ javax/swing/plaf/basic/BasicIconFactory.java	2004-05-27 02:19:07.000000000 -0600
@@ -82,7 +82,34 @@
   }
   public static Icon getMenuArrowIcon()
   {
-    return new DummyIcon();
+    return new Icon()
+      {
+	public int getIconHeight()
+	{
+	  return 12;
+	}
+
+	public int getIconWidth()
+	{
+	  return 12;
+	}
+
+	public void paintIcon(Component c, Graphics g, int x, int y)
+	{
+	  g.translate(x, y);
+
+	  Color saved = g.getColor();
+
+	  g.setColor(Color.BLACK);
+
+	  g.fillPolygon(new Polygon(new int[] { 3, 9, 3 },
+                                  new int[] { 2, 6, 10 },
+                                  3));
+
+	  g.setColor(saved);
+	  g.translate(-x, -y);
+	}
+      };
   }
 
   public static Icon getCheckBoxIcon()
