--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicTabbedPaneUI.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/plaf/basic/BasicTabbedPaneUI.java	2004-05-27 02:19:08.000000000 -0600
@@ -223,10 +223,11 @@
         {
 	  Component visible = getVisibleComponent();
 	  Insets insets = getContentBorderInsets(tabPane.getTabPlacement());
-	  visible.setBounds(contentRect.x + insets.left,
-	                    contentRect.y + insets.top,
-	                    contentRect.width - insets.left - insets.right,
-	                    contentRect.height - insets.top - insets.bottom);
+	  if (visible != null)
+	    visible.setBounds(contentRect.x + insets.left,
+	                      contentRect.y + insets.top,
+	                      contentRect.width - insets.left - insets.right,
+	                      contentRect.height - insets.top - insets.bottom);
         }
     }
 
@@ -1510,6 +1511,7 @@
     tabPane.setFont(defaults.getFont("TabbedPane.font"));
     tabPane.setForeground(defaults.getColor("TabbedPane.foreground"));
     tabPane.setBackground(defaults.getColor("TabbedPane.background"));
+    tabPane.setOpaque(true);
 
     highlight = defaults.getColor("TabbedPane.highlight");
     lightHighlight = defaults.getColor("TabbedPane.lightHighlight");
@@ -1790,7 +1792,7 @@
     // FIXME: Paint little folding corner and jagged edge clipped tab.
     if (icon != null)
       paintIcon(g, tabPlacement, tabIndex, icon, iconRect, isSelected);
-    if (title == null || ! title.equals(""))
+    if (title != null && ! title.equals(""))
       paintText(g, tabPlacement, tabPane.getFont(), fm, tabIndex, title,
                 textRect, isSelected);
   }
