--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RemoteStub.java	Sun Mar 23 02:18:41 2003
+++ java/rmi/server/RemoteStub.java	Mon Mar 31 02:17:49 2003
@@ -37,8 +37,7 @@
 
 package java.rmi.server;
 
-public abstract class RemoteStub
-  extends RemoteObject
+public abstract class RemoteStub extends RemoteObject
 {
   static final long serialVersionUID = -1585587260594494182l;
 
@@ -52,6 +51,9 @@
     super (ref);
   }
 
+  /**
+   * @deprecated
+   */
   protected static void setRef (RemoteStub stub,  RemoteRef ref)
   {
     stub.ref = ref;
