--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2003-09-19 02:19:01.000000000 -0600
+++ java/net/URLStreamHandler.java	2003-11-12 02:18:14.000000000 -0700
@@ -478,15 +478,13 @@
       }
 
     if (host.length() != 0)
-      sb.append("//").append(host);
+      {
+        sb.append("//").append(host);
 
-    // Note that this produces different results from JDK 1.2 as JDK 1.2
-    // ignores a non-default port if host is null or "".  That is inconsistent
-    // with the spec since the result of this method is spec'ed so it can be
-    // used to construct a new URL that is equivalent to the original.
-    boolean port_needed = port > 0 && port != getDefaultPort();
-    if (port_needed)
-      sb.append(':').append(port);
+        // Append port if port was in URL spec.
+        if (port >= 0)
+          sb.append(':').append(port);
+      }
 
     sb.append(file);
 
