--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ShortBuffer.java	Tue Mar 11 10:27:52 2003
+++ java/nio/ShortBuffer.java	Thu Feb 20 02:17:58 2003
@@ -77,7 +77,7 @@
     return wrap(array, 0, array.length);
   }
 
-  protected ShortBuffer (int capacity, int limit, int position, int mark)
+  ShortBuffer (int capacity, int limit, int position, int mark)
   {
     super (capacity, limit, position, mark);
     array_offset = 0;
