--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/NamingException.java	2004-04-19 02:20:15.000000000 -0600
+++ javax/naming/NamingException.java	2003-06-28 02:18:23.000000000 -0600
@@ -93,7 +93,7 @@
   /**
    * Creates a new NamingException without a message. Does not set any of the
    * <code>rootException</code>, <code>resolvedName</code>,
-   * <code>resolvedObj</code> or <code>remainingObject</code> fields.
+   * <code>resolvedObj</code> or <code>remainingObject,<code> fields.
    * These fields can be set later.
    */
   public NamingException ()
@@ -104,7 +104,7 @@
   /**
    * Creates a new NamingException with a detailed message. Does not set
    * the <code>rootException</code>, <code>resolvedName</code>,
-   * <code>resolvedObj</code> or <code>remainingObject,</code> fields.
+   * <code>resolvedObj</code> or <code>remainingObject,<code> fields.
    * These fields can be set later.
    */
   public NamingException (String msg)
