--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/CellRendererPane.java	2004-02-06 02:21:26.000000000 -0700
+++ javax/swing/CellRendererPane.java	2004-06-18 02:18:51.000000000 -0600
@@ -56,50 +56,30 @@
 {
   private static final long serialVersionUID = -7642183829532984273L;
 
-	//-------------------------------------------------------------
-	// Classes ----------------------------------------------------
-	//-------------------------------------------------------------
-
-	/**
-	 * AccessibleCellRendererPane
-	 */
-	protected class AccessibleCellRendererPane extends AccessibleAWTContainer {
-
-          private static final long serialVersionUID = 7300340301783504481L;
-
-		//-------------------------------------------------------------
-		// Initialization ---------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * Constructor AccessibleCellRendererPane
-		 * @param component TODO
-		 */
-		protected AccessibleCellRendererPane(CellRendererPane component) {
-			super();
-			// TODO
-		} // AccessibleCellRendererPane()
-
-
-		//-------------------------------------------------------------
-		// Methods ----------------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * getAccessibleRole
-		 * @returns AccessibleRole
-		 */
-		public AccessibleRole getAccessibleRole() {
-			return AccessibleRole.PANEL;
-		} // getAccessibleRole()
-
-
-	} // AccessibleCellRendererPane
-
-
-	//-------------------------------------------------------------
-	// Variables --------------------------------------------------
-	//-------------------------------------------------------------
+  /**
+   * AccessibleCellRendererPane
+   */
+  protected class AccessibleCellRendererPane extends AccessibleAWTContainer
+  {
+    private static final long serialVersionUID = -8981090083147391074L;
+
+    /**
+     * Constructor AccessibleCellRendererPane
+     * @param component TODO
+     */
+    protected AccessibleCellRendererPane()
+    {
+    }
+
+    /**
+     * getAccessibleRole
+     * @returns AccessibleRole
+     */
+    public AccessibleRole getAccessibleRole()
+    {
+      return AccessibleRole.PANEL;
+    }
+  }
 
 	/**
 	 * accessibleContext
@@ -209,16 +189,15 @@
 		// TODO
 	} // paintComponent()
 
-	/**
-	 * getAccessibleContext
-	 * @returns AccessibleContext
-	 */
-	public AccessibleContext getAccessibleContext() {
-		if (accessibleContext == null) {
-			accessibleContext = new AccessibleCellRendererPane(this);
-		} // if
-		return accessibleContext;
-	} // getAccessibleContext()
-
-
-} // CellRendererPane
+  /**
+   * getAccessibleContext
+   * @return AccessibleContext
+   */
+  public AccessibleContext getAccessibleContext()
+  {
+    if (accessibleContext == null)
+      accessibleContext = new AccessibleCellRendererPane();
+
+    return accessibleContext;
+  }
+}
