--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ViewportLayout.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/ViewportLayout.java	2004-03-18 02:18:23.000000000 -0700
@@ -67,7 +67,7 @@
           return null;
 	}
 	public void layoutContainer(Container parent) {
-          if (parent.getComponentCount() == 1)
+          if (parent.countComponents() == 1)
             {
               // This should usually be true, but if it's not it is
               // probably nicer if we do not panic.
