--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormatSymbols.java	2003-11-16 02:19:40.000000000 -0700
+++ java/text/DecimalFormatSymbols.java	2002-01-22 16:01:37.000000000 -0700
@@ -39,7 +39,6 @@
 package java.text;
 
 import java.io.Serializable;
-import java.util.Currency;
 import java.util.Locale;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
@@ -196,18 +195,6 @@
   }
 
   /**
-   * Returns the currency corresponding to the currency symbol stored
-   * in the instance of <code>DecimalFormatSymbols</code>.
-   *
-   * @return A new instance of <code>Currency</code> if
-   * the currency code matches a known one.
-   */
-  public Currency getCurrency ()
-  {
-    return Currency.getInstance (currencySymbol);
-  }
-
-  /**
    * This method returns the currency symbol in local format.  For example,
    * "$" for Canadian dollars.
    *
@@ -367,16 +354,6 @@
   }
 
   /**
-   * This method sets the currency to the specified value.
-   *
-   * @param currency The new currency
-   */
-  public void setCurrency (Currency currency)
-  {
-    setCurrencySymbol (currency.getSymbol());
-  }
-
-  /**
    * This method sets the currency symbol to the specified value.
    *
    * @param currencySymbol The new currency symbol
