--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/spec/DSAParameterSpec.java	2004-10-28 02:27:36.000000000 -0600
+++ java/security/spec/DSAParameterSpec.java	2004-04-21 02:18:11.000000000 -0600
@@ -37,17 +37,16 @@
 
 
 package java.security.spec;
-
-import java.math.BigInteger;
 import java.security.interfaces.DSAParams;
+import java.math.BigInteger;
 
 /**
- * DSA Parameter class Specification. Used to maintain the DSA
- * Parameters.
- *
- * @since 1.2
- *
- * @author Mark Benvenuto
+	DSA Parameter class Specification. Used to maintain the DSA
+	Parameters.
+
+	@since JDK 1.2
+
+	@author Mark Benvenuto
 */
 public class DSAParameterSpec implements AlgorithmParameterSpec, DSAParams
 {
@@ -56,46 +55,46 @@
   private BigInteger g = null;
 
   /**
-   * Constructs a new DSAParameterSpec with the specified p, q, and g.
-   *
-   * @param p the prime
-   * @param q the sub-prime
-   * @param g the base
-   */
+     Constructs a new DSAParameterSpec with the specified p, q, and g.
+
+     @param p the prime
+     @param q the sub-prime
+     @param g the base
+  */
   public DSAParameterSpec(BigInteger p, BigInteger q, BigInteger g) 
   {
     this.p = p;
     this.q = q;
     this.g = g;
   }
-  
   /**
-   * Returns p for the DSA algorithm.
-   *
-   * @return Returns the requested BigInteger
-   */
+     Returns p for the DSA algorithm.
+
+     @return Returns the requested BigInteger
+  */
   public BigInteger getP() 
   {
     return this.p;
   }
 
   /**
-   * Returns p for the DSA algorithm.
-   *
-   * @return Returns the requested BigInteger
-   */
+     Returns p for the DSA algorithm.
+
+     @return Returns the requested BigInteger
+  */
   public BigInteger getQ() 
   {
     return this.q;
   }
 
   /**
-   * Returns g for the DSA algorithm.
-   *
-   * @return Returns the requested BigInteger
-   */
+     Returns g for the DSA algorithm.
+
+     @return Returns the requested BigInteger
+  */
   public BigInteger getG()
   {
     return this.g;
   }
-}
\ No newline at end of file
+
+}
