--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/SwingUtilities.java	2004-01-28 02:27:14.000000000 -0700
+++ javax/swing/SwingUtilities.java	2003-07-14 02:18:15.000000000 -0600
@@ -97,8 +97,8 @@
     // view rect 'vr' already ok, 
     // we need to compute ir (icon rect) and tr (text-rect)
 	
-    int next_x = vr.x;
-    int next_y = vr.y;
+    int next_x = 0;//vr.x;
+    int next_y = 0;//vr.y;
 	
     ir.height = ir.width = ir.y = ir.x = 0;
 
@@ -115,10 +115,10 @@
       }
 	
     tr.x = next_x;
-    tr.y = vr.y; // + (vr.height/2);
+    tr.y = vr.y + (vr.height/2);
 
     tr.width  = fm.stringWidth(text);
-    tr.height = fm.getHeight(); // +  fm.getAscent()/2;
+    tr.height = fm.getHeight() +  fm.getAscent()/2;
 
     return text;
   }
