--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectOutput.java	Mon Apr  7 02:24:27 2003
+++ java/io/ObjectOutput.java	Mon Mar 24 02:17:54 2003
@@ -45,11 +45,10 @@
   * <code>OutputStream</code> like.
   *
   * @author Aaron M. Renn (arenn@urbanophile.com)
-  *
-  * @see DataOutput
   */
 public interface ObjectOutput extends DataOutput
 {
+
   /**
     * This method writes the specified byte to the output stream.
     *
