--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/HashMap.java	2003-11-08 02:19:25.000000000 -0700
+++ java/util/HashMap.java	2003-12-08 02:18:24.000000000 -0700
@@ -223,7 +223,7 @@
   public HashMap(Map m)
   {
     this(Math.max(m.size() * 2, DEFAULT_CAPACITY), DEFAULT_LOAD_FACTOR);
-    putAllInternal(m);
+    putAll(m);
   }
 
   /**
@@ -699,9 +699,9 @@
   }
 
   /**
-   * A simplified, more efficient internal implementation of putAll(). The 
-   * Map constructor and clone() should not call putAll or put, in order to 
-   * be compatible with the JDK implementation with respect to subclasses.
+   * A simplified, more efficient internal implementation of putAll(). clone() 
+   * should not call putAll or put, in order to be compatible with the JDK 
+   * implementation with respect to subclasses.
    *
    * @param m the map to initialize this from
    */
