--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/FileChannelImpl.java	2004-01-03 02:20:34.000000000 -0700
+++ java/nio/channels/FileChannelImpl.java	2003-12-20 02:18:39.000000000 -0700
@@ -48,7 +48,7 @@
 import java.nio.MappedByteBuffer;
 import java.nio.MappedByteBufferImpl;
 import gnu.classpath.Configuration;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
