--- /home/tromey/gnu/classpath/classpath/java/awt/CheckboxMenuItem.java	Mon Dec 30 15:39:03 2002
+++ java/awt/CheckboxMenuItem.java	Sun Mar 24 16:08:55 2002
@@ -84,9 +84,6 @@
 /**
   * Initializes a new instance of <code>CheckboxMenuItem</code> with no
   * label and an initial state of off.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true.
   */
 public
 CheckboxMenuItem()
@@ -101,9 +98,6 @@
   * specified label and an initial state of off.
   *
   * @param label The label of the menu item.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true.
   */
 public
 CheckboxMenuItem(String label)
@@ -120,18 +114,12 @@
   * @param label The label of the menu item.
   * @param state The initial state of the menu item, where <code>true</code>
   * is on, and <code>false</code> is off.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true.
   */
 public
 CheckboxMenuItem(String label, boolean state)
 {
   super(label);
   this.state = state;
-
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
