--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/geom/GeneralPath.java	2003-10-21 02:19:01.000000000 -0600
+++ java/awt/geom/GeneralPath.java	2003-01-14 14:32:16.000000000 -0700
@@ -197,7 +197,7 @@
   {
     if (subpath < 0)
       return null;
-    return new Point2D.Float(points[index - 2], points[index - 1]);
+    return new Point2D.Float(points[subpath], points[subpath + 1]);
   }
   public void reset()
   {
