--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/directory/DirContext.java	2003-10-13 02:22:19.000000000 -0600
+++ javax/naming/directory/DirContext.java	2003-12-06 02:18:23.000000000 -0700
@@ -38,7 +38,10 @@
 
 package javax.naming.directory;
 
-import javax.naming.*;
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
 
 /**
  * @author Warren Levy <warrenl@redhat.com>
