--- /home/tromey/gnu/classpath/classpath/java/nio/channels/spi/AbstractInterruptibleChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/spi/AbstractInterruptibleChannel.java	Wed Nov 13 06:52:47 2002
@@ -38,7 +38,6 @@
 package java.nio.channels.spi;
 
 import java.io.IOException;
-import java.nio.channels.AsynchronousCloseException;
 import java.nio.channels.Channel;
 import java.nio.channels.InterruptibleChannel;
 
@@ -85,7 +84,6 @@
    * I/O operation was interrupted.
    */
   protected final void end (boolean completed)
-    throws AsynchronousCloseException
   {
   }   
 
