--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicButtonUI.java	2003-06-19 02:34:52.000000000 -0600
+++ javax/swing/plaf/basic/BasicButtonUI.java	2003-05-10 02:19:19.000000000 -0600
@@ -74,7 +74,14 @@
     public Dimension getPreferredSize(JComponent c) 
     {
 	AbstractButton b = (AbstractButton)c;
-	Dimension d = BasicGraphicsUtils.getPreferredButtonSize(b, gap);
+	Dimension d = BasicGraphicsUtils.getPreferredSize(b, 
+							  gap,
+							  b.getText(),
+							  b.getIcon(),
+							  b.getVerticalAlignment(),
+							  b.getHorizontalAlignment(),
+							  b.getHorizontalTextPosition(),
+							  b.getVerticalTextPosition());
 	//	System.out.println("^^^^^^^^^^^^^^^^^^^^^^   BASIC-PREF="+d + ",T="+b.text);
 	return d;
     }
@@ -92,7 +99,7 @@
 
         g.setFont(f);
 
-        FontMetrics fm = g.getFontMetrics(f);
+        FontMetrics fm = SwingUtilities.getFontMetrics(f);
 
         Insets i = c.getInsets();
 
@@ -191,7 +198,7 @@
 
         g.setFont(f);
 
-        FontMetrics fm = g.getFontMetrics(f);
+        FontMetrics fm = SwingUtilities.getFontMetrics(f);
 
 	g.setColor(c.isEnabled() ? textColor : disabledTextColor);
 
