--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/ldap/ControlFactory.java	2002-11-23 14:44:28.000000000 -0700
+++ javax/naming/ldap/ControlFactory.java	2003-12-06 02:18:23.000000000 -0700
@@ -38,9 +38,10 @@
 
 package javax.naming.ldap;
 
-import javax.naming.*;
-import java.util.StringTokenizer;
 import java.util.Hashtable;
+import java.util.StringTokenizer;
+import javax.naming.Context;
+import javax.naming.NamingException;
 
 /**
  * @author Tom Tromey <tromey@redhat.com>
