--- /home/tromey/gnu/classpath/classpath/java/awt/geom/PathIterator.java	Mon Dec 30 15:39:04 2002
+++ java/awt/geom/PathIterator.java	Tue Dec 10 19:20:25 2002
@@ -84,7 +84,7 @@
    * (P1), and final interpolated control point (P2):
    * <pre>
    *  P(t) = B(2,0)*CP + B(2,1)*P1 + B(2,2)*P2
-   *    0 &lt;= t &lt;= 1
+   *    0 <= t <= 1
    *  B(n,m) = mth coefficient of nth degree Bernstein polynomial
    *         = C(n,m) * t^(m) * (1 - t)^(n-m)
    *  C(n,m) = Combinations of n things, taken m at a time
@@ -100,7 +100,7 @@
    * (P2), and final interpolated control point (P3):
    * <pre>
    *  P(t) = B(3,0)*CP + B(3,1)*P1 + B(3,2)*P2 + B(3,3)*P3
-   *    0 &lt;= t &lt;= 1
+   *    0 <= t <= 1
    *  B(n,m) = mth coefficient of nth degree Bernstein polynomial
    *         = C(n,m) * t^(m) * (1 - t)^(n-m)
    *  C(n,m) = Combinations of n things, taken m at a time
