--- /home/tromey/gnu/classpath/classpath/java/io/ObjectStreamField.java	Mon Mar 25 00:05:17 2002
+++ java/io/ObjectStreamField.java	Tue Jan 22 16:01:33 2002
@@ -47,23 +47,8 @@
   {
     this.name = name;
     this.type = type;
-    this.typename = TypeSignature.getEncodingOfClass(type);
   }
- 
-  /**
-   * There're many cases you can't get java.lang.Class from typename if your context
-   * class loader can't load it, then use typename to construct the field
-   */
-  ObjectStreamField (String name, String typename){
-    this.name = name;
-    this.typename = typename;
-    try{
-      type = TypeSignature.getClassForEncoding(typename);
-    }catch(ClassNotFoundException e){
-      type = Object.class; //??
-    }
-  }
-  
+
   public String getName ()
   {
     return name;
@@ -76,13 +61,12 @@
 
   public char getTypeCode ()
   {
-    return typename.charAt (0);
+    return TypeSignature.getEncodingOfClass (type).charAt (0);
   }
 
   public String getTypeString ()
   {
-    // use intern()
-    return typename.intern();
+    return TypeSignature.getEncodingOfClass (type);
   }
 
   public int getOffset ()
@@ -122,6 +106,5 @@
 
   private String name;
   private Class type;
-  private String typename;
   private int offset = -1; // XXX make sure this is correct
 }
