--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/PrivateKey.java	2003-05-07 02:23:22.000000000 -0600
+++ java/security/PrivateKey.java	2003-05-10 02:19:17.000000000 -0600
@@ -58,5 +58,5 @@
   /**
    * The version identifier used for serialization.
    */
-  long serialVersionUID = 6034044314589513430L;
+  public static final long serialVersionUID = 6034044314589513430L;
 } // interface PrivateKey
