--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIDefaults.java	2003-11-18 02:23:11.000000000 -0700
+++ javax/swing/UIDefaults.java	2004-02-12 02:18:27.000000000 -0700
@@ -61,7 +61,7 @@
 /**
  * UIDefaults is a database where all settings and interface bindings are
  * stored into. An PLAF implementation fills one of these (see for example
- * plaf/basic/BasicDefaults.java) with "JButton" -> new BasicButtonUI().
+ * plaf/basic/BasicLookAndFeel.java) with "ButtonUI" -> new BasicButtonUI().
  *
  * @author Ronald Veldema (rveldema@cs.vu.nl)
  */
