--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectStreamClass.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/ObjectStreamClass.java	2004-02-29 02:18:20.000000000 -0700
@@ -104,17 +104,17 @@
       }
   }
 
+
   /**
    * Returns the name of the class that this
    * <code>ObjectStreamClass</code> represents.
-   *
-   * @return the name of the class.
    */
   public String getName()
   {
     return name;
   }
 
+
   /**
    * Returns the class that this <code>ObjectStreamClass</code>
    * represents.  Null could be returned if this
@@ -129,27 +129,24 @@
     return clazz;
   }
 
+
   /**
    * Returns the serial version stream-unique identifier for the class
    * represented by this <code>ObjectStreamClass</code>.  This SUID is
    * either defined by the class as <code>static final long
    * serialVersionUID</code> or is calculated as specified in
    * Javasoft's "Object Serialization Specification" XXX: add reference
-   *
-   * @return the serial version UID.
    */
   public long getSerialVersionUID()
   {
     return uid;
   }
 
-  /**
-   * Returns the serializable (non-static and non-transient) Fields
-   * of the class represented by this ObjectStreamClass.  The Fields
-   * are sorted by name.
-   *
-   * @return the fields.
-   */
+
+  // Returns the serializable (non-static and non-transient) Fields
+  // of the class represented by this ObjectStreamClass.  The Fields
+  // are sorted by name.
+  // XXX doc
   public ObjectStreamField[] getFields()
   {
     ObjectStreamField[] copy = new ObjectStreamField[ fields.length ];
@@ -157,6 +154,7 @@
     return copy;
   }
 
+
   // XXX doc
   // Can't do binary search since fields is sorted by name and
   // primitiveness.
@@ -168,6 +166,7 @@
     return null;
   }
 
+
   /**
    * Returns a textual representation of this
    * <code>ObjectStreamClass</code> object including the name of the
@@ -182,6 +181,7 @@
     return "java.io.ObjectStreamClass< " + name + ", " + uid + " >";
   }
 
+
   // Returns true iff the class that this ObjectStreamClass represents
   // has the following method:
   //
