--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollBar.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JScrollBar.java	2004-04-17 02:18:07.000000000 -0600
@@ -41,14 +41,16 @@
 import java.awt.Dimension;
 import java.awt.event.AdjustmentEvent;
 import java.awt.event.AdjustmentListener;
-
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
 import javax.accessibility.AccessibleStateSet;
 import javax.accessibility.AccessibleValue;
+import javax.swing.SwingConstants;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
+import javax.swing.event.EventListenerList;
+import javax.swing.plaf.ProgressBarUI;
 import javax.swing.plaf.ScrollBarUI;
 
 
