--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/MappedByteBufferImpl.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/MappedByteBufferImpl.java	2004-03-01 02:18:07.000000000 -0700
@@ -39,9 +39,10 @@
 package java.nio;
 
 import java.io.IOException;
-import gnu.classpath.RawData;
+import gnu.java.nio.channels.FileChannelImpl;
+import gnu.gcj.RawData;
 
-final class MappedByteBufferImpl extends MappedByteBuffer
+class MappedByteBufferImpl extends MappedByteBuffer
 {
   boolean readOnly;
   RawData address;
@@ -200,133 +201,133 @@
     return new DoubleViewBufferImpl (this, remaining() >> 3);
   }
 
-  public char getChar ()
+  final public char getChar ()
   {
     return ByteBufferHelper.getChar(this, order());
   }
   
-  public ByteBuffer putChar (char value)
+  final public ByteBuffer putChar (char value)
   {
     ByteBufferHelper.putChar(this, value, order());
     return this;
   }
   
-  public char getChar (int index)
+  final public char getChar (int index)
   {
     return ByteBufferHelper.getChar(this, index, order());
   }
   
-  public ByteBuffer putChar (int index, char value)
+  final public ByteBuffer putChar (int index, char value)
   {
     ByteBufferHelper.putChar(this, index, value, order());
     return this;
   }
 
-  public short getShort ()
+  final public short getShort ()
   {
     return ByteBufferHelper.getShort(this, order());
   }
   
-  public ByteBuffer putShort (short value)
+  final public ByteBuffer putShort (short value)
   {
     ByteBufferHelper.putShort(this, value, order());
     return this;
   }
   
-  public short getShort (int index)
+  final public short getShort (int index)
   {
     return ByteBufferHelper.getShort(this, index, order());
   }
   
-  public ByteBuffer putShort (int index, short value)
+  final public ByteBuffer putShort (int index, short value)
   {
     ByteBufferHelper.putShort(this, index, value, order());
     return this;
   }
 
-  public int getInt ()
+  final public int getInt ()
   {
     return ByteBufferHelper.getInt(this, order());
   }
   
-  public ByteBuffer putInt (int value)
+  final public ByteBuffer putInt (int value)
   {
     ByteBufferHelper.putInt(this, value, order());
     return this;
   }
   
-  public int getInt (int index)
+  final public int getInt (int index)
   {
     return ByteBufferHelper.getInt(this, index, order());
   }
   
-  public ByteBuffer putInt (int index, int value)
+  final public ByteBuffer putInt (int index, int value)
   {
     ByteBufferHelper.putInt(this, index, value, order());
     return this;
   }
 
-  public long getLong ()
+  final public long getLong ()
   {
     return ByteBufferHelper.getLong(this, order());
   }
   
-  public ByteBuffer putLong (long value)
+  final public ByteBuffer putLong (long value)
   {
     ByteBufferHelper.putLong (this, value, order());
     return this;
   }
   
-  public long getLong (int index)
+  final public long getLong (int index)
   {
     return ByteBufferHelper.getLong (this, index, order());
   }
   
-  public ByteBuffer putLong (int index, long value)
+  final public ByteBuffer putLong (int index, long value)
   {
     ByteBufferHelper.putLong (this, index, value, order());
     return this;
   }
 
-  public float getFloat ()
+  final public float getFloat ()
   {
     return ByteBufferHelper.getFloat (this, order());
   }
   
-  public ByteBuffer putFloat (float value)
+  final public ByteBuffer putFloat (float value)
   {
     ByteBufferHelper.putFloat (this, value, order());
     return this;
   }
   
-  public float getFloat (int index)
+  public final float getFloat (int index)
   {
     return ByteBufferHelper.getFloat (this, index, order());
   }
 
-  public ByteBuffer putFloat (int index, float value)
+  final public ByteBuffer putFloat (int index, float value)
   {
     ByteBufferHelper.putFloat (this, index, value, order());
     return this;
   }
 
-  public double getDouble ()
+  final public double getDouble ()
   {
     return ByteBufferHelper.getDouble (this, order());
   }
 
-  public ByteBuffer putDouble (double value)
+  final public ByteBuffer putDouble (double value)
   {
     ByteBufferHelper.putDouble (this, value, order());
     return this;
   }
   
-  public double getDouble (int index)
+  final public double getDouble (int index)
   {
     return ByteBufferHelper.getDouble (this, index, order());
   }
   
-  public ByteBuffer putDouble (int index, double value)
+  final public ByteBuffer putDouble (int index, double value)
   {
     ByteBufferHelper.putDouble (this, index, value, order());
     return this;
