--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/AccessController.java	2004-04-13 02:20:25.000000000 -0600
+++ java/security/AccessController.java	2002-05-31 12:14:17.000000000 -0600
@@ -91,7 +91,7 @@
    *
    * @param action the <code>PrivilegedAction</code> whose <code>run()</code>
    * should be be called.
-   * @return the result of the <code>action.run()</code> method.
+   * @returns the result of the <code>action.run()</code> method.
    */
   public static Object doPrivileged(PrivilegedAction action)
   {
@@ -110,7 +110,7 @@
    * should be be called.
    * @param context the <code>AccessControlContext</code> whose protection
    * domains should be added to the protection domain of the calling class.
-   * @return the result of the <code>action.run()</code> method.
+   * @returns the result of the <code>action.run()</code> method.
    */
   public static Object doPrivileged(PrivilegedAction action,
 				    AccessControlContext context)
@@ -130,7 +130,7 @@
    *
    * @param action the <code>PrivilegedExceptionAction</code> whose
    * <code>run()</code> should be be called.
-   * @return the result of the <code>action.run()</code> method.
+   * @returns the result of the <code>action.run()</code> method.
    * @exception PrivilegedActionException wrapped around any exception that
    * is thrown in the <code>run()</code> method.
    */
@@ -162,7 +162,7 @@
    * <code>run()</code> should be be called.
    * @param context the <code>AccessControlContext</code> whose protection
    * domains should be added to the protection domain of the calling class.
-   * @return the result of the <code>action.run()</code> method.
+   * @returns the result of the <code>action.run()</code> method.
    * @exception PrivilegedActionException wrapped around any exception that
    * is thrown in the <code>run()</code> method.
    */
