--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Boolean.java	Mon Jul 23 14:20:24 2001
+++ java/lang/Boolean.java	Fri Oct 26 16:18:42 2001
@@ -60,7 +60,7 @@
      * The primitive type <code>boolean</code> is represented by this 
      * <code>Class</code> object.
      */
-    public static final Class TYPE = VMClassLoader.getPrimitiveClass("boolean");
+    public static final Class TYPE = VMClassLoader.getPrimitiveClass('Z');
     
     /**
      * The immutable value of this Boolean.
