--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/image/Raster.java	2004-04-13 02:20:18.000000000 -0600
+++ java/awt/image/Raster.java	2003-09-26 02:18:35.000000000 -0600
@@ -452,25 +452,4 @@
 				  y-sampleModelTranslateY,
 				  w, h, b, dArray, dataBuffer);
   }
-  
-  /**
-   * Create a String representing the stat of this Raster.
-   * @return A String representing the stat of this Raster.
-   * @see java.lang.Object#toString()
-   */
-  public String toString()
-  {
-    StringBuffer result = new StringBuffer();
-    
-    result.append(getClass().getName());
-    result.append("[(");
-    result.append(minX).append(",").append(minY).append("), ");
-    result.append(width).append(" x ").append(height).append(",");
-    result.append(sampleModel).append(",");
-    result.append(dataBuffer);
-    result.append("]");
-    
-    return result.toString();
-  }
-  
 }
