--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTabbedPane.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JTabbedPane.java	2004-04-18 02:18:29.000000000 -0600
@@ -44,13 +44,15 @@
 import java.awt.event.MouseEvent;
 import java.io.Serializable;
 import java.util.Vector;
-
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
 import javax.accessibility.AccessibleSelection;
+import javax.accessibility.AccessibleStateSet;
+import javax.accessibility.AccessibleValue;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
+import javax.swing.event.EventListenerList;
 import javax.swing.plaf.TabbedPaneUI;
 import javax.swing.plaf.UIResource;
 
