--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicTextUI.java	2003-07-02 02:19:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicTextUI.java	2003-06-28 02:18:24.000000000 -0600
@@ -38,25 +38,12 @@
 
 package javax.swing.plaf.basic;
 
-import java.awt.Color;
-import java.awt.Dimension;
-import java.awt.Graphics;
-import java.awt.Point;
-import java.awt.Rectangle;
-import javax.swing.JComponent;
-import javax.swing.plaf.ComponentUI;
-import javax.swing.plaf.TextUI;
-import javax.swing.text.BadLocationException;
-import javax.swing.text.DefaultEditorKit;
-import javax.swing.text.EditorKit;
-import javax.swing.text.Element;
-import javax.swing.text.JTextComponent;
-import javax.swing.text.Position;
-import javax.swing.text.View;
-import javax.swing.text.ViewFactory;
+import javax.swing.text.*;
+import javax.swing.plaf.*;
+import java.awt.*;
+import javax.swing.*;
 
 public class BasicTextUI extends TextUI
-  implements ViewFactory
 {
     int gap = 3;
     View view = null; // was: new RootView();
@@ -180,10 +167,9 @@
     {
 	return 0;
     } 
-
-  public View create (Element elem)
-  {
-    // subclasses have to implement this to get this functionality
-    return null;
-  }
 }
+
+
+
+
+
