--- /home/tromey/gnu/classpath/classpath/java/net/Authenticator.java	Sun Jul 14 23:54:26 2002
+++ java/net/Authenticator.java	Tue Jan 22 16:01:35 2002
@@ -38,10 +38,10 @@
 package java.net;
 
 /**
-  * This abstract class provides a model for obtaining authentication
-  * information (in the form of a username and password) required by
-  * some network operations (such as hitting a password protected
-  * web site).
+  * Sometimes a network operation (such as hitting a password protected
+  * web site) will require authentication information in the form of a
+  * username and password.  This abstract class provides a model for 
+  * obtaining that information.
   * <p>
   * To make use of this feature, a programmer must create a subclass of
   * Authenticator that knows how to obtain the necessary info.  An example
