--- /home/tromey/gnu/classpath/classpath/java/rmi/activation/ActivationGroupDesc.java	Wed Oct  2 11:21:38 2002
+++ java/rmi/activation/ActivationGroupDesc.java	Tue Jan 22 15:40:25 2002
@@ -41,9 +41,8 @@
 import java.util.Properties;
 import java.rmi.MarshalledObject;
 
-public final class ActivationGroupDesc implements Serializable
-{
-  static final long serialVersionUID = -4936225423168276595L;
+public final class ActivationGroupDesc
+	implements Serializable {
 
 public static class CommandEnvironment
 	implements Serializable {
