--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/BreakIterator.java	2002-01-22 15:27:01.000000000 -0700
+++ java/text/BreakIterator.java	2003-05-01 02:21:57.000000000 -0600
@@ -77,6 +77,21 @@
   }
 
   /**
+   * Create a clone of this object.
+   */
+  public Object clone ()
+  {
+    try
+      {
+        return super.clone();
+      }
+    catch (CloneNotSupportedException e)
+      {
+        return null;
+      }
+  }
+  
+  /**
    * This method returns the index of the current text element boundary.
    *
    * @return The current text boundary.
