--- /home/tromey/gnu/classpath/classpath/java/lang/NoClassDefFoundError.java	Sat Mar  2 12:13:12 2002
+++ java/lang/NoClassDefFoundError.java	Tue Jan 22 16:01:35 2002
@@ -1,5 +1,5 @@
-/* NoClassDefFoundError.java -- thrown when a ClassLoader cannot find a class
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* NoClassDefFoundError.java 
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,39 +38,43 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct. 
+ */
+
 /**
  * A <code>NoClassDefFoundError</code> is thrown when a classloader or the
  * Java Virtual Machine tries to load a class and no definition of the class
  * can be found.  This could happen when using the <code>new</code> expression
- * or during a normal method call.  The reason this would occur at runtime is
+ * or during a normal method call.  The reason this would occur at runtime is 
  * because the missing class definition existed when the currently executing 
  * class was compiled, but now that definition cannot be found.
  *
+ * @since JDK 1.0
+ * 
  * @author Brian Jones
  * @author Tom Tromey <tromey@cygnus.com>
- * @status updated to 1.4
+ * @date October 1, 1998
  */
 public class NoClassDefFoundError extends LinkageError
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = 9095859863287012458L;
+  static final long serialVersionUID = 9095859863287012458L;
 
   /**
    * Create an error without a message.
    */
   public NoClassDefFoundError()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an error with a message.
-   *
-   * @param s the message
    */
   public NoClassDefFoundError(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
