--- /home/tromey/gnu/classpath/classpath/java/nio/channels/SocketChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/SocketChannel.java	Tue Dec 10 19:20:33 2002
@@ -49,7 +49,6 @@
  * @since 1.4
  */
 abstract public class SocketChannel extends AbstractSelectableChannel
-  implements ByteChannel, ScatteringByteChannel, GatheringByteChannel
 {
   /**
    * Initializes this socket.
@@ -101,7 +100,7 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException If this channel is not yet connected.
    */
-  public final long read (ByteBuffer[] dsts) throws IOException
+  public final long read (ByteBuffer[] dsts)
   {
     long b = 0;
     
@@ -119,7 +118,7 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException If this channel is not yet connected.
    */
-  public final long write (ByteBuffer[] dsts) throws IOException
+  public final long write (ByteBuffer[] dsts)
   {
     long b = 0;
 
@@ -145,7 +144,7 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException If this channel is not yet connected.
    */
-  public abstract int read (ByteBuffer dst) throws IOException;
+  public abstract int read (ByteBuffer dst);
 
   /**
    * Connects the channel's socket to the remote address.
@@ -180,7 +179,7 @@
    * @exception NoConnectionPendingException If this channel is not connected
    * and a connection operation has not been initiated.
    */
-  public abstract boolean finishConnect () throws IOException;
+  public abstract boolean finishConnect ();
  
   /**
    * Tells whether or not the channel's socket is connected.
@@ -198,8 +197,7 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException If this channel is not yet connected.
    */
-  public abstract long read (ByteBuffer[] dsts, int offset, int length)
-    throws IOException;
+  public abstract long read (ByteBuffer[] dsts, int offset, int length);
  
   /**
    * Retrieves the channel's socket.
@@ -212,7 +210,7 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException If this channel is not yet connected.
    */
-  public abstract int write (ByteBuffer src) throws IOException;
+  public abstract int write (ByteBuffer src);
   
   /**
    * Writes data to the channel.
@@ -220,6 +218,5 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException If this channel is not yet connected.
    */
-  public abstract long write (ByteBuffer[] srcs, int offset, int length)
-    throws IOException;
+  public abstract long write (ByteBuffer[] srcs, int offset, int length);
 }
