--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/FlowLayout.java	2003-10-09 02:19:58.000000000 -0600
+++ java/awt/FlowLayout.java	2003-12-20 02:18:39.000000000 -0700
@@ -214,7 +214,8 @@
 		if (comps[k].visible)
 		  {
 		    Dimension c = comps[k].getPreferredSize ();
-		    comps[k].setBounds (x, y, c.width, new_h);
+		    comps[k].setBounds (x, y + (new_h - c.height) / 2,
+					c.width, c.height);
 		    x += c.width + hgap;
 		  }
 	      }
