--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DirectByteBufferImpl.java	2004-11-23 02:24:10.000000000 -0700
+++ java/nio/DirectByteBufferImpl.java	2004-11-25 02:16:36.000000000 -0700
@@ -38,7 +38,7 @@
 
 package java.nio;
 
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 abstract class DirectByteBufferImpl extends ByteBuffer
 {
@@ -89,6 +89,11 @@
       super(capacity);
     }
 
+    ReadWrite(RawData address, int capacity)
+    {
+      super(address, capacity);
+    }
+    
     ReadWrite(Object owner, RawData address,
 	      int capacity, int limit,
 	      int position)
@@ -109,6 +114,13 @@
     this.address = VMDirectByteBuffer.allocate(capacity);
   }
 
+  DirectByteBufferImpl(RawData address, int capacity)
+  {
+    super(capacity, capacity, 0, -1);
+    this.owner = this;
+    this.address = address;
+  }
+  
   DirectByteBufferImpl(Object owner, RawData address,
 		       int capacity, int limit,
 		       int position)
