--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/AbstractButton.java	2004-02-15 02:21:04.000000000 -0700
+++ javax/swing/AbstractButton.java	2004-02-12 02:18:27.000000000 -0700
@@ -1612,7 +1612,7 @@
    *
    * @return The current rollover selected icon
    */
-  public Icon getRolloverSelectedIcon()
+  Icon getRolloverSelectedIcon()
   {
     return rollover_selected_icon;
   }
@@ -1648,7 +1648,7 @@
    *
    * @return The current selected icon
    */
-  public Icon getSelectedIcon()
+  Icon getSelectedIcon()
   {
     return selected_icon;
   }
