--- /home/tromey/gnu/classpath/classpath/java/awt/Cursor.java	Mon Dec 30 15:39:04 2002
+++ java/awt/Cursor.java	Tue Dec 10 19:20:24 2002
@@ -133,14 +133,11 @@
    * type.
    *
    * @param type The cursor type.
-   *
-   * @exception IllegalArgumentException If the specified cursor type is invalid
    */
   public Cursor(int type)
   {
     if (type < 0 || type >= PREDEFINED_COUNT)
       throw new IllegalArgumentException ("invalid cursor " + type);
-
     this.type = type;
     // FIXME: lookup and set name?
   }
@@ -174,20 +171,9 @@
     return predefined[type];
   }
 
-  /**
-   * Retrieves the system specific custom Cursor named Cursor names are,
-   * for example: "Invalid.16x16".
-   *
-   * @exception AWTException
-   * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-   * returns true.
-   */
   public static Cursor getSystemCustomCursor(String name)
                                       throws AWTException
   {
-    if (GraphicsEnvironment.isHeadless())
-      throw new HeadlessException ();
-
     // FIXME
     return null;
   }
