--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JProgressBar.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JProgressBar.java	2004-06-18 02:18:53.000000000 -0600
@@ -85,14 +85,15 @@
   protected class AccessibleJProgressBar extends AccessibleJComponent
     implements AccessibleValue
   {
+    private static final long serialVersionUID = -2938130009392721813L;
+  
     /**
      * Constructor AccessibleJProgressBar
      *
      * @param component TODO
      */
-    protected AccessibleJProgressBar(JProgressBar component)
+    protected AccessibleJProgressBar()
     {
-      super(component);
     } 
 
     /**
@@ -168,6 +169,8 @@
     } 
   } 
 
+  private static final long serialVersionUID = 1980046021813598781L;
+  
   /** Fired in a PropertyChangeEvent when the "borderPainted" property changes. */
   public static final String BORDER_PAINTED_CHANGED_PROPERTY = "borderPainted";
   
@@ -661,7 +664,8 @@
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      accessibleContext = new AccessibleJProgressBar(this);
+      accessibleContext = new AccessibleJProgressBar();
+    
     return accessibleContext;
   } 
 }
