--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicPanelUI.java	2004-01-11 02:20:39.000000000 -0700
+++ javax/swing/plaf/basic/BasicPanelUI.java	2004-02-12 02:18:27.000000000 -0700
@@ -44,8 +44,6 @@
 
 public class BasicPanelUI extends PanelUI
 {
-    int gap = 3;
-
     public static ComponentUI createUI(JComponent x) 
     {
         return new BasicPanelUI();
@@ -53,6 +51,6 @@
 
     public void installUI(JComponent c)
     {
-	super.installUI(c);
+      super.installUI(c);
     }
 }
