--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/dnd/DropTarget.java	Sun Mar 23 02:18:38 2003
+++ java/awt/dnd/DropTarget.java	Mon Mar  3 02:17:48 2003
@@ -1,5 +1,5 @@
 /* DropTarget.java -- 
-   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -60,9 +60,7 @@
    */
   private static final long serialVersionUID = -6283860791671019047L;
 
-  /** @specnote According to the online documentation, this is
-   * protected, but in reality it is public.  */
-  public static class DropTargetAutoScroller
+  protected static class DropTargetAutoScroller
     implements ActionListener
   {
     private Component component;
