--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/SecureRandom.java	2003-04-24 02:22:11.000000000 -0600
+++ java/security/SecureRandom.java	2003-04-20 02:20:19.000000000 -0600
@@ -41,8 +41,6 @@
 import java.util.Random;
 import java.util.Enumeration;
 
-import gnu.java.security.Engine;
-
 /**
  * An interface to a cryptographically secure pseudo-random number
  * generator (PRNG). Random (or at least unguessable) numbers are used
@@ -236,10 +234,6 @@
           Engine.getInstance(SECURE_RANDOM, algorithm, provider),
           provider);
       }
-    catch (java.lang.reflect.InvocationTargetException ite)
-      {
-	throw new NoSuchAlgorithmException(algorithm);
-      }
     catch (ClassCastException cce)
       {
         throw new NoSuchAlgorithmException(algorithm);
