--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Compiler.java	2004-04-13 02:20:22.000000000 -0600
+++ java/lang/Compiler.java	2002-06-15 13:30:18.000000000 -0600
@@ -81,7 +81,7 @@
   }
 
   /**
-   * Compile the classes whose name matches <code>classNames</code>.
+   * Compile the classes whose name matches <code>classNames/code>.
    *
    * @param classNames the name of classes to compile
    * @return <code>false</code> if no compiler is available or
