--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileOutputStream.java	Mon Apr  7 02:24:27 2003
+++ java/io/FileOutputStream.java	Fri Mar 28 02:18:03 2003
@@ -84,20 +84,9 @@
     SecurityManager s = System.getSecurityManager();
     if (s != null)
       s.checkWrite(path);
-
-    fd = new FileDescriptor();
-
-    try 
-      {
-        if (append)
-          fd.open(path, "a");
-        else
-          fd.open(path, "w");
-      }
-    catch(IOException e)
-      {
-        throw new FileNotFoundException(path + ": " + e.getMessage());
-      }
+    fd = new FileDescriptor (path, (append
+				    ? FileDescriptor.APPEND
+				    : FileDescriptor.WRITE));
   }
 
   /**
@@ -244,7 +233,7 @@
   public void write (byte[] buf)
     throws IOException
   {
-    write (buf, 0, buf.length);
+    fd.write (buf, 0, buf.length);
   }
 
   /**
