--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/AbstractButton.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/AbstractButton.java	2004-02-25 02:18:21.000000000 -0700
@@ -37,6 +37,8 @@
 
 package javax.swing;
 
+import java.awt.AWTEvent;
+import java.awt.AWTEventMulticaster;
 import java.awt.Graphics;
 import java.awt.Image;
 import java.awt.Insets;
@@ -49,10 +51,12 @@
 import java.awt.event.FocusListener;
 import java.awt.event.ItemEvent;
 import java.awt.event.ItemListener;
+import java.awt.event.MouseEvent;
 import java.beans.PropertyChangeEvent;
 import java.beans.PropertyChangeListener;
+import java.io.Serializable;
+import java.util.Vector;
 import java.util.EventListener;
-
 import javax.accessibility.AccessibleAction;
 import javax.accessibility.AccessibleIcon;
 import javax.accessibility.AccessibleRelationSet;
@@ -147,8 +151,8 @@
  *
  * </ul>
  *
- * @author Ronald Veldema (rveldema@cs.vu.nl)
- * @author Graydon Hoare (graydon@redhat.com)
+ * @author Ronald Veldema (rveldema&064;cs.vu.nl)
+ * @author Graydon Hoare (graydon&064;redhat.com)
  */
 
 public abstract class AbstractButton extends JComponent
