--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/CharSequence.java	2004-04-18 02:22:16.000000000 -0600
+++ java/lang/CharSequence.java	2002-06-15 13:27:28.000000000 -0600
@@ -78,7 +78,7 @@
   int length();
 
   /**
-   * Returns a new <code>CharSequence</code> of the indicated range.
+   * Returns a new <code>CharSequence</char> of the indicated range.
    *
    * @param begin the start index (inclusive)
    * @param end the end index (exclusive)
