--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2004-07-09 02:22:40.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2004-06-18 02:18:58.000000000 -0600
@@ -1,5 +1,5 @@
 /* BasicLookAndFeel.java --
-   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -228,7 +228,7 @@
   {
     Object[] uiDefaults;
     
-    // The default Look and Feel happens to use these three purple shades
+    // The JDK's default L&F happens to use these three purple shades
     // extensively.
     Color lightPurple = new Color(0xCC, 0xCC, 0xFF);
     Color midPurple = new Color(0x99, 0x99, 0xCC);
