--- /home/tromey/gnu/Nightly/classpath/classpath/javax/crypto/Mac.java	2004-11-17 02:24:09.000000000 -0700
+++ javax/crypto/Mac.java	2004-10-28 02:16:12.000000000 -0600
@@ -290,7 +290,7 @@
    *
    * @return The MAC length.
    */
-  public final int getMacLength()
+  public int getMacLength()
   {
     return macSpi.engineGetMacLength();
   }
@@ -404,7 +404,7 @@
    * @throws java.lang.CloneNotSupportedException If the underlying
    *         implementation is not cloneable.
    */
-  public final Object clone() throws CloneNotSupportedException
+  public Object clone() throws CloneNotSupportedException
   {
     Mac result = new Mac((MacSpi) macSpi.clone(), provider, algorithm);
     result.virgin = virgin;
