--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Locale.java	2003-08-13 02:19:50.000000000 -0600
+++ java/util/Locale.java	2003-06-28 02:18:23.000000000 -0600
@@ -489,7 +489,7 @@
          + "WS,YE,YT,YU,ZA,ZM,ZR,ZW")
       .indexOf(country);
 
-    if (index % 3 != 0 || country.length() != 2)
+    if (index % 3 != 0 || language.length() != 2)
       throw new MissingResourceException
         ("Can't find ISO3 country for " + country,
          "java.util.Locale", country);
