--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/dnd/DropTargetContext.java	Sun Mar 23 02:18:38 2003
+++ java/awt/dnd/DropTargetContext.java	Mon Mar  3 02:17:48 2003
@@ -1,5 +1,5 @@
 /* DropTargetContext.java --
-   Copyright (C) 2002, 2003 Free Software Foundation
+   Copyright (C) 2002 Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -55,9 +55,7 @@
 {
   static final long serialVersionUID = -634158968993743371L;
 
-  /** @specnote According to the online documentation, this is
-   * protected, but in reality it is public.  */
-  public class TransferableProxy implements Transferable
+  protected class TransferableProxy implements Transferable
   {
     protected boolean isLocal;
     protected Transferable transferable;
