--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/charset/Charset.java	2004-10-28 02:27:35.000000000 -0600
+++ java/nio/charset/Charset.java	2004-10-28 02:16:12.000000000 -0600
@@ -123,6 +123,8 @@
     Charset cs = charsetForName (charsetName);
     if (cs == null)
       throw new UnsupportedCharsetException (charsetName);
+    cachedDecoder = null;
+    cachedEncoder = null;
     return cs;
   }
 
