--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/Operation.java	Sun Mar 23 02:18:41 2003
+++ java/rmi/server/Operation.java	Mon Mar 31 02:17:49 2003
@@ -40,20 +40,31 @@
 /**
  * @deprecated
  */
-public class Operation {
+public class Operation
+{
+  private String operation;
 
-private String operation;
+  /**
+   * @deprecated
+   */
+  public Operation (String op)
+  {
+    operation = op;
+  }
 
-public Operation(String op) {
-	operation = op;
-}
-
-public String getOperation() {
-	return (operation);
-}
-
-public String toString() {
-	return (operation);
-}
+  /**
+   * @deprecated
+   */
+  public String getOperation ()
+  {
+    return operation;
+  }
 
+  /**
+   * @deprecated
+   */
+  public String toString ()
+  {
+    return operation;
+  }
 }
