--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JToolBar.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JToolBar.java	2004-01-11 02:18:05.000000000 -0700
@@ -37,14 +37,14 @@
 
 package javax.swing;
 
-import java.awt.Component;
 import java.awt.Dimension;
+import java.awt.Component;
+import java.awt.Container;
 import java.awt.Graphics;
 import java.awt.Insets;
 import java.beans.PropertyChangeListener;
 import java.io.IOException;
 import java.io.ObjectOutputStream;
-
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
