--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JFrame.java	2003-11-22 02:22:37.000000000 -0700
+++ javax/swing/JFrame.java	2004-02-12 02:18:27.000000000 -0700
@@ -38,6 +38,7 @@
 
 package javax.swing;
 
+import java.awt.AWTEvent;
 import java.awt.BorderLayout;
 import java.awt.Component;
 import java.awt.Container;
@@ -105,6 +106,7 @@
     protected  void frameInit()
     {
       super.setLayout(new BorderLayout(1, 1));
+      enableEvents(AWTEvent.WINDOW_EVENT_MASK);
       getRootPane(); // will do set/create
     }
   
@@ -201,9 +203,7 @@
 
     protected  void processWindowEvent(WindowEvent e)
     {
-	//	System.out.println("PROCESS_WIN_EV-1: " + e);
 	super.processWindowEvent(e); 
-	//	System.out.println("PROCESS_WIN_EV-2: " + e);
 	switch (e.getID())
 	    {
 	    case WindowEvent.WINDOW_CLOSING:
@@ -212,13 +212,11 @@
 			{
 			case EXIT_ON_CLOSE:
 			    {
-				System.out.println("user requested exit on close");
 				System.exit(1);
 				break;
 			    }
 			case DISPOSE_ON_CLOSE:
 			    {
-				System.out.println("user requested dispose on close");
 				dispose();
 				break;
 			    }
