--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Package.java	Fri Oct  5 09:36:25 2001
+++ java/lang/Package.java	Wed Apr 25 09:45:12 2001
@@ -235,7 +235,8 @@
   public static Package getPackage(String name)
   {
     // Get the caller's classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
 
     if (cl != null)
@@ -251,7 +252,8 @@
   public static Package[] getPackages()
   {
     // Get the caller's classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
 
     if (cl != null)
