--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Package.java	Mon Mar  3 02:19:19 2003
+++ java/lang/Package.java	Mon Mar  3 02:17:48 2003
@@ -269,7 +269,8 @@
   public static Package getPackage(String name)
   {
     // Get the caller's classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
     return cl != null ? cl.getPackage(name) : null;
   }
@@ -283,7 +284,8 @@
   public static Package[] getPackages()
   {
     // Get the caller's classloader
-    Class c = VMSecurityManager.getClassContext()[1];
+    SecurityManager sm = System.getSecurityManager();
+    Class c = sm.getClassContext()[1];
     ClassLoader cl = c.getClassLoader();
     // Sun's implementation returns the packages loaded by the bootstrap
     // classloader if cl is null, but right now our bootstrap classloader
@@ -311,7 +313,12 @@
    */
   public String toString()
   {
-    return "package " + name + (specTitle == null ? "" : ", " + specTitle)
-      + (specVersion == null ? "" : ", version " + specVersion);
+    return "package: " + name +
+	   " spec: " + specTitle +
+	   " version: " + specVersion +
+	   " vendor: " + specVendor +
+	   " implementation: " + implTitle +
+	   " version: " + implVersion +
+	   " vendor: " + implVendor + " sealed: " + sealed;
   }
 } // class Package
