--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JComboBox.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JComboBox.java	2004-01-29 02:18:52.000000000 -0700
@@ -48,12 +48,13 @@
 import java.io.IOException;
 import java.io.ObjectOutputStream;
 import java.util.Vector;
-
 import javax.accessibility.Accessible;
-import javax.accessibility.AccessibleAction;
 import javax.accessibility.AccessibleContext;
+import javax.accessibility.AccessibleAction;
 import javax.accessibility.AccessibleRole;
 import javax.accessibility.AccessibleSelection;
+import javax.swing.event.MenuEvent;
+import javax.swing.event.MenuListener;
 import javax.swing.event.ListDataEvent;
 import javax.swing.event.ListDataListener;
 import javax.swing.event.PopupMenuListener;
