--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/PublicKey.java	2002-03-29 01:30:51.000000000 -0700
+++ java/security/PublicKey.java	2003-05-04 02:19:12.000000000 -0600
@@ -54,7 +54,7 @@
 public interface PublicKey extends Key
 {
   /**
-   * The verion identifier used for serialization.
+   * The version identifier used for serialization.
    */
   long serialVersionUID = 7187392471159151072L;
 } // interface PublicKey
