--- /home/tromey/gnu/egcs/classpath/classpath/java/io/UnsupportedEncodingException.java	Sat Mar  2 12:13:10 2002
+++ java/io/UnsupportedEncodingException.java	Tue Jan 22 16:01:33 2002
@@ -1,5 +1,5 @@
-/* UnsupportedEncodingException.java -- the requested encoding isn't supported
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* UnsupportedEncodingException.java -- The requested encoding isn't supported
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,36 +38,49 @@
 
 package java.io;
 
-/**
- * This exception is thrown when the requested character encoding is 
- * not supported.
- *
- * @author Aaron M. Renn (arenn@urbanophile.com)
- * @author Per Bothner <bothner@cygnus.com>
- * @since 1.1
- * @status updated to 1.4
+/* Written using "Java Class Libraries", 2nd edition, plus online
+ * API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
  */
+ 
+/**
+  * This exception is thrown when the requested character encoding is 
+  * not supported.
+  *
+  * @version 0.0
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @author Per Bothner <bothner@cygnus.com>
+  * @date April 17, 1998.
+  */
 public class UnsupportedEncodingException extends IOException
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -4274276298326136670L;
-
-  /**
-   * Create an exception without a descriptive error message.
-   */
-  public UnsupportedEncodingException()
-  {
-  }
-
-  /**
-   * Create an exception with a descriptive error message.
-   *
-   * @param message the descriptive error message
-   */
-  public UnsupportedEncodingException(String message)
-  {
-    super(message);
-  }
+
+/*
+ * Constructors
+ */
+
+/**
+  * Create a new UnsupportedEncodingException without a descriptive error message
+  */
+public
+UnsupportedEncodingException()
+{
+  super();
+}
+
+/*************************************************************************/
+
+/**
+  * Create a new UnsupportedEncodingException with a descriptive error message String
+  *
+  * @param message The descriptive error message
+  */
+public
+UnsupportedEncodingException(String message)
+{
+  super(message);
+}
+
 } // class UnsupportedEncodingException
+
