--- /home/tromey/gnu/classpath/classpath/java/nio/channels/InterruptibleChannel.java	Tue Apr 30 17:58:37 2002
+++ java/nio/channels/InterruptibleChannel.java	Mon Oct  7 01:59:12 2002
@@ -41,8 +41,10 @@
 
 public interface InterruptibleChannel extends Channel
 {
-    void close() throws IOException;
+  /**
+   * Closes this channel
+   *
+   * @exception IOException If an error occurs
+   */
+  void close() throws IOException;
 }
-
-    
-
