--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RemoteServer.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/server/RemoteServer.java	2004-03-21 02:18:06.000000000 -0700
@@ -1,5 +1,5 @@
-/* RemoteServer.java --
-   Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
