--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/InflaterInputStream.java	2004-06-21 06:24:02.000000000 -0600
+++ java/util/zip/InflaterInputStream.java	2004-09-23 02:16:13.000000000 -0600
@@ -70,9 +70,8 @@
    */
   protected int len;
 
-  /*
-   * We just use this if we are decoding one byte at a time with the read() call
-   */
+  // We just use this if we are decoding one byte at a time with the
+  // read() call.
   private byte[] onebytebuffer = new byte[1];
 
   /**
@@ -139,9 +138,8 @@
    */
   public synchronized void close() throws IOException
   {
-    if (in != null)
-      in.close();
-    in = null;
+    inf = null;
+    super.close();
   }
 
   /**
@@ -167,11 +165,9 @@
    */
   public int read() throws IOException
   { 
-    int nread = read(onebytebuffer, 0, 1); //read one byte
-    
+    int nread = read(onebytebuffer, 0, 1);
     if (nread > 0)
       return onebytebuffer[0] & 0xff;
-    
     return -1;
   }
 
@@ -188,14 +184,28 @@
       throw new IOException("stream closed");
     if (len == 0)
       return 0;
+    if (inf.finished())
+      return -1;
 
     int count = 0;
     for (;;)
       {
+	if (inf.needsInput())
+	  fill();
 	
 	try
 	  {
 	    count = inf.inflate(b, off, len);
+	    if (count == 0)
+	      {
+		if (this.len == -1)
+		  {
+		    // Couldn't get any more data to feed to the Inflater
+		    return -1;
+		  }
+		if (inf.needsDictionary())
+		  throw new ZipException("Inflater needs Dictionary");
+	      }
 	  } 
 	catch (DataFormatException dfe) 
 	  {
@@ -204,14 +214,6 @@
 
 	if (count > 0)
 	  return count;
-	
-	if (inf.needsDictionary()
-	    | inf.finished())
-	  return -1;
-	else if (inf.needsInput())
-	  fill();
-	else
-	  throw new InternalError("Don't know what to do");
       }
   }
 
