--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/FileChannelImpl.java	2003-10-21 02:19:06.000000000 -0600
+++ java/nio/channels/FileChannelImpl.java	2003-12-20 02:18:39.000000000 -0700
@@ -48,7 +48,7 @@
 import java.nio.MappedByteBuffer;
 import java.nio.MappedByteBufferImpl;
 import gnu.classpath.Configuration;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
@@ -169,7 +169,9 @@
     byte[] buffer = new byte [dst.remaining ()];
     
     result = implRead (buffer, 0, buffer.length);
-    dst.put (buffer, 0, result);
+
+    if (result > 0)
+      dst.put (buffer, 0, result);
 
     return result;
   }
