--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/FieldPosition.java	2004-04-13 02:20:26.000000000 -0600
+++ java/text/FieldPosition.java	2003-12-03 02:18:10.000000000 -0700
@@ -72,13 +72,13 @@
   /**
    * This method initializes a new instance of <code>FieldPosition</code>
    * to have the specified field attribute. The attribute will be used as
-   * an id. It is formally equivalent to calling FieldPosition(field, -1).
+   * an id.
    *
    * @param field The field format attribute.
    */
   public FieldPosition (Format.Field field)
   {
-    this(field, -1);
+    this.field_attribute = field;
   }
 
   /**
