--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RMIClassLoader.java	2004-11-17 02:24:08.000000000 -0700
+++ java/rmi/server/RMIClassLoader.java	2004-11-07 02:15:45.000000000 -0700
@@ -55,11 +55,6 @@
  */
 public class RMIClassLoader
 {
-  /**
-   * This class isn't intended to be instantiated.
-   */
-  private RMIClassLoader() {}
-
   private static class MyClassLoader extends URLClassLoader
   {
     // Package-private to avoid a trampoline constructor.
