--- /home/tromey/gnu/classpath/classpath/java/nio/channels/spi/AbstractSelectableChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/spi/AbstractSelectableChannel.java	Tue Dec 10 19:20:33 2002
@@ -75,7 +75,6 @@
    * Adjusts this channel's blocking mode.
    */
   public final SelectableChannel configureBlocking (boolean block)
-    throws IOException
   {
     synchronized (LOCK)
       {
@@ -91,7 +90,7 @@
    *
    * @exception IOException If an error occurs
    */
-  protected final void implCloseChannel () throws IOException
+  protected final void implCloseChannel ()
   {
     implCloseSelectableChannel ();
   }
@@ -99,13 +98,12 @@
   /**
    * Closes this selectable channel.
    */
-  protected abstract void implCloseSelectableChannel () throws IOException;
+  protected abstract void implCloseSelectableChannel ();
   
   /**
    * Adjusts this channel's blocking mode.
    */
-  protected abstract void implConfigureBlocking (boolean block)
-    throws IOException;
+  protected abstract void implConfigureBlocking (boolean block);
 
   /**
    * Tells whether or not every I/O operation on this channel will block
