--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/filechooser/FileSystemView.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/filechooser/FileSystemView.java	2003-06-13 02:27:45.000000000 -0600
@@ -1,5 +1,5 @@
 /* FileSystemView.java --
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,6 +40,7 @@
 
 import java.io.File;
 import java.io.IOException;
+import javax.swing.Icon;
 
 /**
  * FileSystemView
