--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/InputStream.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/InputStream.java	2002-01-22 16:01:33.000000000 -0700
@@ -105,7 +105,7 @@
    * @param readLimit The number of bytes that can be read before the
    *                  mark becomes invalid
    */
-  public void mark(int readLimit)
+  public void mark(int readlimit)
   {
     // Do nothing
   }
