--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/Socket.java	2004-07-21 02:19:57.000000000 -0600
+++ java/net/Socket.java	2004-07-18 02:18:49.000000000 -0600
@@ -311,7 +311,8 @@
     // that default.  JDK 1.2 doc infers not to do a bind.
   }
 
-  private SocketImpl getImpl() throws SocketException
+  // This has to be accessible from java.net.ServerSocket.
+  SocketImpl getImpl() throws SocketException
   {
     try
       {
