--- /home/tromey/gnu/classpath/classpath/java/awt/dnd/DragSourceAdapter.java	Mon Sep 30 13:37:10 2002
+++ java/awt/dnd/DragSourceAdapter.java	Thu Aug  8 22:26:15 2002
@@ -108,7 +108,7 @@
    *
    * @param e the event
    */
-  public void dragExit(DragSourceEvent e)
+  public void dragExit(DragSourceDragEvent e)
   {
   }
 
@@ -120,7 +120,7 @@
    *
    * @param e the event
    */
-  public void dragDropEnd(DragSourceDropEvent e)
+  public void dragDropEnd(DragSourceDragEvent e)
   {
   }
 } // class DragSourceAdapter
