--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-04-18 02:22:12.000000000 -0600
+++ java/awt/Component.java	2004-04-17 02:18:07.000000000 -0600
@@ -105,7 +105,7 @@
  * in inner classes, rather than using this object itself as the listener, if
  * external objects do not need to save the state of this object.
  *
- * <pre>
+ * <p><pre>
  * import java.awt.*;
  * import java.awt.event.*;
  * import java.io.Serializable;
@@ -127,7 +127,6 @@
  *     aButton.addActionListener(new MyActionListener());
  *   }
  * }
- * </pre>
  *
  * <p>Status: Incomplete. The event dispatch mechanism is implemented. All
  * other methods defined in the J2SE 1.3 API javadoc exist, but are mostly
