--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Key.java	2002-03-29 01:30:51.000000000 -0700
+++ java/security/Key.java	2003-05-04 02:19:12.000000000 -0600
@@ -62,7 +62,7 @@
 public interface Key extends Serializable
 {
   /**
-   * The verion identifier used for serialization.
+   * The version identifier used for serialization.
    */
   long serialVersionUID = 6603384152749567654L;
 
