--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/SignatureSpi.java	2004-04-18 02:22:22.000000000 -0600
+++ java/security/SignatureSpi.java	2003-04-20 02:20:20.000000000 -0600
@@ -279,7 +279,7 @@
    * @param param the string name of the parameter.
    * @return the object that represents the parameter value, or <code>null</code>
    * if there is none.
-   * @throws InvalidParameterException if <code>param</code> is an invalid
+   * @throws InvalidParameterException if <code>param<?code> is an invalid
    * parameter for this engine, or another exception occurs while trying to get
    * this parameter.
    * @deprecated
