--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/MenuItem.java	2004-01-03 02:20:32.000000000 -0700
+++ java/awt/MenuItem.java	2004-01-20 02:18:20.000000000 -0700
@@ -1,5 +1,5 @@
 /* MenuItem.java -- An item in a menu
-   Copyright (C) 1999, 2000, 2001, 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000, 2001, 2002, 2003, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -361,7 +361,7 @@
 public void
 addNotify()
 {
-  if (peer != null)
+  if (peer == null)
     peer = getToolkit ().createMenuItem (this);
 }
 
