--- /home/tromey/gnu/egcs/classpath/classpath/java/util/ArrayList.java	Tue Jan 22 15:50:06 2002
+++ java/util/ArrayList.java	Tue Jan 22 16:01:39 2002
@@ -71,6 +71,7 @@
  * non-deterministic behavior.
  *
  * @author Jon A. Zeppieri
+ * @author Bryce McKinlay
  * @author Eric Blake <ebb9@email.byu.edu>
  * @see Collection
  * @see List
@@ -463,7 +464,7 @@
     // use of a negative index will cause an ArrayIndexOutOfBoundsException,
     // a subclass of the required exception, with no effort on our part.
     if (index > size)
-      throw new IndexOutOfBoundsException("Index: " + index + ", Size:"
+      throw new IndexOutOfBoundsException("Index: " + index + ", Size: "
                                           + size);
   }
 
@@ -479,7 +480,7 @@
     // use of a negative index will cause an ArrayIndexOutOfBoundsException,
     // a subclass of the required exception, with no effort on our part.
     if (index >= size)
-      throw new IndexOutOfBoundsException("Index: " + index + ", Size:"
+      throw new IndexOutOfBoundsException("Index: " + index + ", Size: "
                                           + size);
   }
 
