--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileOutputStream.java	2003-04-22 02:26:41.000000000 -0600
+++ java/io/FileOutputStream.java	2003-03-28 02:18:03.000000000 -0700
@@ -84,11 +84,9 @@
     SecurityManager s = System.getSecurityManager();
     if (s != null)
       s.checkWrite(path);
-
-    int flags = FileDescriptor.WRITE;
-    if (append)
-      flags |= FileDescriptor.APPEND;
-    fd = new FileDescriptor(path, flags);
+    fd = new FileDescriptor (path, (append
+				    ? FileDescriptor.APPEND
+				    : FileDescriptor.WRITE));
   }
 
   /**
@@ -179,6 +177,11 @@
   public FileOutputStream (FileDescriptor fdObj)
     throws SecurityException
   {
+    // Hmm, no other exception but this one to throw, but if the descriptor
+    // isn't valid, we surely don't have "permission" to write to it.
+    if (!fdObj.valid())
+      throw new SecurityException("Invalid FileDescriptor");
+
     SecurityManager s = System.getSecurityManager();
     if (s != null)
       s.checkWrite(fdObj);
@@ -230,7 +233,7 @@
   public void write (byte[] buf)
     throws IOException
   {
-    write (buf, 0, buf.length);
+    fd.write (buf, 0, buf.length);
   }
 
   /**
