--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/CloneNotSupportedException.java	Mon Sep 10 21:24:13 2001
+++ java/lang/CloneNotSupportedException.java	Sat Sep  1 23:13:33 2001
@@ -36,42 +36,36 @@
  */
 
 /**
- * Thrown to indicate an object should not or could not be cloned. This
- * includes the case when {@link Object#clone()} is called on an object
- * which does not implement the {@link Cloneable} interface.
- * <p>
- *
- * Notice that calling <code>clone()</code> on an array will never produce
- * this exception, as the VM will always succeed in copying the array, or
- * cause an OutOfMemoryError first.
+ * Exceptions may be thrown by one part of a Java program and caught
+ * by another in order to deal with exceptional conditions.  
+ * Thrown to indicate an object should not or could not be cloned.  
+ * For example <code>CloneNotSupportedException</code> is thrown by
+ * the <code>clone</code> method of <code>Object</code> to indicate 
+ * that object does not implement the <code>Cloneable</code> interface.
  *
+ * @since JDK 1.0
+ * 
  * @author Brian Jones
  * @author Warren Levy <warrenl@cygnus.com>
- * @author Eric Blake <ebb9@email.byu.edu>
- * @since 1.0
- * @see Cloneable
- * @see Object#clone()
+ * @date September 18, 1998.
  */
 public class CloneNotSupportedException extends Exception
 {
-  /**
-   * compatible with JDK 1.0+
-   */
-  private static final long serialVersionUID = 5195511250079656443L;
+  static final long serialVersionUID = 5195511250079656443L;
 
   /**
    * Create an exception without a message.
    */
   public CloneNotSupportedException()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an exception with a message.
-   * @param s the error message
    */
   public CloneNotSupportedException(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
