--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-06-08 02:24:44.000000000 -0600
+++ java/awt/Component.java	2004-06-18 02:18:50.000000000 -0600
@@ -941,7 +941,7 @@
   {
     if (foreground != null)
       return foreground;
-    return parent == null ? null : parent.getForeground();
+    return parent == null ? SystemColor.windowText : parent.getForeground();
   }
 
   /**
@@ -982,7 +982,7 @@
   {
     if (background != null)
       return background;
-    return parent == null ? null : parent.getBackground();
+    return parent == null ? SystemColor.window : parent.getBackground();
   }
 
   /**
@@ -1031,7 +1031,7 @@
     if (parent != null)
       return parent.getFont ();
     else
-      return new Font ("Fixed", Font.PLAIN, 12);
+      return new Font ("Dialog", Font.PLAIN, 12);
   }
 
   /**
