--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/DefaultTableCellRenderer.java	2004-04-28 02:20:18.000000000 -0600
+++ javax/swing/table/DefaultTableCellRenderer.java	2004-01-29 02:18:52.000000000 -0700
@@ -54,7 +54,7 @@
 {
   static final long serialVersionUID = 7878911414715528324L;
 
-  protected static Border noFocusBorder = new EmptyBorder(0, 0, 0, 0);
+  protected static Border noFocusBorder;
 
   public static class UIResource extends DefaultTableCellRenderer
     implements javax.swing.plaf.UIResource
@@ -70,6 +70,7 @@
   public DefaultTableCellRenderer()
   {
     super();
+    this.noFocusBorder = new EmptyBorder(0, 0, 0, 0);
   }
 
   /**
