--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/PasswordAuthentication.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/PasswordAuthentication.java	2003-06-13 02:27:36.000000000 -0600
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,7 +37,6 @@
 
 package java.net;
 
-
 /**
   * This class serves a container for username/password pairs.
   *
@@ -51,7 +50,7 @@
    */
 
   /**
-    * The username
+    * The username 
     */
   private String username;
 
@@ -67,7 +66,7 @@
    */
 
   /**
-    * Creates a new <code>PasswordAuthentication</code> object from the
+    * Creates a new <code>PasswordAuthentication</code> object from the 
     * specified username and password.
     *
     * @param username The username for this object
@@ -92,9 +91,9 @@
     */
   public String getUserName()
   {
-    return (username);
+    return(username);
   }
-
+   
   /**
     * Returns the password associated with this object
     *
@@ -102,6 +101,8 @@
     */
   public char[] getPassword()
   {
-    return (password);
+    return(password);
   }
+
 } // class PasswordAuthentication
+
