--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/SetOfIntegerSyntax.java	2004-04-28 02:20:18.000000000 -0600
+++ javax/print/attribute/SetOfIntegerSyntax.java	2004-04-21 02:18:12.000000000 -0600
@@ -1,5 +1,5 @@
 /* SetOfIntegerSyntax.java -- 
-   Copyright (C) 2003, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -199,7 +199,7 @@
    *
    * @param obj the object to test
    *
-   * @return true if both objects are equal, false otherwise.
+   * @returns true if both objects are equal, false otherwise.
    */
   public boolean equals(Object obj)
   {
