--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/StringWriter.java	2004-04-13 02:20:22.000000000 -0600
+++ java/io/StringWriter.java	2003-02-04 13:42:48.000000000 -0700
@@ -60,8 +60,6 @@
   /**
    * This method closes the stream.  The contents of the internal buffer
    * can still be retrieved, but future writes are not guaranteed to work.
-   *
-   * @excepttion IOException If an error orrurs.
    */
   public void close () throws IOException
   {
