--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultBoundedRangeModel.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/DefaultBoundedRangeModel.java	2004-01-08 02:18:18.000000000 -0700
@@ -203,7 +203,7 @@
    * Changes the current value of this bounded range model. In a
    * scroll bar visualization of a {@link BoundedRangeModel}, the
    * <code>value</code> is displayed as the position of the thumb;
-   * changing the <code>value</code> of a scroll bar's model
+   * changing the <code>value</code> of a scroll bar&#x2019;s model
    * thus moves the thumb to a different position.
    */
   public void setValue(int value)
