--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/SocketPermission.java	2004-10-16 02:22:18.000000000 -0600
+++ java/net/SocketPermission.java	2004-04-21 02:18:10.000000000 -0600
@@ -151,13 +151,13 @@
   public boolean equals(Object obj)
   {
     if (! (obj instanceof SocketPermission))
-      return false;
+      return (false);
 
     if (((SocketPermission) obj).hostport.equals(hostport))
       if (((SocketPermission) obj).actions.equals(actions))
-	return true;
+	return (true);
 
-    return false;
+    return (false);
   }
 
   /**
@@ -216,7 +216,7 @@
     else if (actions.indexOf("resolve") != -1)
       sb.append("resolve");
 
-    return sb.toString();
+    return (sb.toString());
   }
 
   /**
@@ -229,7 +229,7 @@
   {
     // FIXME: Implement
 
-    return null;
+    return (null);
   }
 
   /**
@@ -265,14 +265,14 @@
     if (perm instanceof SocketPermission)
       p = (SocketPermission) perm;
     else
-      return false;
+      return (false);
 
     // Next check the actions
     String ourlist = getActions();
     String theirlist = p.getActions();
 
     if (! ourlist.startsWith(theirlist))
-      return false;
+      return (false);
 
     // Now check ports
     int ourfirstport = 0;
@@ -347,7 +347,7 @@
 
     // Now check them
     if ((theirfirstport < ourfirstport) || (theirlastport > ourlastport))
-      return false;
+      return (false);
 
     // Finally we can check the hosts
     String ourhost;
@@ -369,7 +369,7 @@
 
     // Are they equal?
     if (ourhost.equals(theirhost))
-      return true;
+      return (true);
 
     // Try the canonical names
     String ourcanonical = null;
@@ -392,17 +392,17 @@
       }
 
     if (ourcanonical.equals(theircanonical))
-      return true;
+      return (true);
 
     // Well, last chance.  Try for a wildcard
     if (ourhost.indexOf("*.") != -1)
       {
 	String wild_domain = ourhost.substring(ourhost.indexOf("*" + 1));
 	if (theircanonical.endsWith(wild_domain))
-	  return true;
+	  return (true);
       }
 
     // Didn't make it
-    return false;
+    return (false);
   }
 }
