--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/AlgorithmParameterGenerator.java	2003-04-24 02:22:11.000000000 -0600
+++ java/security/AlgorithmParameterGenerator.java	2003-04-20 02:20:19.000000000 -0600
@@ -39,8 +39,6 @@
 
 import java.security.spec.AlgorithmParameterSpec;
 
-import gnu.java.security.Engine;
-
 /**
  * <p>The <code>AlgorithmParameterGenerator</code> class is used to generate a
  * set of parameters to be used with a certain algorithm. Parameter generators
@@ -203,11 +201,7 @@
 	  (AlgorithmParameterGeneratorSpi) Engine.getInstance(
 	    ALGORITHM_PARAMETER_GENERATOR, algorithm, provider),
 	  provider, algorithm);
-      }
-    catch (java.lang.reflect.InvocationTargetException ite)
-      {
-	throw new NoSuchAlgorithmException(algorithm);
-      }
+        }
     catch (ClassCastException cce)
       {
 	throw new NoSuchAlgorithmException(algorithm);
