--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/BufferedReader.java	2004-03-13 02:21:03.000000000 -0700
+++ java/io/BufferedReader.java	2004-01-30 02:18:16.000000000 -0700
@@ -112,7 +112,7 @@
    * @param in The subordinate stream to read from
    * @param size The buffer size to use
    *
-   * @exception IllegalArgumentException if size &lt;= 0
+   * @exception IllegalArgumentException if size &lt;&eq; 0
    */
   public BufferedReader(Reader in, int size)
   {
