--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectInputStream.java	2004-01-09 02:20:36.000000000 -0700
+++ java/io/ObjectInputStream.java	2004-01-09 02:18:04.000000000 -0700
@@ -1728,7 +1728,11 @@
    * @param sm SecurityManager instance which should be called.
    * @return The current class loader in the calling stack.
    */
-  private static native ClassLoader currentClassLoader (SecurityManager sm);
+  private static ClassLoader currentClassLoader (SecurityManager sm)
+  {
+    // FIXME: This is too simple.
+    return ClassLoader.getSystemClassLoader ();
+  }
 
   /**
    * This method tries to access a precise field called in the
