--- /home/tromey/gnu/egcs/classpath/classpath/java/util/ArrayList.java	Fri Oct 26 16:06:57 2001
+++ java/util/ArrayList.java	Sat Dec 15 09:15:59 2001
@@ -60,6 +60,7 @@
  * non-deterministic behavior.
  *
  * @author Jon A. Zeppieri
+ * @author Bryce McKinlay
  * @author Eric Blake <ebb9@email.byu.edu>
  * @see Collection
  * @see List
@@ -452,7 +453,7 @@
     // use of a negative index will cause an ArrayIndexOutOfBoundsException,
     // a subclass of the required exception, with no effort on our part.
     if (index > size)
-      throw new IndexOutOfBoundsException("Index: " + index + ", Size:"
+      throw new IndexOutOfBoundsException("Index: " + index + ", Size: "
                                           + size);
   }
 
@@ -468,7 +469,7 @@
     // use of a negative index will cause an ArrayIndexOutOfBoundsException,
     // a subclass of the required exception, with no effort on our part.
     if (index >= size)
-      throw new IndexOutOfBoundsException("Index: " + index + ", Size:"
+      throw new IndexOutOfBoundsException("Index: " + index + ", Size: "
                                           + size);
   }
 
