--- /home/tromey/gnu/egcs/classpath/classpath/java/util/EventListener.java	Fri Feb 22 16:38:24 2002
+++ java/util/EventListener.java	Tue Jan 22 16:01:40 2002
@@ -1,5 +1,5 @@
-/* EventListener.java -- tagging interface for all event listeners
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* An interface that all event listener interfaces must extend
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,6 +38,11 @@
 
 package java.util;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * Status:  Believed complete and correct.
+ */
+
 /**
  * Empty interface that is implemented by classes that need to receive
  * events. Subinterfaces define methods that can be called to fire an
@@ -46,8 +51,7 @@
  * take as argument an subclass of <code>EventObject</code>.
  *
  * @author Tom Tromey <tromey@cygnus.com>
- * @see EventObject
- * @status updated to 1.4
+ * @date December 12, 1998
  */
 public interface EventListener
 {
