--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/IdentityScope.java	2004-04-18 02:22:22.000000000 -0600
+++ java/security/IdentityScope.java	2003-05-10 02:19:17.000000000 -0600
@@ -123,7 +123,7 @@
   }
 
   /**
-   * Sets the system's identity scope.
+   * <p>Sets the system's identity scope.</p>
    *
    * <p>First, if there is a security manager, its <code>checkSecurityAccess()
    * </code> method is called with <code>"setSystemScope"</code> as its argument
@@ -131,7 +131,7 @@
    *
    * @param scope the scope to set.
    * @throws SecurityException if a security manager exists and its
-   * <code>checkSecurityAccess()</code> method doesn't allow setting the
+   * <code>checkSecurityAccess()<code> method doesn't allow setting the
    * identity scope.
    * @see #getSystemScope()
    * @see SecurityManager#checkSecurityAccess(String)
