--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/SecurityManager.java	2003-08-01 02:19:05.000000000 -0600
+++ java/lang/SecurityManager.java	2004-01-06 02:18:20.000000000 -0700
@@ -187,8 +187,7 @@
    */
   protected ClassLoader currentClassLoader()
   {
-    Class c = currentLoadedClass();
-    return c != null ? c.getClassLoader() : null;
+    return VMSecurityManager.currentClassLoader();
   }
 
   /**
@@ -1023,8 +1022,9 @@
              index != -1; index = list.indexOf(packageName, index + 1))
           {
             // Exploit package visibility for speed.
-            if (index + packageName.length() == list.length()
-                || list.charAt(index + packageName.length()) == ',')
+	    int packageNameCount = packageName.length();
+            if (index + packageNameCount == list.length()
+                || list.charAt(index + packageNameCount) == ',')
               {
                 checkPermission(p);
                 return;
@@ -1043,4 +1043,3 @@
 		this.classes = classes;
 	}
 }
-
