--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Frame.java	2003-06-16 02:23:41.000000000 -0600
+++ java/awt/Frame.java	2003-09-03 02:17:57.000000000 -0600
@@ -237,12 +237,15 @@
 {
   super();
   this.title = title;
+  // Top-level frames are initially invisible.
+  visible = false;
 }
 
 public
 Frame(GraphicsConfiguration gc)
 {
   super(gc);
+  visible = false;
 }
 
 public
@@ -250,6 +253,7 @@
 {
   super(gc);
   setTitle(title);
+  visible = false;
 }
 
 /*************************************************************************/
