--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/GregorianCalendar.java	2003-11-26 02:20:09.000000000 -0700
+++ java/util/GregorianCalendar.java	2003-12-31 02:18:37.000000000 -0700
@@ -402,7 +402,11 @@
       {
 	hour = fields[HOUR];
         if (isSet[AM_PM] && fields[AM_PM] == PM)
-	  hour += 12;
+	  if (hour != 12) /* not Noon */
+            hour += 12;
+	/* Fix the problem of the status of 12:00 AM (midnight). */
+	if (isSet[AM_PM] && fields[AM_PM] == AM && hour == 12)
+	  hour = 0;
       }
 
     int minute = isSet[MINUTE] ? fields[MINUTE] : 0;
@@ -606,7 +610,7 @@
 	calculateDay(++day, gregorian);
       }
 
-    fields[DAY_OF_WEEK_IN_MONTH] = (fields[DAY_OF_MONTH] + 12) / 7;
+    fields[DAY_OF_WEEK_IN_MONTH] = (fields[DAY_OF_MONTH] + 6) / 7;
 
     // which day of the week are we (0..6), relative to getFirstDayOfWeek
     int relativeWeekday = (7 + fields[DAY_OF_WEEK] - getFirstDayOfWeek()) % 7;
