--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/PropertyPermission.java	Sun Mar 23 02:18:41 2003
+++ java/util/PropertyPermission.java	Mon Jun 17 23:03:38 2002
@@ -121,7 +121,7 @@
     super(name);
     if (actions == null)
       throw new IllegalArgumentException();
-    setActions(actions);
+    setActions(actions.toLowerCase());
   }
 
   /**
@@ -134,37 +134,14 @@
    */
   private void setActions(String str)
   {
-    // Initialising the class java.util.Locale ...
-    //    tries to initialise the Locale.defaultLocale static
-    //    which calls System.getProperty, 
-    //    which calls SecurityManager.checkPropertiesAccess,
-    //    which creates a PropertyPermission with action "read,write",
-    //    which calls setActions("read,write").
-    // If we now were to call toLowerCase on 'str',
-    //    this would call Locale.getDefault() which returns null
-    //       because Locale.defaultLocale hasn't been set yet
-    //    then toLowerCase will fail with a null pointer exception.
-    // 
-    // The solution is to take a punt on 'str' being lower case, and
-    // test accordingly.  If that fails, we convert 'str' to lower case 
-    // and try the tests again.
     if ("read".equals(str))
       actions = READ;
     else if ("write".equals(str))
       actions = WRITE;
     else if ("read,write".equals(str) || "write,read".equals(str))
       actions = READ | WRITE;
-    else {
-      String lstr = str.toLowerCase();
-      if ("read".equals(lstr))
-	actions = READ;
-      else if ("write".equals(lstr))
-	actions = WRITE;
-      else if ("read,write".equals(lstr) || "write,read".equals(lstr))
-	actions = READ | WRITE;
-      else
-	throw new IllegalArgumentException("illegal action " + str);
-    }
+    else
+      throw new IllegalArgumentException("illegal action " + str);
   }
 
   /**
