--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Dialog.java	2003-09-26 02:21:09.000000000 -0600
+++ java/awt/Dialog.java	2004-01-06 02:18:20.000000000 -0700
@@ -187,7 +187,8 @@
 {
   super (parent, gc);
 
-  this.title = title;
+  // A null title is equivalent to an empty title  
+  this.title = (title != null) ? title : "";
   this.modal = modal;
   visible = false;
 
@@ -254,8 +255,9 @@
 Dialog (Dialog parent, String title, boolean modal, GraphicsConfiguration gc)
 {
   super (parent, parent.getGraphicsConfiguration ());
-  
-  this.title = title;
+
+  // A null title is equivalent to an empty title  
+  this.title = (title != null) ? title : "";
   this.modal = modal;
   visible = false;
 
@@ -289,7 +291,9 @@
 public synchronized void
 setTitle(String title)
 {
-  this.title = title;
+  // A null title is equivalent to an empty title  
+  this.title = (title != null) ? title : "";
+
   if (peer != null)
     {
       DialogPeer d = (DialogPeer) peer;
