--- /home/tromey/gnu/egcs/classpath/classpath/java/util/jar/Attributes.java	Fri Nov 17 19:29:20 2000
+++ java/util/jar/Attributes.java	Fri Nov  2 21:44:52 2001
@@ -45,7 +45,7 @@
  * manipulating the Attributes more or less type safe.
  * <p>
  * Most of the methods are wrappers to implement the Map interface. The really
- * usefull and often used methods are <code>getValue(Name)</code> and
+ * useful and often used methods are <code>getValue(Name)</code> and
  * <code>getValue(String)</code>. If you actually want to set attributes you
  * may want to use the <code>putValue(String, String)</code> method
  * (sorry there is no public type safe <code>putValue(Name, String)</code>
