--- /home/tromey/gnu/classpath/classpath/java/awt/ContainerOrderFocusTraversalPolicy.java	Wed Oct  2 11:21:37 2002
+++ java/awt/ContainerOrderFocusTraversalPolicy.java	Thu Aug  8 22:26:14 2002
@@ -46,8 +46,6 @@
 public class ContainerOrderFocusTraversalPolicy extends FocusTraversalPolicy
   implements Serializable
 {
-  static final long serialVersionUID = 486933713763926351L;
-
   private boolean downCycle = true;
 
   public ContainerOrderFocusTraversalPolicy()
