--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/StringReader.java	Mon Mar 10 02:19:33 2003
+++ java/io/StringReader.java	Tue Jan 22 16:01:33 2002
@@ -1,5 +1,5 @@
 /* StringReader.java -- permits a String to be read as a character input stream
-   Copyright (C) 1998, 1999, 2000, 2003  Free Software Foundation
+   Copyright (C) 1998, 1999, 2000  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -51,6 +51,8 @@
  * normal.  If no mark has been set, then calling the <code>reset()</code>
  * method rewinds the read pointer to the beginning of the <code>String</code>.
  *
+ * @version 0.0
+ *
  * @author Aaron M. Renn (arenn@urbanophile.com)
  * @author Warren Levy <warrenl@cygnus.com>
  * @date October 19, 1998.  
@@ -71,8 +73,8 @@
 
   /**
    * Create a new <code>StringReader</code> that will read chars from the 
-   * passed in <code>String</code>.  This stream will read from the beginning 
-   * to the end of the <code>String</code>.
+   * passed in <code>String</code>.  This stream will read from the beginning to the 
+   * end of the <code>String</code>.
    *
    * @param s The <code>String</code> this stream will read from.
    */
@@ -206,4 +208,3 @@
     }
   }
 }
-
