--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JFrame.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JFrame.java	2004-04-21 02:18:12.000000000 -0600
@@ -45,10 +45,10 @@
 import java.awt.Dimension;
 import java.awt.Frame;
 import java.awt.Graphics;
+import java.awt.GraphicsConfiguration;
 import java.awt.LayoutManager;
 import java.awt.event.KeyEvent;
 import java.awt.event.WindowEvent;
-
 import javax.accessibility.AccessibleContext;
 
 /**
