--- /home/tromey/gnu/classpath/classpath/java/rmi/StubNotFoundException.java	Sat Mar  2 12:13:14 2002
+++ java/rmi/StubNotFoundException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* StubNotFoundException.java -- thrown if a valid stub is not found
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,41 +37,15 @@
 
 package java.rmi;
 
-/**
- * Thrown if a valid stub class is not found for an object when it is exported.
- *
- * @author unknown
- * @see UnicastRemoteObject
- * @see Activatable
- * @since 1.1
- * @status updated to 1.4
- */
-public class StubNotFoundException extends RemoteException
-{
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = -7088199405468872373L;
-
-
-  /**
-   * Create an exception with a message.
-   *
-   * @param s the message
-   */
-  public StubNotFoundException(String s)
-  {
-    super(s);
-  }
-
-  /**
-   * Create an exception with a message and a cause.
-   *
-   * @param s the message
-   * @param e the cause
-   */
-  public StubNotFoundException(String s, Exception e)
-  {
-    super(s, e);
-  }
+public class StubNotFoundException
+	extends RemoteException {
+
+public StubNotFoundException(String s) {
+	super(s);
+}
+
+public StubNotFoundException(String s, Exception e) {
+	super(s, e);
+}
+
 }
