--- /home/tromey/gnu/egcs/classpath/classpath/java/security/spec/InvalidParameterSpecException.java	Sat Mar  2 12:13:21 2002
+++ java/security/spec/InvalidParameterSpecException.java	Tue Jan 22 16:01:37 2002
@@ -1,5 +1,5 @@
-/* InvalidParameterSpecException.java --- invalid ParameterSpec Exception
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+/* InvalidParameterSpecException.java --- Invalid ParameterSpec Exception
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,40 +37,34 @@
 
 
 package java.security.spec;
-
 import java.security.GeneralSecurityException;
 
 /**
- * Exception for an invalid algorithm specification.
- *
- * @author Mark Benvenuto
- * @see AlogorithmParameters
- * @see AlogorithmParameterSpec
- * @see DSAParameterSpec
- * @since 1.2
- * @status updated to 1.4
+	Exception for an invalid algorithm specification.
+
+	@since JDK 1.2
+
+	@author Mark Benvenuto
 */
 public class InvalidParameterSpecException extends GeneralSecurityException
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = -970468769593399342L;
 
   /**
-   * Constructs an InvalidParameterSpecException without a message string.
-   */
+     Constructs an InvalidParameterSpecException without a message string.
+  */
   public InvalidParameterSpecException() 
   {
+    super();
   }
 
   /**
-   * Constructs an InvalidParameterSpecException with a message string.
-   *
-   * @param msg a message to display with exception
-   */
-  public InvalidParameterSpecException(String msg)
+     Constructs an InvalidParameterSpecException with a message string.
+
+     @param msg A message to display with exception
+  */
+  public InvalidParameterSpecException(String msg) 
   {
-    super(msg);
+    super( msg );
   }
+
 }
