--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipInputStream.java	2004-10-28 02:27:40.000000000 -0600
+++ java/util/zip/ZipInputStream.java	2003-06-18 02:22:33.000000000 -0600
@@ -1,5 +1,5 @@
-/* ZipInputStream.java --
-   Copyright (C) 2001, 2002, 2003, 2004  Free Software Foundation, Inc.
+/* java.util.zip.ZipInputStream
+   Copyright (C) 2001, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,8 +39,8 @@
 package java.util.zip;
 
 import java.io.EOFException;
-import java.io.IOException;
 import java.io.InputStream;
+import java.io.IOException;
 
 /**
  * This is a FilterInputStream that reads the files in an zip archive
@@ -106,7 +106,7 @@
       }
   }
   
-  private int readLeByte() throws IOException
+  private final int readLeByte() throws IOException
   {
     if (avail <= 0)
       {
@@ -120,7 +120,7 @@
   /**
    * Read an unsigned short in little endian byte order.
    */
-  private int readLeShort() throws IOException 
+  private final int readLeShort() throws IOException 
   {
     return readLeByte() | (readLeByte() << 8);
   }
@@ -128,7 +128,7 @@
   /**
    * Read an int in little endian byte order.
    */
-  private int readLeInt() throws IOException 
+  private final int readLeInt() throws IOException 
   {
     return readLeShort() | (readLeShort() << 16);
   }
