--- /home/tromey/gnu/classpath/classpath/java/awt/Choice.java	Mon Dec 30 15:39:03 2002
+++ java/awt/Choice.java	Tue Dec 10 19:20:24 2002
@@ -85,17 +85,13 @@
  * Constructors
  */
 
-  /**
-   * Initializes a new instance of <code>Choice</code>.
-   *
-   * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-   * returns true
-   */
-  public Choice()
-  {
-    if (GraphicsEnvironment.isHeadless())
-      throw new HeadlessException ();
-  }
+/**
+  * Initializes a new instance of <code>Choice</code>.
+  */
+public
+Choice()
+{
+}
 
 /*************************************************************************/
 
@@ -150,16 +146,12 @@
   * Adds the specified item to this choice box.
   *
   * @param item The item to add.
-  *
-  * @exception NullPointerException If the item's value is null
-  *
-  * @since 1.1
   */
 public synchronized void
 add(String item)
 {
   if (item == null)
-    throw new NullPointerException ("item must be non-null");
+    throw new IllegalArgumentException ("item must be non-null");
 
   pItems.addElement(item);
 
@@ -179,12 +171,7 @@
 /**
   * Adds the specified item to this choice box.
   *
-  * This method is oboslete since Java 2 platform 1.1. Please use @see add
-  * instead.
-  *
   * @param item The item to add.
-  *
-  * @exception NullPointerException If the item's value is equal to null
   */
 public synchronized void
 addItem(String item)
@@ -202,15 +189,10 @@
  *
  * @param item The item to add.
  * @param index The index at which the item should be inserted.
- *
- * @exception IllegalArgumentException If index is less than 0
  */
 public synchronized void
 insert(String item, int index)
 {
-  if (index < 0)
-    throw new IllegalArgumentException ("index may not be less then 0");
-
   if (index > getItemCount ())
     index = getItemCount ();
 
