--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/table/TableColumn.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/table/TableColumn.java	2003-06-13 02:27:45.000000000 -0600
@@ -218,10 +218,9 @@
 	 * @param oldValue TODO
 	 * @param newValue TODO
 	 */
-	private void firePropertyChange(String property, boolean oldValue, boolean newValue)
-	{
+	private void firePropertyChange(String property, boolean oldValue, boolean newValue) {
 		firePropertyChange(property, new Boolean(oldValue), new Boolean(newValue));
-	}
+	} // firePropertyChange()
 
 	/**
 	 * setModelIndex
