--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Map.java	2004-04-03 02:21:03.000000000 -0700
+++ java/util/Map.java	2003-10-13 02:17:51.000000000 -0600
@@ -1,6 +1,6 @@
 /* Map.java: interface Map -- An object that maps keys to values
              interface Map.Entry -- an Entry in a Map
-   Copyright (C) 1998, 2001, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -264,7 +264,7 @@
    * @since 1.2
    * @status updated to 1.4
    */
-  interface Entry
+  static interface Entry
   {
     /**
      * Get the key corresponding to this entry.
