--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicBorders.java	2003-07-01 02:19:59.000000000 -0600
+++ javax/swing/plaf/basic/BasicBorders.java	2003-06-28 02:18:24.000000000 -0600
@@ -1300,7 +1300,7 @@
    * painted if the orientation of the enclosed JSplitPane is
    * <code>JSplitPane.VERTICAL_SPLIT</code> (at least in versions
    * 1.3.1 and 1.4.1).  GNU Classpath does not replicate this bug. A
-   * report has been filed with Sun (bug ID 4885629).
+   * report has been filed with Sun (review ID 188773).
    *
    * <p>Note that the bottom left pixel of the border has a different
    * color depending on the orientation of the enclosed JSplitPane.
