--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Color.java	2004-04-28 02:20:15.000000000 -0600
+++ java/awt/Color.java	2003-08-13 02:17:59.000000000 -0600
@@ -319,12 +319,7 @@
   {
     if ((red & 255) != red || (green & 255) != green || (blue & 255) != blue
         || (alpha & 255) != alpha)
-      throw new IllegalArgumentException("Bad RGB values"
-                                        +" red=0x"+Integer.toHexString(red)
-                                        +" green=0x"+Integer.toHexString(green)
-                                        +" blue=0x"+Integer.toHexString(blue)
-                                        +" alpha=0x"+Integer.toHexString(alpha)  );
-
+      throw new IllegalArgumentException("Bad RGB values");
     value = (alpha << 24) | (red << 16) | (green << 8) | blue;
     falpha = 1;
     cs = null;
@@ -955,7 +950,7 @@
    * object, regardless of the parameters. Subclasses, however, may have a
    * mutable result.
    *
-   * @param cm the requested color model
+   * @param cm the requested color model, ignored
    * @param deviceBounds the bounding box in device coordinates, ignored
    * @param userBounds the bounding box in user coordinates, ignored
    * @param xform the bounds transformation, ignored
@@ -967,8 +962,8 @@
                                     AffineTransform xform,
                                     RenderingHints hints)
   {
-    if (context == null || !context.getColorModel().equals(cm))
-      context = new ColorPaintContext(cm,value);
+    if (context == null)
+      context = new ColorPaintContext(value);
     return context;
   }
 
