--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Hashtable.java	2003-11-29 02:20:16.000000000 -0700
+++ java/util/Hashtable.java	2003-11-27 02:18:13.000000000 -0700
@@ -333,11 +333,7 @@
    */
   public synchronized boolean contains(Object value)
   {
-    /* delegate to non-overridable worker method 
-     * to avoid blowing up the stack, when called 
-     * from overridden contains[Value]() method.
-     */
-    return internalContainsValue(value);
+    return containsValue(value);
   }
 
   /**
@@ -354,25 +350,6 @@
    */
   public boolean containsValue(Object value)
   {
-    /* delegate to older method to make sure code overwriting it 
-     * continues to work.
-     */
-    return contains(value);
-  }
-
-  /**
-   * Returns true if this Hashtable contains a value <code>o</code>, such that
-   * <code>o.equals(value)</code>. This is an internal worker method
-   * called by <code>contains()</code> and <code>containsValue()</code>.
-   *
-   * @param value the value to search for in this Hashtable
-   * @return true if at least one key maps to the value
-   * @see #contains(Object)
-   * @see #containsKey(Object)
-   * @throws NullPointerException if <code>value</code> is null
-   */
-  private boolean internalContainsValue(Object value)
-  {
     for (int i = buckets.length - 1; i >= 0; i--)
       {
         HashEntry e = buckets[i];
