--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JViewport.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JViewport.java	2004-04-21 02:18:12.000000000 -0600
@@ -1,5 +1,5 @@
 /* JViewport.java -- 
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,12 +39,15 @@
 package javax.swing;
 
 import java.awt.Component;
+import java.awt.Container;
 import java.awt.Dimension;
 import java.awt.Graphics;
+import java.awt.Image;
 import java.awt.Insets;
 import java.awt.Point;
 import java.awt.Rectangle;
-
+import java.awt.image.ImageObserver;
+import javax.accessibility.Accessible;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
 import javax.swing.plaf.ViewportUI;
