--- /home/tromey/gnu/classpath/classpath/java/awt/PopupMenu.java	Mon Dec 30 15:39:04 2002
+++ java/awt/PopupMenu.java	Tue Jan 22 16:18:21 2002
@@ -65,9 +65,6 @@
 
 /**
   * Initializes a new instance of <code>PopupMenu</code>.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true.
   */
 public
 PopupMenu()
@@ -81,17 +78,11 @@
   * label.
   *
   * @param label The label for this popup menu.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless()
-  * returns true.
   */
 public
 PopupMenu(String label)
 {
   super(label);
-
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
