--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/im/InputContext.java	2004-05-21 02:26:17.000000000 -0600
+++ java/awt/im/InputContext.java	2003-07-12 02:17:58.000000000 -0600
@@ -86,20 +86,20 @@
   private static final ArrayList descriptors = new ArrayList();
   static
   {
-    Enumeration e;
+    Enumeration enum;
     try
       {
-        e = ClassLoader.getSystemResources
+        enum = ClassLoader.getSystemResources
           ("META_INF/services/java.awt.im.spi.InputMethodDescriptor");
       }
     catch (IOException ex)
       {
         // XXX Should we do something else?
-        e = EmptyEnumeration.getInstance();
+        enum = EmptyEnumeration.getInstance();
       }
-    while (e.hasMoreElements())
+    while (enum.hasMoreElements())
       {
-        URL url = (URL) e.nextElement();
+        URL url = (URL) enum.nextElement();
         BufferedReader in;
         String line;
         try
@@ -125,7 +125,7 @@
                   }
                 line = in.readLine().trim();
               }
-            catch (IOException ex)
+            catch (IOException e)
               {
                 continue outer;
               }
