--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/reflect/ReflectPermission.java	Sun Oct 21 10:39:27 2001
+++ java/lang/reflect/ReflectPermission.java	Mon Feb 19 15:25:11 2001
@@ -1,5 +1,5 @@
 /* ReflectPermission.java - named permission for reflaction
-   Copyright (C) 2000, 2001 Free Software Foundation, Inc.
+   Copyright (C) 2000, 2001  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -27,6 +27,7 @@
 /* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
  * "The Java Language Specification", ISBN 0-201-63451-1
  * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status: Believed complete and correct to version 1.2.
  */
 
 package java.lang.reflect;
@@ -35,57 +36,29 @@
 
 /**
  * This class implements permissions for reflection.  This is a named
- * permission, and the only defined name is suppressAccessChecks, which
- * allows suppression of normal Java objects when using reflection.
- *
- * <table>
- *  <tr>
- *   <th>Permission Target Name</th>
- *   <th>What Permission Allows</th>
- *   <th>Risk of Allowing Permission</th>
- *  </tr>
- *  <tr>
- *   <td><code>suppressAccessChecks</code></td>
- *   <td>Ability to access fields, invoke methods, and construct objects
- *       via reflection, including non-public members in contexts where
- *       such access is not legal at compile-time.</td>
- *   <td>This is dangerous. It exposes possibly confidential information,
- *       and malicious code could interfere with the internals of the Virtual
- *       Machine by corrupting private data.</td>
- *  </tr>
- * </table>
+ * permission, and the only defined name is suppressAccessChecks.
  *
  * @author Tom Tromey <tromey@redhat.com>
- * @author Eric Blake <ebb9@email.byu.edu>
- * @since 1.2
- * @status updated to 1.4
+ * @date November 18, 2000
  */
-public final class ReflectPermission
-  extends BasicPermission
+public final class ReflectPermission extends BasicPermission
 {
   /**
-   * Compatible with JDK 1.2.
-   */
-  private static final long serialVersionUID = 7412737110241507485L;
-
-  /**
    * Construct a ReflectPermission with the given name.
-   *
    * @param name The permission name
    */
-  public ReflectPermission(String name)
+  public ReflectPermission (String name)
   {
-    super(name);
+    super (name);
   }
 
   /**
    * Construct a ReflectPermission with the given name.
-   *
    * @param name The permission name
-   * @param actions The actions; this is ignored and should be null
+   * @param actions The actions; this is ignored and should be null.
    */
-  public ReflectPermission(String name, String actions)
+  public ReflectPermission (String name, String actions)
   {
-    super(name, actions);
+    super (name, actions);
   }
 }
