--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ListModel.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/ListModel.java	2004-03-18 02:18:22.000000000 -0700
@@ -42,7 +42,7 @@
  * This is an interface to general list-like data, typically used as the
  * model object of a {@link JList} component.
  *
- * @author Graydon Hoare (graydon@redhat.com)
+ * @author Graydon Hoare (graydon&064;redhat.com)
  */
 public interface ListModel
 {
