--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URISyntaxException.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/URISyntaxException.java	2002-08-21 12:14:01.000000000 -0600
@@ -37,7 +37,6 @@
 
 package java.net;
 
-
 /**
  * This exception is thrown when a String cannot be parsed as a URI.
  *
@@ -138,7 +137,7 @@
    */
   public String getMessage()
   {
-    return (super.getMessage() + (index >= 0 ? " at index " + index : "")
-           + ": " + input);
+    return super.getMessage() + (index >= 0 ? " at index " + index : "")
+      + ": " + input;
   }
 }
