--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/ObjID.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/server/ObjID.java	2002-01-22 16:01:36.000000000 -0700
@@ -1,5 +1,5 @@
-/* ObjID.java
-   Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -41,8 +41,10 @@
 import java.io.ObjectOutput;
 import java.io.ObjectInput;
 import java.io.IOException;
+import java.lang.Math;
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.util.Random;
 
 public final class ObjID
 	implements Serializable {
