--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/MappedByteBufferImpl.java	2004-02-18 02:21:29.000000000 -0700
+++ java/nio/MappedByteBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -40,7 +40,7 @@
 
 import java.io.IOException;
 import java.nio.channels.FileChannelImpl;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 public class MappedByteBufferImpl extends MappedByteBuffer
 {
