--- /home/tromey/gnu/classpath/classpath/java/awt/event/HierarchyBoundsListener.java	Sun Mar 31 23:08:05 2002
+++ java/awt/event/HierarchyBoundsListener.java	Tue Jan 22 16:01:31 2002
@@ -1,5 +1,4 @@
-/* HierarchyBoundsListener.java -- listens to bounds changes of parents
-   Copyright (C) 2000, 2002 Free Software Foundation
+/* Copyright (C) 2000, 2002  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -35,36 +34,17 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package java.awt.event;
 
-import java.util.EventListener;
-
 /**
- * This listens for changes in an ancestors size or location. Normally it is
- * not necessary to process these events since the AWT handles them
- * internally, taking all appropriate actions. To watch a subset of these
- * events, use a HierarchyBoundsAdapter.
- *
- * @author Bryce McKinlay
- * @see HierarchyBoundsAdapter
- * @see HierarchyEvent
  * @since 1.3
- * @status updated to 1.4
+ * @author Bryce McKinlay
  */
-public interface HierarchyBoundsListener extends EventListener
+
+/* Status:  Believed complete and correct. */
+
+public interface HierarchyBoundsListener extends java.util.EventListener
 {
-  /**
-   * Called when an ancestor component of the source is moved.
-   *
-   * @param e the event describing the ancestor's motion
-   */
-  void ancestorMoved(HierarchyEvent e);
-
-  /**
-   * Called when an ancestor component is resized.
-   *
-   * @param e the event describing the ancestor's resizing
-   */
-  void ancestorResized(HierarchyEvent e);
-} // interface HierarchyBoundsListener
+  public void ancestorMoved(HierarchyEvent e);
+  public void ancestorResized(HierarchyEvent e);
+}
