--- /home/tromey/gnu/egcs/classpath/classpath/java/security/KeyStoreException.java	Sat Mar  2 12:13:17 2002
+++ java/security/KeyStoreException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
 /* KeyStoreException.java -- Indicates a problem with the key store
-   Copyright (C) 1998, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,28 +40,26 @@
 /**
  * Indicates a problem with the key store.
  *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @since 1.2
- * @status updated to 1.4
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
  */
 public class KeyStoreException extends GeneralSecurityException
 {
   /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = -1119353179322377262L;
-
-  /**
-   * Create a new instance detailed error message.
+   * This method initializes a new <code>KeyStoreException</code> with no
+   * detailed error message.
    */
   public KeyStoreException()
   {
+    super();
   }
 
   /**
-   * Create a new instance with a detailed error message.
+   * This method initializes a new <code>KeyStoreException</code> with a
+   * detailed error message.
    *
-   * @param msg the descriptive error message
+   * @param msg The descriptive error message.
    */
   public KeyStoreException(String msg)
   {
