--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Polygon.java	2004-04-13 02:20:17.000000000 -0600
+++ java/awt/Polygon.java	2004-02-04 02:18:37.000000000 -0700
@@ -294,7 +294,7 @@
             else if (y > maxy)
               maxy = y;
           }
-        bounds = new Rectangle (minx, miny, maxx - minx, maxy - miny);
+        bounds = new Rectangle (minx, maxy, maxx - minx, maxy - miny);
       }
     return bounds;
   }
