--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/Collator.java	2004-04-28 02:20:17.000000000 -0600
+++ java/text/Collator.java	2004-04-21 02:18:11.000000000 -0600
@@ -1,5 +1,5 @@
 /* Collator.java -- Perform locale dependent String comparisons.
-   Copyright (C) 1998, 1999, 2000, 2001, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,10 +38,10 @@
 
 package java.text;
 
-import java.util.Comparator;
 import java.util.Locale;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
+import java.util.Comparator;
 
 /**
  * This class is the abstract superclass of classes which perform 
@@ -380,12 +380,7 @@
   }
 
   // Decompose a single character and append results to the buffer.
-  // FIXME: for libgcj this is a native method which handles
-  // decomposition.  For Classpath, for now, it does nothing.
-  final void decomposeCharacter (char c, StringBuffer buf)
-  {
-    buf.append (c);
-  }
+  native final void decomposeCharacter (char c, StringBuffer buf);
 
   /**
    * This is the current collation decomposition setting.
