--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipEntry.java	2004-10-28 02:27:40.000000000 -0600
+++ java/util/zip/ZipEntry.java	2004-06-08 02:19:58.000000000 -0600
@@ -1,4 +1,4 @@
-/* ZipEntry.java --
+/* java.util.zip.ZipEntry
    Copyright (C) 2001, 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -51,10 +51,10 @@
  */
 public class ZipEntry implements ZipConstants, Cloneable
 {
-  private static final int KNOWN_SIZE   = 1;
-  private static final int KNOWN_CSIZE  = 2;
-  private static final int KNOWN_CRC    = 4;
-  private static final int KNOWN_TIME   = 8;
+  private static int KNOWN_SIZE   = 1;
+  private static int KNOWN_CSIZE  = 2;
+  private static int KNOWN_CRC    = 4;
+  private static int KNOWN_TIME   = 8;
 
   private static Calendar cal;
 
@@ -71,14 +71,15 @@
   int flags;              /* used by ZipOutputStream */
   int offset;             /* used by ZipFile and ZipOutputStream */
 
+
   /**
    * Compression method.  This method doesn't compress at all.
    */
-  public static final int STORED = 0;
+  public final static int STORED      =  0;
   /**
    * Compression method.  This method uses the Deflater.
    */
-  public static final int DEFLATED = 8;
+  public final static int DEFLATED    =  8;
 
   /**
    * Creates a zip entry with the given name.
