--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormatSymbols.java	2003-11-23 02:20:10.000000000 -0700
+++ java/text/DecimalFormatSymbols.java	2003-11-20 02:18:07.000000000 -0700
@@ -158,7 +158,6 @@
     percent = safeGetChar (res, "percent", '%');
     perMill = safeGetChar (res, "perMill", '\u2030');
     zeroDigit = safeGetChar (res, "zeroDigit", '0');
-    locale = loc;
   }
 
   /**
@@ -581,20 +580,13 @@
   /**
    * @serial This value represents the type of object being de-serialized.
    * 0 indicates a pre-Java 1.1.6 version, 1 indicates 1.1.6 or later.
-   * 0 indicates a pre-Java 1.1.6 version, 1 indicates 1.1.6 or later,
-   * 2 indicates 1.4 or later
-    */
-  private int serialVersionOnStream = 2;
+   */
+  private int serialVersionOnStream = 1;
   /**
    * @serial This is the character used to represent 0.
    */
   private char zeroDigit;
 
-  /**
-   * @serial The locale of these currency symbols.
-   */
-  private Locale locale;
- 
   private static final long serialVersionUID = 5772796243397350300L;
 
   private void readObject(ObjectInputStream stream)
@@ -605,10 +597,7 @@
       {
         monetarySeparator = decimalSeparator;
 	exponential = 'E';
+	serialVersionOnStream = 1;
       }
-    if (serialVersionOnStream < 2)
-	locale = Locale.getDefault();
-
-    serialVersionOnStream = 2;
   }
 }
