--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/interfaces/DSAPrivateKey.java	2004-04-28 02:20:17.000000000 -0600
+++ java/security/interfaces/DSAPrivateKey.java	2003-10-13 02:17:51.000000000 -0600
@@ -1,5 +1,5 @@
 /* DSAPublicKey.java -- A Digital Signature Algorithm private key
-   Copyright (C) 1998, 2000, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,8 +37,8 @@
 
 package java.security.interfaces;
 
-import java.math.BigInteger;
 import java.security.PrivateKey;
+import java.math.BigInteger;
 
 /**
  * This interface models a Digital Signature Algorithm (DSA) private key
