--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/CodeSource.java	2004-04-18 02:22:21.000000000 -0600
+++ java/security/CodeSource.java	2002-05-31 12:14:17.000000000 -0600
@@ -159,22 +159,17 @@
    * object must have all the certifcates this object has (but may have more),
    * and must have a location that is a subset of this object's.  In order
    * for this object to imply the specified object, the following must be
-   * true:
-   *
-   * <ol>
+   * true:<ol>
    * <li><em>codesource</em> must not be <code>null</code>.</li>
    * <li>If <em>codesource</em> has a certificate list, all of it's
    *     certificates must be present in the certificate list of this
    *     code source.</li>
    * <li>If this object does not have a <code>null</code> location, then
-   *     the following addtional tests must be passed.
-   *
-   *     <ol>
+   *     the following addtional tests must be passed.<ol>
    *     <li><em>codesource</em> must not have a <code>null</code>
    *         location.</li>
    *     <li><em>codesource</em>'s location must be equal to this object's
-   *         location, or
-   *         <ul>
+   *         location, or<br><ul>
    *         <li><em>codesource</em>'s location protocol, port, and ref (aka,
    *             anchor) must equal this objects</li>
    *         <li><em>codesource</em>'s location host must imply this object's
@@ -188,21 +183,19 @@
    *             location file must start with this object's location file
    *             with the '/' character appended to it.</li>
    *         </ul></li>
-   *     </ol></li>
+   *     </ol>
    * </ol>
    *
    * <p>For example, each of these locations imply the location
-   * "http://java.sun.com/classes/foo.jar":</p>
-   * 
+   * "http://java.sun.com/classes/foo.jar":<ul>
    * <pre>
    * http:
    * http://*.sun.com/classes/*
    * http://java.sun.com/classes/-
    * http://java.sun.com/classes/foo.jar
    * </pre>
-   * 
-   * <p>Note that the code source with null location and null certificates implies
-   * all other code sources.</p>
+   * Note that the code source with null location and null certificates implies
+   * all other code sources.
    *
    * @param cs the <code>CodeSource</code> to test against this object
    * @return true if this specified <code>CodeSource</code> is implied
@@ -252,7 +245,7 @@
   /**
    * This method returns a <code>String</code> that represents this object.
    * The result is in the format <code>"(" + getLocation()</code> followed
-   * by a space separated list of certificates (or "&lt;no certificates&gt;"),
+   * by a space separated list of certificates (or "<no certificates>"),
    * followed by <code>")"</code>.
    *
    * @return a <code>String</code> for this object
