--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/TextComponent.java	2004-04-28 02:20:15.000000000 -0600
+++ java/awt/TextComponent.java	2003-12-05 02:18:14.000000000 -0700
@@ -41,6 +41,7 @@
 import java.awt.event.TextEvent;
 import java.awt.event.TextListener;
 import java.awt.peer.TextComponentPeer;
+import java.awt.peer.ComponentPeer;
 import java.io.Serializable;
 import java.util.EventListener;
 
@@ -233,11 +234,11 @@
   * specified start and end positions.  Illegal values for these
   * positions are silently fixed.
   *
-  * @param selectionStart The new start position for the selected text.
-  * @param selectionEnd The new end position for the selected text.
+  * @param startSelection The new start position for the selected text.
+  * @param endSelection The new end position for the selected text.
   */
 public synchronized void
-select(int selectionStart, int selectionEnd)
+select(int selectionStart, int endSelection)
 {
   if (selectionStart < 0)
     selectionStart = 0;
