--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/ResolutionSyntax.java	2004-04-28 02:20:18.000000000 -0600
+++ javax/print/attribute/ResolutionSyntax.java	2004-04-21 02:18:12.000000000 -0600
@@ -1,5 +1,5 @@
 /* ResolutionSyntax.java -- 
-   Copyright (C) 2003, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -86,7 +86,7 @@
    *
    * @param obj the object to test
    *
-   * @return true if both objects are equal, false otherwise.
+   * @returns true if both objects are equal, false otherwise.
    */
   public boolean equals(Object obj)
   {
@@ -111,6 +111,7 @@
     if (units < 1)
       throw new IllegalArgumentException("units may not be less then 1");
 
+    int rount = units / 2;
     return (crossFeedResolution + units) / units;
   }
 
@@ -136,6 +137,7 @@
     if (units < 1)
       throw new IllegalArgumentException("units may not be less then 1");
 
+    int rount = units / 2;
     return (crossFeedResolution + units) / units;
   }
 
