--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/Socket.java	2003-05-01 02:23:12.000000000 -0600
+++ java/net/Socket.java	2003-03-11 02:17:52.000000000 -0700
@@ -291,41 +291,16 @@
     if (sm != null)
       sm.checkConnect(raddr.getHostName(), rport);
 
-    // create socket
     impl.create(stream);
 
-    // bind/connect to address/port
     // FIXME: JCL p. 1586 says if localPort is unspecified, bind to any port,
     // i.e. '0' and if localAddr is unspecified, use getLocalAddress() as
     // that default.  JDK 1.2 doc infers not to do a bind.
-    if (laddr != null) {
-      try {
-        impl.bind(laddr, lport);
-      } catch (IOException exception) {
-        impl.close();
-        throw exception;
-      } catch (RuntimeException exception) {
-        impl.close();
-        throw exception;
-      } catch (Error error) {
-        impl.close();
-        throw error;
-      }
-    }
-    if (raddr != null) {
-      try {
-        impl.connect(raddr, rport);
-      } catch (IOException exception) {
-        impl.close();
-        throw exception;
-      } catch (RuntimeException exception) {
-        impl.close();
-        throw exception;
-      } catch (Error error) {
-        impl.close();
-        throw error;
-      }
-    }
+    if (laddr != null)
+      impl.bind(laddr, lport);
+
+    if (raddr != null)
+      impl.connect(raddr, rport);
   }
 
   /**
