--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2004-01-26 02:24:17.000000000 -0700
+++ java/awt/Component.java	2003-11-13 02:18:10.000000000 -0700
@@ -869,14 +869,9 @@
    */
   public void show()
   {
-    // We must set visible before showing the peer.  Otherwise the
-    // peer could post paint events before visible is true, in which
-    // case lightweight components are not initially painted --
-    // Container.paint first calls isShowing () before painting itself
-    // and its children.
-    this.visible = true;
     if (peer != null)
       peer.setVisible(true);
+    this.visible = true;
   }
 
   /**
