--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/MessageFormat.java	2004-11-17 02:24:09.000000000 -0700
+++ java/text/MessageFormat.java	2004-10-28 02:16:12.000000000 -0600
@@ -401,10 +401,9 @@
     return formatInternal(arguments, appendBuf, fp, null);
   }
 
-  private StringBuffer formatInternal (Object arguments[],
-                                       StringBuffer appendBuf,
-				       FieldPosition fp,
-				       FormatCharacterIterator output_iterator)
+  protected final StringBuffer formatInternal (Object arguments[], StringBuffer appendBuf,
+					       FieldPosition fp,
+					       FormatCharacterIterator output_iterator)
   {
     appendBuf.append(leader);
     if (output_iterator != null)
