--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicSliderUI.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/plaf/basic/BasicSliderUI.java	2004-05-27 02:19:08.000000000 -0600
@@ -681,6 +681,7 @@
     highlightColor = defaults.getColor("Slider.highlight");
     focusColor = defaults.getColor("Slider.focus");
     slider.setBorder(defaults.getBorder("Slider.border"));
+    slider.setOpaque(true);
 
     thumbHeight = defaults.getInt("Slider.thumbHeight");
     thumbWidth = defaults.getInt("Slider.thumbWidth");
