--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FilePermission.java	2004-04-28 02:20:16.000000000 -0600
+++ java/io/FilePermission.java	2004-03-20 02:18:20.000000000 -0700
@@ -229,6 +229,7 @@
 	        return false;
 	      /* Make sure there are no subdirectories specified 
                  underneath this one */
+	      String sub2 = f2.substring(sub1.length()+1);
 	      if(f2.substring(sub1.length()+1).indexOf(File.separatorChar) 
                  != -1)
 	        return false;
