--- /home/tromey/gnu/classpath/classpath/java/net/URLDecoder.java	Mon Dec 30 15:39:05 2002
+++ java/net/URLDecoder.java	Tue Dec 10 19:20:32 2002
@@ -82,12 +82,12 @@
   {
     try
       {
-        return decode(s, "UTF-8");
+	return decode(s, "UTF-8");
       }
     catch (UnsupportedEncodingException uee)
       {
-        // Should never happen since UTF-8 encoding should always be supported
-        return s;
+	// Should never happen since UTF-8 encoding should always be supported
+	return s;
       }
   }
 
