--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/activation/Activator.java	Sun Mar 23 02:18:41 2003
+++ java/rmi/activation/Activator.java	Tue Jan 22 16:01:36 2002
@@ -42,8 +42,8 @@
 import java.rmi.MarshalledObject;
 
 public interface Activator
-  extends Remote
-{
-  public MarshalledObject activate(ActivationID id, boolean force)
-    throws ActivationException, UnknownObjectException, RemoteException;
+	extends Remote {
+
+public MarshalledObject activate(ActivationID id, boolean force) throws ActivationException, UnknownObjectException, RemoteException;
+
 }
