--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/MessageFormat.java	Tue Jul  2 13:40:35 2002
+++ java/text/MessageFormat.java	Sun Mar 23 02:18:02 2003
@@ -143,6 +143,8 @@
 
 public class MessageFormat extends Format
 {
+  private static final long serialVersionUID = 6479157306784022952L;
+  
   // Helper that returns the text up to the next format opener.  The
   // text is put into BUFFER.  Returns index of character after end of
   // string.  Throws IllegalArgumentException on error.
