--- /home/tromey/gnu/egcs/classpath/classpath/java/util/List.java	Mon Oct 15 10:45:18 2001
+++ java/util/List.java	Fri Nov  2 21:44:52 2001
@@ -288,7 +288,7 @@
   Object remove(int index);
 
   /**
-   * Remove the first occurence of an object from this list (optional
+   * Remove the first occurrence of an object from this list (optional
    * operation). That is, remove the first element e such that
    * <code>o == null ? e == null : o.equals(e)</code>.
    *
