--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/CharBuffer.java	Tue Mar 11 10:27:52 2003
+++ java/nio/CharBuffer.java	Wed Mar 12 02:17:47 2003
@@ -83,9 +83,6 @@
    */
   final public static CharBuffer wrap (CharSequence a, int offset, int length)
   {
-    // FIXME: implement better handling of java.lang.String.
-    // Probably share data with String via reflection.
-	  
     if ((offset < 0)
         || (offset > a.length ())
         || (length < 0)
@@ -99,9 +96,7 @@
         buffer [i] = a.charAt (i);
       }
     
-    // FIXME: make it read-only.
-    //return wrap (buffer, offset, length);
-    return wrap (buffer, offset, length).asReadOnlyBuffer ();
+    return wrap (buffer, offset, length);
   }
   
   /**
@@ -112,7 +107,7 @@
     return wrap (array, 0, array.length);
   }
  
-  protected CharBuffer (int cap, int lim, int pos, int mark)
+  CharBuffer (int cap, int lim, int pos, int mark)
   {
     super (cap, lim, pos, mark);
   }
@@ -363,9 +358,7 @@
    */
   public String toString ()
   {
-    char[] data = new char [length ()];
-    get (data, 0, length ());
-    return new String (data, position (), length ());
+    return new String (array (), position (), length ());
   }
 
   /**
