--- /home/tromey/gnu/classpath/classpath/java/awt/geom/NoninvertibleTransformException.java	Sat Mar  2 12:13:09 2002
+++ java/awt/geom/NoninvertibleTransformException.java	Tue Jan 22 16:01:31 2002
@@ -1,5 +1,4 @@
-/* NoninvertibleTransformException.java -- a transform can't be inverted
-   Copyright (C) 2000, 2002 Free Software Foundation
+/* Copyright (C) 2000, 2002  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -38,28 +37,14 @@
 package java.awt.geom;
 
 /**
- * Thrown if an operation requires an inverse of an
- * <code>AffineTransform</code>, but the transform is in a non-invertible
- * state.
- *
  * @author Tom Tromey <tromey@cygnus.com>
- * @see AffineTransform
- * @status updated to 1.4
+ * @date July 15, 2000
  */
+
 public class NoninvertibleTransformException extends Exception
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = 6137225240503990466L;
-
-  /**
-   * Create an exception with a message.
-   *
-   * @param s the message
-   */
-  public NoninvertibleTransformException(String s)
+  public NoninvertibleTransformException (String s)
   {
-    super(s);
+    super (s);
   }
 }
