--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileInputStream.java	2003-04-22 02:26:41.000000000 -0600
+++ java/io/FileInputStream.java	2003-04-07 02:22:41.000000000 -0600
@@ -82,7 +82,6 @@
     SecurityManager s = System.getSecurityManager();
     if (s != null)
       s.checkRead(name);
-
     fd = new FileDescriptor(name, FileDescriptor.READ);
   }
 
@@ -269,15 +268,15 @@
    *
    * @exception IOException If an error occurs
    */
-  public synchronized long skip (long numBytes) throws IOException
+  public long skip (long numBytes) throws IOException
   {
     if (numBytes < 0)
-      throw new IllegalArgumentException ("Can't skip negative bytes: " + 
-                                          numBytes);
+      throw new IllegalArgumentException ( "Can't skip negative bytes: " +
+                                           numBytes);
 
     if (numBytes == 0)
       return 0;
-
+    
     long curPos = fd.getFilePointer ();
     long newPos = fd.seek (numBytes, FileDescriptor.CUR, true);
     return newPos - curPos;
