--- /home/tromey/gnu/classpath/classpath/java/nio/channels/spi/SelectorProvider.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/spi/SelectorProvider.java	Wed Nov 13 06:52:47 2002
@@ -37,8 +37,7 @@
 
 package java.nio.channels.spi;
 
-import gnu.java.nio.SelectorProviderImpl;
-import java.io.IOException;
+/* import gnu.java.nio.channels.SelectorProviderImpl; */
 import java.nio.channels.DatagramChannel;
 import java.nio.channels.Pipe;
 import java.nio.channels.ServerSocketChannel;
@@ -68,27 +67,27 @@
   /**
    * Opens a datagram channel.
    */
-  public abstract DatagramChannel openDatagramChannel () throws IOException;
+  public abstract DatagramChannel openDatagramChannel ();
   
   /**
    * Opens a pipe.
    */
-  public abstract Pipe openPipe () throws IOException;
+  public abstract Pipe openPipe ();
   
   /**
    * Opens a selector.
    */
-  public abstract AbstractSelector openSelector () throws IOException;
+  public abstract AbstractSelector openSelector ();
   
   /**
    * Opens a server socket channel.
    */
-  public abstract ServerSocketChannel openServerSocketChannel () throws IOException;
+  public abstract ServerSocketChannel openServerSocketChannel ();
   
   /**
    * Opens a socket channel.
    */
-  public abstract SocketChannel openSocketChannel () throws IOException;
+  public abstract SocketChannel openSocketChannel ();
     
   /**
    * Returns the system-wide default selector provider for this invocation
@@ -96,10 +95,8 @@
    */
   public static SelectorProvider provider ()
   {
-    if (pr == null)
-      {
-        pr = new SelectorProviderImpl ();
-      }
+/*    if (pr == null) */
+/*      pr = new SelectorProviderImpl (); */
     
     return pr;
   }
