--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/prefs/AbstractPreferences.java	2004-04-30 02:23:42.000000000 -0600
+++ java/util/prefs/AbstractPreferences.java	2004-04-22 02:18:15.000000000 -0600
@@ -119,7 +119,7 @@
             throw new IllegalArgumentException("Illegal name argument '"
                                                + name
                                                + "' (parent is "
-                                               + (parent == null ? "" : "not ")
+                                               + parent == null ? "" : "not "
                                                + "null)");
         this.parent = parent;
         this.name = name;
