--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/DataOutputStream.java	2004-10-12 02:18:12.000000000 -0600
+++ java/io/DataOutputStream.java	2003-05-09 02:27:01.000000000 -0600
@@ -404,7 +404,7 @@
    *
    * @see DataInput#readUTF
    */
-  public final synchronized void writeUTF(String value) throws IOException
+  public synchronized final void writeUTF (String value) throws IOException
   {
     int len = value.length();
     int sum = 0;
