--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormat.java	2004-05-02 02:20:13.000000000 -0600
+++ java/text/DecimalFormat.java	2004-05-05 02:18:55.000000000 -0600
@@ -42,11 +42,13 @@
 import gnu.java.text.FormatCharacterIterator;
 import gnu.java.text.StringFormatBuffer;
 
-import java.io.IOException;
-import java.io.ObjectInputStream;
 import java.util.Currency;
 import java.util.HashMap;
 import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.io.ObjectInputStream;
+import java.io.IOException;
 
 /**
  * @author Tom Tromey <tromey@cygnus.com>
