--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicCheckBoxMenuItemUI.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicCheckBoxMenuItemUI.java	2004-04-16 12:44:38.000000000 -0600
@@ -39,6 +39,8 @@
 
 import java.awt.event.MouseEvent;
 
+import javax.swing.AbstractButton;
+import javax.swing.Icon;
 import javax.swing.JComponent;
 import javax.swing.JMenuItem;
 import javax.swing.MenuElement;
