--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/IllegalBlockingModeException.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/IllegalBlockingModeException.java	2002-09-25 03:05:53.000000000 -0600
@@ -1,4 +1,4 @@
-/* IllegalBlockingModeException.java --
+/* IllegalBlockingModeException.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -37,7 +37,6 @@
 
 package java.nio.channels;
 
-
 /**
  * @author Michael Koch <konqueror@gmx.de>
  * @since 1.4
