--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/border/LineBorder.java	2003-07-29 02:19:59.000000000 -0600
+++ javax/swing/border/LineBorder.java	2003-06-28 02:18:23.000000000 -0600
@@ -128,7 +128,7 @@
    * Constructs a LineBorder given its color, thickness, and whether
    * it has rounded corners.
    * 
-   * <p><img src="doc-files/LineBorder-1.png" width="500" height="200"
+   * <p><img src="LineBorder-1.png" width="500" height="200"
    * alt="[An illustration of two LineBorders] />
    *
    * <p>Note that the enlarged view in the right-hand picture shows
