--- /home/tromey/gnu/classpath/classpath/java/nio/channels/DatagramChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/DatagramChannel.java	Tue Dec 10 19:20:33 2002
@@ -88,7 +88,7 @@
    * @exception IOException If an error occurs
    * @exception NotYetConnectedException The channel's socket is not connected.
    */
-  public final long write (ByteBuffer[] srcs) throws IOException
+  public final long write (ByteBuffer[] srcs)
   {
     long b = 0;
     
@@ -111,15 +111,14 @@
    * @exception SecurityException If a security manager has been installed and
    * it does not permit datagrams to be sent to the given address.
    */
-  public abstract DatagramChannel connect (SocketAddress remote)
-    throws IOException;
+  public abstract DatagramChannel connect (SocketAddress remote);
 
   /**
    * Disonnects this channel's socket.
    *
    * @exception IOException If an error occurs
    */
-  public abstract DatagramChannel disconnect () throws IOException;
+  public abstract DatagramChannel disconnect ();
 
   /**
    * Tells whether or not this channel's socket is connected.
@@ -132,7 +131,7 @@
   /**
    * Reads data from this channel.
    */
-  public abstract int read (ByteBuffer dst) throws IOException;
+  public abstract int read (ByteBuffer dst);
   
   /**
    * Reads data from this channel.
@@ -140,8 +139,7 @@
    * @exception IOException If an error occurs.
    * @exception NotYetConnectedException The channel's socket is not connected.
    */
-  public abstract long read (ByteBuffer[] dsts, int offset, int length)
-    throws IOException;
+  public abstract long read (ByteBuffer[] dsts, int offset, int length);
  
   /**
    * Receives a datagram via this channel.
@@ -156,7 +154,7 @@
    * @exception SecurityException If a security manager has been installed and
    * it does not permit datagrams to be sent to the given address.
    */
-  public abstract SocketAddress receive (ByteBuffer dst) throws IOException;
+  public abstract SocketAddress receive (ByteBuffer dst);
  
   /**
    * Sends a datagram via this channel.
@@ -171,7 +169,7 @@
    * @exception SecurityException If a security manager has been installed and
    * it does not permit datagrams to be sent to the given address.
    */
-  public abstract int send (ByteBuffer src, SocketAddress target) throws IOException;
+  public abstract int send (ByteBuffer src, SocketAddress target);
  
   /**
    * Retrieves the channel's socket.
@@ -184,7 +182,7 @@
    * @exception IOException If an error occurs.
    * @exception NotYetConnectedException The channel's socket is not connected.
    */
-  public abstract int write (ByteBuffer src) throws IOException;
+  public abstract int write (ByteBuffer src);
   
   /**
    * Writes data to this channel.
@@ -192,8 +190,7 @@
    * @exception IOException If an error occurs.
    * @exception NotYetConnectedException The channel's socket is not connected.
    */
-  public abstract long write (ByteBuffer[] srcs, int offset, int length)
-    throws IOException;
+  public abstract long write (ByteBuffer[] srcs, int offset, int length);
 
   /**
    * Retrieves the valid operations for this channel.
