--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/activation/Activator.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/activation/Activator.java	2003-10-13 02:17:50.000000000 -0600
@@ -1,5 +1,5 @@
-/* Activator.java --
-   Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
