--- /home/tromey/gnu/classpath/classpath/java/awt/color/ProfileDataException.java	Mon Dec 30 15:39:04 2002
+++ java/awt/color/ProfileDataException.java	Thu Aug  8 22:26:14 2002
@@ -45,7 +45,7 @@
   * @author Eric Blake <ebb9@email.byu.edu>
   * @status updated to 1.4
   */
-public class ProfileDataException extends RuntimeException
+public class ProfileDataException extends Exception
 {
   /**
    * Compatible with JDK 1.2+.
