--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RMIClassLoader.java	2003-10-07 02:20:27.000000000 -0600
+++ java/rmi/server/RMIClassLoader.java	2003-11-12 02:18:14.000000000 -0700
@@ -113,8 +113,7 @@
      */
     public boolean equals (Object theOther)
     {
-      if (theOther != null
-          && theOther instanceof CacheKey)
+      if (theOther instanceof CacheKey)
       {
       	CacheKey key = (CacheKey) theOther;
 	
