--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DoubleViewBufferImpl.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/DoubleViewBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -38,7 +38,7 @@
 
 package java.nio;
 
-final class DoubleViewBufferImpl extends DoubleBuffer
+class DoubleViewBufferImpl extends DoubleBuffer
 {
   /** Position in bb (i.e. a byte offset) where this buffer starts. */
   private int offset;
