--- /home/tromey/gnu/classpath/classpath/java/awt/dnd/DragSourceContext.java	Wed Oct  2 11:21:37 2002
+++ java/awt/dnd/DragSourceContext.java	Thu Aug  8 22:26:15 2002
@@ -50,8 +50,6 @@
 public class DragSourceContext
   implements DragSourceListener, DragSourceMotionListener, Serializable
 {
-  static final long serialVersionUID = -115407898692194719L;
-
   protected static final int DEFAULT = 0;
   protected static final int ENTER = 1;
   protected static final int OVER = 2;
