--- /home/tromey/gnu/Nightly/classpath/classpath/javax/rmi/PortableRemoteObject.java	2004-04-20 02:21:38.000000000 -0600
+++ javax/rmi/PortableRemoteObject.java	2003-03-29 14:23:25.000000000 -0700
@@ -1,5 +1,5 @@
 /* PortableRemoteObject.java -- 
-   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,14 +38,13 @@
 
 package javax.rmi;
 
-import gnu.javax.rmi.CORBA.DelegateFactory;
-import gnu.javax.rmi.CORBA.GetDelegateInstanceException;
-
-import java.rmi.NoSuchObjectException;
 import java.rmi.Remote;
 import java.rmi.RemoteException;
-
+import java.rmi.NoSuchObjectException;
+import gnu.javax.rmi.CORBA.DelegateFactory;
+import gnu.javax.rmi.CORBA.GetDelegateInstanceException;
 import javax.rmi.CORBA.PortableRemoteObjectDelegate;
+import javax.rmi.CORBA.Util;
 
 public class PortableRemoteObject
   implements Remote /* why doc doesn't say should implement Remote */
