--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/IntBuffer.java	2003-03-11 10:27:52.000000000 -0700
+++ java/nio/IntBuffer.java	2003-02-20 02:17:58.000000000 -0700
@@ -77,7 +77,7 @@
     return wrap(array, 0, array.length);
   }
 
-  protected IntBuffer (int capacity, int limit, int position, int mark)
+  IntBuffer (int capacity, int limit, int position, int mark)
   {
     super (capacity, limit, position, mark);
     array_offset = 0;
