--- /home/tromey/gnu/egcs/classpath/classpath/java/util/Arrays.java	Tue Jan 22 15:50:06 2002
+++ java/util/Arrays.java	Tue Jan 22 16:01:39 2002
@@ -57,6 +57,7 @@
  * sets that cause other quicksorts to degrade to quadratic performance.
  *
  * @author Original author unknown
+ * @author Bryce McKinlay
  * @author Eric Blake <ebb9@email.byu.edu>
  * @see Comparable
  * @see Comparator
@@ -2345,8 +2346,6 @@
       }
   }
 
-
-// asList
   /**
    * Returns a list "view" of the specified array. This method is intended to
    * make it easy to use the Collections API with existing array-based APIs and
