--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/regex/Pattern.java	Tue Apr 30 15:37:27 2002
+++ java/util/regex/Pattern.java	Tue Mar 25 02:17:56 2003
@@ -38,7 +38,9 @@
 // Stub class until java.util.regex is implemented.
 package java.util.regex;
 
-public class Pattern
+import java.io.Serializable;
+
+public class Pattern implements Serializable
 {
   public static Pattern compile(String regex)
   {
