--- /home/tromey/gnu/classpath/classpath/java/net/ServerSocket.java	Sat Nov  9 16:09:31 2002
+++ java/net/ServerSocket.java	Thu Oct  3 11:50:56 2002
@@ -184,6 +184,19 @@
   public void bind (SocketAddress endpoint)
     throws IOException
   {
+    if (impl == null)
+      throw new IOException ("Cannot initialize Socket implementation");
+
+    if (! (endpoint instanceof InetSocketAddress))
+      throw new IllegalArgumentException ("Address type not supported");
+
+    InetSocketAddress tmp = (InetSocketAddress) endpoint;
+    
+    SecurityManager s = System.getSecurityManager ();
+    if (s != null)
+      s.checkListen (tmp.getPort ());
+
+    impl.bind (tmp.getAddress (), tmp.getPort ());
   }
  
   /**
@@ -201,6 +214,20 @@
    */
   public void bind (SocketAddress endpoint, int backlog) throws IOException
   {
+    if (impl == null)
+      throw new IOException ("Cannot initialize Socket implementation");
+
+    if (! (endpoint instanceof InetSocketAddress))
+      throw new IllegalArgumentException ("Address type not supported");
+
+    InetSocketAddress tmp = (InetSocketAddress) endpoint;
+    
+    SecurityManager s = System.getSecurityManager ();
+    if (s != null)
+      s.checkListen (tmp.getPort ());
+
+    impl.bind (tmp.getAddress (), tmp.getPort ());
+    impl.listen(backlog);
   }
   
   /**
@@ -339,6 +366,17 @@
   }
 
   /**
+   * Returns true if the socket is closed, otherwise false
+   * 
+   * @since 1.4
+   */
+  public boolean isClosed()
+  {
+    // FIXME: implement this
+    return false;
+  }
+
+  /**
    * Sets the value of SO_TIMEOUT.  A value of 0 implies that SO_TIMEOUT is
    * disabled (ie, operations never time out).  This is the number of 
    * milliseconds a socket operation can block before an
