--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JButton.java	2004-02-15 02:21:04.000000000 -0700
+++ javax/swing/JButton.java	2004-06-18 02:18:51.000000000 -0600
@@ -87,6 +87,7 @@
     protected  void configurePropertiesFromAction(Action a)
     {
 	//Factory method which sets the AbstractButton's properties according to values from the Action instance. 
+	super.configurePropertiesFromAction(a);
     }
     
     public AccessibleContext getAccessibleContext()
