--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLookAndFeel.java	2004-05-21 02:27:02.000000000 -0600
+++ javax/swing/plaf/basic/BasicLookAndFeel.java	2004-04-17 02:18:08.000000000 -0600
@@ -205,15 +205,15 @@
   private void loadResourceBundle(UIDefaults defaults)
   {
     ResourceBundle bundle;
-    Enumeration e;
+    Enumeration enum;
     String key;
     String value;
     bundle = ResourceBundle.getBundle("resources/basic");
     // Process Resources
-    e = bundle.getKeys();
-    while (e.hasMoreElements())
+    enum = bundle.getKeys();
+    while (enum.hasMoreElements())
       {
-        key = (String) e.nextElement();
+        key = (String) enum.nextElement();
         value = bundle.getString(key);
         defaults.put(key, value);
       }
