--- /home/tromey/gnu/egcs/classpath/classpath/java/io/PushbackInputStream.java	Mon Mar 25 00:05:17 2002
+++ java/io/PushbackInputStream.java	Mon Mar 25 00:18:40 2002
@@ -201,7 +201,6 @@
   public synchronized int read(byte[] b, int off, int len) throws IOException
   {
     int numBytes = Math.min(buf.length - pos, len);
-
     if (numBytes > 0)
       {
 	System.arraycopy (buf, pos, b, off, numBytes);
@@ -210,10 +209,10 @@
 	off += numBytes;
       }
 
-    if (len > 0) 
+    if (len > 0)
       {
         len = super.read(b, off, len);
-        if (len == -1) //EOF
+	if (len == -1) // EOF
 	  return numBytes > 0 ? numBytes : -1;
 	numBytes += len;
       }
@@ -319,8 +318,7 @@
 	int numread = (int) Math.min((long) (buf.length - pos), n);
 	pos += numread;
 	n -= numread;
-	if (n > 0)
-	  n -= super.skip(n);
+	n -= super.skip(n);
       }
 
     return origN - n;
