--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollBar.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JScrollBar.java	2004-05-27 02:19:05.000000000 -0600
@@ -169,7 +169,7 @@
   protected BoundedRangeModel model;
 
   /** The orientation of the scroll bar. */
-  protected int orientation = SwingConstants.HORIZONTAL;
+  protected int orientation = SwingConstants.VERTICAL;
 
   /** How much the thumb moves when moving in a unit. */
   protected int unitIncrement = 1;
@@ -186,7 +186,7 @@
    */
   public JScrollBar()
   {
-    this(SwingConstants.HORIZONTAL, 0, 10, 0, 100);
+    this(SwingConstants.VERTICAL, 0, 10, 0, 100);
   }
 
   /**
