--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/FloatBuffer.java	2003-03-11 10:27:52.000000000 -0700
+++ java/nio/FloatBuffer.java	2003-05-10 02:19:17.000000000 -0600
@@ -41,8 +41,8 @@
 
 public abstract class FloatBuffer extends Buffer implements Comparable
 {
-  protected float [] backing_buffer;
-  protected int array_offset;
+  int array_offset;
+  float[] backing_buffer;
 
   public static FloatBuffer allocateDirect(int capacity)
   {
@@ -77,7 +77,7 @@
     return wrap(array, 0, array.length);
   }
 
-  protected FloatBuffer (int capacity, int limit, int position, int mark)
+  FloatBuffer (int capacity, int limit, int position, int mark)
   {
     super (capacity, limit, position, mark);
     array_offset = 0;                    
