--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/FontMetrics.java	2004-04-18 02:22:13.000000000 -0600
+++ java/awt/FontMetrics.java	2004-02-04 02:18:37.000000000 -0700
@@ -47,12 +47,12 @@
   * least the following methods:
   * <p>
   * <ul>
-  * <li>getAscent()</li>
-  * <li>getDescent()</li>
-  * <li>getLeading()</li>
-  * <li>getMaxAdvance()</li>
-  * <li>charWidth(char)</li>
-  * <li>charsWidth(char[], int, int)</li>
+  * <li>getAscent
+  * <li>getDescent
+  * <li>getLeading()
+  * <li>getMaxAdvance()
+  * <li>charWidth(char)
+  * <li>charsWidth(char[], int, int)
   * </ul>
   *
   * @author Aaron M. Renn (arenn@urbanophile.com)
@@ -292,7 +292,8 @@
 {
   int total_width = 0;
   for (int i = offset; i < len; i++)
-    total_width += charWidth(buf[i]);
+    total_width = charWidth(buf[i]);
+
   return(total_width);
 }
 
@@ -327,12 +328,7 @@
 public int[]
 getWidths()
 {
-  int [] result = new int[256];
-  for(char i = 0; i < 256; i++)
-  {
-    result[i]= charWidth(i);
-  }
-  return(result);
+  return(new int[256]);
 }
 
 /*************************************************************************/
@@ -351,4 +347,3 @@
 
 } // class FontMetrics 
 
-
