--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Key.java	2003-05-07 02:23:22.000000000 -0600
+++ java/security/Key.java	2003-05-10 02:19:17.000000000 -0600
@@ -64,7 +64,7 @@
   /**
    * The version identifier used for serialization.
    */
-  long serialVersionUID = 6603384152749567654L;
+  public static final long serialVersionUID = 6603384152749567654L;
 
   /**
    * This method returns the name of the algorithm for this key.  This is a
