--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocketImplFactory.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/DatagramSocketImplFactory.java	2003-10-13 02:17:50.000000000 -0600
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,12 +37,12 @@
 
 package java.net;
 
-
 /** Written using on-line Java Platform 1.4 API Specification.
   * Status: Believed complete and correct.
   */
+
 /**
-  * This interface defines one method which returns a
+  * This interface defines one method which returns a 
   * <code>DatagramSocketImpl</code> object.
   * This should not be needed by ordinary applications.
   *
@@ -57,4 +57,5 @@
    * @return A DatagramSocketImpl object
    */
   DatagramSocketImpl createDatagramSocketImpl();
+
 } // interface DatagramSocketImplFactory
