--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSeparator.java	2004-04-29 02:24:16.000000000 -0600
+++ javax/swing/JSeparator.java	2004-06-18 02:18:53.000000000 -0600
@@ -55,14 +55,15 @@
    */
   protected class AccessibleJSeparator extends AccessibleJComponent
   {
+    private static final long serialVersionUID = 916332890553201095L;
+  
     /**
      * Constructor AccessibleJSeparator
      *
      * @param component TODO
      */
-    protected AccessibleJSeparator(JSeparator component)
+    protected AccessibleJSeparator()
     {
-      super(component);
     }
 
     /**
@@ -76,6 +77,8 @@
     }
   }
 
+  private static final long serialVersionUID = 125301223445282357L;
+  
   /** The orientation of the JSeparator. */
   private transient int orientation = HORIZONTAL;
 
@@ -198,7 +201,8 @@
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      accessibleContext = new AccessibleJSeparator(this);
+      accessibleContext = new AccessibleJSeparator();
+    
     return accessibleContext;
   }
 }
