--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PrintWriter.java	Mon Mar 10 02:19:32 2003
+++ java/io/PrintWriter.java	Tue Jan 22 16:01:33 2002
@@ -48,6 +48,8 @@
   * class which also auto-flushes when it encounters a newline character
   * in the chars written).
   *
+  * @version 0.0
+  *
   * @author Per Bothner <bothner@cygnus.com>
   * @author Aaron M. Renn (arenn@urbanophile.com)
   * @date April 17, 1998.  
@@ -97,8 +99,7 @@
    * every line is terminated or newline character is written.
    *
    * @param wr The <code>Writer</code> to write to.
-   * @param autoflush <code>true</code> to flush the stream after every 
-   * line, <code>false</code> otherwise
+   * @param autoflush <code>true</code> to flush the stream after every line, <code>false</code> otherwise
    */
   public PrintWriter(Writer wr, boolean autoflush)
   {
@@ -129,8 +130,7 @@
    * constructor allows auto-flush functionality to be enabled if desired
    *
    * @param out The <code>OutputStream</code> to write to
-   * @param autoflush <code>true</code> to flush the stream after every 
-   * <code>println</code> call, <code>false</code> otherwise.
+   * @param autoflush <code>true</code> to flush the stream after every <code>println</code> call, <code>false</code> otherwise.
    */
   public PrintWriter(OutputStream out, boolean autoflush)
   {
@@ -153,8 +153,7 @@
    * <code>true</code> forever for this stream.  Before checking for an
    * error condition, this method flushes the stream.
    *
-   * @return <code>true</code> if an error has occurred, 
-   * <code>false</code> otherwise
+   * @return <code>true</code> if an error has occurred, <code>false</code> otherwise
    */
   public boolean checkError()
   {
@@ -310,7 +309,7 @@
    * This is the system dependent line separator
    */
   private static final char[] line_separator
-    = System.getProperty("line.separator").toCharArray();
+  = System.getProperty("line.separator").toCharArray();
 
   /**
    * This method prints a line separator sequence to the stream.  The value
@@ -568,4 +567,3 @@
     write(str, 0, str.length());
   }  
 }
-
