--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/DataInputStream.java	2004-10-12 02:18:12.000000000 -0600
+++ java/io/DataInputStream.java	2004-10-10 02:15:37.000000000 -0600
@@ -651,7 +651,7 @@
    *
    * @see DataInput#readUTF
    */
-  public static final String readUTF(DataInput in) throws IOException
+  public final static String readUTF (DataInput in) throws IOException
   {
     final int UTFlen = in.readUnsignedShort ();
     byte[] buf = new byte [UTFlen];
