--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/event/EventContext.java	2003-10-13 02:22:21.000000000 -0600
+++ javax/naming/event/EventContext.java	2003-12-06 02:18:23.000000000 -0700
@@ -37,7 +37,10 @@
 
 
 package javax.naming.event;
-import javax.naming.*;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingException;
  
 /**
  * @author Warren Levy <warrenl@redhat.com>
