--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileOutputStream.java	2003-05-07 02:23:22.000000000 -0600
+++ java/io/FileOutputStream.java	2003-05-07 02:22:00.000000000 -0600
@@ -84,11 +84,9 @@
     SecurityManager s = System.getSecurityManager();
     if (s != null)
       s.checkWrite(path);
-
-    int flags = FileDescriptor.WRITE;
-    if (append)
-      flags |= FileDescriptor.APPEND;
-    fd = new FileDescriptor(path, flags);
+    fd = new FileDescriptor (path, (append
+				    ? FileDescriptor.APPEND
+				    : FileDescriptor.WRITE));
   }
 
   /**
