--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/spec/DSAPublicKeySpec.java	2004-04-03 02:21:02.000000000 -0700
+++ java/security/spec/DSAPublicKeySpec.java	2003-04-30 02:19:07.000000000 -0600
@@ -1,5 +1,5 @@
 /* DSAPublicKeySpec.java --- DSA Public Key Specificaton class
-   Copyright (C) 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -47,7 +47,7 @@
 
 	@author Mark Benvenuto
 */
-public class DSAPublicKeySpec implements KeySpec 
+public class DSAPublicKeySpec extends Object implements KeySpec 
 {
   private BigInteger y = null;
   private BigInteger p = null;
