--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLabelUI.java	2003-06-19 02:34:52.000000000 -0600
+++ javax/swing/plaf/basic/BasicLabelUI.java	2003-05-10 02:19:19.000000000 -0600
@@ -65,10 +65,6 @@
     public Dimension getPreferredSize(JComponent c) 
     {
 	JLabel b = (JLabel)c;
-        /*
-          We cannot use this method because it is not part of the
-          official Swing API.
-
 	Dimension d = BasicGraphicsUtils.getPreferredSize(b, 
 							  gap,
 							  b.getText(),
@@ -78,8 +74,7 @@
 							  b.getHorizontalTextPosition(),
 							  b.getVerticalTextPosition());
 	System.out.println("JLABEL->^^^^^^^^^^^^^^^^^^^^^^   BASIC-PREF="+d + ",T="+b.getText());
-        */
-        return new Dimension(100, 30);
+	return d;
     }
     
 
@@ -95,7 +90,7 @@
 
         g.setFont(f);
 
-        FontMetrics fm = g.getFontMetrics(f);
+        FontMetrics fm = SwingUtilities.getFontMetrics(f);
 
         Insets i = c.getInsets();
 
