--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/BufferedWriter.java	Mon Mar 10 02:19:32 2003
+++ java/io/BufferedWriter.java	Tue Jan 22 16:01:32 2002
@@ -51,6 +51,8 @@
   * efficient mechanism for writing versus doing numerous small unbuffered
   * writes.
   *
+  * @version 0.0
+  *
   * @author Aaron M. Renn (arenn@urbanophile.com)
   * @author Tom Tromey <tromey@cygnus.com>
   * @date September 25, 1998 
