--- /home/tromey/gnu/classpath/classpath/java/net/NetworkInterface.java	Tue Nov 12 00:23:38 2002
+++ java/net/NetworkInterface.java	Thu Oct 10 15:45:05 2002
@@ -59,12 +59,8 @@
     this.inetAddresses.add (address);
   }
 
-  private static Vector getRealNetworkInterfaces ()
-    throws SocketException
-  {
-    throw new InternalError (
-      "NetworkInterface::getRealNetworkInterfaces not implemented");
-  }
+  private native static Vector getRealNetworkInterfaces ()
+    throws SocketException;
 
   /**
    *  Returns the name of the network interface
