--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ByteArrayOutputStream.java	2003-03-24 02:18:30.000000000 -0700
+++ java/io/ByteArrayOutputStream.java	2003-11-09 02:18:13.000000000 -0700
@@ -198,7 +198,7 @@
   // Resize buffer to accommodate new bytes.
   private void resize (int add)
   {
-    if (count + add >= buf.length)
+    if (count + add > buf.length)
       {
 	int newlen = buf.length * 2;
 	if (count + add > newlen)
