--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/JTextComponent.java	2003-06-28 02:19:25.000000000 -0600
+++ javax/swing/text/JTextComponent.java	2004-02-12 02:18:27.000000000 -0700
@@ -406,7 +406,7 @@
     public String getUIClassID()
     {
 	//          Returns a string that specifies the name of the l&f class that renders this component.  
-	return "JTextComponent";
+	return "TextComponentUI";
     }
     public int getVerticalAlignment()
     {
