--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/Writer.java	2004-04-13 02:20:22.000000000 -0600
+++ java/io/Writer.java	2003-03-24 02:17:54.000000000 -0700
@@ -75,8 +75,8 @@
    * This method initializes a <code>Writer</code> that will synchronize
    * on the specified <code>Object</code>.
    *
-   * @param lock The <code>Object</code> to use for synchronizing critical
-   *             sections
+   * @param obj The <code>Object</code> to use for synchronizing critical
+   *            sections
    */
   protected Writer(Object lock)
   {
@@ -157,7 +157,7 @@
    *
    * @param str The <code>String</code> whose chars are to be written.
    *
-   * @exception IOException If an error occurs
+   * @param IOException If an error occurs
    */
   public void write(String str) throws IOException
   {
