--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSlider.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JSlider.java	2004-04-22 02:18:16.000000000 -0600
@@ -37,14 +37,16 @@
 
 package javax.swing;
 
-import java.awt.Dimension;
+import java.awt.ComponentOrientation;
 import java.awt.MenuContainer;
+import java.awt.Dimension;
 import java.awt.image.ImageObserver;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
 import java.io.Serializable;
 import java.util.Dictionary;
 import java.util.Enumeration;
 import java.util.Hashtable;
-
 import javax.accessibility.Accessible;
 import javax.accessibility.AccessibleContext;
 import javax.accessibility.AccessibleRole;
@@ -52,6 +54,7 @@
 import javax.accessibility.AccessibleValue;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
+import javax.swing.event.EventListenerList;
 import javax.swing.plaf.SliderUI;
 
 
