--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/cert/X509Certificate.java	2004-10-28 02:27:36.000000000 -0600
+++ java/security/cert/X509Certificate.java	2004-10-28 02:16:12.000000000 -0600
@@ -135,9 +135,7 @@
  * @author Mark Benvenuto
  * @author Casey Marshall (rsdio@metastatic.org)
  */
-public abstract class X509Certificate
-  extends java.security.cert.Certificate  // XXX workaround for gcj bug #17845
-  implements X509Extension
+public abstract class X509Certificate extends Certificate implements X509Extension
 {
   private static final long serialVersionUID = -2491127588187038216L;
 
