--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/SocketChannel.java	2002-12-21 04:42:37.000000000 -0700
+++ java/nio/channels/SocketChannel.java	2004-02-27 02:18:22.000000000 -0700
@@ -87,11 +87,7 @@
   public static SocketChannel open (SocketAddress remote) throws IOException
   {
     SocketChannel ch = open ();
-	
-    if (ch.connect (remote))
-      {
-      }
-    
+    ch.connect(remote);
     return ch;
   }
     
