--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/LogStream.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/server/LogStream.java	2003-03-31 02:17:49.000000000 -0700
@@ -1,5 +1,5 @@
-/* LogStream.java --
-   Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,6 +39,7 @@
 
 import java.io.PrintStream;
 import java.io.OutputStream;
+import java.io.IOException;
 
 /**
  * @deprecated
