--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/event/DocumentEvent.java	2004-01-11 02:20:39.000000000 -0700
+++ javax/swing/event/DocumentEvent.java	2004-06-18 02:18:57.000000000 -0600
@@ -87,68 +87,49 @@
 
 	} // ElementChange
 
-	/**
-	 * EventType
-	 */
-	class EventType {
-
-		//-------------------------------------------------------------
-		// Variables --------------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * INSERT
-		 */
-		EventType INSERT = new EventType("INSERT"); // TODO
-
-		/**
-		 * REMOVE
-		 */
-		EventType REMOVE = new EventType("REMOVE"); // TODO
-
-		/**
-		 * CHANGE
-		 */
-		EventType CHANGE = new EventType("CHANGE"); // TODO
-
-		/**
-		 * typeString
-		 */
-		private String type;
-
-
-		//-------------------------------------------------------------
-		// Initialization ---------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * Constructor EventType
-		 * @param type TODO
-		 */
-		private EventType(String type) {
-			this.type = type;
-		} // EventType()
-
-
-		//-------------------------------------------------------------
-		// Methods ----------------------------------------------------
-		//-------------------------------------------------------------
-
-		/**
-		 * toString
-		 * @returns String
-		 */
-		public String toString() {
-			return type; // TODO
-		} // toString()
-
-
-	} // EventType
-
-
-	//-------------------------------------------------------------
-	// Methods ----------------------------------------------------
-	//-------------------------------------------------------------
+  /**
+   * EventType
+   */
+  class EventType
+  {
+    /**
+     * INSERT
+     */
+    public static final EventType INSERT = new EventType("INSERT"); // TODO
+
+    /**
+     * REMOVE
+     */
+    public static final EventType REMOVE = new EventType("REMOVE"); // TODO
+
+    /**
+     * CHANGE
+     */
+    public static final EventType CHANGE = new EventType("CHANGE"); // TODO
+
+    /**
+     * typeString
+     */
+    private String type;
+
+    /**
+     * Constructor EventType
+     * @param type TODO
+     */
+    private EventType(String type)
+    {
+      this.type = type;
+    }
+
+    /**
+     * toString
+     * @returns String
+     */
+    public String toString()
+    {
+      return type;
+    }
+  }
 
 	/**
 	 * getType
