--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JApplet.java	2004-02-04 02:21:40.000000000 -0700
+++ javax/swing/JApplet.java	2004-05-27 02:19:03.000000000 -0600
@@ -49,7 +49,7 @@
 import java.awt.event.WindowEvent;
 import javax.accessibility.AccessibleContext;
 
-public class JApplet extends Applet
+public class JApplet extends Applet implements RootPaneContainer
 {
 
     public final static int HIDE_ON_CLOSE        = 0;
@@ -87,20 +87,20 @@
   public  void setLayout(LayoutManager manager)
   {    super.setLayout(manager);  }
 
-    void setLayeredPane(JLayeredPane layeredPane) 
+   public void setLayeredPane(JLayeredPane layeredPane) 
     {   getRootPane().setLayeredPane(layeredPane);   }
   
-    JLayeredPane getLayeredPane()
+   public JLayeredPane getLayeredPane()
     {   return getRootPane().getLayeredPane();     }
   
-    JRootPane getRootPane()
+   public JRootPane getRootPane()
     {
         if (rootPane == null)
             setRootPane(createRootPane());
         return rootPane;          
     }
 
-    void setRootPane(JRootPane root)
+   public void setRootPane(JRootPane root)
     {
         if (rootPane != null)
             remove(rootPane);
@@ -109,19 +109,19 @@
         add(rootPane, BorderLayout.CENTER);
     }
 
-    JRootPane createRootPane()
+   public JRootPane createRootPane()
     {   return new JRootPane();    }
 
-    Container getContentPane()
+   public Container getContentPane()
     {    return getRootPane().getContentPane();     }
 
-    void setContentPane(Container contentPane)
+   public void setContentPane(Container contentPane)
     {    getRootPane().setContentPane(contentPane);    }
   
-    Component getGlassPane()
+   public  Component getGlassPane()
     {    return getRootPane().getGlassPane();   }
   
-    void setGlassPane(Component glassPane)
+   public void setGlassPane(Component glassPane)
     {   getRootPane().setGlassPane(glassPane);   }
 
 
