--- /home/tromey/gnu/classpath/classpath/java/net/URLStreamHandler.java	Mon Dec 30 15:39:05 2002
+++ java/net/URLStreamHandler.java	Mon Dec 30 15:43:47 2002
@@ -426,11 +426,9 @@
     int size = protocol.length() + host.length() + file.length() + 24;
     StringBuffer sb = new StringBuffer(size);
 
-    if (protocol != null && protocol.length() > 0){
-	sb.append(protocol);
-	sb.append(":");
-    }
-  
+    sb.append(protocol);
+    sb.append(':');
+
     if (host.length() != 0)
       sb.append("//").append(host);
 
