--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectInputStream.java	2004-02-10 02:21:46.000000000 -0700
+++ java/io/ObjectInputStream.java	2004-02-07 02:18:36.000000000 -0700
@@ -491,6 +491,7 @@
     ObjectStreamField[] stream_fields = osc.fields;
     ObjectStreamField[] real_fields = ObjectStreamClass.lookup(clazz).fields;
     ObjectStreamField[] fieldmapping = new ObjectStreamField[2 * Math.max(stream_fields.length, real_fields.length)];
+    osc.fieldMapping = fieldmapping;
 
     int stream_idx = 0;
     int real_idx = 0;
@@ -542,21 +543,9 @@
 	  }
 	if (real_field != null && !real_field.isToSet())
 	    real_field = null;
-	/* If some of stream_fields does not correspond to any of real_fields,
-	 * or the opposite, then fieldmapping will go short.
-	 */
-	if (map_idx == fieldmapping.length)
-	  {
-	    ObjectStreamField[] newfieldmapping =
-	      new ObjectStreamField[fieldmapping.length + 2];
-	    System.arraycopy(fieldmapping, 0,
-	      newfieldmapping, 0, fieldmapping.length);
-	    fieldmapping = newfieldmapping;
-	  }
 	fieldmapping[map_idx++] = stream_field;
 	fieldmapping[map_idx++] = real_field;
       }
-    osc.fieldMapping = fieldmapping;
 
     return osc;
   }
@@ -1748,7 +1737,11 @@
    * @param sm SecurityManager instance which should be called.
    * @return The current class loader in the calling stack.
    */
-  private static native ClassLoader currentClassLoader (SecurityManager sm);
+  private static ClassLoader currentClassLoader (SecurityManager sm)
+  {
+    // FIXME: This is too simple.
+    return ClassLoader.getSystemClassLoader ();
+  }
 
   private void callReadMethod (Method readObject, Class klass, Object obj) throws IOException
   {
