--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/SecureRandom.java	2004-04-28 02:20:17.000000000 -0600
+++ java/security/SecureRandom.java	2003-05-10 02:19:17.000000000 -0600
@@ -37,10 +37,11 @@
 
 package java.security;
 
-import gnu.java.security.Engine;
-
-import java.util.Enumeration;
+import java.io.Serializable;
 import java.util.Random;
+import java.util.Enumeration;
+
+import gnu.java.security.Engine;
 
 /**
  * An interface to a cryptographically secure pseudo-random number
@@ -93,7 +94,7 @@
     String key;
 
     String classname = null;
-    int i;
+    int i, flag = 0;
     Enumeration e;
     for (i = 0; i < p.length; i++)
       {
