--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/CollationKey.java	2004-04-18 02:22:23.000000000 -0600
+++ java/text/CollationKey.java	2004-01-08 02:18:06.000000000 -0700
@@ -78,13 +78,24 @@
   /**
    * This is the bit value for this key.
    */
-  private byte[] key;
+  private int[] key;
 
-  CollationKey (Collator collator, String originalText, byte[] key)
+  CollationKey(Collator collator, CollationElementIterator iter,
+	       String originalText, int strength)
   {
     this.collator = collator;
     this.originalText = originalText;
-    this.key = key;
+
+    // Compute size of required array.
+    int size = 0;
+    while (RuleBasedCollator.next(iter, strength)
+	   != CollationElementIterator.NULLORDER)
+      ++size;
+
+    iter.reset();
+    key = new int[size];
+    for (int i = 0; i < size; i++)
+      key[i] = RuleBasedCollator.next(iter, strength);
   }
 
   /**
@@ -129,12 +140,12 @@
    * this object.  This will be true if and only if:
    * <p>
    * <ul>
-   * <li>The specified object must not be <code>null</code></li>
-   * <li>The specified object is an instance of <code>CollationKey</code>.</li>
+   * <li>The specified object must not be <code>null</code>
+   * <li>The specified object is an instance of <code>CollationKey</code>.
    * <li>The specified object was created from the same <code>Collator</code>
-   * as this object.</li>
+   * as this object.
    * <li>The specified object has the same source string and bit key as
-   * this object.</li>
+   * this object.
    * </ul>
    *
    * @param obj The <code>Object</code> to test for equality.
@@ -194,6 +205,15 @@
    */
   public byte[] toByteArray()
   {
-    return key;
+    byte[] r = new byte[4 * key.length];
+    int off = 0;
+    for (int i = 0; i < key.length; ++i)
+      {
+	r[off++] = (byte) ((key[i] >>> 24) & 255);
+	r[off++] = (byte) ((key[i] >>> 16) & 255);
+	r[off++] = (byte) ((key[i] >>>  8) & 255);
+	r[off++] = (byte) ((key[i]       ) & 255);
+      }
+    return r;
   }
 }
