--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Compiler.java	2004-08-17 02:28:53.000000000 -0600
+++ java/lang/Compiler.java	2004-04-21 02:18:09.000000000 -0600
@@ -1,5 +1,5 @@
 /* Compiler.java -- placeholder for Java-to-native runtime compilers
-   Copyright (C) 1998, 1999, 2001, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -76,7 +76,8 @@
    */
   public static boolean compileClass(Class oneClass)
   {
-    return VMCompiler.compileClass(oneClass);
+    // Never succeed.
+    return false;
   }
 
   /**
@@ -89,7 +90,8 @@
    */
   public static boolean compileClasses(String classNames)
   {
-    return VMCompiler.compileClasses(classNames);
+    // Note the incredibly lame interface.  Always fail.
+    return false;
   }
 
   /**
@@ -103,7 +105,8 @@
    */
   public static Object command(Object arg)
   {
-    return VMCompiler.command(arg);
+    // Our implementation defines this to a no-op.
+    return null;
   }
 
   /**
@@ -113,7 +116,6 @@
    */
   public static void enable()
   {
-    VMCompiler.enable();
   }
 
   /**
@@ -122,6 +124,5 @@
    */
   public static void disable()
   {
-    VMCompiler.disable();
   }
 }
