--- /home/tromey/gnu/classpath/classpath/java/awt/Menu.java	Mon Dec 30 15:39:04 2002
+++ java/awt/Menu.java	Tue Jan 22 16:18:21 2002
@@ -95,8 +95,6 @@
 /**
   * Initializes a new instance of <code>Menu</code> with no label and that
   * is not a tearoff;
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 Menu()
@@ -110,8 +108,6 @@
   * that has the specified label.
   *
   * @param label The menu label.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 Menu(String label)
@@ -128,8 +124,6 @@
   * @param label The label for this menu
   * @param isTearOff <code>true</code> if this menu is a tear off menu,
   * <code>false</code> otherwise.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 Menu(String label, boolean isTearOff)
@@ -140,9 +134,6 @@
 
   if (label.equals("Help"))
     isHelpMenu = true;
-
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
