--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/ArrayIndexOutOfBoundsException.java	Sat Mar  2 12:13:10 2002
+++ java/lang/ArrayIndexOutOfBoundsException.java	Tue Jan 22 16:01:33 2002
@@ -1,6 +1,6 @@
 /* ArrayIndexOutOfBoundsException.java -- exception thrown when accessing
    an illegal index.
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -8,7 +8,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -39,49 +39,49 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
+ */
+
+
 /**
- * Thrown when attempting to access a position outside the valid range of
- * an array. For example:<br>
- * <pre>
- * int[] i = { 1 };
- * i[1] = 2;
- * </pre>
+ * Exceptions may be thrown by one part of a Java program and caught
+ * by another in order to deal with exceptional conditions, in this case
+ * when trying to access an illegal index.  This exception is thrown when
+ * accessing an index which is either negative or greater than the size of
+ * the array minus one.
+ *
+ * @since JDK 1.0
  *
  * @author Brian Jones
  * @author Warren Levy <warrenl@cygnus.com>
- * @status updated to 1.4
+ * @date September 18, 1998.
  */
 public class ArrayIndexOutOfBoundsException extends IndexOutOfBoundsException
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = -5116101128118950844L;
+  static final long serialVersionUID = -5116101128118950844L;
 
   /**
    * Create an exception without a message.
    */
-  public ArrayIndexOutOfBoundsException()
-  {
+  public ArrayIndexOutOfBoundsException() {
+    super();
   }
 
   /**
    * Create an exception with a message.
-   *
-   * @param s the message
    */
-  public ArrayIndexOutOfBoundsException(String s)
-  {
+  public ArrayIndexOutOfBoundsException(String s) {
     super(s);
   }
 
   /**
    * Create an exception indicating the illegal index.
-   *
-   * @param index the invalid index
    */
-  public ArrayIndexOutOfBoundsException(int index)
-  {
+  public ArrayIndexOutOfBoundsException(int index) {
     super("Array index out of range: " + index);
   }
+
 }
