--- /home/tromey/gnu/egcs/classpath/classpath/java/util/zip/DataFormatException.java	Wed Sep 26 08:18:48 2001
+++ java/util/zip/DataFormatException.java	Fri Nov 17 14:42:28 2000
@@ -1,5 +1,5 @@
-/* java.util.zip.DataFormatException
-   Copyright (C) 2001 Free Software Foundation, Inc.
+/* DataformatException.java - Exception thrown when compressed data is corrupt
+   Copyright (C) 1999, 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -27,26 +27,26 @@
 package java.util.zip;
 
 /**
- * May be thrown by methods that are decompressing (inflating) compressed data.
- *
- *
- * @author John Leuner
- * @since JDK 1.1
+ * @author Tom Tromey
+ * @date May 17, 1999
  */
 
+/* Written using on-line Java Platform 1.2 API Specification.
+ * Believed complete and correct.
+ */
 
-public class DataFormatException extends Exception 
+/**
+ * Exception thrown when compressed data is corrupt.
+ */
+public class DataFormatException extends Exception
 {
-  public DataFormatException()
+  public DataFormatException ()
   {
-    this("");
+    super();
   }
 
-  public DataFormatException(String m)
+  public DataFormatException (String msg)
   {
-    super(m);
+    super(msg);
   }
 }
-
-
-
