--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/Skeleton.java	Sun Mar 23 02:18:41 2003
+++ java/rmi/server/Skeleton.java	Mon Mar 31 02:17:49 2003
@@ -38,7 +38,6 @@
 package java.rmi.server;
 
 import java.rmi.Remote;
-import java.lang.Exception;
 import java.rmi.server.RemoteCall;
 
 /**
@@ -46,8 +45,14 @@
  */
 public interface Skeleton
 {
+  /**
+   * @deprecated
+   */
   public void dispatch (Remote obj, RemoteCall theCall, int opnum, long hash)
     throws Exception;
 
+  /**
+   * @deprecated
+   */
   public Operation[] getOperations();
 }
