--- /home/tromey/gnu/classpath/classpath/java/nio/charset/CharacterCodingException.java	Sat Nov  9 16:09:32 2002
+++ java/nio/charset/CharacterCodingException.java	Mon Oct  7 01:59:12 2002
@@ -39,10 +39,7 @@
 
 import java.io.IOException;
 
-/**
- * @since 1.4
- */
-public class CharacterCodingException extends IOException
+class CharacterCodingException extends IOException
 {
   /**
    * Creates the exception
