--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JCheckBox.java	2003-06-16 02:24:06.000000000 -0600
+++ javax/swing/JCheckBox.java	2004-02-12 02:18:27.000000000 -0700
@@ -70,6 +70,8 @@
     public JCheckBox(String text, Icon icon)
     {
 	super(text, icon);
+        paint_border = false;
+        content_area_filled = false;
     }
 
     
@@ -82,7 +84,7 @@
     public String getUIClassID()
     {
 	//Returns a string that specifies the name of the L&F class that renders this component.  
-	return "JCheckBox";
+	return "CheckBoxUI";
     }
   
     protected  String paramString()
