--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/ConsoleHandler.java	2004-04-18 02:22:28.000000000 -0600
+++ java/util/logging/ConsoleHandler.java	2003-06-21 04:31:55.000000000 -0600
@@ -67,7 +67,7 @@
  *
  * <li><code>java.util.logging.ConsoleHandler.encoding</code> - specifies
  *     the name of the character encoding. Default value:
- *     the default platform encoding.</li>
+ *     the default platform encoding.
  *
  * </ul>
  *
