--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2003-11-15 02:20:01.000000000 -0700
+++ javax/swing/UIManager.java	2004-05-27 02:19:06.000000000 -0600
@@ -178,8 +178,7 @@
    */
   public static Dimension getDimension(Object key)
   {
-    System.out.println("UIManager.getDim");
-    return new Dimension(200,100);
+    return (Dimension) getLookAndFeel().getDefaults().get(key);
   }
 
   /**
