--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2004-06-08 02:25:41.000000000 -0600
+++ javax/swing/UIManager.java	2004-06-08 02:20:01.000000000 -0600
@@ -54,15 +54,22 @@
   {
     String name, clazz;
 	
-    LookAndFeelInfo(String name, 
-                    String clazz)
+    public LookAndFeelInfo(String name, 
+			   String clazz)
     {
       this.name  = name;
       this.clazz = clazz;
     }
 
-    String getName()      { return name;  }
-    String getClassName() { return clazz; }
+    public String getName()
+    {
+      return name;
+    }
+    
+    public String getClassName()
+    {
+      return clazz;
+    }
   }
 
   private static final long serialVersionUID = -5547433830339189365L;
