--- /home/tromey/gnu/classpath/classpath/java/net/Socket.java	Mon Dec  2 14:22:32 2002
+++ java/net/Socket.java	Thu Oct  3 11:50:56 2002
@@ -892,9 +892,6 @@
     if (sm != null)
       sm.checkSetFactory();
 
-    if (fac == null)
-      throw new SocketException("SocketImplFactory cannot be null");
-
     factory = fac;
   }
 
@@ -1031,6 +1028,15 @@
   }
 
   /**
+   * Checks if the socket is closed.
+   */
+  public boolean isClosed ()
+  {
+    // FIXME: implement this.
+    return false;
+  }
+
+  /**
    * Checks if the socket's input stream is shutdown
    */
   public boolean isInputShutdown ()
