--- /home/tromey/gnu/classpath/classpath/java/lang/reflect/Proxy.java	Mon Dec 30 15:39:04 2002
+++ java/lang/reflect/Proxy.java	Tue Dec 10 19:20:31 2002
@@ -185,7 +185,7 @@
    * Proxy.
    * @serial invocation handler associated with this proxy instance
    */
-  protected InvocationHandler h;
+  protected final InvocationHandler h;
 
   /**
    * Constructs a new Proxy from a subclass (usually a proxy class),
