--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/Socket.java	2004-09-21 02:40:01.000000000 -0600
+++ java/net/Socket.java	2004-07-22 02:19:23.000000000 -0600
@@ -479,8 +479,7 @@
 
   /**
    * Returns the local address to which this socket is bound.  If this socket
-   * is not connected, then a wildcard address, for which
-   * @see isAnyLocalAddress() is <code>true</code>, is returned.
+   * is not connected, then <code>null</code> is returned.
    *
    * @return The local address
    *
@@ -489,7 +488,7 @@
   public InetAddress getLocalAddress()
   {
     if (! isBound())
-      return InetAddress.ANY_IF;
+      return null;
 
     InetAddress addr = null;
 
