--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/MulticastSocket.java	2003-12-03 02:20:29.000000000 -0700
+++ java/net/MulticastSocket.java	2004-02-12 02:18:26.000000000 -0700
@@ -255,7 +255,7 @@
     if (isClosed())
       throw new SocketException("socket is closed");
 
-    getImpl().setOption (SocketOptions.IP_MULTICAST_LOOP, new Boolean (disable));
+    getImpl().setOption (SocketOptions.IP_MULTICAST_LOOP, Boolean.valueOf(disable));
   }
 
   /**
