--- /home/tromey/gnu/classpath/classpath/java/nio/charset/CodingErrorAction.java	Sat Nov  9 16:09:33 2002
+++ java/nio/charset/CodingErrorAction.java	Fri Nov  1 05:03:39 2002
@@ -37,30 +37,18 @@
 
 package java.nio.charset;
 
-public class CodingErrorAction
-{
-  public static final CodingErrorAction IGNORE
-	  = new CodingErrorAction("ignore");
-  public static final CodingErrorAction REPLACE
-	  = new CodingErrorAction("replace");
-  public static final CodingErrorAction REPORT
-	  = new CodingErrorAction("report");
-
-  private final String name;
 
-  /**
-   * Private constructor only used to create the constant CodingErrorActions.
-   */
-  private CodingErrorAction(String name)
-  {
-    this.name = name;
-  }
+class CodingErrorAction
+{
+  public static final CodingErrorAction IGNORE;
+  public static final CodingErrorAction REPLACE;
+  public static final CodingErrorAction REPORT;
 
   /**
-   * Returns the name of the CodingErrorAction.
+   * FIXME
    */
   public String toString ()
   {
-    return name;
+    return "";
   }
 }
