--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2003-06-16 02:24:07.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-02-12 02:18:27.000000000 -0700
@@ -86,7 +86,7 @@
     public String getUIClassID()
     {
 	//Returns a string that specifies the name of the L&F class that renders this component.  
-	return "JScrollPane";
+	return "ScrollPaneUI";
     }
 
     public JViewport getViewport()
