--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/NoRouteToHostException.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/NoRouteToHostException.java	2002-06-15 13:45:54.000000000 -0600
@@ -35,8 +35,8 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-package java.net;
 
+package java.net;
 
 /**
   * This exception indicates that there is no TCP/IP route to the requested
