--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Security.java	2004-10-12 02:18:14.000000000 -0600
+++ java/security/Security.java	2004-07-18 02:18:50.000000000 -0600
@@ -37,12 +37,10 @@
 
 package java.security;
 
-import gnu.java.security.action.GetPropertyAction;
 import java.io.FileNotFoundException;
 import java.io.InputStream;
 import java.io.IOException;
 import java.net.URL;
-import java.security.AccessController;
 import java.security.Provider;
 import java.util.Collections;
 import java.util.Enumeration;
@@ -70,11 +68,9 @@
   
   static
     {
-      GetPropertyAction getProp = new GetPropertyAction("gnu.classpath.home.url");
-      String base = (String) AccessController.doPrivileged(getProp);
-      getProp = new GetPropertyAction("gnu.classpath.vm.shortname");
-      String vendor = (String) AccessController.doPrivileged(getProp);
-
+      String base = System.getProperty ("gnu.classpath.home.url");
+      String vendor = System.getProperty ("gnu.classpath.vm.shortname");
+    
       // Try VM specific security file
       boolean loaded = loadProviders (base, vendor);
     
