--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/UnsupportedOperationException.java	Sat Mar  2 12:13:13 2002
+++ java/lang/UnsupportedOperationException.java	Tue Jan 22 16:01:35 2002
@@ -1,6 +1,6 @@
-/* UnsupportedOperationException.java -- thrown when an operation is not
-   supported
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* UnsupportedOperationException.java -- Exception thrown when an
+   unsupported operation is attempted on an object
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -8,7 +8,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -39,34 +39,37 @@
 
 package java.lang;
 
-/**
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
+ */
+
+/** 
  * This exception is thrown by an object when an operation is
  * requested of it that it does not support.
  *
+ * @since JDK 1.2
+ *
  * @author Warren Levy <warrenl@cygnus.com>
- * @since 1.2
- * @status updated to 1.4
+ * @date September 18, 1998.
  */
 public class UnsupportedOperationException extends RuntimeException
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = -1242599979055084673L;
+  static final long serialVersionUID = -1242599979055084673L;
 
   /**
    * Create an exception without a message.
    */
-  public UnsupportedOperationException()
+  public UnsupportedOperationException() 
   {
+    super();
   }
 
   /**
    * Create an exception with a message.
-   *
-   * @param s the message
    */
-  public UnsupportedOperationException(String s)
+  public UnsupportedOperationException( String s )
   {
     super(s);
   }
