--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/prefs/AbstractPreferences.java	2003-06-19 02:34:49.000000000 -0600
+++ java/util/prefs/AbstractPreferences.java	2003-12-19 02:18:14.000000000 -0700
@@ -183,6 +183,16 @@
     }
 
     /**
+     * Returns all known unremoved children of this node.
+     *
+     * @return All known unremoved children of this node
+     */
+    protected final AbstractPreferences[] cachedChildren()
+    {
+      return (AbstractPreferences[]) childCache.values().toArray();
+    }
+
+    /**
      * Returns all the direct sub nodes of this preferences node.
      * Needs access to the backing store to give a meaningfull answer.
      * <p>
