--- /home/tromey/gnu/classpath/classpath/java/util/zip/ZipFile.java	Thu Feb 20 18:57:11 2003
+++ java/util/zip/ZipFile.java	Thu Jan  2 19:57:45 2003
@@ -305,15 +305,6 @@
   }
 
   /**
-   * Calls the <code>close()</code> method when this ZipFile has not yet
-   * been explicitly closed.
-   */
-  protected void finalize() throws IOException
-  {
-    if (!closed) close();
-  }
-
-  /**
    * Returns an enumeration of all Zip entries in this Zip file.
    */
   public Enumeration entries()
