--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Math.java	2003-09-26 02:21:24.000000000 -0600
+++ java/lang/Math.java	2003-08-27 02:17:51.000000000 -0600
@@ -575,9 +575,6 @@
    */
   public static int round(float a)
   {
-    // this check for NaN, from JLS 15.21.1, saves a method call
-    if (a != a)
-      return 0;
     return (int) floor(a + 0.5f);
   }
 
@@ -594,9 +591,6 @@
    */
   public static long round(double a)
   {
-    // this check for NaN, from JLS 15.21.1, saves a method call
-    if (a != a)
-      return 0;
     return (long) floor(a + 0.5d);
   }
 
@@ -630,7 +624,7 @@
    */
   public static double toRadians(double degrees)
   {
-    return (degrees * PI) / 180;
+    return degrees * (PI / 180);
   }
 
   /**
@@ -644,6 +638,6 @@
    */
   public static double toDegrees(double rads)
   {
-    return (rads * 180) / PI;
+    return rads * (180 / PI);
   }
 }
