--- /home/tromey/gnu/classpath/classpath/java/net/ServerSocket.java	Mon Dec 30 15:39:05 2002
+++ java/net/ServerSocket.java	Tue Dec 10 19:20:31 2002
@@ -77,8 +77,6 @@
    */
   private ServerSocketChannel ch;
 
-  private boolean closed = false;
-  
   /**
    * Constructor that simply sets the implementation.
    * 
@@ -202,9 +200,6 @@
    */
   public void bind (SocketAddress endpoint, int backlog) throws IOException
   {
-    if (closed)
-      throw new SocketException ("ServerSocket is closed");
-    
     if (impl == null)
       throw new IOException ("Cannot initialize Socket implementation");
 
@@ -320,13 +315,7 @@
    */
   public void close () throws IOException
   {
-    if (impl != null)
-      impl.close ();
-
-    if (ch != null)
-      ch.close ();
-    
-    closed = true;
+    impl.close();
   }
 
   /**
@@ -356,7 +345,7 @@
       }
     catch (SocketException e)
       {
-        return false;
+	return false;
       }
     
     return true;
@@ -369,7 +358,8 @@
    */
   public boolean isClosed()
   {
-    return closed;
+    // FIXME: implement this
+    return false;
   }
 
   /**
