--- /home/tromey/gnu/classpath/classpath/java/awt/GridBagLayout.java	Thu Oct 31 12:43:54 2002
+++ java/awt/GridBagLayout.java	Thu Oct 31 12:29:32 2002
@@ -40,13 +40,11 @@
 import java.io.Serializable;
 
 /**
- * Stub implementation.
+ * Stub implementeation.
  */
 public class GridBagLayout
   implements Serializable, LayoutManager2
 {
-  static final long serialVersionUID = 8838754796412211005L;
-
   public void addLayoutComponent(String name, Component component)
   {
   }
