--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/regex/Pattern.java	2004-04-28 02:20:17.000000000 -0600
+++ java/util/regex/Pattern.java	2004-04-22 02:18:16.000000000 -0600
@@ -67,6 +67,12 @@
 
   private final RE re;
 
+  private Pattern (String regex)
+    throws PatternSyntaxException
+  {
+    this (regex, 0);
+  }
+
   private Pattern (String regex, int flags)
     throws PatternSyntaxException
   {
