--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/ResourceBundle.java	2004-04-28 02:20:17.000000000 -0600
+++ java/util/ResourceBundle.java	2004-04-21 02:18:11.000000000 -0600
@@ -42,8 +42,7 @@
 import java.lang.ref.SoftReference;
 import java.io.InputStream;
 import java.io.IOException;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
+import gnu.classpath.Configuration;
 
 /**
  * A resource bundle contains locale-specific data. If you need localized
@@ -104,62 +103,7 @@
    */
   private Locale locale;
 
-  private static Class resourceBundleClass;
-  private static Class securityClass;
-
-  static
-  {
-    try
-      {
-	resourceBundleClass = Class.forName("java.util.ResourceBundle");
-	securityClass = Class.forName("java.util.ResourceBundle$Security");
-      }
-    catch (ClassNotFoundException e)
-      {
-      }
-  }
-      
-  /**
-   * We override SecurityManager in order to access getClassContext().
-   */
-  private static final class Security extends SecurityManager
-  {
-    /**
-     * Avoid accessor method of private constructor.
-     */
-    Security()
-    {
-    }
-
-    /**
-     * Return the ClassLoader of the class which called into this
-     * ResourceBundle, or null if it cannot be determined.
-     */
-    ClassLoader getCallingClassLoader()
-    {
-      Class[] stack = getClassContext();
-      for (int i = 0; i < stack.length; i++)
-	{
-	  if (stack[i] != securityClass && stack[i] != resourceBundleClass)
-	    return stack[i].getClassLoader();
-	}
-
-      return null;
-    }
-  }
-
-  /** A security context for grabbing the correct class loader. */
-  private static final Security security
-    = (Security) AccessController.doPrivileged(new PrivilegedAction()
-      {
-        // This will always work since java.util classes have (all) system
-        // permissions.
-        public Object run()
-        {
-          return new Security();
-        }
-      }
-    );
+  private static native ClassLoader getCallingClassLoader();
 
   /**
    * The resource bundle cache. This is a two-level hash map: The key
@@ -280,7 +224,7 @@
   public static final ResourceBundle getBundle(String baseName)
   {
     return getBundle(baseName, Locale.getDefault(),
-                     security.getCallingClassLoader());
+                     getCallingClassLoader());
   }
 
   /**
@@ -298,7 +242,7 @@
   public static final ResourceBundle getBundle(String baseName,
                                                Locale locale)
   {
-    return getBundle(baseName, locale, security.getCallingClassLoader());
+    return getBundle(baseName, locale, getCallingClassLoader());
   }
 
   /**
@@ -401,6 +345,7 @@
     else if (cache.containsKey(name))
       {
 	Reference ref = (Reference) cache.get(name);
+	ResourceBundle result = null;
 	// If REF is null, that means that we added a `null' value to
 	// the hash map.  That means we failed to find the bundle
 	// previously, and we cached that fact.  The JDK does this, so
@@ -495,6 +440,7 @@
     if (cache.containsKey(localizedName))
       {
 	Reference ref = (Reference) cache.get(localizedName);
+	ResourceBundle result = null;
 	// If REF is null, that means that we added a `null' value to
 	// the hash map.  That means we failed to find the bundle
 	// previously, and we cached that fact.  The JDK does this, so
