--- /home/tromey/gnu/Nightly/classpath/classpath/javax/security/auth/login/Configuration.java	2004-10-28 02:27:41.000000000 -0600
+++ javax/security/auth/login/Configuration.java	2004-09-21 02:33:33.000000000 -0600
@@ -106,15 +106,4 @@
   public abstract AppConfigurationEntry[] getAppConfigurationEntry (String applicationName);
 
   public abstract void refresh();
-
-  // Package-private methods.
-  // -------------------------------------------------------------------------
-
-  /**
-   * Get the current configuration, bypassing security checks.
-   */
-  static Configuration getConfig()
-  {
-    return config;
-  }
 }
