--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/MarshalledObject.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/MarshalledObject.java	2004-04-21 02:18:11.000000000 -0600
@@ -1,5 +1,5 @@
-/* MarshalledObject.java --
-   Copyright (c) 1996, 1997, 1998, 1999, 2004 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,16 +37,18 @@
 
 package java.rmi;
 
+import java.io.Serializable;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 import gnu.java.rmi.RMIMarshalledObjectInputStream;
 import gnu.java.rmi.RMIMarshalledObjectOutputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.Serializable;
 
 /**
  * FIXME - doc missing
  */
 public final class MarshalledObject implements Serializable 
 {
+  
   //The following fields are from Java API Documentation "Serialized form"
   private static final long serialVersionUID = 8988374069173025854L;
   byte[] objBytes;
