--- /home/tromey/gnu/classpath/classpath/java/net/NetPermission.java	Tue Jan 22 15:49:58 2002
+++ java/net/NetPermission.java	Wed Sep 25 13:48:13 2002
@@ -50,6 +50,8 @@
 public final class NetPermission extends BasicPermission
   implements java.io.Serializable
 {
+  static final long serialVersionUID = -8343910153355041693L;
+
   /**
    * Initializes a new instance of <code>NetPermission</code> with the
    * specified name.
