--- /home/tromey/gnu/Nightly/classpath/classpath/javax/rmi/CORBA/PortableRemoteObjectDelegate.java	2004-04-20 02:21:38.000000000 -0600
+++ javax/rmi/CORBA/PortableRemoteObjectDelegate.java	2003-03-29 14:23:25.000000000 -0700
@@ -1,5 +1,5 @@
 /* PortableRemoteObjectDelegate.java -- Interface supporting PortableRemoteObject
-   Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,9 +38,7 @@
 
 package javax.rmi.CORBA;
 
-import java.rmi.NoSuchObjectException;
-import java.rmi.Remote;
-import java.rmi.RemoteException;
+import java.rmi.*;
 
 /**
  * A delegate is a singleton class that support delegation for method
