--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/MarshalledObject.java	2002-10-31 11:35:23.000000000 -0700
+++ java/rmi/MarshalledObject.java	2003-11-12 02:18:14.000000000 -0700
@@ -76,7 +76,7 @@
   
   public boolean equals(Object obj) 
   {
-    if(obj == null || !(obj instanceof MarshalledObject) )
+    if (! (obj instanceof MarshalledObject))
       return false;
 
     // hashCode even differs, don't do the time-consuming comparisons
