--- /home/tromey/gnu/egcs/classpath/classpath/java/security/InvalidParameterException.java	Wed Mar 22 12:26:19 2000
+++ java/security/InvalidParameterException.java	Wed Apr 25 11:13:32 2001
@@ -1,5 +1,4 @@
-/* InvalidParameterException.java -- What it says
-   Copyright (C) 1998 Free Software Foundation, Inc.
+/* Copyright (C) 2000  Free Software Foundation
 
 This file is part of GNU Classpath.
 
@@ -7,7 +6,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,19 +23,27 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 
 /**
-  * This exception is thrown when an invalid parameter is passed to
-  * a method.
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
-public class InvalidParameterException extends IllegalArgumentException
-{
+ * @author Warren Levy <warrenl@cygnus.com>
+ * @date February 1, 2000.
+ */
 
-} // class InvalidParameterException
+/**
+ * Written using on-line Java Platform 1.2 API Specification.
+ * Status:  Believed complete and correct.
+ */
 
+public class InvalidParameterException extends IllegalArgumentException
+{
+  public InvalidParameterException()
+  {
+    super();
+  }
+
+  public InvalidParameterException(String msg)
+  {
+    super(msg);
+  }
+}
