--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FilePermission.java	2004-03-13 02:21:04.000000000 -0700
+++ java/io/FilePermission.java	2003-12-05 02:18:15.000000000 -0700
@@ -271,9 +271,9 @@
       break;
 */
         default:
-          if(f2.charAt(f2.length() - 1) == File.separatorChar) 
+          if(f2.charAt(f2.length()) == File.separatorChar) 
             {
-              if(!f1.equals(f2.substring(0,f2.length() - 1)))
+              if(!f1.equals(f2.substring(0,f2.length()-1)))
 	        return false;
             } 
           else 
