--- /home/tromey/gnu/classpath/classpath/java/awt/event/WindowEvent.java	Mon Dec 30 15:39:04 2002
+++ java/awt/event/WindowEvent.java	Mon Aug 12 10:11:06 2002
@@ -109,7 +109,7 @@
   public static final int WINDOW_STATE_CHANGED = 209;
 
   /** This is the last id in the range of event ids used by this class. */
-  public static final int WINDOW_LAST = 209;
+  public static final int WINDOW_LAST = 207;
 
   /**
    * The other Window involved in a focus or activation change. For
