--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/font/FontRenderContext.java	2003-09-24 02:19:29.000000000 -0600
+++ java/awt/font/FontRenderContext.java	2003-02-17 08:26:30.000000000 -0700
@@ -1,5 +1,5 @@
 /* FontRenderContext.java
-   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -88,22 +88,11 @@
             && usesFractionalMetrics == rhs.usesFractionalMetrics ());
   }
 
-
-  /**
-   * Retrieves the affine transform for scaling typographical points
-   * to raster pixels.
-   *
-   * @return a clone of the transform object.
-   */
   public AffineTransform getTransform ()
   {
-    if (affineTransform == null)
-      return new AffineTransform ();
-    else
-      return new AffineTransform (affineTransform);
+    return affineTransform;
   }
 
-
   /**
    * Returns the hash code of the font render context.
    */
