--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/HashMap.java	2003-11-08 02:19:25.000000000 -0700
+++ java/util/HashMap.java	2003-08-01 02:18:18.000000000 -0600
@@ -381,7 +381,8 @@
   public void putAll(Map m)
   {
     Iterator itr = m.entrySet().iterator();
-    while (itr.hasNext())
+    int msize = m.size();
+    while (msize-- > 0)
       {
         Map.Entry e = (Map.Entry) itr.next();
         // Optimize in case the Entry is one of our own.
@@ -708,10 +709,10 @@
   void putAllInternal(Map m)
   {
     Iterator itr = m.entrySet().iterator();
-    size = 0;
-    while (itr.hasNext())
+    int msize = m.size();
+    size = msize;
+    while (msize-- > 0)
       {
-        size++;
 	Map.Entry e = (Map.Entry) itr.next();
 	Object key = e.getKey();
 	int idx = hash(key);
