--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Scrollbar.java	2004-02-04 02:21:19.000000000 -0700
+++ java/awt/Scrollbar.java	2004-02-04 02:18:37.000000000 -0700
@@ -550,7 +550,7 @@
 public synchronized void
 setBlockIncrement(int blockIncrement)
 {
-  setPageIncrement (pageIncrement);
+  setPageIncrement (blockIncrement);
 }
 
 /*************************************************************************/
