--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/ReadableByteChannel.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/ReadableByteChannel.java	2003-10-13 02:17:50.000000000 -0600
@@ -1,5 +1,5 @@
-/* ReadableByteChannel.java --
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+/* ReadableByteChannel.java -- 
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,16 +40,11 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 
-
 public interface ReadableByteChannel extends Channel
 {
   /**
    * Reads a sequence of bytes from this channel into the given buffer
    *
-   * @param dst the buffer to put the read data into
-   *
-   * @return the numer of bytes read
-   *
    * @exception AsynchronousCloseException If another thread closes this
    * channel while the read operation is in progress
    * @exception ClosedByInterruptException If another thread interrupts the
@@ -60,5 +55,5 @@
    * @exception NonReadableChannelException If this channel was not opened for
    * reading
    */
-  int read(ByteBuffer dst) throws IOException;
+  int read (ByteBuffer dst) throws IOException;
 }
