--- /home/tromey/gnu/egcs/classpath/classpath/java/util/jar/JarInputStream.java	Tue Nov  7 10:43:04 2000
+++ java/util/jar/JarInputStream.java	Fri Nov  2 21:44:52 2001
@@ -75,7 +75,7 @@
    * tries to verify all the entry signatures while reading.
    *
    * @param in InputStream to read the jar from
-   * @param verify wheter or not to verify the manifest entries
+   * @param verify whether or not to verify the manifest entries
    * @exception IOException when an error occurs when opening or reading
    */
   public JarInputStream(InputStream in, boolean verify) throws IOException
