--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/CharArrayWriter.java	2004-10-03 02:35:36.000000000 -0600
+++ java/io/CharArrayWriter.java	2004-04-21 02:18:08.000000000 -0600
@@ -249,7 +249,7 @@
    * application programmers by setting their initial buffer size big
    * enough to hold everything if possible.
    */
-  private void resize (int len)
+  private final void resize (int len)
   {
     if (count + len >= buf.length)
       {
