--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/DefaultListSelectionModel.java	2004-03-20 02:21:49.000000000 -0700
+++ javax/swing/DefaultListSelectionModel.java	2004-06-18 02:18:51.000000000 -0600
@@ -62,6 +62,7 @@
                                                   ListSelectionModel,
                                                   Serializable
 {
+  private static final long serialVersionUID = -5718799865110415860L;
 
   /** The list of ListSelectionListeners subscribed to this selection model. */
   protected EventListenerList listenerList = new EventListenerList();
