--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Calendar.java	2003-08-13 02:19:48.000000000 -0600
+++ java/util/Calendar.java	2003-12-10 02:18:05.000000000 -0700
@@ -37,8 +37,12 @@
 
 
 package java.util;
+
 import java.lang.reflect.InvocationTargetException;
-import java.io.*;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
 
 /**
  * This class is an abstract base class for Calendars, which can be
