--- /home/tromey/gnu/classpath/classpath/java/rmi/server/UnicastRemoteObject.java	Mon Mar 25 00:05:18 2002
+++ java/rmi/server/UnicastRemoteObject.java	Tue Jan 22 16:01:36 2002
@@ -42,17 +42,10 @@
 import java.rmi.server.RemoteRef;
 import java.rmi.NoSuchObjectException;
 import gnu.java.rmi.server.UnicastServerRef;
-import gnu.java.rmi.server.UnicastServer;
 
 public class UnicastRemoteObject
 	extends RemoteServer {
 
-private static final long serialVersionUID = 4974527148936298033L;
-//The following serialized fields are from Java API Documentation "Serialized form"
-private int port = 0;
-private RMIClientSocketFactory csf = null;
-private RMIServerSocketFactory ssf = null;
-
 protected UnicastRemoteObject() throws RemoteException {
 	this(0);
 }
@@ -62,21 +55,11 @@
 }
 
 protected UnicastRemoteObject(int port, RMIClientSocketFactory csf, RMIServerSocketFactory ssf) throws RemoteException {
-  this.port = port;
-  //Is RMIXXXSocketFactory serializable
-  //this.csf = csf;
-  //this.ssf = ssf;
-  this.ref = new UnicastServerRef(new ObjID(), port, ssf);
-  //Should we export it here?
-  // if we export, we got infinite recursive call:
-  //  UnicastRemoteObject.<init>->...->UnicastServer.startDGC()->UnicastRemoteObject.<init>->...
-  //exportObject(this);
+	super(new UnicastServerRef(new ObjID(), port, ssf));
 }
 
 protected UnicastRemoteObject(RemoteRef ref) throws RemoteException {
 	super((UnicastServerRef)ref);
-       //Should we export it here?
-       //exportObject(this);
 }
 
 public Object clone() throws CloneNotSupportedException {
@@ -88,46 +71,16 @@
 	return (sref.exportObject(obj));
 }
 
-  public static Remote exportObject(Remote obj, int port) throws RemoteException 
-  {
-    return exportObject(obj, port, null);
-  }
-  
-  protected static Remote exportObject(Remote obj, int port, RMIServerSocketFactory ssf) 
-    throws RemoteException 
-  {
-    UnicastServerRef sref = null;
-    if (obj instanceof RemoteObject)
-      sref = (UnicastServerRef)((RemoteObject)obj).getRef ();
-    if(sref == null)
-      {
-	sref = new UnicastServerRef(new ObjID (), port, ssf);
-      }
-    return (sref.exportObject (obj)); 
-  }
-
-  /**
-   * FIX ME
-   */
-  public static Remote exportObject(Remote obj, int port, RMIClientSocketFactory csf, 
-				    RMIServerSocketFactory ssf) 
-    throws RemoteException 
-  {
-    return (exportObject(obj, port, ssf));
-  }
-
-  public static boolean unexportObject(Remote obj, boolean force) 
-    throws RemoteException, NoSuchObjectException 
-  {
-    if (obj instanceof RemoteObject)
-      {
-	UnicastServerRef sref = (UnicastServerRef)((RemoteObject)obj).getRef();
-	return sref.unexportObject(obj, force);
-      }
-    else
-      //FIX ME
-      ;
-    return true;
-  }
+public static Remote exportObject(Remote obj, int port) throws RemoteException {
+	return (exportObject(obj));
+}
+
+public static Remote exportObject(Remote obj, int port, RMIClientSocketFactory csf, RMIServerSocketFactory ssf) throws RemoteException {
+	return (exportObject(obj));
+}
+
+public static boolean unexportObject(Remote obj, boolean force) throws NoSuchObjectException {
+	throw new Error("Not implemented");
+}
 
 }
