--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/spi/DirectoryManager.java	2002-11-23 14:44:28.000000000 -0700
+++ javax/naming/spi/DirectoryManager.java	2003-12-06 02:18:23.000000000 -0700
@@ -38,11 +38,19 @@
 
 package javax.naming.spi;
 
-import javax.naming.*;
-import javax.naming.directory.*;
+import java.util.Enumeration;
 import java.util.Hashtable;
 import java.util.StringTokenizer;
-import java.util.Enumeration;
+import javax.naming.CannotProceedException;
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.NamingException;
+import javax.naming.RefAddr;
+import javax.naming.Reference;
+import javax.naming.Referenceable;
+import javax.naming.StringRefAddr;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
 
 /**
  * @author Tom Tromey <tromey@redhat.com>
