--- /home/tromey/gnu/egcs/classpath/classpath/java/net/BindException.java	Sat Mar  2 12:13:13 2002
+++ java/net/BindException.java	Tue Jan 22 16:01:35 2002
@@ -1,5 +1,5 @@
 /* BindException.java -- An exception occurred while binding to a socket
-   Copyright (C) 1998, 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,35 +38,49 @@
 package java.net;
 
 /**
- * This exception indicates that an error occurred while attempting to bind
- * socket to a particular port.
- *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @author Warren Levy <warrenl@cygnus.com>
- * @status updated to 1.4
+ * Written using on-line Java Platform 1.2 API Specification, as well
+ * as "The Java Class Libraries", 2nd edition (Addison-Wesley, 1998).
+ * Status:  Believed complete and correct.
  */
+
+/**
+  * This exception indicates that an error occurred while attempting to bind
+  * socket to a particular port.
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @author Warren Levy <warrenl@cygnus.com>
+  * @date March 5, 1999.
+  */
 public class BindException extends SocketException
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -5945005768251722951L;
-
-  /**
-   * Create a new instance without a descriptive error message.
-   */
-  public BindException()
-  {
-  }
-
-  /**
-   * Create a new instance with a descriptive error message, such as the
-   * text from strerror(3).
-   *
-   * @param message a message describing the error that occurred
-   */
-  public BindException(String message)
-  {
-    super(message);
-  }
+
+/*
+ * Constructors
+ */
+
+/**
+  * Initializes a new instance of <code>BindException</code> without
+  * a descriptive error message.
+  */
+public
+BindException()
+{
+  super();
+}
+
+/*************************************************************************/
+
+/**
+  * Initializes a new instance of <code>BindException</code> with
+  * a descriptive error message, such as the text from strerror(3).
+  *
+  * @param message A message describing the error that occurred.
+  */
+public
+BindException(String message)
+{
+  super(message);
+}
+
 } // class BindException
+
