--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/WritableByteChannel.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/WritableByteChannel.java	2003-10-13 02:17:50.000000000 -0600
@@ -1,4 +1,4 @@
-/* WritableByteChannel.java --
+/* WritableByteChannel.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -40,8 +40,8 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 
-
-public interface WritableByteChannel extends Channel
+public interface WritableByteChannel
+  extends Channel
 {
   /**
    * Writes a sequence of bytes to this channel from the given buffer
@@ -56,5 +56,5 @@
    * @exception NonWritableChannelException If this channel was not opened for
    * writing
    */
-  int write(ByteBuffer src) throws IOException;
+  int write (ByteBuffer src) throws IOException;
 }
