--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicScrollPaneUI.java	2004-04-29 02:24:29.000000000 -0600
+++ javax/swing/plaf/basic/BasicScrollPaneUI.java	2004-03-18 02:18:23.000000000 -0700
@@ -40,8 +40,8 @@
 
 import java.awt.Dimension;
 import java.awt.Graphics;
-
 import javax.swing.JComponent;
+import javax.swing.JScrollBar;
 import javax.swing.JScrollPane;
 import javax.swing.ScrollPaneLayout;
 import javax.swing.plaf.ComponentUI;
