--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/CheckedInputStream.java	2004-10-28 02:27:40.000000000 -0600
+++ java/util/zip/CheckedInputStream.java	2002-01-22 16:01:46.000000000 -0700
@@ -1,5 +1,5 @@
 /* CheckedInputStream.java - Compute checksum of data being read
-   Copyright (C) 1999, 2000, 2004  Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -35,12 +35,11 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package java.util.zip;
 
 import java.io.FilterInputStream;
-import java.io.IOException;
 import java.io.InputStream;
+import java.io.IOException;
 
 /* Written using on-line Java Platform 1.2 API Specification
  * and JCL book.
