--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Long.java	2004-04-13 02:20:22.000000000 -0600
+++ java/lang/Long.java	2002-06-13 11:39:51.000000000 -0600
@@ -225,7 +225,7 @@
    *
    * @param s the <code>String</code> to convert
    * @param radix the radix (base) to use in the conversion
-   * @return the <code>String</code> argument converted to <code>long</code>
+   * @return the <code>String</code> argument converted to </code>long</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as a
    *         <code>long</code>
    */
