--- /home/tromey/gnu/egcs/classpath/classpath/java/util/AbstractSet.java	Tue Jan 22 15:50:06 2002
+++ java/util/AbstractSet.java	Tue Jan 22 16:01:39 2002
@@ -126,7 +126,7 @@
     Iterator i;
     if (oldsize < count)
       {
-        for (i = iterator(), count = oldsize; count > 0; count--)
+	for (i = iterator(), count = oldsize; count > 0; count--)
           if (c.contains(i.next()))
             i.remove();
       }
