--- /home/tromey/gnu/classpath/classpath/java/nio/channels/ServerSocketChannel.java	Tue Apr 30 17:58:37 2002
+++ java/nio/channels/ServerSocketChannel.java	Fri Nov  1 12:46:58 2002
@@ -37,29 +37,14 @@
 
 package java.nio.channels;
 
-import java.nio.channels.spi.*;
-import java.net.*;
-import java.nio.*;
-import java.io.*;
+import java.nio.channels.spi.AbstractSelectableChannel;
+import java.nio.channels.spi.SelectorProvider;
 
 public abstract class ServerSocketChannel
-    extends AbstractSelectableChannel
+  extends AbstractSelectableChannel
 {
-    protected ServerSocketChannel(SelectorProvider provider)
-    {
-    }
- 
-
-    public abstract  SocketChannel accept();
-    public abstract  ServerSocket socket();
-    
-    public static ServerSocketChannel open() throws IOException
-    {
-	return SelectorProvider.provider().openServerSocketChannel();
-    }
-
-    public int validOps()
-    {
-	return SelectionKey.OP_ACCEPT;
-    } 
+  public ServerSocketChannel (SelectorProvider provider)
+  {
+    super (provider);
+  }
 }
