--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/KeyStore.java	2003-04-24 02:22:11.000000000 -0600
+++ java/security/KeyStore.java	2003-04-20 02:20:19.000000000 -0600
@@ -43,8 +43,6 @@
 import java.util.Date;
 import java.util.Enumeration;
 
-import gnu.java.security.Engine;
-
 /**
  * Keystore represents an in-memory collection of keys and 
  * certificates. There are two types of entries:
@@ -196,10 +194,6 @@
       {
         throw new KeyStoreException(type);
       }
-    catch (java.lang.reflect.InvocationTargetException ite)
-      {
-	throw new KeyStoreException(type);
-      }
     catch (ClassCastException cce)
       {
         throw new KeyStoreException(type);
