--- /home/tromey/gnu/classpath/classpath/java/awt/dnd/DragGestureRecognizer.java	Mon Dec 30 15:39:04 2002
+++ java/awt/dnd/DragGestureRecognizer.java	Thu Aug  8 22:26:15 2002
@@ -130,12 +130,6 @@
     throw new Error("not implemented");
   }
 
-  /**
-   * Register a new DragGestureListener.
-   *
-   * @exception TooManyListenersException If a DragGestureListener has already
-   * been added.
-   */
   public void addDragGestureListener(DragGestureListener dgl)
     throws TooManyListenersException
   {
