--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/StackTraceElement.java	2004-09-21 02:40:01.000000000 -0600
+++ java/lang/StackTraceElement.java	2002-07-12 09:55:53.000000000 -0600
@@ -77,7 +77,7 @@
    *
    * @serial the enclosing class, if known
    */
-  private final String declaringClass;
+  private final String className;
 
   /**
    * The method name in the class, null if unknown.
@@ -106,7 +106,7 @@
   {
     this.fileName = fileName;
     this.lineNumber = lineNumber;
-    this.declaringClass = className;
+    this.className = className;
     this.methodName = methodName;
     this.isNative = isNative;
   }
@@ -142,7 +142,7 @@
    */
   public String getClassName()
   {
-    return declaringClass;
+    return className;
   }
 
   /**
@@ -183,9 +183,9 @@
   public String toString()
   {
     StringBuffer sb = new StringBuffer();
-    if (declaringClass != null)
+    if (className != null)
       {
-        sb.append(declaringClass);
+        sb.append(className);
         if (methodName != null)
           sb.append('.');
       }
@@ -217,7 +217,7 @@
     StackTraceElement e = (StackTraceElement) o;
     return equals(fileName, e.fileName)
       && lineNumber == e.lineNumber
-      && equals(declaringClass, e.declaringClass)
+      && equals(className, e.className)
       && equals(methodName, e.methodName);
   }
 
@@ -230,7 +230,7 @@
    */
   public int hashCode()
   {
-    return hashCode(fileName) ^ lineNumber ^ hashCode(declaringClass)
+    return hashCode(fileName) ^ lineNumber ^ hashCode(className)
       ^ hashCode(methodName);
   }
 
