--- /home/tromey/gnu/classpath/classpath/java/nio/charset/Charset.java	Sun Nov 24 16:32:34 2002
+++ java/nio/charset/Charset.java	Sat Feb 15 18:59:11 2003
@@ -187,9 +187,10 @@
     return canonicalName;
   }
 
-  public final boolean isRegistered (String name)
+  public final boolean isRegistered ()
   {
-    return !name.startsWith ("x-") && !name.startsWith ("X-");
+    return (!canonicalName.startsWith ("x-")
+            && !canonicalName.startsWith ("X-"));
   }
 
   public abstract boolean contains (Charset cs);
