--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/MarshalledObject.java	2004-04-03 02:21:02.000000000 -0700
+++ java/rmi/MarshalledObject.java	2003-11-12 02:18:14.000000000 -0700
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 1996, 1997, 1998, 1999, 2004 Free Software Foundation, Inc.
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -46,7 +46,8 @@
 /**
  * FIXME - doc missing
  */
-public final class MarshalledObject implements Serializable 
+public final class MarshalledObject
+  extends Object implements Serializable 
 {
   
   //The following fields are from Java API Documentation "Serialized form"
