--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/PortUnreachableException.java	2004-04-13 02:20:23.000000000 -0600
+++ java/net/PortUnreachableException.java	2002-08-21 12:13:02.000000000 -0600
@@ -37,7 +37,6 @@
 
 package java.net;
 
-
 /**
  * This exception signals that an ICMP port unreachable datagram has been
  * received.
@@ -70,3 +69,4 @@
     super(message);
   }
 } // class PortUnreachableException
+
