--- /home/tromey/gnu/classpath/classpath/java/util/TreeMap.java	Mon Dec 30 15:39:05 2002
+++ java/util/TreeMap.java	Mon Jun 17 22:56:21 2002
@@ -865,7 +865,7 @@
     int rowsize;
 
     // Fill each row that is completely full of nodes.
-    for (rowsize = 2; rowsize + rowsize <= count; rowsize <<= 1)
+    for (rowsize = 2; rowsize + rowsize < count; rowsize <<= 1)
       {
         Node parent = row;
         Node last = null;
@@ -1468,10 +1468,10 @@
      */
     public void remove()
     {
-      if (last == null)
-        throw new IllegalStateException();
       if (knownMod != modCount)
         throw new ConcurrentModificationException();
+      if (last == null)
+        throw new IllegalStateException();
 
       removeNode(last);
       last = null;
