--- /home/tromey/gnu/classpath/classpath/java/awt/dnd/DropTarget.java	Mon Dec 30 15:39:04 2002
+++ java/awt/dnd/DropTarget.java	Tue Dec 10 19:20:25 2002
@@ -42,13 +42,10 @@
 import java.awt.datatransfer.FlavorMap;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.io.Serializable;
-import java.util.EventListener;
 import java.util.TooManyListenersException;
 
 /** STUB CLASS ONLY */
 public class DropTarget
-  implements DropTargetListener, EventListener, Serializable
 {
   protected static class DropTargetAutoScroller
     implements ActionListener
@@ -70,9 +67,6 @@
     }
   }
   
-  // FIXME: check the correctness of default value.
-  private boolean isActive = false;
-    
   /**
    * FIXME
    *
@@ -139,12 +133,11 @@
 
   public void setActive(boolean isActive)
   {
-    this.isActive = isActive;
   }
 
   public boolean isActive()
   {
-    return this.isActive;
+    return false;
   }
 
   /**
