--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URLStreamHandler.java	2004-10-16 02:22:18.000000000 -0600
+++ java/net/URLStreamHandler.java	2004-07-12 02:19:00.000000000 -0600
@@ -180,8 +180,8 @@
 	      }
 	    catch (NumberFormatException e)
 	      {
-		// Ignore invalid port values; port is already set to u's
-		// port.
+		; // Ignore invalid port values; port is already set to u's
+		  // port.
 	      }
 
 	    // Now we must cut the port number in the original string.
