--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicProgressBarUI.java	2004-06-08 02:25:42.000000000 -0600
+++ javax/swing/plaf/basic/BasicProgressBarUI.java	2004-06-18 02:18:58.000000000 -0600
@@ -134,8 +134,7 @@
   }
 
   /** The timer used to move the bouncing box. */
-  private transient Timer animationTimer = new Timer();
-
+  private transient Timer animationTimer;
 
   // The total number of frames must be an even number.
   // The total number of frames is calculated from
@@ -681,6 +680,8 @@
    * This method starts the animation timer. It is called
    * when the propertyChangeListener detects that the progressBar
    * has changed to indeterminate mode.
+   *
+   * @since 1.4
    */
   protected void startAnimationTimer()
   {
@@ -692,6 +693,8 @@
    * This method stops the animation timer. It is called when
    * the propertyChangeListener detects that the progressBar
    * has changed to determinate mode.
+   *
+   * @since 1.4
    */
   protected void stopAnimationTimer()
   {
@@ -794,7 +797,7 @@
       {
 	progressBar = (JProgressBar) c;
 
-	animationTimer = new Timer();
+	animationTimer = new Timer(200, null);
 	animationTimer.setRepeats(true);
 
 	installDefaults();
