--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/charset/spi/CharsetProvider.java	2004-04-13 02:20:25.000000000 -0600
+++ java/nio/charset/spi/CharsetProvider.java	2002-12-10 19:20:35.000000000 -0700
@@ -40,7 +40,6 @@
 import java.nio.charset.Charset;
 import java.util.Iterator;
 
-
 /**
  * This class allows an implementor to provide additional character sets. The
  * subclass must have a nullary constructor, and be attached to charset
@@ -83,8 +82,6 @@
   /**
    * Returns the named charset, by canonical name or alias.
    *
-   * @param name the name of the character
-   *
    * @return the charset, or null if not supported
    */
   public abstract Charset charsetForName(String name);
