--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/KeyFactory.java	2003-04-24 02:22:11.000000000 -0600
+++ java/security/KeyFactory.java	2003-04-20 02:20:19.000000000 -0600
@@ -41,8 +41,6 @@
 import java.security.spec.InvalidKeySpecException;
 import java.security.NoSuchAlgorithmException;
 
-import gnu.java.security.Engine;
-
 /**
  * <p>Key factories are used to convert keys (opaque cryptographic keys of type
  * {@link Key}) into key specifications (transparent representations of the
@@ -194,10 +192,6 @@
 	  Engine.getInstance(KEY_FACTORY, algorithm, provider),
           provider, algorithm);
       }
-    catch (java.lang.reflect.InvocationTargetException ite)
-      {
-	throw new NoSuchAlgorithmException(algorithm);
-      }
     catch (ClassCastException cce)
       {
 	throw new NoSuchAlgorithmException(algorithm);
