--- /home/tromey/gnu/egcs/classpath/classpath/java/security/UnrecoverableKeyException.java	Wed Mar 22 12:26:24 2000
+++ java/security/UnrecoverableKeyException.java	Wed Apr 25 09:45:12 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,44 +24,35 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 
 /**
-  * This exception is thrown when a key cannot be recovered from the key
-  * store.
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This exception is thrown when a key cannot be recovered from the key
+ * store.
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public class UnrecoverableKeyException extends GeneralSecurityException
 {
-
-/**
-  * This method initializes a new instance of <code>UnrecoverableKeyException</code>
-  * with no descriptive error message.
-  */
-public
-UnrecoverableKeyException()
-{
-  super();
+  /**
+   * This method initializes a new instance of <code>UnrecoverableKeyException</code>
+   * with no descriptive error message.
+   */
+  public UnrecoverableKeyException()
+  {
+    super();
+  }
+
+  /**
+   * This method initializes a new instance of <code>UnrecoverableKeyException</code>
+   * with a descriptive error message.
+   *
+   * @param msg The descriptive error message.
+   */
+  public UnrecoverableKeyException(String msg)
+  {
+    super(msg);
+  }
 }
-
-/*************************************************************************/
-
-/**
-  * This method initializes a new instance of <code>UnrecoverableKeyException</code>
-  * with a descriptive error message.
-  *
-  * @param msg The descriptive error message.
-  */
-public
-UnrecoverableKeyException(String msg)
-{
-  super(msg);
-}
-
-
-} // class UnrecoverableKeyException
-
