--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/UIManager.java	2004-07-09 02:22:38.000000000 -0600
+++ javax/swing/UIManager.java	2004-06-08 02:20:01.000000000 -0600
@@ -1,5 +1,5 @@
 /* UIManager.java -- 
-   Copyright (C) 2002, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -194,7 +194,7 @@
    *
    * @param key an Object that specifies the font. Typically,
    *        this is a String such as
-   *        <code>TitledBorder.font</code>.
+   *        <code>&quot;TitledBorder.font&quot;</code>.
    */
   public static Font getFont(Object key)
   {
@@ -253,7 +253,7 @@
   }
 
   public static ComponentUI getUI(JComponent target)
-    // Returns the Look and Feel object that renders the target component. 
+    // Returns the L&F object that renders the target component. 
   {
     ComponentUI ui = getDefaults().getUI(target);
     //System.out.println("GET-UI-> " + ui + ", for " + target);
