--- /home/tromey/gnu/egcs/classpath/classpath/java/security/PublicKey.java	Sun Mar 31 23:08:12 2002
+++ java/security/PublicKey.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* PublicKey.java -- tagging interface for all public keys
-   Copyright (C) 1998, 2001, 2002 Free Software Foundation, Inc.
+/* PublicKey.java -- "Super-interface" for all public keys
+   Copyright (C) 1998, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -41,20 +41,10 @@
  * This interface specified no methods.  In simply provides a common
  * super-interface for all algorithm specific public key values.
  *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @see Key
- * @see PrivateKey
- * @see Certificate
- * @see Signature#initVerify(PublicKey)
- * @see DSAPublicKey
- * @see RSAPublicKey
- * @since 1.1
- * @status updated to 1.4
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
  */
 public interface PublicKey extends Key
 {
-  /**
-   * The verion identifier used for serialization.
-   */
-  long serialVersionUID = 7187392471159151072L;
-} // interface PublicKey
+}
