--- /home/tromey/gnu/egcs/classpath/classpath/java/security/BasicPermission.java	Sun Feb 10 17:34:21 2002
+++ java/security/BasicPermission.java	Tue Jan 22 16:01:36 2002
@@ -61,9 +61,8 @@
  *
  * @author Aaron M. Renn (arenn@urbanophile.com)
  */
-public abstract class BasicPermission extends java.security.Permission
-  implements Serializable
-  // FIXME extends with fully qualified classname as workaround for gcj 3.0.4
+public abstract class BasicPermission extends Permission implements
+  Serializable
 {
   /**
    * This method initializes a new instance of <code>BasicPermission</code>
