--- /home/tromey/gnu/classpath/classpath/java/awt/Label.java	Mon Dec 30 15:39:04 2002
+++ java/awt/Label.java	Tue Jan 22 16:01:31 2002
@@ -99,8 +99,6 @@
 
 /**
   * Initializes a new instance of <code>Label</code> with no text.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 Label()
@@ -115,8 +113,6 @@
   * text that is aligned to the left.
   *
   * @param text The text of the label.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 Label(String text)
@@ -134,17 +130,12 @@
   * @param alignment The desired alignment for the text in this label,
   * which must be one of <code>LEFT</code>, <code>CENTER</code>, or
   * <code>RIGHT</code>.
-  *
-  * @exception HeadlessException If GraphicsEnvironment.isHeadless() is true.
   */
 public
 Label(String text, int alignment)
 {
   setAlignment (alignment);
   setText (text);
-
-  if (GraphicsEnvironment.isHeadless())
-    throw new HeadlessException ();
 }
 
 /*************************************************************************/
