--- /home/tromey/gnu/Nightly/classpath/classpath/javax/naming/spi/ResolveResult.java	2004-04-20 02:21:37.000000000 -0600
+++ javax/naming/spi/ResolveResult.java	2003-12-06 02:18:23.000000000 -0700
@@ -1,5 +1,5 @@
 /* ResolveResult.java --
-   Copyright (C) 2001, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,10 +39,10 @@
 package javax.naming.spi;
 
 import java.io.Serializable;
-
+import java.util.EventObject;
+import javax.naming.Name;
 import javax.naming.CompositeName;
 import javax.naming.InvalidNameException;
-import javax.naming.Name;
  
 /**
  * @author Warren Levy <warrenl@redhat.com>
