--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicButtonUI.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicButtonUI.java	2004-05-27 02:19:07.000000000 -0600
@@ -92,6 +92,7 @@
     b.setBackground(defaults.getColor("Button.background"));
     b.setMargin(defaults.getInsets("Button.margin"));
     b.setBorder(defaults.getBorder("Button.border"));
+    b.setOpaque(true);
   }
 
   protected void uninstallDefaults(AbstractButton b)
