--- /home/tromey/gnu/classpath/classpath/java/nio/FloatBuffer.java	Mon Dec 30 15:39:05 2002
+++ java/nio/FloatBuffer.java	Mon Nov 18 06:56:58 2002
@@ -39,7 +39,7 @@
 
 import gnu.java.nio.FloatBufferImpl;
 
-public abstract class FloatBuffer extends Buffer implements Comparable
+public abstract class FloatBuffer extends Buffer
 {
   private ByteOrder endian = ByteOrder.BIG_ENDIAN;
   protected float [] backing_buffer;
@@ -77,7 +77,7 @@
     return wrap(array, 0, array.length);
   }
 
-  public FloatBuffer get(float[] dst, int offset, int length)
+  final public FloatBuffer get(float[] dst, int offset, int length)
   {
     for (int i = offset; i < offset + length; i++)
       {
@@ -87,12 +87,12 @@
     return this;
   }
 
-  public FloatBuffer get(float[] dst)
+  final public FloatBuffer get(float[] dst)
   {
     return get(dst, 0, dst.length);
   }
 
-  public FloatBuffer put(FloatBuffer src)
+  final public FloatBuffer put(FloatBuffer src)
   {
     while (src.hasRemaining())
       put(src.get());
@@ -100,7 +100,7 @@
     return this;
   }
 
-  public FloatBuffer put(float[] src, int offset, int length)
+  final public FloatBuffer put(float[] src, int offset, int length)
   {
     for (int i = offset; i < offset + length; i++)
       put(src[i]);
@@ -171,7 +171,7 @@
     return 0;
   }
 
-  public ByteOrder order()
+  public final ByteOrder order()
   {
     return endian;
   }
@@ -191,4 +191,34 @@
   public abstract FloatBuffer slice();
   public abstract FloatBuffer duplicate();
   public abstract FloatBuffer asReadOnlyBuffer();
+  public abstract ShortBuffer asShortBuffer();
+  public abstract CharBuffer asCharBuffer();
+  public abstract IntBuffer asIntBuffer();
+  public abstract LongBuffer asLongBuffer();
+  public abstract FloatBuffer asFloatBuffer();
+  public abstract DoubleBuffer asDoubleBuffer();
+  public abstract char getChar();
+  public abstract FloatBuffer putChar(char value);
+  public abstract char getChar(int index);
+  public abstract FloatBuffer putChar(int index, char value);
+  public abstract short getShort();
+  public abstract FloatBuffer putShort(short value);
+  public abstract short getShort(int index);
+  public abstract FloatBuffer putShort(int index, short value);
+  public abstract int getInt();
+  public abstract FloatBuffer putInt(int value);
+  public abstract int getInt(int index);
+  public abstract FloatBuffer putInt(int index, int value);
+  public abstract long getLong();
+  public abstract FloatBuffer putLong(long value);
+  public abstract long getLong(int index);
+  public abstract FloatBuffer putLong(int index, long value);
+  public abstract float getFloat();
+  public abstract FloatBuffer putFloat(float value);
+  public abstract float getFloat(int index);
+  public abstract FloatBuffer putFloat(int index, float value);
+  public abstract double getDouble();
+  public abstract FloatBuffer putDouble(double value);
+  public abstract double getDouble(int index);
+  public abstract FloatBuffer putDouble(int index, double value);
 }
