--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/Channels.java	2004-11-17 02:24:08.000000000 -0700
+++ java/nio/channels/Channels.java	2004-10-17 02:15:36.000000000 -0600
@@ -60,11 +60,6 @@
 public final class Channels
 {
   /**
-   * This class isn't intended to be instantiated.
-   */
-  private Channels() {}
-
-  /**
    * Constructs a stream that reads bytes from the given channel.
    */
   public static InputStream newInputStream(ReadableByteChannel ch)
