--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Component.java	2003-08-23 02:18:44.000000000 -0600
+++ java/awt/Component.java	2003-08-26 02:18:05.000000000 -0600
@@ -1866,12 +1866,17 @@
    * @param height the height of the image
    * @return the requested image, or null if it is not supported
    */
-  public Image createImage(int width, int height)
+  public Image createImage (int width, int height)
   {
-    if (GraphicsEnvironment.isHeadless())
-      return null;
-    GraphicsConfiguration config = getGraphicsConfiguration();
-    return config == null ? null : config.createCompatibleImage(width, height);
+    Image returnValue = null;
+    if (!GraphicsEnvironment.isHeadless ())
+    {
+      if (isLightweight () && parent != null)
+        returnValue = parent.createImage (width, height);
+      else if (peer != null)
+        returnValue = peer.createImage (width, height);
+    }
+    return returnValue;
   }
 
   /**
