--- /home/tromey/gnu/classpath/classpath/java/nio/charset/UnmappableCharacterException.java	Fri Nov  8 05:15:32 2002
+++ java/nio/charset/UnmappableCharacterException.java	Mon Oct  7 01:59:12 2002
@@ -37,10 +37,7 @@
 
 package java.nio.charset;
 
-/**
- * @since 1.4
- */
-public class UnmappableCharacterException extends CharacterCodingException
+class UnmappableCharacterException extends CharacterCodingException
 {
   private int inputLength;
   
@@ -66,6 +63,7 @@
    */
   public String getMessage ()
   {
-    return "Input length = " + inputLength;
+    // FIXME: I think this is not correct
+    return "";
   }
 }
