--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/InputStream.java	2004-04-18 02:22:15.000000000 -0600
+++ java/io/InputStream.java	2002-01-22 16:01:33.000000000 -0700
@@ -105,7 +105,7 @@
    * @param readLimit The number of bytes that can be read before the
    *                  mark becomes invalid
    */
-  public void mark(int readLimit)
+  public void mark(int readlimit)
   {
     // Do nothing
   }
@@ -117,7 +117,7 @@
    * point.
    * <p>
    * This method always returns <code>false</code> in this class, but
-   * subclasses can override this method to return <code>true</code>
+   * subclasses can override this method to return </code>true</code>
    * if they support mark/reset functionality.
    *
    * @return <code>true</code> if mark/reset functionality is
