--- /home/tromey/gnu/egcs/classpath/classpath/java/net/MulticastSocket.java	Mon Oct  1 14:44:13 2001
+++ java/net/MulticastSocket.java	Mon Oct  1 14:45:05 2001
@@ -64,7 +64,7 @@
   */
   public MulticastSocket() throws IOException
   {
-    super(0, null);
+    super(0, ServerSocket.ANY_IF);
   }
 
 /**
@@ -76,7 +76,7 @@
   */
   public MulticastSocket(int port) throws IOException
   {
-    super(port, null);
+    super(port, ServerSocket.ANY_IF);
   }
 
 /**
