--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/cert/PKIXCertPathBuilderResult.java	2003-05-01 02:23:13.000000000 -0600
+++ java/security/cert/PKIXCertPathBuilderResult.java	2003-04-30 01:23:42.000000000 -0600
@@ -47,7 +47,6 @@
  * @see CertPathBuilderResult
  */
 public class PKIXCertPathBuilderResult extends PKIXCertPathValidatorResult
-  implements CertPathBuilderResult
 {
 
   // Fields.
