--- /home/tromey/gnu/egcs/classpath/classpath/java/security/acl/LastOwnerException.java	Wed Mar 22 12:26:24 2000
+++ java/security/acl/LastOwnerException.java	Wed Apr 25 09:45:13 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,29 +24,24 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security.acl;
 
 /**
-  * This exception is thrown when an attempt is made to delete the last owner
-  * of an access control list (ACL)
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This exception is thrown when an attempt is made to delete the last owner
+ * of an access control list (ACL)
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public class LastOwnerException extends java.lang.Exception
 {
-
-/**
-  * Initialize a new instance of <code>LastOwnerException</code> that does
-  * not have a log message.
-  */
-public
-LastOwnerException()
-{
-  super();
+  /**
+   * Initialize a new instance of <code>LastOwnerException</code> that does
+   * not have a log message.
+   */
+  public LastOwnerException()
+  {
+    super();
+  }
 }
-
-} // class LastOwnerException 
-
