--- /home/tromey/gnu/classpath/classpath/javax/naming/RefAddr.java	Tue Jan 22 15:50:12 2002
+++ javax/naming/RefAddr.java	Wed Sep 25 10:10:50 2002
@@ -61,7 +61,7 @@
    * The string resprenstation of the type of address.
    * Set by the constructor and returned by the getType() method.
    */
-  protected final String addrType;
+  protected String addrType;
 
   /**
    * Protected constructor for use by subclasses.
@@ -98,7 +98,7 @@
    *         is the same as this addrType and the content is equals to the
    *         content of this object.
    */
-  public boolean equal(Object o)
+  public boolean equals(Object o)
   {
     if (o instanceof RefAddr)
       {
