--- /home/tromey/gnu/egcs/classpath/classpath/java/util/zip/ZipException.java	Sat Mar  2 12:13:23 2002
+++ java/util/zip/ZipException.java	Tue Jan 22 16:01:46 2002
@@ -1,5 +1,5 @@
-/* ZipException.java - exception representing a zip related error
-   Copyright (C) 1998, 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
+/* ZipException.java - Exception representing a zip related error
+   Copyright (C) 1998, 1999, 2000, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,34 +37,24 @@
 
 package java.util.zip;
 
-import java.io.IOException;
+/* Written using on-line Java Platform 1.2 API Specification.
+ * Believed complete and correct.
+ */
 
 /**
- * Thrown during the creation or input of a zip file.
+ * Is thrown during the creation or input of a zip file.
  *
  * @author Jochen Hoenicke
  * @author Per Bothner
- * @status updated to 1.4
+ * @since JDK 1.1
  */
-public class ZipException extends IOException
+public class ZipException extends java.io.IOException
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = 8000196834066748623L;
-
-  /**
-   * Create an exception without a message.
-   */
-  public ZipException()
+  public ZipException ()
   {
+    super();
   }
 
-  /**
-   * Create an exception with a message.
-   *
-   * @param msg the message
-   */
   public ZipException (String msg)
   {
     super(msg);
