--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocket.java	2004-01-03 02:20:33.000000000 -0700
+++ java/net/DatagramSocket.java	2004-02-04 02:18:37.000000000 -0700
@@ -39,6 +39,7 @@
 package java.net;
 
 import gnu.java.net.PlainDatagramSocketImpl;
+import gnu.java.nio.DatagramChannelImpl;
 import java.io.IOException;
 import java.nio.channels.DatagramChannel;
 import java.nio.channels.IllegalBlockingModeException;
@@ -565,7 +566,8 @@
         ("Socket connected to a multicast address my not receive");
 
     if (getChannel() != null
-        && !getChannel().isBlocking ())
+        && !getChannel().isBlocking ()
+        && !((DatagramChannelImpl) getChannel()).isInChannelOperation())
       throw new IllegalBlockingModeException ();
 
     getImpl().receive(p);
@@ -618,7 +620,8 @@
     // use getTimeToLive for TTL val.
 
     if (getChannel() != null
-        && !getChannel().isBlocking ())
+        && !getChannel().isBlocking ()
+        && !((DatagramChannelImpl) getChannel()).isInChannelOperation())
       throw new IllegalBlockingModeException ();
 
     getImpl().send(p);
