--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/URI.java	2004-10-16 02:22:18.000000000 -0600
+++ java/net/URI.java	2004-08-26 02:20:09.000000000 -0600
@@ -1,4 +1,4 @@
-/* URI.java - An URI class --
+/* URI.java - An URI class
    Copyright (C) 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -134,14 +134,14 @@
   {
     this.string = (String) is.readObject();
     try
-      {
-	parseURI(this.string);
-      }
+    {
+      parseURI(this.string);
+    }
     catch (URISyntaxException x)
-      {
-	// Should not happen.
-	throw new RuntimeException(x);
-      }
+    {
+      // Should not happen.
+      throw new RuntimeException(x);
+    }
   }
 
   private void writeObject(ObjectOutputStream os) throws IOException
@@ -618,7 +618,7 @@
    */
   public boolean isAbsolute()
   {
-    return scheme != null;
+    return (scheme != null);
   }
 
   /**
