--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/logging/LogManager.java	2003-08-13 02:19:51.000000000 -0600
+++ java/util/logging/LogManager.java	2003-06-21 04:31:55.000000000 -0600
@@ -52,7 +52,6 @@
 import java.util.Enumeration;
 import java.util.Iterator;
 import java.util.Map;
-import java.util.StringTokenizer;
 import java.lang.ref.WeakReference;
 
 /**
@@ -168,7 +167,6 @@
      * the order in which classes are initialized.
      */
     Logger.getLogger("global").setParent(rootLogger);
-    Logger.getLogger("global").setUseParentHandlers(true);
   }
 
 
@@ -522,7 +520,7 @@
 
   public synchronized void readConfiguration(InputStream inputStream)
     throws IOException, SecurityException
-  {	
+  {
     Properties   newProperties;
     Enumeration  keys;
 
@@ -534,36 +532,12 @@
 
     while (keys.hasMoreElements())
     {
-      String key = ((String) keys.nextElement()).trim();
+      String key = (String) keys.nextElement();
       String value = newProperties.getProperty(key);
-  
+
       if (value == null)
 	continue;
-	
-	  value = value.trim();
-	
-	  if("handlers".equals(key))
-        {
-          StringTokenizer tokenizer = new StringTokenizer(value);
-          while(tokenizer.hasMoreTokens())
-            {
-              String handlerName = tokenizer.nextToken();	
-              try
-                {
-              	  Class handlerClass = Class.forName(handlerName);
-              	  getLogger("").addHandler((Handler)handlerClass.newInstance()); 
-                }
-              catch (ClassCastException ex)
-                {
-                  System.err.println("[LogManager] class " + handlerName + " is not subclass of java.util.logging.Handler");
-                }
-              catch (Exception ex)
-                {
-                  //System.out.println("[LogManager.readConfiguration]"+ex);
-                }
-            }
-        }
-	  
+
       if (key.endsWith(".level"))
       {
 	String loggerName = key.substring(0, key.length() - 6);
@@ -576,7 +550,6 @@
 	  }
 	  catch (Exception _)
 	  {
-        //System.out.println("[LogManager.readConfiguration] "+_);
 	  }
 	  continue;
 	}
