--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Throwable.java	2004-04-18 02:22:19.000000000 -0600
+++ java/lang/Throwable.java	2002-12-10 19:20:27.000000000 -0700
@@ -392,7 +392,7 @@
    * then a line starting with three spaces and the string "... X more" is
    * printed, where X is the number of remaining stackframes.
    *
-   * @param pw the PrintWriter to write the trace to
+   * @param w the PrintWriter to write the trace to
    * @since 1.1
    */
   public void printStackTrace (PrintWriter pw)
