--- /home/tromey/gnu/classpath/classpath/java/awt/datatransfer/DataFlavor.java	Mon Dec 30 15:39:04 2002
+++ java/awt/datatransfer/DataFlavor.java	Tue Dec 10 19:20:24 2002
@@ -281,7 +281,6 @@
   *
   * @exception IllegalArgumentException If the representation class
   * specified cannot be loaded.
-  * @exception ClassNotFoundException If the class is not loaded.
   */
 public
 DataFlavor(String mimeType, String humanPresentableName, 
@@ -505,8 +504,6 @@
   *
   * @return <code>true</code> if the MIME type is equal to this object's
   * MIME type, <code>false</code> otherwise.
-  *
-  * @exception NullPointerException If mimeType is null.
   */
 public final boolean
 isMimeTypeEqual(String mimeType)
@@ -655,10 +652,6 @@
   * Returns a copy of this object.
   *
   * @return A copy of this object.
-  *
-  * @exception CloneNotSupportedException If the object's class does not support
-  * the Cloneable interface. Subclasses that override the clone method can also
-  * throw this exception to indicate that an instance cannot be cloned.
   */
 public Object
 clone()
@@ -820,8 +813,6 @@
   * Serialize this class.
   *
   * @param stream The <code>ObjectOutput</code> stream to serialize to.
-  *
-  * @exception IOException If an error occurs.
   */
 public void
 writeExternal(ObjectOutput stream) throws IOException
@@ -835,10 +826,6 @@
   * De-serialize this class.
   *
   * @param stream The <code>ObjectInput</code> stream to deserialize from.
-  *
-  * @exception IOException If an error ocurs.
-  * @exception ClassNotFoundException If the class for an object being restored
-  * cannot be found.
   */
 public void
 readExternal(ObjectInput stream) throws IOException, ClassNotFoundException
@@ -963,10 +950,6 @@
   *
   * @param transferable The <code>Transferable</code> for which a text
   *                     <code>Reader</code> is requested.
-  *
-  * @exception IllegalArgumentException If the representation class is not one
-  * of the seven listed above or the Transferable has null data.
-  * @exception NullPointerException If the Transferable is null.
   * @exception UnsupportedFlavorException when the transferable doesn't
   * support this <code>DataFlavor</code>. Or if the representable class
   * isn't a (subclass of) <code>Reader</code>, <code>String</code>,
