--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/AssertionError.java	Sat Feb 23 21:25:16 2002
+++ java/lang/AssertionError.java	Sun Mar 31 23:32:04 2002
@@ -76,8 +76,8 @@
   public AssertionError(Object msg)
   {
     super("" + msg);
-    if (msg instanceof Throwable)
-      initCause((Throwable) msg);
+//      if (msg instanceof Throwable)
+//        initCause((Throwable) msg);
   }
 
   /**
