--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/ButtonModel.java	2003-07-02 02:19:16.000000000 -0600
+++ javax/swing/ButtonModel.java	2002-08-08 22:26:10.000000000 -0600
@@ -76,4 +76,12 @@
 
     void setSelected(boolean b);
     boolean isSelected();
+
+
+    // there are not in the spec !!
+
+
+    void fireItemStateChanged(ItemEvent event);
+    void fireStateChanged(ChangeEvent event);    
+    void fireActionPerformed(ActionEvent event);
 }
