--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipInputStream.java	2003-02-12 13:08:36.000000000 -0700
+++ java/util/zip/ZipInputStream.java	2003-06-18 02:22:33.000000000 -0600
@@ -35,11 +35,12 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package java.util.zip;
+
 import java.io.EOFException;
 import java.io.InputStream;
 import java.io.IOException;
-import java.util.Enumeration;
 
 /**
  * This is a FilterInputStream that reads the files in an zip archive
