--- /home/tromey/gnu/egcs/classpath/classpath/java/security/KeyException.java	Wed Mar 22 12:26:19 2000
+++ java/security/KeyException.java	Wed Apr 25 11:13:33 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,42 +24,34 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 
 /**
-  * This exception is thrown when there is a problem with a key.
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This exception is thrown when there is a problem with a key.
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public class KeyException extends GeneralSecurityException
 {
-
-/**
-  * This method initializes a new instance of <code>KeyException</code>
-  * with no descriptive message.
-  */
-public
-KeyException()
-{
-  super();
-}
-
-/*************************************************************************/
-
-/**
-  * This method initializes a new instance of <code>KeyException</code>
-  * with a descriptive message.
-  *
-  * @param msg The descriptive message.
-  */
-public
-KeyException(String msg)
-{
-  super(msg);
+  /**
+   * This method initializes a new instance of <code>KeyException</code>
+   * with no descriptive message.
+   */
+  public KeyException()
+  {
+    super();
+  }
+
+  /**
+   * This method initializes a new instance of <code>KeyException</code>
+   * with a descriptive message.
+   *
+   * @param msg The descriptive message.
+   */
+  public KeyException(String msg)
+  {
+    super(msg);
+  }
 }
-
-} // class KeyException
-
