--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JSlider.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JSlider.java	2004-06-18 02:18:54.000000000 -0600
@@ -122,14 +122,15 @@
   protected class AccessibleJSlider extends JComponent.AccessibleJComponent
     implements AccessibleValue
   {
+    private static final long serialVersionUID = -6301740148041106789L;
+  
     /**
      * Creates a new AccessibleJSlider object.
      *
      * @param value0 DOCUMENT ME!
      */
-    protected AccessibleJSlider(JSlider value0)
+    protected AccessibleJSlider()
     {
-      super(value0);
     }
 
     /**
@@ -933,7 +934,8 @@
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      accessibleContext = new AccessibleJSlider(this);
+      accessibleContext = new AccessibleJSlider();
+    
     return accessibleContext;
   }
 }
