--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectStreamField.java	2004-05-18 02:42:48.000000000 -0600
+++ java/io/ObjectStreamField.java	2004-09-26 02:16:22.000000000 -0600
@@ -367,46 +367,109 @@
 
   final void setBooleanField(Object obj, boolean val)
   {
-    VMObjectStreamClass.setBooleanNative(field, obj, val);  
+    try
+      {
+	field.setBoolean(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
 
   final void setByteField(Object obj, byte val)
   {
-    VMObjectStreamClass.setByteNative(field, obj, val);
+    try
+      {
+	field.setByte(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
   
   final void setCharField(Object obj, char val)
   {
-    VMObjectStreamClass.setCharNative(field, obj, val);
+    try
+      {
+	field.setChar(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
   
   final void setShortField(Object obj, short val)
   {
-    VMObjectStreamClass.setShortNative(field, obj, val);
+    try
+      {
+	field.setShort(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
 
   final void setIntField(Object obj, int val)
   {
-    VMObjectStreamClass.setIntNative(field, obj, val);
+    try
+      {
+	field.setInt(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
   
   final void setLongField(Object obj, long val)
   {
-    VMObjectStreamClass.setLongNative(field, obj, val);
+    try
+      {
+	field.setLong(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
   
   final void setFloatField(Object obj, float val)
   {
-    VMObjectStreamClass.setFloatNative(field, obj, val);
+    try
+      {
+	field.setFloat(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
   
   final void setDoubleField(Object obj, double val)
   {
-    VMObjectStreamClass.setDoubleNative(field, obj, val);
+    try
+      {
+	field.setDouble(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
   
   final void setObjectField(Object obj, Object val)
   { 
-    VMObjectStreamClass.setObjectNative(field, obj, val);
+    try
+      {
+	field.set(obj, val);
+      }
+    catch(IllegalAccessException x)
+      {
+	throw new InternalError(x.getMessage());
+      }
   }
 }
