--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormat.java	2004-07-01 02:24:48.000000000 -0600
+++ java/text/DateFormat.java	2004-06-15 02:19:32.000000000 -0600
@@ -1,6 +1,5 @@
 /* DateFormat.java -- Class for formatting/parsing date/times
-   Copyright (C) 1998, 1999, 2000, 2001, 2003, 2004
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -326,7 +325,7 @@
     try
       {
 	res = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation",
-				       loc, ClassLoader.getSystemClassLoader());
+				       loc, DateFormat.class.getClassLoader());
       }
     catch (MissingResourceException x)
       {
@@ -592,7 +591,6 @@
   /**
    * This method parses the specified date/time string.
    *
-   * @param source The string to parse.
    * @return The resultant date.
    *
    * @exception ParseException If the specified string cannot be parsed.
@@ -606,8 +604,7 @@
 	int index = pos.getErrorIndex();
 	if (index < 0)
 	  index = pos.getIndex();
-	throw new ParseException("invalid Date syntax in \""
-				 + source + '\"', index);
+	throw new ParseException("invalid Date syntax", index);
       }
     return result;
   }
