--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/LinkedList.java	2002-05-06 23:13:05.000000000 -0600
+++ java/util/LinkedList.java	2003-10-02 02:18:38.000000000 -0600
@@ -1,4 +1,4 @@
-/* LinkedList.java -- Linked list implementation of the List interface
+ /* LinkedList.java -- Linked list implementation of the List interface
    Copyright (C) 1998, 1999, 2000, 2001, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
