--- /home/tromey/gnu/classpath/classpath/java/net/DatagramPacket.java	Wed Oct  2 11:21:38 2002
+++ java/net/DatagramPacket.java	Wed Sep 25 10:10:47 2002
@@ -402,9 +402,9 @@
   /**
    * Sets the length of the data in the buffer. 
    *
-   * @param length The new length.  (Where len &lt;= buf.length)
+   * @param length The new length.  (Where len <= buf.length)
    *
-   * @exception IllegalArgumentException If the length is negative or
+   * @exception IllegalArgumentException f the length is negative or
    * if the length is greater than the packet's data buffer length
    *
    * @since 1.1
