--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/KeyStroke.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/KeyStroke.java	2004-06-18 02:18:55.000000000 -0600
@@ -149,6 +149,7 @@
 	 * @param keyChar TODO
 	 * @param onKeyRelease TODO
 	 * @returns KeyStroke
+	 * @deprecated Use getKeyStroke(char)
 	 */
 	public static KeyStroke getKeyStroke(char keyChar, boolean onKeyRelease) {
 		KeyStroke key = getKeyStroke(keyChar);
