--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/ClosedChannelException.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/ClosedChannelException.java	2002-12-10 19:20:33.000000000 -0700
@@ -1,4 +1,4 @@
-/* ClosedChannelException.java --
+/* ClosedChannelException.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -39,7 +39,6 @@
 
 import java.io.IOException;
 
-
 /**
  * @author Michael Koch
  * @since 1.4
