--- /home/tromey/gnu/classpath/classpath/java/lang/AssertionError.java	Sun Jul 14 23:54:26 2002
+++ java/lang/AssertionError.java	Mon Jun 10 22:15:47 2002
@@ -42,7 +42,7 @@
  * An assertion error normally occurs as a result of the <code>assert</code>
  * statement added in JDK 1.4, to indicate that an assertion failed. There
  * are enough constructors to ensure that
- * <code>new AssertionError(<em>expression</em>)</code> will work for all
+ * <code>new AssertionError(<em>expression</em)</code> will work for all
  * espressions, regardless of type, as if the error message were given by
  * the string <code>"" + <em>expression</em></code>. This extends Error,
  * because you usually do not want to inadvertently trap an assertion failure.
