--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/border/AbstractBorder.java	2003-06-16 02:24:07.000000000 -0600
+++ javax/swing/border/AbstractBorder.java	2003-06-13 02:27:43.000000000 -0600
@@ -91,16 +91,15 @@
 
 
   /**
-   * Measures the width of this border.
+   * Determines the insets of this border. The implementation provided
+   * by AbstractButton returns Insets for a zero-width border, whose
+   * <code>left</code>, <code>right</code>, <code>top</code> and
+   * <code>bottom</code> fields are all zero.
    *
    * @param c the component whose border is to be measured.
    *
-   * @return an Insets object whose <code>left</code>, <code>right</code>,
-   *         <code>top</code> and <code>bottom</code> fields indicate the
-   *         width of the border at the respective edge, which is zero
-   *         for the default implementation provided by AbstractButton.
-   *
-   * @see #getBorderInsets(java.awt.Component, java.awt.Insets)
+   * @return a newly created Insets object, indicating a zero-width
+   *         border.
    */
   public Insets getBorderInsets (Component c)
   {
@@ -114,11 +113,12 @@
    * <code>top</code> and <code>bottom</code> fields of the passed
    * <code>insets</code> parameter to zero.
    *
-   * @param c the component whose border is to be measured.
+   * @param c the component in the center of this border.
    *
-   * @return the same object that was passed for <code>insets</code>.
+   * @param insets an Insets object for holding the insets of this
+   *        border.
    *
-   * @see #getBorderInsets()
+   * @return the <code>insets</code> object.
    */
   public Insets getBorderInsets (Component c, Insets insets) 
   {
