--- /home/tromey/gnu/egcs/classpath/classpath/java/io/SyncFailedException.java	Sat Mar  2 12:13:10 2002
+++ java/io/SyncFailedException.java	Tue Jan 22 16:01:33 2002
@@ -1,5 +1,5 @@
-/* SyncFailedException.java -- a file sync failed
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* SyncFailedException.java -- The sync failed (?)
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,29 +38,49 @@
 
 package java.io;
 
-/**
- * Thrown when a file synchronization fails.
- *
- * @author Aaron M. Renn (arenn@urbanophile.com)
- * @author Tom Tromey <tromey@cygnus.com>
- * @see FileDescriptor#sync()
- * @since 1.1
- * @status updated to 1.4
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * Status:  Complete to 1.1.
  */
+
+/**
+  * I really wish I knew what caused this exception to be thrown.
+  *
+  * @version 0.0
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @author Tom Tromey <tromey@cygnus.com>
+  * @date September 24, 1998
+  */
 public class SyncFailedException extends IOException
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -2353342684412443330L;
-
-  /**
-   * Create an exception with a descriptive error message.
-   *
-   * @param message the descriptive error message
-   */
-  public SyncFailedException(String message)
-  {
-    super(message);
-  }
+
+private static final long serialVersionUID = -2353342684412443330L;
+
+/*
+ * Constructors
+ */
+
+/**
+  * Create a new SyncFailedException without a descriptive error message
+  */
+SyncFailedException()
+{
+  super();
+}
+
+/*************************************************************************/
+
+/**
+  * Create a new SyncFailedException with a descriptive error message String
+  *
+  * @param message The descriptive error message
+  */
+public
+SyncFailedException(String message)
+{
+  super(message);
+}
+
 } // class SyncFailedException
+
