--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/activation/ActivationException.java	2004-04-19 02:20:14.000000000 -0600
+++ java/rmi/activation/ActivationException.java	2002-06-17 13:21:57.000000000 -0600
@@ -94,7 +94,7 @@
    * This method returns a message indicating what went wrong, in this
    * format:
    * <code>super.getMessage() + (detail == null ? ""
-   *    : "; nested exception is:\n\t" + detail)</code>.
+   *    : "; nested exception is:\n\t" + detail)<code>.
    *
    * @return the chained message
    */
