--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/RemoteException.java	2004-04-19 02:20:14.000000000 -0600
+++ java/rmi/RemoteException.java	2002-06-16 15:01:45.000000000 -0600
@@ -99,7 +99,7 @@
    * This method returns a message indicating what went wrong, in this
    * format:
    * <code>super.getMessage() + (detail == null ? ""
-   *    : "; nested exception is:\n\t" + detail)</code>.
+   *    : "; nested exception is:\n\t" + detail)<code>.
    *
    * @return the chained message
    */
