--- /home/tromey/gnu/egcs/classpath/classpath/java/io/WriteAbortedException.java	Sun Jul 23 11:57:08 2000
+++ java/io/WriteAbortedException.java	Fri Nov  2 21:44:51 2001
@@ -1,4 +1,4 @@
-/* WriteAbortedException.java -- An exception occured while writing a 
+/* WriteAbortedException.java -- An exception occurred while writing a 
    serialization stream
    Copyright (C) 1998, 2000 Free Software Foundation, Inc.
 
