--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipOutputStream.java	2004-05-21 02:26:42.000000000 -0600
+++ java/util/zip/ZipOutputStream.java	2003-06-18 02:22:33.000000000 -0600
@@ -338,10 +338,10 @@
     int numEntries = 0;
     int sizeEntries = 0;
     
-    Enumeration e = entries.elements();
-    while (e.hasMoreElements())
+    Enumeration enum = entries.elements();
+    while (enum.hasMoreElements())
       {
-	ZipEntry entry = (ZipEntry) e.nextElement();
+	ZipEntry entry = (ZipEntry) enum.nextElement();
 	
 	int method = entry.getMethod();
 	writeLeInt(CENSIG);
