--- /home/tromey/gnu/classpath/classpath/java/rmi/MarshalledObject.java	Thu Oct 31 12:43:13 2002
+++ java/rmi/MarshalledObject.java	Mon Sep 30 16:06:33 2002
@@ -78,10 +78,6 @@
   {
     if(obj == null || !(obj instanceof MarshalledObject) )
       return false;
-
-    // hashCode even differs, don't do the time-consuming comparisons
-    if (obj.hashCode() != hash)
-      return false;
     
     MarshalledObject aobj = (MarshalledObject)obj;
     if (objBytes == null || aobj.objBytes == null)
