--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Timer.java	2003-06-28 02:19:21.000000000 -0600
+++ java/util/Timer.java	2002-03-05 15:42:21.000000000 -0700
@@ -601,7 +601,7 @@
    * Tells the scheduler that the Timer task died
    * so there will be no more new tasks scheduled.
    */
-  protected void finalize() throws Throwable
+  protected void finalize()
   {
     queue.setNullOnEmpty(true);
   }
