--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/ServerSocket.java	2003-12-03 02:20:30.000000000 -0700
+++ java/net/ServerSocket.java	2003-12-26 02:18:07.000000000 -0700
@@ -221,14 +221,20 @@
       throw new IllegalArgumentException ("Address type not supported");
 
     InetSocketAddress tmp = (InetSocketAddress) endpoint;
-    
+
     SecurityManager s = System.getSecurityManager ();
     if (s != null)
       s.checkListen (tmp.getPort ());
 
+    InetAddress addr = tmp.getAddress();
+    
+    // Initialize addr with 0.0.0.0.
+    if (addr == null)
+      addr = InetAddress.ANY_IF;
+    
     try
       {
-	impl.bind (tmp.getAddress (), tmp.getPort ());
+	impl.bind(addr, tmp.getPort());
 	impl.listen(backlog);
 	bound = true;
       }
@@ -353,15 +359,15 @@
    */
   public void close () throws IOException
   {
-    if (!isClosed())
-      {
-	impl.close();
-	impl = null;
-	bound = false;
+    if (isClosed())
+      return;
+    
+    impl.close();
+    impl = null;
+    bound = false;
 
-	if (getChannel() != null)
-	  getChannel().close();
-      }
+    if (getChannel() != null)
+      getChannel().close();
   }
 
   /**
