--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/RandomAccessFile.java	Tue Apr  1 02:19:07 2003
+++ java/io/RandomAccessFile.java	Tue Apr  1 02:18:00 2003
@@ -62,18 +62,13 @@
 
   // The underlying file.
   private FileDescriptor fd;
+  // The corresponding input and output streams.
+  private DataOutputStream out;
+  private DataInputStream in;
   
   private FileChannel ch; /* cached associated file-channel */
   
   /**
-   * Whether or not this file is open in read only mode
-   */
-  private boolean readOnly;
-  
-  // Used for DataOutput methods writing values to the underlying file
-  private byte[] buf = new byte[8];
-  
-  /**
    * This method initializes a new instance of <code>RandomAccessFile</code>
    * to read from the specified <code>File</code> object with the specified 
    * access mode.   The access mode is either "r" for read only access or "rw" 
@@ -125,28 +120,27 @@
         !mode.equals("rwd"))
       throw new IllegalArgumentException("Bad mode value: " + mode);
   
+    int fdmode;
+    if (mode.compareTo ("r") == 0)
+      fdmode = FileDescriptor.READ;
+    else if (mode.compareTo ("rw") == 0)
+      fdmode = FileDescriptor.READ | FileDescriptor.WRITE;
+    else
+      throw new IllegalArgumentException ("invalid mode: " + mode);
+
     // The obligatory SecurityManager stuff
     SecurityManager s = System.getSecurityManager();
     if (s != null)
       {
         s.checkRead(fileName);
 
-        if (!mode.equals("r"))
+        if ((fdmode & FileDescriptor.WRITE) != 0)
           s.checkWrite(fileName);
       }
-  
-    if (mode.equals("r"))
-      readOnly = true;
-  
-    fd = new FileDescriptor();
-    try
-      {
-        fd.open(fileName, mode);
-      }
-    catch(IOException e)
-      {
-        throw new FileNotFoundException(e.getMessage()); 
-      }
+
+    fd = new FileDescriptor (fileName, fdmode);
+    out = new DataOutputStream (new FileOutputStream (fd));
+    in = new DataInputStream (new FileInputStream (fd));
   }
 
   /**
@@ -203,12 +197,9 @@
    *
    * @exception IOException If an error occurs
    */
-  public void setLength(long newlen) throws IOException
+  public void setLength (long pos) throws IOException
   {
-    if (readOnly)
-      throw new IOException("File is open read only");
-  
-    fd.setLength(newlen);
+    fd.setLength(pos);
   }
 
   /**
@@ -220,7 +211,7 @@
    */
   public long length () throws IOException
   {
-    return fd.getLength();
+    return fd.length();
   }
 
   /**
@@ -233,7 +224,7 @@
    */
   public int read () throws IOException
   {
-    return fd.read();
+    return in.read();
   }
 
   /**
@@ -249,7 +240,7 @@
    */
   public int read (byte[] buffer) throws IOException
   {
-    return read (buffer, 0, buffer.length);
+    return in.read (buffer);
   }
 
   /**
@@ -266,7 +257,7 @@
    */
   public int read (byte[] buffer, int offset, int len) throws IOException
   {
-    return fd.read (buffer, offset, len);
+    return in.read (buffer, offset, len);
   }
 
   /**
@@ -288,12 +279,7 @@
    */
   public final boolean readBoolean () throws IOException
   {
-    int byte_read = read();
-  
-    if (byte_read == -1)
-      throw new EOFException("Unexpected end of stream");
-  
-    return byte_read != 0;
+    return in.readBoolean ();
   }
 
   /**
@@ -313,12 +299,7 @@
    */
   public final byte readByte () throws IOException
   {
-    int byte_read = read ();
-  
-    if (byte_read == -1)
-      throw new EOFException ("Unexpected end of stream");
-  
-    return (byte) byte_read;
+    return in.readByte ();
   }
 
   /**
@@ -346,11 +327,9 @@
    *
    * @see DataOutput
    */
-  public final synchronized char readChar () throws IOException
+  public final char readChar () throws IOException
   {
-    readFully (buf, 0, 2);
-  
-    return (char) ((buf[0] << 8) | (buf[1] & 0xff));
+    return in.readChar();
   }
 
   /**
@@ -377,9 +356,7 @@
    */
   public final double readDouble () throws IOException
   {
-    long val = readLong();
-  
-    return Double.longBitsToDouble(val);
+    return in.readDouble ();
   }
 
   /**
@@ -404,9 +381,7 @@
    */
   public final float readFloat () throws IOException
   {
-    int val = readInt();
-  
-    return Float.intBitsToFloat(val);
+    return in.readFloat();
   }
 
   /**
@@ -423,7 +398,7 @@
    */
   public final void readFully (byte[] buffer) throws IOException
   {
-    readFully (buffer, 0, buffer.length);
+    in.readFully(buffer);
   }
 
   /**
@@ -443,19 +418,10 @@
    * the buffer
    * @exception IOException If any other error occurs
    */
-  public synchronized final void readFully (byte[] buffer, int offset, int len) 
+  public final void readFully (byte[] buffer, int offset, int count)
     throws IOException
   {
-    int total_read = 0;
-  
-    while (total_read < len)
-      {
-        int bytes_read = read (buffer, offset + total_read, len - total_read);
-        if (bytes_read == -1)
-          throw new EOFException("Unexpected end of stream");
-  
-        total_read += bytes_read;
-      }
+    in.readFully (buffer, offset, count);
   }
 
   /**
@@ -486,12 +452,9 @@
    *
    * @see DataOutput
    */
-  public final synchronized int readInt() throws IOException
+  public final int readInt () throws IOException
   {
-    readFully(buf, 0, 4);
-  
-    return (((buf[0] & 0xff) << 24) | ((buf[1] & 0xff) << 16) |
-            ((buf[2] & 0xff) << 8) | (buf[3] & 0xff));
+    return in.readInt();
   }
 
   /**
@@ -521,33 +484,9 @@
    *
    * @deprecated
    */
-  public synchronized final String readLine () throws IOException
+  public final String readLine () throws IOException
   {
-    StringBuffer sb = new StringBuffer ("");
-  
-    for (;;)
-      {
-        int byte_read = read ();
-   
-        if (byte_read == -1)
-          return sb.toString();
-  
-        char c = (char) byte_read;
-  
-        if (c == '\r')
-          {
-            byte_read = read();
-            if (((char)byte_read) != '\n')
-              seek (getFilePointer() - 1);
-  
-            return sb.toString();
-          }
-  
-        if (c == '\n')
-          return sb.toString();
-  
-        sb.append (c);
-      }
+    return in.readLine ();
   }
 
   /**
@@ -581,18 +520,9 @@
    *
    * @see DataOutput
    */
-  public final synchronized long readLong() throws IOException
+  public final long readLong () throws IOException
   {
-    readFully(buf, 0, 8);
-  
-    return (((long)(buf[0] & 0xff) << 56) |
-            ((long)(buf[1] & 0xff) << 48) |
-            ((long)(buf[2] & 0xff) << 40) |
-            ((long)(buf[3] & 0xff) << 32) |
-            ((long)(buf[4] & 0xff) << 24) |
-            ((long)(buf[5] & 0xff) << 16) |
-            ((long)(buf[6] & 0xff) <<  8) |
-            ((long)(buf[7] & 0xff)));
+    return in.readLong();
   }
 
   /**
@@ -622,11 +552,9 @@
    *
    * @see DataOutput
    */
-  public final synchronized short readShort () throws IOException
+  public final short readShort () throws IOException
   {
-    readFully (buf, 0, 2);
-    
-    return (short) ((buf[0] << 8) | (buf[1] & 0xff));
+    return in.readShort();
   }
 
   /**
@@ -647,12 +575,7 @@
    */
   public final int readUnsignedByte () throws IOException
   {
-    int byte_read = read ();
-  
-    if (byte_read == -1)
-      throw new EOFException ("Unexpected end of stream");
-  
-    return byte_read & 0xFF;
+    return in.readUnsignedByte();
   }
 
   /**
@@ -680,12 +603,9 @@
    * @exception EOFException If end of file is reached before reading the value
    * @exception IOException If any other error occurs
    */
-  public final synchronized int readUnsignedShort () 
-    throws IOException
+  public final int readUnsignedShort () throws IOException
   {
-    readFully(buf, 0, 2);
-    
-    return (((buf[0] & 0xff) << 8) | (buf[1] & 0xff));
+    return in.readUnsignedShort();
   }
 
   /**
@@ -760,16 +680,9 @@
    *
    * @see DataOutput
    */
-  public synchronized final String readUTF () throws IOException
+  public final String readUTF () throws IOException
   {
-    StringBuffer sb = new StringBuffer("");
-  
-    int num_bytes = readUnsignedShort();
-    byte[] buf = new byte[num_bytes];
-    readFully(buf);
-  
-    // FIXME: Look to migrate to new String(buf, "UTF-8") if performance ok
-    return DataInputStream.convertFromUTF(buf);
+    return in.readUTF();
   }
 
   /**
@@ -824,10 +737,7 @@
    */
   public void write (int oneByte) throws IOException
   {
-    if (readOnly)
-      throw new IOException("File is open read only");
-  
-    fd.write (oneByte);
+    out.write(oneByte);
   }
 
   /**
@@ -838,7 +748,7 @@
    */
   public void write (byte[] buffer) throws IOException
   {
-    write (buffer, 0, buffer.length);
+    out.write(buffer);
   }
 
   /**
@@ -853,10 +763,7 @@
    */
   public void write (byte[] buffer, int offset, int len) throws IOException
   {
-    if (readOnly)
-      throw new IOException("File is open read only");
-  
-    fd.write (buffer, offset, len);
+    out.write (buffer, offset, len);
   }
 
   /**
@@ -870,7 +777,7 @@
    */
   public final void writeBoolean (boolean val) throws IOException
   {
-    write (val ? 1 : 0);
+    out.writeBoolean(val);
   }
 
   /**
@@ -884,7 +791,7 @@
    */
   public final void writeByte (int v) throws IOException
   {
-    write (v & 0xFF);
+    out.writeByte(v);
   }
 
   /**
@@ -896,12 +803,9 @@
    *
    * @exception IOException If an error occurs
    */
-  public final synchronized void writeShort (int s) throws IOException
+  public final void writeShort (int v) throws IOException
   {
-    buf[0] = (byte)((s & 0xFF00) >> 8);
-    buf[1] = (byte)(s & 0x00FF);
-  
-    write(buf, 0, 2);
+    out.writeShort(v);
   }
 
   /**
@@ -913,12 +817,9 @@
    *
    * @exception IOException If an error occurs
    */
-  public final synchronized void writeChar (int v) throws IOException
+  public final void writeChar (int v) throws IOException
   {
-    buf[0] = (byte)((v & 0xFF00) >> 8);
-    buf[1] = (byte)((int)v & 0x00FF);
-  
-    write(buf, 0, 2);
+    out.writeChar(v);
   }
 
   /**
@@ -929,14 +830,9 @@
    *
    * @exception IOException If an error occurs
    */
-  public final synchronized void writeInt (int v) throws IOException
+  public final void writeInt (int v) throws IOException
   {
-    buf[0] = (byte)((v & 0xFF000000) >> 24);
-    buf[1] = (byte)((v & 0x00FF0000) >> 16);
-    buf[2] = (byte)((v & 0x0000FF00) >> 8);
-    buf[3] = (byte)(v & 0x000000FF);
-  
-    write (buf, 0, 4);
+    out.writeInt(v);
   }
 
   /**
@@ -947,18 +843,9 @@
    *
    * @exception IOException If an error occurs
    */
-  public final synchronized void writeLong (long v) throws IOException
+  public final void writeLong (long v) throws IOException
   {
-    buf[0] = (byte)((v & 0xFF00000000000000L) >> 56);
-    buf[1] = (byte)((v & 0x00FF000000000000L) >> 48);
-    buf[2] = (byte)((v & 0x0000FF0000000000L) >> 40);
-    buf[3] = (byte)((v & 0x000000FF00000000L) >> 32);
-    buf[4] = (byte)((v & 0x00000000FF000000L) >> 24);
-    buf[5] = (byte)((v & 0x0000000000FF0000L) >> 16);
-    buf[6] = (byte)((v & 0x000000000000FF00L) >> 8);
-    buf[7] = (byte)(v & 0x00000000000000FFL);
-  
-    write (buf, 0, 8);
+    out.writeLong(v);
   }
 
   /**
@@ -977,8 +864,7 @@
    */
   public final void writeFloat (float v) throws IOException
   {
-    int i = Float.floatToIntBits (v);
-    writeInt (i);
+    out.writeFloat(v);
   }
 
   /**
@@ -998,8 +884,7 @@
    */
   public final void writeDouble (double v) throws IOException
   {
-    long l = Double.doubleToLongBits (v);
-    writeLong (l);
+    out.writeDouble(v);
   }
 
   /**
@@ -1013,17 +898,7 @@
    */
   public final void writeBytes (String s) throws IOException
   {
-    int len = s.length();
-
-    if (len == 0)
-      return;
-  
-    byte[] buf = new byte[len];
-  
-    for (int i = 0; i < len; i++)
-      buf[i] = (byte)(s.charAt(i) & 0xFF);
-  
-    write(buf);
+    out.writeBytes(s);
   }
   
   /**
@@ -1037,19 +912,7 @@
    */
   public final void writeChars (String s) throws IOException
   {
-    int len = s.length();
-    if (len == 0)
-      return;
-  
-    byte[] buf = new byte[len * 2];
-  
-    for (int i = 0; i < len; i++)
-      {
-        buf[i * 2] = (byte)((s.charAt(i) & 0xFF00) >> 8);
-        buf[(i * 2) + 1] = (byte)(s.charAt(i) & 0x00FF);
-      }
-  
-    write(buf, 0, buf.length);
+    out.writeChars(s);
   }
   
   /**
@@ -1083,11 +946,7 @@
    */
   public final void writeUTF (String s) throws IOException
   {
-    // FIXME:  Look to migrate to s.getBytes("UTF-8") if performance ok
-    byte[] buf = DataOutputStream.convertToUTF(s);
-  
-    writeShort(buf.length);
-    write(buf);
+    out.writeUTF(s);
   }
   
   /**
@@ -1099,7 +958,7 @@
   public synchronized FileChannel getChannel ()
   {
     if (ch == null)
-      ch = new FileChannelImpl ((int) (fd.getNativeFd() & 0xFFFF), this);
+      ch = new FileChannelImpl (fd, true, this);
 
     return ch;
   }
