--- /home/tromey/gnu/egcs/classpath/classpath/java/security/SignedObject.java	Wed Sep 26 08:18:45 2001
+++ java/security/SignedObject.java	Fri Nov  2 21:44:51 2001
@@ -80,7 +80,7 @@
      @param signingKey the key to sign with
      @param signingEngine the signature engine to use
 
-     @throws IOException serialization error occured
+     @throws IOException serialization error occurred
      @throws InvalidKeyException invalid key
      @throws SignatureException signing error
    */
@@ -108,8 +108,8 @@
 
      @return the encapsulated object
 
-     @throws IOException de-serialization error occured
-     @throws ClassNotFoundException de-serialization error occured
+     @throws IOException de-serialization error occurred
+     @throws ClassNotFoundException de-serialization error occurred
    */
   public Object getObject() throws IOException, ClassNotFoundException
   {
