--- /home/tromey/gnu/classpath/classpath/java/io/FilePermission.java	Wed Oct  2 11:21:38 2002
+++ java/io/FilePermission.java	Tue Jan 22 15:40:14 2002
@@ -41,10 +41,7 @@
 import java.security.*;
 
 
-public final class FilePermission extends Permission implements Serializable
-{
-  static final long serialVersionUID = 7930732926638008763L;
-
+public final class FilePermission extends Permission implements Serializable {
   private static final String CURRENT_DIRECTORY = System.getProperty("user.dir");
   private boolean usingPerms = false;
   private boolean readPerm = false;
