--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/FileDialog.java	2004-04-28 02:20:15.000000000 -0600
+++ java/awt/FileDialog.java	2004-01-06 02:18:20.000000000 -0700
@@ -41,7 +41,6 @@
 import java.awt.peer.FileDialogPeer;
 import java.awt.peer.DialogPeer;
 import java.io.FilenameFilter;
-import java.io.Serializable;
 
 /**
   * This class implements a file selection dialog box widget.
@@ -49,7 +48,7 @@
   * @author Aaron M. Renn (arenn@urbanophile.com)
   * @author Tom Tromey <tromey@redhat.com>
   */
-public class FileDialog extends Dialog implements Serializable
+public class FileDialog extends Dialog implements java.io.Serializable
 {
 
 /*
