--- /home/tromey/gnu/classpath/classpath/java/nio/charset/IllegalCharsetNameException.java	Fri Nov  8 05:15:32 2002
+++ java/nio/charset/IllegalCharsetNameException.java	Mon Oct  7 01:59:12 2002
@@ -37,17 +37,17 @@
 
 package java.nio.charset;
 
-/**
- * @since 1.4
- */
-public class IllegalCharsetNameException extends IllegalArgumentException
+class IllegalCharsetNameException extends IllegalArgumentException
 {
+  private String charsetName;
+  
   /**
    * Creates the exception
    */
   public IllegalCharsetNameException (String charsetName)
   {
-    super (charsetName);
+    super ();
+    this.charsetName = charsetName;
   }
 
   /**
@@ -55,6 +55,6 @@
    */
   public String getCharsetName ()
   {
-    return getMessage ();
+    return charsetName;
   }
 }
