--- /home/tromey/gnu/classpath/classpath/java/nio/channels/Channels.java	Sun Nov 24 16:32:34 2002
+++ java/nio/channels/Channels.java	Wed Feb 12 03:10:28 2003
@@ -56,7 +56,7 @@
    */
   public static InputStream newInputStream (ReadableByteChannel ch)
   {
-    return null;
+    throw new Error ("not implemented");
   }
   
   /**
@@ -64,7 +64,7 @@
    */
   public static OutputStream newOutputStream (WritableByteChannel ch) 
   {
-    return null;
+    throw new Error ("not implemented");
   }
   
   /**
@@ -72,7 +72,7 @@
    */
   public static ReadableByteChannel newChannel (InputStream in)
   {
-    return null;
+    throw new Error ("not implemented");
   }
   
   /**
@@ -80,7 +80,7 @@
    */
   public static WritableByteChannel newChannel (OutputStream out)
   {
-    return null;	
+    throw new Error ("not implemented");
   }
 
   /**
@@ -90,7 +90,7 @@
   public static Reader newReader (ReadableByteChannel ch, CharsetDecoder dec,
                                   int minBufferCap)
   {
-    return null;
+    throw new Error ("not implemented");
   }
 
   /**
@@ -112,7 +112,7 @@
   public static Writer newWriter (WritableByteChannel ch, CharsetEncoder enc,
                                   int minBufferCap)
   {
-    return null;
+    throw new Error ("not implemented");
   }
 
   /**
