--- /home/tromey/gnu/Nightly/classpath/classpath/javax/imageio/ImageTranscoder.java	2004-10-04 02:33:45.000000000 -0600
+++ javax/imageio/ImageTranscoder.java	2004-04-21 02:20:30.000000000 -0600
@@ -1,5 +1,5 @@
 /* ImageTranscoder.java -- Image metadata transcoder.
-   Copyright (C) 2004  Free Software Foundation, Inc.
+   Copyright (C) 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,17 +38,9 @@
 
 package javax.imageio;
 
-import javax.imageio.metadata.IIOMetadata;
-
-/**
- * @author Michael Koch (konqueror@gmx.de)
- */
 public interface ImageTranscoder
 {
-  IIOMetadata convertImageMetadata(IIOMetadata inData,
-		                   ImageTypeSpecifier imageType,
-				   ImageWriteParam param);
-
-  IIOMetadata convertStreamMetadata(IIOMetadata inData,
-				    ImageWriteParam param);
+  // FIXME: Incomplete. This interface is merely present in order to
+  // allow compilation of the javax.imageio.spi package, for which GNU
+  // Classpath does provide an implementation.
 }
