--- /home/tromey/gnu/Nightly/classpath/classpath/javax/imageio/event/IIOWriteProgressListener.java	2004-10-28 02:27:40.000000000 -0600
+++ javax/imageio/event/IIOWriteProgressListener.java	2004-07-09 08:27:32.000000000 -0600
@@ -35,13 +35,12 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package javax.imageio.event;
 
 import java.util.EventListener;
-
 import javax.imageio.ImageWriter;
 
+
 public interface IIOWriteProgressListener extends EventListener
 {
   /**
