--- /home/tromey/gnu/classpath/classpath/java/util/jar/JarException.java	Sat Mar  2 12:13:22 2002
+++ java/util/jar/JarException.java	Tue Jan 22 16:01:45 2002
@@ -1,5 +1,5 @@
-/* JarException.java -- thrown to indicate an problem with a jar file
-   Copyright (C) 2000, 2002 Free Software Foundation, Inc.
+/* Attributes.java -- exception thrown to indicate an problem with a jar file
+   Copyright (C) 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -41,34 +41,35 @@
 
 /**
  * This exception is thrown to indicate an problem with a jar file.
+ * It can be constructed with or without a descriptive message of the problem.
+ * <p>
  * Note that none of the methods in the java.util.jar package actually declare
  * to throw this exception, most just declare that they throw an IOException
  * which is super class of JarException.
- *
- * @author Mark Wielaard (mark@klomp.org)
+ * 
  * @since 1.2
+ * @author Mark Wielaard (mark@klomp.org)
  */
+
 public class JarException extends ZipException
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = 7159778400963954473L;
+  // Constructors
 
   /**
    * Create a new JarException without a descriptive error message.
    */
   public JarException()
   {
+    super();
   }
 
   /**
    * Create a new JarException with a descriptive error message indicating
    * what went wrong. This message can later be retrieved by calling the
    * <code>getMessage()</code> method.
+   * @see java.lang.Throwable@getMessage()
    *
    * @param message The descriptive error message
-   * @see #getMessage()
    */
   public JarException(String message)
   {
