--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/spi/AbstractSelectableChannel.java	2004-02-06 02:21:25.000000000 -0700
+++ java/nio/channels/spi/AbstractSelectableChannel.java	2004-02-27 02:18:22.000000000 -0700
@@ -48,7 +48,6 @@
 
 public abstract class AbstractSelectableChannel extends SelectableChannel
 {
-  private int registered;
   private boolean blocking = true;
   private Object LOCK = new Object();
   private SelectorProvider provider;
@@ -135,9 +134,15 @@
    */
   public final SelectionKey keyFor(Selector selector)
   {
+    if (! isOpen())
+      return null;
+    
     try
       {
-        return register (selector, 0, null);
+        synchronized(blockingLock())
+	  {
+	    return locate (selector);
+	  }
       }
     catch (Exception e)
       {
@@ -196,7 +201,8 @@
 
         if (key != null)
           {
-            key.attach (att);
+	    if (att != null)
+	      key.attach (att);
           }
         else
           {
