--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DirectByteBufferImpl.java	2003-06-19 02:34:46.000000000 -0600
+++ java/nio/DirectByteBufferImpl.java	2003-06-18 02:22:30.000000000 -0600
@@ -38,14 +38,14 @@
 
 package java.nio;
 
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 public class DirectByteBufferImpl extends ByteBuffer
 {
   RawData address;
   private int offset;
   private boolean readOnly;
-  
+
   public DirectByteBufferImpl (RawData address, long len)
   {
     this (address, 0, (int) len, (int) len, 0, -1, false);
