--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/Calendar.java	2003-08-13 02:19:48.000000000 -0600
+++ java/util/Calendar.java	2003-06-13 02:27:40.000000000 -0600
@@ -361,21 +361,11 @@
   static final long serialVersionUID = -1807547505821590642L;
 
   /**
-   * The name of the resource bundle. Used only by getBundle()
+   * The name of the resource bundle.
    */
   private static final String bundleName = "gnu.java.locale.Calendar";
 
   /**
-   * get resource bundle: 
-   * The resources should be loaded via this method only. Iff an application
-   * uses this method, the resourcebundle is required. 
-   */
-  private static ResourceBundle getBundle(Locale locale) 
-  {
-    return ResourceBundle.getBundle(bundleName, locale);
-  }
-
-  /**
    * Constructs a new Calendar with the default time zone and the default
    * locale.
    */
@@ -395,7 +385,7 @@
     this.zone = zone;
     lenient = true;
 
-    ResourceBundle rb = getBundle(locale);
+    ResourceBundle rb = ResourceBundle.getBundle(bundleName, locale);
 
     firstDayOfWeek = ((Integer) rb.getObject("firstDayOfWeek")).intValue();
     minimalDaysInFirstWeek =
@@ -440,7 +430,7 @@
   public static synchronized Calendar getInstance(TimeZone zone, Locale locale)
   {
     String calendarClassName = null;
-    ResourceBundle rb = getBundle(locale);
+    ResourceBundle rb = ResourceBundle.getBundle(bundleName, locale);
     calendarClassName = rb.getString("calendarClass");
     if (calendarClassName != null)
       {
@@ -471,7 +461,8 @@
    */
   public static synchronized Locale[] getAvailableLocales()
   {
-    ResourceBundle rb = getBundle(new Locale("", ""));
+    ResourceBundle rb = ResourceBundle.getBundle(bundleName,
+						 new Locale("", ""));
     return (Locale[]) rb.getObject("availableLocales");
   }
 
