--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormatSymbols.java	2004-04-18 02:22:25.000000000 -0600
+++ java/text/DecimalFormatSymbols.java	2003-11-28 02:18:12.000000000 -0700
@@ -167,9 +167,9 @@
    * regard to the specified object:
    * <p>
    * <ul>
-   * <li>It is not <code>null</code>.</li>
-   * <li>It is an instance of <code>DecimalFormatSymbols</code>.</li>
-   * <li>All of its symbols are identical to the symbols in this object.</li>
+   * <li>It is not <code>null</code>.
+   * <li>It is an instance of <code>DecimalFormatSymbols</code>
+   * <li>All of its symbols are identical to the symbols in this object.
    * </ul>
    *
    * @return <code>true</code> if the specified object is equal to this
