--- /home/tromey/gnu/egcs/classpath/classpath/java/security/acl/NotOwnerException.java	Sat Mar  2 12:13:20 2002
+++ java/security/acl/NotOwnerException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
 /* NotOwnerException.java -- Attempt to modify an unowned ACL
-   Copyright (C) 1998, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -42,15 +42,12 @@
  * the caller to be the owner of the access control list (ACL) when the caller
  * is in fact not the owner of the ACL.
  *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @status updated to 1.4
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
  */
-public class NotOwnerException extends Exception
+public class NotOwnerException extends java.lang.Exception
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -5555597911163362399L;
 
   /**
    * Initializes a new instance of <code>NotOwnerException</code> that does
@@ -58,5 +55,6 @@
    */
   public NotOwnerException()
   {
+    super();
   }
 }
