--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Frame.java	2003-09-26 02:21:10.000000000 -0600
+++ java/awt/Frame.java	2004-01-20 02:18:20.000000000 -0700
@@ -1,5 +1,5 @@
 /* Frame.java -- AWT toplevel window
-   Copyright (C) 1999, 2000, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2000, 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -342,6 +342,8 @@
 setMenuBar(MenuBar menuBar)
 {
   this.menuBar = menuBar;
+  if (menuBar != null)
+    menuBar.addNotify(); 
   if (peer != null)
     ((FramePeer) peer).setMenuBar(menuBar);
 }
