--- /home/tromey/gnu/classpath/classpath/java/nio/channels/ClosedChannelException.java	Tue Apr 30 17:58:37 2002
+++ java/nio/channels/ClosedChannelException.java	Mon Oct  7 01:59:12 2002
@@ -37,12 +37,14 @@
 
 package java.nio.channels;
 
-import java.io.*;
-
+import java.io.IOException;
 
 public class ClosedChannelException extends IOException
 {
-    public ClosedChannelException()
-    {
-    }
+  /**
+   * Creates the exception
+   */
+  public ClosedChannelException()
+  {
+  }
 }
