--- /home/tromey/gnu/egcs/classpath/classpath/java/util/jar/JarOutputStream.java	Tue Nov  7 10:43:04 2000
+++ java/util/jar/JarOutputStream.java	Fri Nov  2 21:44:52 2001
@@ -93,7 +93,7 @@
    * This implementation just calls <code>super.putNextEntre()</code>.
    *
    * @param entry The information for the next entry
-   * @exception IOException when some unexpected I/O exception occured
+   * @exception IOException when some unexpected I/O exception occurred
    */
   public void putNextEntry(ZipEntry entry) throws IOException
   {
