--- /home/tromey/gnu/egcs/classpath/classpath/java/io/CharConversionException.java	Sun Apr 15 11:57:24 2001
+++ java/io/CharConversionException.java	Fri Nov  2 21:44:51 2001
@@ -33,7 +33,7 @@
  */
 
 /**
-  * This exception is thrown to indicate that a problem occured with
+  * This exception is thrown to indicate that a problem occurred with
   * an attempted character conversion.
   *
   * @version 0.0
