--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/ServerSocket.java	2003-06-20 02:22:09.000000000 -0600
+++ java/net/ServerSocket.java	2003-06-28 02:18:20.000000000 -0600
@@ -164,6 +164,14 @@
     bind (new InetSocketAddress (bindAddr, port), backlog);
   }
 
+  /*
+   * This method may only be used by java.nio.channels.ServerSocketChannel.open.
+   */
+  void setChannel (ServerSocketChannel ch)
+  {
+    this.ch = ch;
+  }
+
   /**
    * Binds the server socket to a specified socket address
    *
