--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/BorderUIResource.java	2003-07-29 02:20:01.000000000 -0600
+++ javax/swing/plaf/BorderUIResource.java	2003-06-28 02:18:23.000000000 -0600
@@ -277,8 +277,7 @@
      * constructed by this method will automatically reflect a change
      * to the component&#x2019;s background color.
      *
-     * <p><img src="../border/doc-files/BevelBorder-1.png"
-     * width="500" height="150"
+     * <p><img src="../border/BevelBorder-1.png" width="500" height="150"
      * alt="[An illustration showing raised and lowered BevelBorders]" />
      *
      * @param bevelType the desired appearance of the border. The value
@@ -298,9 +297,9 @@
      * Constructs a BevelBorderUIResource given its appearance type
      * and two colors for its highlight and shadow.
      *
-     * <p><img src="../border/doc-files/BevelBorder-2.png" width="500"
-     * height="150" alt="[An illustration showing BevelBorders that were
-     * constructed with this method]" />
+     * <p><img src="../border/BevelBorder-2.png" width="500" height="150"
+     * alt="[An illustration showing BevelBorders that were constructed
+     * with this method]" />
      *
      * @param bevelType the desired appearance of the border. The value
      *        must be either {@link javax.swing.border.BevelBorder#RAISED}
@@ -338,9 +337,9 @@
      * Constructs a BevelBorderUIResource given its appearance type
      * and all its colors.
      *
-     * <p><img src="../border/doc-files/BevelBorder-3.png" width="500"
-     * height="150" alt="[An illustration showing BevelBorders that
-     * were constructed with this method]" />
+     * <p><img src="../border/BevelBorder-3.png" width="500" height="150"
+     * alt="[An illustration showing BevelBorders that were constructed
+     * with this method]" />
      *
      * @param bevelType the desired appearance of the border. The value
      *        must be either {@link javax.swing.border.BevelBorder#RAISED}
@@ -430,8 +429,7 @@
    * always install borders that implement <code>UIResource</code>,
    * such as the borders provided by this class.
    *
-   * <p><img src="../border/doc-files/EmptyBorder-1.png"
-   * width="290" height="200"
+   * <p><img src="../border/EmptyBorder-1.png" width="290" height="200"
    * alt="[An illustration of EmptyBorder]" />
    *
    * @author Brian Jones (cbj@gnu.org)
@@ -485,9 +483,8 @@
    * always install borders that implement <code>UIResource</code>,
    * such as the borders provided by this class.
    *
-   * <p><img src="../border/doc-files/EtchedBorder-1.png" width="500"
-   * height="200" alt="[An illustration of the two EtchedBorder
-   * variants]" />
+   * <p><img src="../border/EtchedBorder-1.png" width="500" height="200"
+   * alt="[An illustration of the two EtchedBorder variants]" />
    *
    * @author Brian Jones (cbj@gnu.org)
    * @author Sascha Brawer (brawer@dandelis.ch)
@@ -512,9 +509,8 @@
      * appearance. The colors will be derived from the background
      * color of the enclosed Component when the border gets painted.
      *
-     * <p><img src="../border/doc-files/EtchedBorder-1.png"
-     * width="500" height="200" alt="[An illustration of the two
-     * EtchedBorder variants]" />
+     * <p><img src="../border/EtchedBorder-1.png" width="500" height="200"
+     * alt="[An illustration of the two EtchedBorder variants]" />
      *
      * @param etchType the desired appearance of the border. The value
      *        must be either {@link javax.swing.border.EtchedBorder#RAISED}
@@ -553,7 +549,7 @@
      * appearance, explicitly selecting the colors that will be used
      * for highlight and shadow.
      *
-     * <p><img src="../border/doc-files/EtchedBorder-2.png" width="500"
+     * <p><img src="../border/EtchedBorder-2.png" width="500"
      * height="200" alt="[An illustration that shows which pixels get
      * painted in what color]" />
      *
@@ -587,8 +583,8 @@
    * always install borders that implement <code>UIResource</code>,
    * such as the borders provided by this class.
    *
-   * <p><img src="../border/doc-files/LineBorder-1.png" width="500"
-   * height="200" alt="[An illustration of two LineBorders] />
+   * <p><img src="../border/LineBorder-1.png" width="500" height="200"
+   * alt="[An illustration of two LineBorders] />
    *
    * @author Brian Jones (cbj@gnu.org)
    * @author Sascha Brawer (brawer@dandelis.ch)
@@ -642,8 +638,8 @@
    * always install borders that implement <code>UIResource</code>,
    * such as the borders provided by this class.
    *
-   * <p><img src="../border/doc-files/MatteBorder-1.png" width="500"
-   * height="150" alt="[An illustration of two MatteBorders] />
+   * <p><img src="../border/MatteBorder-1.png" width="500" height="150"
+   * alt="[An illustration of two MatteBorders] />
    *
    * @author Brian Jones (cbj@gnu.org)
    * @author Sascha Brawer (brawer@dandelis.ch)
@@ -656,9 +652,8 @@
      * Constructs a MatteBorderUIResource given the width on each side
      * and a fill color.
      *
-     * <p><img src="../border/doc-files/MatteBorder-2.png" width="500"
-     * height="150" alt="[A picture of a MatteBorder made by this
-     * constructor]" />
+     * <p><img src="../border/MatteBorder-2.png" width="500" height="150"
+     * alt="[A picture of a MatteBorder made by this constructor]" />
      *
      * @param top the width of the border at its top edge.
      * @param left the width of the border at its left edge.
@@ -678,7 +673,7 @@
      * Constructs a MatteBorderUIResource given the width on each side
      * and an icon for tiling the border area.
      *
-     * <p><img src="../border/doc-files/MatteBorder-4.png" width="500"
+     * <p><img src="../border/MatteBorder-4.png" width="500"
      * height="150" alt="[A picture of a MatteBorder made by this
      * constructor]" />
      *
@@ -702,11 +697,10 @@
      * the left and right edge, the icon height for the top and bottom
      * edge.
      *
-     * <p><img src="../border/doc-files/MatteBorder-6.png" width="500"
-     * height="150" alt="[A picture of a MatteBorder made by this
-     * constructor]" />
+     * <p><img src="../border/MatteBorder-6.png" width="500" height="150"
+     * alt="[A picture of a MatteBorder made by this constructor]" />
      *
-     * @param tileIcon an icon for tiling the border area. 
+     * @param tileIcon an icon for tiling the border area.
      */
     public MatteBorderUIResource(Icon tileIcon)
     {
