--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/Integer.java	Mon Jul 30 12:07:52 2001
+++ java/lang/Integer.java	Fri Oct 26 16:18:42 2001
@@ -58,7 +58,7 @@
    * The primitive type <code>int</code> is represented by this 
    * <code>Class</code> object.
    */
-  public static final Class TYPE = VMClassLoader.getPrimitiveClass ("int");
+  public static final Class TYPE = VMClassLoader.getPrimitiveClass ('I');
 
   /**
    * The immutable value of this Integer.
