--- /home/tromey/gnu/classpath/classpath/javax/naming/BinaryRefAddr.java	Tue Jan 22 15:50:12 2002
+++ javax/naming/BinaryRefAddr.java	Tue Jan 22 16:01:47 2002
@@ -97,17 +97,17 @@
    *         the addrType is the same as this addrType and the bytes of the
    *         content are the same.
    */
-  public boolean equal(Object o)
+  public boolean equals (Object o)
   {
     if (o instanceof BinaryRefAddr)
       {
         BinaryRefAddr refAddr = (BinaryRefAddr) o;
         if (this.getType().equals(refAddr.getType()))
-        {
-          byte[] c1 = (byte[]) this.getContent();
-          byte[] c2 = (byte[]) refAddr.getContent();
-	  return Arrays.equals(c1, c2);
-        }
+	  {
+	    byte[] c1 = (byte[]) this.getContent();
+	    byte[] c2 = (byte[]) refAddr.getContent();
+	    return Arrays.equals(c1, c2);
+	  }
       }
     return false;
   }
