--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JToolTip.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JToolTip.java	2004-06-18 02:18:54.000000000 -0600
@@ -42,12 +42,18 @@
 
 public class JToolTip extends JComponent implements Accessible
 {
-    String text;
+  private static final long serialVersionUID = -1138929898906751643L;
+  
+  String text;
 
-    JToolTip(String text)
-    {
-	this.text = text;
-    }
+  public JToolTip()
+  {
+  }
+
+  public void setTipText(String newText)
+  {
+    this.text = newText;
+  }
 }
 
 
