--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/AttributedString.java	2004-03-09 02:21:57.000000000 -0700
+++ java/text/AttributedString.java	2002-01-22 16:01:37.000000000 -0700
@@ -1,5 +1,5 @@
 /* AttributedString.java -- Models text with attributes
-   Copyright (C) 1998, 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,7 +39,6 @@
 package java.text;
 
 import java.util.Iterator;
-import java.util.HashMap;
 import java.util.Hashtable;
 import java.util.HashSet;
 import java.util.Map;
@@ -330,7 +329,7 @@
   * of the string.
   *
   * @param attrib The attribute to add.
-  * @param value The value of the attribute, which may be null.
+  * @param value The value of the attribute.
   * @param begin_index The beginning index of the subrange.
   * @param end_index The ending index of the subrange.
   *
@@ -343,10 +342,10 @@
   if (attrib == null)
     throw new IllegalArgumentException("null attribute");
 
-  HashMap hm = new HashMap();
-  hm.put(attrib, value);
+  Hashtable ht = new Hashtable();
+  ht.put(attrib, value);
 
-  addAttributes(hm, begin_index, end_index);
+  addAttributes(ht, begin_index, end_index);
 }
 
 /*************************************************************************/
