--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PipedReader.java	Mon Mar 10 02:19:32 2003
+++ java/io/PipedReader.java	Tue Jan 22 16:01:33 2002
@@ -118,8 +118,7 @@
   }
 
   /**
-    * This method connects this stream to the passed in 
-    * <code>PipedWriter</code>.
+    * This method connects this stream to the passed in <code>PipedWriter</code>.
     * This stream is then ready for reading.  If this stream is already
     * connected or has been previously closed, then an exception is thrown
     *
@@ -208,12 +207,9 @@
   
   /**
     * This method reads chars from the stream into a caller supplied buffer.
-    * It starts storing chars at position <code>offset</code> into the 
-    * buffer and
-    * reads a maximum of <code>len</code> chars.  Note that this method 
-    * can actually
-    * read fewer than <code>len</code> chars.  The actual number of chars 
-    * read is
+    * It starts storing chars at position <code>offset</code> into the buffer and
+    * reads a maximum of <code>len</code> chars.  Note that this method can actually
+    * read fewer than <code>len</code> chars.  The actual number of chars read is
     * returned.  A -1 is returned to indicated that no chars can be read
     * because the end of the stream was reached.  If the stream is already
     * closed, a -1 will again be returned to indicate the end of the stream.
@@ -241,11 +237,10 @@
   }
   
   /**
-    * This method reads characters from the stream into a caller supplied 
-    * buffer. It starts storing chars at position <code>offset</code> into 
-    * the buffer and reads a maximum of <code>len</code> chars.  Note that 
-    * this method can actually read fewer than <code>len</code> chars.  
-    * The actual number of chars read is
+    * This method reads characters from the stream into a caller supplied buffer.
+    * It starts storing chars at position <code>offset</code> into the buffer and
+    * reads a maximum of <code>len</code> chars.  Note that this method can actually
+    * read fewer than <code>len</code> chars.  The actual number of chars read is
     * returned.  A -1 is returned to indicated that no chars can be read
     * because the end of the stream was reached - ie close() was called on the
     * connected PipedWriter.
@@ -366,4 +361,3 @@
     }
   }
 }
-
