--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PipedInputStream.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/PipedInputStream.java	2003-03-23 02:18:02.000000000 -0700
@@ -277,7 +277,7 @@
     * @param offset The index into the buffer at which to start writing.
     * @param len The maximum number of bytes to read.
     *
-    * @exception IOException If <code>close()</code> was called on this Piped
+    * @exception IOException If <code>close()/code> was called on this Piped
     *                        InputStream.
     */  
   public synchronized int read(byte[] buf, int offset, int len)
