--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ByteOrder.java	2003-10-21 02:19:05.000000000 -0600
+++ java/nio/ByteOrder.java	2003-03-11 02:17:52.000000000 -0700
@@ -38,8 +38,6 @@
 
 package java.nio;
 
-import gnu.classpath.Configuration;
-
 /**
  * @author Michael Koch
  * @since 1.4
@@ -49,15 +47,6 @@
   public static final ByteOrder BIG_ENDIAN     = new ByteOrder();
   public static final ByteOrder LITTLE_ENDIAN  = new ByteOrder();
 
-  static
-  {
-    // load the shared library needed for native methods.
-    if (Configuration.INIT_LOAD_LIBRARY)
-      {
-        System.loadLibrary ("javanio");
-      }
-  }
-  
   /**
    * Returns the native byte order of the platform currently running.
    */
