--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/StringBuffer.java	Mon Oct  1 14:44:13 2001
+++ java/lang/StringBuffer.java	Tue Oct  2 17:30:13 2001
@@ -100,10 +100,7 @@
    *  @return this <code>StringBuffer</code>.
    *  @see java.lang.String#valueOf(int)
    */
-  public StringBuffer append (int inum)
-  {
-    return append (String.valueOf(inum));
-  }
+  public native StringBuffer append (int inum);
 
   /** Append the <code>String</code> value of the argument to this <code>StringBuffer</code>.
    *  Uses <code>String.valueOf()</code> to convert to
