--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/border/CompoundBorder.java	2003-06-16 02:24:07.000000000 -0600
+++ javax/swing/border/CompoundBorder.java	2003-06-13 02:27:43.000000000 -0600
@@ -51,10 +51,6 @@
 public class CompoundBorder
   extends AbstractBorder
 {
-  /**
-   * Determined using the <code>serialver</code> tool
-   * of Apple/Sun JDK 1.3.1 on MacOS X 10.1.5.
-   */
   static final long serialVersionUID = 9054540377030555103L;
 
 
