--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/standard/JobName.java	2004-05-06 02:21:14.000000000 -0600
+++ javax/print/attribute/standard/JobName.java	2004-01-09 04:26:42.000000000 -0700
@@ -42,10 +42,6 @@
 import javax.print.attribute.PrintRequestAttribute;
 import javax.print.attribute.TextSyntax;
 
-
-/**
- * @author Michael Koch (konqueror@gmx.de)
- */
 public final class JobName extends TextSyntax
   implements PrintJobAttribute, PrintRequestAttribute
 {
@@ -68,7 +64,7 @@
    *
    * @param obj the object to test
    *
-   * @return true if both objects are equal, false otherwise.
+   * @returns true if both objects are equal, false otherwise.
    */
   public boolean equals(Object obj)
   {
