--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/cert/Certificate.java	2003-05-01 02:23:13.000000000 -0600
+++ java/security/cert/Certificate.java	2003-04-30 02:19:06.000000000 -0600
@@ -47,7 +47,6 @@
 import java.io.ByteArrayInputStream;
 import java.io.InvalidObjectException;
 import java.io.ObjectStreamException;
-import java.io.Serializable;
 
 /**
  * The Certificate class is an abstract class used to manage 
@@ -72,9 +71,9 @@
  * @author Mark Benvenuto
  * @author Casey Marshall
  */
-public abstract class Certificate implements Serializable
+public abstract class Certificate
 {
-  private static final long serialVersionUID = -6751606818319535583L;
+  static final long serialVersionUID = -6751606818319535583L;
 	
   private String type;
 
@@ -228,7 +227,7 @@
    * @throws ObjectStreamException If the replacement could not be
    * created.
    */
-  protected Object writeReplace() throws ObjectStreamException
+  public Object writeReplace() throws ObjectStreamException
   {
     try
       {
