--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JFrame.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JFrame.java	2004-06-18 02:18:52.000000000 -0600
@@ -59,10 +59,23 @@
  */
 public class JFrame extends Frame implements WindowConstants, RootPaneContainer
 {
-    protected  AccessibleContext accessibleContext;
+  private static final long serialVersionUID = -3362141868504252139L;
+  
+  protected  AccessibleContext accessibleContext;
+  
+  private int close_action = HIDE_ON_CLOSE;    
+  
+  private static boolean defaultLookAndFeelDecorated = false;    
 
-    private int close_action = HIDE_ON_CLOSE;    
-    
+  public static void setDefaultLookAndFeelDecorated(boolean d)
+  {
+    defaultLookAndFeelDecorated = d;
+  }
+
+  public static boolean isDefaultLookAndFeelDecorated()
+  {
+    return defaultLookAndFeelDecorated;
+  }
 
     /***************************************************
      *
@@ -111,10 +124,10 @@
     return d;
   }
 
-    JMenuBar getJMenuBar()
+  public JMenuBar getJMenuBar()
     {    return getRootPane().getJMenuBar();   }
     
-    void setJMenuBar(JMenuBar menubar)
+  public void setJMenuBar(JMenuBar menubar)
     {    getRootPane().setJMenuBar(menubar); }
     
 
