--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RemoteRef.java	Tue Jan 22 15:27:00 2002
+++ java/rmi/server/RemoteRef.java	Mon Mar 31 02:17:49 2003
@@ -43,19 +43,37 @@
 import java.rmi.RemoteException;
 import java.io.ObjectOutput;
 
-public interface RemoteRef
-	extends Externalizable {
+public interface RemoteRef extends Externalizable
+{
+  public static final long serialVersionUID = 0;
+  
+  public static final String packagePrefix = "gnu.java.rmi.server";
 
-public static final long serialVersionUID = 0;
-public static final String packagePrefix = "gnu.java.rmi.server";
-
-public void invoke(RemoteCall call) throws Exception;
-public Object invoke(Remote obj, Method method, Object[] params, long opnum) throws Exception;
-public RemoteCall newCall(RemoteObject obj, Operation[] op, int opnum, long hash) throws RemoteException;
-public void done(RemoteCall call) throws RemoteException;
-public boolean remoteEquals(RemoteRef ref);
-public int remoteHashCode();
-public String getRefClass(ObjectOutput out);
-public String remoteToString();
+  /**
+   * @deprecated
+   */
+  public void invoke(RemoteCall call) throws Exception;
+  
+  public Object invoke (Remote obj, Method method, Object[] params, long opnum)
+    throws Exception;
+  
+  /**
+   * @deprecated
+   */
+  public RemoteCall newCall (RemoteObject obj, Operation[] op, int opnum,
+                             long hash)
+    throws RemoteException;
 
+  /**
+   * @deprecated
+   */
+  public void done (RemoteCall call) throws RemoteException;
+  
+  public boolean remoteEquals (RemoteRef ref);
+  
+  public int remoteHashCode ();
+  
+  public String getRefClass (ObjectOutput out);
+  
+  public String remoteToString ();
 }
