--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/UnsupportedClassVersionError.java	Sat Mar  2 12:13:13 2002
+++ java/lang/UnsupportedClassVersionError.java	Tue Jan 22 16:01:35 2002
@@ -1,6 +1,5 @@
-/* UnsupportedClassVersionError.java -- thrown when a class file version
-   exceeds the capability of the virtual machine
-   Copyright (C) 1998, 2002 Free Software Foundation, Inc.
+/* UnsupportedClassVersionError.java 
+   Copyright (C) 1998 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -8,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -44,31 +43,27 @@
  * Java Virtual Machine determines it does not support the major and minor
  * version numbers in the class file it is attempting to read.
  *
+ * @since JDK 1.2
+ * 
  * @author Brian Jones
- * @since 1.2
- * @status updated to 1.4
  */
 public class UnsupportedClassVersionError extends ClassFormatError
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = -7123279212883497373L;
+  static final long serialVersionUID = -7123279212883497373L;
 
   /**
    * Create an error without a message.
    */
   public UnsupportedClassVersionError()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an error with a message.
-   *
-   * @param s the message
    */
   public UnsupportedClassVersionError(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
