--- /home/tromey/gnu/egcs/classpath/classpath/java/util/Arrays.java	Mon May  6 10:40:03 2002
+++ java/util/Arrays.java	Sun Apr  7 15:27:50 2002
@@ -2205,9 +2205,9 @@
    *         comparable
    * @throws NullPointerException if an element is null (since
    *         null.compareTo cannot work)
-   * @throws ArrayIndexOutOfBoundsException if fromIndex and toIndex
+   * @throws ArrayIndexOutOfBoundsException, if fromIndex and toIndex
    *         are not in range.
-   * @throws IllegalArgumentException if fromIndex &gt; toIndex
+   * @throws IllegalArgumentException if fromIndex > toIndex
    */
   public static void sort(Object[] a, int fromIndex, int toIndex)
   {
@@ -2229,9 +2229,9 @@
    *        the elements' natural order
    * @throws ClassCastException if any two elements are not mutually
    *         comparable by the Comparator provided
-   * @throws ArrayIndexOutOfBoundsException if fromIndex and toIndex
+   * @throws ArrayIndexOutOfBoundsException, if fromIndex and toIndex
    *         are not in range.
-   * @throws IllegalArgumentException if fromIndex &gt; toIndex
+   * @throws IllegalArgumentException if fromIndex > toIndex
    * @throws NullPointerException if a null element is compared with natural
    *         ordering (only possible when c is null)
    */
