--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/Writer.java	2004-05-18 02:42:48.000000000 -0600
+++ java/io/Writer.java	2004-05-17 02:18:28.000000000 -0600
@@ -76,7 +76,7 @@
    * on the specified <code>Object</code>.
    *
    * @param lock The <code>Object</code> to use for synchronizing critical
-   *             sections. Must not be null.
+   *             sections. Must be not be null.
    */
   protected Writer(Object lock)
   {
