--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/Document.java	2004-01-11 02:20:39.000000000 -0700
+++ javax/swing/text/Document.java	2004-06-18 02:18:59.000000000 -0600
@@ -1,4 +1,4 @@
-/* Document.java -- 
+/* Document.java --
    Copyright (C) 2002, 2004 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -40,27 +40,44 @@
 import javax.swing.event.DocumentListener;
 import javax.swing.event.UndoableEditListener;
 
+
 public interface Document
-{ 
+{
   String StreamDescriptionProperty = "stream";
-  
   String TitleProperty = "text";
 
   void addDocumentListener(DocumentListener listener);
+
   void addUndoableEditListener(UndoableEditListener listener);
+
   Position createPosition(int offs);
+
   Element getDefaultRootElement();
+
   Position getEndPosition();
+
   int getLength();
+
   Object getProperty(Object key);
+
   Element[] getRootElements();
+
   Position getStartPosition();
+
   String getText(int offset, int length);
+
   void getText(int offset, int length, Segment txt);
-  void insertString(int offset, String str, AttributeSet a);
+
+  void insertString(int offset, String str, AttributeSet a)
+    throws BadLocationException;
+
   void putProperty(Object key, Object value);
+
   void remove(int offs, int len);
+
   void removeDocumentListener(DocumentListener listener);
+
   void removeUndoableEditListener(UndoableEditListener listener);
+
   void render(Runnable r);
 }
