--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/Naming.java	2004-04-19 02:20:14.000000000 -0600
+++ java/rmi/Naming.java	2003-12-27 02:18:02.000000000 -0700
@@ -44,16 +44,14 @@
 
 public final class Naming {
 
-/**
+/** <pre>
  * Looks for the remote object that is associated with the named service.
  * Name and location is given in form of a URL without a scheme:
- *
- * <pre>
- * //host:port/service-name
- * </pre>
+ * 
+ *   //host:port/service-name
  *  
  * The port is optional.
- * 
+ * </pre>
  * @param name the service name and location
  * @return Remote-object that implements the named service
  * @throws NotBoundException if no object implements the service
