--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JDialog.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JDialog.java	2004-06-18 02:18:51.000000000 -0600
@@ -62,6 +62,7 @@
 public class JDialog extends Dialog implements Accessible, WindowConstants,
                                                RootPaneContainer
 {
+  private static final long serialVersionUID = -864070866424508218L;
 
   /** DOCUMENT ME! */
   protected AccessibleContext accessibleContext;
