--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/SignedObject.java	2004-04-18 02:22:22.000000000 -0600
+++ java/security/SignedObject.java	2003-05-10 02:19:17.000000000 -0600
@@ -98,7 +98,7 @@
  * <p>The signature algorithm can be, among others, the NIST standard <i>DSS</i>,
  * using <i>DSA</i> and <i>SHA-1</i>. The algorithm is specified using the same
  * convention as that for signatures. The <i>DSA</i> algorithm using the
- * <i>SHA-1</i> message digest algorithm can be specified, for example, as
+ * </i>SHA-1</i> message digest algorithm can be specified, for example, as
  * <code>"SHA/DSA"</code> or <code>"SHA-1/DSA"</code> (they are equivalent). In
  * the case of <i>RSA</i>, there are multiple choices for the message digest
  * algorithm, so the signing algorithm could be specified as, for example,
