--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/SimpleDateFormat.java	2003-10-07 02:20:30.000000000 -0600
+++ java/text/SimpleDateFormat.java	2003-09-28 02:18:21.000000000 -0600
@@ -117,8 +117,7 @@
       field = formatData.getLocalPatternChars().indexOf(thisChar);
       if (field == -1) {
 	current = null;
-	if (Character.isLowerCase (thisChar)
-	    || Character.isUpperCase (thisChar)) {
+	if (Character.isLetter(thisChar)) {
 	  // Not a valid letter
 	  tokens.add(new FieldSizePair(-1,0));
 	} else if (thisChar == '\'') {
