--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Integer.java	2004-04-18 02:22:17.000000000 -0600
+++ java/lang/Integer.java	2002-06-13 10:47:09.000000000 -0600
@@ -220,9 +220,9 @@
    * within <code>MIN_VALUE</code> to <code>MAX_VALUE</code>, inclusive.
    * Unlike Double.parseDouble, you may not have a leading '+'.
    *
-   * @param str the <code>String</code> to convert
+   * @param s the <code>String</code> to convert
    * @param radix the radix (base) to use in the conversion
-   * @return the <code>String</code> argument converted to <code>int</code>
+   * @return the <code>String</code> argument converted to </code>int</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as an
    *         <code>int</code>
    */
@@ -414,7 +414,7 @@
    * interpret the value of the property.
    *
    * @param nm the name of the system property
-   * @param def the default value
+   * @param val the default value
    * @return the value of the system property, or the default
    * @throws SecurityException if accessing the system property is forbidden
    * @see System#getProperty(String)
@@ -461,7 +461,7 @@
    * Finally, the value must be in the range <code>MIN_VALUE</code> to
    * <code>MAX_VALUE</code>, or an exception is thrown.
    *
-   * @param str the <code>String</code> to interpret
+   * @param s the <code>String</code> to interpret
    * @return the value of the String as an <code>Integer</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as a
    *         <code>int</code>
