--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DecimalFormat.java	2004-05-21 02:26:38.000000000 -0600
+++ java/text/DecimalFormat.java	2004-05-05 02:18:55.000000000 -0600
@@ -42,11 +42,13 @@
 import gnu.java.text.FormatCharacterIterator;
 import gnu.java.text.StringFormatBuffer;
 
-import java.io.IOException;
-import java.io.ObjectInputStream;
 import java.util.Currency;
 import java.util.HashMap;
 import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+import java.io.ObjectInputStream;
+import java.io.IOException;
 
 /**
  * @author Tom Tromey <tromey@cygnus.com>
@@ -847,7 +849,7 @@
     // FIXME: handle Inf and NaN.
 
     // FIXME: do we have to respect minimum digits?
-    // What about multiplier?
+    // What about leading zeros?  What about multiplier?
 
     StringBuffer buf = int_buf;
     StringBuffer frac_buf = null;
@@ -855,13 +857,7 @@
     int start_index = index;
     int max = str.length();
     int exp_index = -1;
-    int last = index + maximumIntegerDigits; 
-
-    if (maximumFractionDigits > 0)
-      last += maximumFractionDigits + 1;
-    
-    if (useExponentialNotation)
-      last += minExponentDigits + 1;
+    int last = index + MAXIMUM_INTEGER_DIGITS;
 
     if (last > 0 && max > last)
       max = last;
