--- /home/tromey/gnu/egcs/classpath/classpath/java/security/interfaces/RSAPublicKey.java	Wed Mar 22 12:26:27 2000
+++ java/security/interfaces/RSAPublicKey.java	Wed Apr 25 11:13:37 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,31 +24,24 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security.interfaces;
 
 import java.security.PublicKey;
 import java.math.BigInteger;
 
 /**
-  * This interface provides access to information about an RSA public key.
-  *
-  * @version 0.1
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This interface provides access to information about an RSA public key.
+ *
+ * @version 0.1
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public interface RSAPublicKey extends PublicKey, RSAKey
 {
-
-/*************************************************************************/
-
-/**
-  * Returns the public exponent value for this key
-  *
-  * @return The public exponent value for this key
-  */
-public abstract BigInteger
-getPublicExponent();
-
-} // interface RSAPublicKey 
-
+  /**
+   * Returns the public exponent value for this key
+   *
+   * @return The public exponent value for this key
+   */
+  public abstract BigInteger getPublicExponent();
+}
