--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/AWTEvent.java	2004-06-08 02:24:42.000000000 -0600
+++ java/awt/AWTEvent.java	2004-06-18 02:18:50.000000000 -0600
@@ -238,8 +238,16 @@
    */
   public String toString ()
   {
-    return getClass ().getName () + "[" + paramString () + "] on "
-      + ((Component) source).getName ();
+    String string = null;
+
+    if (source instanceof Component)
+      string = getClass ().getName () + "[" + paramString () + "] on "
+        + ((Component) source).getName ();
+    else if (source instanceof MenuComponent)
+      string = getClass ().getName () + "[" + paramString () + "] on "
+        + ((MenuComponent) source).getName ();
+
+    return string;
   }
 
   /**
