--- /home/tromey/gnu/classpath/classpath/java/nio/channels/spi/SelectorProvider.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/spi/SelectorProvider.java	Wed Feb 19 14:18:33 2003
@@ -83,7 +83,8 @@
   /**
    * Opens a server socket channel.
    */
-  public abstract ServerSocketChannel openServerSocketChannel () throws IOException;
+  public abstract ServerSocketChannel openServerSocketChannel ()
+    throws IOException;
   
   /**
    * Opens a socket channel.
