--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/BasicPermission.java	2004-09-25 02:36:40.000000000 -0600
+++ java/security/BasicPermission.java	2004-08-06 02:20:24.000000000 -0600
@@ -71,9 +71,8 @@
  * @since 1.1
  * @status updated to 1.4
  */
-public abstract class BasicPermission extends java.security.Permission
+public abstract class BasicPermission extends Permission
   implements Serializable
-  // FIXME extends with fully qualified classname as workaround for gcj 3.3.
 {
   /**
    * Compatible with JDK 1.1+.
