--- /home/tromey/gnu/egcs/classpath/classpath/java/text/ParseException.java	Sat Mar  2 12:13:22 2002
+++ java/text/ParseException.java	Tue Jan 22 16:01:38 2002
@@ -1,5 +1,5 @@
-/* ParseException.java -- an error occurred while parsing
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* ParseException.java -- An error occurred while parsing.
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,49 +38,65 @@
 
 package java.text;
 
-/**
- * This exception is thrown when an unexpected error occurs during parsing.
- *
- * @author Aaron M. Renn (arenn@urbanophile.com)
- * @author Per Bothner <bothner@cygnus.com>
- * @see Format
- * @see FieldPosition
- * @status updated to 1.4
+/* Written using "Java Class Libraries", 2nd edition, plus online
+ * API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct.
  */
+
+/**
+  * This exception is thrown when an unexpected error occurs during parsing.
+  *
+  * @version 0.0
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  * @author Per Bothner <bothner@cygnus.com>
+  * @date October 25, 1998.
+  */
 public class ParseException extends Exception
 {
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = 2703218443322787634L;
-
-  /**
-   * This is the position where the error was encountered.
-   *
-   * @serial the zero-based offset in the string where the error occurred
-   */
-  private final int errorOffset;
-
-  /**
-   * This method initializes a new instance of <code>ParseException</code>
-   * with a detailed error message and a error position.
-   *
-   * @param msg the descriptive message describing the error
-   * @param offset the position where the error was encountered
-   */
-  public ParseException(String s, int offset)
-  {
-    super(s);
-    errorOffset = offset;
-  }
-
-  /**
-   * This method returns the position where the error occurred.
-   *
-   * @return the position where the error occurred
-   */
-  public int getErrorOffset()
-  {
-    return errorOffset;
-  }
+
+/*
+ * Instance Variables
+ */
+
+/**
+  * This is the position where the error was encountered.
+  */
+private int errorOffset;
+
+/*************************************************************************/
+
+/*
+ * Constructors
+ */
+
+/**
+  * This method initializes a new instance of <code>ParseException</code>
+  * with a detailed error message and a error position.
+  *
+  * @param msg The descriptive message describing the error.
+  * @param offset The position where the error was encountered.
+  */
+public
+ParseException(String s, int offset)
+{
+  super(s);
+  
+  errorOffset = offset;
+}
+
+/*************************************************************************/
+
+/**
+  * This method returns the position where the error occurred.
+  * 
+  * @return The position where the error occurred.
+  */
+public int
+getErrorOffset()
+{
+  return(errorOffset);
+}
+
 } // class ParseException
+
