--- /home/tromey/gnu/classpath/classpath/java/rmi/NoSuchObjectException.java	Sat Mar  2 12:13:14 2002
+++ java/rmi/NoSuchObjectException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* NoSuchObjectException.java -- thrown if the remote object no longer exists
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,32 +37,13 @@
 
 package java.rmi;
 
-/**
- * Thrown on an attempt to invoke a call on an object that no longer exists
- * in the remote Virtual Machine. The call may be retransmitted and still
- * obey the semantics of "at most once".
- *
- * @author unknown
- * @see RemoteObject#toStub(Remote)
- * @see UnicastRemoteObject#unexportObject(Remote, boolean)
- * @see Activatable#unexportObject(Remote, boolean)
- * @since 1.1
- * @status updated to 1.4
- */
-public class NoSuchObjectException extends RemoteException
-{
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = 6619395951570472985L;
-
-  /**
-   * Create an exception with a message.
-   *
-   * @param s the message
-   */
-  public NoSuchObjectException(String s)
-  {
-    super(s);
-  }
+public class NoSuchObjectException
+	extends RemoteException {
+
+public static final long serialVersionUID = 6619395951570472985L;
+
+public NoSuchObjectException(String s) {
+	super(s);
+}
+
 }
