--- /home/tromey/gnu/egcs/classpath/classpath/java/security/SecureClassLoader.java	Mon Mar 25 00:05:18 2002
+++ java/security/SecureClassLoader.java	Tue Jan 22 16:01:36 2002
@@ -51,16 +51,12 @@
   protected SecureClassLoader(ClassLoader parent)
   {
     super(parent);
-    SecurityManager sm = System.getSecurityManager();
-    if(sm != null)
-      sm.checkCreateClassLoader();
+    // FIXME: What else?
   }
 
   protected SecureClassLoader()
   {
-    SecurityManager sm = System.getSecurityManager();
-    if(sm != null)
-      sm.checkCreateClassLoader();
+    // FIXME: What do we need to do here?
   }
 
   /** 
