--- /home/tromey/gnu/classpath/classpath/java/awt/ScrollPaneAdjustable.java	Tue Jan 22 15:49:51 2002
+++ java/awt/ScrollPaneAdjustable.java	Wed Feb 19 14:18:32 2003
@@ -39,34 +39,28 @@
 package java.awt;
 
 /**
-  * Need this class since the serialization spec for ScrollPane
-  * uses it.
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * Need this class since the serialization spec for ScrollPane
+ * uses it.
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 class ScrollPaneAdjustable extends Scrollbar
 {
-
-public
-ScrollPaneAdjustable(int orientation)
-{
-  super(orientation);
-}
-
-public void
-setMaximum(int maximum)
-{
-}
-
-public void
-setMinimum(int minimum)
-{
-}
-
-public void
-setVisibleAmount(int visibleAmount)
-{
-}
-
+  public ScrollPaneAdjustable (int orientation)
+  {
+    super (orientation);
+  }
+
+  public void setMaximum (int maximum)
+  {
+  }
+
+  public void setMinimum (int minimum)
+  {
+  }
+
+  public void setVisibleAmount (int visibleAmount)
+  {
+  }
 } // class ScrollPaneAdjustable
 
