--- /home/tromey/gnu/egcs/classpath/classpath/java/text/Collator.java	Tue Apr 30 11:50:10 2002
+++ java/text/Collator.java	Tue Jan 22 16:01:37 2002
@@ -153,8 +153,8 @@
    * @param str1 The first object to compare
    * @param str2 The second object to compare
    *
-   * @return A negative integer if str1 &lt; str2, 0 if str1 == str2, or
-   * a positive integer if str1 &gt; str2. 
+   * @return A negative integer if str1 < str2, 0 if str1 == str2, or
+   * a positive integer if str1 > str2. 
    */
   public abstract int compare (String source, String target);
 
@@ -167,8 +167,8 @@
    * @param obj1 The first object to compare
    * @param obj2 The second object to compare
    *
-   * @return A negative integer if obj1 &lt; obj2, 0 if obj1 == obj2, or
-   * a positive integer if obj1 &gt; obj2. 
+   * @return A negative integer if obj1 < obj2, 0 if obj1 == obj2, or
+   * a positive integer if obj1 > obj2. 
    *
    * @exception ClassCastException If the arguments are not instances
    * of <code>String</code>. 
@@ -380,12 +380,7 @@
   }
 
   // Decompose a single character and append results to the buffer.
-  // FIXME: for libgcj this is a native method which handles
-  // decomposition.  For Classpath, for now, it does nothing.
-  final void decomposeCharacter (char c, StringBuffer buf)
-  {
-    buf.append (c);
-  }
+  native final void decomposeCharacter (char c, StringBuffer buf);
 
   /**
    * This is the current collation decomposition setting.
