--- /home/tromey/gnu/egcs/classpath/classpath/java/lang/OutOfMemoryError.java	Sat Mar  2 12:13:12 2002
+++ java/lang/OutOfMemoryError.java	Tue Jan 22 16:01:35 2002
@@ -1,5 +1,5 @@
-/* OutOfMemoryError.java -- thrown when a memory allocation fails
-   Copyright (C) 1998, 1999, 2001, 2002 Free Software Foundation, Inc.
+/* OutOfMemoryError.java 
+   Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -38,36 +38,40 @@
 
 package java.lang;
 
+/* Written using "Java Class Libraries", 2nd edition, ISBN 0-201-31002-3
+ * "The Java Language Specification", ISBN 0-201-63451-1
+ * plus online API docs for JDK 1.2 beta from http://www.javasoft.com.
+ * Status:  Believed complete and correct. 
+ */
+
 /**
- * Thrown when the Java Virtual Machine is unable to allocate an object
- * because it is out of memory and no more memory could be made available
- * by the garbage collector.
+ * When the Java Virtual Machine is unable to allocate an object because it 
+ * is out of memory and no more memory could be made available by the 
+ * garbage collector an <code>OutOfMemoryError</code> is thrown.
  *
+ * @since JDK 1.0
+ * 
  * @author Brian Jones
  * @author Tom Tromey <tromey@cygnus.com>
- * @status updated to 1.4
+ * @date October 1, 1998
  */
 public class OutOfMemoryError extends VirtualMachineError
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = 8228564086184010517L;
+  static final long serialVersionUID = 8228564086184010517L;
 
   /**
    * Create an error without a message.
    */
   public OutOfMemoryError()
-  {
-  }
+    {
+      super();
+    }
 
   /**
    * Create an error with a message.
-   *
-   * @param s the message
    */
   public OutOfMemoryError(String s)
-  {
-    super(s);
-  }
+    {
+      super(s);
+    }
 }
