--- /home/tromey/gnu/classpath/classpath/java/awt/CardLayout.java	Wed Oct  2 11:21:37 2002
+++ java/awt/CardLayout.java	Tue Jan 22 15:40:04 2002
@@ -53,8 +53,6 @@
  */
 public class CardLayout implements LayoutManager2, Serializable
 {
-  static final long serialVersionUID = -4328196481005934313L;
-
   /**
    * Initializes a new instance of <code>CardLayout</code> with horizontal
    * and vertical gaps of 0.
