--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/SwingUtilities.java	2004-05-02 02:20:14.000000000 -0600
+++ javax/swing/SwingUtilities.java	2004-03-18 02:18:23.000000000 -0700
@@ -1,5 +1,5 @@
 /* SwingUtilities.java --
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -41,6 +41,7 @@
 import java.awt.Component;
 import java.awt.ComponentOrientation;
 import java.awt.Container;
+import java.awt.EventQueue;
 import java.awt.Font;
 import java.awt.FontMetrics;
 import java.awt.Graphics;
@@ -50,8 +51,11 @@
 import java.awt.Shape;
 import java.awt.Toolkit;
 import java.awt.Window;
+import java.awt.event.KeyEvent;
 import java.awt.event.MouseEvent;
 import java.lang.reflect.InvocationTargetException;
+import javax.accessibility.Accessible;
+import javax.accessibility.AccessibleStateSet;
 
 
 /**
@@ -59,7 +63,7 @@
  * useful when drawing swing components, dispatching events, or calculating
  * regions which need painting.
  *
- * @author Graydon Hoare (graydon@redhat.com)
+ * @author Graydon Hoare (graydon&064;redhat.com)
  */
 public class SwingUtilities implements SwingConstants
 {
