--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/BreakIterator.java	2003-05-10 02:20:42.000000000 -0600
+++ java/text/BreakIterator.java	2004-06-15 02:19:31.000000000 -0600
@@ -136,7 +136,7 @@
       {
 	ResourceBundle res
 	  = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation",
-				     loc);
+				     loc, BreakIterator.class.getClassLoader());
 	className = res.getString(type);
       }
     catch (MissingResourceException x)
