--- /home/tromey/gnu/egcs/classpath/classpath/java/util/zip/ZipException.java	Wed Sep 26 08:18:48 2001
+++ java/util/zip/ZipException.java	Fri Nov 17 14:42:28 2000
@@ -1,5 +1,5 @@
-/* java.util.zip.ZipException
-   Copyright (C) 2001 Free Software Foundation, Inc.
+/* ZipException.java - Exception representing a zip related error
+   Copyright (C) 1998, 1999, 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -25,26 +25,25 @@
 executable file might be covered by the GNU General Public License. */
 
 package java.util.zip;
-import java.io.IOException;
 
 /**
- * Is thrown during the creation or input of a zip file.
- *
- * @author Jochen Hoenicke
- * @since JDK 1.1
+ * @author Per Bothner
+ * @date January 9, 1999.
  */
 
+/* Written using on-line Java Platform 1.2 API Specification.
+ * Believed complete and correct.
+ */
 
-public class ZipException extends IOException 
+public class ZipException extends java.io.IOException
 {
-  public ZipException()
+  public ZipException ()
   {
-    this("");
+    super();
   }
 
-  public ZipException(String m)
+  public ZipException (String msg)
   {
-    super(m);
+    super(msg);
   }
 }
-
