--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicGraphicsUtils.java	2004-03-18 02:18:23.000000000 -0700
@@ -45,9 +45,11 @@
 import java.awt.Graphics2D;
 import java.awt.Insets;
 import java.awt.Rectangle;
+
 import java.awt.font.FontRenderContext;
 import java.awt.font.LineMetrics;
 import java.awt.font.TextLayout;
+
 import java.awt.geom.Rectangle2D;
 
 import javax.swing.AbstractButton;
