--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/CharArrayWriter.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/CharArrayWriter.java	2003-03-24 02:17:53.000000000 -0700
@@ -50,8 +50,8 @@
   * following two properties:
   * <p>
   * <ul>
-  * <li><xmp>gnu.java.io.CharArrayWriter.initialBufferSize</xmp></li>
-  * <li><xmp>gnu.java.io.CharArrayWriter.bufferIncrementSize</xmp></li>
+  * <li><xmp>gnu.java.io.CharArrayWriter.initialBufferSize</xmp>
+  * <li><xmp>gnu.java.io.CharArrayWriter.bufferIncrementSize</xmp>
   * </ul>
   * <p>
   * There is a constructor that specified the initial buffer size and
