--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/ObjectOutputStream.java	2004-07-24 02:24:53.000000000 -0600
+++ java/io/ObjectOutputStream.java	2004-07-24 02:18:38.000000000 -0600
@@ -334,6 +334,8 @@
 
 	    if (obj instanceof Serializable)
 	      {
+		Object prevObject = this.currentObject;
+		ObjectStreamClass prevObjectStreamClass = this.currentObjectStreamClass;
 		currentObject = obj;
 		ObjectStreamClass[] hierarchy =
 		  ObjectStreamClass.getObjectStreamClasses(clazz);
@@ -362,8 +364,8 @@
 		  }
 		  }
 
-		currentObject = null;
-		currentObjectStreamClass = null;
+		this.currentObject = prevObject;
+		this.currentObjectStreamClass = prevObjectStreamClass;
 		currentPutField = null;
 		break;
 	      }
