--- /home/tromey/gnu/egcs/classpath/classpath/java/security/ProtectionDomain.java	Mon Mar 25 00:05:18 2002
+++ java/security/ProtectionDomain.java	Tue Jan 22 16:01:36 2002
@@ -132,14 +132,12 @@
   public String toString()
   {
     StringBuffer sb = new StringBuffer("");
-    sb.append("ProtectionDomain (" + linesep);
-    if (code_source == null)
-      sb.append("CodeSource:null" + linesep);
-    else
-      sb.append(code_source + linesep);
-    sb.append(perms);
-    sb.append(linesep + ")" + linesep);
-    
+
+    sb.append(super.toString() + " (" + linesep);
+    sb.append(code_source.toString());
+    sb.append(perms.toString());
+    sb.append(")" + linesep);
+
     return sb.toString();
   }
 }
