--- /home/tromey/gnu/classpath/classpath/java/awt/PrintGraphics.java	Wed Mar 20 17:01:39 2002
+++ java/awt/PrintGraphics.java	Wed Jan 23 18:57:56 2002
@@ -1,5 +1,5 @@
-/* PrintGraphics.java -- a print graphics context
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+/* PrintGraphics.java -- A print graphics context.
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,19 +39,21 @@
 package java.awt;
 
 /**
- * This interface allows the originating print job to be obtained.
- *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @since 1.0
- * @status updated to 1.4
- */
+  * This interface allows the originating print job to be obtained.
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  */
 public interface PrintGraphics
 {
-  /**
-   * Returns the <code>PrintJob</code> that this object is being
-   * managed by.
-   *
-   * @return the print job for this object
-   */
-  PrintJob getPrintJob();
+
+/**
+  * Returns the <code>PrintJob</code> that this object is being
+  * managed by.
+  *
+  * @return The print job for this object.
+  */
+public abstract PrintJob
+getPrintJob();
+
 } // interface PrintGraphics
+
