--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Container.java	2003-07-02 02:19:06.000000000 -0600
+++ java/awt/Container.java	2003-06-13 02:27:29.000000000 -0600
@@ -201,14 +201,12 @@
   /**
    * Adds the specified component to the container at the end of the
    * component list.  This method should not be used. Instead, use
-   * <code>add(Component, Object)</code>.
+   * <code>add(Component, Object</code>.
    *
    * @param name The name of the component to be added.
    * @param component The component to be added.
    *
    * @return The same component that was added.
-   *
-   * @see #add(Component,Object)
    */
   public Component add(String name, Component comp)
   {
