--- /home/tromey/gnu/egcs/classpath/classpath/java/io/PipedInputStream.java	Sun Jan  7 16:06:09 2001
+++ java/io/PipedInputStream.java	Fri Nov  2 21:44:51 2001
@@ -194,7 +194,7 @@
 	else
 	  copylen = Math.min (len, buffer.length - in);
 
-	// Copy bytes until the pipe is filled, wrapping if neccessary.
+	// Copy bytes until the pipe is filled, wrapping if necessary.
 	System.arraycopy(buf, bufpos, buffer, in, copylen);
 	len -= copylen;
 	bufpos += copylen;
