--- /home/tromey/gnu/classpath/classpath/java/nio/channels/ServerSocketChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/ServerSocketChannel.java	Tue Dec 10 19:20:33 2002
@@ -73,7 +73,7 @@
    * @exception SecurityException If a security manager has been installed and
    * it does not permit access to the remote endpoint of the new connection.
    */
-  public abstract SocketChannel accept () throws IOException;
+  public abstract SocketChannel accept ();
   
   /**
    * Retrieves the channels socket.
