--- /home/tromey/gnu/Nightly/classpath/classpath/java/applet/AudioClip.java	2004-09-22 02:30:00.000000000 -0600
+++ java/applet/AudioClip.java	2002-12-10 19:20:24.000000000 -0700
@@ -1,5 +1,5 @@
 /* AudioClip.java -- play an audio clip in an applet
-   Copyright (C) 1999, 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -38,7 +38,6 @@
 
 package java.applet;
 
-
 /**
  * This interface provides a simple mechanism for playing audio clips.
  * If multiple clips are played at once, the browser combines them into a
