--- /home/tromey/gnu/classpath/classpath/java/nio/channels/SocketChannel.java	Sun Nov 24 16:32:34 2002
+++ java/nio/channels/SocketChannel.java	Wed Nov 13 09:30:36 2002
@@ -60,8 +60,6 @@
  
   /**
    * Opens a socket channel.
-   *
-   * @exception IOException If an error occurs
    */
   public static SocketChannel open () throws IOException
   {
@@ -70,8 +68,6 @@
   
   /**
    * Opens a channel and connects it to a remote address.
-   *
-   * @exception IOException If an error occurs
    */
   public static SocketChannel open (SocketAddress remote) throws IOException
   {
@@ -86,8 +82,6 @@
     
   /**
    * Reads data from the channel.
-   *
-   * @exception IOException If an error occurs
    */
   public final long read (ByteBuffer[] dsts)
   {
@@ -103,8 +97,6 @@
     
   /**
    * Writes data to the channel.
-   *
-   * @exception IOException If an error occurs
    */
   public final long write (ByteBuffer[] dsts)
   {
@@ -128,22 +120,16 @@
 
   /**
    * Reads data from the channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract int read (ByteBuffer dst);
 
   /**
    * Connects the channel's socket to the remote address.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract boolean connect (SocketAddress remote) throws IOException;
   
   /**
    * Finishes the process of connecting a socket channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract boolean finishConnect ();
  
@@ -159,8 +145,6 @@
   
   /**
    * Reads data from the channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract long read (ByteBuffer[] dsts, int offset, int length);
  
@@ -171,15 +155,11 @@
   
   /**
    * Writes data to the channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract int write (ByteBuffer src);
   
   /**
    * Writes data to the channel.
-   *
-   * @exception IOException If an error occurs
    */
   public abstract long write (ByteBuffer[] srcs, int offset, int length);
 }
