--- /home/tromey/gnu/classpath/classpath/javax/naming/BinaryRefAddr.java	Wed Oct  2 11:21:39 2002
+++ javax/naming/BinaryRefAddr.java	Mon Oct  7 09:09:22 2002
@@ -51,7 +51,7 @@
 public class BinaryRefAddr extends RefAddr
 {
   static final long serialVersionUID = -3415254970957330361L;
-	
+
   /**
    * The possibly null content of this RefAddr.
    * Set by the constructor and returned by getContent.
