--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PushbackInputStream.java	2004-04-18 02:22:15.000000000 -0600
+++ java/io/PushbackInputStream.java	2003-06-13 02:27:34.000000000 -0600
@@ -263,7 +263,7 @@
   /**
    * This method pushed back bytes from the passed in array into the
    * pushback buffer.  The bytes from <code>b[offset]</code> to
-   * <code>b[offset + len]</code> are pushed in reverse order so that
+   * <cdoe>b[offset + len]</code> are pushed in reverse order so that
    * the next byte read from the stream after this operation will be
    * <code>b[offset]</code> followed by <code>b[offset + 1]</code>,
    * etc.
