--- /home/tromey/gnu/egcs/classpath/classpath/java/net/SocketException.java	Sun Apr 15 11:57:25 2001
+++ java/net/SocketException.java	Fri Nov  2 21:44:51 2001
@@ -1,4 +1,4 @@
-/* SocketException.java -- An exception occured while performing a socket op
+/* SocketException.java -- An exception occurred while performing a socket op
    Copyright (C) 1998, 1999, 2001 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -31,7 +31,7 @@
  */
 
 /**
-  * This exception indicates that a generic error occured related to an
+  * This exception indicates that a generic error occurred related to an
   * operation on a socket.  Check the descriptive message (if any) for
   * details on the nature of this error
   *
