--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JTextField.java	2004-04-29 02:24:17.000000000 -0600
+++ javax/swing/JTextField.java	2003-06-13 02:27:43.000000000 -0600
@@ -1,5 +1,5 @@
 /* JTextField.java -- 
-   Copyright (C) 2002, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -40,8 +40,9 @@
 
 import java.awt.event.ActionListener;
 import java.util.Vector;
-
 import javax.accessibility.AccessibleStateSet;
+import javax.swing.text.Document;
+import javax.swing.text.JTextComponent;
 
 public class JTextField extends JEditorPane
 {
