--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JRadioButtonMenuItem.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JRadioButtonMenuItem.java	2004-06-18 02:18:53.000000000 -0600
@@ -49,6 +49,8 @@
  */
 public class JRadioButtonMenuItem extends JMenuItem implements Accessible
 {
+  private static final long serialVersionUID = 8482658191548521743L;
+
   private static final String uiClassID = "RadioButtonMenuItemUI";
 
   /**
@@ -185,7 +187,7 @@
   public AccessibleContext getAccessibleContext()
   {
     if (accessibleContext == null)
-      accessibleContext = new AccessibleJRadioButtonMenuItem(this);
+      accessibleContext = new AccessibleJRadioButtonMenuItem();
 
     return accessibleContext;
   }
@@ -195,16 +197,13 @@
    */
   protected class AccessibleJRadioButtonMenuItem extends AccessibleJMenuItem
   {
+    private static final long serialVersionUID = 4381471510145292179L;
+
     /**
      * Creates a new AccessibleJRadioButtonMenuItem object.
-     *
-     * @param component DOCUMENT ME!
      */
-    protected AccessibleJRadioButtonMenuItem(JRadioButtonMenuItem component)
+    protected AccessibleJRadioButtonMenuItem()
     {
-      super(component);
-
-      // TODO
     }
 
     /**
