--- /home/tromey/gnu/egcs/classpath/classpath/java/security/cert/CertificateExpiredException.java	Sat Mar  2 12:13:21 2002
+++ java/security/cert/CertificateExpiredException.java	Tue Jan 22 16:01:37 2002
@@ -1,5 +1,5 @@
 /* CertificateExpiredException.java --- Certificate Expired Exception
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -39,33 +39,31 @@
 package java.security.cert;
 
 /**
- * Exception for a Certificate Expiring.
- *
- * @author Mark Benvenuto
- * @since 1.2
- * @status updated to 1.4
- */
+   Exception for a Certificate Expiring.
+   
+   @since JDK 1.2
+   
+   @author Mark Benvenuto
+*/
 public class CertificateExpiredException extends CertificateException
 {
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = 9071001339691533771L;
 
   /**
-   * Constructs an exception without a message string.
-   */
+     Constructs an CertificateExpiredException without a message string.
+  */
   public CertificateExpiredException()
   {
+    super();
   }
 
   /**
-   * Constructs an exception with a message string.
-   *
-   * @param msg a message to display with exception
-   */
+     Constructs an CertificateExpiredException with a message string.
+
+     @param msg A message to display with exception
+  */
   public CertificateExpiredException(String msg)
   {
-    super(msg);
+    super( msg );
   }
+
 }
