--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/HttpURLConnection.java	2004-10-12 02:18:13.000000000 -0600
+++ java/net/HttpURLConnection.java	2004-04-21 02:18:09.000000000 -0600
@@ -291,7 +291,7 @@
   /**
    * This is a list of valid request methods, separated by "|" characters.
    */
-  private static final String valid_methods =
+  private static String valid_methods =
     "|GET|POST|HEAD|OPTIONS|PUT|DELETE|TRACE|";
 
   // Instance Variables
