--- /home/tromey/gnu/egcs/classpath/classpath/java/security/cert/X509Extension.java	Thu Jul 19 11:38:28 2001
+++ java/security/cert/X509Extension.java	Sat Aug 19 12:19:41 2000
@@ -29,7 +29,7 @@
 import java.util.Set;
 
 /**
-	Public interface for the X.509 Extension.
+	Public abstract interface for the X.509 Extension.
 
 	This is used for X.509 v3 Certificates and CRL v2 (Certificate
 	Revocation Lists) for managing attributes assoicated with
@@ -61,7 +61,7 @@
 
 	@since JDK 1.2
 */
-public interface X509Extension
+public abstract interface X509Extension
 {
 
   /**
