--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JWindow.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JWindow.java	2004-06-18 02:18:54.000000000 -0600
@@ -60,6 +60,8 @@
  */
 public class JWindow extends Window implements Accessible, RootPaneContainer
 {
+  private static final long serialVersionUID = 5420698392125238833L;
+  
     public final static int HIDE_ON_CLOSE        = 0;
     public final static int EXIT_ON_CLOSE        = 1;
     public final static int DISPOSE_ON_CLOSE     = 2;
@@ -114,13 +116,6 @@
     return d;
   }
 
-    JMenuBar getJMenuBar()
-    {    return getRootPane().getJMenuBar();   }
-    
-    void setJMenuBar(JMenuBar menubar)
-    {    getRootPane().setJMenuBar(menubar); }
-    
-
   public  void setLayout(LayoutManager manager)
   {    super.setLayout(manager);  }
 
