--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormatSymbols.java	2004-06-21 06:24:01.000000000 -0600
+++ java/text/DateFormatSymbols.java	2004-06-15 02:19:32.000000000 -0600
@@ -99,7 +99,7 @@
   {
     ResourceBundle res
       = ResourceBundle.getBundle("gnu.java.locale.LocaleInformation", locale,
-      				 ClassLoader.getSystemClassLoader());
+      				 getClass().getClassLoader());
 
     ampms = res.getStringArray ("ampms");
     eras = res.getStringArray ("eras");
