--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Short.java	2004-04-13 02:20:22.000000000 -0600
+++ java/lang/Short.java	2002-06-15 13:42:20.000000000 -0600
@@ -60,13 +60,13 @@
 
   /**
    * The minimum value a <code>short</code> can represent is -32768 (or
-   * -2<sup>15</sup>).
+   * -2<sup>15</sup).
    */
   public static final short MIN_VALUE = -32768;
 
   /**
    * The minimum value a <code>short</code> can represent is 32767 (or
-   * 2<sup>15</sup>).
+   * 2<sup>15</sup).
    */
   public static final short MAX_VALUE = 32767;
 
@@ -144,7 +144,7 @@
    *
    * @param s the <code>String</code> to convert
    * @param radix the radix (base) to use in the conversion
-   * @return the <code>String</code> argument converted to <code>short</code>
+   * @return the <code>String</code> argument converted to </code>short</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as a
    *         <code>short</code>
    */
