--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/PublicKey.java	2003-05-07 02:23:22.000000000 -0600
+++ java/security/PublicKey.java	2003-05-10 02:19:17.000000000 -0600
@@ -56,5 +56,5 @@
   /**
    * The version identifier used for serialization.
    */
-  long serialVersionUID = 7187392471159151072L;
+  public static final long serialVersionUID = 7187392471159151072L;
 } // interface PublicKey
