--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/UnknownHostException.java	2004-04-13 02:20:24.000000000 -0600
+++ java/net/UnknownHostException.java	2002-06-15 13:46:56.000000000 -0600
@@ -39,7 +39,6 @@
 
 import java.io.IOException;
 
-
 /**
  * This exception indicates that an attempt was made to reference a hostname
  * or IP address that is not valid.  This could possibly indicate that a
