--- /home/tromey/gnu/classpath/classpath/java/util/jar/JarFile.java	Sun Nov 24 16:32:34 2002
+++ java/util/jar/JarFile.java	Tue Jan 22 15:40:41 2002
@@ -301,7 +301,7 @@
    * Returns the manifest for this JarFile or null when the JarFile does not
    * contain a manifest file.
    */
-  public Manifest getManifest() throws IOException
+  public Manifest getManifest()
   {
     return manifest;
   }
