--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/event/InternalFrameEvent.java	2003-07-02 02:19:17.000000000 -0600
+++ javax/swing/event/InternalFrameEvent.java	2003-06-28 02:18:23.000000000 -0600
@@ -81,7 +81,7 @@
   /**
    * Internal frame iconified event
    */
-  public static final int INTERNAL_FRAME_ICONIFIED = 25552;
+  public static final int INTERNAL_FRAME_ICONIFIED = 2552;
 
   /**
    * Internal frame last event
@@ -91,7 +91,7 @@
   /**
    * Internal frame opened event
    */
-  public static final int INTERNAL_FRAME_OPENED = 25549;
+  public static final int INTERNAL_FRAME_OPENED = 25550;
 
   /**
    * Creates a <code>JInternalFrameEvent</code> object.
