--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Long.java	2004-04-18 02:22:17.000000000 -0600
+++ java/lang/Long.java	2002-06-13 11:39:51.000000000 -0600
@@ -223,9 +223,9 @@
    * 'L' as the last character is only valid in radices 22 or greater, where
    * it is a digit and not a type indicator.
    *
-   * @param str the <code>String</code> to convert
+   * @param s the <code>String</code> to convert
    * @param radix the radix (base) to use in the conversion
-   * @return the <code>String</code> argument converted to <code>long</code>
+   * @return the <code>String</code> argument converted to </code>long</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as a
    *         <code>long</code>
    */
@@ -306,7 +306,7 @@
    * <code>MAX_VALUE</code>, or an exception is thrown. Note that you cannot
    * use a trailing 'l' or 'L', unlike in Java source code.
    *
-   * @param str the <code>String</code> to interpret
+   * @param s the <code>String</code> to interpret
    * @return the value of the String as a <code>Long</code>
    * @throws NumberFormatException if <code>s</code> cannot be parsed as a
    *         <code>long</code>
@@ -455,7 +455,7 @@
    * interpret the value of the property.
    *
    * @param nm the name of the system property
-   * @param def the default value
+   * @param val the default value
    * @return the value of the system property, or the default
    * @throws SecurityException if accessing the system property is forbidden
    * @see System#getProperty(String)
