--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/CharBuffer.java	2003-03-11 10:27:52.000000000 -0700
+++ java/nio/CharBuffer.java	2003-05-02 02:20:07.000000000 -0600
@@ -99,9 +99,7 @@
         buffer [i] = a.charAt (i);
       }
     
-    // FIXME: make it read-only.
-    //return wrap (buffer, offset, length);
-    return wrap (buffer, offset, length).asReadOnlyBuffer ();
+    return wrap (buffer, offset, length);
   }
   
   /**
@@ -112,7 +110,7 @@
     return wrap (array, 0, array.length);
   }
  
-  protected CharBuffer (int cap, int lim, int pos, int mark)
+  CharBuffer (int cap, int lim, int pos, int mark)
   {
     super (cap, lim, pos, mark);
   }
@@ -363,9 +361,7 @@
    */
   public String toString ()
   {
-    char[] data = new char [length ()];
-    get (data, 0, length ());
-    return new String (data, position (), length ());
+    return new String (array (), position (), length ());
   }
 
   /**
