--- /home/tromey/gnu/classpath/classpath/java/net/SocketAddress.java	Sun Nov 17 14:22:21 2002
+++ java/net/SocketAddress.java	Wed Sep 25 13:48:13 2002
@@ -37,21 +37,20 @@
 
 package java.net;
 
-import java.io.Serializable;
+import java.io.*;
 
 /** 
  * Abstract base class for InetSocketAddress.
  * InetSocketAddress is to my knowledge the only derived
  * class. [Ronald]
- *
- * @since 1.4
  */
+
 public abstract class SocketAddress implements Serializable
 {
-  /**
-   * Initializes the socket address.
-   */
+  static final long serialVersionUID = 5215720748342549866L;
+
   public SocketAddress()
   {
   }
 }
+
