--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/SerializablePermission.java	2004-04-13 02:20:21.000000000 -0600
+++ java/io/SerializablePermission.java	2003-03-24 02:17:54.000000000 -0700
@@ -48,9 +48,9 @@
   * There are currently two allowable permission names for this class:
   * <ul>
   * <li><code>enableSubclassImplementation</code> - Allows a subclass to
-  * override the default serialization behavior of objects.</li>
+  * override the default serialization behavior of objects.
   * <li><code>enableSubstitution</code> - Allows substitution of one object
-  * for another during serialization or deserialization.</li>
+  * for another during serialization or deserialization.
   * </ul>
   *
   * @see java.security.BasicPermission
