--- /home/tromey/gnu/classpath/classpath/java/net/ServerSocket.java	Wed Nov 13 10:15:52 2002
+++ java/net/ServerSocket.java	Thu Oct  3 11:50:56 2002
@@ -366,6 +366,17 @@
   }
 
   /**
+   * Returns true if the socket is closed, otherwise false
+   * 
+   * @since 1.4
+   */
+  public boolean isClosed()
+  {
+    // FIXME: implement this
+    return false;
+  }
+
+  /**
    * Sets the value of SO_TIMEOUT.  A value of 0 implies that SO_TIMEOUT is
    * disabled (ie, operations never time out).  This is the number of 
    * milliseconds a socket operation can block before an
