--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/IdentityHashMap.java	2002-04-30 16:06:46.000000000 -0600
+++ java/util/IdentityHashMap.java	2003-12-10 02:18:06.000000000 -0700
@@ -38,7 +38,10 @@
 
 package java.util;
 
-import java.io.*;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
 
 /**
  * This class provides a hashtable-backed implementation of the
