--- /home/tromey/gnu/classpath/classpath/java/io/ObjectStreamConstants.java	Sat Nov  9 16:09:31 2002
+++ java/io/ObjectStreamConstants.java	Sat Jun 15 12:59:15 2002
@@ -70,7 +70,7 @@
   final static byte TC_PROXYCLASSDESC = (byte)125;  //0x7D
 
   final static byte TC_BASE = TC_NULL;
-  final static byte TC_MAX = TC_PROXYCLASSDESC;
+  final static byte TC_MAX = TC_EXCEPTION;
 
   final static int baseWireHandle = 0x7e0000;
 
