--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/prefs/PreferencesFactory.java	2002-01-22 15:27:02.000000000 -0700
+++ java/util/prefs/PreferencesFactory.java	2003-06-18 02:22:33.000000000 -0600
@@ -35,9 +35,8 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-package java.util.prefs;
 
-import java.util.EventListener;
+package java.util.prefs;
 
 /**
  * Preferences system and user root factory interface. Defines how to get
