--- /home/tromey/gnu/classpath/classpath/java/util/jar/Attributes.java	Mon Nov  4 20:06:02 2002
+++ java/util/jar/Attributes.java	Tue Jun 18 09:40:12 2002
@@ -1,5 +1,5 @@
 /* Attributes.java -- Represents attribute name/value pairs from a Manifest
-   Copyright (C) 2000, 2002 Free Software Foundation, Inc.
+   Copyright (C) 2000 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -285,7 +285,6 @@
       // all other characters must be alphanums, '-' or '_'
       for (int i = 1; i < chars.length; i++)
 	{
-	  c = chars[i];
 	  if (!((c >= 'a' && c <= 'z') ||
 		(c >= 'A' && c <= 'Z') ||
 		(c >= '0' && c <= '9') || (c == '-') || (c == '_')))
