--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Toolkit.java	2003-09-12 02:19:50.000000000 -0600
+++ java/awt/Toolkit.java	2003-09-12 02:17:59.000000000 -0600
@@ -96,7 +96,7 @@
 {
   /** The default toolkit name. */
   private static String default_toolkit_name
-    = "gnu.java.awt.peer.gtk.GtkToolkit";
+    = gnu.classpath.Configuration.default_awt_peer_toolkit;
 
   /**
    * The toolkit in use.  Once we load it, we don't ever change it
@@ -807,7 +807,7 @@
   /**
    * Returns the event queue that is suitable for the calling context.
    *
-   * <p>Despite the word &#x201c;System&#x201d; the name of this
+   * <p>Despite the word “System” in the name of this
    * method, a toolkit may provide different event queues for each
    * applet. There is no guarantee that the same queue is shared
    * system-wide.
@@ -842,7 +842,7 @@
   /**
    * Returns the event queue that is suitable for the calling context.
    *
-   * <p>Despite the word &#x201c;System&#x201d; in the name of this
+   * <p>Despite the word “,System”, in the name of this
    * method, a toolkit may provide different event queues for each
    * applet. There is no guarantee that the same queue is shared
    * system-wide.
