--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RMIClassLoader.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/server/RMIClassLoader.java	2003-11-12 02:18:14.000000000 -0700
@@ -1,6 +1,6 @@
-/* RMIClassLoader.java --
-   Copyright (c) 1996, 1997, 1998, 1999, 2002, 2003
-   Free Software Foundation, Inc.
+/* RMIClassLoader.java
+  Copyright (c) 1996, 1997, 1998, 1999, 2002, 2003
+  Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
