--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicLabelUI.java	2004-04-29 02:24:18.000000000 -0600
+++ javax/swing/plaf/basic/BasicLabelUI.java	2004-05-27 02:19:07.000000000 -0600
@@ -168,23 +168,20 @@
       vr.width = 0;
     if (vr.height < 0)
       vr.height = 0;
-
+      
     Icon icon = (b.isEnabled()) ? b.getIcon() : b.getDisabledIcon();
 
     String text = layoutCL(b, fm, b.getText(), icon, vr, ir, tr);
     
-    if (b.isOpaque())
-    {
-      g.setColor(b.getBackground());
-      g.fillRect(vr.x, vr.y, vr.width, vr.height);
-    }
-
     if (icon != null)
       icon.paintIcon(b, g, ir.x, ir.y);
-    if (b.isEnabled())
-      paintEnabledText(b, g, text, tr.x, tr.y + fm.getAscent());
-    else
-      paintDisabledText(b, g, text, tr.x, tr.y + fm.getAscent());
+    if (text != null && ! text.equals(""))
+    {
+      if (b.isEnabled())
+        paintEnabledText(b, g, text, tr.x, tr.y + fm.getAscent());
+      else
+        paintDisabledText(b, g, text, tr.x, tr.y + fm.getAscent());
+    }
     g.setFont(saved_font);
   }
 
@@ -354,6 +351,7 @@
     c.setBackground(defaults.getColor("Label.background"));
     c.setFont(defaults.getFont("Label.font"));
     c.setBorder(defaults.getBorder("Label.border"));
+    c.setOpaque(true);
     //XXX: There are properties we don't use called disabledForeground
     //and disabledShadow.
   }
