--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/FileDescriptor.java	2004-10-03 02:35:36.000000000 -0600
+++ java/io/FileDescriptor.java	2004-04-21 02:18:08.000000000 -0600
@@ -39,10 +39,9 @@
 
 package java.io;
 
-import gnu.java.nio.channels.FileChannelImpl;
-
 import java.nio.channels.ByteChannel;
 import java.nio.channels.FileChannel;
+import gnu.java.nio.channels.FileChannelImpl;
 
 /**
  * This class represents an opaque file handle as a Java class.  It should
