--- /home/tromey/gnu/egcs/classpath/classpath/java/rmi/activation/UnknownObjectException.java	Sat Mar  2 12:13:15 2002
+++ java/rmi/activation/UnknownObjectException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* UnknownObjectException.java -- thrown on an invalid ActivationID
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -37,33 +37,11 @@
 
 package java.rmi.activation;
 
-/**
- * Thrown when an <code>ActivationID</code> parameter is invalid or unknown.
- *
- * @author unknown
- * @see Activatable
- * @see ActivationGroup
- * @see ActivationID
- * @see ActivationMonitor
- * @see ActivationSystem
- * @see Activator
- * @since 1.2
- * @status updated to 1.4
- */
-public class UnknownObjectException extends ActivationException
-{
-  /**
-   * Compatible with JDK 1.2+.
-   */
-  private static final long serialVersionUID = 3425547551622251430L;
-
-  /**
-   * Create an exception with an error message.
-   *
-   * @param s the message
-   */
-  public UnknownObjectException(String s)
-  {
-    super(s);
-  }
+public class UnknownObjectException
+	extends ActivationException {
+
+public UnknownObjectException(String s) {
+	super(s);
+}
+
 }
