--- /home/tromey/gnu/classpath/classpath/java/lang/reflect/Proxy.java	Sat Mar 23 09:42:57 2002
+++ java/lang/reflect/Proxy.java	Sun Sep 29 23:19:09 2002
@@ -790,7 +790,7 @@
                 in_package = true;
                 data.pack = inter.getPackage();
               }
-          for (int j = i; j >= 0; j--)
+          for (int j = i-1; j >= 0; j--)
             if (data.interfaces[j] == inter)
               throw new IllegalArgumentException("duplicate interface: "
                                                  + inter);
