--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/PipedInputStream.java	2004-10-03 02:35:37.000000000 -0600
+++ java/io/PipedInputStream.java	2004-04-21 02:18:08.000000000 -0600
@@ -35,12 +35,12 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-package java.io;
-
 // NOTE: This implementation is very similar to that of PipedReader.  If you 
 // fix a bug in here, chances are you should make a similar change to the 
 // PipedReader code.
 
+package java.io;
+
 /**
   * An input stream that reads its bytes from an output stream
   * to which it is connected. 
