--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Policy.java	2004-10-12 02:18:14.000000000 -0600
+++ java/security/Policy.java	2003-04-20 02:20:19.000000000 -0600
@@ -95,7 +95,7 @@
  */
 public abstract class Policy
 {
-  static private Policy currentPolicy;
+  static private Policy currentPolicy = null;
 
   /** Map of ProtectionDomains to PermissionCollections for this instance. */
   private Map pd2pc = null;
