--- /home/tromey/gnu/egcs/classpath/classpath/java/text/ChoiceFormat.java	Tue Apr 30 11:50:10 2002
+++ java/text/ChoiceFormat.java	Tue Jan 22 16:01:37 2002
@@ -52,10 +52,10 @@
  * This sounds complicated, but that is because I did a poor job of
  * explaining it.  Consider the following example:
  * <p>
- *
-<pre>terminators = { 1, ChoiceFormat.nextDouble(1) }
-formats = { "file", "files" }</pre>
- *
+ * <pre>
+ * terminators = { 1, ChoiceFormat.nextDouble(1) }
+ * formats = { "file", "files" }
+ * </pre>
  * <p>
  * In this case if the actual number tested is one or less, then the word
  * "file" is used as the format value.  If the number tested is greater than
