--- /home/tromey/gnu/egcs/classpath/classpath/java/security/cert/X509Certificate.java	Wed Mar 22 12:26:26 2000
+++ java/security/cert/X509Certificate.java	Fri Nov  2 21:44:51 2001
@@ -288,7 +288,7 @@
 
      @return the DER encoded tbsCertificate
 
-     @throws CertificateEncodingException if encoding error occured
+     @throws CertificateEncodingException if encoding error occurred
   */
   public abstract byte[] getTBSCertificate() throws CertificateEncodingException;
 
