--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/UnknownServiceException.java	2004-04-13 02:20:24.000000000 -0600
+++ java/net/UnknownServiceException.java	2002-06-15 13:47:07.000000000 -0600
@@ -39,7 +39,6 @@
 
 import java.io.IOException;
 
-
 /**
  * Contrary to what you might think, this does not indicate that the
  * TCP/IP service name specified was invalid.  Instead it indicates that
