--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/Format.java	2004-04-18 02:22:25.000000000 -0600
+++ java/text/Format.java	2003-12-12 02:18:27.000000000 -0700
@@ -123,7 +123,7 @@
    * This method parses a <code>String</code> and converts the parsed 
    * contents into an <code>Object</code>.
    *
-   * @param str The <code>String</code> to parse.
+   * @param str The <code>String to parse.
    *
    * @return The resulting <code>Object</code>.
    *
