--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/Pipe.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/channels/Pipe.java	2003-02-11 09:41:07.000000000 -0700
@@ -1,4 +1,4 @@
-/* Pipe.java --
+/* Pipe.java -- 
    Copyright (C) 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -41,22 +41,22 @@
 import java.nio.channels.spi.AbstractSelectableChannel;
 import java.nio.channels.spi.SelectorProvider;
 
-
 /**
  * @author Michael Koch
  * @since 1.4
  */
 public abstract class Pipe
 {
-  public abstract static class SinkChannel extends AbstractSelectableChannel
+  public abstract static class SinkChannel
+    extends AbstractSelectableChannel
     implements WritableByteChannel, GatheringByteChannel
   {
     /**
      * Initializes the channel.
      */
-    protected SinkChannel(SelectorProvider provider)
+    protected SinkChannel (SelectorProvider provider)
     {
-      super(provider);
+      super (provider);
     }
 
     /**
@@ -64,21 +64,22 @@
      *
      * The only valid operation on this channel is @see SelectionKey.OP_WRITE.
      */
-    public final int validOps()
+    public final int validOps ()
     {
       return SelectionKey.OP_WRITE;
     }
   }
 
-  public abstract static class SourceChannel extends AbstractSelectableChannel
+  public abstract static class SourceChannel
+    extends AbstractSelectableChannel
     implements ReadableByteChannel, ScatteringByteChannel
   {
     /**
      * Initializes the channel.
      */
-    protected SourceChannel(SelectorProvider provider)
+    protected SourceChannel (SelectorProvider provider)
     {
-      super(provider);
+      super (provider);
     }
 
     /**
@@ -86,7 +87,7 @@
      *
      * The only valid operation on this channel is @see SelectionKey.OP_READ.
      */
-    public final int validOps()
+    public final int validOps ()
     {
       return SelectionKey.OP_READ;
     }
@@ -101,12 +102,12 @@
 
   /**
    * Opens a pipe.
-   *
+   * 
    * @exception IOException If an error occurs
    */
   public static Pipe open() throws IOException
   {
-    return SelectorProvider.provider().openPipe();
+    return SelectorProvider.provider ().openPipe();
   }
 
   /**
@@ -117,5 +118,5 @@
   /**
    * Returns a pipe's source channel
    */
-  public abstract Pipe.SourceChannel source();
+  public abstract Pipe.SourceChannel source();   
 }
