--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/JScrollPane.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/JScrollPane.java	2004-06-18 02:18:53.000000000 -0600
@@ -79,20 +79,23 @@
   extends JComponent 
   implements Accessible, ScrollPaneConstants
 {
-  JViewport columnHeader;
-  JViewport rowHeader;
+  private static final long serialVersionUID = 5203525440012340014L;
+  
+  protected JViewport columnHeader;
+  protected JViewport rowHeader;
 
-  Component lowerLeft;
-  Component lowerRight;
-  Component upperLeft;
-  Component upperRight;
-
-  JScrollBar horizontalScrollBar;
-  int horizontalScrollBarPolicy;
-  JScrollBar verticalScrollBar;
-  int verticalScrollBarPolicy;
+  protected Component lowerLeft;
+  protected Component lowerRight;
+  protected Component upperLeft;
+  protected Component upperRight;
+
+  protected JScrollBar horizontalScrollBar;
+  protected int horizontalScrollBarPolicy;
+  protected JScrollBar verticalScrollBar;
+  protected int verticalScrollBarPolicy;
 
-  JViewport viewport;
+  protected JViewport viewport;
+  
   Border viewportBorder;
   boolean wheelScrollingEnabled;
   ChangeListener scrollListener;  
