--- /home/tromey/gnu/egcs/classpath/classpath/java/rmi/server/LogStream.java	Wed Mar 20 17:01:41 2002
+++ java/rmi/server/LogStream.java	Tue Jan 22 16:01:36 2002
@@ -45,8 +45,8 @@
 	extends PrintStream {
 
 public static final int SILENT = 0;
-public static final int BRIEF = 10;
-public static final int VERBOSE = 20;
+public static final int BRIEF = 1;
+public static final int VERBOSE = 2;
 
 private static PrintStream defStream;
 
