--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipEntry.java	2003-02-22 02:18:57.000000000 -0700
+++ java/util/zip/ZipEntry.java	2003-06-18 02:22:33.000000000 -0600
@@ -35,9 +35,10 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package java.util.zip;
+
 import java.util.Calendar;
-import java.util.TimeZone;
 import java.util.Date;
 
 /**
