--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/datatransfer/DataFlavor.java	2004-04-18 02:22:13.000000000 -0600
+++ java/awt/datatransfer/DataFlavor.java	2004-02-06 02:18:31.000000000 -0700
@@ -270,7 +270,7 @@
 /**
  * Initializes a new instance of <code>DataFlavor</code> with the
  * specified MIME type and description.  If the MIME type has a
- * "class=&lt;rep class&gt;" parameter then the representation class will
+ * "class=<rep class>" parameter then the representation class will
  * be the class name specified. Otherwise the class defaults to
  * <code>java.io.InputStream</code>. If the human readable name
  * is not specified (<code>null</code>) then the human readable name
@@ -319,7 +319,7 @@
 /**
  * Initializes a new instance of <code>DataFlavor</code> with the
  * specified MIME type and description.  If the MIME type has a
- * "class=&lt;rep class&gt;" parameter then the representation class will
+ * "class=<rep class>" parameter then the representation class will
  * be the class name specified. Otherwise the class defaults to
  * <code>java.io.InputStream</code>. If the human readable name
  * is not specified (<code>null</code>) then the human readable name
@@ -709,10 +709,10 @@
  * are met:
  * <p>
  * <ul>
- * <li>The object is not <code>null</code>.</li>
- * <li>The object is an instance of <code>DataFlavor</code>.</li>
+ * <li>The object is not <code>null</code>.
+ * <li>The object is an instance of <code>DataFlavor</code>.
  * <li>The object's MIME type and representation class are equal to
- * this object's.</li>
+ * this object's.
  * </ul>
  *
  * @param obj The <code>Object</code> to test against.
