--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/regex/Pattern.java	2004-11-17 02:24:09.000000000 -0700
+++ java/util/regex/Pattern.java	2004-05-05 02:18:56.000000000 -0600
@@ -50,7 +50,7 @@
  *
  * @since 1.4
  */
-public final class Pattern implements Serializable
+public class Pattern implements Serializable
 {
   private static final long serialVersionUID = 5073258162644648461L;
   
