--- /home/tromey/gnu/classpath/classpath/java/nio/channels/Pipe.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/Pipe.java	Wed Nov 13 06:52:47 2002
@@ -37,7 +37,6 @@
 
 package java.nio.channels;
 
-import java.io.IOException;
 import java.nio.channels.spi.AbstractSelectableChannel;
 import java.nio.channels.spi.SelectorProvider;
 
@@ -105,7 +104,7 @@
    * 
    * @exception IOException If an error occurs
    */
-  public static Pipe open() throws IOException
+  public static Pipe open()
   {
     return SelectorProvider.provider ().openPipe();
   }
