--- /home/tromey/gnu/classpath/classpath/java/security/SecureClassLoader.java	Mon Aug 19 13:47:07 2002
+++ java/security/SecureClassLoader.java	Fri May 31 12:14:23 2002
@@ -71,7 +71,7 @@
      @param b the data representing the classfile, in classfile format.
      @param off the offset into the data where the classfile starts.
      @param len the length of the classfile data in the array.
-     @param cs the CodeSource for the class or null when unknown.
+     @param cs the CodeSource for the class
 
      @return the class that was defined and optional CodeSource.
 
@@ -81,14 +81,16 @@
 				    CodeSource cs)
   {
     // FIXME: Need to cache ProtectionDomains according to 1.3 docs.
-    if (cs != null)
+    ProtectionDomain protectionDomain =
+      new ProtectionDomain(cs, getPermissions(cs));
+    try
       {
-	ProtectionDomain protectionDomain
-		= new ProtectionDomain(cs, getPermissions(cs));
 	return super.defineClass(name, b, off, len, protectionDomain);
-      } 
-    else
-      return super.defineClass(name, b, off, len);
+      }
+    catch (ClassFormatError cfe)
+      {
+	return null;
+      }
   }
 
   /**
