--- /home/tromey/gnu/egcs/classpath/classpath/java/awt/IllegalComponentStateException.java	Sat Mar  2 12:13:08 2002
+++ java/awt/IllegalComponentStateException.java	Tue Jan 22 16:01:31 2002
@@ -1,5 +1,5 @@
-/* IllegalComponentStateException.java -- bad component state
-   Copyright (C) 1999, 2002 Free Software Foundation, Inc.
+/* IllegalComponentStateException.java -- Bad component state
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,33 +39,37 @@
 package java.awt;
 
 /**
- * This exception is thrown when the requested operation failed because
- * a component was not in the proper state.
- *
- * @author Aaron M. Renn <arenn@urbanophile.com>
- * @status updated to 1.4
- */
+  * This exception is thrown when the requested operation failed because
+  * a component was not in the proper state.
+  *
+  * @author Aaron M. Renn (arenn@urbanophile.com)
+  */
 public class IllegalComponentStateException extends IllegalStateException
 {
-  /**
-   * Compatible with JDK 1.0+.
-   */
-  private static final long serialVersionUID = -1889339587208144238L;
-
-  /**
-   * Create a new instance with no detailed error message.
-   */
-  public IllegalComponentStateException()
-  {
-  }
-
-  /**
-   * Create a new instance with the specified detailed error message.
-   *
-   * @param message the detailed error message
-   */
-  public IllegalComponentStateException(String message)
-  {
-    super(message);
-  }
+
+/**
+  * Initializes a new instance of <code>IllegalComponentStateException</code>
+  * with the specified detailed error message.
+  *
+  * @param message The detailed error message.
+  */
+public
+IllegalComponentStateException(String message)
+{
+  super(message);
+}
+
+/*************************************************************************/
+
+/**
+  * Initializes a new instance of <code>IllegalComponentStateException</code>
+  * with no detailed error message.
+  */
+public
+IllegalComponentStateException()
+{
+  super();
+}
+
 } // class IllegalComponentStateException
+
