--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/Inet6Address.java	2004-10-16 02:22:17.000000000 -0600
+++ java/net/Inet6Address.java	2004-04-21 02:18:09.000000000 -0600
@@ -1,5 +1,5 @@
-/* Inet6Address.java --
-   Copyright (C) 2002, 2003, 2004  Free Software Foundation, Inc.
+/* Inet6Address.java
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -35,17 +35,20 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
-
 package java.net;
 
 import java.util.Arrays;
 
+
+/**
+ * @author Michael Koch
+ * @date August 3, 2002.
+ */
+
 /*
  * Written using on-line Java Platform 1.4 API Specification and
  * RFC 1884 (http://www.ietf.org/rfc/rfc1884.txt)
- * 
- * @author Michael Koch
- * @status Believed complete and correct.
+ * Status: Believed complete and correct.
  */
 public final class Inet6Address extends InetAddress
 {
