--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/MappedByteBufferImpl.java	2003-07-14 02:19:03.000000000 -0600
+++ java/nio/MappedByteBufferImpl.java	2003-07-14 02:18:14.000000000 -0600
@@ -40,7 +40,7 @@
 
 import java.io.IOException;
 import java.nio.channels.FileChannelImpl;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 public class MappedByteBufferImpl extends MappedByteBuffer
 {
