--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/MenuSelectionManager.java	2004-06-08 02:25:40.000000000 -0600
+++ javax/swing/MenuSelectionManager.java	2004-05-27 02:19:06.000000000 -0600
@@ -43,7 +43,6 @@
 import java.awt.event.KeyEvent;
 import java.awt.event.MouseEvent;
 
-import java.util.ArrayList;
 import java.util.Vector;
 
 import javax.swing.JMenu;
@@ -275,8 +274,8 @@
    */
   private MenuElement[] getPath (Component c)
   {
-    ArrayList path = new ArrayList();
-    path.add(0, c);
+    Vector path = new Vector();
+    path.add (c);
 
     Component parent = c.getParent ();
 
@@ -285,11 +284,15 @@
            || parent instanceof JMenuItem 
            || parent instanceof JMenuBar)
       {
-        path.add(0, parent);
+        path.add (parent);
         parent = parent.getParent ();
       }
 
-    return (MenuElement[]) path.toArray();
+    MenuElement[] pathArray = new MenuElement[path.size ()];
+
+    for (int i = 0; i < path.size (); i++)
+      pathArray[i] = (MenuElement) path.get (path.size () - i - 1);
+    return pathArray;
   }
   
 } // class MenuSelectionManager
