--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/DatagramSocket.java	2003-12-03 02:20:29.000000000 -0700
+++ java/net/DatagramSocket.java	2003-12-10 02:18:05.000000000 -0700
@@ -216,32 +216,32 @@
    */
   public void close()
   {
-    if (!isClosed())
+    if (isClosed())
+      return;
+    
+    try
+      {
+	getImpl().close();
+      }
+    catch (SocketException e)
+      {
+	// Ignore this case, just close the socket in finally clause.
+      }
+    finally
+      {
+	remoteAddress = null;
+	remotePort = -1;
+	impl = null;
+      }
+
+    try
+      {
+	if (getChannel() != null)
+	  getChannel().close();
+      }
+    catch (IOException e)
       {
-	try
-	  {
-	    getImpl().close();
-	  }
-	catch (SocketException e)
-	  {
-	    // Ignore this case, just close the socket in finally clause.
-	  }
-	finally
-	  {
-	    remoteAddress = null;
-	    remotePort = -1;
-	    impl = null;
-	  }
-
-	try
-	  {
-	    if (getChannel() != null)
-	      getChannel().close();
-	  }
-	catch (IOException e)
-	  {
-	    // Do nothing.
-	  }
+	// Do nothing.
       }
   }
 
