--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/DoubleBuffer.java	2003-03-11 10:27:52.000000000 -0700
+++ java/nio/DoubleBuffer.java	2003-05-10 02:19:17.000000000 -0600
@@ -41,8 +41,8 @@
 
 public abstract class DoubleBuffer extends Buffer implements Comparable
 {
-  protected double [] backing_buffer;
-  protected int array_offset;
+  int array_offset;
+  double[] backing_buffer;
 
   public static DoubleBuffer allocateDirect(int capacity)
   {
@@ -77,7 +77,7 @@
     return wrap(array, 0, array.length);
   }
 
-  protected DoubleBuffer (int capacity, int limit, int position, int mark)
+  DoubleBuffer (int capacity, int limit, int position, int mark)
   {
     super (capacity, limit, position, mark);
   }
@@ -108,7 +108,7 @@
   public DoubleBuffer put (double[] src, int offset, int length)
   {
     for (int i = offset; i < offset + length; i++)
-      put(src[i]);
+      put (src [i]);
 
     return this;
   }
