--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/DateFormat.java	2004-11-17 02:24:08.000000000 -0700
+++ java/text/DateFormat.java	2004-10-28 02:16:12.000000000 -0600
@@ -137,7 +137,7 @@
     public static final DateFormat.Field TIME_ZONE
 	= new Field("timezone", Calendar.ZONE_OFFSET);
  
-    static final DateFormat.Field[] allFields =
+    public static final DateFormat.Field[] allFields =
     {
       ERA, YEAR, MONTH, DAY_OF_MONTH, HOUR_OF_DAY1,
       HOUR_OF_DAY0, MINUTE, SECOND, MILLISECOND,
