--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Button.java	2004-02-11 02:21:43.000000000 -0700
+++ java/awt/Button.java	2004-06-18 02:18:50.000000000 -0600
@@ -81,6 +81,11 @@
 // List of ActionListeners for this class.
 private transient ActionListener action_listeners;
 
+  /*
+   * The number used to generate the name returned by getName.
+   */
+  private static transient long next_button_number = 0;
+
 /*************************************************************************/
 
 /*
@@ -305,9 +310,24 @@
 protected String
 paramString()
 {
-  return ("label=" + getLabel() + ",actionCommand=" + getActionCommand()
-	  + "," + super.paramString());
+  return getName () + "," + getX () + "," + getY () + ","
+    + getWidth () + "x" + getHeight () + ",label=" + getLabel ();
 }
 
+  /**
+   * Generate a unique name for this button.
+   *
+   * @return A unique name for this button.
+   */
+  String generateName ()
+  {
+    return "button" + getUniqueLong ();
+  }
+
+  private static synchronized long getUniqueLong ()
+  {
+    return next_button_number++;
+  }
+
 } // class Button 
 
