--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/CharBufferImpl.java	2004-04-13 02:20:24.000000000 -0600
+++ java/nio/CharBufferImpl.java	2004-02-17 02:18:13.000000000 -0700
@@ -104,7 +104,7 @@
     return false;
   }
 
-  public CharSequence subSequence (int start, int end)
+  final public CharSequence subSequence (int start, int end)
   {
     if (start < 0
         || start > length ()
@@ -118,7 +118,7 @@
   /**
    * Relative get method. Reads the next <code>char</code> from the buffer.
    */
-  public char get ()
+  final public char get ()
   {
     char result = backing_buffer [position ()];
     position (position () + 1);
@@ -131,7 +131,7 @@
    * 
    * @exception ReadOnlyBufferException If this buffer is read-only.
    */
-  public CharBuffer put (char value)
+  final public CharBuffer put (char value)
   {
     if (readOnly)
       throw new ReadOnlyBufferException ();
@@ -148,7 +148,7 @@
    * @exception IndexOutOfBoundsException If index is negative or not smaller
    * than the buffer's limit.
    */
-  public char get (int index)
+  final public char get (int index)
   {
     if (index < 0
         || index >= limit ())
@@ -165,7 +165,7 @@
    * than the buffer's limit.
    * @exception ReadOnlyBufferException If this buffer is read-only.
    */
-  public CharBuffer put (int index, char value)
+  final public CharBuffer put (int index, char value)
   {
     if (index < 0
         || index >= limit ())
@@ -178,7 +178,7 @@
     return this;
   }
   
-  public ByteOrder order ()
+  final public ByteOrder order ()
   {
     return ByteOrder.nativeOrder ();
   }
