--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/CollationKey.java	2003-05-02 02:21:16.000000000 -0600
+++ java/text/CollationKey.java	2003-05-01 02:21:57.000000000 -0600
@@ -1,6 +1,6 @@
 /* CollationKey.java -- Precomputed collation value
-   Copyright (C) 1998 Free Software Foundation, Inc.
 
+/* Copyright (C) 1999, 2000  Free Software Foundation
 This file is part of GNU Classpath.
 
 GNU Classpath is free software; you can redistribute it and/or modify
@@ -61,30 +61,34 @@
  *
  * @author Aaron M. Renn <arenn@urbanophile.com>
  * @author Tom Tromey <tromey@cygnus.com>
- * @date March 25, 1999
  */
 public final class CollationKey implements Comparable
 {
   /**
-   * This is the <code>Collator</code> this object was created from.
-   */
-  private Collator collator;
-
-  /**
    * This is the <code>String</code> this object represents.
    */
-  private String str;
+  private String originalText;
 
   /**
    * This is the bit value for this key.
    */
-  private byte[] key;
+  private int[] key;
 
-  CollationKey (Collator collator, String str, byte[] key)
+  CollationKey (CollationElementIterator iter, String originalText,
+		int strength)
   {
-    this.collator = collator;
-    this.str = str;
-    this.key = key;
+    this.originalText = originalText;
+
+    // Compute size of required array.
+    int size = 0;
+    while (RuleBasedCollator.next(iter, strength)
+	   != CollationElementIterator.NULLORDER)
+      ++size;
+
+    iter.reset();
+    key = new int[size];
+    for (int i = 0; i < size; i++)
+      key[i] = RuleBasedCollator.next(iter, strength);
   }
 
   /**
@@ -98,23 +102,15 @@
    */
   public int compareTo (CollationKey ck)
   {
-    int i;
-    for (i = 0; i < key.length; i++)
-      {
-        if (ck.key.length <= i)
-          return(1);
-
-        if (key[i] < ck.key[i])
-          return(-1);
+    int max = Math.min (key.length, ck.key.length);
 
-        if (key[i] > ck.key[i])
-          return(1);
+    for (int i = 0; i < max; ++i)
+      {
+	if (key[i] != ck.key[i])
+	  return key[i] - ck.key[i];
       }
 
-    if (i == ck.key.length)
-      return(0);
-    else
-      return(-1);
+    return key.length - ck.key.length;
   }
 
   /**
@@ -151,24 +147,19 @@
    */
   public boolean equals (Object obj)
   {
-    if (obj == null)
-      return(false);
-
-    if (!(obj instanceof CollationKey))
-      return(false);
-
-    CollationKey ck = (CollationKey)obj;
+    if (! (obj instanceof CollationKey))
+      return false;
 
-    if (!ck.collator.equals(collator))
-      return(false);
+    CollationKey ck = (CollationKey) obj;
 
-    if (!ck.getSourceString().equals(getSourceString()))
-      return(false);
+    if (key.length != ck.key.length)
+      return false;
 
-    if (!ck.toByteArray().equals(toByteArray()))
-      return(false);
+    for (int i = 0; i < key.length; ++i)
+      if (key[i] != ck.key[i])
+	return false;
 
-    return(true);
+    return true;
   }
 
   /**
@@ -179,7 +170,7 @@
    */
   public String getSourceString ()
   {
-    return str;
+    return originalText;
   }
 
   /**
@@ -191,7 +182,11 @@
    */
   public int hashCode ()
   {
-    return key.hashCode();
+    // We just follow BitSet instead of thinking up something new.
+    long h = originalText.hashCode();
+    for (int i = key.length - 1; i >= 0; --i)
+      h ^= key[i] * (i + 1);
+    return (int) ((h >> 32) ^ h);
   }
   
   /**
@@ -201,6 +196,15 @@
    */
   public byte[] toByteArray ()
   {
-    return key;
+    byte[] r = new byte[4 * key.length];
+    int off = 0;
+    for (int i = 0; i < key.length; ++i)
+      {
+	r[off++] = (byte) ((key[i] >>> 24) & 255);
+	r[off++] = (byte) ((key[i] >>> 16) & 255);
+	r[off++] = (byte) ((key[i] >>>  8) & 255);
+	r[off++] = (byte) ((key[i]       ) & 255);
+      }
+    return r;
   }
 }
