--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/PropertyResourceBundle.java	2004-04-18 02:22:27.000000000 -0600
+++ java/util/PropertyResourceBundle.java	2003-08-27 02:17:52.000000000 -0600
@@ -54,7 +54,7 @@
  *
  * If there is also a class for this resource and the same locale, the
  * class will be chosen. The properties file should have the name of the
- * resource bundle, appended with the locale (e.g. <code>_de</code> and the
+ * resource bundle, appended with the locale (e.g. <code>_de</code) and the
  * extension <code>.properties</code>. The file should have the same format
  * as for <code>Properties.load()</code>
  *
