--- /home/tromey/gnu/egcs/classpath/classpath/java/security/ProviderException.java	Wed Mar 22 12:26:21 2000
+++ java/security/ProviderException.java	Wed Apr 25 09:45:12 2001
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
- 
+
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -24,43 +24,35 @@
 This exception does not however invalidate any other reasons why the
 executable file might be covered by the GNU General Public License. */
 
-
 package java.security;
 
 /**
-  * This exception indicates that a runtime problem was encounterd with
-  * a security provider. 
-  *
-  * @version 0.0
-  *
-  * @author Aaron M. Renn (arenn@urbanophile.com)
-  */
+ * This exception indicates that a runtime problem was encounterd with
+ * a security provider. 
+ *
+ * @version 0.0
+ *
+ * @author Aaron M. Renn (arenn@urbanophile.com)
+ */
 public class ProviderException extends RuntimeException
 {
-
-/**
-  * This method initializes a new instance of <code>ProviderException</code>
-  * with no descriptive error message.
-  */
-public
-ProviderException()
-{
-  super();
-}
-
-/*************************************************************************/
-
-/**
-  * This method initializes a new instance of <code>ProviderException</code>
-  * with a descriptive error message.
-  *
-  * @param msg The descriptive error message.
-  */
-public
-ProviderException(String msg)
-{
-  super(msg);
+  /**
+   * This method initializes a new instance of <code>ProviderException</code>
+   * with no descriptive error message.
+   */
+  public ProviderException()
+  {
+    super();
+  }
+
+  /**
+   * This method initializes a new instance of <code>ProviderException</code>
+   * with a descriptive error message.
+   *
+   * @param msg The descriptive error message.
+   */
+  public ProviderException(String msg)
+  {
+    super(msg);
+  }
 }
-
-} // class ProviderException
-
