--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/RemoteObject.java	2004-10-16 02:22:18.000000000 -0600
+++ java/rmi/server/RemoteObject.java	2002-12-10 19:20:35.000000000 -0700
@@ -1,5 +1,5 @@
-/* RemoteObject.java --
-   Copyright (c) 1996, 1997, 1998, 1999, 2004  Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -37,14 +37,18 @@
 
 package java.rmi.server;
 
-import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
 import java.io.Serializable;
-import java.lang.reflect.Constructor;
-import java.rmi.NoSuchObjectException;
 import java.rmi.Remote;
+import java.rmi.NoSuchObjectException;
 import java.rmi.UnmarshalException;
+import java.rmi.server.RemoteRef;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.IOException;
+import java.lang.ClassNotFoundException;
+import java.lang.InstantiationException;
+import java.lang.IllegalAccessException;
+import java.lang.reflect.Constructor;
 
 public abstract class RemoteObject
 	implements Remote, Serializable {
