--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Container.java	2004-01-13 02:21:01.000000000 -0700
+++ java/awt/Container.java	2004-01-08 02:18:05.000000000 -0700
@@ -1262,9 +1262,9 @@
     if (dispatcher != null 
         && dispatcher.handleEvent (e))
       return;
-      
+
     if ((e.id <= ContainerEvent.CONTAINER_LAST
-         && e.id >= ContainerEvent.CONTAINER_FIRST)
+             && e.id >= ContainerEvent.CONTAINER_FIRST)
         && (containerListener != null
             || (eventMask & AWTEvent.CONTAINER_EVENT_MASK) != 0))
       processEvent(e);
@@ -1332,7 +1332,7 @@
           {
             component[i].addNotify();
             if (component[i].isLightweight ())
-              {
+	      {
 
                 // If we're not lightweight, and we just got a lightweight
                 // child, we need a lightweight dispatcher to feed it events.
@@ -1344,10 +1344,10 @@
                   }	
 	  
 
-                enableEvents(component[i].eventMask);
-                if (peer != null && !isLightweight ())
-                  enableEvents (AWTEvent.PAINT_EVENT_MASK);
-              }
+		enableEvents(component[i].eventMask);
+		if (peer != null && !isLightweight ())
+		  enableEvents (AWTEvent.PAINT_EVENT_MASK);
+	      }
           }
       }
   }
@@ -1533,7 +1533,7 @@
   private Component focus;
   private Cursor nativeCursor;
   private long eventMask;
-
+  
   private transient Component mouseEventTarget;
   
   LightweightDispatcher(Container c)
@@ -1553,7 +1553,7 @@
   void mouseExit (MouseEvent me, int x, int y)
   {
   }
-    
+
   void acquireComponentForMouseEvent (MouseEvent me)
   {
     int x = me.getX ();
@@ -1574,14 +1574,14 @@
     if (candidate == null
         || candidate_is_container_with_children
         || candidate_does_not_contain_point)
-	    {
+  {
         // Try to reacquire.
         candidate = nativeContainer.findComponentAt (x, y);
-	    }
+  }
 
     if (mouseEventTarget != null
         && mouseEventTarget != candidate)
-	    {		
+  {
         int nx = x - mouseEventTarget.getX ();
         int ny = y - mouseEventTarget.getY ();
         MouseEvent exited = new MouseEvent (mouseEventTarget, 
@@ -1594,7 +1594,7 @@
                                             me.getButton ());
         mouseEventTarget.dispatchEvent (exited); 
         mouseEventTarget = null;
-	    }
+  }
 
     if (candidate != null)
 	    {
@@ -1602,7 +1602,7 @@
         if (candidate.isLightweight() 
             && candidate != nativeContainer
             && candidate != mouseEventTarget)
-          {
+  {
 			
             mouseEventTarget = candidate;
 			
@@ -1622,14 +1622,14 @@
           }
 	    }
   }
-    
+
   boolean handleEvent (AWTEvent e)
   {
     if ((eventMask & e.getID ()) == 0)
 	    return false;
-      
+
     if (e instanceof MouseEvent)
-	    {
+  {
         MouseEvent me = (MouseEvent) e;
         acquireComponentForMouseEvent (me);
 
@@ -1642,7 +1642,7 @@
           }
 	    }
     else if (e instanceof KeyEvent && focus != null)
-	    {
+  {
         focus.processKeyEvent ((KeyEvent) e);
 	    }
       
