--- /home/tromey/gnu/classpath/classpath/java/nio/charset/Charset.java	Sun Nov 17 14:22:22 2002
+++ java/nio/charset/Charset.java	Wed Nov 13 09:30:36 2002
@@ -55,9 +55,6 @@
  */
 public abstract class Charset implements Comparable
 {
-  private static CharsetEncoder cachedEncoder;
-  private static CharsetDecoder cachedDecoder;
-  
   private final String canonicalName;
   private final String[] aliases;
   
@@ -198,14 +195,10 @@
   {
     try
       {
-	if (cachedEncoder == null)
-          {
-            cachedEncoder = newEncoder ()
-              .onMalformedInput (CodingErrorAction.REPLACE)
-              .onUnmappableCharacter (CodingErrorAction.REPLACE);
-	  }
-		  
-        return cachedEncoder.encode (cb);
+        // TODO: cache encoders between sucessive invocations
+        return newEncoder ().onMalformedInput (CodingErrorAction.REPLACE)
+                            .onUnmappableCharacter (CodingErrorAction.REPLACE)
+                            .encode (cb);
       }
     catch (CharacterCodingException e)
       {
@@ -222,14 +215,10 @@
   {
     try
      {
-        if (cachedDecoder == null)
-          {
-	    cachedDecoder = newDecoder ()
-              .onMalformedInput (CodingErrorAction.REPLACE)
-              .onUnmappableCharacter (CodingErrorAction.REPLACE);
-	  }
-	
-        return cachedDecoder.decode (bb);
+        // TODO: cache encoders between sucessive invocations
+        return newDecoder ().onMalformedInput (CodingErrorAction.REPLACE)
+                            .onUnmappableCharacter (CodingErrorAction.REPLACE)
+                            .decode (bb);
       }
     catch (CharacterCodingException e)
       {
