--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicViewportUI.java	2004-04-18 02:22:35.000000000 -0600
+++ javax/swing/plaf/basic/BasicViewportUI.java	2004-05-27 02:19:08.000000000 -0600
@@ -47,6 +47,7 @@
 import java.awt.image.ImageObserver;
 import javax.swing.JComponent;
 import javax.swing.JViewport;
+import javax.swing.ViewportLayout;
 import javax.swing.event.ChangeEvent;
 import javax.swing.event.ChangeListener;
 import javax.swing.plaf.ComponentUI;
@@ -70,7 +71,8 @@
   }
 
   void installDefaults(JComponent c)
-  {
+  {    
+    c.setOpaque(true);
   }
 
   void uninstallDefaults(JComponent c)
@@ -100,6 +102,7 @@
   public void installUI(JComponent c) 
   {
     super.installUI(c);
+    c.setLayout(new ViewportLayout());
     installListeners(c);
   }
 
@@ -111,9 +114,8 @@
 
   public Dimension getPreferredSize(JComponent c) 
   {
-    // FIXME: integrate with Scrollable
-    Dimension d = new Dimension(100,100);
-    return d;
+    // let the ViewportLayout decide
+    return null;
   }
 
   public void paint(Graphics g, JComponent c)
@@ -129,6 +131,12 @@
     Rectangle viewBounds = view.getBounds();
     Rectangle portBounds = v.getBounds();
 
+    if (viewBounds.width == 0 
+        || viewBounds.height == 0
+        || portBounds.width == 0
+        || portBounds.height == 0)
+      return;
+
     if (backingStoreImage == null 
         || backingStoreWidth != viewBounds.width
         || backingStoreHeight != viewBounds.height)
