--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/IntBuffer.java	2004-04-30 02:23:42.000000000 -0600
+++ java/nio/IntBuffer.java	2004-02-17 02:18:13.000000000 -0700
@@ -1,5 +1,5 @@
 /* IntBuffer.java -- 
-   Copyright (C) 2002, 2003, 2004  Free Software Foundation, Inc.
+   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -265,27 +265,32 @@
    */
   public int compareTo (Object obj)
   {
-    IntBuffer other = (IntBuffer) obj;
+    IntBuffer a = (IntBuffer) obj;
 
-    int num = Math.min(remaining(), other.remaining());
-    int pos_this = position();
-    int pos_other = other.position();
-    
-    for (int count = 0; count < num; count++)
+    if (a.remaining () != remaining ())
+      return 1;
+
+    if (! hasArray () ||
+        ! a.hasArray ())
+      {
+        return 1;
+      }
+
+    int r = remaining ();
+    int i1 = position ();
+    int i2 = a.position ();
+
+    for (int i = 0; i < r; i++)
       {
-      	 int a = get(pos_this++);
-      	 int b = other.get(pos_other++);
-      	 
-      	 if (a == b)
-      	   continue;
-      	   
-      	 if (a < b)
-      	   return -1;
-      	   
-      	 return 1;
+        int t = (int) (get (i1) - a.get (i2));
+
+        if (t != 0)
+          {
+            return (int) t;
+          }
       }
-      
-     return remaining() - other.remaining();
+
+    return 0;
   }
 
   /**
