--- /home/tromey/gnu/Nightly/classpath/classpath/java/rmi/server/LoaderHandler.java	Sun Mar 23 02:18:41 2003
+++ java/rmi/server/LoaderHandler.java	Mon Mar 31 02:17:49 2003
@@ -47,11 +47,20 @@
 {
   public static final String packagePrefix = "";
 
+  /**
+   * @deprecated
+   */
   public Class loadClass(String name)
     throws MalformedURLException, ClassNotFoundException;
 
+  /**
+   * @deprecated
+   */
   public Class loadClass(URL codebase, String name)
     throws MalformedURLException, ClassNotFoundException;
 
+  /**
+   * @deprecated
+   */
   public Object getSecurityContext(ClassLoader loader);
 }
