--- /home/tromey/gnu/Nightly/classpath/classpath/java/text/StringCharacterIterator.java	2004-04-18 02:22:26.000000000 -0600
+++ java/text/StringCharacterIterator.java	2002-01-22 16:01:38.000000000 -0700
@@ -314,10 +314,10 @@
    * object.  This will be true if and only if the specified object:
    * <p>
    * <ul>
-   * <li>is not <code>null</code>.</li>
-   * <li>is an instance of <code>StringCharacterIterator</code></li>
-   * <li>has the same text as this object</li>
-   * <li>has the same beginning, ending, and current index as this object.</li>
+   * <li>is not <code>null</code>.
+   * <li>is an instance of <code>StringCharacterIterator</code>
+   * <li>has the same text as this object
+   * <li>has the same beginning, ending, and current index as this object.
    * </ul>
    *
    * @param obj The object to test for equality against.
