--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Process.java	2004-10-28 02:27:35.000000000 -0600
+++ java/lang/Process.java	2003-03-23 02:18:02.000000000 -0700
@@ -1,6 +1,5 @@
 /* Process.java - Represent spawned system process
-   Copyright (C) 1998, 1999, 2001, 2002, 2003, 2004
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2001, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -39,8 +38,8 @@
 
 package java.lang;
 
-import java.io.InputStream;
 import java.io.OutputStream;
+import java.io.InputStream;
 
 /**
  * An instance of a subclass of <code>Process</code> is created by the
