--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/Style.java	2004-01-11 02:20:39.000000000 -0700
+++ javax/swing/text/Style.java	2004-06-18 02:18:59.000000000 -0600
@@ -39,9 +39,26 @@
 
 import javax.swing.event.ChangeListener;
 
-public interface Style
+public interface Style extends MutableAttributeSet
 {
-    void addChangeListener(ChangeListener l);
-    String getName();
-    void removeChangeListener(ChangeListener l);
+  /**
+   * Returns the name of the style.
+   *
+   * @return the name
+   */
+  String getName();
+
+  /**
+   * Adds a <code>ChangeListener</code> object to the style.
+   *
+   * @param listener the listener object to add
+   */
+  void addChangeListener(ChangeListener listener);
+
+  /**
+   * Removes a <code>ChangeListener</code> from to the style.
+   *
+   * @param listener the listener object to remove,
+   */
+  void removeChangeListener(ChangeListener listener);
 }
