--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/acl/Acl.java	2003-08-17 02:19:09.000000000 -0600
+++ java/security/acl/Acl.java	2002-01-22 16:01:36.000000000 -0700
@@ -52,7 +52,7 @@
  * <code>Principal</code> belongs have an ACL entry, the permissions for
  * the individual <code>Principal</code> take precedence over the 
  * permissions of the <code>Group</code> if there is a conflict.
- * <p>
+ * <p
  * Additionally, the ACL interface extends the <code>Owner</code> interface
  * and so an ACL has owners.  Actions which modify the ACL are restricted
  * to owners.
