--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/text/JTextComponent.java	2004-07-09 02:22:42.000000000 -0600
+++ javax/swing/text/JTextComponent.java	2004-06-18 02:18:59.000000000 -0600
@@ -371,7 +371,7 @@
   }
 
   /**
-   * Returns a string that specifies the name of the Look and Feel class
+   * Returns a string that specifies the name of the l&amp;f class
    * that renders this component.
    *
    * @return the string "TextComponentUI"
