--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/SortedSet.java	2004-07-30 02:31:35.000000000 -0600
+++ java/util/SortedSet.java	2002-01-22 16:01:42.000000000 -0700
@@ -88,7 +88,6 @@
    * Returns the first (lowest sorted) element in the map.
    *
    * @return the first element
-   * @throws NoSuchElementException if the set is empty.
    */
   Object first();
 
@@ -119,7 +118,6 @@
    * Returns the last (highest sorted) element in the map.
    *
    * @return the last element
-   * @throws NoSuchElementException if the set is empty.
    */
   Object last();
 
