--- /home/tromey/gnu/Nightly/classpath/classpath/java/awt/Window.java	2003-03-23 02:18:38.000000000 -0700
+++ java/awt/Window.java	2003-04-29 02:23:15.000000000 -0600
@@ -213,14 +213,14 @@
       addNotify();
 
     validate();
-    setVisible (true);
+    super.show();
     toFront();
   }
 
   public void hide()
   {
     // FIXME: call hide() on any "owned" children here.
-    setVisible (false);
+    super.hide();
   }
 
   public boolean isDisplayable()
