--- /home/tromey/gnu/Nightly/classpath/classpath/java/net/NetworkInterface.java	2004-10-16 02:22:18.000000000 -0600
+++ java/net/NetworkInterface.java	2004-05-04 02:18:23.000000000 -0600
@@ -1,4 +1,4 @@
-/* NetworkInterface.java --
+/* NetworkInterface.java
    Copyright (C) 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
@@ -49,7 +49,7 @@
  * For example, an ethernet interface may be named "eth0" and have the
  * address 192.168.1.101 assigned to it.
  *
- * @author Michael Koch (konqueror@gmx.de)
+ * @author Michael Koch <konqueror@gmx.de>
  * @since 1.4
  */
 public final class NetworkInterface
@@ -113,7 +113,6 @@
 	  }
 	catch (SecurityException e)
 	  {
-	    // Ignore.
 	  }
       }
 
