--- /home/tromey/gnu/egcs/classpath/classpath/java/rmi/server/ExportException.java	Sat Mar  2 12:13:15 2002
+++ java/rmi/server/ExportException.java	Tue Jan 22 16:01:36 2002
@@ -1,5 +1,5 @@
-/* ExportException.java -- an export attempt failed
-   Copyright (c) 1996, 1997, 1998, 1999, 2002 Free Software Foundation, Inc.
+/*
+  Copyright (c) 1996, 1997, 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -7,7 +7,7 @@
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.
-
+ 
 GNU Classpath is distributed in the hope that it will be useful, but
 WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
@@ -39,40 +39,15 @@
 
 import java.rmi.RemoteException;
 
-/**
- * Thrown if an attempt to export a remote object fails.
- *
- * @author unknown
- * @see UnicastRemoteObject
- * @see Activatable
- * @since 1.1
- * @status updated to 1.4
- */
-public class ExportException extends RemoteException
-{
-  /**
-   * Compatible with JDK 1.1+.
-   */
-  private static final long serialVersionUID = -9155485338494060170L;
-
-  /**
-   * Create an exception with the specified message.
-   *
-   * @param s the message
-   */
-  public ExportException(String s)
-  {
-    super(s);
-  }
-
-  /**
-   * Create an exception with the specified message and cause.
-   *
-   * @param s the message
-   * @param e the cause
-   */
-  public ExportException(String s, Exception e)
-  {
-    super(s, e);
-  }
+public class ExportException
+	extends RemoteException {
+
+public ExportException(String s) {
+	super(s);
+}
+
+public ExportException(String s, Exception e) {
+	super(s, e);
+}
+
 }
