--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipOutputStream.java	2002-10-27 17:21:39.000000000 -0700
+++ java/util/zip/ZipOutputStream.java	2003-06-18 02:22:33.000000000 -0600
@@ -35,12 +35,13 @@
 obligated to do so.  If you do not wish to do so, delete this
 exception statement from your version. */
 
+
 package java.util.zip;
-import java.io.OutputStream;
+
 import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-import java.util.Vector;
+import java.io.OutputStream;
 import java.util.Enumeration;
+import java.util.Vector;
 
 /**
  * This is a FilterOutputStream that writes the files into a zip
