--- /home/tromey/gnu/classpath/classpath/java/nio/channels/SelectableChannel.java	Mon Dec 30 15:39:05 2002
+++ java/nio/channels/SelectableChannel.java	Wed Nov 13 06:52:47 2002
@@ -37,7 +37,6 @@
 
 package java.nio.channels;
 
-import java.io.IOException;
 import java.nio.channels.spi.AbstractInterruptibleChannel;
 import java.nio.channels.spi.SelectorProvider;
 
@@ -68,8 +67,7 @@
    * is registered with one or more selectors.
    * @exception IOException If an error occurs.
    */
-  public abstract SelectableChannel configureBlocking (boolean block)
-    throws IOException;
+  public abstract SelectableChannel configureBlocking (boolean block);
   
   /**
    * Tells whether this channel is blocking or not.
