--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/HashMap.java	2004-04-30 02:23:42.000000000 -0600
+++ java/util/HashMap.java	2004-04-23 02:18:17.000000000 -0600
@@ -743,9 +743,18 @@
           {
             int idx = hash(e.key);
             HashEntry dest = buckets[idx];
+
+            if (dest != null)
+              {
+                while (dest.next != null)
+                  dest = dest.next;
+                dest.next = e;
+              }
+            else
+              buckets[idx] = e;
+
             HashEntry next = e.next;
-            e.next = buckets[idx];
-            buckets[idx] = e;
+            e.next = null;
             e = next;
           }
       }
