--- /home/tromey/gnu/egcs/classpath/classpath/java/io/PrintWriter.java	Fri Feb  9 00:36:24 2001
+++ java/io/PrintWriter.java	Fri Nov  2 21:44:51 2001
@@ -57,7 +57,7 @@
   private boolean autoflush;
 
   /**
-   * This boolean indicates whether or not an error has ever occured
+   * This boolean indicates whether or not an error has ever occurred
    * on this stream.
    */
   private boolean error;
