--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/ByteBuffer.java	2004-09-18 02:39:16.000000000 -0600
+++ java/nio/ByteBuffer.java	2004-08-31 02:19:52.000000000 -0600
@@ -260,27 +260,11 @@
 
   /**
    * Calculates a hash code for this buffer.
-   *
-   * This is done with <code>int</code> arithmetic,
-   * where ** represents exponentiation, by this formula:<br>
-   * <code>s[position()] + 31 + (s[position()+1] + 30)*31**1 + ... +
-   * (s[limit()-1]+30)*31**(limit()-1)</code>.
-   * Where s is the buffer data. Note that the hashcode is dependent
-   * on buffer content, and therefore is not useful if the buffer
-   * content may change.
-   *
-   * @return the hash code
    */
   public int hashCode ()
   {
-    int hashCode = get(position()) + 31;
-    int multiplier = 1;
-    for (int i = position() + 1; i < limit(); ++i)
-      {
-	  multiplier *= 31;
-	  hashCode += (get(i) + 30)*multiplier;
-      }
-    return hashCode;
+    // FIXME: Check what SUN calculates here.
+    return super.hashCode ();
   }
 
   /**
