--- /home/tromey/gnu/Nightly/classpath/classpath/java/lang/Float.java	2004-04-13 02:20:22.000000000 -0600
+++ java/lang/Float.java	2003-11-22 02:18:18.000000000 -0700
@@ -1,6 +1,5 @@
 /* Float.java -- object wrapper for float
-   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003
-   Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -391,9 +390,9 @@
    * <code>Float.NaN</code> as equal, but treats <code>0.0</code> and
    * <code>-0.0</code> as unequal.
    *
-   * <p>Note that <code>f1.equals(f2)</code> is identical to
+   * <p>Note that <code>f1.equals(f2)<code> is identical to
    * <code>floatToIntBits(f1.floatValue()) ==
-   *    floatToIntBits(f2.floatValue())</code>.
+   *    floatToIntBits(f2.floatValue())<code>.
    *
    * @param obj the object to compare
    * @return whether the objects are semantically equal
@@ -426,10 +425,9 @@
    * @return the bits of the <code>float</code>
    * @see #intBitsToFloat(int)
    */
-  public static int floatToIntBits(float value)
-  {
-    return VMFloat.floatToIntBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native int floatToIntBits(float value);
+  // END GCJ LOCAL
 
   /**
    * Convert the float to the IEEE 754 floating-point "single format" bit
@@ -444,10 +442,9 @@
    * @return the bits of the <code>float</code>
    * @see #intBitsToFloat(int)
    */
-  public static int floatToRawIntBits(float value)
-  {
-    return VMFloat.floatToRawIntBits(value);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native int floatToRawIntBits(float value);
+  // END GCJ LOCAL
 
   /**
    * Convert the argument in IEEE 754 floating-point "single format" bit
@@ -462,10 +459,9 @@
    * @see #floatToIntBits(float)
    * @see #floatToRawIntBits(float)
    */
-  public static float intBitsToFloat(int bits)
-  {
-    return VMFloat.intBitsToFloat(bits);
-  }
+  // GCJ LOCAL: We diverge from Classpath for efficiency.
+  public static native float intBitsToFloat(int bits);
+  // END GCJ LOCAL
 
   /**
    * Compare two Floats numerically by comparing their <code>float</code>
