--- /home/tromey/gnu/Nightly/classpath/classpath/javax/swing/plaf/basic/BasicSliderUI.java	2004-06-08 02:25:43.000000000 -0600
+++ javax/swing/plaf/basic/BasicSliderUI.java	2004-06-18 02:18:58.000000000 -0600
@@ -618,8 +618,7 @@
 	insetCache = slider.getInsets();
 	leftToRightCache = ! slider.getInverted();
 
-	scrollTimer = new Timer();
-	scrollTimer.setDelay(200);
+	scrollTimer = new Timer(200, null);
 	scrollTimer.setRepeats(true);
 
 	installDefaults(slider);
