--- /home/tromey/gnu/classpath/classpath/java/net/Socket.java	Mon Dec 30 15:39:05 2002
+++ java/net/Socket.java	Tue Dec 10 19:20:31 2002
@@ -85,8 +85,6 @@
 
   SocketChannel ch; // this field must have been set if created by SocketChannel
 
-  private boolean closed = false;
-
   // Constructors
 
   /**
@@ -310,9 +308,6 @@
    */
   public void bind (SocketAddress bindpoint) throws IOException
   {
-    if (closed)
-      throw new SocketException ("Socket is closed");
-    
     if ( !(bindpoint instanceof InetSocketAddress))
       throw new IllegalArgumentException ();
 
@@ -335,9 +330,6 @@
   public void connect (SocketAddress endpoint)
     throws IOException
   {
-    if (closed)
-      throw new SocketException ("Socket is closed");
-    
     if (! (endpoint instanceof InetSocketAddress))
       throw new IllegalArgumentException ("Address type not supported");
 
@@ -365,9 +357,6 @@
   public void connect (SocketAddress endpoint, int timeout)
     throws IOException
   {
-    if (closed)
-      throw new SocketException ("Socket is closed");
-    
     if (! (endpoint instanceof InetSocketAddress))
       throw new IllegalArgumentException ("Address type not supported");
 
@@ -864,11 +853,6 @@
   {
     if (impl != null)
       impl.close();
-
-    if (ch != null)
-      ch.close();
-    
-    closed = true;
   }
 
   /**
@@ -1051,7 +1035,8 @@
    */
   public boolean isClosed ()
   {
-    return closed;
+    // FIXME: implement this.
+    return false;
   }
 
   /**
