--- /home/tromey/gnu/Nightly/classpath/classpath/java/io/DataInputStream.java	2004-10-02 02:25:49.000000000 -0600
+++ java/io/DataInputStream.java	2004-04-21 02:18:08.000000000 -0600
@@ -279,9 +279,6 @@
    */
   public final void readFully (byte[] buf, int offset, int len) throws IOException
   {
-    if (len < 0)
-      throw new IndexOutOfBoundsException("Negative length: " + len);
-    
     while (len > 0)
       {
 	// in.read will block until some data is available.
