--- /home/tromey/gnu/Nightly/classpath/classpath/javax/print/attribute/standard/JobImpressions.java	2004-04-28 02:20:18.000000000 -0600
+++ javax/print/attribute/standard/JobImpressions.java	2003-12-28 02:18:10.000000000 -0700
@@ -69,7 +69,7 @@
    *
    * @param obj the object to test
    *
-   * @return true if both objects are equal, false otherwise.
+   * @returns true if both objects are equal, false otherwise.
    */
   public boolean equals(Object obj)
   {
