--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/AlgorithmParameters.java	2004-04-18 02:22:21.000000000 -0600
+++ java/security/AlgorithmParameters.java	2003-04-30 02:19:05.000000000 -0600
@@ -71,7 +71,7 @@
  * parameter specification or parameter encoding.</p>
  *
  * <p>A transparent parameter specification is obtained from an
- * <code>AlgorithmParameters</code> object via a call to
+ * <ocde>AlgorithmParameters</code> object via a call to
  * <code>getParameterSpec()</code>, and a byte encoding of the parameters is
  * obtained via a call to <code>getEncoded()</code>.</p>
  *
