--- /home/tromey/gnu/Nightly/classpath/classpath/java/nio/channels/FileChannelImpl.java	2003-10-21 02:19:06.000000000 -0600
+++ java/nio/channels/FileChannelImpl.java	2003-10-22 02:21:48.000000000 -0600
@@ -48,7 +48,7 @@
 import java.nio.MappedByteBuffer;
 import java.nio.MappedByteBufferImpl;
 import gnu.classpath.Configuration;
-import gnu.classpath.RawData;
+import gnu.gcj.RawData;
 
 /**
  * This file is not user visible !
