--- /home/tromey/gnu/Nightly/classpath/classpath/java/util/zip/ZipEntry.java	2004-10-12 02:18:15.000000000 -0600
+++ java/util/zip/ZipEntry.java	2004-06-08 02:19:58.000000000 -0600
@@ -51,10 +51,10 @@
  */
 public class ZipEntry implements ZipConstants, Cloneable
 {
-  private static final int KNOWN_SIZE   = 1;
-  private static final int KNOWN_CSIZE  = 2;
-  private static final int KNOWN_CRC    = 4;
-  private static final int KNOWN_TIME   = 8;
+  private static int KNOWN_SIZE   = 1;
+  private static int KNOWN_CSIZE  = 2;
+  private static int KNOWN_CRC    = 4;
+  private static int KNOWN_TIME   = 8;
 
   private static Calendar cal;
 
