--- /home/tromey/gnu/Nightly/classpath/classpath/java/security/Signature.java	2003-04-25 02:26:30.000000000 -0600
+++ java/security/Signature.java	2003-04-20 02:20:20.000000000 -0600
@@ -41,8 +41,6 @@
 import java.security.cert.X509Certificate;
 import java.security.spec.AlgorithmParameterSpec;
 
-import gnu.java.security.Engine;
-
 /**
  * <p>This <code>Signature</code> class is used to provide applications the
  * functionality of a digital signature algorithm. Digital signatures are used
@@ -239,15 +237,7 @@
       throw new IllegalArgumentException("Illegal provider");
 
     Signature result = null;
-    Object o = null;
-    try
-      {
-        o = Engine.getInstance(SIGNATURE, algorithm, provider);
-      }
-    catch (java.lang.reflect.InvocationTargetException ite)
-      {
-        throw new NoSuchAlgorithmException(algorithm);
-      }
+    Object o = Engine.getInstance(SIGNATURE, algorithm, provider);
 
     if (o instanceof SignatureSpi)
       {
