--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-06-27 02:20:49.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-04-21 02:18:12.000000000 -0600
@@ -198,7 +198,7 @@
       jclass cx = (*gdk_env)->GetObjectClass (gdk_env, peer_obj);
       hideID = (*gdk_env)->GetMethodID (gdk_env, cx, "gtkHideFileDialog", "()V");
       gtkSetFilenameID = (*gdk_env)->GetMethodID (gdk_env, cx,
-                                   "gtkSetFilename", "(Ljava/lang/String;)V");
+                                   "gtkSetFilename", "(Ljava.lang.String;)V");
       isIDSet = 1;
     }
     
@@ -230,7 +230,7 @@
       jclass cx = (*gdk_env)->GetObjectClass (gdk_env, peer_obj);
       hideID = (*gdk_env)->GetMethodID (gdk_env, cx, "gtkHideFileDialog", "()V");
       gtkSetFilenameID = (*gdk_env)->GetMethodID (gdk_env, cx,
-                                   "gtkSetFilename", "(Ljava/lang/String;)V");
+                                   "gtkSetFilename", "(Ljava.lang.String;)V");
       isIDSet = 1;
     }
     
