--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuPeer.c	2004-04-18 02:22:35.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkMenuPeer.c	2004-04-17 02:18:08.000000000 -0600
@@ -1,5 +1,5 @@
 /* gtkmenupeer.c -- Native implementation of GtkMenuPeer
-   Copyright (C) 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -46,12 +46,8 @@
   GtkAccelGroup *accel;
 
   accel = gtk_menu_get_accel_group (GTK_MENU (menu_item->submenu));
-  /* FIXME: _gtk_accel_group_attach is a GTK-private function, so
-     we'll need a different approach here: */
-#if 0
   _gtk_accel_group_attach (accel, 
     G_OBJECT (gtk_widget_get_toplevel (GTK_WIDGET(menu_item))));
-#endif
 }
 
 JNIEXPORT void JNICALL Java_gnu_java_awt_peer_gtk_GtkMenuPeer_setupAccelGroup
@@ -112,7 +108,7 @@
 
   gtk_menu_item_set_submenu (GTK_MENU_ITEM (menu_title), menu);
 
-  /* Allow this menu to grab the pointer. */
+  // Allow this menu to grab the pointer.
   GtkWidget *toplevel = gtk_widget_get_toplevel (menu);
   if (GTK_IS_WINDOW (toplevel))
     {
