--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkPixbufDecoder.c	2004-03-29 02:24:15.000000000 -0700
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkPixbufDecoder.c	2004-06-18 02:19:00.000000000 -0600
@@ -72,7 +72,7 @@
     return;
 
   width = gdk_pixbuf_get_width (pixbuf); 
-  height = gdk_pixbuf_get_height (pixbuf), 
+  height = gdk_pixbuf_get_height (pixbuf);
 
   gdk_threads_leave ();
 
