--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkImagePainter.c	2004-03-29 02:24:15.000000000 -0700
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkImagePainter.c	2004-05-27 02:19:10.000000000 -0600
@@ -58,6 +58,9 @@
 
   g = (struct graphics *) NSA_GET_PTR (env, gc_obj);
 
+  if (!jpixels)
+    return;
+
   elems = (*env)->GetIntArrayElements (env, jpixels, NULL);
   num_pixels = (*env)->GetArrayLength (env, jpixels);
  
