--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkClasspathFontPeer.c	2004-03-29 02:24:15.000000000 -0700
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GdkClasspathFontPeer.c	2004-05-27 02:19:09.000000000 -0600
@@ -112,9 +112,10 @@
   struct peerfont *pfont = NULL;
   PangoFontMap *map = NULL; 
   char const *family_name = NULL;
+  enum java_awt_font_style style;
 
   gdk_threads_enter ();
-  enum java_awt_font_style style = (enum java_awt_font_style) style_int;
+  style = (enum java_awt_font_style) style_int;
 
   g_assert (self != NULL);
   pfont = (struct peerfont *)NSA_GET_FONT_PTR (env, self);
