--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextComponentPeer.c	2004-04-13 02:20:44.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextComponentPeer.c	2003-12-13 02:18:19.000000000 -0700
@@ -1,5 +1,5 @@
 /* gtktextcomponentpeer.c -- Native implementation of GtkTextComponentPeer
-   Copyright (C) 1998, 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -112,8 +112,8 @@
 {
   void *ptr;
   int pos = 0;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
   GtkTextMark *mark;
   GtkTextIter iter;
@@ -157,8 +157,8 @@
   (JNIEnv *env, jobject obj, jint pos)
 {
   void *ptr;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
   GtkTextIter iter;
 
@@ -198,8 +198,8 @@
 {
   void *ptr;
   int pos = 0;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
   GtkTextIter start;
   GtkTextIter end;
@@ -255,8 +255,8 @@
 {
   void *ptr;
   int pos = 0;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
   GtkTextIter start;
   GtkTextIter end;
@@ -311,8 +311,8 @@
   (JNIEnv *env, jobject obj, jint start, jint end)
 {
   void *ptr;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
   GtkTextIter iter;
 
@@ -357,8 +357,8 @@
   (JNIEnv *env, jobject obj, jboolean state)
 {
   void *ptr;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
 
   ptr = NSA_GET_PTR (env, obj);
 
@@ -396,8 +396,8 @@
   void *ptr;
   char *contents = NULL;
   jstring jcontents;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;    // type of GtkEntry    (TextField)
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
   GtkTextIter start, end;
 
@@ -444,7 +444,7 @@
 {
   void *ptr;
   const char *str;
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkWidget *text = NULL;   // type of GtkTextView (TextArea)
   GtkTextBuffer *buf;
 
   ptr = NSA_GET_PTR (env, obj);
