--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-04-13 02:20:44.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-04-17 02:18:08.000000000 -0600
@@ -1,5 +1,5 @@
 /* gtkfiledialogpeer.c -- Native implementation of GtkFileDialogPeer
-   Copyright (C) 1998, 1999, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2002 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -156,7 +156,7 @@
   static jmethodID disposeID;
   void *ptr;
 
-  /* We only need this for the case when the user closed the window. */
+  // We only need this for the case when the user closed the window
   if (responseId != GTK_RESPONSE_DELETE_EVENT)
     return;
 
