--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkCheckboxMenuItemPeer.c	2004-05-01 02:26:14.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkCheckboxMenuItemPeer.c	2003-12-13 02:18:19.000000000 -0700
@@ -37,7 +37,7 @@
 
 
 #include "gtkpeer.h"
-#include "gnu_java_awt_peer_gtk_GtkCheckboxMenuItemPeer.h"
+#include "gnu_java_awt_peer_gtk_GtkMenuItemPeer.h"
 
 JNIEXPORT void JNICALL
 Java_gnu_java_awt_peer_gtk_GtkCheckboxMenuItemPeer_create
