--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkListPeer.c	2004-04-13 02:20:44.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkListPeer.c	2003-12-13 02:18:19.000000000 -0700
@@ -1,5 +1,5 @@
 /* gtklistpeer.c -- Native implementation of GtkListPeer
-   Copyright (C) 1998, 1999, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999 Free Software Foundation, Inc.
 
 This file is part of GNU Classpath.
 
@@ -324,6 +324,7 @@
 	     GdkEventButton *event __attribute__((unused)), 
 	     jobject peer_obj)
 {
+  //g_print ("select_row\n");
   (*gdk_env)->CallVoidMethod (gdk_env, peer_obj,
 			      postListItemEventID,
 			      row,
@@ -337,6 +338,7 @@
 	       GdkEventButton *event __attribute__((unused)),
 	       jobject peer_obj)
 {
+  //g_print ("unselect_row\n");
   (*gdk_env)->CallVoidMethod (gdk_env, peer_obj,
 			      postListItemEventID,
 			      row,
