--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextComponentPeer.c	2004-04-13 02:20:44.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkTextComponentPeer.c	2004-05-27 02:19:10.000000000 -0600
@@ -40,10 +40,6 @@
 #include "gnu_java_awt_peer_gtk_GtkComponentPeer.h"
 #include "gnu_java_awt_peer_gtk_GtkTextComponentPeer.h"
 
-static void textcomponent_commit_cb (GtkIMContext *context,
-                                 const gchar  *str,
-                                 jobject peer);
-
 static void textcomponent_changed_cb (GtkEditable *editable,
                                   jobject peer);
 
@@ -61,9 +57,6 @@
 
   if (GTK_IS_ENTRY(ptr))
     {
-      g_signal_connect (GTK_ENTRY (ptr)->im_context, "commit",
-                        G_CALLBACK (textcomponent_commit_cb), *gref);
-
       g_signal_connect (GTK_EDITABLE (ptr), "changed",
                         G_CALLBACK (textcomponent_changed_cb), *gref);
 
@@ -85,9 +78,6 @@
 
       if (text)
 	{
-          g_signal_connect (text->im_context, "commit",
-                            G_CALLBACK (textcomponent_commit_cb), *gref);
-
           buf = gtk_text_view_get_buffer (text);
           if (buf)
             g_signal_connect (buf, "changed",
@@ -112,8 +102,8 @@
 {
   void *ptr;
   int pos = 0;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextMark *mark;
   GtkTextIter iter;
@@ -157,8 +147,8 @@
   (JNIEnv *env, jobject obj, jint pos)
 {
   void *ptr;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextIter iter;
 
@@ -198,8 +188,8 @@
 {
   void *ptr;
   int pos = 0;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextIter start;
   GtkTextIter end;
@@ -255,8 +245,8 @@
 {
   void *ptr;
   int pos = 0;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextIter start;
   GtkTextIter end;
@@ -311,8 +301,8 @@
   (JNIEnv *env, jobject obj, jint start, jint end)
 {
   void *ptr;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextIter iter;
 
@@ -357,8 +347,8 @@
   (JNIEnv *env, jobject obj, jboolean state)
 {
   void *ptr;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
 
   ptr = NSA_GET_PTR (env, obj);
 
@@ -396,8 +386,8 @@
   void *ptr;
   char *contents = NULL;
   jstring jcontents;
-  GtkEditable *editable;    /* type of GtkEntry    (TextField) */
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkEditable *editable;
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
   GtkTextIter start, end;
 
@@ -444,7 +434,7 @@
 {
   void *ptr;
   const char *str;
-  GtkWidget *text = NULL;   /* type of GtkTextView (TextArea) */
+  GtkWidget *text = NULL;
   GtkTextBuffer *buf;
 
   ptr = NSA_GET_PTR (env, obj);
@@ -480,28 +470,6 @@
 }
 
 static void
-textcomponent_commit_cb (GtkIMContext *context __attribute__((unused)),
-                         const gchar  *str,
-                         jobject peer)
-{
-  /* str is a \0-terminated UTF-8 encoded character. */
-  gunichar2 *jc = g_utf8_to_utf16 (str, -1, NULL, NULL, NULL);
-  GdkEvent *event = gtk_get_current_event ();
-
-  if (jc)
-    (*gdk_env)->CallVoidMethod (gdk_env, peer,
-                                postKeyEventID,
-                                (jint) AWT_KEY_TYPED,
-                                (jlong) event->key.time,
-                                keyevent_state_to_awt_mods (event),
-                                VK_UNDEFINED,
-                                (jchar) jc[0],
-                                AWT_KEY_LOCATION_UNKNOWN);
-  g_free (jc);
-  gdk_event_free (event);
-}
-
-static void
 textcomponent_changed_cb (GtkEditable *editable __attribute__((unused)),
 			  jobject peer)
 {
