--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkLabelPeer.c	2004-03-29 02:24:15.000000000 -0700
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkLabelPeer.c	2004-06-18 02:19:02.000000000 -0600
@@ -54,7 +54,7 @@
   str = (*env)->GetStringUTFChars (env, text, 0);
 
   gdk_threads_enter ();
-  
+
   ebox = gtk_event_box_new ();
   ebox_container = GTK_CONTAINER (ebox);
   label = gtk_label_new (str);
@@ -80,17 +80,17 @@
 
   ptr = NSA_GET_PTR (env, obj);
 
-  label = gtk_bin_get_child (GTK_BIN(ptr));
+  font_name = (*env)->GetStringUTFChars (env, name, NULL);
 
-  if (!label)
-      return;
+  gdk_threads_enter ();
 
-  font_name = (*env)->GetStringUTFChars (env, name, NULL);
+  label = gtk_bin_get_child (GTK_BIN (ptr));
 
-  gdk_threads_enter();
+  if (!label)
+    return;
 
   font_desc = pango_font_description_from_string (font_name);
-  pango_font_description_set_size (font_desc, size * PANGO_SCALE);
+  pango_font_description_set_size (font_desc, size * dpi_conversion_factor);
 
   if (style & AWT_STYLE_BOLD)
     pango_font_description_set_weight (font_desc, PANGO_WEIGHT_BOLD);
@@ -98,11 +98,11 @@
   if (style & AWT_STYLE_ITALIC)
     pango_font_description_set_style (font_desc, PANGO_STYLE_OBLIQUE);
 
-  gtk_widget_modify_font (GTK_WIDGET(label), font_desc);
+  gtk_widget_modify_font (GTK_WIDGET (label), font_desc);
 
   pango_font_description_free (font_desc);
 
-  gdk_threads_leave();
+  gdk_threads_leave ();
 
   (*env)->ReleaseStringUTFChars (env, name, font_name);
 }
