--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkPopupMenuPeer.c	2004-04-18 02:22:35.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkPopupMenuPeer.c	2004-05-27 02:19:10.000000000 -0600
@@ -39,6 +39,8 @@
 #include "gtkpeer.h"
 #include "gnu_java_awt_peer_gtk_GtkPopupMenuPeer.h"
 
+void menu_pos (GtkMenu *, gint *, gint *, gboolean *, gpointer);
+
 struct pos
 {
   gint x;
@@ -91,8 +93,7 @@
   gdk_threads_enter ();
   menu = GTK_MENU (GTK_MENU_ITEM (ptr1)->submenu);
   gtk_menu_set_accel_group (menu, gtk_accel_group_new ());
-  /* FIXME: _gtk_accel_group_attach is a GTK-private function, so
-     we'll need a different approach here: */
+  /* FIXME: update this to use GTK-2.4 GtkActions. */
 #if 0
   _gtk_accel_group_attach (gtk_menu_get_accel_group (menu),
 			   G_OBJECT (gtk_widget_get_toplevel (ptr2)));
