--- /home/tromey/gnu/Nightly/classpath/classpath/native/jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-07-26 02:25:23.000000000 -0600
+++ jni/gtk-peer/gnu_java_awt_peer_gtk_GtkFileDialogPeer.c	2004-07-23 02:18:51.000000000 -0600
@@ -151,7 +151,6 @@
   jclass cx;
   jmethodID id;
   jstring *filename;
-  gboolean accepted;
 
   cx = (*gdk_env)->GetObjectClass (gdk_env, (jobject) obj);
   id = (*gdk_env)->GetMethodID (gdk_env, cx, "filenameFilterCallback",
@@ -160,7 +159,7 @@
   filename = (*gdk_env)->NewStringUTF(gdk_env, filter_info->filename);
   
   gdk_threads_leave();
-  accepted = (*gdk_env)->CallBooleanMethod(gdk_env, obj, id, filename);
+  gboolean accepted = (*gdk_env)->CallBooleanMethod(gdk_env, obj, id, filename);
   gdk_threads_enter();
 
   return accepted;
